<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>17 Interactivity with the R/Spatial ecosystem | Workshop: Spatial multi-omics data analysis with Giotto Suite</title>
  <meta name="description" content="17 Interactivity with the R/Spatial ecosystem | Workshop: Spatial multi-omics data analysis with Giotto Suite" />
  <meta name="generator" content="bookdown 0.39 and GitBook 2.6.7" />

  <meta property="og:title" content="17 Interactivity with the R/Spatial ecosystem | Workshop: Spatial multi-omics data analysis with Giotto Suite" />
  <meta property="og:type" content="book" />
  
  
  <meta name="github-repo" content="drieslab/giotto_workshop_2024" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="17 Interactivity with the R/Spatial ecosystem | Workshop: Spatial multi-omics data analysis with Giotto Suite" />
  
  
  

<meta name="author" content="Ruben Dries, Jiaji George Chen, Joselyn Cristina ChÃ¡vez-Fuentes, Junxiang Xu ,Edward Ruiz, Jeff Sheridan, Iqra Amin, Wen Wang" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="interoperability-with-isolated-tools.html"/>
<link rel="next" href="contributing-to-giotto.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Giotto Suite Workshop 2024</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#instructors"><i class="fa fa-check"></i><b>1.1</b> Instructors</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#topics-and-schedule"><i class="fa fa-check"></i><b>1.2</b> Topics and Schedule:</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#license"><i class="fa fa-check"></i><b>1.3</b> License</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="datasets-packages.html"><a href="datasets-packages.html"><i class="fa fa-check"></i><b>2</b> Datasets &amp; Packages</a>
<ul>
<li class="chapter" data-level="2.1" data-path="datasets-packages.html"><a href="datasets-packages.html#datasets-to-download"><i class="fa fa-check"></i><b>2.1</b> Datasets to download</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="datasets-packages.html"><a href="datasets-packages.html#zenodo-repository"><i class="fa fa-check"></i><b>2.1.1</b> Zenodo repository</a></li>
<li class="chapter" data-level="2.1.2" data-path="datasets-packages.html"><a href="datasets-packages.html#x-genomics-visium-mouse-brain-section-coronal-dataset"><i class="fa fa-check"></i><b>2.1.2</b> 10X Genomics Visium Mouse Brain Section (Coronal) dataset</a></li>
<li class="chapter" data-level="2.1.3" data-path="datasets-packages.html"><a href="datasets-packages.html#x-genomics-visium-hd-ffpe-human-colon-cancer"><i class="fa fa-check"></i><b>2.1.3</b> 10X Genomics Visium HD: FFPE Human Colon Cancer</a></li>
<li class="chapter" data-level="2.1.4" data-path="datasets-packages.html"><a href="datasets-packages.html#x-genomics-multi-modal-dataset"><i class="fa fa-check"></i><b>2.1.4</b> 10X Genomics multi-modal dataset</a></li>
<li class="chapter" data-level="2.1.5" data-path="datasets-packages.html"><a href="datasets-packages.html#x-genomics-multi-omics-visium-cytassist-human-tonsil-dataset"><i class="fa fa-check"></i><b>2.1.5</b> 10X Genomics multi-omics Visium CytAssist Human Tonsil dataset</a></li>
<li class="chapter" data-level="2.1.6" data-path="datasets-packages.html"><a href="datasets-packages.html#x-genomics-human-prostate-cancer-adenocarcinoma-with-invasive-carcinoma-ffpe"><i class="fa fa-check"></i><b>2.1.6</b> 10X Genomics Human Prostate Cancer Adenocarcinoma with Invasive Carcinoma (FFPE)</a></li>
<li class="chapter" data-level="2.1.7" data-path="datasets-packages.html"><a href="datasets-packages.html#x-genomics-normal-human-prostate-ffpe"><i class="fa fa-check"></i><b>2.1.7</b> 10X Genomics Normal Human Prostate (FFPE)</a></li>
<li class="chapter" data-level="2.1.8" data-path="datasets-packages.html"><a href="datasets-packages.html#xenium"><i class="fa fa-check"></i><b>2.1.8</b> Xenium</a></li>
<li class="chapter" data-level="2.1.9" data-path="datasets-packages.html"><a href="datasets-packages.html#merfish-cortex-dataset"><i class="fa fa-check"></i><b>2.1.9</b> MERFISH cortex dataset</a></li>
<li class="chapter" data-level="2.1.10" data-path="datasets-packages.html"><a href="datasets-packages.html#lunaphore-if-dataset"><i class="fa fa-check"></i><b>2.1.10</b> Lunaphore IF dataset</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="datasets-packages.html"><a href="datasets-packages.html#needed-packages"><i class="fa fa-check"></i><b>2.2</b> Needed packages</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="datasets-packages.html"><a href="datasets-packages.html#cran-dependencies"><i class="fa fa-check"></i><b>2.2.1</b> CRAN dependencies:</a></li>
<li class="chapter" data-level="2.2.2" data-path="datasets-packages.html"><a href="datasets-packages.html#terra-installation"><i class="fa fa-check"></i><b>2.2.2</b> <em>terra</em> installation</a></li>
<li class="chapter" data-level="2.2.3" data-path="datasets-packages.html"><a href="datasets-packages.html#matrix-installation"><i class="fa fa-check"></i><b>2.2.3</b> <em>Matrix</em> installation</a></li>
<li class="chapter" data-level="2.2.4" data-path="datasets-packages.html"><a href="datasets-packages.html#rtools-installation"><i class="fa fa-check"></i><b>2.2.4</b> <em>Rtools</em> installation</a></li>
<li class="chapter" data-level="2.2.5" data-path="datasets-packages.html"><a href="datasets-packages.html#giotto-installation"><i class="fa fa-check"></i><b>2.2.5</b> <em>Giotto</em> installation</a></li>
<li class="chapter" data-level="2.2.6" data-path="datasets-packages.html"><a href="datasets-packages.html#irlba-install"><i class="fa fa-check"></i><b>2.2.6</b> <em>irlba</em> install</a></li>
<li class="chapter" data-level="2.2.7" data-path="datasets-packages.html"><a href="datasets-packages.html#arrow-install"><i class="fa fa-check"></i><b>2.2.7</b> <em>arrow</em> install</a></li>
<li class="chapter" data-level="2.2.8" data-path="datasets-packages.html"><a href="datasets-packages.html#bioconductor-dependencies"><i class="fa fa-check"></i><b>2.2.8</b> Bioconductor dependencies:</a></li>
<li class="chapter" data-level="2.2.9" data-path="datasets-packages.html"><a href="datasets-packages.html#cran-packages"><i class="fa fa-check"></i><b>2.2.9</b> CRAN packages:</a></li>
<li class="chapter" data-level="2.2.10" data-path="datasets-packages.html"><a href="datasets-packages.html#packages-from-github"><i class="fa fa-check"></i><b>2.2.10</b> Packages from GitHub</a></li>
<li class="chapter" data-level="2.2.11" data-path="datasets-packages.html"><a href="datasets-packages.html#python-environments"><i class="fa fa-check"></i><b>2.2.11</b> Python environments</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html"><i class="fa fa-check"></i><b>3</b> Spatial omics technologies</a>
<ul>
<li class="chapter" data-level="3.1" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#presentation"><i class="fa fa-check"></i><b>3.1</b> Presentation</a></li>
<li class="chapter" data-level="3.2" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#short-summary"><i class="fa fa-check"></i><b>3.2</b> Short summary</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#why-do-we-need-spatial-omics-technologies"><i class="fa fa-check"></i><b>3.2.1</b> Why do we need spatial omics technologies?</a></li>
<li class="chapter" data-level="3.2.2" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#what-is-spatial-omics"><i class="fa fa-check"></i><b>3.2.2</b> What is spatial omics?</a></li>
<li class="chapter" data-level="3.2.3" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#what-are-the-main-spatial-omics-technologies"><i class="fa fa-check"></i><b>3.2.3</b> What are the main spatial omics technologies?</a></li>
<li class="chapter" data-level="3.2.4" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#other-spatial-omics-atac-seq-cuttag-lipidomics-etc"><i class="fa fa-check"></i><b>3.2.4</b> Other Spatial omics: ATAC-seq, CUT&amp;Tag, lipidomics, etc</a></li>
<li class="chapter" data-level="3.2.5" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#what-are-the-different-types-of-spatial-downstream-analyses"><i class="fa fa-check"></i><b>3.2.5</b> What are the different types of spatial downstream analyses?</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html"><i class="fa fa-check"></i><b>4</b> Introduction to the Giotto package</a>
<ul>
<li class="chapter" data-level="4.1" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html#presentation-1"><i class="fa fa-check"></i><b>4.1</b> Presentation</a></li>
<li class="chapter" data-level="4.2" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html#ecosystem"><i class="fa fa-check"></i><b>4.2</b> Ecosystem</a></li>
<li class="chapter" data-level="4.3" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html#installation-python-environment"><i class="fa fa-check"></i><b>4.3</b> Installation + python environment</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html#giotto-installation-1"><i class="fa fa-check"></i><b>4.3.1</b> Giotto installation</a></li>
<li class="chapter" data-level="4.3.2" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html#python-environment"><i class="fa fa-check"></i><b>4.3.2</b> Python environment</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html#giotto-instructions"><i class="fa fa-check"></i><b>4.4</b> Giotto instructions</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html"><i class="fa fa-check"></i><b>5</b> Data formatting and Pre-processing</a>
<ul>
<li class="chapter" data-level="5.1" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#data-formats"><i class="fa fa-check"></i><b>5.1</b> Data formats</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#general-formats"><i class="fa fa-check"></i><b>5.1.1</b> General formats</a></li>
<li class="chapter" data-level="5.1.2" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#matrix-formats"><i class="fa fa-check"></i><b>5.1.2</b> Matrix formats</a></li>
<li class="chapter" data-level="5.1.3" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#tabular-formats"><i class="fa fa-check"></i><b>5.1.3</b> Tabular formats</a></li>
<li class="chapter" data-level="5.1.4" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#spatial-formats"><i class="fa fa-check"></i><b>5.1.4</b> Spatial formats</a></li>
<li class="chapter" data-level="5.1.5" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#mask-files"><i class="fa fa-check"></i><b>5.1.5</b> Mask files</a></li>
<li class="chapter" data-level="5.1.6" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#images"><i class="fa fa-check"></i><b>5.1.6</b> images</a></li>
<li class="chapter" data-level="5.1.7" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#jsonlike-formats"><i class="fa fa-check"></i><b>5.1.7</b> jsonlike formats</a></li>
<li class="chapter" data-level="5.1.8" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#hierarchical-formats"><i class="fa fa-check"></i><b>5.1.8</b> Hierarchical formats</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#pre-processing"><i class="fa fa-check"></i><b>5.2</b> Pre-processing</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#expression-matrix"><i class="fa fa-check"></i><b>5.2.1</b> Expression matrix</a></li>
<li class="chapter" data-level="5.2.2" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#spatial-locations"><i class="fa fa-check"></i><b>5.2.2</b> Spatial locations</a></li>
<li class="chapter" data-level="5.2.3" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#giottopoints"><i class="fa fa-check"></i><b>5.2.3</b> giottoPoints</a></li>
<li class="chapter" data-level="5.2.4" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#giottopolygon"><i class="fa fa-check"></i><b>5.2.4</b> giottoPolygon</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#subobject-utility-functions"><i class="fa fa-check"></i><b>5.3</b> Subobject utility functions</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#ids"><i class="fa fa-check"></i><b>5.3.1</b> IDs</a></li>
<li class="chapter" data-level="5.3.2" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#bracket-subsetting-and-extraction"><i class="fa fa-check"></i><b>5.3.2</b> Bracket subsetting and extraction</a></li>
<li class="chapter" data-level="5.3.3" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#nesting-metadata-generics"><i class="fa fa-check"></i><b>5.3.3</b> Nesting metadata generics</a></li>
<li class="chapter" data-level="5.3.4" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#appending-to-a-giotto-object"><i class="fa fa-check"></i><b>5.3.4</b> Appending to a giotto object</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html"><i class="fa fa-check"></i><b>6</b> Creating a Giotto object</a>
<ul>
<li class="chapter" data-level="6.1" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#overview"><i class="fa fa-check"></i><b>6.1</b> Overview</a></li>
<li class="chapter" data-level="6.2" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#giottodata-modular-package"><i class="fa fa-check"></i><b>6.2</b> GiottoData modular package</a></li>
<li class="chapter" data-level="6.3" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#from-matrix-locations"><i class="fa fa-check"></i><b>6.3</b> From matrix + locations</a></li>
<li class="chapter" data-level="6.4" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#from-subcellular-raw-data-transcripts-or-images-polygons"><i class="fa fa-check"></i><b>6.4</b> From subcellular raw data (transcripts or images) + polygons</a></li>
<li class="chapter" data-level="6.5" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#from-piece-wise"><i class="fa fa-check"></i><b>6.5</b> From piece-wise</a></li>
<li class="chapter" data-level="6.6" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#using-convenience-functions-for-popular-technologies-vizgen-xenium-cosmx"><i class="fa fa-check"></i><b>6.6</b> Using convenience functions for popular technologies (Vizgen, Xenium, CosMx, â¦)</a></li>
<li class="chapter" data-level="6.7" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#plotting"><i class="fa fa-check"></i><b>6.7</b> Plotting</a>
<ul>
<li class="chapter" data-level="6.7.1" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#subobject-plotting"><i class="fa fa-check"></i><b>6.7.1</b> Subobject plotting</a></li>
<li class="chapter" data-level="6.7.2" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#additive-subobject-plotting"><i class="fa fa-check"></i><b>6.7.2</b> Additive subobject plotting</a></li>
<li class="chapter" data-level="6.7.3" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#giotto-object-plotting"><i class="fa fa-check"></i><b>6.7.3</b> Giotto object plotting</a></li>
</ul></li>
<li class="chapter" data-level="6.8" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#subsetting"><i class="fa fa-check"></i><b>6.8</b> Subsetting</a></li>
<li class="chapter" data-level="6.9" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#giottodata-mini-objects"><i class="fa fa-check"></i><b>6.9</b> <em>GiottoData</em> Mini objects</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="visium-part-i.html"><a href="visium-part-i.html"><i class="fa fa-check"></i><b>7</b> Visium Part I</a>
<ul>
<li class="chapter" data-level="7.1" data-path="visium-part-i.html"><a href="visium-part-i.html#the-visium-technology"><i class="fa fa-check"></i><b>7.1</b> The Visium technology</a></li>
<li class="chapter" data-level="7.2" data-path="visium-part-i.html"><a href="visium-part-i.html#introduction-to-the-spatial-dataset"><i class="fa fa-check"></i><b>7.2</b> Introduction to the spatial dataset</a></li>
<li class="chapter" data-level="7.3" data-path="visium-part-i.html"><a href="visium-part-i.html#download-dataset"><i class="fa fa-check"></i><b>7.3</b> Download dataset</a></li>
<li class="chapter" data-level="7.4" data-path="visium-part-i.html"><a href="visium-part-i.html#create-the-giotto-object"><i class="fa fa-check"></i><b>7.4</b> Create the Giotto object</a></li>
<li class="chapter" data-level="7.5" data-path="visium-part-i.html"><a href="visium-part-i.html#subset-on-spots-that-were-covered-by-tissue"><i class="fa fa-check"></i><b>7.5</b> Subset on spots that were covered by tissue</a></li>
<li class="chapter" data-level="7.6" data-path="visium-part-i.html"><a href="visium-part-i.html#quality-control"><i class="fa fa-check"></i><b>7.6</b> Quality control</a></li>
<li class="chapter" data-level="7.7" data-path="visium-part-i.html"><a href="visium-part-i.html#filtering"><i class="fa fa-check"></i><b>7.7</b> Filtering</a></li>
<li class="chapter" data-level="7.8" data-path="visium-part-i.html"><a href="visium-part-i.html#normalization"><i class="fa fa-check"></i><b>7.8</b> Normalization</a></li>
<li class="chapter" data-level="7.9" data-path="visium-part-i.html"><a href="visium-part-i.html#feature-selection"><i class="fa fa-check"></i><b>7.9</b> Feature selection</a>
<ul>
<li class="chapter" data-level="7.9.1" data-path="visium-part-i.html"><a href="visium-part-i.html#highly-variable-features"><i class="fa fa-check"></i><b>7.9.1</b> Highly Variable Features:</a></li>
</ul></li>
<li class="chapter" data-level="7.10" data-path="visium-part-i.html"><a href="visium-part-i.html#dimension-reduction"><i class="fa fa-check"></i><b>7.10</b> Dimension Reduction</a>
<ul>
<li class="chapter" data-level="7.10.1" data-path="visium-part-i.html"><a href="visium-part-i.html#pca"><i class="fa fa-check"></i><b>7.10.1</b> PCA</a></li>
<li class="chapter" data-level="7.10.2" data-path="visium-part-i.html"><a href="visium-part-i.html#umap"><i class="fa fa-check"></i><b>7.10.2</b> UMAP</a></li>
<li class="chapter" data-level="7.10.3" data-path="visium-part-i.html"><a href="visium-part-i.html#t-sne"><i class="fa fa-check"></i><b>7.10.3</b> t-SNE</a></li>
</ul></li>
<li class="chapter" data-level="7.11" data-path="visium-part-i.html"><a href="visium-part-i.html#clustering"><i class="fa fa-check"></i><b>7.11</b> Clustering</a>
<ul>
<li class="chapter" data-level="7.11.1" data-path="visium-part-i.html"><a href="visium-part-i.html#calculate-leiden-clustering"><i class="fa fa-check"></i><b>7.11.1</b> Calculate Leiden clustering</a></li>
<li class="chapter" data-level="7.11.2" data-path="visium-part-i.html"><a href="visium-part-i.html#calculate-louvain-clustering"><i class="fa fa-check"></i><b>7.11.2</b> Calculate Louvain clustering</a></li>
</ul></li>
<li class="chapter" data-level="7.12" data-path="visium-part-i.html"><a href="visium-part-i.html#save-the-object"><i class="fa fa-check"></i><b>7.12</b> Save the object</a></li>
<li class="chapter" data-level="7.13" data-path="visium-part-i.html"><a href="visium-part-i.html#session-info"><i class="fa fa-check"></i><b>7.13</b> Session info</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="visium-part-ii.html"><a href="visium-part-ii.html"><i class="fa fa-check"></i><b>8</b> Visium Part II</a>
<ul>
<li class="chapter" data-level="8.1" data-path="visium-part-ii.html"><a href="visium-part-ii.html#load-the-object"><i class="fa fa-check"></i><b>8.1</b> Load the object</a></li>
<li class="chapter" data-level="8.2" data-path="visium-part-ii.html"><a href="visium-part-ii.html#differential-expression"><i class="fa fa-check"></i><b>8.2</b> Differential expression</a>
<ul>
<li class="chapter" data-level="8.2.1" data-path="visium-part-ii.html"><a href="visium-part-ii.html#gini-markers"><i class="fa fa-check"></i><b>8.2.1</b> Gini markers</a></li>
<li class="chapter" data-level="8.2.2" data-path="visium-part-ii.html"><a href="visium-part-ii.html#scran-markers"><i class="fa fa-check"></i><b>8.2.2</b> Scran markers</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="visium-part-ii.html"><a href="visium-part-ii.html#enrichment-deconvolution"><i class="fa fa-check"></i><b>8.3</b> Enrichment &amp; Deconvolution</a>
<ul>
<li class="chapter" data-level="8.3.1" data-path="visium-part-ii.html"><a href="visium-part-ii.html#pagerank"><i class="fa fa-check"></i><b>8.3.1</b> PAGE/Rank</a></li>
<li class="chapter" data-level="8.3.2" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatialdwls"><i class="fa fa-check"></i><b>8.3.2</b> SpatialDWLS</a></li>
</ul></li>
<li class="chapter" data-level="8.4" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatial-expression-patterns"><i class="fa fa-check"></i><b>8.4</b> Spatial expression patterns</a>
<ul>
<li class="chapter" data-level="8.4.1" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatial-variable-genes"><i class="fa fa-check"></i><b>8.4.1</b> Spatial variable genes</a></li>
<li class="chapter" data-level="8.4.2" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatial-co-expression-modules"><i class="fa fa-check"></i><b>8.4.2</b> Spatial co-expression modules</a></li>
</ul></li>
<li class="chapter" data-level="8.5" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatially-informed-clusters"><i class="fa fa-check"></i><b>8.5</b> Spatially informed clusters</a></li>
<li class="chapter" data-level="8.6" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatial-domains-hmrf"><i class="fa fa-check"></i><b>8.6</b> Spatial domains HMRF</a></li>
<li class="chapter" data-level="8.7" data-path="visium-part-ii.html"><a href="visium-part-ii.html#interactive-tools"><i class="fa fa-check"></i><b>8.7</b> Interactive tools</a></li>
<li class="chapter" data-level="8.8" data-path="visium-part-ii.html"><a href="visium-part-ii.html#save-the-object-1"><i class="fa fa-check"></i><b>8.8</b> Save the object</a></li>
<li class="chapter" data-level="8.9" data-path="visium-part-ii.html"><a href="visium-part-ii.html#session-info-1"><i class="fa fa-check"></i><b>8.9</b> Session info</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="visium-hd.html"><a href="visium-hd.html"><i class="fa fa-check"></i><b>9</b> Visium HD</a>
<ul>
<li class="chapter" data-level="9.1" data-path="visium-hd.html"><a href="visium-hd.html#objective"><i class="fa fa-check"></i><b>9.1</b> Objective</a></li>
<li class="chapter" data-level="9.2" data-path="visium-hd.html"><a href="visium-hd.html#background"><i class="fa fa-check"></i><b>9.2</b> Background</a>
<ul>
<li class="chapter" data-level="9.2.1" data-path="visium-hd.html"><a href="visium-hd.html#visium-hd-technology"><i class="fa fa-check"></i><b>9.2.1</b> Visium HD Technology</a></li>
<li class="chapter" data-level="9.2.2" data-path="visium-hd.html"><a href="visium-hd.html#colorectal-cancer-sample"><i class="fa fa-check"></i><b>9.2.2</b> Colorectal Cancer Sample</a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="visium-hd.html"><a href="visium-hd.html#data-ingestion"><i class="fa fa-check"></i><b>9.3</b> Data Ingestion</a>
<ul>
<li class="chapter" data-level="9.3.1" data-path="visium-hd.html"><a href="visium-hd.html#visium-hd-output-data-format"><i class="fa fa-check"></i><b>9.3.1</b> Visium HD output data format</a></li>
<li class="chapter" data-level="9.3.2" data-path="visium-hd.html"><a href="visium-hd.html#mini-visium-hd-dataset"><i class="fa fa-check"></i><b>9.3.2</b> Mini Visium HD dataset</a></li>
<li class="chapter" data-level="9.3.3" data-path="visium-hd.html"><a href="visium-hd.html#giotto-visium-hd-convenience-function"><i class="fa fa-check"></i><b>9.3.3</b> Giotto Visium HD convenience function</a></li>
<li class="chapter" data-level="9.3.4" data-path="visium-hd.html"><a href="visium-hd.html#read-in-data-manually"><i class="fa fa-check"></i><b>9.3.4</b> Read in data manually</a></li>
</ul></li>
<li class="chapter" data-level="9.4" data-path="visium-hd.html"><a href="visium-hd.html#giotto-object-at-hexbin-400"><i class="fa fa-check"></i><b>9.4</b> Giotto object at Hexbin 400</a>
<ul>
<li class="chapter" data-level="9.4.1" data-path="visium-hd.html"><a href="visium-hd.html#create-giotto-points"><i class="fa fa-check"></i><b>9.4.1</b> create giotto points</a></li>
<li class="chapter" data-level="9.4.2" data-path="visium-hd.html"><a href="visium-hd.html#create-giotto-polygons"><i class="fa fa-check"></i><b>9.4.2</b> create giotto polygons</a></li>
<li class="chapter" data-level="9.4.3" data-path="visium-hd.html"><a href="visium-hd.html#combine-giotto-points-and-polygons-to-create-giotto-object"><i class="fa fa-check"></i><b>9.4.3</b> combine Giotto points and polygons to create Giotto object</a></li>
<li class="chapter" data-level="9.4.4" data-path="visium-hd.html"><a href="visium-hd.html#process-giotto-object"><i class="fa fa-check"></i><b>9.4.4</b> Process Giotto object</a></li>
<li class="chapter" data-level="9.4.5" data-path="visium-hd.html"><a href="visium-hd.html#dimension-reduction-clustering"><i class="fa fa-check"></i><b>9.4.5</b> Dimension reduction + clustering</a></li>
</ul></li>
<li class="chapter" data-level="9.5" data-path="visium-hd.html"><a href="visium-hd.html#hexbin-100"><i class="fa fa-check"></i><b>9.5</b> Hexbin 100</a>
<ul>
<li class="chapter" data-level="9.5.1" data-path="visium-hd.html"><a href="visium-hd.html#standard-subcellular-pipeline"><i class="fa fa-check"></i><b>9.5.1</b> Standard subcellular pipeline</a></li>
<li class="chapter" data-level="9.5.2" data-path="visium-hd.html"><a href="visium-hd.html#spatial-expression-patterns-1"><i class="fa fa-check"></i><b>9.5.2</b> Spatial expression patterns</a></li>
</ul></li>
<li class="chapter" data-level="9.6" data-path="visium-hd.html"><a href="visium-hd.html#hexbin-25"><i class="fa fa-check"></i><b>9.6</b> Hexbin 25</a>
<ul>
<li class="chapter" data-level="9.6.1" data-path="visium-hd.html"><a href="visium-hd.html#subcellular-workflow-and-projection-functions"><i class="fa fa-check"></i><b>9.6.1</b> Subcellular workflow and projection functions</a></li>
</ul></li>
<li class="chapter" data-level="9.7" data-path="visium-hd.html"><a href="visium-hd.html#database-backend---work-in-progress-but-coming-soon"><i class="fa fa-check"></i><b>9.7</b> Database backend - Work in progress, but coming soon!</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="xenium-1.html"><a href="xenium-1.html"><i class="fa fa-check"></i><b>10</b> Xenium</a>
<ul>
<li class="chapter" data-level="10.1" data-path="xenium-1.html"><a href="xenium-1.html#introduction-to-spatial-dataset"><i class="fa fa-check"></i><b>10.1</b> Introduction to spatial dataset</a>
<ul>
<li class="chapter" data-level="10.1.1" data-path="xenium-1.html"><a href="xenium-1.html#output-directory-structure"><i class="fa fa-check"></i><b>10.1.1</b> Output directory structure</a></li>
<li class="chapter" data-level="10.1.2" data-path="xenium-1.html"><a href="xenium-1.html#mini-xenium-dataset"><i class="fa fa-check"></i><b>10.1.2</b> Mini Xenium Dataset</a></li>
</ul></li>
<li class="chapter" data-level="10.2" data-path="xenium-1.html"><a href="xenium-1.html#data-prep"><i class="fa fa-check"></i><b>10.2</b> Data prep</a>
<ul>
<li class="chapter" data-level="10.2.1" data-path="xenium-1.html"><a href="xenium-1.html#image-conversion-may-change"><i class="fa fa-check"></i><b>10.2.1</b> Image conversion (may change)</a></li>
</ul></li>
<li class="chapter" data-level="10.3" data-path="xenium-1.html"><a href="xenium-1.html#convenience-function"><i class="fa fa-check"></i><b>10.3</b> Convenience function</a>
<ul>
<li class="chapter" data-level="10.3.1" data-path="xenium-1.html"><a href="xenium-1.html#specific-filepaths"><i class="fa fa-check"></i><b>10.3.1</b> Specific filepaths</a></li>
<li class="chapter" data-level="10.3.2" data-path="xenium-1.html"><a href="xenium-1.html#quality-value"><i class="fa fa-check"></i><b>10.3.2</b> Quality value</a></li>
<li class="chapter" data-level="10.3.3" data-path="xenium-1.html"><a href="xenium-1.html#transcript-type-splitting"><i class="fa fa-check"></i><b>10.3.3</b> Transcript type splitting</a></li>
<li class="chapter" data-level="10.3.4" data-path="xenium-1.html"><a href="xenium-1.html#centroids-calculation"><i class="fa fa-check"></i><b>10.3.4</b> Centroids calculation</a></li>
<li class="chapter" data-level="10.3.5" data-path="xenium-1.html"><a href="xenium-1.html#simple-visualization"><i class="fa fa-check"></i><b>10.3.5</b> Simple visualization</a></li>
</ul></li>
<li class="chapter" data-level="10.4" data-path="xenium-1.html"><a href="xenium-1.html#piecewise-loading"><i class="fa fa-check"></i><b>10.4</b> Piecewise loading</a>
<ul>
<li class="chapter" data-level="10.4.1" data-path="xenium-1.html"><a href="xenium-1.html#load-giottopoints-transcripts"><i class="fa fa-check"></i><b>10.4.1</b> load giottoPoints transcripts</a></li>
<li class="chapter" data-level="10.4.2" data-path="xenium-1.html"><a href="xenium-1.html#optional-loading-pre-aggregated-data"><i class="fa fa-check"></i><b>10.4.2</b> (optional) Loading pre-aggregated data</a></li>
</ul></li>
<li class="chapter" data-level="10.5" data-path="xenium-1.html"><a href="xenium-1.html#xenium-images"><i class="fa fa-check"></i><b>10.5</b> Xenium Images</a>
<ul>
<li class="chapter" data-level="10.5.1" data-path="xenium-1.html"><a href="xenium-1.html#image-metadata"><i class="fa fa-check"></i><b>10.5.1</b> Image metadata</a></li>
<li class="chapter" data-level="10.5.2" data-path="xenium-1.html"><a href="xenium-1.html#image-loading"><i class="fa fa-check"></i><b>10.5.2</b> Image loading</a></li>
</ul></li>
<li class="chapter" data-level="10.6" data-path="xenium-1.html"><a href="xenium-1.html#spatial-aggregation"><i class="fa fa-check"></i><b>10.6</b> Spatial aggregation</a></li>
<li class="chapter" data-level="10.7" data-path="xenium-1.html"><a href="xenium-1.html#aggregate-analyses-workflow"><i class="fa fa-check"></i><b>10.7</b> Aggregate analyses workflow</a>
<ul>
<li class="chapter" data-level="10.7.1" data-path="xenium-1.html"><a href="xenium-1.html#filtering-1"><i class="fa fa-check"></i><b>10.7.1</b> Filtering</a></li>
<li class="chapter" data-level="10.7.2" data-path="xenium-1.html"><a href="xenium-1.html#normalization-1"><i class="fa fa-check"></i><b>10.7.2</b> Normalization</a></li>
<li class="chapter" data-level="10.7.3" data-path="xenium-1.html"><a href="xenium-1.html#dimension-reduction-1"><i class="fa fa-check"></i><b>10.7.3</b> Dimension Reduction</a></li>
<li class="chapter" data-level="10.7.4" data-path="xenium-1.html"><a href="xenium-1.html#clustering-1"><i class="fa fa-check"></i><b>10.7.4</b> Clustering</a></li>
</ul></li>
<li class="chapter" data-level="10.8" data-path="xenium-1.html"><a href="xenium-1.html#niche-clustering"><i class="fa fa-check"></i><b>10.8</b> Niche clustering</a>
<ul>
<li class="chapter" data-level="10.8.1" data-path="xenium-1.html"><a href="xenium-1.html#spatial-network"><i class="fa fa-check"></i><b>10.8.1</b> Spatial network</a></li>
<li class="chapter" data-level="10.8.2" data-path="xenium-1.html"><a href="xenium-1.html#niche-calculation"><i class="fa fa-check"></i><b>10.8.2</b> Niche calculation</a></li>
<li class="chapter" data-level="10.8.3" data-path="xenium-1.html"><a href="xenium-1.html#kmeans-clustering-based-on-niche-signature"><i class="fa fa-check"></i><b>10.8.3</b> kmeans clustering based on niche signature</a></li>
</ul></li>
<li class="chapter" data-level="10.9" data-path="xenium-1.html"><a href="xenium-1.html#cell-proximity-enrichment"><i class="fa fa-check"></i><b>10.9</b> Cell proximity enrichment</a></li>
<li class="chapter" data-level="10.10" data-path="xenium-1.html"><a href="xenium-1.html#pseudovisium"><i class="fa fa-check"></i><b>10.10</b> Pseudovisium</a>
<ul>
<li class="chapter" data-level="10.10.1" data-path="xenium-1.html"><a href="xenium-1.html#pseudovisium-aggregation-and-workflow"><i class="fa fa-check"></i><b>10.10.1</b> Pseudovisium aggregation and workflow</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html"><i class="fa fa-check"></i><b>11</b> Spatial proteomics: Multiplexed Immunofluorescence</a>
<ul>
<li class="chapter" data-level="11.1" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#spatial-proteomics-technologies"><i class="fa fa-check"></i><b>11.1</b> Spatial Proteomics Technologies</a></li>
<li class="chapter" data-level="11.2" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#raw-data-type-coming-out-of-different-technologies"><i class="fa fa-check"></i><b>11.2</b> Raw data type coming out of different technologies</a>
<ul>
<li class="chapter" data-level="11.2.1" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#use-ome.tiff-as-an-example-output-data-to-begin-with"><i class="fa fa-check"></i><b>11.2.1</b> Use ome.tiff as an example output data to begin with</a></li>
<li class="chapter" data-level="11.2.2" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#use-single-channel-images-as-an-example-output-data-to-begin-with"><i class="fa fa-check"></i><b>11.2.2</b> Use single channel images as an example output data to begin with</a></li>
</ul></li>
<li class="chapter" data-level="11.3" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#cell-segmentation-file-to-get-single-cell-level-protein-expression"><i class="fa fa-check"></i><b>11.3</b> Cell Segmentation file to get single cell level protein expression</a>
<ul>
<li class="chapter" data-level="11.3.1" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#using-segmentation-output-file-from-deepcellmesmer-as-an-example."><i class="fa fa-check"></i><b>11.3.1</b> Using segmentation output file from DeepCell(mesmer) as an example.</a></li>
<li class="chapter" data-level="11.3.2" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#using-giotto-wrapper-of-cellpose-to-perform-segmentation"><i class="fa fa-check"></i><b>11.3.2</b> Using Giotto wrapper of Cellpose to perform segmentation</a></li>
</ul></li>
<li class="chapter" data-level="11.4" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#create-a-giotto-object-using-list-of-gitto-large-images-and-polygons"><i class="fa fa-check"></i><b>11.4</b> Create a Giotto Object using list of gitto large images and polygons</a>
<ul>
<li class="chapter" data-level="11.4.1" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#overlap-to-matrix"><i class="fa fa-check"></i><b>11.4.1</b> Overlap to matrix</a></li>
<li class="chapter" data-level="11.4.2" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#manipulate-expression-information"><i class="fa fa-check"></i><b>11.4.2</b> Manipulate Expression information</a></li>
<li class="chapter" data-level="11.4.3" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#rescale-polygons"><i class="fa fa-check"></i><b>11.4.3</b> Rescale polygons</a></li>
<li class="chapter" data-level="11.4.4" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#perform-clustering-and-differential-expression"><i class="fa fa-check"></i><b>11.4.4</b> Perform clustering and differential expression</a></li>
<li class="chapter" data-level="11.4.5" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#give-the-cluster-an-annotation-based-on-expression-values"><i class="fa fa-check"></i><b>11.4.5</b> Give the cluster an annotation based on expression values</a></li>
<li class="chapter" data-level="11.4.6" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#spatial-network-1"><i class="fa fa-check"></i><b>11.4.6</b> spatial network</a></li>
<li class="chapter" data-level="11.4.7" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#cell-neighborhood-cell-typecell-type-interactions"><i class="fa fa-check"></i><b>11.4.7</b> Cell Neighborhood: Cell-Type/Cell-Type Interactions</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="12" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html"><i class="fa fa-check"></i><b>12</b> Working with multiple samples</a>
<ul>
<li class="chapter" data-level="12.1" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#objective-1"><i class="fa fa-check"></i><b>12.1</b> Objective</a></li>
<li class="chapter" data-level="12.2" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#background-1"><i class="fa fa-check"></i><b>12.2</b> Background</a>
<ul>
<li class="chapter" data-level="12.2.1" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#dataset"><i class="fa fa-check"></i><b>12.2.1</b> Dataset</a></li>
<li class="chapter" data-level="12.2.2" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#visium-technology"><i class="fa fa-check"></i><b>12.2.2</b> Visium technology</a></li>
</ul></li>
<li class="chapter" data-level="12.3" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#create-individual-giotto-objects"><i class="fa fa-check"></i><b>12.3</b> Create individual giotto objects</a>
<ul>
<li class="chapter" data-level="12.3.1" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#download-the-data"><i class="fa fa-check"></i><b>12.3.1</b> Download the data</a></li>
</ul></li>
<li class="chapter" data-level="12.4" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#extracting-the-downloaded-files"><i class="fa fa-check"></i><b>12.4</b> Extracting the downloaded files</a>
<ul>
<li class="chapter" data-level="12.4.1" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#create-giotto-instructions"><i class="fa fa-check"></i><b>12.4.1</b> Create giotto instructions</a></li>
<li class="chapter" data-level="12.4.2" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#load-visium-data-into-giotto"><i class="fa fa-check"></i><b>12.4.2</b> Load visium data into Giotto</a></li>
<li class="chapter" data-level="12.4.3" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#healthy-prostate-tissue-coverage"><i class="fa fa-check"></i><b>12.4.3</b> Healthy prostate tissue coverage</a></li>
<li class="chapter" data-level="12.4.4" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#adenocarcinoma-prostate-tissue-coverage"><i class="fa fa-check"></i><b>12.4.4</b> Adenocarcinoma prostate tissue coverage</a></li>
<li class="chapter" data-level="12.4.5" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#showing-the-data-strucutre-for-the-inidividual-objects"><i class="fa fa-check"></i><b>12.4.5</b> Showing the data strucutre for the inidividual objects</a></li>
</ul></li>
<li class="chapter" data-level="12.5" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#join-giotto-objects"><i class="fa fa-check"></i><b>12.5</b> Join Giotto Objects</a></li>
<li class="chapter" data-level="12.6" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#visualizing-combined-datasets"><i class="fa fa-check"></i><b>12.6</b> Visualizing combined datasets</a>
<ul>
<li class="chapter" data-level="12.6.1" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#vizualizing-in-the-same-plot"><i class="fa fa-check"></i><b>12.6.1</b> Vizualizing in the same plot</a></li>
<li class="chapter" data-level="12.6.2" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#vizualizing-on-separate-plots"><i class="fa fa-check"></i><b>12.6.2</b> Vizualizing on separate plots</a></li>
</ul></li>
<li class="chapter" data-level="12.7" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#splitting-combined-dataset"><i class="fa fa-check"></i><b>12.7</b> Splitting combined dataset</a></li>
<li class="chapter" data-level="12.8" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#analyzing-joined-objects"><i class="fa fa-check"></i><b>12.8</b> Analyzing joined objects</a>
<ul>
<li class="chapter" data-level="12.8.1" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#normalization-and-adding-statistics"><i class="fa fa-check"></i><b>12.8.1</b> Normalization and adding statistics</a></li>
<li class="chapter" data-level="12.8.2" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#clustering-the-datasets"><i class="fa fa-check"></i><b>12.8.2</b> Clustering the datasets</a></li>
<li class="chapter" data-level="12.8.3" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#vizualizing-spatial-location-of-clusters"><i class="fa fa-check"></i><b>12.8.3</b> Vizualizing spatial location of clusters</a></li>
<li class="chapter" data-level="12.8.4" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#vizualizing-tissue-contribution-to-clusters"><i class="fa fa-check"></i><b>12.8.4</b> Vizualizing tissue contribution to clusters</a></li>
</ul></li>
<li class="chapter" data-level="12.9" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#perform-harmony-and-default-workflows"><i class="fa fa-check"></i><b>12.9</b> Perform Harmony and default workflows</a>
<ul>
<li class="chapter" data-level="12.9.1" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#clustering-harmonized-object"><i class="fa fa-check"></i><b>12.9.1</b> Clustering harmonized object</a></li>
<li class="chapter" data-level="12.9.2" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#vizualizing-the-tissue-contribution"><i class="fa fa-check"></i><b>12.9.2</b> Vizualizing the tissue contribution</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="13" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html"><i class="fa fa-check"></i><b>13</b> Spatial multi-modal analysis</a>
<ul>
<li class="chapter" data-level="13.1" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#overview-1"><i class="fa fa-check"></i><b>13.1</b> Overview</a></li>
<li class="chapter" data-level="13.2" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#spatial-manipulation"><i class="fa fa-check"></i><b>13.2</b> Spatial manipulation</a>
<ul>
<li class="chapter" data-level="13.2.1" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#spatial-transforms"><i class="fa fa-check"></i><b>13.2.1</b> Spatial transforms:</a></li>
<li class="chapter" data-level="13.2.2" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#spatial-utilities"><i class="fa fa-check"></i><b>13.2.2</b> Spatial utilities:</a></li>
<li class="chapter" data-level="13.2.3" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#spatial-classes"><i class="fa fa-check"></i><b>13.2.3</b> Spatial classes:</a></li>
</ul></li>
<li class="chapter" data-level="13.3" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#examples-of-the-simple-transforms-with-a-giottopolygon"><i class="fa fa-check"></i><b>13.3</b> Examples of the simple transforms with a giottoPolygon</a></li>
<li class="chapter" data-level="13.4" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#affine-transforms"><i class="fa fa-check"></i><b>13.4</b> Affine transforms</a></li>
<li class="chapter" data-level="13.5" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#image-transforms"><i class="fa fa-check"></i><b>13.5</b> Image transforms</a></li>
<li class="chapter" data-level="13.6" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#the-practical-usage-of-multi-modality-co-registration"><i class="fa fa-check"></i><b>13.6</b> The practical usage of multi-modality co-registration</a>
<ul>
<li class="chapter" data-level="13.6.1" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#example-dataset-xenium-breast-cancer-pre-release-pack"><i class="fa fa-check"></i><b>13.6.1</b> Example dataset: Xenium Breast Cancer pre-release pack</a></li>
<li class="chapter" data-level="13.6.2" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#target-coordinate-system"><i class="fa fa-check"></i><b>13.6.2</b> Target Coordinate system</a></li>
<li class="chapter" data-level="13.6.3" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#visium-to-register"><i class="fa fa-check"></i><b>13.6.3</b> Visium to register</a></li>
<li class="chapter" data-level="13.6.4" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#register-post-xenium-he-and-if-image"><i class="fa fa-check"></i><b>13.6.4</b> Register post-Xenium H&amp;E and IF image</a></li>
<li class="chapter" data-level="13.6.5" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#automatic-alignment-via-sift-feature-descriptor-matching-and-affine-transformation"><i class="fa fa-check"></i><b>13.6.5</b> Automatic alignment via SIFT feature descriptor matching and affine transformation</a></li>
<li class="chapter" data-level="13.6.6" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#final-notes"><i class="fa fa-check"></i><b>13.6.6</b> Final Notes</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="14" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html"><i class="fa fa-check"></i><b>14</b> Multi-omics integration</a>
<ul>
<li class="chapter" data-level="14.1" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#the-cytassist-technology"><i class="fa fa-check"></i><b>14.1</b> The CytAssist technology</a></li>
<li class="chapter" data-level="14.2" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#introduction-to-the-spatial-dataset-1"><i class="fa fa-check"></i><b>14.2</b> Introduction to the spatial dataset</a></li>
<li class="chapter" data-level="14.3" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#download-dataset-1"><i class="fa fa-check"></i><b>14.3</b> Download dataset</a></li>
<li class="chapter" data-level="14.4" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#create-the-giotto-object-1"><i class="fa fa-check"></i><b>14.4</b> Create the Giotto object</a></li>
<li class="chapter" data-level="14.5" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#subset-on-spots-that-were-covered-by-tissue-1"><i class="fa fa-check"></i><b>14.5</b> Subset on spots that were covered by tissue</a></li>
<li class="chapter" data-level="14.6" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#rna-processing"><i class="fa fa-check"></i><b>14.6</b> RNA processing</a></li>
<li class="chapter" data-level="14.7" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#protein-processing"><i class="fa fa-check"></i><b>14.7</b> Protein processing</a></li>
<li class="chapter" data-level="14.8" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#multi-omics-integration-1"><i class="fa fa-check"></i><b>14.8</b> Multi-omics integration</a></li>
<li class="chapter" data-level="14.9" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#session-info-2"><i class="fa fa-check"></i><b>14.9</b> Session info</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html"><i class="fa fa-check"></i><b>15</b> Interoperability with other frameworks</a>
<ul>
<li class="chapter" data-level="15.1" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#load-giotto-object"><i class="fa fa-check"></i><b>15.1</b> Load Giotto object</a></li>
<li class="chapter" data-level="15.2" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#seurat"><i class="fa fa-check"></i><b>15.2</b> Seurat</a>
<ul>
<li class="chapter" data-level="15.2.1" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#conversion-of-giotto-object-to-seurat-object"><i class="fa fa-check"></i><b>15.2.1</b> Conversion of Giotto Object to Seurat Object</a></li>
<li class="chapter" data-level="15.2.2" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#conversion-of-seurat-object-back-to-giotto-object"><i class="fa fa-check"></i><b>15.2.2</b> Conversion of Seurat object Back to Giotto Object</a></li>
</ul></li>
<li class="chapter" data-level="15.3" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#spatialexperiment"><i class="fa fa-check"></i><b>15.3</b> SpatialExperiment</a>
<ul>
<li class="chapter" data-level="15.3.1" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#convert-giotto-object-to-spatialexperiment-object"><i class="fa fa-check"></i><b>15.3.1</b> Convert Giotto Object to SpatialExperiment Object</a></li>
<li class="chapter" data-level="15.3.2" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#conversion-of-spatialexperiment-object-back-to-giotto"><i class="fa fa-check"></i><b>15.3.2</b> Conversion of SpatialExperiment object back to Giotto</a></li>
</ul></li>
<li class="chapter" data-level="15.4" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#anndata"><i class="fa fa-check"></i><b>15.4</b> AnnData</a>
<ul>
<li class="chapter" data-level="15.4.1" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#load-required-libraries"><i class="fa fa-check"></i><b>15.4.1</b> Load Required Libraries</a></li>
<li class="chapter" data-level="15.4.2" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#specify-path-for-results"><i class="fa fa-check"></i><b>15.4.2</b> Specify Path for Results</a></li>
<li class="chapter" data-level="15.4.3" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#giotto-to-anndata"><i class="fa fa-check"></i><b>15.4.3</b> Giotto To AnnData</a></li>
<li class="chapter" data-level="15.4.4" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#convert-anndata-to-giotto"><i class="fa fa-check"></i><b>15.4.4</b> Convert AnnData to Giotto</a></li>
</ul></li>
<li class="chapter" data-level="15.5" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#create-mini-vizgen-object"><i class="fa fa-check"></i><b>15.5</b> Create mini Vizgen object</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html"><i class="fa fa-check"></i><b>16</b> Interoperability with isolated tools</a>
<ul>
<li class="chapter" data-level="16.1" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#spatial-niche-trajectory-analysis-ontrac"><i class="fa fa-check"></i><b>16.1</b> Spatial niche trajectory analysis (ONTraC)</a>
<ul>
<li class="chapter" data-level="16.1.1" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#introduction-to-ontrac"><i class="fa fa-check"></i><b>16.1.1</b> Introduction to ONTraC</a></li>
<li class="chapter" data-level="16.1.2" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#introduction-to-merfish"><i class="fa fa-check"></i><b>16.1.2</b> Introduction to MERFISH</a></li>
<li class="chapter" data-level="16.1.3" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#settings"><i class="fa fa-check"></i><b>16.1.3</b> Settings</a></li>
<li class="chapter" data-level="16.1.4" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#dataset-1"><i class="fa fa-check"></i><b>16.1.4</b> Dataset</a></li>
<li class="chapter" data-level="16.1.5" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#create-the-giotto-object-2"><i class="fa fa-check"></i><b>16.1.5</b> Create the Giotto object</a></li>
<li class="chapter" data-level="16.1.6" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#perform-spatial-niche-trajectory-analysis-using-ontrac"><i class="fa fa-check"></i><b>16.1.6</b> Perform spatial niche trajectory analysis using ONTraC</a></li>
<li class="chapter" data-level="16.1.7" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#visualization"><i class="fa fa-check"></i><b>16.1.7</b> Visualization</a></li>
</ul></li>
<li class="chapter" data-level="16.2" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#session-info-3"><i class="fa fa-check"></i><b>16.2</b> Session info</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html"><i class="fa fa-check"></i><b>17</b> Interactivity with the R/Spatial ecosystem</a>
<ul>
<li class="chapter" data-level="17.1" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#kriging"><i class="fa fa-check"></i><b>17.1</b> Kriging</a>
<ul>
<li class="chapter" data-level="17.1.1" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#visium-technology-1"><i class="fa fa-check"></i><b>17.1.1</b> Visium technology</a></li>
<li class="chapter" data-level="17.1.2" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#dataset-2"><i class="fa fa-check"></i><b>17.1.2</b> Dataset</a></li>
<li class="chapter" data-level="17.1.3" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#generating-a-geojson-file-of-nuclei-location"><i class="fa fa-check"></i><b>17.1.3</b> Generating a geojson file of nuclei location</a></li>
</ul></li>
<li class="chapter" data-level="17.2" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#downloading-the-dataset"><i class="fa fa-check"></i><b>17.2</b> Downloading the dataset</a></li>
<li class="chapter" data-level="17.3" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#extracting-the-files"><i class="fa fa-check"></i><b>17.3</b> Extracting the files</a></li>
<li class="chapter" data-level="17.4" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#downloading-giotto-object-and-nuclei-segmentation"><i class="fa fa-check"></i><b>17.4</b> Downloading giotto object and nuclei segmentation</a></li>
<li class="chapter" data-level="17.5" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#importing-visium-data"><i class="fa fa-check"></i><b>17.5</b> Importing visium data</a>
<ul>
<li class="chapter" data-level="17.5.1" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#identifying-spatially-organized-features"><i class="fa fa-check"></i><b>17.5.1</b> Identifying spatially organized features</a></li>
</ul></li>
<li class="chapter" data-level="17.6" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#adding-cell-polygons-to-giotto-object"><i class="fa fa-check"></i><b>17.6</b> Adding cell polygons to Giotto object</a>
<ul>
<li class="chapter" data-level="17.6.1" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#read-in-the-poly-information"><i class="fa fa-check"></i><b>17.6.1</b> Read in the poly information</a></li>
<li class="chapter" data-level="17.6.2" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#vizualizing-polygons"><i class="fa fa-check"></i><b>17.6.2</b> Vizualizing polygons</a></li>
<li class="chapter" data-level="17.6.3" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#showing-giotto-object-prior-to-polygon-addition"><i class="fa fa-check"></i><b>17.6.3</b> Showing Giotto object prior to polygon addition</a></li>
<li class="chapter" data-level="17.6.4" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#adding-polygons-to-giotto-object"><i class="fa fa-check"></i><b>17.6.4</b> Adding polygons to giotto object</a></li>
<li class="chapter" data-level="17.6.5" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#check-polygon-information"><i class="fa fa-check"></i><b>17.6.5</b> Check polygon information</a></li>
</ul></li>
<li class="chapter" data-level="17.7" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#performing-kriging"><i class="fa fa-check"></i><b>17.7</b> Performing kriging</a>
<ul>
<li class="chapter" data-level="17.7.1" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#interpolating-features"><i class="fa fa-check"></i><b>17.7.1</b> Interpolating features</a></li>
<li class="chapter" data-level="17.7.2" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#expression-overlap"><i class="fa fa-check"></i><b>17.7.2</b> Expression overlap</a></li>
</ul></li>
<li class="chapter" data-level="17.8" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#reading-in-larger-dataset"><i class="fa fa-check"></i><b>17.8</b> Reading in larger dataset</a></li>
<li class="chapter" data-level="17.9" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#analyzing-interpolated-features"><i class="fa fa-check"></i><b>17.9</b> Analyzing interpolated features</a>
<ul>
<li class="chapter" data-level="17.9.1" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#filter-and-normalization"><i class="fa fa-check"></i><b>17.9.1</b> Filter and normalization</a></li>
<li class="chapter" data-level="17.9.2" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#visualizing-gene-expression-from-interpolated-expression"><i class="fa fa-check"></i><b>17.9.2</b> Visualizing gene expression from interpolated expression</a></li>
<li class="chapter" data-level="17.9.3" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#run-pca"><i class="fa fa-check"></i><b>17.9.3</b> Run PCA</a></li>
<li class="chapter" data-level="17.9.4" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#clustering-2"><i class="fa fa-check"></i><b>17.9.4</b> Clustering</a></li>
<li class="chapter" data-level="17.9.5" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#visualizing-clustering"><i class="fa fa-check"></i><b>17.9.5</b> Visualizing clustering</a></li>
<li class="chapter" data-level="17.9.6" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#cropping-objects"><i class="fa fa-check"></i><b>17.9.6</b> Cropping objects</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="18" data-path="contributing-to-giotto.html"><a href="contributing-to-giotto.html"><i class="fa fa-check"></i><b>18</b> Contributing to Giotto</a>
<ul>
<li class="chapter" data-level="18.1" data-path="contributing-to-giotto.html"><a href="contributing-to-giotto.html#contribution-guideline"><i class="fa fa-check"></i><b>18.1</b> Contribution guideline</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Workshop: Spatial multi-omics data analysis with Giotto Suite</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="interactivity-with-the-rspatial-ecosystem" class="section level1 hasAnchor" number="17">
<h1><span class="header-section-number">17</span> Interactivity with the R/Spatial ecosystem<a href="interactivity-with-the-rspatial-ecosystem.html#interactivity-with-the-rspatial-ecosystem" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Jeff Sheridan</p>
<p>August 7th 2024</p>
<div id="kriging" class="section level2 hasAnchor" number="17.1">
<h2><span class="header-section-number">17.1</span> Kriging<a href="interactivity-with-the-rspatial-ecosystem.html#kriging" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Low resolution spatial data typically covers multiple cells making it difficult to delineate the cell contribution to gene expression. Using a process called kriging we can interpolate gene expression at the single cell levels from low resolution datasets. Kriging is a spatial interpolation technique that estimates unknown values at specific locations by weighing nearby known values based on distance and spatial trends. It uses a model to account for both the distance between points and the overall pattern in the data to make accurate predictions. By taking discrete measurement spots, such as those used for visium, we can interpolate gene expression to a finer scale using kriging.</p>
<div id="visium-technology-1" class="section level3 hasAnchor" number="17.1.1">
<h3><span class="header-section-number">17.1.1</span> Visium technology<a href="interactivity-with-the-rspatial-ecosystem.html#visium-technology-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-597"></span>
<img src="img/03_session1/visium_overview.png" alt="Overview of Visium. Source: 10X Genomics." width="80%" />
<p class="caption">
Figure 17.1: Overview of Visium. Source: 10X Genomics.
</p>
</div>
<p>Visium by 10x Genomics is a spatial gene expression platform that allows for the mapping of gene expression to high-resolution histology through RNA sequencing The process involves placing a tissue section on a specially prepared slide with an array of barcoded spots, which are 55 Âµm in diameter with a spot to spot distance of 100 Âµm. Each spot contains unique barcodes that capture the mRNA from the tissue section, preserving the spatial information. After the tissue is imaged and RNA is captured, the mRNA is sequenced, and the data is mapped back to the tissueâs spatial coordinates. This technology is particularly useful in understanding complex tissue environments, such as tumors, by providing insights into how gene expression varies across different regions.</p>
</div>
<div id="dataset-2" class="section level3 hasAnchor" number="17.1.2">
<h3><span class="header-section-number">17.1.2</span> Dataset<a href="interactivity-with-the-rspatial-ecosystem.html#dataset-2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>For this tutorial weâll be using the mouse brain dataset described in section 6. Visium datasets require a high resolution H&amp;E or IF image to align spots to. Using these images we can identify individual nuclei and cells to be used for kriging. Identifying nuclei is outside the scope of the current tutorial but is required to perform kriging.</p>
</div>
<div id="generating-a-geojson-file-of-nuclei-location" class="section level3 hasAnchor" number="17.1.3">
<h3><span class="header-section-number">17.1.3</span> Generating a geojson file of nuclei location<a href="interactivity-with-the-rspatial-ecosystem.html#generating-a-geojson-file-of-nuclei-location" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>For the following sections we will need to create a geojson that contains polygon information for the nuclei in the sample. We will be providing this in the following <a href="https://zenodo.org/records/13144556">link</a>, however when using for your own datasets this will need to be done outside of Giotto. A tutorial for this using qupath can be found <a href="https://qupath.readthedocs.io/en/stable/docs/tutorials/cell_detection.html">here</a>.</p>
</div>
</div>
<div id="downloading-the-dataset" class="section level2 hasAnchor" number="17.2">
<h2><span class="header-section-number">17.2</span> Downloading the dataset<a href="interactivity-with-the-rspatial-ecosystem.html#downloading-the-dataset" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We first need to import a dataset that we want to perform kriging on.</p>
<div class="sourceCode" id="cb426"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb426-1"><a href="interactivity-with-the-rspatial-ecosystem.html#cb426-1" tabindex="-1"></a>data_directory <span class="ot">&lt;-</span> <span class="st">&quot;data/03_session6&quot;</span></span>
<span id="cb426-2"><a href="interactivity-with-the-rspatial-ecosystem.html#cb426-2" tabindex="-1"></a><span class="fu">dir.create</span>(data_directory)</span>
<span id="cb426-3"><a href="interactivity-with-the-rspatial-ecosystem.html#cb426-3" tabindex="-1"></a></span>
<span id="cb426-4"><a href="interactivity-with-the-rspatial-ecosystem.html#cb426-4" tabindex="-1"></a><span class="fu">download.file</span>(<span class="at">url =</span> <span class="st">&quot;https://cf.10xgenomics.com/samples/spatial-exp/1.1.0/V1_Adult_Mouse_Brain/V1_Adult_Mouse_Brain_raw_feature_bc_matrix.tar.gz&quot;</span>,</span>
<span id="cb426-5"><a href="interactivity-with-the-rspatial-ecosystem.html#cb426-5" tabindex="-1"></a>              <span class="at">destfile =</span> <span class="fu">file.path</span>(data_directory, <span class="st">&quot;V1_Adult_Mouse_Brain_raw_feature_bc_matrix.tar.gz&quot;</span>))</span>
<span id="cb426-6"><a href="interactivity-with-the-rspatial-ecosystem.html#cb426-6" tabindex="-1"></a></span>
<span id="cb426-7"><a href="interactivity-with-the-rspatial-ecosystem.html#cb426-7" tabindex="-1"></a><span class="fu">download.file</span>(<span class="at">url =</span> <span class="st">&quot;https://cf.10xgenomics.com/samples/spatial-exp/1.1.0/V1_Adult_Mouse_Brain/V1_Adult_Mouse_Brain_spatial.tar.gz&quot;</span>,</span>
<span id="cb426-8"><a href="interactivity-with-the-rspatial-ecosystem.html#cb426-8" tabindex="-1"></a>              <span class="at">destfile =</span> <span class="fu">file.path</span>(data_directory, <span class="st">&quot;V1_Adult_Mouse_Brain_spatial.tar.gz&quot;</span>))</span></code></pre></div>
</div>
<div id="extracting-the-files" class="section level2 hasAnchor" number="17.3">
<h2><span class="header-section-number">17.3</span> Extracting the files<a href="interactivity-with-the-rspatial-ecosystem.html#extracting-the-files" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb427"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb427-1"><a href="interactivity-with-the-rspatial-ecosystem.html#cb427-1" tabindex="-1"></a><span class="fu">untar</span>(<span class="at">tarfile =</span> <span class="fu">file.path</span>(data_directory, <span class="st">&quot;V1_Adult_Mouse_Brain_raw_feature_bc_matrix.tar.gz&quot;</span>), </span>
<span id="cb427-2"><a href="interactivity-with-the-rspatial-ecosystem.html#cb427-2" tabindex="-1"></a>      <span class="at">exdir =</span> data_directory)</span>
<span id="cb427-3"><a href="interactivity-with-the-rspatial-ecosystem.html#cb427-3" tabindex="-1"></a></span>
<span id="cb427-4"><a href="interactivity-with-the-rspatial-ecosystem.html#cb427-4" tabindex="-1"></a><span class="fu">untar</span>(<span class="at">tarfile =</span> <span class="fu">file.path</span>(data_directory, <span class="st">&quot;V1_Adult_Mouse_Brain_spatial.tar.gz&quot;</span>), </span>
<span id="cb427-5"><a href="interactivity-with-the-rspatial-ecosystem.html#cb427-5" tabindex="-1"></a>      <span class="at">exdir =</span> data_directory)</span></code></pre></div>
</div>
<div id="downloading-giotto-object-and-nuclei-segmentation" class="section level2 hasAnchor" number="17.4">
<h2><span class="header-section-number">17.4</span> Downloading giotto object and nuclei segmentation<a href="interactivity-with-the-rspatial-ecosystem.html#downloading-giotto-object-and-nuclei-segmentation" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We will need nuclei/cell segmentations to perform the kriging. Later in the tutorial weâll also be using a pre-made giotto object. Download them using the following:</p>
<div class="sourceCode" id="cb428"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb428-1"><a href="interactivity-with-the-rspatial-ecosystem.html#cb428-1" tabindex="-1"></a>destfile <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_directory, <span class="st">&quot;subcellular_gobject.zip&quot;</span>)</span>
<span id="cb428-2"><a href="interactivity-with-the-rspatial-ecosystem.html#cb428-2" tabindex="-1"></a></span>
<span id="cb428-3"><a href="interactivity-with-the-rspatial-ecosystem.html#cb428-3" tabindex="-1"></a><span class="fu">options</span>(<span class="at">timeout =</span> <span class="cn">Inf</span>) <span class="co"># Needed to download large files</span></span>
<span id="cb428-4"><a href="interactivity-with-the-rspatial-ecosystem.html#cb428-4" tabindex="-1"></a></span>
<span id="cb428-5"><a href="interactivity-with-the-rspatial-ecosystem.html#cb428-5" tabindex="-1"></a><span class="fu">download.file</span>(<span class="st">&quot;https://zenodo.org/records/13144556/files/Day3_Session6.zip?download=1&quot;</span>, <span class="at">destfile =</span> destfile)</span>
<span id="cb428-6"><a href="interactivity-with-the-rspatial-ecosystem.html#cb428-6" tabindex="-1"></a></span>
<span id="cb428-7"><a href="interactivity-with-the-rspatial-ecosystem.html#cb428-7" tabindex="-1"></a><span class="fu">unzip</span>(<span class="fu">file.path</span>(data_directory, <span class="st">&quot;subcellular_gobject.zip&quot;</span>), <span class="at">exdir =</span> data_directory)</span></code></pre></div>
</div>
<div id="importing-visium-data" class="section level2 hasAnchor" number="17.5">
<h2><span class="header-section-number">17.5</span> Importing visium data<a href="interactivity-with-the-rspatial-ecosystem.html#importing-visium-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Weâre going to begin by creating a Giotto object for the visium mouse brain dataset. This tutorial wonât go into detail about each of these steps as these have been covered for this dataset in section 6. To get the best results when performing gene expression interpolation we need to identify spatially distinct genes. Therefore, we need to perform nearest neighbor to create a spatial network.</p>
<p>If you have a Giotto object from day 1 session 5, feel free to load that in and skip this first step.</p>
<div class="sourceCode" id="cb429"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb429-1"><a href="interactivity-with-the-rspatial-ecosystem.html#cb429-1" tabindex="-1"></a><span class="fu">library</span>(Giotto)</span>
<span id="cb429-2"><a href="interactivity-with-the-rspatial-ecosystem.html#cb429-2" tabindex="-1"></a></span>
<span id="cb429-3"><a href="interactivity-with-the-rspatial-ecosystem.html#cb429-3" tabindex="-1"></a>save_directory <span class="ot">&lt;-</span> <span class="st">&quot;results/03_session6&quot;</span></span>
<span id="cb429-4"><a href="interactivity-with-the-rspatial-ecosystem.html#cb429-4" tabindex="-1"></a>visium_save_directory <span class="ot">&lt;-</span> <span class="fu">file.path</span>(save_directory, <span class="st">&quot;visium_mouse_brain&quot;</span>)</span>
<span id="cb429-5"><a href="interactivity-with-the-rspatial-ecosystem.html#cb429-5" tabindex="-1"></a>subcell_save_directory <span class="ot">&lt;-</span> <span class="fu">file.path</span>(save_directory, <span class="st">&quot;pseudo_subcellular/&quot;</span>)</span>
<span id="cb429-6"><a href="interactivity-with-the-rspatial-ecosystem.html#cb429-6" tabindex="-1"></a></span>
<span id="cb429-7"><a href="interactivity-with-the-rspatial-ecosystem.html#cb429-7" tabindex="-1"></a>instrs <span class="ot">&lt;-</span> <span class="fu">createGiottoInstructions</span>(<span class="at">show_plot =</span> <span class="cn">TRUE</span>,</span>
<span id="cb429-8"><a href="interactivity-with-the-rspatial-ecosystem.html#cb429-8" tabindex="-1"></a>                                   <span class="at">save_plot =</span> <span class="cn">TRUE</span>,</span>
<span id="cb429-9"><a href="interactivity-with-the-rspatial-ecosystem.html#cb429-9" tabindex="-1"></a>                                   <span class="at">save_dir =</span> visium_save_directory)</span>
<span id="cb429-10"><a href="interactivity-with-the-rspatial-ecosystem.html#cb429-10" tabindex="-1"></a></span>
<span id="cb429-11"><a href="interactivity-with-the-rspatial-ecosystem.html#cb429-11" tabindex="-1"></a>v_brain <span class="ot">&lt;-</span> <span class="fu">createGiottoVisiumObject</span>(data_directory, </span>
<span id="cb429-12"><a href="interactivity-with-the-rspatial-ecosystem.html#cb429-12" tabindex="-1"></a>                                    <span class="at">gene_column_index =</span> <span class="dv">2</span>, </span>
<span id="cb429-13"><a href="interactivity-with-the-rspatial-ecosystem.html#cb429-13" tabindex="-1"></a>                                    <span class="at">instructions =</span> instrs)</span>
<span id="cb429-14"><a href="interactivity-with-the-rspatial-ecosystem.html#cb429-14" tabindex="-1"></a></span>
<span id="cb429-15"><a href="interactivity-with-the-rspatial-ecosystem.html#cb429-15" tabindex="-1"></a><span class="co"># Subset to in tissue only</span></span>
<span id="cb429-16"><a href="interactivity-with-the-rspatial-ecosystem.html#cb429-16" tabindex="-1"></a>cm <span class="ot">&lt;-</span> <span class="fu">pDataDT</span>(v_brain)</span>
<span id="cb429-17"><a href="interactivity-with-the-rspatial-ecosystem.html#cb429-17" tabindex="-1"></a>in_tissue_barcodes <span class="ot">&lt;-</span> cm[in_tissue <span class="sc">==</span> <span class="dv">1</span>]<span class="sc">$</span>cell_ID</span>
<span id="cb429-18"><a href="interactivity-with-the-rspatial-ecosystem.html#cb429-18" tabindex="-1"></a>v_brain <span class="ot">&lt;-</span> <span class="fu">subsetGiotto</span>(v_brain, </span>
<span id="cb429-19"><a href="interactivity-with-the-rspatial-ecosystem.html#cb429-19" tabindex="-1"></a>                        <span class="at">cell_ids =</span> in_tissue_barcodes)</span>
<span id="cb429-20"><a href="interactivity-with-the-rspatial-ecosystem.html#cb429-20" tabindex="-1"></a></span>
<span id="cb429-21"><a href="interactivity-with-the-rspatial-ecosystem.html#cb429-21" tabindex="-1"></a><span class="co"># Filter</span></span>
<span id="cb429-22"><a href="interactivity-with-the-rspatial-ecosystem.html#cb429-22" tabindex="-1"></a>v_brain <span class="ot">&lt;-</span> <span class="fu">filterGiotto</span>(<span class="at">gobject =</span> v_brain,</span>
<span id="cb429-23"><a href="interactivity-with-the-rspatial-ecosystem.html#cb429-23" tabindex="-1"></a>                        <span class="at">expression_threshold =</span> <span class="dv">1</span>,</span>
<span id="cb429-24"><a href="interactivity-with-the-rspatial-ecosystem.html#cb429-24" tabindex="-1"></a>                        <span class="at">feat_det_in_min_cells =</span> <span class="dv">50</span>,</span>
<span id="cb429-25"><a href="interactivity-with-the-rspatial-ecosystem.html#cb429-25" tabindex="-1"></a>                        <span class="at">min_det_feats_per_cell =</span> <span class="dv">1000</span>,</span>
<span id="cb429-26"><a href="interactivity-with-the-rspatial-ecosystem.html#cb429-26" tabindex="-1"></a>                        <span class="at">expression_values =</span> <span class="st">&quot;raw&quot;</span>)</span>
<span id="cb429-27"><a href="interactivity-with-the-rspatial-ecosystem.html#cb429-27" tabindex="-1"></a></span>
<span id="cb429-28"><a href="interactivity-with-the-rspatial-ecosystem.html#cb429-28" tabindex="-1"></a><span class="co"># Normalize</span></span>
<span id="cb429-29"><a href="interactivity-with-the-rspatial-ecosystem.html#cb429-29" tabindex="-1"></a>v_brain <span class="ot">&lt;-</span> <span class="fu">normalizeGiotto</span>(<span class="at">gobject =</span> v_brain,</span>
<span id="cb429-30"><a href="interactivity-with-the-rspatial-ecosystem.html#cb429-30" tabindex="-1"></a>                           <span class="at">scalefactor =</span> <span class="dv">6000</span>,</span>
<span id="cb429-31"><a href="interactivity-with-the-rspatial-ecosystem.html#cb429-31" tabindex="-1"></a>                           <span class="at">verbose =</span> <span class="cn">TRUE</span>)</span>
<span id="cb429-32"><a href="interactivity-with-the-rspatial-ecosystem.html#cb429-32" tabindex="-1"></a></span>
<span id="cb429-33"><a href="interactivity-with-the-rspatial-ecosystem.html#cb429-33" tabindex="-1"></a><span class="co"># Add stats</span></span>
<span id="cb429-34"><a href="interactivity-with-the-rspatial-ecosystem.html#cb429-34" tabindex="-1"></a>v_brain <span class="ot">&lt;-</span> <span class="fu">addStatistics</span>(<span class="at">gobject =</span> v_brain)</span>
<span id="cb429-35"><a href="interactivity-with-the-rspatial-ecosystem.html#cb429-35" tabindex="-1"></a></span>
<span id="cb429-36"><a href="interactivity-with-the-rspatial-ecosystem.html#cb429-36" tabindex="-1"></a><span class="co"># ID HVF</span></span>
<span id="cb429-37"><a href="interactivity-with-the-rspatial-ecosystem.html#cb429-37" tabindex="-1"></a>v_brain <span class="ot">&lt;-</span> <span class="fu">calculateHVF</span>(<span class="at">gobject =</span> v_brain, </span>
<span id="cb429-38"><a href="interactivity-with-the-rspatial-ecosystem.html#cb429-38" tabindex="-1"></a>                        <span class="at">method =</span> <span class="st">&quot;cov_loess&quot;</span>)</span>
<span id="cb429-39"><a href="interactivity-with-the-rspatial-ecosystem.html#cb429-39" tabindex="-1"></a></span>
<span id="cb429-40"><a href="interactivity-with-the-rspatial-ecosystem.html#cb429-40" tabindex="-1"></a>fm <span class="ot">&lt;-</span> <span class="fu">fDataDT</span>(v_brain)</span>
<span id="cb429-41"><a href="interactivity-with-the-rspatial-ecosystem.html#cb429-41" tabindex="-1"></a>hv_feats <span class="ot">&lt;-</span> fm[hvf <span class="sc">==</span> <span class="st">&quot;yes&quot;</span> <span class="sc">&amp;</span> perc_cells <span class="sc">&gt;</span> <span class="dv">3</span> <span class="sc">&amp;</span> mean_expr_det <span class="sc">&gt;</span> <span class="fl">0.4</span>]<span class="sc">$</span>feat_ID</span>
<span id="cb429-42"><a href="interactivity-with-the-rspatial-ecosystem.html#cb429-42" tabindex="-1"></a><span class="fu">length</span>(hv_feats)</span>
<span id="cb429-43"><a href="interactivity-with-the-rspatial-ecosystem.html#cb429-43" tabindex="-1"></a></span>
<span id="cb429-44"><a href="interactivity-with-the-rspatial-ecosystem.html#cb429-44" tabindex="-1"></a><span class="co"># Dimension Reductions</span></span>
<span id="cb429-45"><a href="interactivity-with-the-rspatial-ecosystem.html#cb429-45" tabindex="-1"></a>v_brain <span class="ot">&lt;-</span> <span class="fu">runPCA</span>(<span class="at">gobject =</span> v_brain,</span>
<span id="cb429-46"><a href="interactivity-with-the-rspatial-ecosystem.html#cb429-46" tabindex="-1"></a>                  <span class="at">feats_to_use =</span> hv_feats)</span>
<span id="cb429-47"><a href="interactivity-with-the-rspatial-ecosystem.html#cb429-47" tabindex="-1"></a></span>
<span id="cb429-48"><a href="interactivity-with-the-rspatial-ecosystem.html#cb429-48" tabindex="-1"></a>v_brain <span class="ot">&lt;-</span> <span class="fu">runUMAP</span>(v_brain,</span>
<span id="cb429-49"><a href="interactivity-with-the-rspatial-ecosystem.html#cb429-49" tabindex="-1"></a>                   <span class="at">dimensions_to_use =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>,</span>
<span id="cb429-50"><a href="interactivity-with-the-rspatial-ecosystem.html#cb429-50" tabindex="-1"></a>                   <span class="at">n_neighbors =</span> <span class="dv">15</span>,</span>
<span id="cb429-51"><a href="interactivity-with-the-rspatial-ecosystem.html#cb429-51" tabindex="-1"></a>                   <span class="at">set_seed =</span> <span class="cn">TRUE</span>)</span>
<span id="cb429-52"><a href="interactivity-with-the-rspatial-ecosystem.html#cb429-52" tabindex="-1"></a></span>
<span id="cb429-53"><a href="interactivity-with-the-rspatial-ecosystem.html#cb429-53" tabindex="-1"></a><span class="co"># NN Network</span></span>
<span id="cb429-54"><a href="interactivity-with-the-rspatial-ecosystem.html#cb429-54" tabindex="-1"></a>v_brain <span class="ot">&lt;-</span> <span class="fu">createNearestNetwork</span>(<span class="at">gobject =</span> v_brain,</span>
<span id="cb429-55"><a href="interactivity-with-the-rspatial-ecosystem.html#cb429-55" tabindex="-1"></a>                                <span class="at">dimensions_to_use =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>,</span>
<span id="cb429-56"><a href="interactivity-with-the-rspatial-ecosystem.html#cb429-56" tabindex="-1"></a>                                <span class="at">k =</span> <span class="dv">15</span>)</span>
<span id="cb429-57"><a href="interactivity-with-the-rspatial-ecosystem.html#cb429-57" tabindex="-1"></a><span class="co"># Leiden Cluster</span></span>
<span id="cb429-58"><a href="interactivity-with-the-rspatial-ecosystem.html#cb429-58" tabindex="-1"></a>v_brain <span class="ot">&lt;-</span> <span class="fu">doLeidenCluster</span>(<span class="at">gobject =</span> v_brain,</span>
<span id="cb429-59"><a href="interactivity-with-the-rspatial-ecosystem.html#cb429-59" tabindex="-1"></a>                           <span class="at">resolution =</span> <span class="fl">0.4</span>,</span>
<span id="cb429-60"><a href="interactivity-with-the-rspatial-ecosystem.html#cb429-60" tabindex="-1"></a>                           <span class="at">n_iterations =</span> <span class="dv">1000</span>,</span>
<span id="cb429-61"><a href="interactivity-with-the-rspatial-ecosystem.html#cb429-61" tabindex="-1"></a>                           <span class="at">set_seed =</span> <span class="cn">TRUE</span>)</span>
<span id="cb429-62"><a href="interactivity-with-the-rspatial-ecosystem.html#cb429-62" tabindex="-1"></a></span>
<span id="cb429-63"><a href="interactivity-with-the-rspatial-ecosystem.html#cb429-63" tabindex="-1"></a><span class="co"># Spatial Network (kNN)</span></span>
<span id="cb429-64"><a href="interactivity-with-the-rspatial-ecosystem.html#cb429-64" tabindex="-1"></a>v_brain <span class="ot">&lt;-</span> <span class="fu">createSpatialNetwork</span>(<span class="at">gobject =</span> v_brain,</span>
<span id="cb429-65"><a href="interactivity-with-the-rspatial-ecosystem.html#cb429-65" tabindex="-1"></a>                                <span class="at">method =</span> <span class="st">&quot;kNN&quot;</span>,</span>
<span id="cb429-66"><a href="interactivity-with-the-rspatial-ecosystem.html#cb429-66" tabindex="-1"></a>                                <span class="at">k =</span> <span class="dv">5</span>,</span>
<span id="cb429-67"><a href="interactivity-with-the-rspatial-ecosystem.html#cb429-67" tabindex="-1"></a>                                <span class="at">maximum_distance_knn =</span> <span class="dv">400</span>,</span>
<span id="cb429-68"><a href="interactivity-with-the-rspatial-ecosystem.html#cb429-68" tabindex="-1"></a>                                <span class="at">name =</span> <span class="st">&quot;spatial_network&quot;</span>)</span>
<span id="cb429-69"><a href="interactivity-with-the-rspatial-ecosystem.html#cb429-69" tabindex="-1"></a></span>
<span id="cb429-70"><a href="interactivity-with-the-rspatial-ecosystem.html#cb429-70" tabindex="-1"></a><span class="fu">spatPlot2D</span>(<span class="at">gobject =</span> v_brain,</span>
<span id="cb429-71"><a href="interactivity-with-the-rspatial-ecosystem.html#cb429-71" tabindex="-1"></a>           <span class="at">spat_unit =</span> <span class="st">&quot;cell&quot;</span>,</span>
<span id="cb429-72"><a href="interactivity-with-the-rspatial-ecosystem.html#cb429-72" tabindex="-1"></a>           <span class="at">cell_color =</span> <span class="st">&quot;leiden_clus&quot;</span>,</span>
<span id="cb429-73"><a href="interactivity-with-the-rspatial-ecosystem.html#cb429-73" tabindex="-1"></a>           <span class="at">show_image =</span> <span class="cn">TRUE</span>,</span>
<span id="cb429-74"><a href="interactivity-with-the-rspatial-ecosystem.html#cb429-74" tabindex="-1"></a>           <span class="at">point_size =</span> <span class="fl">1.5</span>,</span>
<span id="cb429-75"><a href="interactivity-with-the-rspatial-ecosystem.html#cb429-75" tabindex="-1"></a>           <span class="at">point_shape =</span> <span class="st">&quot;no_border&quot;</span>,</span>
<span id="cb429-76"><a href="interactivity-with-the-rspatial-ecosystem.html#cb429-76" tabindex="-1"></a>           <span class="at">background_color =</span> <span class="st">&quot;black&quot;</span>,</span>
<span id="cb429-77"><a href="interactivity-with-the-rspatial-ecosystem.html#cb429-77" tabindex="-1"></a>           <span class="at">show_legend =</span> <span class="cn">TRUE</span>,</span>
<span id="cb429-78"><a href="interactivity-with-the-rspatial-ecosystem.html#cb429-78" tabindex="-1"></a>           <span class="at">save_plot =</span> <span class="cn">TRUE</span>,</span>
<span id="cb429-79"><a href="interactivity-with-the-rspatial-ecosystem.html#cb429-79" tabindex="-1"></a>           <span class="at">save_param =</span> <span class="fu">list</span>(<span class="at">save_name =</span> <span class="st">&quot;03_ses6_1_vis_spat&quot;</span>))</span></code></pre></div>
<p>Here we can see the clustering of the regular visium spots is able to identify distinct regions of the mouse brain.</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-602"></span>
<img src="img/03_session6/03_ses6_1_vis_spat.png" alt="Mouse brain spatial plot showing leiden clustering" width="75%" />
<p class="caption">
Figure 17.2: Mouse brain spatial plot showing leiden clustering
</p>
</div>
<div id="identifying-spatially-organized-features" class="section level3 hasAnchor" number="17.5.1">
<h3><span class="header-section-number">17.5.1</span> Identifying spatially organized features<a href="interactivity-with-the-rspatial-ecosystem.html#identifying-spatially-organized-features" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>We need to identify genes to be used for interpolation. This works best with genes that are spatially distinct. To identify these genes weâll use binSpect(). For this tutorial weâll only use the top 15 spatially distinct genes. The more genes used for interpolation the longer the analysis will take. When running this for your own datasets you should use more genes. We are only using 15 here to minimize analysis time.</p>
<div class="sourceCode" id="cb430"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb430-1"><a href="interactivity-with-the-rspatial-ecosystem.html#cb430-1" tabindex="-1"></a><span class="co"># Spatially Variable Features</span></span>
<span id="cb430-2"><a href="interactivity-with-the-rspatial-ecosystem.html#cb430-2" tabindex="-1"></a>ranktest <span class="ot">&lt;-</span> <span class="fu">binSpect</span>(v_brain,</span>
<span id="cb430-3"><a href="interactivity-with-the-rspatial-ecosystem.html#cb430-3" tabindex="-1"></a>                     <span class="at">bin_method =</span> <span class="st">&quot;rank&quot;</span>,</span>
<span id="cb430-4"><a href="interactivity-with-the-rspatial-ecosystem.html#cb430-4" tabindex="-1"></a>                     <span class="at">calc_hub =</span> <span class="cn">TRUE</span>,</span>
<span id="cb430-5"><a href="interactivity-with-the-rspatial-ecosystem.html#cb430-5" tabindex="-1"></a>                     <span class="at">hub_min_int =</span> <span class="dv">5</span>,</span>
<span id="cb430-6"><a href="interactivity-with-the-rspatial-ecosystem.html#cb430-6" tabindex="-1"></a>                     <span class="at">spatial_network_name =</span> <span class="st">&quot;spatial_network&quot;</span>,</span>
<span id="cb430-7"><a href="interactivity-with-the-rspatial-ecosystem.html#cb430-7" tabindex="-1"></a>                     <span class="at">do_parallel =</span> <span class="cn">TRUE</span>,</span>
<span id="cb430-8"><a href="interactivity-with-the-rspatial-ecosystem.html#cb430-8" tabindex="-1"></a>                     <span class="at">cores =</span> <span class="dv">8</span>) <span class="co">#not able to provide a seed number, so do not set one</span></span>
<span id="cb430-9"><a href="interactivity-with-the-rspatial-ecosystem.html#cb430-9" tabindex="-1"></a></span>
<span id="cb430-10"><a href="interactivity-with-the-rspatial-ecosystem.html#cb430-10" tabindex="-1"></a><span class="co"># Getting the top 15 spatially organized genes</span></span>
<span id="cb430-11"><a href="interactivity-with-the-rspatial-ecosystem.html#cb430-11" tabindex="-1"></a>ext_spatial_features <span class="ot">&lt;-</span> ranktest[<span class="dv">1</span><span class="sc">:</span><span class="dv">15</span>,]<span class="sc">$</span>feats</span></code></pre></div>
</div>
</div>
<div id="adding-cell-polygons-to-giotto-object" class="section level2 hasAnchor" number="17.6">
<h2><span class="header-section-number">17.6</span> Adding cell polygons to Giotto object<a href="interactivity-with-the-rspatial-ecosystem.html#adding-cell-polygons-to-giotto-object" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="read-in-the-poly-information" class="section level3 hasAnchor" number="17.6.1">
<h3><span class="header-section-number">17.6.1</span> Read in the poly information<a href="interactivity-with-the-rspatial-ecosystem.html#read-in-the-poly-information" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>First we need to read in the geojson file that contains the cell polygons that weâll interpolate gene expression onto. These will then be added to the Giotto object as a new polygon object. This wonât affect the visium polygons. Both polygons will be stored within the same Giotto object.</p>
<div class="sourceCode" id="cb431"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb431-1"><a href="interactivity-with-the-rspatial-ecosystem.html#cb431-1" tabindex="-1"></a><span class="co"># Read in the data</span></span>
<span id="cb431-2"><a href="interactivity-with-the-rspatial-ecosystem.html#cb431-2" tabindex="-1"></a>stardist_cell_poly_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_directory, <span class="st">&quot;segmentations/stardist_only_cell_bounds.geojson&quot;</span>)</span>
<span id="cb431-3"><a href="interactivity-with-the-rspatial-ecosystem.html#cb431-3" tabindex="-1"></a></span>
<span id="cb431-4"><a href="interactivity-with-the-rspatial-ecosystem.html#cb431-4" tabindex="-1"></a>stardist_cell_gpoly <span class="ot">&lt;-</span> <span class="fu">createGiottoPolygonsFromGeoJSON</span>(<span class="at">GeoJSON =</span> stardist_cell_poly_path,</span>
<span id="cb431-5"><a href="interactivity-with-the-rspatial-ecosystem.html#cb431-5" tabindex="-1"></a>                                                       <span class="at">name =</span> <span class="st">&quot;stardist_cell&quot;</span>,</span>
<span id="cb431-6"><a href="interactivity-with-the-rspatial-ecosystem.html#cb431-6" tabindex="-1"></a>                                                       <span class="at">calc_centroids =</span> <span class="cn">TRUE</span>)</span>
<span id="cb431-7"><a href="interactivity-with-the-rspatial-ecosystem.html#cb431-7" tabindex="-1"></a></span>
<span id="cb431-8"><a href="interactivity-with-the-rspatial-ecosystem.html#cb431-8" tabindex="-1"></a>stardist_cell_gpoly <span class="ot">&lt;-</span> <span class="fu">flip</span>(stardist_cell_gpoly)</span></code></pre></div>
</div>
<div id="vizualizing-polygons" class="section level3 hasAnchor" number="17.6.2">
<h3><span class="header-section-number">17.6.2</span> Vizualizing polygons<a href="interactivity-with-the-rspatial-ecosystem.html#vizualizing-polygons" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Below we can see a visualization of the polygons for the visium and the nuclei we identified from the H&amp;E image. The visium dataset has 2698 spots compared to the 36694 nuclei we identified. Just using the visium spots weâre therefore losing a lot of the spatial data for individual cells. With the increased number of spots and them directly correlating with the tissue, through the spots alone we are able to better see the actual structure of the mouse brain.</p>
<div class="sourceCode" id="cb432"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb432-1"><a href="interactivity-with-the-rspatial-ecosystem.html#cb432-1" tabindex="-1"></a><span class="fu">plot</span>(<span class="fu">getPolygonInfo</span>(v_brain))</span>
<span id="cb432-2"><a href="interactivity-with-the-rspatial-ecosystem.html#cb432-2" tabindex="-1"></a></span>
<span id="cb432-3"><a href="interactivity-with-the-rspatial-ecosystem.html#cb432-3" tabindex="-1"></a><span class="fu">plot</span>(stardist_cell_gpoly, <span class="at">max_poly =</span> <span class="fl">1e6</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-606"></span>
<img src="img/03_session6/visium_polys.png" alt="Mouse brain cell polygons from the visium dataset" width="100%" />
<p class="caption">
Figure 17.3: Mouse brain cell polygons from the visium dataset
</p>
</div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-607"></span>
<img src="img/03_session6/03_ses6_2_cell_poly_flipped.png" alt="Mouse brain cell polygons with artifacts removed and flipped" width="100%" />
<p class="caption">
Figure 17.4: Mouse brain cell polygons with artifacts removed and flipped
</p>
</div>
</div>
<div id="showing-giotto-object-prior-to-polygon-addition" class="section level3 hasAnchor" number="17.6.3">
<h3><span class="header-section-number">17.6.3</span> Showing Giotto object prior to polygon addition<a href="interactivity-with-the-rspatial-ecosystem.html#showing-giotto-object-prior-to-polygon-addition" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Before we add the polygons we can see the gobject contains âcellâ as a spatial unit and a polygon.</p>
<div class="sourceCode" id="cb433"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb433-1"><a href="interactivity-with-the-rspatial-ecosystem.html#cb433-1" tabindex="-1"></a><span class="fu">print</span>(v_brain)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-609"></span>
<img src="img/03_session6/visium_gobject.png" alt="Giotto object before adding subcellular polygons." width="40%" />
<p class="caption">
Figure 17.5: Giotto object before adding subcellular polygons.
</p>
</div>
</div>
<div id="adding-polygons-to-giotto-object" class="section level3 hasAnchor" number="17.6.4">
<h3><span class="header-section-number">17.6.4</span> Adding polygons to giotto object<a href="interactivity-with-the-rspatial-ecosystem.html#adding-polygons-to-giotto-object" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>After we add the nuclei polygons we can see that a new polygon name, âstardist_cellâ has been added to the gobject.</p>
<div class="sourceCode" id="cb434"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb434-1"><a href="interactivity-with-the-rspatial-ecosystem.html#cb434-1" tabindex="-1"></a>v_brain <span class="ot">&lt;-</span> <span class="fu">addGiottoPolygons</span>(v_brain, </span>
<span id="cb434-2"><a href="interactivity-with-the-rspatial-ecosystem.html#cb434-2" tabindex="-1"></a>                             <span class="at">gpolygons =</span> <span class="fu">list</span>(<span class="st">&quot;stardist_cell&quot;</span> <span class="ot">=</span> stardist_cell_gpoly))</span>
<span id="cb434-3"><a href="interactivity-with-the-rspatial-ecosystem.html#cb434-3" tabindex="-1"></a></span>
<span id="cb434-4"><a href="interactivity-with-the-rspatial-ecosystem.html#cb434-4" tabindex="-1"></a><span class="fu">print</span>(v_brain)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-611"></span>
<img src="img/03_session6/03_ses6_gobject_sub.png" alt="Giotto object after to adding subcellular polygons." width="40%" />
<p class="caption">
Figure 17.6: Giotto object after to adding subcellular polygons.
</p>
</div>
</div>
<div id="check-polygon-information" class="section level3 hasAnchor" number="17.6.5">
<h3><span class="header-section-number">17.6.5</span> Check polygon information<a href="interactivity-with-the-rspatial-ecosystem.html#check-polygon-information" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>We can now see the addition of the new polygons under the name âstardist_cellâ. Each of the new polyons is given a unique poly_ID as shown below. Each polygon is also added into same space as the original visium spots, therefore line up with the same image as the visium spots.</p>
<div class="sourceCode" id="cb435"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb435-1"><a href="interactivity-with-the-rspatial-ecosystem.html#cb435-1" tabindex="-1"></a>poly_info <span class="ot">&lt;-</span> <span class="fu">getPolygonInfo</span>(v_brain,</span>
<span id="cb435-2"><a href="interactivity-with-the-rspatial-ecosystem.html#cb435-2" tabindex="-1"></a>                            <span class="at">polygon_name =</span> <span class="st">&quot;stardist_cell&quot;</span>)</span>
<span id="cb435-3"><a href="interactivity-with-the-rspatial-ecosystem.html#cb435-3" tabindex="-1"></a></span>
<span id="cb435-4"><a href="interactivity-with-the-rspatial-ecosystem.html#cb435-4" tabindex="-1"></a><span class="fu">print</span>(poly_info)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-613"></span>
<img src="img/03_session6/stardist_poly_info.png" alt="Polygon information for stardist_cell." width="60%" />
<p class="caption">
Figure 17.7: Polygon information for stardist_cell.
</p>
</div>
</div>
</div>
<div id="performing-kriging" class="section level2 hasAnchor" number="17.7">
<h2><span class="header-section-number">17.7</span> Performing kriging<a href="interactivity-with-the-rspatial-ecosystem.html#performing-kriging" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="interpolating-features" class="section level3 hasAnchor" number="17.7.1">
<h3><span class="header-section-number">17.7.1</span> Interpolating features<a href="interactivity-with-the-rspatial-ecosystem.html#interpolating-features" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Now we can perform the first step in interpolating expression data onto cell polygons. This involves creating a raster image for the gene expression of each of the selected genes. The steps from here can be time consuming and require large amounts of memory.</p>
<p>We will only be analyzing 15 genes to show the process of expression interpolation. For clustering and other analyses more genes are required.</p>
<div class="sourceCode" id="cb436"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb436-1"><a href="interactivity-with-the-rspatial-ecosystem.html#cb436-1" tabindex="-1"></a>future<span class="sc">::</span><span class="fu">plan</span>(future<span class="sc">::</span><span class="fu">multisession</span>()) <span class="co"># comment out for single threading</span></span>
<span id="cb436-2"><a href="interactivity-with-the-rspatial-ecosystem.html#cb436-2" tabindex="-1"></a></span>
<span id="cb436-3"><a href="interactivity-with-the-rspatial-ecosystem.html#cb436-3" tabindex="-1"></a>subcell_v_brain <span class="ot">&lt;-</span> <span class="fu">interpolateFeature</span>(v_brain,</span>
<span id="cb436-4"><a href="interactivity-with-the-rspatial-ecosystem.html#cb436-4" tabindex="-1"></a>                                      <span class="at">spat_unit =</span> <span class="st">&quot;cell&quot;</span>,</span>
<span id="cb436-5"><a href="interactivity-with-the-rspatial-ecosystem.html#cb436-5" tabindex="-1"></a>                                      <span class="at">feat_type =</span> <span class="st">&quot;rna&quot;</span>,</span>
<span id="cb436-6"><a href="interactivity-with-the-rspatial-ecosystem.html#cb436-6" tabindex="-1"></a>                                      <span class="at">ext =</span> <span class="fu">ext</span>(v_brain),</span>
<span id="cb436-7"><a href="interactivity-with-the-rspatial-ecosystem.html#cb436-7" tabindex="-1"></a>                                      <span class="at">feats =</span> ext_spatial_features,</span>
<span id="cb436-8"><a href="interactivity-with-the-rspatial-ecosystem.html#cb436-8" tabindex="-1"></a>                                      <span class="at">overwrite =</span> <span class="cn">TRUE</span>)</span>
<span id="cb436-9"><a href="interactivity-with-the-rspatial-ecosystem.html#cb436-9" tabindex="-1"></a></span>
<span id="cb436-10"><a href="interactivity-with-the-rspatial-ecosystem.html#cb436-10" tabindex="-1"></a><span class="fu">print</span>(subcell_v_brain)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-615"></span>
<img src="img/03_session6/03_ses6_gobject_sub_interpolate.png" alt="Giotto object after to interpolating features. Addition of images for each interoplated feature (left) and an example of rasterized gene expression image (right)." width="100%" />
<p class="caption">
Figure 17.8: Giotto object after to interpolating features. Addition of images for each interoplated feature (left) and an example of rasterized gene expression image (right).
</p>
</div>
<p>For each gene that we interpolate a raster image is exported based on the gene expression. Shown below is an example of an output for the gene Pantr1.</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-616"></span>
<img src="img/03_session6/Pantr1_raster.png" alt="Raster of gene expression interpolation for Pantr1" width="50%" />
<p class="caption">
Figure 17.9: Raster of gene expression interpolation for Pantr1
</p>
</div>
</div>
<div id="expression-overlap" class="section level3 hasAnchor" number="17.7.2">
<h3><span class="header-section-number">17.7.2</span> Expression overlap<a href="interactivity-with-the-rspatial-ecosystem.html#expression-overlap" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The raster we created above gives the gene expression in a graphical form. We next need to determine how that relates to the nuclei location. To determine that we will calculate the overlap of the rasterized gene expression image to the polygons supplied earlier.</p>
<p>This step also takes more time the more genes that are provided. For large datasets please allow up to multiple hours for these steps to run.</p>
<div class="sourceCode" id="cb437"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb437-1"><a href="interactivity-with-the-rspatial-ecosystem.html#cb437-1" tabindex="-1"></a>subcell_v_brain <span class="ot">&lt;-</span> <span class="fu">calculateOverlapPolygonImages</span>(<span class="at">gobject =</span> subcell_v_brain,</span>
<span id="cb437-2"><a href="interactivity-with-the-rspatial-ecosystem.html#cb437-2" tabindex="-1"></a>                                                 <span class="at">name_overlap =</span> <span class="st">&quot;rna&quot;</span>,</span>
<span id="cb437-3"><a href="interactivity-with-the-rspatial-ecosystem.html#cb437-3" tabindex="-1"></a>                                                 <span class="at">spatial_info =</span> <span class="st">&quot;stardist_cell&quot;</span>,</span>
<span id="cb437-4"><a href="interactivity-with-the-rspatial-ecosystem.html#cb437-4" tabindex="-1"></a>                                                 <span class="at">image_names =</span> ext_spatial_features)</span>
<span id="cb437-5"><a href="interactivity-with-the-rspatial-ecosystem.html#cb437-5" tabindex="-1"></a></span>
<span id="cb437-6"><a href="interactivity-with-the-rspatial-ecosystem.html#cb437-6" tabindex="-1"></a>subcell_v_brain <span class="ot">&lt;-</span> Giotto<span class="sc">::</span><span class="fu">overlapToMatrix</span>(<span class="at">x =</span> subcell_v_brain,</span>
<span id="cb437-7"><a href="interactivity-with-the-rspatial-ecosystem.html#cb437-7" tabindex="-1"></a>                                           <span class="at">poly_info =</span> <span class="st">&quot;stardist_cell&quot;</span>,</span>
<span id="cb437-8"><a href="interactivity-with-the-rspatial-ecosystem.html#cb437-8" tabindex="-1"></a>                                           <span class="at">feat_info =</span> <span class="st">&quot;rna&quot;</span>,</span>
<span id="cb437-9"><a href="interactivity-with-the-rspatial-ecosystem.html#cb437-9" tabindex="-1"></a>                                           <span class="at">aggr_function =</span> <span class="st">&quot;sum&quot;</span>,</span>
<span id="cb437-10"><a href="interactivity-with-the-rspatial-ecosystem.html#cb437-10" tabindex="-1"></a>                                           <span class="at">type=</span><span class="st">&quot;intensity&quot;</span>)</span></code></pre></div>
<p>After performing the overlap we now have expression data for each gene provided. This can be seen below where we see the interpolated gene expression for genes in each of the nuclei we identified.</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-618"></span>
<img src="img/03_session6/03_ses6_interpolate_gene_express.png" alt="Gene expression for cells based on interpolation." width="100%" />
<p class="caption">
Figure 17.10: Gene expression for cells based on interpolation.
</p>
</div>
</div>
</div>
<div id="reading-in-larger-dataset" class="section level2 hasAnchor" number="17.8">
<h2><span class="header-section-number">17.8</span> Reading in larger dataset<a href="interactivity-with-the-rspatial-ecosystem.html#reading-in-larger-dataset" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>For better results more genes are required. The above data used only 15 genes. We will now read in a dataset that has 1500 interpolated genes an use this for the remained of the tutorial. If you havenât downloaded this dataset please download it <a href="https://zenodo.org/records/13144556">here</a>.</p>
<div class="sourceCode" id="cb438"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb438-1"><a href="interactivity-with-the-rspatial-ecosystem.html#cb438-1" tabindex="-1"></a>subcell_v_brain <span class="ot">&lt;-</span> <span class="fu">loadGiotto</span>(<span class="fu">file.path</span>(data_directory, <span class="st">&quot;subcellular_gobject&quot;</span>))</span></code></pre></div>
</div>
<div id="analyzing-interpolated-features" class="section level2 hasAnchor" number="17.9">
<h2><span class="header-section-number">17.9</span> Analyzing interpolated features<a href="interactivity-with-the-rspatial-ecosystem.html#analyzing-interpolated-features" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="filter-and-normalization" class="section level3 hasAnchor" number="17.9.1">
<h3><span class="header-section-number">17.9.1</span> Filter and normalization<a href="interactivity-with-the-rspatial-ecosystem.html#filter-and-normalization" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Now that we have a valid spat unit and gene expression data for each of the provided genes we can now perform the same analyses we used for the regular visium data. Please note that due to the differences in cell number that the values used for the current analysis arenât identical to the visium analysis.</p>
<div class="sourceCode" id="cb439"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb439-1"><a href="interactivity-with-the-rspatial-ecosystem.html#cb439-1" tabindex="-1"></a>subcell_v_brain <span class="ot">&lt;-</span> <span class="fu">filterGiotto</span>(<span class="at">gobject =</span> subcell_v_brain,</span>
<span id="cb439-2"><a href="interactivity-with-the-rspatial-ecosystem.html#cb439-2" tabindex="-1"></a>                                <span class="at">spat_unit =</span> <span class="st">&quot;stardist_cell&quot;</span>,</span>
<span id="cb439-3"><a href="interactivity-with-the-rspatial-ecosystem.html#cb439-3" tabindex="-1"></a>                                <span class="at">expression_values =</span> <span class="st">&quot;raw&quot;</span>,</span>
<span id="cb439-4"><a href="interactivity-with-the-rspatial-ecosystem.html#cb439-4" tabindex="-1"></a>                                <span class="at">expression_threshold =</span> <span class="dv">1</span>,</span>
<span id="cb439-5"><a href="interactivity-with-the-rspatial-ecosystem.html#cb439-5" tabindex="-1"></a>                                <span class="at">feat_det_in_min_cells =</span> <span class="dv">0</span>,</span>
<span id="cb439-6"><a href="interactivity-with-the-rspatial-ecosystem.html#cb439-6" tabindex="-1"></a>                                <span class="at">min_det_feats_per_cell =</span> <span class="dv">1</span>)</span>
<span id="cb439-7"><a href="interactivity-with-the-rspatial-ecosystem.html#cb439-7" tabindex="-1"></a></span>
<span id="cb439-8"><a href="interactivity-with-the-rspatial-ecosystem.html#cb439-8" tabindex="-1"></a>subcell_v_brain <span class="ot">&lt;-</span> <span class="fu">normalizeGiotto</span>(<span class="at">gobject =</span> subcell_v_brain,</span>
<span id="cb439-9"><a href="interactivity-with-the-rspatial-ecosystem.html#cb439-9" tabindex="-1"></a>                                   <span class="at">spat_unit =</span> <span class="st">&quot;stardist_cell&quot;</span>,</span>
<span id="cb439-10"><a href="interactivity-with-the-rspatial-ecosystem.html#cb439-10" tabindex="-1"></a>                                   <span class="at">scalefactor =</span> <span class="dv">6000</span>,</span>
<span id="cb439-11"><a href="interactivity-with-the-rspatial-ecosystem.html#cb439-11" tabindex="-1"></a>                                   <span class="at">verbose =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
</div>
<div id="visualizing-gene-expression-from-interpolated-expression" class="section level3 hasAnchor" number="17.9.2">
<h3><span class="header-section-number">17.9.2</span> Visualizing gene expression from interpolated expression<a href="interactivity-with-the-rspatial-ecosystem.html#visualizing-gene-expression-from-interpolated-expression" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Since we have the gene expression information for both the visium and the interpolated gene expression we can visualize gene expression for both from the same Giotto object. We will look at the expression for two genes âSparcâ and âPantr1â for both the visium and interpolated data.</p>
<div class="sourceCode" id="cb440"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb440-1"><a href="interactivity-with-the-rspatial-ecosystem.html#cb440-1" tabindex="-1"></a><span class="fu">spatFeatPlot2D</span>(subcell_v_brain,</span>
<span id="cb440-2"><a href="interactivity-with-the-rspatial-ecosystem.html#cb440-2" tabindex="-1"></a>               <span class="at">spat_unit =</span> <span class="st">&quot;cell&quot;</span>,</span>
<span id="cb440-3"><a href="interactivity-with-the-rspatial-ecosystem.html#cb440-3" tabindex="-1"></a>               <span class="at">gradient_style =</span> <span class="st">&quot;sequential&quot;</span>,</span>
<span id="cb440-4"><a href="interactivity-with-the-rspatial-ecosystem.html#cb440-4" tabindex="-1"></a>               <span class="at">cell_color_gradient =</span> <span class="st">&quot;Geyser&quot;</span>,</span>
<span id="cb440-5"><a href="interactivity-with-the-rspatial-ecosystem.html#cb440-5" tabindex="-1"></a>               <span class="at">feats =</span> <span class="st">&quot;Sparc&quot;</span>,</span>
<span id="cb440-6"><a href="interactivity-with-the-rspatial-ecosystem.html#cb440-6" tabindex="-1"></a>               <span class="at">point_size =</span> <span class="dv">2</span>,</span>
<span id="cb440-7"><a href="interactivity-with-the-rspatial-ecosystem.html#cb440-7" tabindex="-1"></a>               <span class="at">save_plot =</span> <span class="cn">TRUE</span>,</span>
<span id="cb440-8"><a href="interactivity-with-the-rspatial-ecosystem.html#cb440-8" tabindex="-1"></a>               <span class="at">show_image =</span> <span class="cn">TRUE</span>,</span>
<span id="cb440-9"><a href="interactivity-with-the-rspatial-ecosystem.html#cb440-9" tabindex="-1"></a>               <span class="at">save_param =</span> <span class="fu">list</span>(<span class="at">save_name =</span> <span class="st">&quot;03_ses6_sparc_vis&quot;</span>))</span>
<span id="cb440-10"><a href="interactivity-with-the-rspatial-ecosystem.html#cb440-10" tabindex="-1"></a></span>
<span id="cb440-11"><a href="interactivity-with-the-rspatial-ecosystem.html#cb440-11" tabindex="-1"></a><span class="fu">spatFeatPlot2D</span>(subcell_v_brain,</span>
<span id="cb440-12"><a href="interactivity-with-the-rspatial-ecosystem.html#cb440-12" tabindex="-1"></a>               <span class="at">spat_unit =</span> <span class="st">&quot;stardist_cell&quot;</span>,</span>
<span id="cb440-13"><a href="interactivity-with-the-rspatial-ecosystem.html#cb440-13" tabindex="-1"></a>               <span class="at">gradient_style =</span> <span class="st">&quot;sequential&quot;</span>,</span>
<span id="cb440-14"><a href="interactivity-with-the-rspatial-ecosystem.html#cb440-14" tabindex="-1"></a>               <span class="at">cell_color_gradient =</span> <span class="st">&quot;Geyser&quot;</span>,</span>
<span id="cb440-15"><a href="interactivity-with-the-rspatial-ecosystem.html#cb440-15" tabindex="-1"></a>               <span class="at">feats =</span> <span class="st">&quot;Sparc&quot;</span>,</span>
<span id="cb440-16"><a href="interactivity-with-the-rspatial-ecosystem.html#cb440-16" tabindex="-1"></a>               <span class="at">point_size =</span> <span class="fl">0.6</span>,</span>
<span id="cb440-17"><a href="interactivity-with-the-rspatial-ecosystem.html#cb440-17" tabindex="-1"></a>               <span class="at">save_plot =</span> <span class="cn">TRUE</span>,</span>
<span id="cb440-18"><a href="interactivity-with-the-rspatial-ecosystem.html#cb440-18" tabindex="-1"></a>               <span class="at">show_image =</span> <span class="cn">TRUE</span>,</span>
<span id="cb440-19"><a href="interactivity-with-the-rspatial-ecosystem.html#cb440-19" tabindex="-1"></a>               <span class="at">save_param =</span> <span class="fu">list</span>(<span class="at">save_name =</span> <span class="st">&quot;03_ses6_sparc&quot;</span>))</span>
<span id="cb440-20"><a href="interactivity-with-the-rspatial-ecosystem.html#cb440-20" tabindex="-1"></a></span>
<span id="cb440-21"><a href="interactivity-with-the-rspatial-ecosystem.html#cb440-21" tabindex="-1"></a><span class="fu">spatFeatPlot2D</span>(subcell_v_brain,</span>
<span id="cb440-22"><a href="interactivity-with-the-rspatial-ecosystem.html#cb440-22" tabindex="-1"></a>               <span class="at">spat_unit =</span> <span class="st">&quot;cell&quot;</span>,</span>
<span id="cb440-23"><a href="interactivity-with-the-rspatial-ecosystem.html#cb440-23" tabindex="-1"></a>               <span class="at">gradient_style =</span> <span class="st">&quot;sequential&quot;</span>,</span>
<span id="cb440-24"><a href="interactivity-with-the-rspatial-ecosystem.html#cb440-24" tabindex="-1"></a>               <span class="at">feats =</span> <span class="st">&quot;Pantr1&quot;</span>,</span>
<span id="cb440-25"><a href="interactivity-with-the-rspatial-ecosystem.html#cb440-25" tabindex="-1"></a>               <span class="at">cell_color_gradient =</span> <span class="st">&quot;Geyser&quot;</span>,</span>
<span id="cb440-26"><a href="interactivity-with-the-rspatial-ecosystem.html#cb440-26" tabindex="-1"></a>               <span class="at">point_size =</span> <span class="dv">2</span>,</span>
<span id="cb440-27"><a href="interactivity-with-the-rspatial-ecosystem.html#cb440-27" tabindex="-1"></a>               <span class="at">save_plot =</span> <span class="cn">TRUE</span>,</span>
<span id="cb440-28"><a href="interactivity-with-the-rspatial-ecosystem.html#cb440-28" tabindex="-1"></a>               <span class="at">show_image =</span> <span class="cn">TRUE</span>,</span>
<span id="cb440-29"><a href="interactivity-with-the-rspatial-ecosystem.html#cb440-29" tabindex="-1"></a>               <span class="at">save_param =</span> <span class="fu">list</span>(<span class="at">save_name =</span> <span class="st">&quot;03_ses6_pantr1_vis&quot;</span>))</span>
<span id="cb440-30"><a href="interactivity-with-the-rspatial-ecosystem.html#cb440-30" tabindex="-1"></a></span>
<span id="cb440-31"><a href="interactivity-with-the-rspatial-ecosystem.html#cb440-31" tabindex="-1"></a><span class="fu">spatFeatPlot2D</span>(subcell_v_brain,</span>
<span id="cb440-32"><a href="interactivity-with-the-rspatial-ecosystem.html#cb440-32" tabindex="-1"></a>               <span class="at">spat_unit =</span> <span class="st">&quot;stardist_cell&quot;</span>,</span>
<span id="cb440-33"><a href="interactivity-with-the-rspatial-ecosystem.html#cb440-33" tabindex="-1"></a>               <span class="at">gradient_style =</span> <span class="st">&quot;sequential&quot;</span>,</span>
<span id="cb440-34"><a href="interactivity-with-the-rspatial-ecosystem.html#cb440-34" tabindex="-1"></a>               <span class="at">cell_color_gradient =</span> <span class="st">&quot;Geyser&quot;</span>,</span>
<span id="cb440-35"><a href="interactivity-with-the-rspatial-ecosystem.html#cb440-35" tabindex="-1"></a>               <span class="at">feats =</span> <span class="st">&quot;Pantr1&quot;</span>,</span>
<span id="cb440-36"><a href="interactivity-with-the-rspatial-ecosystem.html#cb440-36" tabindex="-1"></a>               <span class="at">point_size =</span> <span class="fl">0.6</span>,</span>
<span id="cb440-37"><a href="interactivity-with-the-rspatial-ecosystem.html#cb440-37" tabindex="-1"></a>               <span class="at">save_plot =</span> <span class="cn">TRUE</span>,</span>
<span id="cb440-38"><a href="interactivity-with-the-rspatial-ecosystem.html#cb440-38" tabindex="-1"></a>               <span class="at">show_image =</span> <span class="cn">TRUE</span>,</span>
<span id="cb440-39"><a href="interactivity-with-the-rspatial-ecosystem.html#cb440-39" tabindex="-1"></a>               <span class="at">save_param =</span> <span class="fu">list</span>(<span class="at">save_name =</span> <span class="st">&quot;03_ses6_pantr1&quot;</span>))</span></code></pre></div>
<p>Below we can see the gene expression for both datatypes. With the interpolated gene expression weâre able to get a better idea as to the cells that are expressing each of the genes. This is especially clear with Pantr1, which clearly localizes to the pyramidal layer.</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-622"></span>
<img src="img/03_session6/03_ses6_gene_expression.png" alt="Gene expression for visium (left) and interpolated (right) expression for Sparc (top) and Pantr1 (bottom)." width="100%" />
<p class="caption">
Figure 17.11: Gene expression for visium (left) and interpolated (right) expression for Sparc (top) and Pantr1 (bottom).
</p>
</div>
</div>
<div id="run-pca" class="section level3 hasAnchor" number="17.9.3">
<h3><span class="header-section-number">17.9.3</span> Run PCA<a href="interactivity-with-the-rspatial-ecosystem.html#run-pca" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb441"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb441-1"><a href="interactivity-with-the-rspatial-ecosystem.html#cb441-1" tabindex="-1"></a>subcell_v_brain <span class="ot">&lt;-</span> <span class="fu">runPCA</span>(<span class="at">gobject =</span> subcell_v_brain,</span>
<span id="cb441-2"><a href="interactivity-with-the-rspatial-ecosystem.html#cb441-2" tabindex="-1"></a>                          <span class="at">spat_unit =</span> <span class="st">&quot;stardist_cell&quot;</span>,</span>
<span id="cb441-3"><a href="interactivity-with-the-rspatial-ecosystem.html#cb441-3" tabindex="-1"></a>                          <span class="at">expression_values =</span> <span class="st">&quot;normalized&quot;</span>,</span>
<span id="cb441-4"><a href="interactivity-with-the-rspatial-ecosystem.html#cb441-4" tabindex="-1"></a>                          <span class="at">feats_to_use =</span> <span class="cn">NULL</span>)</span></code></pre></div>
</div>
<div id="clustering-2" class="section level3 hasAnchor" number="17.9.4">
<h3><span class="header-section-number">17.9.4</span> Clustering<a href="interactivity-with-the-rspatial-ecosystem.html#clustering-2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb442"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb442-1"><a href="interactivity-with-the-rspatial-ecosystem.html#cb442-1" tabindex="-1"></a><span class="co"># UMAP</span></span>
<span id="cb442-2"><a href="interactivity-with-the-rspatial-ecosystem.html#cb442-2" tabindex="-1"></a>subcell_v_brain <span class="ot">&lt;-</span> <span class="fu">runUMAP</span>(subcell_v_brain,</span>
<span id="cb442-3"><a href="interactivity-with-the-rspatial-ecosystem.html#cb442-3" tabindex="-1"></a>                           <span class="at">spat_unit =</span> <span class="st">&quot;stardist_cell&quot;</span>,</span>
<span id="cb442-4"><a href="interactivity-with-the-rspatial-ecosystem.html#cb442-4" tabindex="-1"></a>                           <span class="at">dimensions_to_use =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">15</span>,</span>
<span id="cb442-5"><a href="interactivity-with-the-rspatial-ecosystem.html#cb442-5" tabindex="-1"></a>                           <span class="at">n_neighbors =</span> <span class="dv">1000</span>,</span>
<span id="cb442-6"><a href="interactivity-with-the-rspatial-ecosystem.html#cb442-6" tabindex="-1"></a>                           <span class="at">min_dist =</span> <span class="fl">0.001</span>,</span>
<span id="cb442-7"><a href="interactivity-with-the-rspatial-ecosystem.html#cb442-7" tabindex="-1"></a>                           <span class="at">spread =</span> <span class="dv">1</span>)</span>
<span id="cb442-8"><a href="interactivity-with-the-rspatial-ecosystem.html#cb442-8" tabindex="-1"></a></span>
<span id="cb442-9"><a href="interactivity-with-the-rspatial-ecosystem.html#cb442-9" tabindex="-1"></a><span class="co"># NN Network</span></span>
<span id="cb442-10"><a href="interactivity-with-the-rspatial-ecosystem.html#cb442-10" tabindex="-1"></a>subcell_v_brain <span class="ot">&lt;-</span> <span class="fu">createNearestNetwork</span>(<span class="at">gobject =</span> subcell_v_brain,</span>
<span id="cb442-11"><a href="interactivity-with-the-rspatial-ecosystem.html#cb442-11" tabindex="-1"></a>                                        <span class="at">spat_unit =</span> <span class="st">&quot;stardist_cell&quot;</span>,</span>
<span id="cb442-12"><a href="interactivity-with-the-rspatial-ecosystem.html#cb442-12" tabindex="-1"></a>                                        <span class="at">dimensions_to_use =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>,</span>
<span id="cb442-13"><a href="interactivity-with-the-rspatial-ecosystem.html#cb442-13" tabindex="-1"></a>                                        <span class="at">feats_to_use =</span> hv_feats,</span>
<span id="cb442-14"><a href="interactivity-with-the-rspatial-ecosystem.html#cb442-14" tabindex="-1"></a>                                        <span class="at">expression_values =</span> <span class="st">&quot;normalized&quot;</span>,</span>
<span id="cb442-15"><a href="interactivity-with-the-rspatial-ecosystem.html#cb442-15" tabindex="-1"></a>                                        <span class="at">k =</span> <span class="dv">70</span>)</span>
<span id="cb442-16"><a href="interactivity-with-the-rspatial-ecosystem.html#cb442-16" tabindex="-1"></a></span>
<span id="cb442-17"><a href="interactivity-with-the-rspatial-ecosystem.html#cb442-17" tabindex="-1"></a>subcell_v_brain <span class="ot">&lt;-</span> <span class="fu">doLeidenCluster</span>(<span class="at">gobject =</span> subcell_v_brain,</span>
<span id="cb442-18"><a href="interactivity-with-the-rspatial-ecosystem.html#cb442-18" tabindex="-1"></a>                                   <span class="at">spat_unit =</span> <span class="st">&quot;stardist_cell&quot;</span>,</span>
<span id="cb442-19"><a href="interactivity-with-the-rspatial-ecosystem.html#cb442-19" tabindex="-1"></a>                                   <span class="at">resolution =</span> <span class="fl">0.15</span>,</span>
<span id="cb442-20"><a href="interactivity-with-the-rspatial-ecosystem.html#cb442-20" tabindex="-1"></a>                                   <span class="at">n_iterations =</span> <span class="dv">100</span>,</span>
<span id="cb442-21"><a href="interactivity-with-the-rspatial-ecosystem.html#cb442-21" tabindex="-1"></a>                                   <span class="at">partition_type =</span> <span class="st">&quot;RBConfigurationVertexPartition&quot;</span>)</span>
<span id="cb442-22"><a href="interactivity-with-the-rspatial-ecosystem.html#cb442-22" tabindex="-1"></a></span>
<span id="cb442-23"><a href="interactivity-with-the-rspatial-ecosystem.html#cb442-23" tabindex="-1"></a><span class="fu">plotUMAP</span>(subcell_v_brain, </span>
<span id="cb442-24"><a href="interactivity-with-the-rspatial-ecosystem.html#cb442-24" tabindex="-1"></a>         <span class="at">spat_unit =</span> <span class="st">&quot;stardist_cell&quot;</span>, </span>
<span id="cb442-25"><a href="interactivity-with-the-rspatial-ecosystem.html#cb442-25" tabindex="-1"></a>         <span class="at">cell_color =</span> <span class="st">&quot;leiden_clus&quot;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-625"></span>
<img src="img/03_session6/03_ses6_subcell_umap.png" alt="UMAP for stardist_cell based on the 1500 interpolated gene expressions. Colored based on leiden clustering." width="100%" />
<p class="caption">
Figure 17.12: UMAP for stardist_cell based on the 1500 interpolated gene expressions. Colored based on leiden clustering.
</p>
</div>
</div>
<div id="visualizing-clustering" class="section level3 hasAnchor" number="17.9.5">
<h3><span class="header-section-number">17.9.5</span> Visualizing clustering<a href="interactivity-with-the-rspatial-ecosystem.html#visualizing-clustering" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Visualizing the clustering for both the visium dataset and the interpolated dataset we can get similar clusters. However, with the interpolated dataset we are able to see finer detail for each cluster.</p>
<div class="sourceCode" id="cb443"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb443-1"><a href="interactivity-with-the-rspatial-ecosystem.html#cb443-1" tabindex="-1"></a><span class="fu">spatPlot2D</span>(<span class="at">gobject =</span> subcell_v_brain,</span>
<span id="cb443-2"><a href="interactivity-with-the-rspatial-ecosystem.html#cb443-2" tabindex="-1"></a>           <span class="at">spat_unit =</span> <span class="st">&quot;cell&quot;</span>,</span>
<span id="cb443-3"><a href="interactivity-with-the-rspatial-ecosystem.html#cb443-3" tabindex="-1"></a>           <span class="at">cell_color =</span> <span class="st">&quot;leiden_clus&quot;</span>,</span>
<span id="cb443-4"><a href="interactivity-with-the-rspatial-ecosystem.html#cb443-4" tabindex="-1"></a>           <span class="at">show_image =</span> <span class="cn">TRUE</span>,</span>
<span id="cb443-5"><a href="interactivity-with-the-rspatial-ecosystem.html#cb443-5" tabindex="-1"></a>           <span class="at">point_size =</span> <span class="fl">0.5</span>,</span>
<span id="cb443-6"><a href="interactivity-with-the-rspatial-ecosystem.html#cb443-6" tabindex="-1"></a>           <span class="at">point_shape =</span> <span class="st">&quot;no_border&quot;</span>,</span>
<span id="cb443-7"><a href="interactivity-with-the-rspatial-ecosystem.html#cb443-7" tabindex="-1"></a>           <span class="at">background_color =</span> <span class="st">&quot;black&quot;</span>,</span>
<span id="cb443-8"><a href="interactivity-with-the-rspatial-ecosystem.html#cb443-8" tabindex="-1"></a>           <span class="at">save_plot =</span> <span class="cn">FALSE</span>,</span>
<span id="cb443-9"><a href="interactivity-with-the-rspatial-ecosystem.html#cb443-9" tabindex="-1"></a>           <span class="at">show_legend =</span> <span class="cn">TRUE</span>)</span>
<span id="cb443-10"><a href="interactivity-with-the-rspatial-ecosystem.html#cb443-10" tabindex="-1"></a></span>
<span id="cb443-11"><a href="interactivity-with-the-rspatial-ecosystem.html#cb443-11" tabindex="-1"></a><span class="fu">spatPlot2D</span>(<span class="at">gobject =</span> subcell_v_brain,</span>
<span id="cb443-12"><a href="interactivity-with-the-rspatial-ecosystem.html#cb443-12" tabindex="-1"></a>           <span class="at">spat_unit =</span> <span class="st">&quot;stardist_cell&quot;</span>,</span>
<span id="cb443-13"><a href="interactivity-with-the-rspatial-ecosystem.html#cb443-13" tabindex="-1"></a>           <span class="at">cell_color =</span> <span class="st">&quot;leiden_clus&quot;</span>,</span>
<span id="cb443-14"><a href="interactivity-with-the-rspatial-ecosystem.html#cb443-14" tabindex="-1"></a>           <span class="at">show_image =</span> <span class="cn">TRUE</span>,</span>
<span id="cb443-15"><a href="interactivity-with-the-rspatial-ecosystem.html#cb443-15" tabindex="-1"></a>           <span class="at">point_size =</span> <span class="fl">0.1</span>,</span>
<span id="cb443-16"><a href="interactivity-with-the-rspatial-ecosystem.html#cb443-16" tabindex="-1"></a>           <span class="at">point_shape =</span> <span class="st">&quot;no_border&quot;</span>,</span>
<span id="cb443-17"><a href="interactivity-with-the-rspatial-ecosystem.html#cb443-17" tabindex="-1"></a>           <span class="at">background_color =</span> <span class="st">&quot;black&quot;</span>,</span>
<span id="cb443-18"><a href="interactivity-with-the-rspatial-ecosystem.html#cb443-18" tabindex="-1"></a>           <span class="at">show_legend =</span> <span class="cn">TRUE</span>,</span>
<span id="cb443-19"><a href="interactivity-with-the-rspatial-ecosystem.html#cb443-19" tabindex="-1"></a>           <span class="at">save_plot =</span> <span class="cn">TRUE</span>,</span>
<span id="cb443-20"><a href="interactivity-with-the-rspatial-ecosystem.html#cb443-20" tabindex="-1"></a>           <span class="at">save_param =</span> <span class="fu">list</span>(<span class="at">save_name =</span> <span class="st">&quot;03_ses6_subcell_spat&quot;</span>))</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-627"></span>
<img src="img/03_session6/03_ses6_spat_plots.png" alt="Spatial plots showing leiden clustering mapped onto the base visium spots (left) and individual nuceli through interpolation (right)" width="100%" />
<p class="caption">
Figure 17.13: Spatial plots showing leiden clustering mapped onto the base visium spots (left) and individual nuceli through interpolation (right)
</p>
</div>
</div>
<div id="cropping-objects" class="section level3 hasAnchor" number="17.9.6">
<h3><span class="header-section-number">17.9.6</span> Cropping objects<a href="interactivity-with-the-rspatial-ecosystem.html#cropping-objects" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>We are also able to crop both spat units simultaneously to zoom in on specific regions of the tissue such as seen below.</p>
<div class="sourceCode" id="cb444"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb444-1"><a href="interactivity-with-the-rspatial-ecosystem.html#cb444-1" tabindex="-1"></a>subcell_v_brain_crop <span class="ot">&lt;-</span> <span class="fu">subsetGiottoLocs</span>(<span class="at">gobject =</span> subcell_v_brain,</span>
<span id="cb444-2"><a href="interactivity-with-the-rspatial-ecosystem.html#cb444-2" tabindex="-1"></a>                                         <span class="at">spat_unit =</span> <span class="st">&quot;:all:&quot;</span>,</span>
<span id="cb444-3"><a href="interactivity-with-the-rspatial-ecosystem.html#cb444-3" tabindex="-1"></a>                                         <span class="at">x_min =</span> <span class="dv">4000</span>,</span>
<span id="cb444-4"><a href="interactivity-with-the-rspatial-ecosystem.html#cb444-4" tabindex="-1"></a>                                         <span class="at">x_max =</span> <span class="dv">7000</span>,</span>
<span id="cb444-5"><a href="interactivity-with-the-rspatial-ecosystem.html#cb444-5" tabindex="-1"></a>                                         <span class="at">y_min =</span> <span class="sc">-</span><span class="dv">6500</span>,</span>
<span id="cb444-6"><a href="interactivity-with-the-rspatial-ecosystem.html#cb444-6" tabindex="-1"></a>                                         <span class="at">y_max =</span> <span class="sc">-</span><span class="dv">3500</span>,</span>
<span id="cb444-7"><a href="interactivity-with-the-rspatial-ecosystem.html#cb444-7" tabindex="-1"></a>                                         <span class="at">z_max =</span> <span class="cn">NULL</span>,</span>
<span id="cb444-8"><a href="interactivity-with-the-rspatial-ecosystem.html#cb444-8" tabindex="-1"></a>                                         <span class="at">z_min =</span> <span class="cn">NULL</span>)</span>
<span id="cb444-9"><a href="interactivity-with-the-rspatial-ecosystem.html#cb444-9" tabindex="-1"></a></span>
<span id="cb444-10"><a href="interactivity-with-the-rspatial-ecosystem.html#cb444-10" tabindex="-1"></a><span class="fu">spatPlot2D</span>(<span class="at">gobject =</span> subcell_v_brain_crop,</span>
<span id="cb444-11"><a href="interactivity-with-the-rspatial-ecosystem.html#cb444-11" tabindex="-1"></a>           <span class="at">spat_unit =</span> <span class="st">&quot;cell&quot;</span>,</span>
<span id="cb444-12"><a href="interactivity-with-the-rspatial-ecosystem.html#cb444-12" tabindex="-1"></a>           <span class="at">cell_color =</span> <span class="st">&quot;leiden_clus&quot;</span>,</span>
<span id="cb444-13"><a href="interactivity-with-the-rspatial-ecosystem.html#cb444-13" tabindex="-1"></a>           <span class="at">show_image =</span> <span class="cn">TRUE</span>,</span>
<span id="cb444-14"><a href="interactivity-with-the-rspatial-ecosystem.html#cb444-14" tabindex="-1"></a>           <span class="at">point_size =</span> <span class="dv">2</span>,</span>
<span id="cb444-15"><a href="interactivity-with-the-rspatial-ecosystem.html#cb444-15" tabindex="-1"></a>           <span class="at">point_shape =</span> <span class="st">&quot;no_border&quot;</span>,</span>
<span id="cb444-16"><a href="interactivity-with-the-rspatial-ecosystem.html#cb444-16" tabindex="-1"></a>           <span class="at">background_color =</span> <span class="st">&quot;black&quot;</span>,</span>
<span id="cb444-17"><a href="interactivity-with-the-rspatial-ecosystem.html#cb444-17" tabindex="-1"></a>           <span class="at">show_legend =</span> <span class="cn">TRUE</span>,</span>
<span id="cb444-18"><a href="interactivity-with-the-rspatial-ecosystem.html#cb444-18" tabindex="-1"></a>           <span class="at">save_plot =</span> <span class="cn">TRUE</span>,</span>
<span id="cb444-19"><a href="interactivity-with-the-rspatial-ecosystem.html#cb444-19" tabindex="-1"></a>           <span class="at">save_param =</span> <span class="fu">list</span>(<span class="at">save_name =</span> <span class="st">&quot;03_ses6_vis_spat_crop&quot;</span>))</span>
<span id="cb444-20"><a href="interactivity-with-the-rspatial-ecosystem.html#cb444-20" tabindex="-1"></a></span>
<span id="cb444-21"><a href="interactivity-with-the-rspatial-ecosystem.html#cb444-21" tabindex="-1"></a><span class="fu">spatPlot2D</span>(<span class="at">gobject =</span> subcell_v_brain_crop,</span>
<span id="cb444-22"><a href="interactivity-with-the-rspatial-ecosystem.html#cb444-22" tabindex="-1"></a>           <span class="at">spat_unit =</span> <span class="st">&quot;stardist_cell&quot;</span>,</span>
<span id="cb444-23"><a href="interactivity-with-the-rspatial-ecosystem.html#cb444-23" tabindex="-1"></a>           <span class="at">cell_color =</span> <span class="st">&quot;leiden_clus&quot;</span>,</span>
<span id="cb444-24"><a href="interactivity-with-the-rspatial-ecosystem.html#cb444-24" tabindex="-1"></a>           <span class="at">show_image =</span> <span class="cn">TRUE</span>,</span>
<span id="cb444-25"><a href="interactivity-with-the-rspatial-ecosystem.html#cb444-25" tabindex="-1"></a>           <span class="at">point_size =</span> <span class="fl">0.1</span>,</span>
<span id="cb444-26"><a href="interactivity-with-the-rspatial-ecosystem.html#cb444-26" tabindex="-1"></a>           <span class="at">point_shape =</span> <span class="st">&quot;no_border&quot;</span>,</span>
<span id="cb444-27"><a href="interactivity-with-the-rspatial-ecosystem.html#cb444-27" tabindex="-1"></a>           <span class="at">background_color =</span> <span class="st">&quot;black&quot;</span>,</span>
<span id="cb444-28"><a href="interactivity-with-the-rspatial-ecosystem.html#cb444-28" tabindex="-1"></a>           <span class="at">show_legend =</span> <span class="cn">TRUE</span>,</span>
<span id="cb444-29"><a href="interactivity-with-the-rspatial-ecosystem.html#cb444-29" tabindex="-1"></a>           <span class="at">save_plot =</span> <span class="cn">TRUE</span>,</span>
<span id="cb444-30"><a href="interactivity-with-the-rspatial-ecosystem.html#cb444-30" tabindex="-1"></a>           <span class="at">save_param =</span> <span class="fu">list</span>(<span class="at">save_name =</span> <span class="st">&quot;03_ses6_subcell_spat_crop&quot;</span>))</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-629"></span>
<img src="img/03_session6/03_ses_spat_crop.png" alt="Spatial plots showing leiden clustering mapped onto the base visium spots (left) and individual nuceli through interpolation (right)" width="100%" />
<p class="caption">
Figure 17.14: Spatial plots showing leiden clustering mapped onto the base visium spots (left) and individual nuceli through interpolation (right)
</p>
</div>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="interoperability-with-isolated-tools.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="contributing-to-giotto.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
