<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>16 Interoperability with isolated tools | Workshop: Spatial multi-omics data analysis with Giotto Suite</title>
  <meta name="description" content="16 Interoperability with isolated tools | Workshop: Spatial multi-omics data analysis with Giotto Suite" />
  <meta name="generator" content="bookdown 0.39 and GitBook 2.6.7" />

  <meta property="og:title" content="16 Interoperability with isolated tools | Workshop: Spatial multi-omics data analysis with Giotto Suite" />
  <meta property="og:type" content="book" />
  
  
  <meta name="github-repo" content="drieslab/giotto_workshop_2024" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="16 Interoperability with isolated tools | Workshop: Spatial multi-omics data analysis with Giotto Suite" />
  
  
  

<meta name="author" content="Ruben Dries, Jiaji George Chen, Joselyn Cristina ChÃ¡vez-Fuentes, Junxiang Xu ,Edward Ruiz, Jeff Sheridan, Iqra Amin, Wen Wang" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="interoperability-with-other-frameworks.html"/>
<link rel="next" href="interactivity-with-the-rspatial-ecosystem.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Giotto Suite Workshop 2024</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#instructors"><i class="fa fa-check"></i><b>1.1</b> Instructors</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#topics-and-schedule"><i class="fa fa-check"></i><b>1.2</b> Topics and Schedule:</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#license"><i class="fa fa-check"></i><b>1.3</b> License</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="datasets-packages.html"><a href="datasets-packages.html"><i class="fa fa-check"></i><b>2</b> Datasets &amp; Packages</a>
<ul>
<li class="chapter" data-level="2.1" data-path="datasets-packages.html"><a href="datasets-packages.html#datasets-to-download"><i class="fa fa-check"></i><b>2.1</b> Datasets to download</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="datasets-packages.html"><a href="datasets-packages.html#zenodo-repository"><i class="fa fa-check"></i><b>2.1.1</b> Zenodo repository</a></li>
<li class="chapter" data-level="2.1.2" data-path="datasets-packages.html"><a href="datasets-packages.html#x-genomics-visium-mouse-brain-section-coronal-dataset"><i class="fa fa-check"></i><b>2.1.2</b> 10X Genomics Visium Mouse Brain Section (Coronal) dataset</a></li>
<li class="chapter" data-level="2.1.3" data-path="datasets-packages.html"><a href="datasets-packages.html#x-genomics-visium-hd-ffpe-human-colon-cancer"><i class="fa fa-check"></i><b>2.1.3</b> 10X Genomics Visium HD: FFPE Human Colon Cancer</a></li>
<li class="chapter" data-level="2.1.4" data-path="datasets-packages.html"><a href="datasets-packages.html#x-genomics-multi-modal-dataset"><i class="fa fa-check"></i><b>2.1.4</b> 10X Genomics multi-modal dataset</a></li>
<li class="chapter" data-level="2.1.5" data-path="datasets-packages.html"><a href="datasets-packages.html#x-genomics-multi-omics-visium-cytassist-human-tonsil-dataset"><i class="fa fa-check"></i><b>2.1.5</b> 10X Genomics multi-omics Visium CytAssist Human Tonsil dataset</a></li>
<li class="chapter" data-level="2.1.6" data-path="datasets-packages.html"><a href="datasets-packages.html#x-genomics-human-prostate-cancer-adenocarcinoma-with-invasive-carcinoma-ffpe"><i class="fa fa-check"></i><b>2.1.6</b> 10X Genomics Human Prostate Cancer Adenocarcinoma with Invasive Carcinoma (FFPE)</a></li>
<li class="chapter" data-level="2.1.7" data-path="datasets-packages.html"><a href="datasets-packages.html#x-genomics-normal-human-prostate-ffpe"><i class="fa fa-check"></i><b>2.1.7</b> 10X Genomics Normal Human Prostate (FFPE)</a></li>
<li class="chapter" data-level="2.1.8" data-path="datasets-packages.html"><a href="datasets-packages.html#xenium"><i class="fa fa-check"></i><b>2.1.8</b> Xenium</a></li>
<li class="chapter" data-level="2.1.9" data-path="datasets-packages.html"><a href="datasets-packages.html#merfish-cortex-dataset"><i class="fa fa-check"></i><b>2.1.9</b> MERFISH cortex dataset</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="datasets-packages.html"><a href="datasets-packages.html#needed-packages"><i class="fa fa-check"></i><b>2.2</b> Needed packages</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="datasets-packages.html"><a href="datasets-packages.html#cran-dependencies"><i class="fa fa-check"></i><b>2.2.1</b> CRAN dependencies:</a></li>
<li class="chapter" data-level="2.2.2" data-path="datasets-packages.html"><a href="datasets-packages.html#terra-installation"><i class="fa fa-check"></i><b>2.2.2</b> <em>terra</em> installation</a></li>
<li class="chapter" data-level="2.2.3" data-path="datasets-packages.html"><a href="datasets-packages.html#matrix-installation"><i class="fa fa-check"></i><b>2.2.3</b> <em>Matrix</em> installation</a></li>
<li class="chapter" data-level="2.2.4" data-path="datasets-packages.html"><a href="datasets-packages.html#rtools-installation"><i class="fa fa-check"></i><b>2.2.4</b> <em>Rtools</em> installation</a></li>
<li class="chapter" data-level="2.2.5" data-path="datasets-packages.html"><a href="datasets-packages.html#giotto-installation"><i class="fa fa-check"></i><b>2.2.5</b> <em>Giotto</em> installation</a></li>
<li class="chapter" data-level="2.2.6" data-path="datasets-packages.html"><a href="datasets-packages.html#irlba-install"><i class="fa fa-check"></i><b>2.2.6</b> <em>irlba</em> install</a></li>
<li class="chapter" data-level="2.2.7" data-path="datasets-packages.html"><a href="datasets-packages.html#arrow-install"><i class="fa fa-check"></i><b>2.2.7</b> <em>arrow</em> install</a></li>
<li class="chapter" data-level="2.2.8" data-path="datasets-packages.html"><a href="datasets-packages.html#bioconductor-dependencies"><i class="fa fa-check"></i><b>2.2.8</b> Bioconductor dependencies:</a></li>
<li class="chapter" data-level="2.2.9" data-path="datasets-packages.html"><a href="datasets-packages.html#cran-packages"><i class="fa fa-check"></i><b>2.2.9</b> CRAN packages:</a></li>
<li class="chapter" data-level="2.2.10" data-path="datasets-packages.html"><a href="datasets-packages.html#packages-from-github"><i class="fa fa-check"></i><b>2.2.10</b> Packages from GitHub</a></li>
<li class="chapter" data-level="2.2.11" data-path="datasets-packages.html"><a href="datasets-packages.html#python-environments"><i class="fa fa-check"></i><b>2.2.11</b> Python environments</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html"><i class="fa fa-check"></i><b>3</b> Spatial omics technologies</a>
<ul>
<li class="chapter" data-level="3.1" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#presentation"><i class="fa fa-check"></i><b>3.1</b> Presentation</a></li>
<li class="chapter" data-level="3.2" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#short-summary"><i class="fa fa-check"></i><b>3.2</b> Short summary</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#why-do-we-need-spatial-omics-technologies"><i class="fa fa-check"></i><b>3.2.1</b> Why do we need spatial omics technologies?</a></li>
<li class="chapter" data-level="3.2.2" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#what-is-spatial-omics"><i class="fa fa-check"></i><b>3.2.2</b> What is spatial omics?</a></li>
<li class="chapter" data-level="3.2.3" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#what-are-the-main-spatial-omics-technologies"><i class="fa fa-check"></i><b>3.2.3</b> What are the main spatial omics technologies?</a></li>
<li class="chapter" data-level="3.2.4" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#other-spatial-omics-atac-seq-cuttag-lipidomics-etc"><i class="fa fa-check"></i><b>3.2.4</b> Other Spatial omics: ATAC-seq, CUT&amp;Tag, lipidomics, etc</a></li>
<li class="chapter" data-level="3.2.5" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#what-are-the-different-types-of-spatial-downstream-analyses"><i class="fa fa-check"></i><b>3.2.5</b> What are the different types of spatial downstream analyses?</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html"><i class="fa fa-check"></i><b>4</b> Introduction to the Giotto package</a>
<ul>
<li class="chapter" data-level="4.1" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html#presentation-1"><i class="fa fa-check"></i><b>4.1</b> Presentation</a></li>
<li class="chapter" data-level="4.2" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html#ecosystem"><i class="fa fa-check"></i><b>4.2</b> Ecosystem</a></li>
<li class="chapter" data-level="4.3" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html#installation-python-environment"><i class="fa fa-check"></i><b>4.3</b> Installation + python environment</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html#giotto-installation-1"><i class="fa fa-check"></i><b>4.3.1</b> Giotto installation</a></li>
<li class="chapter" data-level="4.3.2" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html#python-environment"><i class="fa fa-check"></i><b>4.3.2</b> Python environment</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html#giotto-instructions"><i class="fa fa-check"></i><b>4.4</b> Giotto instructions</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html"><i class="fa fa-check"></i><b>5</b> Data formatting and Pre-processing</a>
<ul>
<li class="chapter" data-level="5.1" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#data-formats"><i class="fa fa-check"></i><b>5.1</b> Data formats</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#general-formats"><i class="fa fa-check"></i><b>5.1.1</b> General formats</a></li>
<li class="chapter" data-level="5.1.2" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#matrix-formats"><i class="fa fa-check"></i><b>5.1.2</b> Matrix formats</a></li>
<li class="chapter" data-level="5.1.3" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#tabular-formats"><i class="fa fa-check"></i><b>5.1.3</b> Tabular formats</a></li>
<li class="chapter" data-level="5.1.4" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#spatial-formats"><i class="fa fa-check"></i><b>5.1.4</b> Spatial formats</a></li>
<li class="chapter" data-level="5.1.5" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#mask-files"><i class="fa fa-check"></i><b>5.1.5</b> Mask files</a></li>
<li class="chapter" data-level="5.1.6" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#images"><i class="fa fa-check"></i><b>5.1.6</b> images</a></li>
<li class="chapter" data-level="5.1.7" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#jsonlike-formats"><i class="fa fa-check"></i><b>5.1.7</b> jsonlike formats</a></li>
<li class="chapter" data-level="5.1.8" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#hierarchical-formats"><i class="fa fa-check"></i><b>5.1.8</b> Hierarchical formats</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#pre-processing"><i class="fa fa-check"></i><b>5.2</b> Pre-processing</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#expression-matrix"><i class="fa fa-check"></i><b>5.2.1</b> Expression matrix</a></li>
<li class="chapter" data-level="5.2.2" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#spatial-locations"><i class="fa fa-check"></i><b>5.2.2</b> Spatial locations</a></li>
<li class="chapter" data-level="5.2.3" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#giottopoints"><i class="fa fa-check"></i><b>5.2.3</b> giottoPoints</a></li>
<li class="chapter" data-level="5.2.4" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#giottopolygon"><i class="fa fa-check"></i><b>5.2.4</b> giottoPolygon</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#subobject-utility-functions"><i class="fa fa-check"></i><b>5.3</b> Subobject utility functions</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#ids"><i class="fa fa-check"></i><b>5.3.1</b> IDs</a></li>
<li class="chapter" data-level="5.3.2" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#bracket-subsetting-and-extraction"><i class="fa fa-check"></i><b>5.3.2</b> Bracket subsetting and extraction</a></li>
<li class="chapter" data-level="5.3.3" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#nesting-metadata-generics"><i class="fa fa-check"></i><b>5.3.3</b> Nesting metadata generics</a></li>
<li class="chapter" data-level="5.3.4" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#appending-to-a-giotto-object"><i class="fa fa-check"></i><b>5.3.4</b> Appending to a giotto object</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html"><i class="fa fa-check"></i><b>6</b> Creating a Giotto object</a>
<ul>
<li class="chapter" data-level="6.0.1" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#used-packages"><i class="fa fa-check"></i><b>6.0.1</b> Used packages</a></li>
<li class="chapter" data-level="6.1" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#overview"><i class="fa fa-check"></i><b>6.1</b> Overview</a></li>
<li class="chapter" data-level="6.2" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#from-matrix-locations"><i class="fa fa-check"></i><b>6.2</b> From matrix + locations</a></li>
<li class="chapter" data-level="6.3" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#from-subcellular-raw-data-transcripts-or-images-polygons"><i class="fa fa-check"></i><b>6.3</b> From subcellular raw data (transcripts or images) + polygons</a></li>
<li class="chapter" data-level="6.4" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#from-piece-wise"><i class="fa fa-check"></i><b>6.4</b> From piece-wise</a></li>
<li class="chapter" data-level="6.5" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#using-convenience-functions-for-popular-technologies-vizgen-xenium-cosmx"><i class="fa fa-check"></i><b>6.5</b> Using convenience functions for popular technologies (Vizgen, Xenium, CosMx, â¦)</a></li>
<li class="chapter" data-level="6.6" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#spatial-plots"><i class="fa fa-check"></i><b>6.6</b> Spatial plots</a></li>
<li class="chapter" data-level="6.7" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#subsetting"><i class="fa fa-check"></i><b>6.7</b> Subsetting</a></li>
<li class="chapter" data-level="6.8" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#mini-objects-giottodata"><i class="fa fa-check"></i><b>6.8</b> Mini objects &amp; GiottoData</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="visium-part-i.html"><a href="visium-part-i.html"><i class="fa fa-check"></i><b>7</b> Visium Part I</a>
<ul>
<li class="chapter" data-level="7.1" data-path="visium-part-i.html"><a href="visium-part-i.html#the-visium-technology"><i class="fa fa-check"></i><b>7.1</b> The Visium technology</a></li>
<li class="chapter" data-level="7.2" data-path="visium-part-i.html"><a href="visium-part-i.html#introduction-to-the-spatial-dataset"><i class="fa fa-check"></i><b>7.2</b> Introduction to the spatial dataset</a></li>
<li class="chapter" data-level="7.3" data-path="visium-part-i.html"><a href="visium-part-i.html#download-dataset"><i class="fa fa-check"></i><b>7.3</b> Download dataset</a></li>
<li class="chapter" data-level="7.4" data-path="visium-part-i.html"><a href="visium-part-i.html#create-the-giotto-object"><i class="fa fa-check"></i><b>7.4</b> Create the Giotto object</a></li>
<li class="chapter" data-level="7.5" data-path="visium-part-i.html"><a href="visium-part-i.html#subset-on-spots-that-were-covered-by-tissue"><i class="fa fa-check"></i><b>7.5</b> Subset on spots that were covered by tissue</a></li>
<li class="chapter" data-level="7.6" data-path="visium-part-i.html"><a href="visium-part-i.html#quality-control"><i class="fa fa-check"></i><b>7.6</b> Quality control</a></li>
<li class="chapter" data-level="7.7" data-path="visium-part-i.html"><a href="visium-part-i.html#filtering"><i class="fa fa-check"></i><b>7.7</b> Filtering</a></li>
<li class="chapter" data-level="7.8" data-path="visium-part-i.html"><a href="visium-part-i.html#normalization"><i class="fa fa-check"></i><b>7.8</b> Normalization</a></li>
<li class="chapter" data-level="7.9" data-path="visium-part-i.html"><a href="visium-part-i.html#feature-selection"><i class="fa fa-check"></i><b>7.9</b> Feature selection</a>
<ul>
<li class="chapter" data-level="7.9.1" data-path="visium-part-i.html"><a href="visium-part-i.html#highly-variable-features"><i class="fa fa-check"></i><b>7.9.1</b> Highly Variable Features:</a></li>
</ul></li>
<li class="chapter" data-level="7.10" data-path="visium-part-i.html"><a href="visium-part-i.html#dimension-reduction"><i class="fa fa-check"></i><b>7.10</b> Dimension Reduction</a>
<ul>
<li class="chapter" data-level="7.10.1" data-path="visium-part-i.html"><a href="visium-part-i.html#pca"><i class="fa fa-check"></i><b>7.10.1</b> PCA</a></li>
<li class="chapter" data-level="7.10.2" data-path="visium-part-i.html"><a href="visium-part-i.html#umap"><i class="fa fa-check"></i><b>7.10.2</b> UMAP</a></li>
<li class="chapter" data-level="7.10.3" data-path="visium-part-i.html"><a href="visium-part-i.html#t-sne"><i class="fa fa-check"></i><b>7.10.3</b> t-SNE</a></li>
</ul></li>
<li class="chapter" data-level="7.11" data-path="visium-part-i.html"><a href="visium-part-i.html#clustering"><i class="fa fa-check"></i><b>7.11</b> Clustering</a>
<ul>
<li class="chapter" data-level="7.11.1" data-path="visium-part-i.html"><a href="visium-part-i.html#calculate-leiden-clustering"><i class="fa fa-check"></i><b>7.11.1</b> Calculate Leiden clustering</a></li>
<li class="chapter" data-level="7.11.2" data-path="visium-part-i.html"><a href="visium-part-i.html#calculate-louvain-clustering"><i class="fa fa-check"></i><b>7.11.2</b> Calculate Louvain clustering</a></li>
</ul></li>
<li class="chapter" data-level="7.12" data-path="visium-part-i.html"><a href="visium-part-i.html#save-the-object"><i class="fa fa-check"></i><b>7.12</b> Save the object</a></li>
<li class="chapter" data-level="7.13" data-path="visium-part-i.html"><a href="visium-part-i.html#session-info"><i class="fa fa-check"></i><b>7.13</b> Session info</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="visium-part-ii.html"><a href="visium-part-ii.html"><i class="fa fa-check"></i><b>8</b> Visium Part II</a>
<ul>
<li class="chapter" data-level="8.1" data-path="visium-part-ii.html"><a href="visium-part-ii.html#load-the-object"><i class="fa fa-check"></i><b>8.1</b> Load the object</a></li>
<li class="chapter" data-level="8.2" data-path="visium-part-ii.html"><a href="visium-part-ii.html#differential-expression"><i class="fa fa-check"></i><b>8.2</b> Differential expression</a>
<ul>
<li class="chapter" data-level="8.2.1" data-path="visium-part-ii.html"><a href="visium-part-ii.html#gini-markers"><i class="fa fa-check"></i><b>8.2.1</b> Gini markers</a></li>
<li class="chapter" data-level="8.2.2" data-path="visium-part-ii.html"><a href="visium-part-ii.html#scran-markers"><i class="fa fa-check"></i><b>8.2.2</b> Scran markers</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="visium-part-ii.html"><a href="visium-part-ii.html#enrichment-deconvolution"><i class="fa fa-check"></i><b>8.3</b> Enrichment &amp; Deconvolution</a>
<ul>
<li class="chapter" data-level="8.3.1" data-path="visium-part-ii.html"><a href="visium-part-ii.html#pagerank"><i class="fa fa-check"></i><b>8.3.1</b> PAGE/Rank</a></li>
<li class="chapter" data-level="8.3.2" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatialdwls"><i class="fa fa-check"></i><b>8.3.2</b> SpatialDWLS</a></li>
</ul></li>
<li class="chapter" data-level="8.4" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatial-expression-patterns"><i class="fa fa-check"></i><b>8.4</b> Spatial expression patterns</a>
<ul>
<li class="chapter" data-level="8.4.1" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatial-variable-genes"><i class="fa fa-check"></i><b>8.4.1</b> Spatial variable genes</a></li>
<li class="chapter" data-level="8.4.2" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatial-co-expression-modules"><i class="fa fa-check"></i><b>8.4.2</b> Spatial co-expression modules</a></li>
</ul></li>
<li class="chapter" data-level="8.5" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatially-informed-clusters"><i class="fa fa-check"></i><b>8.5</b> Spatially informed clusters</a></li>
<li class="chapter" data-level="8.6" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatial-domains-hmrf"><i class="fa fa-check"></i><b>8.6</b> Spatial domains HMRF</a></li>
<li class="chapter" data-level="8.7" data-path="visium-part-ii.html"><a href="visium-part-ii.html#interactive-tools"><i class="fa fa-check"></i><b>8.7</b> Interactive tools</a></li>
<li class="chapter" data-level="8.8" data-path="visium-part-ii.html"><a href="visium-part-ii.html#session-info-1"><i class="fa fa-check"></i><b>8.8</b> Session info</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="visium-hd.html"><a href="visium-hd.html"><i class="fa fa-check"></i><b>9</b> Visium HD</a>
<ul>
<li class="chapter" data-level="9.1" data-path="visium-hd.html"><a href="visium-hd.html#objective"><i class="fa fa-check"></i><b>9.1</b> Objective</a></li>
<li class="chapter" data-level="9.2" data-path="visium-hd.html"><a href="visium-hd.html#background"><i class="fa fa-check"></i><b>9.2</b> Background</a>
<ul>
<li class="chapter" data-level="9.2.1" data-path="visium-hd.html"><a href="visium-hd.html#visium-hd-technology"><i class="fa fa-check"></i><b>9.2.1</b> Visium HD Technology</a></li>
<li class="chapter" data-level="9.2.2" data-path="visium-hd.html"><a href="visium-hd.html#colorectal-cancer-sample"><i class="fa fa-check"></i><b>9.2.2</b> Colorectal Cancer Sample</a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="visium-hd.html"><a href="visium-hd.html#data-ingestion"><i class="fa fa-check"></i><b>9.3</b> Data Ingestion</a>
<ul>
<li class="chapter" data-level="9.3.1" data-path="visium-hd.html"><a href="visium-hd.html#visium-hd-output-data-format"><i class="fa fa-check"></i><b>9.3.1</b> Visium HD output data format</a></li>
<li class="chapter" data-level="9.3.2" data-path="visium-hd.html"><a href="visium-hd.html#mini-visium-hd-dataset"><i class="fa fa-check"></i><b>9.3.2</b> Mini Visium HD dataset</a></li>
<li class="chapter" data-level="9.3.3" data-path="visium-hd.html"><a href="visium-hd.html#giotto-visium-hd-convenience-function"><i class="fa fa-check"></i><b>9.3.3</b> Giotto Visium HD convenience function</a></li>
<li class="chapter" data-level="9.3.4" data-path="visium-hd.html"><a href="visium-hd.html#read-in-data-manually"><i class="fa fa-check"></i><b>9.3.4</b> Read in data manually</a></li>
</ul></li>
<li class="chapter" data-level="9.4" data-path="visium-hd.html"><a href="visium-hd.html#create-giotto-object"><i class="fa fa-check"></i><b>9.4</b> Create Giotto object</a>
<ul>
<li class="chapter" data-level="9.4.1" data-path="visium-hd.html"><a href="visium-hd.html#create-giotto-points"><i class="fa fa-check"></i><b>9.4.1</b> create giotto points</a></li>
<li class="chapter" data-level="9.4.2" data-path="visium-hd.html"><a href="visium-hd.html#create-giotto-polygons"><i class="fa fa-check"></i><b>9.4.2</b> create giotto polygons</a></li>
<li class="chapter" data-level="9.4.3" data-path="visium-hd.html"><a href="visium-hd.html#combine-giotto-points-and-polygons-to-create-giotto-object"><i class="fa fa-check"></i><b>9.4.3</b> combine Giotto points and polygons to create Giotto object</a></li>
</ul></li>
<li class="chapter" data-level="9.5" data-path="visium-hd.html"><a href="visium-hd.html#process-giotto-object"><i class="fa fa-check"></i><b>9.5</b> Process Giotto object</a>
<ul>
<li class="chapter" data-level="9.5.1" data-path="visium-hd.html"><a href="visium-hd.html#calculate-overlap-between-points-and-polygons"><i class="fa fa-check"></i><b>9.5.1</b> calculate overlap between points and polygons</a></li>
<li class="chapter" data-level="9.5.2" data-path="visium-hd.html"><a href="visium-hd.html#convert-overlap-results-to-a-bin-by-gene-matrix"><i class="fa fa-check"></i><b>9.5.2</b> convert overlap results to a bin-by-gene matrix</a></li>
<li class="chapter" data-level="9.5.3" data-path="visium-hd.html"><a href="visium-hd.html#default-processing-steps"><i class="fa fa-check"></i><b>9.5.3</b> default processing steps</a></li>
<li class="chapter" data-level="9.5.4" data-path="visium-hd.html"><a href="visium-hd.html#visualize-number-of-features"><i class="fa fa-check"></i><b>9.5.4</b> visualize number of features</a></li>
</ul></li>
<li class="chapter" data-level="9.6" data-path="visium-hd.html"><a href="visium-hd.html#dimension-reduction-clustering"><i class="fa fa-check"></i><b>9.6</b> Dimension reduction + clustering</a>
<ul>
<li class="chapter" data-level="9.6.1" data-path="visium-hd.html"><a href="visium-hd.html#highly-variable-features-pca"><i class="fa fa-check"></i><b>9.6.1</b> Highly variable features + PCA</a></li>
<li class="chapter" data-level="9.6.2" data-path="visium-hd.html"><a href="visium-hd.html#umap-reduction-for-visualization"><i class="fa fa-check"></i><b>9.6.2</b> UMAP reduction for visualization</a></li>
<li class="chapter" data-level="9.6.3" data-path="visium-hd.html"><a href="visium-hd.html#create-network-based-on-expression-similarity-graph-partition-cluster"><i class="fa fa-check"></i><b>9.6.3</b> Create network based on expression similarity + graph partition cluster</a></li>
</ul></li>
<li class="chapter" data-level="9.7" data-path="visium-hd.html"><a href="visium-hd.html#hexbin-100"><i class="fa fa-check"></i><b>9.7</b> Hexbin 100</a>
<ul>
<li class="chapter" data-level="9.7.1" data-path="visium-hd.html"><a href="visium-hd.html#standard-subcellular-pipeline"><i class="fa fa-check"></i><b>9.7.1</b> Standard subcellular pipeline</a></li>
<li class="chapter" data-level="9.7.2" data-path="visium-hd.html"><a href="visium-hd.html#spatial-expression-patterns-1"><i class="fa fa-check"></i><b>9.7.2</b> Spatial expression patterns</a></li>
</ul></li>
<li class="chapter" data-level="9.8" data-path="visium-hd.html"><a href="visium-hd.html#hexbin-25"><i class="fa fa-check"></i><b>9.8</b> Hexbin 25</a>
<ul>
<li class="chapter" data-level="9.8.1" data-path="visium-hd.html"><a href="visium-hd.html#subcellular-workflow-and-projection-functions"><i class="fa fa-check"></i><b>9.8.1</b> Subcellular workflow and projection functions</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="xenium-1.html"><a href="xenium-1.html"><i class="fa fa-check"></i><b>10</b> Xenium</a>
<ul>
<li class="chapter" data-level="10.1" data-path="xenium-1.html"><a href="xenium-1.html#introduction-to-spatial-dataset"><i class="fa fa-check"></i><b>10.1</b> Introduction to spatial dataset</a>
<ul>
<li class="chapter" data-level="10.1.1" data-path="xenium-1.html"><a href="xenium-1.html#output-directory-structure"><i class="fa fa-check"></i><b>10.1.1</b> Output directory structure</a></li>
<li class="chapter" data-level="10.1.2" data-path="xenium-1.html"><a href="xenium-1.html#mini-xenium-dataset"><i class="fa fa-check"></i><b>10.1.2</b> Mini Xenium Dataset</a></li>
</ul></li>
<li class="chapter" data-level="10.2" data-path="xenium-1.html"><a href="xenium-1.html#data-prep"><i class="fa fa-check"></i><b>10.2</b> Data prep</a>
<ul>
<li class="chapter" data-level="10.2.1" data-path="xenium-1.html"><a href="xenium-1.html#image-conversion-may-change"><i class="fa fa-check"></i><b>10.2.1</b> Image conversion (may change)</a></li>
</ul></li>
<li class="chapter" data-level="10.3" data-path="xenium-1.html"><a href="xenium-1.html#convenience-function"><i class="fa fa-check"></i><b>10.3</b> Convenience function</a>
<ul>
<li class="chapter" data-level="10.3.1" data-path="xenium-1.html"><a href="xenium-1.html#specific-filepaths"><i class="fa fa-check"></i><b>10.3.1</b> Specific filepaths</a></li>
<li class="chapter" data-level="10.3.2" data-path="xenium-1.html"><a href="xenium-1.html#quality-value"><i class="fa fa-check"></i><b>10.3.2</b> Quality value</a></li>
<li class="chapter" data-level="10.3.3" data-path="xenium-1.html"><a href="xenium-1.html#transcript-type-splitting"><i class="fa fa-check"></i><b>10.3.3</b> Transcript type splitting</a></li>
<li class="chapter" data-level="10.3.4" data-path="xenium-1.html"><a href="xenium-1.html#centroids-calculation"><i class="fa fa-check"></i><b>10.3.4</b> Centroids calculation</a></li>
<li class="chapter" data-level="10.3.5" data-path="xenium-1.html"><a href="xenium-1.html#simple-visualization"><i class="fa fa-check"></i><b>10.3.5</b> Simple visualization</a></li>
</ul></li>
<li class="chapter" data-level="10.4" data-path="xenium-1.html"><a href="xenium-1.html#piecewise-loading"><i class="fa fa-check"></i><b>10.4</b> Piecewise loading</a>
<ul>
<li class="chapter" data-level="10.4.1" data-path="xenium-1.html"><a href="xenium-1.html#load-giottopoints-transcripts"><i class="fa fa-check"></i><b>10.4.1</b> load giottoPoints transcripts</a></li>
<li class="chapter" data-level="10.4.2" data-path="xenium-1.html"><a href="xenium-1.html#optional-loading-pre-aggregated-data"><i class="fa fa-check"></i><b>10.4.2</b> (optional) Loading pre-aggregated data</a></li>
</ul></li>
<li class="chapter" data-level="10.5" data-path="xenium-1.html"><a href="xenium-1.html#xenium-images"><i class="fa fa-check"></i><b>10.5</b> Xenium Images</a>
<ul>
<li class="chapter" data-level="10.5.1" data-path="xenium-1.html"><a href="xenium-1.html#image-metadata"><i class="fa fa-check"></i><b>10.5.1</b> Image metadata</a></li>
<li class="chapter" data-level="10.5.2" data-path="xenium-1.html"><a href="xenium-1.html#image-loading"><i class="fa fa-check"></i><b>10.5.2</b> Image loading</a></li>
</ul></li>
<li class="chapter" data-level="10.6" data-path="xenium-1.html"><a href="xenium-1.html#spatial-aggregation"><i class="fa fa-check"></i><b>10.6</b> Spatial aggregation</a></li>
<li class="chapter" data-level="10.7" data-path="xenium-1.html"><a href="xenium-1.html#aggregate-analyses-workflow"><i class="fa fa-check"></i><b>10.7</b> Aggregate analyses workflow</a>
<ul>
<li class="chapter" data-level="10.7.1" data-path="xenium-1.html"><a href="xenium-1.html#filtering-1"><i class="fa fa-check"></i><b>10.7.1</b> Filtering</a></li>
<li class="chapter" data-level="10.7.2" data-path="xenium-1.html"><a href="xenium-1.html#normalization-1"><i class="fa fa-check"></i><b>10.7.2</b> Normalization</a></li>
<li class="chapter" data-level="10.7.3" data-path="xenium-1.html"><a href="xenium-1.html#dimension-reduction-1"><i class="fa fa-check"></i><b>10.7.3</b> Dimension Reduction</a></li>
<li class="chapter" data-level="10.7.4" data-path="xenium-1.html"><a href="xenium-1.html#clustering-1"><i class="fa fa-check"></i><b>10.7.4</b> Clustering</a></li>
</ul></li>
<li class="chapter" data-level="10.8" data-path="xenium-1.html"><a href="xenium-1.html#niche-clustering"><i class="fa fa-check"></i><b>10.8</b> Niche clustering</a>
<ul>
<li class="chapter" data-level="10.8.1" data-path="xenium-1.html"><a href="xenium-1.html#spatial-network"><i class="fa fa-check"></i><b>10.8.1</b> Spatial network</a></li>
<li class="chapter" data-level="10.8.2" data-path="xenium-1.html"><a href="xenium-1.html#niche-calculation"><i class="fa fa-check"></i><b>10.8.2</b> Niche calculation</a></li>
<li class="chapter" data-level="10.8.3" data-path="xenium-1.html"><a href="xenium-1.html#kmeans-clustering-based-on-niche-signature"><i class="fa fa-check"></i><b>10.8.3</b> kmeans clustering based on niche signature</a></li>
</ul></li>
<li class="chapter" data-level="10.9" data-path="xenium-1.html"><a href="xenium-1.html#cell-proximity-enrichment"><i class="fa fa-check"></i><b>10.9</b> Cell proximity enrichment</a></li>
<li class="chapter" data-level="10.10" data-path="xenium-1.html"><a href="xenium-1.html#pseudovisium"><i class="fa fa-check"></i><b>10.10</b> Pseudovisium</a>
<ul>
<li class="chapter" data-level="10.10.1" data-path="xenium-1.html"><a href="xenium-1.html#pseudovisium-aggregation-and-workflow"><i class="fa fa-check"></i><b>10.10.1</b> Pseudovisium aggregation and workflow</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html"><i class="fa fa-check"></i><b>11</b> Spatial proteomics: Multiplexed Immunofluorescence</a>
<ul>
<li class="chapter" data-level="11.1" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#spatial-proteomics-technologies"><i class="fa fa-check"></i><b>11.1</b> Spatial Proteomics Technologies</a></li>
<li class="chapter" data-level="11.2" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#raw-data-type-coming-out-of-different-technologies"><i class="fa fa-check"></i><b>11.2</b> Raw data type coming out of different technologies</a>
<ul>
<li class="chapter" data-level="11.2.1" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#use-ome.tiff-as-an-example-output-data-to-begin-with"><i class="fa fa-check"></i><b>11.2.1</b> Use ome.tiff as an example output data to begin with</a></li>
<li class="chapter" data-level="11.2.2" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#use-single-channel-images-as-an-example-output-data-to-begin-with"><i class="fa fa-check"></i><b>11.2.2</b> Use single channel images as an example output data to begin with</a></li>
</ul></li>
<li class="chapter" data-level="11.3" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#cell-segmentation-file-to-get-single-cell-level-protein-expression"><i class="fa fa-check"></i><b>11.3</b> Cell Segmentation file to get single cell level protein expression</a>
<ul>
<li class="chapter" data-level="11.3.1" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#using-segmentation-output-file-from-deepcellmesmer-as-an-example."><i class="fa fa-check"></i><b>11.3.1</b> Using segmentation output file from DeepCell(mesmer) as an example.</a></li>
<li class="chapter" data-level="11.3.2" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#using-giotto-wrapper-of-cellpose-to-perform-segmentation"><i class="fa fa-check"></i><b>11.3.2</b> Using Giotto wrapper of Cellpose to perform segmentation</a></li>
</ul></li>
<li class="chapter" data-level="11.4" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#create-a-giotto-object-using-list-of-gitto-large-images-and-polygons"><i class="fa fa-check"></i><b>11.4</b> Create a Giotto Object using list of gitto large images and polygons</a>
<ul>
<li class="chapter" data-level="11.4.1" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#overlap-to-matrix"><i class="fa fa-check"></i><b>11.4.1</b> Overlap to matrix</a></li>
<li class="chapter" data-level="11.4.2" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#manipulate-expression-information"><i class="fa fa-check"></i><b>11.4.2</b> Manipulate Expression information</a></li>
<li class="chapter" data-level="11.4.3" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#rescale-polygons"><i class="fa fa-check"></i><b>11.4.3</b> Rescale polygons</a></li>
<li class="chapter" data-level="11.4.4" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#perform-clustering-and-differential-expression"><i class="fa fa-check"></i><b>11.4.4</b> Perform clustering and differential expression</a></li>
<li class="chapter" data-level="11.4.5" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#give-the-cluster-an-annotation-based-on-expression-values"><i class="fa fa-check"></i><b>11.4.5</b> Give the cluster an annotation based on expression values</a></li>
<li class="chapter" data-level="11.4.6" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#spatial-network-1"><i class="fa fa-check"></i><b>11.4.6</b> spatial network</a></li>
<li class="chapter" data-level="11.4.7" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#cell-neighborhood-cell-typecell-type-interactions"><i class="fa fa-check"></i><b>11.4.7</b> Cell Neighborhood: Cell-Type/Cell-Type Interactions</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="12" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html"><i class="fa fa-check"></i><b>12</b> Working with multiple samples</a>
<ul>
<li class="chapter" data-level="12.1" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#objective-1"><i class="fa fa-check"></i><b>12.1</b> Objective</a></li>
<li class="chapter" data-level="12.2" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#background-1"><i class="fa fa-check"></i><b>12.2</b> Background</a>
<ul>
<li class="chapter" data-level="12.2.1" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#dataset"><i class="fa fa-check"></i><b>12.2.1</b> Dataset</a></li>
<li class="chapter" data-level="12.2.2" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#visium-technology"><i class="fa fa-check"></i><b>12.2.2</b> Visium technology</a></li>
</ul></li>
<li class="chapter" data-level="12.3" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#create-individual-giotto-objects"><i class="fa fa-check"></i><b>12.3</b> Create individual giotto objects</a>
<ul>
<li class="chapter" data-level="12.3.1" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#download-the-data"><i class="fa fa-check"></i><b>12.3.1</b> Download the data</a></li>
</ul></li>
<li class="chapter" data-level="12.4" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#extracting-the-downloaded-files"><i class="fa fa-check"></i><b>12.4</b> Extracting the downloaded files</a>
<ul>
<li class="chapter" data-level="12.4.1" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#create-giotto-instructions"><i class="fa fa-check"></i><b>12.4.1</b> Create giotto instructions</a></li>
<li class="chapter" data-level="12.4.2" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#load-visium-data-into-giotto"><i class="fa fa-check"></i><b>12.4.2</b> Load visium data into Giotto</a></li>
<li class="chapter" data-level="12.4.3" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#healthy-prostate-tissue-coverage"><i class="fa fa-check"></i><b>12.4.3</b> Healthy prostate tissue coverage</a></li>
<li class="chapter" data-level="12.4.4" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#adenocarcinoma-prostate-tissue-coverage"><i class="fa fa-check"></i><b>12.4.4</b> Adenocarcinoma prostate tissue coverage</a></li>
<li class="chapter" data-level="12.4.5" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#showing-the-data-strucutre-for-the-inidividual-objects"><i class="fa fa-check"></i><b>12.4.5</b> Showing the data strucutre for the inidividual objects</a></li>
</ul></li>
<li class="chapter" data-level="12.5" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#join-giotto-objects"><i class="fa fa-check"></i><b>12.5</b> Join Giotto Objects</a></li>
<li class="chapter" data-level="12.6" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#visualizing-combined-datasets"><i class="fa fa-check"></i><b>12.6</b> Visualizing combined datasets</a>
<ul>
<li class="chapter" data-level="12.6.1" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#vizualizing-in-the-same-plot"><i class="fa fa-check"></i><b>12.6.1</b> Vizualizing in the same plot</a></li>
<li class="chapter" data-level="12.6.2" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#vizualizing-on-separate-plots"><i class="fa fa-check"></i><b>12.6.2</b> Vizualizing on separate plots</a></li>
</ul></li>
<li class="chapter" data-level="12.7" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#splitting-combined-dataset"><i class="fa fa-check"></i><b>12.7</b> Splitting combined dataset</a></li>
<li class="chapter" data-level="12.8" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#analyzing-joined-objects"><i class="fa fa-check"></i><b>12.8</b> Analyzing joined objects</a>
<ul>
<li class="chapter" data-level="12.8.1" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#normalization-and-adding-statistics"><i class="fa fa-check"></i><b>12.8.1</b> Normalization and adding statistics</a></li>
<li class="chapter" data-level="12.8.2" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#clustering-the-datasets"><i class="fa fa-check"></i><b>12.8.2</b> Clustering the datasets</a></li>
<li class="chapter" data-level="12.8.3" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#vizualizing-spatial-location-of-clusters"><i class="fa fa-check"></i><b>12.8.3</b> Vizualizing spatial location of clusters</a></li>
<li class="chapter" data-level="12.8.4" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#vizualizing-tissue-contribution-to-clusters"><i class="fa fa-check"></i><b>12.8.4</b> Vizualizing tissue contribution to clusters</a></li>
</ul></li>
<li class="chapter" data-level="12.9" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#perform-harmony-and-default-workflows"><i class="fa fa-check"></i><b>12.9</b> Perform Harmony and default workflows</a>
<ul>
<li class="chapter" data-level="12.9.1" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#clustering-harmonized-object"><i class="fa fa-check"></i><b>12.9.1</b> Clustering harmonized object</a></li>
<li class="chapter" data-level="12.9.2" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#vizualizing-the-tissue-contribution"><i class="fa fa-check"></i><b>12.9.2</b> Vizualizing the tissue contribution</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="13" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html"><i class="fa fa-check"></i><b>13</b> Spatial multi-modal analysis</a>
<ul>
<li class="chapter" data-level="13.1" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#overview-1"><i class="fa fa-check"></i><b>13.1</b> Overview</a></li>
<li class="chapter" data-level="13.2" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#spatial-manipulation"><i class="fa fa-check"></i><b>13.2</b> Spatial manipulation</a>
<ul>
<li class="chapter" data-level="13.2.1" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#spatial-transforms"><i class="fa fa-check"></i><b>13.2.1</b> Spatial transforms:</a></li>
<li class="chapter" data-level="13.2.2" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#spatial-utilities"><i class="fa fa-check"></i><b>13.2.2</b> Spatial utilities:</a></li>
<li class="chapter" data-level="13.2.3" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#spatial-classes"><i class="fa fa-check"></i><b>13.2.3</b> Spatial classes:</a></li>
</ul></li>
<li class="chapter" data-level="13.3" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#examples-of-the-simple-transforms-with-a-giottopolygon"><i class="fa fa-check"></i><b>13.3</b> Examples of the simple transforms with a giottoPolygon</a></li>
<li class="chapter" data-level="13.4" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#affine-transforms"><i class="fa fa-check"></i><b>13.4</b> Affine transforms</a></li>
<li class="chapter" data-level="13.5" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#image-transforms"><i class="fa fa-check"></i><b>13.5</b> Image transforms</a></li>
<li class="chapter" data-level="13.6" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#the-practical-usage-of-multi-modality-co-registration"><i class="fa fa-check"></i><b>13.6</b> The practical usage of multi-modality co-registration</a>
<ul>
<li class="chapter" data-level="13.6.1" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#example-dataset-xenium-breast-cancer-pre-release-pack"><i class="fa fa-check"></i><b>13.6.1</b> Example dataset: Xenium Breast Cancer pre-release pack</a></li>
<li class="chapter" data-level="13.6.2" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#target-coordinate-system"><i class="fa fa-check"></i><b>13.6.2</b> Target Coordinate system</a></li>
<li class="chapter" data-level="13.6.3" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#visium-to-register"><i class="fa fa-check"></i><b>13.6.3</b> Visium to register</a></li>
<li class="chapter" data-level="13.6.4" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#register-post-xenium-he-and-if-image"><i class="fa fa-check"></i><b>13.6.4</b> Register post-Xenium H&amp;E and IF image</a></li>
<li class="chapter" data-level="13.6.5" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#automatic-alignment-via-sift-feature-descriptor-matching-and-affine-transformation"><i class="fa fa-check"></i><b>13.6.5</b> Automatic alignment via SIFT feature descriptor matching and affine transformation</a></li>
<li class="chapter" data-level="13.6.6" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#final-notes"><i class="fa fa-check"></i><b>13.6.6</b> Final Notes</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="14" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html"><i class="fa fa-check"></i><b>14</b> Multi-omics integration</a>
<ul>
<li class="chapter" data-level="14.1" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#the-cytassist-technology"><i class="fa fa-check"></i><b>14.1</b> The CytAssist technology</a></li>
<li class="chapter" data-level="14.2" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#introduction-to-the-spatial-dataset-1"><i class="fa fa-check"></i><b>14.2</b> Introduction to the spatial dataset</a></li>
<li class="chapter" data-level="14.3" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#download-dataset-1"><i class="fa fa-check"></i><b>14.3</b> Download dataset</a></li>
<li class="chapter" data-level="14.4" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#create-the-giotto-object-1"><i class="fa fa-check"></i><b>14.4</b> Create the Giotto object</a></li>
<li class="chapter" data-level="14.5" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#subset-on-spots-that-were-covered-by-tissue-1"><i class="fa fa-check"></i><b>14.5</b> Subset on spots that were covered by tissue</a></li>
<li class="chapter" data-level="14.6" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#rna-processing"><i class="fa fa-check"></i><b>14.6</b> RNA processing</a></li>
<li class="chapter" data-level="14.7" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#protein-processing"><i class="fa fa-check"></i><b>14.7</b> Protein processing</a></li>
<li class="chapter" data-level="14.8" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#multi-omics-integration-1"><i class="fa fa-check"></i><b>14.8</b> Multi-omics integration</a></li>
<li class="chapter" data-level="14.9" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#session-info-2"><i class="fa fa-check"></i><b>14.9</b> Session info</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html"><i class="fa fa-check"></i><b>15</b> Interoperability with other frameworks</a>
<ul>
<li class="chapter" data-level="15.1" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#load-giotto-object"><i class="fa fa-check"></i><b>15.1</b> Load Giotto object</a></li>
<li class="chapter" data-level="15.2" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#seurat"><i class="fa fa-check"></i><b>15.2</b> Seurat</a>
<ul>
<li class="chapter" data-level="15.2.1" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#conversion-of-giotto-object-to-seurat-object"><i class="fa fa-check"></i><b>15.2.1</b> Conversion of Giotto Object to Seurat Object</a></li>
<li class="chapter" data-level="15.2.2" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#conversion-of-seurat-object-back-to-giotto-object"><i class="fa fa-check"></i><b>15.2.2</b> Conversion of Seurat object Back to Giotto Object</a></li>
</ul></li>
<li class="chapter" data-level="15.3" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#spatialexperiment"><i class="fa fa-check"></i><b>15.3</b> SpatialExperiment</a>
<ul>
<li class="chapter" data-level="15.3.1" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#convert-giotto-object-to-spatialexperiment-object"><i class="fa fa-check"></i><b>15.3.1</b> Convert Giotto Object to SpatialExperiment Object</a></li>
<li class="chapter" data-level="15.3.2" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#conversion-of-spatialexperiment-object-back-to-giotto"><i class="fa fa-check"></i><b>15.3.2</b> Conversion of SpatialExperiment object back to Giotto</a></li>
</ul></li>
<li class="chapter" data-level="15.4" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#anndata"><i class="fa fa-check"></i><b>15.4</b> AnnData</a>
<ul>
<li class="chapter" data-level="15.4.1" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#load-required-libraries"><i class="fa fa-check"></i><b>15.4.1</b> Load Required Libraries</a></li>
<li class="chapter" data-level="15.4.2" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#specify-path-for-results"><i class="fa fa-check"></i><b>15.4.2</b> Specify Path for Results</a></li>
<li class="chapter" data-level="15.4.3" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#giotto-to-anndata"><i class="fa fa-check"></i><b>15.4.3</b> Giotto To AnnData</a></li>
<li class="chapter" data-level="15.4.4" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#convert-anndata-to-giotto"><i class="fa fa-check"></i><b>15.4.4</b> Convert AnnData to Giotto</a></li>
</ul></li>
<li class="chapter" data-level="15.5" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#create-mini-vizgen-object"><i class="fa fa-check"></i><b>15.5</b> Create mini Vizgen object</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html"><i class="fa fa-check"></i><b>16</b> Interoperability with isolated tools</a>
<ul>
<li class="chapter" data-level="16.1" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#spatial-niche-trajectory-analysis-ontrac"><i class="fa fa-check"></i><b>16.1</b> Spatial niche trajectory analysis (ONTraC)</a>
<ul>
<li class="chapter" data-level="16.1.1" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#introduction-to-ontrac"><i class="fa fa-check"></i><b>16.1.1</b> Introduction to ONTraC</a></li>
<li class="chapter" data-level="16.1.2" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#introduction-to-merfish"><i class="fa fa-check"></i><b>16.1.2</b> Introduction to MERFISH</a></li>
<li class="chapter" data-level="16.1.3" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#settings"><i class="fa fa-check"></i><b>16.1.3</b> settings</a></li>
<li class="chapter" data-level="16.1.4" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#dataset-1"><i class="fa fa-check"></i><b>16.1.4</b> Dataset</a></li>
<li class="chapter" data-level="16.1.5" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#create-the-giotto-object-2"><i class="fa fa-check"></i><b>16.1.5</b> Create the Giotto object</a></li>
<li class="chapter" data-level="16.1.6" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#perform-spatial-niche-trajectory-analysis-using-ontrac"><i class="fa fa-check"></i><b>16.1.6</b> Perform spatial niche trajectory analysis using ONTraC</a></li>
<li class="chapter" data-level="16.1.7" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#visualization"><i class="fa fa-check"></i><b>16.1.7</b> visualization</a></li>
</ul></li>
<li class="chapter" data-level="16.2" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#session-info-3"><i class="fa fa-check"></i><b>16.2</b> Session info</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html"><i class="fa fa-check"></i><b>17</b> Interactivity with the R/Spatial ecosystem</a>
<ul>
<li class="chapter" data-level="17.1" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#kriging"><i class="fa fa-check"></i><b>17.1</b> Kriging</a>
<ul>
<li class="chapter" data-level="17.1.1" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#visium-technology-1"><i class="fa fa-check"></i><b>17.1.1</b> Visium technology</a></li>
<li class="chapter" data-level="17.1.2" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#dataset-2"><i class="fa fa-check"></i><b>17.1.2</b> Dataset</a></li>
<li class="chapter" data-level="17.1.3" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#generating-a-geojson-file-of-nuclei-location"><i class="fa fa-check"></i><b>17.1.3</b> Generating a geojson file of nuclei location</a></li>
</ul></li>
<li class="chapter" data-level="17.2" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#downloading-the-dataset"><i class="fa fa-check"></i><b>17.2</b> Downloading the dataset</a></li>
<li class="chapter" data-level="17.3" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#extracting-the-files"><i class="fa fa-check"></i><b>17.3</b> Extracting the files</a></li>
<li class="chapter" data-level="17.4" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#downloading-giotto-object-and-nuclei-segmentation"><i class="fa fa-check"></i><b>17.4</b> Downloading giotto object and nuclei segmentation</a></li>
<li class="chapter" data-level="17.5" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#importing-visium-data"><i class="fa fa-check"></i><b>17.5</b> Importing visium data</a>
<ul>
<li class="chapter" data-level="17.5.1" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#identifying-spatially-organized-features"><i class="fa fa-check"></i><b>17.5.1</b> Identifying spatially organized features</a></li>
</ul></li>
<li class="chapter" data-level="17.6" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#adding-cell-polygons-to-giotto-object"><i class="fa fa-check"></i><b>17.6</b> Adding cell polygons to Giotto object</a>
<ul>
<li class="chapter" data-level="17.6.1" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#read-in-the-poly-information"><i class="fa fa-check"></i><b>17.6.1</b> Read in the poly information</a></li>
<li class="chapter" data-level="17.6.2" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#vizualizing-polygons"><i class="fa fa-check"></i><b>17.6.2</b> Vizualizing polygons</a></li>
<li class="chapter" data-level="17.6.3" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#showing-giotto-object-prior-to-polygon-addition"><i class="fa fa-check"></i><b>17.6.3</b> Showing Giotto object prior to polygon addition</a></li>
<li class="chapter" data-level="17.6.4" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#adding-polygons-to-giotto-object"><i class="fa fa-check"></i><b>17.6.4</b> Adding polygons to giotto object</a></li>
<li class="chapter" data-level="17.6.5" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#check-polygon-information"><i class="fa fa-check"></i><b>17.6.5</b> Check polygon information</a></li>
</ul></li>
<li class="chapter" data-level="17.7" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#performing-kriging"><i class="fa fa-check"></i><b>17.7</b> Performing kriging</a>
<ul>
<li class="chapter" data-level="17.7.1" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#interpolating-features"><i class="fa fa-check"></i><b>17.7.1</b> Interpolating features</a></li>
<li class="chapter" data-level="17.7.2" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#expression-overlap"><i class="fa fa-check"></i><b>17.7.2</b> Expression overlap</a></li>
</ul></li>
<li class="chapter" data-level="17.8" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#reading-in-larger-dataset"><i class="fa fa-check"></i><b>17.8</b> Reading in larger dataset</a></li>
<li class="chapter" data-level="17.9" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#analyzing-interpolated-features"><i class="fa fa-check"></i><b>17.9</b> Analyzing interpolated features</a>
<ul>
<li class="chapter" data-level="17.9.1" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#filter-and-normalization"><i class="fa fa-check"></i><b>17.9.1</b> Filter and normalization</a></li>
<li class="chapter" data-level="17.9.2" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#visualizing-gene-expression-from-interpolated-expression"><i class="fa fa-check"></i><b>17.9.2</b> Visualizing gene expression from interpolated expression</a></li>
<li class="chapter" data-level="17.9.3" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#run-pca"><i class="fa fa-check"></i><b>17.9.3</b> Run PCA</a></li>
<li class="chapter" data-level="17.9.4" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#clustering-2"><i class="fa fa-check"></i><b>17.9.4</b> Clustering</a></li>
<li class="chapter" data-level="17.9.5" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#visualizing-clustering"><i class="fa fa-check"></i><b>17.9.5</b> Visualizing clustering</a></li>
<li class="chapter" data-level="17.9.6" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#cropping-objects"><i class="fa fa-check"></i><b>17.9.6</b> Cropping objects</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="18" data-path="contributing-to-giotto.html"><a href="contributing-to-giotto.html"><i class="fa fa-check"></i><b>18</b> Contributing to Giotto</a>
<ul>
<li class="chapter" data-level="18.1" data-path="contributing-to-giotto.html"><a href="contributing-to-giotto.html#contribution-guideline"><i class="fa fa-check"></i><b>18.1</b> Contribution guideline</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Workshop: Spatial multi-omics data analysis with Giotto Suite</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="interoperability-with-isolated-tools" class="section level1 hasAnchor" number="16">
<h1><span class="header-section-number">16</span> Interoperability with isolated tools<a href="interoperability-with-isolated-tools.html#interoperability-with-isolated-tools" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Wen Wang</p>
<p>August 7th 2024</p>
<div id="spatial-niche-trajectory-analysis-ontrac" class="section level2 hasAnchor" number="16.1">
<h2><span class="header-section-number">16.1</span> Spatial niche trajectory analysis (ONTraC)<a href="interoperability-with-isolated-tools.html#spatial-niche-trajectory-analysis-ontrac" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="introduction-to-ontrac" class="section level3 hasAnchor" number="16.1.1">
<h3><span class="header-section-number">16.1.1</span> Introduction to ONTraC<a href="interoperability-with-isolated-tools.html#introduction-to-ontrac" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>ONTraC (Ordered Niche Trajectory Construction) is a niche-centered, machine learning
method for constructing spatially continuous trajectories. ONTraC differs from existing tools in that it treats a niche, rather than an individual cell, as the basic unit for spatial trajectory analysis. In this context, we define niche as a multicellular, spatially localized region where different cell types may coexist and interact with each other. ONTraC seamlessly integrates cell-type composition and spatial information by using the graph neural network modeling framework. Its output, which is called the niche trajectory, can be viewed as a one dimensional representation of the tissue microenvironment continuum. By disentangling cell-level and niche-level properties, niche trajectory analysis provides a coherent framework to study coordinated responses from all the cells in association with continuous tissue microenvironment variations.</p>
<p><img src="img/03_session5/0-ONTraCWorkFlow.png" width="80%" style="display: block; margin: auto;" /></p>
<p><a href="https://www.biorxiv.org/content/10.1101/2024.04.23.590827v1">ONTraC paper</a></p>
<p><a href="https://github.com/gyuanlab/ONTraC">ONTraC GitHub repository</a></p>
</div>
<div id="introduction-to-merfish" class="section level3 hasAnchor" number="16.1.2">
<h3><span class="header-section-number">16.1.2</span> Introduction to MERFISH<a href="interoperability-with-isolated-tools.html#introduction-to-merfish" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>MERFISH is a massively multiplexed single-molecule imaging technology for spatially resolved transcriptomics capable of simultaneously measuring the copy number and spatial distribution of hundreds to tens of thousands of RNA species in individual cells. For further information, please visit the official <a href="https://vizgen.com/technology/#merfish">website</a>.</p>
<p><img src="img/03_session5/0-MERFISH.png" width="100%" style="display: block; margin: auto;" /></p>
</div>
<div id="settings" class="section level3 hasAnchor" number="16.1.3">
<h3><span class="header-section-number">16.1.3</span> settings<a href="interoperability-with-isolated-tools.html#settings" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb376"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb376-1"><a href="interoperability-with-isolated-tools.html#cb376-1" tabindex="-1"></a><span class="fu">options</span>(<span class="at">timeout=</span><span class="cn">Inf</span>) <span class="co"># In case of network interreupt</span></span>
<span id="cb376-2"><a href="interoperability-with-isolated-tools.html#cb376-2" tabindex="-1"></a>data_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(<span class="st">&quot;data&quot;</span>,<span class="st">&quot;merfish&quot;</span>)</span>
<span id="cb376-3"><a href="interoperability-with-isolated-tools.html#cb376-3" tabindex="-1"></a><span class="fu">dir.create</span>(data_path)</span>
<span id="cb376-4"><a href="interoperability-with-isolated-tools.html#cb376-4" tabindex="-1"></a>results_folder <span class="ot">&lt;-</span> <span class="fu">file.path</span>(<span class="st">&quot;results&quot;</span>,<span class="st">&quot;merfish&quot;</span>)</span>
<span id="cb376-5"><a href="interoperability-with-isolated-tools.html#cb376-5" tabindex="-1"></a><span class="fu">dir.create</span>(results_folder)</span></code></pre></div>
</div>
<div id="dataset-1" class="section level3 hasAnchor" number="16.1.4">
<h3><span class="header-section-number">16.1.4</span> Dataset<a href="interoperability-with-isolated-tools.html#dataset-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>This is a MERFISH mouse motor cortex dataset comprising 61 tissue sections and containing approximately 280,000 cells characterised by a 258-gene panel. The study identified 3 classes of cells, glutamatergic, GABAergic and non-neuronal cell groups, and further clustered into 23 annotated plus 1 other subclass-level cell types.</p>
<p><img src="img/03_session5/0-Dataset.png" width="100%" style="display: block; margin: auto;" /></p>
<p>Pseudotime based methods could generate one dimensional coordinates for specific lineages but lack the ability to generate trajectories for whole samples. By moving our focus from the cell to the niche (local microenvironment), ONTraC could generate niche trajectories for whole samples and map the NT score to each cell.</p>
<div id="dataset-download" class="section level4 hasAnchor" number="16.1.4.1">
<h4><span class="header-section-number">16.1.4.1</span> Dataset download<a href="interoperability-with-isolated-tools.html#dataset-download" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>The MERFISH mouse motor cortex data to run this tutorial can be found
<a href="https://doi.brainimagelibrary.org/doi/10.35077/g.21">here</a></p>
<p>You need to download the processed expression, metadata, and cell segmentation
information by running these commands:</p>
<p>Note 1: there are 61 slices here, we run on two of them to save the time.
Note 2: due to the instability of network, download processing may be interrupt.
We recommend to download these data in advance or download the processing giotto obj from <a href="https://zenodo.org/records/13216587/files/merfish.zip?download=1">Zenodo</a>.</p>
<div class="sourceCode" id="cb377"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb377-1"><a href="interoperability-with-isolated-tools.html#cb377-1" tabindex="-1"></a><span class="fu">download.file</span>(<span class="at">url =</span> <span class="st">&quot;https://download.brainimagelibrary.org/cf/1c/cf1c1a431ef8d021/processed_data/counts.h5ad&quot;</span>,</span>
<span id="cb377-2"><a href="interoperability-with-isolated-tools.html#cb377-2" tabindex="-1"></a>              <span class="at">destfile =</span> <span class="fu">file.path</span>(data_path,<span class="st">&quot;counts.h5ad&quot;</span>))</span>
<span id="cb377-3"><a href="interoperability-with-isolated-tools.html#cb377-3" tabindex="-1"></a><span class="fu">download.file</span>(<span class="at">url =</span> <span class="st">&quot;https://download.brainimagelibrary.org/cf/1c/cf1c1a431ef8d021/processed_data/cell_labels.csv&quot;</span>,</span>
<span id="cb377-4"><a href="interoperability-with-isolated-tools.html#cb377-4" tabindex="-1"></a>              <span class="at">destfile =</span> <span class="fu">file.path</span>(data_path,<span class="st">&quot;cell_labels.csv&quot;</span>))</span>
<span id="cb377-5"><a href="interoperability-with-isolated-tools.html#cb377-5" tabindex="-1"></a><span class="fu">download.file</span>(<span class="at">url =</span> <span class="st">&quot;https://download.brainimagelibrary.org/cf/1c/cf1c1a431ef8d021/processed_data/segmented_cells_mouse2sample1.csv&quot;</span>,</span>
<span id="cb377-6"><a href="interoperability-with-isolated-tools.html#cb377-6" tabindex="-1"></a>              <span class="at">destfile =</span> <span class="fu">file.path</span>(data_path,<span class="st">&quot;segmented_cells_mouse2sample1.csv&quot;</span>))</span>
<span id="cb377-7"><a href="interoperability-with-isolated-tools.html#cb377-7" tabindex="-1"></a><span class="fu">download.file</span>(<span class="at">url =</span> <span class="st">&quot;https://download.brainimagelibrary.org/cf/1c/cf1c1a431ef8d021/processed_data/segmented_cells_mouse2sample6.csv&quot;</span>,</span>
<span id="cb377-8"><a href="interoperability-with-isolated-tools.html#cb377-8" tabindex="-1"></a>              <span class="at">destfile =</span> <span class="fu">file.path</span>(data_path,<span class="st">&quot;segmented_cells_mouse2sample6.csv&quot;</span>))</span></code></pre></div>
</div>
</div>
<div id="create-the-giotto-object-2" class="section level3 hasAnchor" number="16.1.5">
<h3><span class="header-section-number">16.1.5</span> Create the Giotto object<a href="interoperability-with-isolated-tools.html#create-the-giotto-object-2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb378"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb378-1"><a href="interoperability-with-isolated-tools.html#cb378-1" tabindex="-1"></a><span class="fu">library</span>(Giotto)</span>
<span id="cb378-2"><a href="interoperability-with-isolated-tools.html#cb378-2" tabindex="-1"></a><span class="fu">library</span>(reticulate)</span>
<span id="cb378-3"><a href="interoperability-with-isolated-tools.html#cb378-3" tabindex="-1"></a></span>
<span id="cb378-4"><a href="interoperability-with-isolated-tools.html#cb378-4" tabindex="-1"></a><span class="do">## Set instructions</span></span>
<span id="cb378-5"><a href="interoperability-with-isolated-tools.html#cb378-5" tabindex="-1"></a>python_path <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb378-6"><a href="interoperability-with-isolated-tools.html#cb378-6" tabindex="-1"></a></span>
<span id="cb378-7"><a href="interoperability-with-isolated-tools.html#cb378-7" tabindex="-1"></a>instructions <span class="ot">&lt;-</span> <span class="fu">createGiottoInstructions</span>(</span>
<span id="cb378-8"><a href="interoperability-with-isolated-tools.html#cb378-8" tabindex="-1"></a>    <span class="at">save_dir =</span> results_folder,</span>
<span id="cb378-9"><a href="interoperability-with-isolated-tools.html#cb378-9" tabindex="-1"></a>    <span class="at">save_plot =</span> <span class="cn">TRUE</span>,</span>
<span id="cb378-10"><a href="interoperability-with-isolated-tools.html#cb378-10" tabindex="-1"></a>    <span class="at">show_plot =</span> <span class="cn">FALSE</span>,</span>
<span id="cb378-11"><a href="interoperability-with-isolated-tools.html#cb378-11" tabindex="-1"></a>    <span class="at">return_plot =</span> <span class="cn">FALSE</span>,</span>
<span id="cb378-12"><a href="interoperability-with-isolated-tools.html#cb378-12" tabindex="-1"></a>    <span class="at">python_path =</span> python_path</span>
<span id="cb378-13"><a href="interoperability-with-isolated-tools.html#cb378-13" tabindex="-1"></a>)</span>
<span id="cb378-14"><a href="interoperability-with-isolated-tools.html#cb378-14" tabindex="-1"></a></span>
<span id="cb378-15"><a href="interoperability-with-isolated-tools.html#cb378-15" tabindex="-1"></a><span class="do">## create Giotto object from expression counts. This file contains 61 slices here.</span></span>
<span id="cb378-16"><a href="interoperability-with-isolated-tools.html#cb378-16" tabindex="-1"></a>giotto_all_slices_obj <span class="ot">&lt;-</span> <span class="fu">anndataToGiotto</span>(<span class="st">&quot;data/merfish/counts.h5ad&quot;</span>)</span>
<span id="cb378-17"><a href="interoperability-with-isolated-tools.html#cb378-17" tabindex="-1"></a></span>
<span id="cb378-18"><a href="interoperability-with-isolated-tools.html#cb378-18" tabindex="-1"></a><span class="do">## load meta_data</span></span>
<span id="cb378-19"><a href="interoperability-with-isolated-tools.html#cb378-19" tabindex="-1"></a>meta_df <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="fu">file.path</span>(data_path, <span class="st">&quot;cell_labels.csv&quot;</span>), <span class="at">colClasses =</span> <span class="st">&quot;character&quot;</span>) <span class="co"># as the cell IDs are 30 digit numbers, set the type as character to avoid the limitation of R in handling larger integers</span></span>
<span id="cb378-20"><a href="interoperability-with-isolated-tools.html#cb378-20" tabindex="-1"></a><span class="fu">colnames</span>(meta_df)[[<span class="dv">1</span>]] <span class="ot">&lt;-</span> <span class="st">&quot;cell_ID&quot;</span></span>
<span id="cb378-21"><a href="interoperability-with-isolated-tools.html#cb378-21" tabindex="-1"></a></span>
<span id="cb378-22"><a href="interoperability-with-isolated-tools.html#cb378-22" tabindex="-1"></a><span class="do">### we use two slices here to speed up</span></span>
<span id="cb378-23"><a href="interoperability-with-isolated-tools.html#cb378-23" tabindex="-1"></a>slice1_cells <span class="ot">&lt;-</span> meta_df[meta_df<span class="sc">$</span>slice_id <span class="sc">==</span> <span class="st">&quot;mouse2_slice229&quot;</span>,]<span class="sc">$</span>cell_ID</span>
<span id="cb378-24"><a href="interoperability-with-isolated-tools.html#cb378-24" tabindex="-1"></a>slice2_cells <span class="ot">&lt;-</span> meta_df[meta_df<span class="sc">$</span>slice_id <span class="sc">==</span> <span class="st">&quot;mouse2_slice300&quot;</span>,]<span class="sc">$</span>cell_ID</span>
<span id="cb378-25"><a href="interoperability-with-isolated-tools.html#cb378-25" tabindex="-1"></a></span>
<span id="cb378-26"><a href="interoperability-with-isolated-tools.html#cb378-26" tabindex="-1"></a><span class="do">## subset giotto obj by cell ID</span></span>
<span id="cb378-27"><a href="interoperability-with-isolated-tools.html#cb378-27" tabindex="-1"></a>giotto_slice1_obj <span class="ot">&lt;-</span> <span class="fu">subsetGiotto</span>(<span class="at">gobject =</span> giotto_all_slices_obj,</span>
<span id="cb378-28"><a href="interoperability-with-isolated-tools.html#cb378-28" tabindex="-1"></a>                                  <span class="at">cell_ids =</span> slice1_cells)</span>
<span id="cb378-29"><a href="interoperability-with-isolated-tools.html#cb378-29" tabindex="-1"></a>giotto_slice2_obj <span class="ot">&lt;-</span> <span class="fu">subsetGiotto</span>(<span class="at">gobject =</span> giotto_all_slices_obj,</span>
<span id="cb378-30"><a href="interoperability-with-isolated-tools.html#cb378-30" tabindex="-1"></a>                                  <span class="at">cell_ids =</span> slice2_cells)</span>
<span id="cb378-31"><a href="interoperability-with-isolated-tools.html#cb378-31" tabindex="-1"></a></span>
<span id="cb378-32"><a href="interoperability-with-isolated-tools.html#cb378-32" tabindex="-1"></a><span class="do">## add cell metadata</span></span>
<span id="cb378-33"><a href="interoperability-with-isolated-tools.html#cb378-33" tabindex="-1"></a>giotto_slice1_obj <span class="ot">&lt;-</span> <span class="fu">addCellMetadata</span>(<span class="at">gobject =</span> giotto_slice1_obj,</span>
<span id="cb378-34"><a href="interoperability-with-isolated-tools.html#cb378-34" tabindex="-1"></a>                <span class="at">new_metadata =</span> meta_df,</span>
<span id="cb378-35"><a href="interoperability-with-isolated-tools.html#cb378-35" tabindex="-1"></a>                <span class="at">by_column =</span> T)</span>
<span id="cb378-36"><a href="interoperability-with-isolated-tools.html#cb378-36" tabindex="-1"></a>giotto_slice2_obj <span class="ot">&lt;-</span> <span class="fu">addCellMetadata</span>(<span class="at">gobject =</span> giotto_slice2_obj,</span>
<span id="cb378-37"><a href="interoperability-with-isolated-tools.html#cb378-37" tabindex="-1"></a>                <span class="at">new_metadata =</span> meta_df,</span>
<span id="cb378-38"><a href="interoperability-with-isolated-tools.html#cb378-38" tabindex="-1"></a>                <span class="at">by_column =</span> T)</span>
<span id="cb378-39"><a href="interoperability-with-isolated-tools.html#cb378-39" tabindex="-1"></a></span>
<span id="cb378-40"><a href="interoperability-with-isolated-tools.html#cb378-40" tabindex="-1"></a></span>
<span id="cb378-41"><a href="interoperability-with-isolated-tools.html#cb378-41" tabindex="-1"></a></span>
<span id="cb378-42"><a href="interoperability-with-isolated-tools.html#cb378-42" tabindex="-1"></a></span>
<span id="cb378-43"><a href="interoperability-with-isolated-tools.html#cb378-43" tabindex="-1"></a><span class="do">## cell segmentation. Calculate center (median of vertices) of each cell.</span></span>
<span id="cb378-44"><a href="interoperability-with-isolated-tools.html#cb378-44" tabindex="-1"></a>segments_1_df <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="fu">file.path</span>(data_path, <span class="st">&quot;segmented_cells_mouse2sample1.csv&quot;</span>), <span class="at">row.names=</span><span class="dv">1</span>, <span class="at">colClasses =</span> <span class="st">&quot;character&quot;</span>) <span class="co"># as the cell IDs are 30 digit numbers, set the type as character to avoid the limitation of R in handling larger integers</span></span>
<span id="cb378-45"><a href="interoperability-with-isolated-tools.html#cb378-45" tabindex="-1"></a>segments_2_df <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="fu">file.path</span>(data_path, <span class="st">&quot;segmented_cells_mouse2sample6.csv&quot;</span>), <span class="at">row.names=</span><span class="dv">1</span>, <span class="at">colClasses =</span> <span class="st">&quot;character&quot;</span>) <span class="co"># as the cell IDs are 30 digit numbers, set the type as character to avoid the limitation of R in handling larger integers</span></span>
<span id="cb378-46"><a href="interoperability-with-isolated-tools.html#cb378-46" tabindex="-1"></a>segments_df <span class="ot">&lt;-</span> <span class="fu">rbind</span>(segments_1_df, segments_2_df)</span>
<span id="cb378-47"><a href="interoperability-with-isolated-tools.html#cb378-47" tabindex="-1"></a>loc.use <span class="ot">&lt;-</span> segments_df[selected_cells,]</span>
<span id="cb378-48"><a href="interoperability-with-isolated-tools.html#cb378-48" tabindex="-1"></a>loc.x <span class="ot">&lt;-</span> <span class="fu">grep</span>(<span class="st">&#39;boundaryX_&#39;</span>,<span class="fu">colnames</span>(loc.use),<span class="at">value =</span> T)</span>
<span id="cb378-49"><a href="interoperability-with-isolated-tools.html#cb378-49" tabindex="-1"></a>loc.y <span class="ot">&lt;-</span> <span class="fu">grep</span>(<span class="st">&#39;boundaryY_&#39;</span>,<span class="fu">colnames</span>(loc.use),<span class="at">value =</span> T)</span>
<span id="cb378-50"><a href="interoperability-with-isolated-tools.html#cb378-50" tabindex="-1"></a>centr.x <span class="ot">&lt;-</span> <span class="fu">apply</span>(loc.use[,loc.x],<span class="dv">1</span>,<span class="cf">function</span>(x){</span>
<span id="cb378-51"><a href="interoperability-with-isolated-tools.html#cb378-51" tabindex="-1"></a>  temp <span class="ot">&lt;-</span> <span class="fu">lapply</span>(x,<span class="cf">function</span>(y){</span>
<span id="cb378-52"><a href="interoperability-with-isolated-tools.html#cb378-52" tabindex="-1"></a>    <span class="fu">as.numeric</span>(<span class="fu">unlist</span>(<span class="fu">strsplit</span>(y,<span class="st">&#39;, &#39;</span>)))</span>
<span id="cb378-53"><a href="interoperability-with-isolated-tools.html#cb378-53" tabindex="-1"></a>  })</span>
<span id="cb378-54"><a href="interoperability-with-isolated-tools.html#cb378-54" tabindex="-1"></a>  <span class="fu">return</span> (<span class="fu">median</span>(<span class="fu">unname</span>(<span class="fu">unlist</span>(temp))))</span>
<span id="cb378-55"><a href="interoperability-with-isolated-tools.html#cb378-55" tabindex="-1"></a>})</span>
<span id="cb378-56"><a href="interoperability-with-isolated-tools.html#cb378-56" tabindex="-1"></a>centr.y <span class="ot">&lt;-</span> <span class="fu">apply</span>(loc.use[,loc.y],<span class="dv">1</span>,<span class="cf">function</span>(x){</span>
<span id="cb378-57"><a href="interoperability-with-isolated-tools.html#cb378-57" tabindex="-1"></a>  temp <span class="ot">&lt;-</span> <span class="fu">lapply</span>(x,<span class="cf">function</span>(y){</span>
<span id="cb378-58"><a href="interoperability-with-isolated-tools.html#cb378-58" tabindex="-1"></a>    <span class="fu">as.numeric</span>(<span class="fu">unlist</span>(<span class="fu">strsplit</span>(y,<span class="st">&#39;, &#39;</span>)))</span>
<span id="cb378-59"><a href="interoperability-with-isolated-tools.html#cb378-59" tabindex="-1"></a>  })</span>
<span id="cb378-60"><a href="interoperability-with-isolated-tools.html#cb378-60" tabindex="-1"></a>  <span class="fu">return</span> (<span class="fu">median</span>(<span class="fu">unname</span>(<span class="fu">unlist</span>(temp))))</span>
<span id="cb378-61"><a href="interoperability-with-isolated-tools.html#cb378-61" tabindex="-1"></a>})</span>
<span id="cb378-62"><a href="interoperability-with-isolated-tools.html#cb378-62" tabindex="-1"></a></span>
<span id="cb378-63"><a href="interoperability-with-isolated-tools.html#cb378-63" tabindex="-1"></a><span class="do">## create spatial locations object</span></span>
<span id="cb378-64"><a href="interoperability-with-isolated-tools.html#cb378-64" tabindex="-1"></a>spatial_locs_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">cell_ID =</span> selected_cells,<span class="at">sdimx =</span> centr.x,<span class="at">sdimy =</span> centr.y)</span>
<span id="cb378-65"><a href="interoperability-with-isolated-tools.html#cb378-65" tabindex="-1"></a>spatial_locs_slice1_df <span class="ot">&lt;-</span> spatial_locs_df[slice1_cells,]</span>
<span id="cb378-66"><a href="interoperability-with-isolated-tools.html#cb378-66" tabindex="-1"></a>spatial_locs_slice2_df <span class="ot">&lt;-</span> spatial_locs_df[slice2_cells,]</span>
<span id="cb378-67"><a href="interoperability-with-isolated-tools.html#cb378-67" tabindex="-1"></a>spat_locs_slice1_obj <span class="ot">&lt;-</span> <span class="fu">readSpatLocsData</span>(<span class="at">data_list =</span> spatial_locs_slice1_df)</span>
<span id="cb378-68"><a href="interoperability-with-isolated-tools.html#cb378-68" tabindex="-1"></a>spat_locs_slice2_obj <span class="ot">&lt;-</span> <span class="fu">readSpatLocsData</span>(<span class="at">data_list =</span> spatial_locs_slice2_df)</span>
<span id="cb378-69"><a href="interoperability-with-isolated-tools.html#cb378-69" tabindex="-1"></a></span>
<span id="cb378-70"><a href="interoperability-with-isolated-tools.html#cb378-70" tabindex="-1"></a><span class="do">## add spatial location info</span></span>
<span id="cb378-71"><a href="interoperability-with-isolated-tools.html#cb378-71" tabindex="-1"></a>giotto_slice1_obj <span class="ot">&lt;-</span> <span class="fu">setSpatialLocations</span>(<span class="at">gobject =</span> giotto_slice1_obj,</span>
<span id="cb378-72"><a href="interoperability-with-isolated-tools.html#cb378-72" tabindex="-1"></a>                                         <span class="at">x =</span> spat_locs_slice1_obj)</span>
<span id="cb378-73"><a href="interoperability-with-isolated-tools.html#cb378-73" tabindex="-1"></a>giotto_slice2_obj <span class="ot">&lt;-</span> <span class="fu">setSpatialLocations</span>(<span class="at">gobject =</span> giotto_slice2_obj,</span>
<span id="cb378-74"><a href="interoperability-with-isolated-tools.html#cb378-74" tabindex="-1"></a>                                         <span class="at">x =</span> spat_locs_slice2_obj)</span>
<span id="cb378-75"><a href="interoperability-with-isolated-tools.html#cb378-75" tabindex="-1"></a></span>
<span id="cb378-76"><a href="interoperability-with-isolated-tools.html#cb378-76" tabindex="-1"></a><span class="do">## merge two giotto objects together</span></span>
<span id="cb378-77"><a href="interoperability-with-isolated-tools.html#cb378-77" tabindex="-1"></a>giotto_obj <span class="ot">=</span> <span class="fu">joinGiottoObjects</span>(<span class="at">gobject_list =</span> <span class="fu">list</span>(giotto_slice1_obj, giotto_slice2_obj),</span>
<span id="cb378-78"><a href="interoperability-with-isolated-tools.html#cb378-78" tabindex="-1"></a>                              <span class="at">gobject_names =</span> <span class="fu">c</span>(<span class="st">&#39;mouse2_slice229&#39;</span>,</span>
<span id="cb378-79"><a href="interoperability-with-isolated-tools.html#cb378-79" tabindex="-1"></a>                                                <span class="st">&#39;mouse2_slice300&#39;</span>),  <span class="co"># name for each samples</span></span>
<span id="cb378-80"><a href="interoperability-with-isolated-tools.html#cb378-80" tabindex="-1"></a>                              <span class="at">join_method =</span> <span class="st">&#39;z_stack&#39;</span>)</span>
<span id="cb378-81"><a href="interoperability-with-isolated-tools.html#cb378-81" tabindex="-1"></a></span>
<span id="cb378-82"><a href="interoperability-with-isolated-tools.html#cb378-82" tabindex="-1"></a><span class="do">## save giotto obj</span></span>
<span id="cb378-83"><a href="interoperability-with-isolated-tools.html#cb378-83" tabindex="-1"></a><span class="fu">saveRDS</span>(giotto_obj, <span class="at">file =</span> <span class="fu">file.path</span>(results_folder, <span class="st">&quot;giotto_merfish_sbuset.Rds&quot;</span>))</span></code></pre></div>
<p>If you facing network issue when downloading the raw dataset. Please download the processing giotto obj from <a href="https://zenodo.org/records/13216587/files/merfish.zip?download=1">Zenodo</a>, unzip and move it to <code>results</code> folder.s</p>
<div class="sourceCode" id="cb379"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb379-1"><a href="interoperability-with-isolated-tools.html#cb379-1" tabindex="-1"></a>giotto_obj <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="at">file =</span> <span class="fu">file.path</span>(results_folder, <span class="st">&quot;giotto_merfish_sbuset.Rds&quot;</span>))</span></code></pre></div>
<div id="spatial-distribution-of-cell-type" class="section level4 hasAnchor" number="16.1.5.1">
<h4><span class="header-section-number">16.1.5.1</span> spatial distribution of cell type<a href="interoperability-with-isolated-tools.html#spatial-distribution-of-cell-type" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb380"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb380-1"><a href="interoperability-with-isolated-tools.html#cb380-1" tabindex="-1"></a><span class="fu">spatPlot2D</span>(giotto_obj,</span>
<span id="cb380-2"><a href="interoperability-with-isolated-tools.html#cb380-2" tabindex="-1"></a>           <span class="at">group_by =</span> <span class="st">&quot;slice_id&quot;</span>,</span>
<span id="cb380-3"><a href="interoperability-with-isolated-tools.html#cb380-3" tabindex="-1"></a>           <span class="at">cell_color =</span> <span class="st">&quot;subclass&quot;</span>,</span>
<span id="cb380-4"><a href="interoperability-with-isolated-tools.html#cb380-4" tabindex="-1"></a>           <span class="at">point_size =</span> <span class="dv">1</span>,</span>
<span id="cb380-5"><a href="interoperability-with-isolated-tools.html#cb380-5" tabindex="-1"></a>           <span class="at">point_border_stroke =</span> <span class="cn">NA</span>,</span>
<span id="cb380-6"><a href="interoperability-with-isolated-tools.html#cb380-6" tabindex="-1"></a>           <span class="at">legend_text =</span> <span class="dv">6</span>)</span></code></pre></div>
<p><img src="img/03_session5/1-spatialCellTypeDis.png" width="100%" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb381"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb381-1"><a href="interoperability-with-isolated-tools.html#cb381-1" tabindex="-1"></a><span class="co"># We skip the processing process here to save time and use the given cell type</span></span>
<span id="cb381-2"><a href="interoperability-with-isolated-tools.html#cb381-2" tabindex="-1"></a><span class="co"># annotation directly</span></span>
<span id="cb381-3"><a href="interoperability-with-isolated-tools.html#cb381-3" tabindex="-1"></a></span>
<span id="cb381-4"><a href="interoperability-with-isolated-tools.html#cb381-4" tabindex="-1"></a>ONTraC_input <span class="ot">&lt;-</span> <span class="fu">getONTraCv1Input</span>(<span class="at">gobject =</span> giotto_obj,</span>
<span id="cb381-5"><a href="interoperability-with-isolated-tools.html#cb381-5" tabindex="-1"></a>                                 <span class="at">cell_type =</span> <span class="st">&quot;subclass&quot;</span>,</span>
<span id="cb381-6"><a href="interoperability-with-isolated-tools.html#cb381-6" tabindex="-1"></a>                                 <span class="at">output_path =</span> data_path,</span>
<span id="cb381-7"><a href="interoperability-with-isolated-tools.html#cb381-7" tabindex="-1"></a>                                 <span class="at">spat_unit =</span> <span class="st">&quot;cell&quot;</span>,</span>
<span id="cb381-8"><a href="interoperability-with-isolated-tools.html#cb381-8" tabindex="-1"></a>                                 <span class="at">feat_type =</span> <span class="st">&quot;rna&quot;</span>,</span>
<span id="cb381-9"><a href="interoperability-with-isolated-tools.html#cb381-9" tabindex="-1"></a>                                 <span class="at">verbose =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<div class="sourceCode" id="cb382"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb382-1"><a href="interoperability-with-isolated-tools.html#cb382-1" tabindex="-1"></a><span class="fu">head</span>(ONTraC_input)</span></code></pre></div>
<div class="sourceCode" id="cb383"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb383-1"><a href="interoperability-with-isolated-tools.html#cb383-1" tabindex="-1"></a><span class="co">#   Cell_ID                                                     Sample          x           y           Cell_Type          </span></span>
<span id="cb383-2"><a href="interoperability-with-isolated-tools.html#cb383-2" tabindex="-1"></a><span class="co">#   &lt;chr&gt;                                                       &lt;chr&gt;           &lt;dbl&gt;       &lt;dbl&gt;       &lt;chr&gt;</span></span>
<span id="cb383-3"><a href="interoperability-with-isolated-tools.html#cb383-3" tabindex="-1"></a><span class="co">#   mouse2_slice229-100101435705986292663283283043431511315     mouse2_slice229 -4828.728   -2203.4502  L6 CT</span></span>
<span id="cb383-4"><a href="interoperability-with-isolated-tools.html#cb383-4" tabindex="-1"></a><span class="co">#   mouse2_slice229-100104370212612969023746137269354247741     mouse2_slice229 -5405.400   -995.6467   OPC</span></span>
<span id="cb383-5"><a href="interoperability-with-isolated-tools.html#cb383-5" tabindex="-1"></a><span class="co">#   mouse2_slice229-100128078183217482733448056590230529739     mouse2_slice229 -5731.403   -1071.1735  L2/3 IT</span></span>
<span id="cb383-6"><a href="interoperability-with-isolated-tools.html#cb383-6" tabindex="-1"></a><span class="co">#   mouse2_slice229-100209662400867003194056898065587980841     mouse2_slice229 -5468.113   -1286.2465  Oligo</span></span>
<span id="cb383-7"><a href="interoperability-with-isolated-tools.html#cb383-7" tabindex="-1"></a><span class="co">#   mouse2_slice229-100218038012295593766653119076639444055     mouse2_slice229 -6399.986   -959.7440   L2/3 IT</span></span>
<span id="cb383-8"><a href="interoperability-with-isolated-tools.html#cb383-8" tabindex="-1"></a><span class="co">#   mouse2_slice229-100252992997994275968450436343196667192     mouse2_slice229 -6637.847   -1659.6237  Astro</span></span></code></pre></div>
</div>
</div>
<div id="perform-spatial-niche-trajectory-analysis-using-ontrac" class="section level3 hasAnchor" number="16.1.6">
<h3><span class="header-section-number">16.1.6</span> Perform spatial niche trajectory analysis using ONTraC<a href="interoperability-with-isolated-tools.html#perform-spatial-niche-trajectory-analysis-using-ontrac" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="ontrac-installation" class="section level4 hasAnchor" number="16.1.6.1">
<h4><span class="header-section-number">16.1.6.1</span> ONTraC Installation<a href="interoperability-with-isolated-tools.html#ontrac-installation" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>You could run ONTraC on your own laptop or on an HPC with an NVIDIA GPU node.
It will run for less than 10 minutes on this example dataset.
For larger datasets, running on an NVIDIA GPU is recommended, otherwise it will take a long time.</p>
<div class="sourceCode" id="cb384"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb384-1"><a href="interoperability-with-isolated-tools.html#cb384-1" tabindex="-1"></a><span class="bu">source</span> ~/.bash_profile</span>
<span id="cb384-2"><a href="interoperability-with-isolated-tools.html#cb384-2" tabindex="-1"></a><span class="ex">conda</span> create <span class="at">-y</span> <span class="at">-n</span> ONTraC python=3.11</span>
<span id="cb384-3"><a href="interoperability-with-isolated-tools.html#cb384-3" tabindex="-1"></a><span class="ex">conda</span> activate ONTraC</span>
<span id="cb384-4"><a href="interoperability-with-isolated-tools.html#cb384-4" tabindex="-1"></a><span class="ex">pip</span> install ONTraC</span></code></pre></div>
</div>
<div id="running-ontrac" class="section level4 hasAnchor" number="16.1.6.2">
<h4><span class="header-section-number">16.1.6.2</span> Running ONTraC<a href="interoperability-with-isolated-tools.html#running-ontrac" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb385"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb385-1"><a href="interoperability-with-isolated-tools.html#cb385-1" tabindex="-1"></a><span class="bu">source</span> ~/.bash_profile</span>
<span id="cb385-2"><a href="interoperability-with-isolated-tools.html#cb385-2" tabindex="-1"></a><span class="ex">conda</span> activate ONTraC</span>
<span id="cb385-3"><a href="interoperability-with-isolated-tools.html#cb385-3" tabindex="-1"></a></span>
<span id="cb385-4"><a href="interoperability-with-isolated-tools.html#cb385-4" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> output/ONTrac</span>
<span id="cb385-5"><a href="interoperability-with-isolated-tools.html#cb385-5" tabindex="-1"></a><span class="ex">ONTraC</span> <span class="at">-d</span> data/merfish/ONTraC_dataset_input.csv <span class="at">--preprocessing-dir</span> results/merfish/preprocessing_dir <span class="at">--GNN-dir</span> results/merfish/GNN_dir <span class="at">--NTScore-dir</span> results/merfish/NTScore_dir <span class="at">--device</span> cuda     <span class="at">--epochs</span> 1000 <span class="at">-s</span> 42 <span class="at">--patience</span> 100 <span class="at">--min-delta</span> 0.001 <span class="at">--min-epochs</span> 50 <span class="at">--lr</span> 0.03     <span class="at">--hidden-feats</span> 4 <span class="at">-k</span> 6 <span class="at">--modularity-loss-weight</span> 0.3 <span class="at">--purity-loss-weight</span> 300 <span class="at">--regularization-loss-weight</span> 0.3 <span class="at">--beta</span> 0.03 <span class="dv">2</span><span class="op">&gt;&amp;</span><span class="dv">1</span> <span class="kw">|</span> <span class="fu">tee</span> results/merfish/merfish_subset.log</span></code></pre></div>
</div>
</div>
<div id="visualization" class="section level3 hasAnchor" number="16.1.7">
<h3><span class="header-section-number">16.1.7</span> visualization<a href="interoperability-with-isolated-tools.html#visualization" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="load-ontrac-results" class="section level4 hasAnchor" number="16.1.7.1">
<h4><span class="header-section-number">16.1.7.1</span> load ONTraC results<a href="interoperability-with-isolated-tools.html#load-ontrac-results" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb386"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb386-1"><a href="interoperability-with-isolated-tools.html#cb386-1" tabindex="-1"></a>giotto_obj <span class="ot">&lt;-</span> <span class="fu">loadOntraCResults</span>(<span class="at">gobject =</span> giotto_obj,</span>
<span id="cb386-2"><a href="interoperability-with-isolated-tools.html#cb386-2" tabindex="-1"></a>                                <span class="at">ontrac_results_dir =</span> results_folder)</span></code></pre></div>
<p>The NTScore and binarized niche cluster info were stored in cell metadata</p>
<div class="sourceCode" id="cb387"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb387-1"><a href="interoperability-with-isolated-tools.html#cb387-1" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">pDataDT</span>(giotto_obj, <span class="at">spat_unit =</span> <span class="st">&quot;cell&quot;</span>, <span class="at">feat_type =</span> <span class="st">&quot;niche cluster&quot;</span>))</span></code></pre></div>
<div class="sourceCode" id="cb388"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb388-1"><a href="interoperability-with-isolated-tools.html#cb388-1" tabindex="-1"></a><span class="co">#                                                    cell_ID      sample_id        slice_id   class_label subclass    label         list_ID NicheCluster   NTScore</span></span>
<span id="cb388-2"><a href="interoperability-with-isolated-tools.html#cb388-2" tabindex="-1"></a><span class="co">#                                                     &lt;char&gt;         &lt;char&gt;          &lt;char&gt;        &lt;char&gt;   &lt;char&gt;   &lt;char&gt;          &lt;char&gt;        &lt;int&gt;     &lt;num&gt;</span></span>
<span id="cb388-3"><a href="interoperability-with-isolated-tools.html#cb388-3" tabindex="-1"></a><span class="co"># 1: mouse2_slice229-100101435705986292663283283043431511315 mouse2_sample6 mouse2_slice229 Glutamatergic    L6 CT  L6_CT_5 mouse2_slice229            2 0.7998957</span></span>
<span id="cb388-4"><a href="interoperability-with-isolated-tools.html#cb388-4" tabindex="-1"></a><span class="co"># 2: mouse2_slice229-100104370212612969023746137269354247741 mouse2_sample6 mouse2_slice229         Other      OPC      OPC mouse2_slice229            0 0.2003027</span></span>
<span id="cb388-5"><a href="interoperability-with-isolated-tools.html#cb388-5" tabindex="-1"></a><span class="co"># 3: mouse2_slice229-100128078183217482733448056590230529739 mouse2_sample6 mouse2_slice229 Glutamatergic  L2/3 IT L23_IT_4 mouse2_slice229            0 0.2350597</span></span>
<span id="cb388-6"><a href="interoperability-with-isolated-tools.html#cb388-6" tabindex="-1"></a><span class="co"># 4: mouse2_slice229-100209662400867003194056898065587980841 mouse2_sample6 mouse2_slice229         Other    Oligo  Oligo_1 mouse2_slice229            1 0.3990417</span></span>
<span id="cb388-7"><a href="interoperability-with-isolated-tools.html#cb388-7" tabindex="-1"></a><span class="co"># 5: mouse2_slice229-100218038012295593766653119076639444055 mouse2_sample6 mouse2_slice229 Glutamatergic  L2/3 IT L23_IT_4 mouse2_slice229            0 0.2910255</span></span>
<span id="cb388-8"><a href="interoperability-with-isolated-tools.html#cb388-8" tabindex="-1"></a><span class="co"># 6: mouse2_slice229-100252992997994275968450436343196667192 mouse2_sample6 mouse2_slice229         Other    Astro  Astro_2 mouse2_slice229            2 0.8007477</span></span></code></pre></div>
<p>The probability matrix of each cell assigned to each niche cluster and
connectivity between niche cluster were stored here.</p>
<div class="sourceCode" id="cb389"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb389-1"><a href="interoperability-with-isolated-tools.html#cb389-1" tabindex="-1"></a>GiottoClass<span class="sc">::</span><span class="fu">list_expression</span>(giotto_obj)</span></code></pre></div>
<div class="sourceCode" id="cb390"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb390-1"><a href="interoperability-with-isolated-tools.html#cb390-1" tabindex="-1"></a><span class="co">#        spat_unit     feat_type       name</span></span>
<span id="cb390-2"><a href="interoperability-with-isolated-tools.html#cb390-2" tabindex="-1"></a><span class="co">#           &lt;char&gt;        &lt;char&gt;     &lt;char&gt;</span></span>
<span id="cb390-3"><a href="interoperability-with-isolated-tools.html#cb390-3" tabindex="-1"></a><span class="co"># 1:          cell           rna        raw</span></span>
<span id="cb390-4"><a href="interoperability-with-isolated-tools.html#cb390-4" tabindex="-1"></a><span class="co"># 2:          cell niche cluster       prob</span></span>
<span id="cb390-5"><a href="interoperability-with-isolated-tools.html#cb390-5" tabindex="-1"></a><span class="co"># 3: niche cluster  connectivity normalized</span></span></code></pre></div>
</div>
<div id="niche-cluster-probability-distribution" class="section level4 hasAnchor" number="16.1.7.2">
<h4><span class="header-section-number">16.1.7.2</span> niche cluster probability distribution<a href="interoperability-with-isolated-tools.html#niche-cluster-probability-distribution" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb391"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb391-1"><a href="interoperability-with-isolated-tools.html#cb391-1" tabindex="-1"></a><span class="fu">spatFeatPlot2D</span>(<span class="at">gobject =</span> giotto_obj, </span>
<span id="cb391-2"><a href="interoperability-with-isolated-tools.html#cb391-2" tabindex="-1"></a>           <span class="at">spat_unit =</span> <span class="st">&#39;cell&#39;</span>,</span>
<span id="cb391-3"><a href="interoperability-with-isolated-tools.html#cb391-3" tabindex="-1"></a>           <span class="at">feat_type =</span> <span class="st">&#39;niche cluster&#39;</span>,</span>
<span id="cb391-4"><a href="interoperability-with-isolated-tools.html#cb391-4" tabindex="-1"></a>           <span class="at">expression_values =</span> <span class="st">&quot;prob&quot;</span>,</span>
<span id="cb391-5"><a href="interoperability-with-isolated-tools.html#cb391-5" tabindex="-1"></a>           <span class="at">group_by =</span> <span class="st">&#39;list_ID&#39;</span>,</span>
<span id="cb391-6"><a href="interoperability-with-isolated-tools.html#cb391-6" tabindex="-1"></a>           <span class="at">feats =</span> <span class="fu">rownames</span>(giotto_obj<span class="sc">@</span>expression<span class="sc">$</span>cell<span class="sc">$</span><span class="st">`</span><span class="at">niche cluster</span><span class="st">`</span><span class="sc">$</span>prob),</span>
<span id="cb391-7"><a href="interoperability-with-isolated-tools.html#cb391-7" tabindex="-1"></a>           <span class="at">point_border_col =</span> <span class="st">&#39;gray&#39;</span></span>
<span id="cb391-8"><a href="interoperability-with-isolated-tools.html#cb391-8" tabindex="-1"></a>           )</span></code></pre></div>
<p><img src="img/03_session5/2-spatialNicheClusterProb.png" width="60%" style="display: block; margin: auto;" /></p>
</div>
<div id="binarized-niche-cluster-for-each-cell" class="section level4 hasAnchor" number="16.1.7.3">
<h4><span class="header-section-number">16.1.7.3</span> binarized niche cluster for each cell<a href="interoperability-with-isolated-tools.html#binarized-niche-cluster-for-each-cell" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb392"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb392-1"><a href="interoperability-with-isolated-tools.html#cb392-1" tabindex="-1"></a><span class="fu">spatPlot2D</span>(giotto_obj,</span>
<span id="cb392-2"><a href="interoperability-with-isolated-tools.html#cb392-2" tabindex="-1"></a>           <span class="at">spat_unit =</span> <span class="st">&quot;cell&quot;</span>,</span>
<span id="cb392-3"><a href="interoperability-with-isolated-tools.html#cb392-3" tabindex="-1"></a>           <span class="at">group_by =</span> <span class="st">&#39;slice_id&#39;</span>,</span>
<span id="cb392-4"><a href="interoperability-with-isolated-tools.html#cb392-4" tabindex="-1"></a>           <span class="at">cell_color =</span> <span class="st">&quot;NicheCluster&quot;</span>,</span>
<span id="cb392-5"><a href="interoperability-with-isolated-tools.html#cb392-5" tabindex="-1"></a>           <span class="at">color_as_factor =</span> T,</span>
<span id="cb392-6"><a href="interoperability-with-isolated-tools.html#cb392-6" tabindex="-1"></a>           <span class="at">point_size =</span> <span class="dv">1</span>,</span>
<span id="cb392-7"><a href="interoperability-with-isolated-tools.html#cb392-7" tabindex="-1"></a>           <span class="at">point_border_stroke =</span> <span class="cn">NA</span>)</span></code></pre></div>
<p><img src="img/03_session5/3-spatialBinNicheCluster.png" width="80%" style="display: block; margin: auto;" /></p>
</div>
<div id="niche-cluster-spatial-connectivity" class="section level4 hasAnchor" number="16.1.7.4">
<h4><span class="header-section-number">16.1.7.4</span> niche cluster spatial connectivity<a href="interoperability-with-isolated-tools.html#niche-cluster-spatial-connectivity" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb393"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb393-1"><a href="interoperability-with-isolated-tools.html#cb393-1" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">42</span>)  <span class="co"># fix the node positions</span></span>
<span id="cb393-2"><a href="interoperability-with-isolated-tools.html#cb393-2" tabindex="-1"></a><span class="fu">plotNicheClusterConnectivity</span>(<span class="at">gobject =</span> giotto_obj)</span></code></pre></div>
<p><img src="img/03_session5/4-nicheClusterConnectivity.png" width="60%" style="display: block; margin: auto;" /></p>
</div>
<div id="nt-niche-trajectory-score" class="section level4 hasAnchor" number="16.1.7.5">
<h4><span class="header-section-number">16.1.7.5</span> NT (niche trajectory) score<a href="interoperability-with-isolated-tools.html#nt-niche-trajectory-score" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb394"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb394-1"><a href="interoperability-with-isolated-tools.html#cb394-1" tabindex="-1"></a><span class="fu">spatPlot2D</span>(<span class="at">gobject =</span> giotto_obj, </span>
<span id="cb394-2"><a href="interoperability-with-isolated-tools.html#cb394-2" tabindex="-1"></a>           <span class="at">spat_unit =</span> <span class="st">&#39;cell&#39;</span>,</span>
<span id="cb394-3"><a href="interoperability-with-isolated-tools.html#cb394-3" tabindex="-1"></a>           <span class="at">feat_type =</span> <span class="st">&#39;rna&#39;</span>,</span>
<span id="cb394-4"><a href="interoperability-with-isolated-tools.html#cb394-4" tabindex="-1"></a>           <span class="at">group_by =</span> <span class="st">&#39;slice_id&#39;</span>,</span>
<span id="cb394-5"><a href="interoperability-with-isolated-tools.html#cb394-5" tabindex="-1"></a>           <span class="at">cell_color =</span> <span class="st">&quot;NTScore&quot;</span>,</span>
<span id="cb394-6"><a href="interoperability-with-isolated-tools.html#cb394-6" tabindex="-1"></a>           <span class="at">color_as_factor =</span> <span class="cn">FALSE</span>,</span>
<span id="cb394-7"><a href="interoperability-with-isolated-tools.html#cb394-7" tabindex="-1"></a>           <span class="at">cell_color_gradient =</span> <span class="st">&quot;turbo&quot;</span>,</span>
<span id="cb394-8"><a href="interoperability-with-isolated-tools.html#cb394-8" tabindex="-1"></a>           <span class="at">point_size =</span> <span class="dv">1</span>,</span>
<span id="cb394-9"><a href="interoperability-with-isolated-tools.html#cb394-9" tabindex="-1"></a>           <span class="at">point_border_stroke =</span> <span class="cn">NA</span></span>
<span id="cb394-10"><a href="interoperability-with-isolated-tools.html#cb394-10" tabindex="-1"></a>           )</span></code></pre></div>
<p><img src="img/03_session5/5-1-rawSpatialNTScore.png" width="80%" style="display: block; margin: auto;" /></p>
<p>We could change the direction of NT scores here.</p>
<div class="sourceCode" id="cb395"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb395-1"><a href="interoperability-with-isolated-tools.html#cb395-1" tabindex="-1"></a>giotto_obj<span class="sc">@</span>cell_metadata<span class="sc">$</span>cell<span class="sc">$</span>rna<span class="sc">$</span>NTScore <span class="ot">&lt;-</span> <span class="dv">1</span> <span class="sc">-</span> giotto_obj<span class="sc">@</span>cell_metadata<span class="sc">$</span>cell<span class="sc">$</span>rna<span class="sc">$</span>NTScore</span></code></pre></div>
<div class="sourceCode" id="cb396"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb396-1"><a href="interoperability-with-isolated-tools.html#cb396-1" tabindex="-1"></a><span class="fu">spatPlot2D</span>(<span class="at">gobject =</span> giotto_obj, </span>
<span id="cb396-2"><a href="interoperability-with-isolated-tools.html#cb396-2" tabindex="-1"></a>           <span class="at">spat_unit =</span> <span class="st">&#39;cell&#39;</span>,</span>
<span id="cb396-3"><a href="interoperability-with-isolated-tools.html#cb396-3" tabindex="-1"></a>           <span class="at">feat_type =</span> <span class="st">&#39;rna&#39;</span>,</span>
<span id="cb396-4"><a href="interoperability-with-isolated-tools.html#cb396-4" tabindex="-1"></a>           <span class="at">group_by =</span> <span class="st">&#39;slice_id&#39;</span>,</span>
<span id="cb396-5"><a href="interoperability-with-isolated-tools.html#cb396-5" tabindex="-1"></a>           <span class="at">cell_color =</span> <span class="st">&quot;NTScore&quot;</span>,</span>
<span id="cb396-6"><a href="interoperability-with-isolated-tools.html#cb396-6" tabindex="-1"></a>           <span class="at">color_as_factor =</span> <span class="cn">FALSE</span>,</span>
<span id="cb396-7"><a href="interoperability-with-isolated-tools.html#cb396-7" tabindex="-1"></a>           <span class="at">cell_color_gradient =</span> <span class="st">&quot;turbo&quot;</span>,</span>
<span id="cb396-8"><a href="interoperability-with-isolated-tools.html#cb396-8" tabindex="-1"></a>           <span class="at">point_size =</span> <span class="dv">1</span>,</span>
<span id="cb396-9"><a href="interoperability-with-isolated-tools.html#cb396-9" tabindex="-1"></a>           <span class="at">point_border_stroke =</span> <span class="cn">NA</span></span>
<span id="cb396-10"><a href="interoperability-with-isolated-tools.html#cb396-10" tabindex="-1"></a>           )</span></code></pre></div>
<p><img src="img/03_session5/5-2-newSpatialNTScore.png" width="80%" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb397"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb397-1"><a href="interoperability-with-isolated-tools.html#cb397-1" tabindex="-1"></a><span class="fu">plotCellTypeNTScore</span>(<span class="at">gobject =</span> giotto_obj,</span>
<span id="cb397-2"><a href="interoperability-with-isolated-tools.html#cb397-2" tabindex="-1"></a>                    <span class="at">cell_type =</span> <span class="st">&quot;subclass&quot;</span>,</span>
<span id="cb397-3"><a href="interoperability-with-isolated-tools.html#cb397-3" tabindex="-1"></a>                    <span class="at">values =</span> <span class="st">&quot;NTScore&quot;</span>,</span>
<span id="cb397-4"><a href="interoperability-with-isolated-tools.html#cb397-4" tabindex="-1"></a>                    <span class="at">spat_unit =</span> <span class="st">&quot;cell&quot;</span>,</span>
<span id="cb397-5"><a href="interoperability-with-isolated-tools.html#cb397-5" tabindex="-1"></a>                    <span class="at">feat_type =</span> <span class="st">&quot;rna&quot;</span>)</span></code></pre></div>
<p><img src="img/03_session5/6-violinCellTypeOverNTScore.png" width="80%" style="display: block; margin: auto;" /></p>
</div>
<div id="cell-type-composition-within-niche-cluster" class="section level4 hasAnchor" number="16.1.7.6">
<h4><span class="header-section-number">16.1.7.6</span> Cell type composition within niche cluster<a href="interoperability-with-isolated-tools.html#cell-type-composition-within-niche-cluster" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb398"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb398-1"><a href="interoperability-with-isolated-tools.html#cb398-1" tabindex="-1"></a><span class="fu">plotCTCompositionInNicheCluster</span>(<span class="at">gobject =</span> giotto_obj,</span>
<span id="cb398-2"><a href="interoperability-with-isolated-tools.html#cb398-2" tabindex="-1"></a>                                <span class="at">cell_type =</span> <span class="st">&quot;subclass&quot;</span>)</span></code></pre></div>
<p><img src="img/03_session5/7-heatmapCellTypeCompositionInNicheCluster.png" width="80%" style="display: block; margin: auto;" /></p>
</div>
</div>
</div>
<div id="session-info-3" class="section level2 hasAnchor" number="16.2">
<h2><span class="header-section-number">16.2</span> Session info<a href="interoperability-with-isolated-tools.html#session-info-3" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb399"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb399-1"><a href="interoperability-with-isolated-tools.html#cb399-1" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code></pre></div>
<div class="sourceCode" id="cb400"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb400-1"><a href="interoperability-with-isolated-tools.html#cb400-1" tabindex="-1"></a><span class="co"># R version 4.4.0 (2024-04-24)</span></span>
<span id="cb400-2"><a href="interoperability-with-isolated-tools.html#cb400-2" tabindex="-1"></a><span class="co"># Platform: aarch64-apple-darwin20</span></span>
<span id="cb400-3"><a href="interoperability-with-isolated-tools.html#cb400-3" tabindex="-1"></a><span class="co"># Running under: macOS Ventura 13.6.6</span></span>
<span id="cb400-4"><a href="interoperability-with-isolated-tools.html#cb400-4" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb400-5"><a href="interoperability-with-isolated-tools.html#cb400-5" tabindex="-1"></a><span class="co"># Matrix products: default</span></span>
<span id="cb400-6"><a href="interoperability-with-isolated-tools.html#cb400-6" tabindex="-1"></a><span class="co"># BLAS:   /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib </span></span>
<span id="cb400-7"><a href="interoperability-with-isolated-tools.html#cb400-7" tabindex="-1"></a><span class="co"># LAPACK: /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0</span></span>
<span id="cb400-8"><a href="interoperability-with-isolated-tools.html#cb400-8" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb400-9"><a href="interoperability-with-isolated-tools.html#cb400-9" tabindex="-1"></a><span class="co"># locale:</span></span>
<span id="cb400-10"><a href="interoperability-with-isolated-tools.html#cb400-10" tabindex="-1"></a><span class="co"># [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span></span>
<span id="cb400-11"><a href="interoperability-with-isolated-tools.html#cb400-11" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb400-12"><a href="interoperability-with-isolated-tools.html#cb400-12" tabindex="-1"></a><span class="co"># time zone: America/New_York</span></span>
<span id="cb400-13"><a href="interoperability-with-isolated-tools.html#cb400-13" tabindex="-1"></a><span class="co"># tzcode source: internal</span></span>
<span id="cb400-14"><a href="interoperability-with-isolated-tools.html#cb400-14" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb400-15"><a href="interoperability-with-isolated-tools.html#cb400-15" tabindex="-1"></a><span class="co"># attached base packages:</span></span>
<span id="cb400-16"><a href="interoperability-with-isolated-tools.html#cb400-16" tabindex="-1"></a><span class="co"># [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span id="cb400-17"><a href="interoperability-with-isolated-tools.html#cb400-17" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb400-18"><a href="interoperability-with-isolated-tools.html#cb400-18" tabindex="-1"></a><span class="co"># other attached packages:</span></span>
<span id="cb400-19"><a href="interoperability-with-isolated-tools.html#cb400-19" tabindex="-1"></a><span class="co"># [1] ggraph_2.2.1      ggplot2_3.5.1     reticulate_1.37.0 Giotto_4.1.0      GiottoClass_0.3.2</span></span>
<span id="cb400-20"><a href="interoperability-with-isolated-tools.html#cb400-20" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb400-21"><a href="interoperability-with-isolated-tools.html#cb400-21" tabindex="-1"></a><span class="co"># loaded via a namespace (and not attached):</span></span>
<span id="cb400-22"><a href="interoperability-with-isolated-tools.html#cb400-22" tabindex="-1"></a><span class="co">#  [1] tidyselect_1.2.1    viridisLite_0.4.2   dplyr_1.1.4         farver_2.1.2        GiottoVisuals_0.2.4 viridis_0.6.5       fastmap_1.2.0       lazyeval_0.2.2      tweenr_2.0.3        digest_0.6.35       lifecycle_1.0.4    </span></span>
<span id="cb400-23"><a href="interoperability-with-isolated-tools.html#cb400-23" tabindex="-1"></a><span class="co"># [12] terra_1.7-78        magrittr_2.0.3      dbscan_1.1-12       compiler_4.4.0      rlang_1.1.4         tools_4.4.0         igraph_2.0.3        utf8_1.2.4          yaml_2.3.8          data.table_1.15.4   knitr_1.47         </span></span>
<span id="cb400-24"><a href="interoperability-with-isolated-tools.html#cb400-24" tabindex="-1"></a><span class="co"># [23] labeling_0.4.3      graphlayouts_1.1.1  htmlwidgets_1.6.4   sp_2.1-4            plyr_1.8.9          RColorBrewer_1.1-3  withr_3.0.0         purrr_1.0.2         grid_4.4.0          polyclip_1.10-6     fansi_1.0.6        </span></span>
<span id="cb400-25"><a href="interoperability-with-isolated-tools.html#cb400-25" tabindex="-1"></a><span class="co"># [34] colorspace_2.1-0    scales_1.3.0        gtools_3.9.5        MASS_7.3-60.2       cli_3.6.2           rmarkdown_2.27      generics_0.1.3      rstudioapi_0.16.0   httr_1.4.7          reshape2_1.4.4      cachem_1.1.0       </span></span>
<span id="cb400-26"><a href="interoperability-with-isolated-tools.html#cb400-26" tabindex="-1"></a><span class="co"># [45] ggforce_0.4.2       stringr_1.5.1       parallel_4.4.0      matrixStats_1.3.0   vctrs_0.6.5         Matrix_1.7-0        jsonlite_1.8.8      bookdown_0.40       ggrepel_0.9.5       scattermore_1.2     magick_2.8.3       </span></span>
<span id="cb400-27"><a href="interoperability-with-isolated-tools.html#cb400-27" tabindex="-1"></a><span class="co"># [56] GiottoUtils_0.1.10  plotly_4.10.4       tidyr_1.3.1         glue_1.7.0          codetools_0.2-20    cowplot_1.1.3       stringi_1.8.4       gtable_0.3.5        deldir_2.0-4        munsell_0.5.1       tibble_3.2.1       </span></span>
<span id="cb400-28"><a href="interoperability-with-isolated-tools.html#cb400-28" tabindex="-1"></a><span class="co"># [67] pillar_1.9.0        htmltools_0.5.8.1   R6_2.5.1            tidygraph_1.3.1     evaluate_0.24.0     lattice_0.22-6      png_0.1-8           backports_1.5.0     memoise_2.0.1       Rcpp_1.0.12         gridExtra_2.3      </span></span>
<span id="cb400-29"><a href="interoperability-with-isolated-tools.html#cb400-29" tabindex="-1"></a><span class="co"># [78] checkmate_2.3.1     colorRamp2_0.1.0    xfun_0.44           pkgconfig_2.0.3    </span></span></code></pre></div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="interoperability-with-other-frameworks.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="interactivity-with-the-rspatial-ecosystem.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
