<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>14 Multi-omics integration | Workshop: Spatial multi-omics data analysis with Giotto Suite</title>
  <meta name="description" content="14 Multi-omics integration | Workshop: Spatial multi-omics data analysis with Giotto Suite" />
  <meta name="generator" content="bookdown 0.39 and GitBook 2.6.7" />

  <meta property="og:title" content="14 Multi-omics integration | Workshop: Spatial multi-omics data analysis with Giotto Suite" />
  <meta property="og:type" content="book" />
  
  
  <meta name="github-repo" content="drieslab/giotto_workshop_2024" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="14 Multi-omics integration | Workshop: Spatial multi-omics data analysis with Giotto Suite" />
  
  
  

<meta name="author" content="Ruben Dries, Jiaji George Chen, Joselyn Cristina ChÃ¡vez-Fuentes, Junxiang Xu ,Edward Ruiz, Jeff Sheridan, Iqra Amin, Wen Wang" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="spatial-multi-modal-analysis.html"/>
<link rel="next" href="interoperability-with-other-frameworks.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Giotto Suite Workshop 2024</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#instructors"><i class="fa fa-check"></i><b>1.1</b> Instructors</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#topics-and-schedule"><i class="fa fa-check"></i><b>1.2</b> Topics and Schedule:</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#license"><i class="fa fa-check"></i><b>1.3</b> License</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="datasets-packages.html"><a href="datasets-packages.html"><i class="fa fa-check"></i><b>2</b> Datasets &amp; Packages</a>
<ul>
<li class="chapter" data-level="2.1" data-path="datasets-packages.html"><a href="datasets-packages.html#datasets-to-download"><i class="fa fa-check"></i><b>2.1</b> Datasets to download</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="datasets-packages.html"><a href="datasets-packages.html#zenodo-repository"><i class="fa fa-check"></i><b>2.1.1</b> Zenodo repository</a></li>
<li class="chapter" data-level="2.1.2" data-path="datasets-packages.html"><a href="datasets-packages.html#x-genomics-visium-mouse-brain-section-coronal-dataset"><i class="fa fa-check"></i><b>2.1.2</b> 10X Genomics Visium Mouse Brain Section (Coronal) dataset</a></li>
<li class="chapter" data-level="2.1.3" data-path="datasets-packages.html"><a href="datasets-packages.html#x-genomics-visium-hd-ffpe-human-colon-cancer"><i class="fa fa-check"></i><b>2.1.3</b> 10X Genomics Visium HD: FFPE Human Colon Cancer</a></li>
<li class="chapter" data-level="2.1.4" data-path="datasets-packages.html"><a href="datasets-packages.html#x-genomics-multi-modal-dataset"><i class="fa fa-check"></i><b>2.1.4</b> 10X Genomics multi-modal dataset</a></li>
<li class="chapter" data-level="2.1.5" data-path="datasets-packages.html"><a href="datasets-packages.html#x-genomics-multi-omics-visium-cytassist-human-tonsil-dataset"><i class="fa fa-check"></i><b>2.1.5</b> 10X Genomics multi-omics Visium CytAssist Human Tonsil dataset</a></li>
<li class="chapter" data-level="2.1.6" data-path="datasets-packages.html"><a href="datasets-packages.html#x-genomics-human-prostate-cancer-adenocarcinoma-with-invasive-carcinoma-ffpe"><i class="fa fa-check"></i><b>2.1.6</b> 10X Genomics Human Prostate Cancer Adenocarcinoma with Invasive Carcinoma (FFPE)</a></li>
<li class="chapter" data-level="2.1.7" data-path="datasets-packages.html"><a href="datasets-packages.html#x-genomics-normal-human-prostate-ffpe"><i class="fa fa-check"></i><b>2.1.7</b> 10X Genomics Normal Human Prostate (FFPE)</a></li>
<li class="chapter" data-level="2.1.8" data-path="datasets-packages.html"><a href="datasets-packages.html#xenium"><i class="fa fa-check"></i><b>2.1.8</b> Xenium</a></li>
<li class="chapter" data-level="2.1.9" data-path="datasets-packages.html"><a href="datasets-packages.html#merfish-cortex-dataset"><i class="fa fa-check"></i><b>2.1.9</b> MERFISH cortex dataset</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="datasets-packages.html"><a href="datasets-packages.html#needed-packages"><i class="fa fa-check"></i><b>2.2</b> Needed packages</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="datasets-packages.html"><a href="datasets-packages.html#cran-dependencies"><i class="fa fa-check"></i><b>2.2.1</b> CRAN dependencies:</a></li>
<li class="chapter" data-level="2.2.2" data-path="datasets-packages.html"><a href="datasets-packages.html#terra-installation"><i class="fa fa-check"></i><b>2.2.2</b> <em>terra</em> installation</a></li>
<li class="chapter" data-level="2.2.3" data-path="datasets-packages.html"><a href="datasets-packages.html#matrix-installation"><i class="fa fa-check"></i><b>2.2.3</b> <em>Matrix</em> installation</a></li>
<li class="chapter" data-level="2.2.4" data-path="datasets-packages.html"><a href="datasets-packages.html#rtools-installation"><i class="fa fa-check"></i><b>2.2.4</b> <em>Rtools</em> installation</a></li>
<li class="chapter" data-level="2.2.5" data-path="datasets-packages.html"><a href="datasets-packages.html#giotto-installation"><i class="fa fa-check"></i><b>2.2.5</b> <em>Giotto</em> installation</a></li>
<li class="chapter" data-level="2.2.6" data-path="datasets-packages.html"><a href="datasets-packages.html#irlba-install"><i class="fa fa-check"></i><b>2.2.6</b> <em>irlba</em> install</a></li>
<li class="chapter" data-level="2.2.7" data-path="datasets-packages.html"><a href="datasets-packages.html#arrow-install"><i class="fa fa-check"></i><b>2.2.7</b> <em>arrow</em> install</a></li>
<li class="chapter" data-level="2.2.8" data-path="datasets-packages.html"><a href="datasets-packages.html#bioconductor-dependencies"><i class="fa fa-check"></i><b>2.2.8</b> Bioconductor dependencies:</a></li>
<li class="chapter" data-level="2.2.9" data-path="datasets-packages.html"><a href="datasets-packages.html#cran-packages"><i class="fa fa-check"></i><b>2.2.9</b> CRAN packages:</a></li>
<li class="chapter" data-level="2.2.10" data-path="datasets-packages.html"><a href="datasets-packages.html#packages-from-github"><i class="fa fa-check"></i><b>2.2.10</b> Packages from GitHub</a></li>
<li class="chapter" data-level="2.2.11" data-path="datasets-packages.html"><a href="datasets-packages.html#python-environments"><i class="fa fa-check"></i><b>2.2.11</b> Python environments</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html"><i class="fa fa-check"></i><b>3</b> Spatial omics technologies</a>
<ul>
<li class="chapter" data-level="3.1" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#presentation"><i class="fa fa-check"></i><b>3.1</b> Presentation</a></li>
<li class="chapter" data-level="3.2" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#short-summary"><i class="fa fa-check"></i><b>3.2</b> Short summary</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#why-do-we-need-spatial-omics-technologies"><i class="fa fa-check"></i><b>3.2.1</b> Why do we need spatial omics technologies?</a></li>
<li class="chapter" data-level="3.2.2" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#what-is-spatial-omics"><i class="fa fa-check"></i><b>3.2.2</b> What is spatial omics?</a></li>
<li class="chapter" data-level="3.2.3" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#what-are-the-main-spatial-omics-technologies"><i class="fa fa-check"></i><b>3.2.3</b> What are the main spatial omics technologies?</a></li>
<li class="chapter" data-level="3.2.4" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#other-spatial-omics-atac-seq-cuttag-lipidomics-etc"><i class="fa fa-check"></i><b>3.2.4</b> Other Spatial omics: ATAC-seq, CUT&amp;Tag, lipidomics, etc</a></li>
<li class="chapter" data-level="3.2.5" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#what-are-the-different-types-of-spatial-downstream-analyses"><i class="fa fa-check"></i><b>3.2.5</b> What are the different types of spatial downstream analyses?</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html"><i class="fa fa-check"></i><b>4</b> Introduction to the Giotto package</a>
<ul>
<li class="chapter" data-level="4.1" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html#presentation-1"><i class="fa fa-check"></i><b>4.1</b> Presentation</a></li>
<li class="chapter" data-level="4.2" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html#ecosystem"><i class="fa fa-check"></i><b>4.2</b> Ecosystem</a></li>
<li class="chapter" data-level="4.3" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html#installation-python-environment"><i class="fa fa-check"></i><b>4.3</b> Installation + python environment</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html#giotto-installation-1"><i class="fa fa-check"></i><b>4.3.1</b> Giotto installation</a></li>
<li class="chapter" data-level="4.3.2" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html#python-environment"><i class="fa fa-check"></i><b>4.3.2</b> Python environment</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html#giotto-instructions"><i class="fa fa-check"></i><b>4.4</b> Giotto instructions</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html"><i class="fa fa-check"></i><b>5</b> Data formatting and Pre-processing</a>
<ul>
<li class="chapter" data-level="5.1" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#data-formats"><i class="fa fa-check"></i><b>5.1</b> Data formats</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#general-formats"><i class="fa fa-check"></i><b>5.1.1</b> General formats</a></li>
<li class="chapter" data-level="5.1.2" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#matrix-formats"><i class="fa fa-check"></i><b>5.1.2</b> Matrix formats</a></li>
<li class="chapter" data-level="5.1.3" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#tabular-formats"><i class="fa fa-check"></i><b>5.1.3</b> Tabular formats</a></li>
<li class="chapter" data-level="5.1.4" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#spatial-formats"><i class="fa fa-check"></i><b>5.1.4</b> Spatial formats</a></li>
<li class="chapter" data-level="5.1.5" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#mask-files"><i class="fa fa-check"></i><b>5.1.5</b> Mask files</a></li>
<li class="chapter" data-level="5.1.6" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#images"><i class="fa fa-check"></i><b>5.1.6</b> images</a></li>
<li class="chapter" data-level="5.1.7" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#jsonlike-formats"><i class="fa fa-check"></i><b>5.1.7</b> jsonlike formats</a></li>
<li class="chapter" data-level="5.1.8" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#hierarchical-formats"><i class="fa fa-check"></i><b>5.1.8</b> Hierarchical formats</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#pre-processing"><i class="fa fa-check"></i><b>5.2</b> Pre-processing</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#expression-matrix"><i class="fa fa-check"></i><b>5.2.1</b> Expression matrix</a></li>
<li class="chapter" data-level="5.2.2" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#spatial-locations"><i class="fa fa-check"></i><b>5.2.2</b> Spatial locations</a></li>
<li class="chapter" data-level="5.2.3" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#giottopoints"><i class="fa fa-check"></i><b>5.2.3</b> giottoPoints</a></li>
<li class="chapter" data-level="5.2.4" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#giottopolygon"><i class="fa fa-check"></i><b>5.2.4</b> giottoPolygon</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#subobject-utility-functions"><i class="fa fa-check"></i><b>5.3</b> Subobject utility functions</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#ids"><i class="fa fa-check"></i><b>5.3.1</b> IDs</a></li>
<li class="chapter" data-level="5.3.2" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#bracket-subsetting-and-extraction"><i class="fa fa-check"></i><b>5.3.2</b> Bracket subsetting and extraction</a></li>
<li class="chapter" data-level="5.3.3" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#nesting-metadata-generics"><i class="fa fa-check"></i><b>5.3.3</b> Nesting metadata generics</a></li>
<li class="chapter" data-level="5.3.4" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#appending-to-a-giotto-object"><i class="fa fa-check"></i><b>5.3.4</b> Appending to a giotto object</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html"><i class="fa fa-check"></i><b>6</b> Creating a Giotto object</a>
<ul>
<li class="chapter" data-level="6.0.1" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#used-packages"><i class="fa fa-check"></i><b>6.0.1</b> Used packages</a></li>
<li class="chapter" data-level="6.1" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#overview"><i class="fa fa-check"></i><b>6.1</b> Overview</a></li>
<li class="chapter" data-level="6.2" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#from-matrix-locations"><i class="fa fa-check"></i><b>6.2</b> From matrix + locations</a></li>
<li class="chapter" data-level="6.3" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#from-subcellular-raw-data-transcripts-or-images-polygons"><i class="fa fa-check"></i><b>6.3</b> From subcellular raw data (transcripts or images) + polygons</a></li>
<li class="chapter" data-level="6.4" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#from-piece-wise"><i class="fa fa-check"></i><b>6.4</b> From piece-wise</a></li>
<li class="chapter" data-level="6.5" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#using-convenience-functions-for-popular-technologies-vizgen-xenium-cosmx"><i class="fa fa-check"></i><b>6.5</b> Using convenience functions for popular technologies (Vizgen, Xenium, CosMx, â¦)</a></li>
<li class="chapter" data-level="6.6" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#spatial-plots"><i class="fa fa-check"></i><b>6.6</b> Spatial plots</a></li>
<li class="chapter" data-level="6.7" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#subsetting"><i class="fa fa-check"></i><b>6.7</b> Subsetting</a></li>
<li class="chapter" data-level="6.8" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#mini-objects-giottodata"><i class="fa fa-check"></i><b>6.8</b> Mini objects &amp; GiottoData</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="visium-part-i.html"><a href="visium-part-i.html"><i class="fa fa-check"></i><b>7</b> Visium Part I</a>
<ul>
<li class="chapter" data-level="7.1" data-path="visium-part-i.html"><a href="visium-part-i.html#the-visium-technology"><i class="fa fa-check"></i><b>7.1</b> The Visium technology</a></li>
<li class="chapter" data-level="7.2" data-path="visium-part-i.html"><a href="visium-part-i.html#introduction-to-the-spatial-dataset"><i class="fa fa-check"></i><b>7.2</b> Introduction to the spatial dataset</a></li>
<li class="chapter" data-level="7.3" data-path="visium-part-i.html"><a href="visium-part-i.html#download-dataset"><i class="fa fa-check"></i><b>7.3</b> Download dataset</a></li>
<li class="chapter" data-level="7.4" data-path="visium-part-i.html"><a href="visium-part-i.html#create-the-giotto-object"><i class="fa fa-check"></i><b>7.4</b> Create the Giotto object</a></li>
<li class="chapter" data-level="7.5" data-path="visium-part-i.html"><a href="visium-part-i.html#subset-on-spots-that-were-covered-by-tissue"><i class="fa fa-check"></i><b>7.5</b> Subset on spots that were covered by tissue</a></li>
<li class="chapter" data-level="7.6" data-path="visium-part-i.html"><a href="visium-part-i.html#quality-control"><i class="fa fa-check"></i><b>7.6</b> Quality control</a></li>
<li class="chapter" data-level="7.7" data-path="visium-part-i.html"><a href="visium-part-i.html#filtering"><i class="fa fa-check"></i><b>7.7</b> Filtering</a></li>
<li class="chapter" data-level="7.8" data-path="visium-part-i.html"><a href="visium-part-i.html#normalization"><i class="fa fa-check"></i><b>7.8</b> Normalization</a></li>
<li class="chapter" data-level="7.9" data-path="visium-part-i.html"><a href="visium-part-i.html#feature-selection"><i class="fa fa-check"></i><b>7.9</b> Feature selection</a>
<ul>
<li class="chapter" data-level="7.9.1" data-path="visium-part-i.html"><a href="visium-part-i.html#highly-variable-features"><i class="fa fa-check"></i><b>7.9.1</b> Highly Variable Features:</a></li>
</ul></li>
<li class="chapter" data-level="7.10" data-path="visium-part-i.html"><a href="visium-part-i.html#dimension-reduction"><i class="fa fa-check"></i><b>7.10</b> Dimension Reduction</a>
<ul>
<li class="chapter" data-level="7.10.1" data-path="visium-part-i.html"><a href="visium-part-i.html#pca"><i class="fa fa-check"></i><b>7.10.1</b> PCA</a></li>
<li class="chapter" data-level="7.10.2" data-path="visium-part-i.html"><a href="visium-part-i.html#umap"><i class="fa fa-check"></i><b>7.10.2</b> UMAP</a></li>
<li class="chapter" data-level="7.10.3" data-path="visium-part-i.html"><a href="visium-part-i.html#t-sne"><i class="fa fa-check"></i><b>7.10.3</b> t-SNE</a></li>
</ul></li>
<li class="chapter" data-level="7.11" data-path="visium-part-i.html"><a href="visium-part-i.html#clustering"><i class="fa fa-check"></i><b>7.11</b> Clustering</a>
<ul>
<li class="chapter" data-level="7.11.1" data-path="visium-part-i.html"><a href="visium-part-i.html#calculate-leiden-clustering"><i class="fa fa-check"></i><b>7.11.1</b> Calculate Leiden clustering</a></li>
<li class="chapter" data-level="7.11.2" data-path="visium-part-i.html"><a href="visium-part-i.html#calculate-louvain-clustering"><i class="fa fa-check"></i><b>7.11.2</b> Calculate Louvain clustering</a></li>
</ul></li>
<li class="chapter" data-level="7.12" data-path="visium-part-i.html"><a href="visium-part-i.html#save-the-object"><i class="fa fa-check"></i><b>7.12</b> Save the object</a></li>
<li class="chapter" data-level="7.13" data-path="visium-part-i.html"><a href="visium-part-i.html#session-info"><i class="fa fa-check"></i><b>7.13</b> Session info</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="visium-part-ii.html"><a href="visium-part-ii.html"><i class="fa fa-check"></i><b>8</b> Visium Part II</a>
<ul>
<li class="chapter" data-level="8.1" data-path="visium-part-ii.html"><a href="visium-part-ii.html#load-the-object"><i class="fa fa-check"></i><b>8.1</b> Load the object</a></li>
<li class="chapter" data-level="8.2" data-path="visium-part-ii.html"><a href="visium-part-ii.html#differential-expression"><i class="fa fa-check"></i><b>8.2</b> Differential expression</a>
<ul>
<li class="chapter" data-level="8.2.1" data-path="visium-part-ii.html"><a href="visium-part-ii.html#gini-markers"><i class="fa fa-check"></i><b>8.2.1</b> Gini markers</a></li>
<li class="chapter" data-level="8.2.2" data-path="visium-part-ii.html"><a href="visium-part-ii.html#scran-markers"><i class="fa fa-check"></i><b>8.2.2</b> Scran markers</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="visium-part-ii.html"><a href="visium-part-ii.html#enrichment-deconvolution"><i class="fa fa-check"></i><b>8.3</b> Enrichment &amp; Deconvolution</a>
<ul>
<li class="chapter" data-level="8.3.1" data-path="visium-part-ii.html"><a href="visium-part-ii.html#pagerank"><i class="fa fa-check"></i><b>8.3.1</b> PAGE/Rank</a></li>
<li class="chapter" data-level="8.3.2" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatialdwls"><i class="fa fa-check"></i><b>8.3.2</b> SpatialDWLS</a></li>
</ul></li>
<li class="chapter" data-level="8.4" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatial-expression-patterns"><i class="fa fa-check"></i><b>8.4</b> Spatial expression patterns</a>
<ul>
<li class="chapter" data-level="8.4.1" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatial-variable-genes"><i class="fa fa-check"></i><b>8.4.1</b> Spatial variable genes</a></li>
<li class="chapter" data-level="8.4.2" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatial-co-expression-modules"><i class="fa fa-check"></i><b>8.4.2</b> Spatial co-expression modules</a></li>
</ul></li>
<li class="chapter" data-level="8.5" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatially-informed-clusters"><i class="fa fa-check"></i><b>8.5</b> Spatially informed clusters</a></li>
<li class="chapter" data-level="8.6" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatial-domains-hmrf"><i class="fa fa-check"></i><b>8.6</b> Spatial domains HMRF</a></li>
<li class="chapter" data-level="8.7" data-path="visium-part-ii.html"><a href="visium-part-ii.html#interactive-tools"><i class="fa fa-check"></i><b>8.7</b> Interactive tools</a></li>
<li class="chapter" data-level="8.8" data-path="visium-part-ii.html"><a href="visium-part-ii.html#session-info-1"><i class="fa fa-check"></i><b>8.8</b> Session info</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="visium-hd.html"><a href="visium-hd.html"><i class="fa fa-check"></i><b>9</b> Visium HD</a>
<ul>
<li class="chapter" data-level="9.1" data-path="visium-hd.html"><a href="visium-hd.html#objective"><i class="fa fa-check"></i><b>9.1</b> Objective</a></li>
<li class="chapter" data-level="9.2" data-path="visium-hd.html"><a href="visium-hd.html#background"><i class="fa fa-check"></i><b>9.2</b> Background</a>
<ul>
<li class="chapter" data-level="9.2.1" data-path="visium-hd.html"><a href="visium-hd.html#visium-hd-technology"><i class="fa fa-check"></i><b>9.2.1</b> Visium HD Technology</a></li>
<li class="chapter" data-level="9.2.2" data-path="visium-hd.html"><a href="visium-hd.html#colorectal-cancer-sample"><i class="fa fa-check"></i><b>9.2.2</b> Colorectal Cancer Sample</a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="visium-hd.html"><a href="visium-hd.html#data-ingestion"><i class="fa fa-check"></i><b>9.3</b> Data Ingestion</a>
<ul>
<li class="chapter" data-level="9.3.1" data-path="visium-hd.html"><a href="visium-hd.html#visium-hd-output-data-format"><i class="fa fa-check"></i><b>9.3.1</b> Visium HD output data format</a></li>
<li class="chapter" data-level="9.3.2" data-path="visium-hd.html"><a href="visium-hd.html#read-in-raw-data"><i class="fa fa-check"></i><b>9.3.2</b> Read in raw data</a></li>
<li class="chapter" data-level="9.3.3" data-path="visium-hd.html"><a href="visium-hd.html#giotto-visium-hd-convenience-function"><i class="fa fa-check"></i><b>9.3.3</b> Giotto Visium HD convenience function</a></li>
</ul></li>
<li class="chapter" data-level="9.4" data-path="visium-hd.html"><a href="visium-hd.html#tiling-and-aggregation"><i class="fa fa-check"></i><b>9.4</b> Tiling and aggregation</a></li>
<li class="chapter" data-level="9.5" data-path="visium-hd.html"><a href="visium-hd.html#scalability-and-projection-functions"><i class="fa fa-check"></i><b>9.5</b> Scalability and projection functions</a></li>
<li class="chapter" data-level="9.6" data-path="visium-hd.html"><a href="visium-hd.html#spatial-expression-patterns-1"><i class="fa fa-check"></i><b>9.6</b> Spatial expression patterns</a></li>
<li class="chapter" data-level="9.7" data-path="visium-hd.html"><a href="visium-hd.html#spatial-co-expression-modules-1"><i class="fa fa-check"></i><b>9.7</b> Spatial co-expression modules</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="xenium-1.html"><a href="xenium-1.html"><i class="fa fa-check"></i><b>10</b> Xenium</a>
<ul>
<li class="chapter" data-level="10.1" data-path="xenium-1.html"><a href="xenium-1.html#introduction-to-spatial-dataset"><i class="fa fa-check"></i><b>10.1</b> Introduction to spatial dataset</a>
<ul>
<li class="chapter" data-level="10.1.1" data-path="xenium-1.html"><a href="xenium-1.html#output-directory-structure"><i class="fa fa-check"></i><b>10.1.1</b> Output directory structure</a></li>
<li class="chapter" data-level="10.1.2" data-path="xenium-1.html"><a href="xenium-1.html#mini-xenium-dataset"><i class="fa fa-check"></i><b>10.1.2</b> Mini Xenium Dataset</a></li>
</ul></li>
<li class="chapter" data-level="10.2" data-path="xenium-1.html"><a href="xenium-1.html#data-prep"><i class="fa fa-check"></i><b>10.2</b> Data prep</a>
<ul>
<li class="chapter" data-level="10.2.1" data-path="xenium-1.html"><a href="xenium-1.html#image-conversion-may-change"><i class="fa fa-check"></i><b>10.2.1</b> Image conversion (may change)</a></li>
</ul></li>
<li class="chapter" data-level="10.3" data-path="xenium-1.html"><a href="xenium-1.html#convenience-function"><i class="fa fa-check"></i><b>10.3</b> Convenience function</a>
<ul>
<li class="chapter" data-level="10.3.1" data-path="xenium-1.html"><a href="xenium-1.html#specific-filepaths"><i class="fa fa-check"></i><b>10.3.1</b> Specific filepaths</a></li>
<li class="chapter" data-level="10.3.2" data-path="xenium-1.html"><a href="xenium-1.html#quality-value"><i class="fa fa-check"></i><b>10.3.2</b> Quality value</a></li>
<li class="chapter" data-level="10.3.3" data-path="xenium-1.html"><a href="xenium-1.html#transcript-type-splitting"><i class="fa fa-check"></i><b>10.3.3</b> Transcript type splitting</a></li>
<li class="chapter" data-level="10.3.4" data-path="xenium-1.html"><a href="xenium-1.html#centroids-calculation"><i class="fa fa-check"></i><b>10.3.4</b> Centroids calculation</a></li>
<li class="chapter" data-level="10.3.5" data-path="xenium-1.html"><a href="xenium-1.html#simple-visualization"><i class="fa fa-check"></i><b>10.3.5</b> Simple visualization</a></li>
</ul></li>
<li class="chapter" data-level="10.4" data-path="xenium-1.html"><a href="xenium-1.html#piecewise-loading"><i class="fa fa-check"></i><b>10.4</b> Piecewise loading</a>
<ul>
<li class="chapter" data-level="10.4.1" data-path="xenium-1.html"><a href="xenium-1.html#load-giottopoints-transcripts"><i class="fa fa-check"></i><b>10.4.1</b> load giottoPoints transcripts</a></li>
<li class="chapter" data-level="10.4.2" data-path="xenium-1.html"><a href="xenium-1.html#optional-loading-pre-aggregated-data"><i class="fa fa-check"></i><b>10.4.2</b> (optional) Loading pre-aggregated data</a></li>
</ul></li>
<li class="chapter" data-level="10.5" data-path="xenium-1.html"><a href="xenium-1.html#xenium-images"><i class="fa fa-check"></i><b>10.5</b> Xenium Images</a>
<ul>
<li class="chapter" data-level="10.5.1" data-path="xenium-1.html"><a href="xenium-1.html#image-metadata"><i class="fa fa-check"></i><b>10.5.1</b> Image metadata</a></li>
<li class="chapter" data-level="10.5.2" data-path="xenium-1.html"><a href="xenium-1.html#image-loading"><i class="fa fa-check"></i><b>10.5.2</b> Image loading</a></li>
</ul></li>
<li class="chapter" data-level="10.6" data-path="xenium-1.html"><a href="xenium-1.html#spatial-aggregation"><i class="fa fa-check"></i><b>10.6</b> Spatial aggregation</a></li>
<li class="chapter" data-level="10.7" data-path="xenium-1.html"><a href="xenium-1.html#aggregate-analyses-workflow"><i class="fa fa-check"></i><b>10.7</b> Aggregate analyses workflow</a>
<ul>
<li class="chapter" data-level="10.7.1" data-path="xenium-1.html"><a href="xenium-1.html#filtering-1"><i class="fa fa-check"></i><b>10.7.1</b> Filtering</a></li>
<li class="chapter" data-level="10.7.2" data-path="xenium-1.html"><a href="xenium-1.html#normalization-1"><i class="fa fa-check"></i><b>10.7.2</b> Normalization</a></li>
<li class="chapter" data-level="10.7.3" data-path="xenium-1.html"><a href="xenium-1.html#dimension-reduction-1"><i class="fa fa-check"></i><b>10.7.3</b> Dimension Reduction</a></li>
<li class="chapter" data-level="10.7.4" data-path="xenium-1.html"><a href="xenium-1.html#clustering-1"><i class="fa fa-check"></i><b>10.7.4</b> Clustering</a></li>
</ul></li>
<li class="chapter" data-level="10.8" data-path="xenium-1.html"><a href="xenium-1.html#niche-clustering"><i class="fa fa-check"></i><b>10.8</b> Niche clustering</a>
<ul>
<li class="chapter" data-level="10.8.1" data-path="xenium-1.html"><a href="xenium-1.html#spatial-network"><i class="fa fa-check"></i><b>10.8.1</b> Spatial network</a></li>
<li class="chapter" data-level="10.8.2" data-path="xenium-1.html"><a href="xenium-1.html#niche-calculation"><i class="fa fa-check"></i><b>10.8.2</b> Niche calculation</a></li>
<li class="chapter" data-level="10.8.3" data-path="xenium-1.html"><a href="xenium-1.html#kmeans-clustering-based-on-niche-signature"><i class="fa fa-check"></i><b>10.8.3</b> kmeans clustering based on niche signature</a></li>
</ul></li>
<li class="chapter" data-level="10.9" data-path="xenium-1.html"><a href="xenium-1.html#cell-proximity-enrichment"><i class="fa fa-check"></i><b>10.9</b> Cell proximity enrichment</a></li>
<li class="chapter" data-level="10.10" data-path="xenium-1.html"><a href="xenium-1.html#pseudovisium"><i class="fa fa-check"></i><b>10.10</b> Pseudovisium</a>
<ul>
<li class="chapter" data-level="10.10.1" data-path="xenium-1.html"><a href="xenium-1.html#pseudovisium-aggregation-and-workflow"><i class="fa fa-check"></i><b>10.10.1</b> Pseudovisium aggregation and workflow</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html"><i class="fa fa-check"></i><b>11</b> Spatial proteomics: Multiplexed Immunofluorescence</a>
<ul>
<li class="chapter" data-level="11.1" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#spatial-proteomics-technologies"><i class="fa fa-check"></i><b>11.1</b> Spatial Proteomics Technologies</a></li>
<li class="chapter" data-level="11.2" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#raw-data-type-coming-out-of-different-technologies"><i class="fa fa-check"></i><b>11.2</b> Raw data type coming out of different technologies</a>
<ul>
<li class="chapter" data-level="11.2.1" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#use-ome.tiff-as-an-example-output-data-to-begin-with"><i class="fa fa-check"></i><b>11.2.1</b> Use ome.tiff as an example output data to begin with</a></li>
<li class="chapter" data-level="11.2.2" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#use-single-channel-images-as-an-example-output-data-to-begin-with"><i class="fa fa-check"></i><b>11.2.2</b> Use single channel images as an example output data to begin with</a></li>
</ul></li>
<li class="chapter" data-level="11.3" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#cell-segmentation-file-to-get-single-cell-level-protein-expression"><i class="fa fa-check"></i><b>11.3</b> Cell Segmentation file to get single cell level protein expression</a>
<ul>
<li class="chapter" data-level="11.3.1" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#using-segmentation-output-file-from-deepcellmesmer-as-an-example."><i class="fa fa-check"></i><b>11.3.1</b> Using segmentation output file from DeepCell(mesmer) as an example.</a></li>
<li class="chapter" data-level="11.3.2" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#using-giotto-wrapper-of-cellpose-to-perform-segmentation"><i class="fa fa-check"></i><b>11.3.2</b> Using Giotto wrapper of Cellpose to perform segmentation</a></li>
</ul></li>
<li class="chapter" data-level="11.4" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#create-a-giotto-object-using-list-of-gitto-large-images-and-polygons"><i class="fa fa-check"></i><b>11.4</b> Create a Giotto Object using list of gitto large images and polygons</a>
<ul>
<li class="chapter" data-level="11.4.1" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#overlap-to-matrix"><i class="fa fa-check"></i><b>11.4.1</b> Overlap to matrix</a></li>
<li class="chapter" data-level="11.4.2" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#manipulate-expression-information"><i class="fa fa-check"></i><b>11.4.2</b> Manipulate Expression information</a></li>
<li class="chapter" data-level="11.4.3" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#rescale-polygons"><i class="fa fa-check"></i><b>11.4.3</b> Rescale polygons</a></li>
<li class="chapter" data-level="11.4.4" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#perform-clustering-and-differential-expression"><i class="fa fa-check"></i><b>11.4.4</b> Perform clustering and differential expression</a></li>
<li class="chapter" data-level="11.4.5" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#give-the-cluster-an-annotation-based-on-expression-values"><i class="fa fa-check"></i><b>11.4.5</b> Give the cluster an annotation based on expression values</a></li>
<li class="chapter" data-level="11.4.6" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#spatial-network-1"><i class="fa fa-check"></i><b>11.4.6</b> spatial network</a></li>
<li class="chapter" data-level="11.4.7" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#cell-neighborhood-cell-typecell-type-interactions"><i class="fa fa-check"></i><b>11.4.7</b> Cell Neighborhood: Cell-Type/Cell-Type Interactions</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="12" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html"><i class="fa fa-check"></i><b>12</b> Working with multiple samples</a>
<ul>
<li class="chapter" data-level="12.1" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#objective-1"><i class="fa fa-check"></i><b>12.1</b> Objective</a></li>
<li class="chapter" data-level="12.2" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#background-1"><i class="fa fa-check"></i><b>12.2</b> Background</a>
<ul>
<li class="chapter" data-level="12.2.1" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#dataset"><i class="fa fa-check"></i><b>12.2.1</b> Dataset</a></li>
<li class="chapter" data-level="12.2.2" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#visium-technology"><i class="fa fa-check"></i><b>12.2.2</b> Visium technology</a></li>
</ul></li>
<li class="chapter" data-level="12.3" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#create-individual-giotto-objects"><i class="fa fa-check"></i><b>12.3</b> Create individual giotto objects</a>
<ul>
<li class="chapter" data-level="12.3.1" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#download-the-data"><i class="fa fa-check"></i><b>12.3.1</b> Download the data</a></li>
<li class="chapter" data-level="12.3.2" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#create-giotto-instructions"><i class="fa fa-check"></i><b>12.3.2</b> Create giotto instructions</a></li>
<li class="chapter" data-level="12.3.3" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#load-visium-data-into-giotto"><i class="fa fa-check"></i><b>12.3.3</b> Load visium data into Giotto</a></li>
<li class="chapter" data-level="12.3.4" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#healthy-prostate-tissue-coverage"><i class="fa fa-check"></i><b>12.3.4</b> Healthy prostate tissue coverage</a></li>
<li class="chapter" data-level="12.3.5" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#adenocarcinoma-prostate-tissue-coverage"><i class="fa fa-check"></i><b>12.3.5</b> Adenocarcinoma prostate tissue coverage</a></li>
<li class="chapter" data-level="12.3.6" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#showing-the-data-strucutre-for-the-inidividual-objects"><i class="fa fa-check"></i><b>12.3.6</b> Showing the data strucutre for the inidividual objects</a></li>
</ul></li>
<li class="chapter" data-level="12.4" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#join-giotto-objects"><i class="fa fa-check"></i><b>12.4</b> Join Giotto Objects</a></li>
<li class="chapter" data-level="12.5" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#visualizing-combined-datasets"><i class="fa fa-check"></i><b>12.5</b> Visualizing combined datasets</a>
<ul>
<li class="chapter" data-level="12.5.1" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#vizualizing-in-the-same-plot"><i class="fa fa-check"></i><b>12.5.1</b> Vizualizing in the same plot</a></li>
<li class="chapter" data-level="12.5.2" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#vizualizing-on-separate-plots"><i class="fa fa-check"></i><b>12.5.2</b> Vizualizing on separate plots</a></li>
</ul></li>
<li class="chapter" data-level="12.6" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#splitting-combined-dataset"><i class="fa fa-check"></i><b>12.6</b> Splitting combined dataset</a></li>
<li class="chapter" data-level="12.7" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#analyzing-joined-objects"><i class="fa fa-check"></i><b>12.7</b> Analyzing joined objects</a>
<ul>
<li class="chapter" data-level="12.7.1" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#normalization-and-adding-statistics"><i class="fa fa-check"></i><b>12.7.1</b> Normalization and adding statistics</a></li>
<li class="chapter" data-level="12.7.2" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#clustering-the-datasets"><i class="fa fa-check"></i><b>12.7.2</b> Clustering the datasets</a></li>
<li class="chapter" data-level="12.7.3" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#vizualizing-spatial-location-of-clusters"><i class="fa fa-check"></i><b>12.7.3</b> Vizualizing spatial location of clusters</a></li>
<li class="chapter" data-level="12.7.4" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#vizualizing-tissue-contribution-to-clusters"><i class="fa fa-check"></i><b>12.7.4</b> Vizualizing tissue contribution to clusters</a></li>
</ul></li>
<li class="chapter" data-level="12.8" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#perform-harmony-and-default-workflows"><i class="fa fa-check"></i><b>12.8</b> Perform Harmony and default workflows</a>
<ul>
<li class="chapter" data-level="12.8.1" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#clustering-harmonized-object"><i class="fa fa-check"></i><b>12.8.1</b> Clustering harmonized object</a></li>
<li class="chapter" data-level="12.8.2" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#vizualizing-the-tissue-contribution"><i class="fa fa-check"></i><b>12.8.2</b> Vizualizing the tissue contribution</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="13" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html"><i class="fa fa-check"></i><b>13</b> Spatial multi-modal analysis</a>
<ul>
<li class="chapter" data-level="13.1" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#overview-1"><i class="fa fa-check"></i><b>13.1</b> Overview</a></li>
<li class="chapter" data-level="13.2" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#spatial-manipulation"><i class="fa fa-check"></i><b>13.2</b> Spatial manipulation</a>
<ul>
<li class="chapter" data-level="13.2.1" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#spatial-transforms"><i class="fa fa-check"></i><b>13.2.1</b> Spatial transforms:</a></li>
<li class="chapter" data-level="13.2.2" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#spatial-utilities"><i class="fa fa-check"></i><b>13.2.2</b> Spatial utilities:</a></li>
<li class="chapter" data-level="13.2.3" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#spatial-classes"><i class="fa fa-check"></i><b>13.2.3</b> Spatial classes:</a></li>
</ul></li>
<li class="chapter" data-level="13.3" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#examples-of-the-simple-transforms-with-a-giottopolygon"><i class="fa fa-check"></i><b>13.3</b> Examples of the simple transforms with a giottoPolygon</a></li>
<li class="chapter" data-level="13.4" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#affine-transforms"><i class="fa fa-check"></i><b>13.4</b> Affine transforms</a></li>
<li class="chapter" data-level="13.5" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#image-transforms"><i class="fa fa-check"></i><b>13.5</b> Image transforms</a></li>
<li class="chapter" data-level="13.6" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#the-practical-usage-of-multi-modality-co-registration"><i class="fa fa-check"></i><b>13.6</b> The practical usage of multi-modality co-registration</a>
<ul>
<li class="chapter" data-level="13.6.1" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#example-dataset-xenium-breast-cancer-pre-release-pack"><i class="fa fa-check"></i><b>13.6.1</b> Example dataset: Xenium Breast Cancer pre-release pack</a></li>
<li class="chapter" data-level="13.6.2" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#target-coordinate-system"><i class="fa fa-check"></i><b>13.6.2</b> Target Coordinate system</a></li>
<li class="chapter" data-level="13.6.3" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#visium-to-register"><i class="fa fa-check"></i><b>13.6.3</b> Visium to register</a></li>
<li class="chapter" data-level="13.6.4" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#register-post-xenium-he-and-if-image"><i class="fa fa-check"></i><b>13.6.4</b> Register post-Xenium H&amp;E and IF image</a></li>
<li class="chapter" data-level="13.6.5" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#automatic-alignment-via-sift-feature-descriptor-matching-and-affine-transformation"><i class="fa fa-check"></i><b>13.6.5</b> Automatic alignment via SIFT feature descriptor matching and affine transformation</a></li>
<li class="chapter" data-level="13.6.6" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#final-notes"><i class="fa fa-check"></i><b>13.6.6</b> Final Notes</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="14" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html"><i class="fa fa-check"></i><b>14</b> Multi-omics integration</a>
<ul>
<li class="chapter" data-level="14.1" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#the-cytassist-technology"><i class="fa fa-check"></i><b>14.1</b> The CytAssist technology</a></li>
<li class="chapter" data-level="14.2" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#introduction-to-the-spatial-dataset-1"><i class="fa fa-check"></i><b>14.2</b> Introduction to the spatial dataset</a></li>
<li class="chapter" data-level="14.3" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#download-dataset-1"><i class="fa fa-check"></i><b>14.3</b> Download dataset</a></li>
<li class="chapter" data-level="14.4" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#create-the-giotto-object-1"><i class="fa fa-check"></i><b>14.4</b> Create the Giotto object</a></li>
<li class="chapter" data-level="14.5" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#subset-on-spots-that-were-covered-by-tissue-1"><i class="fa fa-check"></i><b>14.5</b> Subset on spots that were covered by tissue</a></li>
<li class="chapter" data-level="14.6" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#rna-processing"><i class="fa fa-check"></i><b>14.6</b> RNA processing</a></li>
<li class="chapter" data-level="14.7" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#protein-processing"><i class="fa fa-check"></i><b>14.7</b> Protein processing</a></li>
<li class="chapter" data-level="14.8" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#multi-omics-integration-1"><i class="fa fa-check"></i><b>14.8</b> Multi-omics integration</a></li>
<li class="chapter" data-level="14.9" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#session-info-2"><i class="fa fa-check"></i><b>14.9</b> Session info</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html"><i class="fa fa-check"></i><b>15</b> Interoperability with other frameworks</a>
<ul>
<li class="chapter" data-level="15.1" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#load-giotto-object"><i class="fa fa-check"></i><b>15.1</b> Load Giotto object</a></li>
<li class="chapter" data-level="15.2" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#seurat"><i class="fa fa-check"></i><b>15.2</b> Seurat</a>
<ul>
<li class="chapter" data-level="15.2.1" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#conversion-of-giotto-object-to-seurat-object"><i class="fa fa-check"></i><b>15.2.1</b> Conversion of Giotto Object to Seurat Object</a></li>
<li class="chapter" data-level="15.2.2" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#conversion-of-seurat-object-back-to-giotto-object"><i class="fa fa-check"></i><b>15.2.2</b> Conversion of Seurat object Back to Giotto Object</a></li>
</ul></li>
<li class="chapter" data-level="15.3" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#spatialexperiment"><i class="fa fa-check"></i><b>15.3</b> SpatialExperiment</a>
<ul>
<li class="chapter" data-level="15.3.1" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#convert-giotto-object-to-spatialexperiment-object"><i class="fa fa-check"></i><b>15.3.1</b> Convert Giotto Object to SpatialExperiment Object</a></li>
<li class="chapter" data-level="15.3.2" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#conversion-of-spatialexperiment-object-back-to-giotto"><i class="fa fa-check"></i><b>15.3.2</b> Conversion of SpatialExperiment object back to Giotto</a></li>
</ul></li>
<li class="chapter" data-level="15.4" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#anndata"><i class="fa fa-check"></i><b>15.4</b> AnnData</a>
<ul>
<li class="chapter" data-level="15.4.1" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#load-required-libraries"><i class="fa fa-check"></i><b>15.4.1</b> Load Required Libraries</a></li>
<li class="chapter" data-level="15.4.2" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#specify-path-for-results"><i class="fa fa-check"></i><b>15.4.2</b> Specify Path for Results</a></li>
<li class="chapter" data-level="15.4.3" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#giotto-to-anndata"><i class="fa fa-check"></i><b>15.4.3</b> Giotto To AnnData</a></li>
<li class="chapter" data-level="15.4.4" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#convert-anndata-to-giotto"><i class="fa fa-check"></i><b>15.4.4</b> Convert AnnData to Giotto</a></li>
</ul></li>
<li class="chapter" data-level="15.5" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#create-mini-vizgen-object"><i class="fa fa-check"></i><b>15.5</b> Create mini Vizgen object</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html"><i class="fa fa-check"></i><b>16</b> Interoperability with isolated tools</a>
<ul>
<li class="chapter" data-level="16.1" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#spatial-niche-trajectory-analysis-ontrac"><i class="fa fa-check"></i><b>16.1</b> Spatial niche trajectory analysis (ONTraC)</a>
<ul>
<li class="chapter" data-level="16.1.1" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#introduction-to-ontrac"><i class="fa fa-check"></i><b>16.1.1</b> Introduction to ONTraC</a></li>
<li class="chapter" data-level="16.1.2" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#introduction-to-merfish"><i class="fa fa-check"></i><b>16.1.2</b> Introduction to MERFISH</a></li>
<li class="chapter" data-level="16.1.3" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#settings"><i class="fa fa-check"></i><b>16.1.3</b> settings</a></li>
<li class="chapter" data-level="16.1.4" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#dataset-1"><i class="fa fa-check"></i><b>16.1.4</b> Dataset</a></li>
<li class="chapter" data-level="16.1.5" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#create-the-giotto-object-2"><i class="fa fa-check"></i><b>16.1.5</b> Create the Giotto object</a></li>
<li class="chapter" data-level="16.1.6" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#perform-spatial-niche-trajectory-analysis-using-ontrac"><i class="fa fa-check"></i><b>16.1.6</b> Perform spatial niche trajectory analysis using ONTraC</a></li>
<li class="chapter" data-level="16.1.7" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#visualization"><i class="fa fa-check"></i><b>16.1.7</b> visualization</a></li>
</ul></li>
<li class="chapter" data-level="16.2" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#session-info-3"><i class="fa fa-check"></i><b>16.2</b> Session info</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html"><i class="fa fa-check"></i><b>17</b> Interactivity with the R/Spatial ecosystem</a>
<ul>
<li class="chapter" data-level="17.1" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#kriging"><i class="fa fa-check"></i><b>17.1</b> Kriging</a>
<ul>
<li class="chapter" data-level="17.1.1" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#visium-technology-1"><i class="fa fa-check"></i><b>17.1.1</b> Visium technology</a></li>
<li class="chapter" data-level="17.1.2" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#dataset-2"><i class="fa fa-check"></i><b>17.1.2</b> Dataset</a></li>
<li class="chapter" data-level="17.1.3" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#generating-a-geojson-file-of-nuclei-location"><i class="fa fa-check"></i><b>17.1.3</b> Generating a geojson file of nuclei location</a></li>
</ul></li>
<li class="chapter" data-level="17.2" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#downloading-the-dataset"><i class="fa fa-check"></i><b>17.2</b> Downloading the dataset</a></li>
<li class="chapter" data-level="17.3" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#importing-visium-data"><i class="fa fa-check"></i><b>17.3</b> Importing visium data</a>
<ul>
<li class="chapter" data-level="17.3.1" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#identifying-spatially-organized-features"><i class="fa fa-check"></i><b>17.3.1</b> Identifying spatially organized features</a></li>
</ul></li>
<li class="chapter" data-level="17.4" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#adding-cell-polygons-to-giotto-object"><i class="fa fa-check"></i><b>17.4</b> Adding cell polygons to Giotto object</a>
<ul>
<li class="chapter" data-level="17.4.1" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#read-in-the-poly-information"><i class="fa fa-check"></i><b>17.4.1</b> Read in the poly information</a></li>
<li class="chapter" data-level="17.4.2" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#vizualizing-polygons"><i class="fa fa-check"></i><b>17.4.2</b> Vizualizing polygons</a></li>
<li class="chapter" data-level="17.4.3" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#showing-giotto-object-prior-to-polygon-addition"><i class="fa fa-check"></i><b>17.4.3</b> Showing Giotto object prior to polygon addition</a></li>
<li class="chapter" data-level="17.4.4" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#adding-polygons-to-giotto-object"><i class="fa fa-check"></i><b>17.4.4</b> Adding polygons to giotto object</a></li>
<li class="chapter" data-level="17.4.5" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#check-polygon-information"><i class="fa fa-check"></i><b>17.4.5</b> Check polygon information</a></li>
</ul></li>
<li class="chapter" data-level="17.5" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#performing-kriging"><i class="fa fa-check"></i><b>17.5</b> Performing kriging</a>
<ul>
<li class="chapter" data-level="17.5.1" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#interpolating-features"><i class="fa fa-check"></i><b>17.5.1</b> Interpolating features</a></li>
<li class="chapter" data-level="17.5.2" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#expression-overlap"><i class="fa fa-check"></i><b>17.5.2</b> Expression overlap</a></li>
</ul></li>
<li class="chapter" data-level="17.6" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#reading-in-larger-dataset"><i class="fa fa-check"></i><b>17.6</b> Reading in larger dataset</a></li>
<li class="chapter" data-level="17.7" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#analyzing-interpolated-features"><i class="fa fa-check"></i><b>17.7</b> Analyzing interpolated features</a>
<ul>
<li class="chapter" data-level="17.7.1" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#filter-and-normalization"><i class="fa fa-check"></i><b>17.7.1</b> Filter and normalization</a></li>
<li class="chapter" data-level="17.7.2" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#visualizing-gene-expression-from-interpolated-expression"><i class="fa fa-check"></i><b>17.7.2</b> Visualizing gene expression from interpolated expression</a></li>
<li class="chapter" data-level="17.7.3" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#run-pca"><i class="fa fa-check"></i><b>17.7.3</b> Run PCA</a></li>
<li class="chapter" data-level="17.7.4" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#clustering-2"><i class="fa fa-check"></i><b>17.7.4</b> Clustering</a></li>
<li class="chapter" data-level="17.7.5" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#visualizing-clustering"><i class="fa fa-check"></i><b>17.7.5</b> Visualizing clustering</a></li>
<li class="chapter" data-level="17.7.6" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#cropping-objects"><i class="fa fa-check"></i><b>17.7.6</b> Cropping objects</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="18" data-path="contributing-to-giotto.html"><a href="contributing-to-giotto.html"><i class="fa fa-check"></i><b>18</b> Contributing to Giotto</a>
<ul>
<li class="chapter" data-level="18.1" data-path="contributing-to-giotto.html"><a href="contributing-to-giotto.html#contribution-guideline"><i class="fa fa-check"></i><b>18.1</b> Contribution guideline</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Workshop: Spatial multi-omics data analysis with Giotto Suite</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="multi-omics-integration" class="section level1 hasAnchor" number="14">
<h1><span class="header-section-number">14</span> Multi-omics integration<a href="multi-omics-integration.html#multi-omics-integration" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Joselyn Cristina ChÃ¡vez Fuentes</p>
<p>August 7th 2024</p>
<div id="the-cytassist-technology" class="section level2 hasAnchor" number="14.1">
<h2><span class="header-section-number">14.1</span> The CytAssist technology<a href="multi-omics-integration.html#the-cytassist-technology" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The Visium CytAssist Spatial Gene and Protein Expression assay is designed to introduce simultaneous Gene Expression and Protein Expression analysis to FFPE samples processed with Visium CytAssist. The assay uses NGS to measure the abundance of oligo-tagged antibodies with spatial resolution, in addition to the whole transcriptome and a morphological image.</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-400"></span>
<img src="img/03_session3/cytassist.png" alt="CystAssits multi-omics diagram. Source: 10X genomics." width="80%" />
<p class="caption">
Figure 14.1: CystAssits multi-omics diagram. Source: 10X genomics.
</p>
</div>
<p>The 10X human immune cell profiling panel features 35 antibodies from Abcam and Biolegend, and includes cell surface and intracellular targets. The rna probes hybridize to ~18,000 genes, or RNA targets, within the tissue section to achieve whole transcriptome gene expression profiling. The remaining steps, starting with probe extension, follow the standard Visium workflow outside of the instrument.</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-401"></span>
<img src="img/03_session3/workflow.png" alt="CytAssist workflow. Source: 10X genomics." width="100%" />
<p class="caption">
Figure 14.2: CytAssist workflow. Source: 10X genomics.
</p>
</div>
</div>
<div id="introduction-to-the-spatial-dataset-1" class="section level2 hasAnchor" number="14.2">
<h2><span class="header-section-number">14.2</span> Introduction to the spatial dataset<a href="multi-omics-integration.html#introduction-to-the-spatial-dataset-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The <a href="https://www.10xgenomics.com/datasets/gene-protein-expression-library-of-human-tonsil-cytassist-ffpe-2-standard">Human Tonsil (FFPE) dataset</a> was obtained from 10X Genomics. The tissue was sectioned as described in Visium CytAssist Spatial Gene and Protein Expression for FFPE â Tissue Preparation Guide (CG000660). 5 Âµm tissue sections were placed on Superfrost glass slides, deparaffinized, H&amp;E stained (CG000658) and coverslipped. Sections were imaged, decoverslipped, followed by decrosslinking per the Staining Demonstrated Protocol (CG000658).</p>
<p>More information about this dataset can be found <a href="https://www.10xgenomics.com/datasets/gene-protein-expression-library-of-human-tonsil-cytassist-ffpe-2-standard">here</a>.</p>
</div>
<div id="download-dataset-1" class="section level2 hasAnchor" number="14.3">
<h2><span class="header-section-number">14.3</span> Download dataset<a href="multi-omics-integration.html#download-dataset-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>You need to download the expression matrix and spatial information by running these commands:</p>
<div class="sourceCode" id="cb287"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb287-1"><a href="multi-omics-integration.html#cb287-1" tabindex="-1"></a><span class="fu">dir.create</span>(<span class="st">&quot;data/03_session3&quot;</span>)</span>
<span id="cb287-2"><a href="multi-omics-integration.html#cb287-2" tabindex="-1"></a></span>
<span id="cb287-3"><a href="multi-omics-integration.html#cb287-3" tabindex="-1"></a><span class="fu">download.file</span>(<span class="at">url =</span> <span class="st">&quot;https://cf.10xgenomics.com/samples/spatial-exp/2.1.0/CytAssist_FFPE_Protein_Expression_Human_Tonsil/CytAssist_FFPE_Protein_Expression_Human_Tonsil_raw_feature_bc_matrix.tar.gz&quot;</span>,</span>
<span id="cb287-4"><a href="multi-omics-integration.html#cb287-4" tabindex="-1"></a>              <span class="at">destfile =</span> <span class="st">&quot;data/03_session3/CytAssist_FFPE_Protein_Expression_Human_Tonsil_raw_feature_bc_matrix.tar.gz&quot;</span>)</span>
<span id="cb287-5"><a href="multi-omics-integration.html#cb287-5" tabindex="-1"></a></span>
<span id="cb287-6"><a href="multi-omics-integration.html#cb287-6" tabindex="-1"></a><span class="fu">download.file</span>(<span class="at">url =</span> <span class="st">&quot;https://cf.10xgenomics.com/samples/spatial-exp/2.1.0/CytAssist_FFPE_Protein_Expression_Human_Tonsil/CytAssist_FFPE_Protein_Expression_Human_Tonsil_spatial.tar.gz&quot;</span>,</span>
<span id="cb287-7"><a href="multi-omics-integration.html#cb287-7" tabindex="-1"></a>              <span class="at">destfile =</span> <span class="st">&quot;data/03_session3/CytAssist_FFPE_Protein_Expression_Human_Tonsil_spatial.tar.gz&quot;</span>)</span></code></pre></div>
<p>After downloading, unzip the gz files. You should get the âraw_feature_bc_matrixâ and âspatialâ folders inside âdata/03_session3/â.</p>
<div class="sourceCode" id="cb288"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb288-1"><a href="multi-omics-integration.html#cb288-1" tabindex="-1"></a><span class="fu">untar</span>(<span class="at">tarfile =</span> <span class="st">&quot;data/03_session3/CytAssist_FFPE_Protein_Expression_Human_Tonsil_raw_feature_bc_matrix.tar.gz&quot;</span>, </span>
<span id="cb288-2"><a href="multi-omics-integration.html#cb288-2" tabindex="-1"></a>      <span class="at">exdir =</span> <span class="st">&quot;data/03_session3&quot;</span>)</span>
<span id="cb288-3"><a href="multi-omics-integration.html#cb288-3" tabindex="-1"></a></span>
<span id="cb288-4"><a href="multi-omics-integration.html#cb288-4" tabindex="-1"></a><span class="fu">untar</span>(<span class="at">tarfile =</span> <span class="st">&quot;data/03_session3/CytAssist_FFPE_Protein_Expression_Human_Tonsil_spatial.tar.gz&quot;</span>, </span>
<span id="cb288-5"><a href="multi-omics-integration.html#cb288-5" tabindex="-1"></a>      <span class="at">exdir =</span> <span class="st">&quot;data/03_session3&quot;</span>)</span></code></pre></div>
</div>
<div id="create-the-giotto-object-1" class="section level2 hasAnchor" number="14.4">
<h2><span class="header-section-number">14.4</span> Create the Giotto object<a href="multi-omics-integration.html#create-the-giotto-object-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The minimum requirements are:</p>
<ul>
<li>matrix with expression information (or the path to)</li>
<li>x,y(,z) coordinates for cells or spots (or the path to)</li>
</ul>
<p>createGiottoVisiumObject() will automatically detect both RNA and Protein modalities in the expression matrix and will create a multi-omics Giotto object.</p>
<div class="sourceCode" id="cb289"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb289-1"><a href="multi-omics-integration.html#cb289-1" tabindex="-1"></a><span class="fu">library</span>(Giotto)</span>
<span id="cb289-2"><a href="multi-omics-integration.html#cb289-2" tabindex="-1"></a></span>
<span id="cb289-3"><a href="multi-omics-integration.html#cb289-3" tabindex="-1"></a><span class="do">## Set instructions</span></span>
<span id="cb289-4"><a href="multi-omics-integration.html#cb289-4" tabindex="-1"></a>results_folder <span class="ot">&lt;-</span> <span class="st">&quot;results/03_session3/&quot;</span></span>
<span id="cb289-5"><a href="multi-omics-integration.html#cb289-5" tabindex="-1"></a></span>
<span id="cb289-6"><a href="multi-omics-integration.html#cb289-6" tabindex="-1"></a>python_path <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb289-7"><a href="multi-omics-integration.html#cb289-7" tabindex="-1"></a></span>
<span id="cb289-8"><a href="multi-omics-integration.html#cb289-8" tabindex="-1"></a>instructions <span class="ot">&lt;-</span> <span class="fu">createGiottoInstructions</span>(</span>
<span id="cb289-9"><a href="multi-omics-integration.html#cb289-9" tabindex="-1"></a>  <span class="at">save_dir =</span> results_folder,</span>
<span id="cb289-10"><a href="multi-omics-integration.html#cb289-10" tabindex="-1"></a>  <span class="at">save_plot =</span> <span class="cn">TRUE</span>,</span>
<span id="cb289-11"><a href="multi-omics-integration.html#cb289-11" tabindex="-1"></a>  <span class="at">show_plot =</span> <span class="cn">FALSE</span>,</span>
<span id="cb289-12"><a href="multi-omics-integration.html#cb289-12" tabindex="-1"></a>  <span class="at">return_plot =</span> <span class="cn">FALSE</span>,</span>
<span id="cb289-13"><a href="multi-omics-integration.html#cb289-13" tabindex="-1"></a>  <span class="at">python_path =</span> python_path</span>
<span id="cb289-14"><a href="multi-omics-integration.html#cb289-14" tabindex="-1"></a>)</span>
<span id="cb289-15"><a href="multi-omics-integration.html#cb289-15" tabindex="-1"></a></span>
<span id="cb289-16"><a href="multi-omics-integration.html#cb289-16" tabindex="-1"></a><span class="co"># Provide the path to the data folder</span></span>
<span id="cb289-17"><a href="multi-omics-integration.html#cb289-17" tabindex="-1"></a>data_path <span class="ot">&lt;-</span> <span class="st">&quot;data/03_session3/&quot;</span></span>
<span id="cb289-18"><a href="multi-omics-integration.html#cb289-18" tabindex="-1"></a></span>
<span id="cb289-19"><a href="multi-omics-integration.html#cb289-19" tabindex="-1"></a><span class="co"># Create object directly from the data folder</span></span>
<span id="cb289-20"><a href="multi-omics-integration.html#cb289-20" tabindex="-1"></a>visium_tonsil <span class="ot">&lt;-</span> <span class="fu">createGiottoVisiumObject</span>(</span>
<span id="cb289-21"><a href="multi-omics-integration.html#cb289-21" tabindex="-1"></a>  <span class="at">visium_dir =</span> data_path,</span>
<span id="cb289-22"><a href="multi-omics-integration.html#cb289-22" tabindex="-1"></a>  <span class="at">expr_data =</span> <span class="st">&quot;raw&quot;</span>,</span>
<span id="cb289-23"><a href="multi-omics-integration.html#cb289-23" tabindex="-1"></a>  <span class="at">png_name =</span> <span class="st">&quot;tissue_lowres_image.png&quot;</span>,</span>
<span id="cb289-24"><a href="multi-omics-integration.html#cb289-24" tabindex="-1"></a>  <span class="at">gene_column_index =</span> <span class="dv">2</span>,</span>
<span id="cb289-25"><a href="multi-omics-integration.html#cb289-25" tabindex="-1"></a>  <span class="at">instructions =</span> instructions</span>
<span id="cb289-26"><a href="multi-omics-integration.html#cb289-26" tabindex="-1"></a>)</span></code></pre></div>
<ul>
<li>Print the information of the object, note that both rna and protein are listed in the expression slot.</li>
</ul>
<div class="sourceCode" id="cb290"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb290-1"><a href="multi-omics-integration.html#cb290-1" tabindex="-1"></a>visium_tonsil</span></code></pre></div>
</div>
<div id="subset-on-spots-that-were-covered-by-tissue-1" class="section level2 hasAnchor" number="14.5">
<h2><span class="header-section-number">14.5</span> Subset on spots that were covered by tissue<a href="multi-omics-integration.html#subset-on-spots-that-were-covered-by-tissue-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb291"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb291-1"><a href="multi-omics-integration.html#cb291-1" tabindex="-1"></a><span class="fu">spatPlot2D</span>(</span>
<span id="cb291-2"><a href="multi-omics-integration.html#cb291-2" tabindex="-1"></a>  <span class="at">gobject =</span> visium_tonsil,</span>
<span id="cb291-3"><a href="multi-omics-integration.html#cb291-3" tabindex="-1"></a>  <span class="at">cell_color =</span> <span class="st">&quot;in_tissue&quot;</span>,</span>
<span id="cb291-4"><a href="multi-omics-integration.html#cb291-4" tabindex="-1"></a>  <span class="at">point_size =</span> <span class="dv">2</span>,</span>
<span id="cb291-5"><a href="multi-omics-integration.html#cb291-5" tabindex="-1"></a>  <span class="at">cell_color_code =</span> <span class="fu">c</span>(<span class="st">&quot;0&quot;</span> <span class="ot">=</span> <span class="st">&quot;lightgrey&quot;</span>, <span class="st">&quot;1&quot;</span> <span class="ot">=</span> <span class="st">&quot;blue&quot;</span>),</span>
<span id="cb291-6"><a href="multi-omics-integration.html#cb291-6" tabindex="-1"></a>  <span class="at">show_image =</span> <span class="cn">TRUE</span>,</span>
<span id="cb291-7"><a href="multi-omics-integration.html#cb291-7" tabindex="-1"></a>  <span class="at">image_name =</span> <span class="st">&quot;image&quot;</span></span>
<span id="cb291-8"><a href="multi-omics-integration.html#cb291-8" tabindex="-1"></a>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-407"></span>
<img src="img/03_session3/0-spatPlot2D.png" alt="Spatial plot of the CytAssist human tonsil sample, color indicates wheter the spot is in tissue (1) or not (0)." width="80%" />
<p class="caption">
Figure 14.3: Spatial plot of the CytAssist human tonsil sample, color indicates wheter the spot is in tissue (1) or not (0).
</p>
</div>
<p>Use the metadata table to identify the spots corresponding to the tissue area, given by the âin_tissueâ column. Then use the spot IDs to subset the giotto object.</p>
<div class="sourceCode" id="cb292"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb292-1"><a href="multi-omics-integration.html#cb292-1" tabindex="-1"></a>metadata <span class="ot">&lt;-</span> <span class="fu">getCellMetadata</span>(<span class="at">gobject =</span> visium_tonsil,</span>
<span id="cb292-2"><a href="multi-omics-integration.html#cb292-2" tabindex="-1"></a>                            <span class="at">output =</span> <span class="st">&quot;data.table&quot;</span>)</span>
<span id="cb292-3"><a href="multi-omics-integration.html#cb292-3" tabindex="-1"></a></span>
<span id="cb292-4"><a href="multi-omics-integration.html#cb292-4" tabindex="-1"></a>in_tissue_barcodes <span class="ot">&lt;-</span> metadata[in_tissue <span class="sc">==</span> <span class="dv">1</span>]<span class="sc">$</span>cell_ID</span>
<span id="cb292-5"><a href="multi-omics-integration.html#cb292-5" tabindex="-1"></a></span>
<span id="cb292-6"><a href="multi-omics-integration.html#cb292-6" tabindex="-1"></a>visium_tonsil <span class="ot">&lt;-</span> <span class="fu">subsetGiotto</span>(visium_tonsil, </span>
<span id="cb292-7"><a href="multi-omics-integration.html#cb292-7" tabindex="-1"></a>                              <span class="at">cell_ids =</span> in_tissue_barcodes)</span></code></pre></div>
</div>
<div id="rna-processing" class="section level2 hasAnchor" number="14.6">
<h2><span class="header-section-number">14.6</span> RNA processing<a href="multi-omics-integration.html#rna-processing" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ul>
<li>Run the Filtering, normalization, and statistics steps using only the RNA feature.</li>
</ul>
<div class="sourceCode" id="cb293"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb293-1"><a href="multi-omics-integration.html#cb293-1" tabindex="-1"></a>visium_tonsil <span class="ot">&lt;-</span> <span class="fu">filterGiotto</span>(</span>
<span id="cb293-2"><a href="multi-omics-integration.html#cb293-2" tabindex="-1"></a>  <span class="at">gobject =</span> visium_tonsil,</span>
<span id="cb293-3"><a href="multi-omics-integration.html#cb293-3" tabindex="-1"></a>  <span class="at">expression_threshold =</span> <span class="dv">1</span>,</span>
<span id="cb293-4"><a href="multi-omics-integration.html#cb293-4" tabindex="-1"></a>  <span class="at">feat_det_in_min_cells =</span> <span class="dv">50</span>,</span>
<span id="cb293-5"><a href="multi-omics-integration.html#cb293-5" tabindex="-1"></a>  <span class="at">min_det_feats_per_cell =</span> <span class="dv">1000</span>,</span>
<span id="cb293-6"><a href="multi-omics-integration.html#cb293-6" tabindex="-1"></a>  <span class="at">expression_values =</span> <span class="st">&quot;raw&quot;</span>,</span>
<span id="cb293-7"><a href="multi-omics-integration.html#cb293-7" tabindex="-1"></a>  <span class="at">verbose =</span> <span class="cn">TRUE</span>)</span>
<span id="cb293-8"><a href="multi-omics-integration.html#cb293-8" tabindex="-1"></a></span>
<span id="cb293-9"><a href="multi-omics-integration.html#cb293-9" tabindex="-1"></a>visium_tonsil <span class="ot">&lt;-</span> <span class="fu">normalizeGiotto</span>(<span class="at">gobject =</span> visium_tonsil,</span>
<span id="cb293-10"><a href="multi-omics-integration.html#cb293-10" tabindex="-1"></a>                                 <span class="at">scalefactor =</span> <span class="dv">6000</span>,</span>
<span id="cb293-11"><a href="multi-omics-integration.html#cb293-11" tabindex="-1"></a>                                 <span class="at">verbose =</span> <span class="cn">TRUE</span>)</span>
<span id="cb293-12"><a href="multi-omics-integration.html#cb293-12" tabindex="-1"></a></span>
<span id="cb293-13"><a href="multi-omics-integration.html#cb293-13" tabindex="-1"></a>visium_tonsil <span class="ot">&lt;-</span> <span class="fu">addStatistics</span>(<span class="at">gobject =</span> visium_tonsil)</span></code></pre></div>
<ul>
<li>Dimension reduction</li>
</ul>
<p>Identify the highly variable features using the RNA features, then calculate the principal components based on the HVFs.</p>
<div class="sourceCode" id="cb294"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb294-1"><a href="multi-omics-integration.html#cb294-1" tabindex="-1"></a>visium_tonsil <span class="ot">&lt;-</span> <span class="fu">calculateHVF</span>(<span class="at">gobject =</span> visium_tonsil)</span>
<span id="cb294-2"><a href="multi-omics-integration.html#cb294-2" tabindex="-1"></a></span>
<span id="cb294-3"><a href="multi-omics-integration.html#cb294-3" tabindex="-1"></a>visium_tonsil <span class="ot">&lt;-</span> <span class="fu">runPCA</span>(<span class="at">gobject =</span> visium_tonsil)</span></code></pre></div>
<ul>
<li>Clustering</li>
</ul>
<p>Calculate the UMAP, tSNE, and shared nearest neighbor network using the first 10 principal components for the RNA modality.</p>
<div class="sourceCode" id="cb295"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb295-1"><a href="multi-omics-integration.html#cb295-1" tabindex="-1"></a>visium_tonsil <span class="ot">&lt;-</span> <span class="fu">runUMAP</span>(visium_tonsil,</span>
<span id="cb295-2"><a href="multi-omics-integration.html#cb295-2" tabindex="-1"></a>                         <span class="at">dimensions_to_use =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>)</span>
<span id="cb295-3"><a href="multi-omics-integration.html#cb295-3" tabindex="-1"></a></span>
<span id="cb295-4"><a href="multi-omics-integration.html#cb295-4" tabindex="-1"></a>visium_tonsil <span class="ot">&lt;-</span> <span class="fu">runtSNE</span>(visium_tonsil, </span>
<span id="cb295-5"><a href="multi-omics-integration.html#cb295-5" tabindex="-1"></a>                         <span class="at">dimensions_to_use =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>)</span>
<span id="cb295-6"><a href="multi-omics-integration.html#cb295-6" tabindex="-1"></a></span>
<span id="cb295-7"><a href="multi-omics-integration.html#cb295-7" tabindex="-1"></a>visium_tonsil <span class="ot">&lt;-</span> <span class="fu">createNearestNetwork</span>(<span class="at">gobject =</span> visium_tonsil,</span>
<span id="cb295-8"><a href="multi-omics-integration.html#cb295-8" tabindex="-1"></a>                                      <span class="at">dimensions_to_use =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>,</span>
<span id="cb295-9"><a href="multi-omics-integration.html#cb295-9" tabindex="-1"></a>                                      <span class="at">k =</span> <span class="dv">30</span>)</span></code></pre></div>
<p>Calculate the RNA-based Leiden clusters.</p>
<div class="sourceCode" id="cb296"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb296-1"><a href="multi-omics-integration.html#cb296-1" tabindex="-1"></a>visium_tonsil <span class="ot">&lt;-</span> <span class="fu">doLeidenCluster</span>(<span class="at">gobject =</span> visium_tonsil,</span>
<span id="cb296-2"><a href="multi-omics-integration.html#cb296-2" tabindex="-1"></a>                                 <span class="at">resolution =</span> <span class="dv">1</span>,</span>
<span id="cb296-3"><a href="multi-omics-integration.html#cb296-3" tabindex="-1"></a>                                 <span class="at">n_iterations =</span> <span class="dv">1000</span>)</span></code></pre></div>
<ul>
<li>Visualization</li>
</ul>
<p>Plot the RNA-based UMAP with the corresponding RNA-based Leiden cluster per spot.</p>
<div class="sourceCode" id="cb297"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb297-1"><a href="multi-omics-integration.html#cb297-1" tabindex="-1"></a><span class="fu">plotUMAP</span>(<span class="at">gobject =</span> visium_tonsil,</span>
<span id="cb297-2"><a href="multi-omics-integration.html#cb297-2" tabindex="-1"></a>         <span class="at">cell_color =</span> <span class="st">&quot;leiden_clus&quot;</span>,</span>
<span id="cb297-3"><a href="multi-omics-integration.html#cb297-3" tabindex="-1"></a>         <span class="at">show_NN_network =</span> <span class="cn">TRUE</span>,</span>
<span id="cb297-4"><a href="multi-omics-integration.html#cb297-4" tabindex="-1"></a>         <span class="at">point_size =</span> <span class="dv">2</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-414"></span>
<img src="img/03_session3/1-UMAP.png" alt="RNA UMAP, color indicates the RNA-based Leiden clusters." width="80%" />
<p class="caption">
Figure 14.4: RNA UMAP, color indicates the RNA-based Leiden clusters.
</p>
</div>
<p>Plot the spatial distribution of the RNA-based Leiden cluster per spot.</p>
<div class="sourceCode" id="cb298"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb298-1"><a href="multi-omics-integration.html#cb298-1" tabindex="-1"></a><span class="fu">spatPlot2D</span>(<span class="at">gobject =</span> visium_tonsil,</span>
<span id="cb298-2"><a href="multi-omics-integration.html#cb298-2" tabindex="-1"></a>           <span class="at">show_image =</span> <span class="cn">TRUE</span>,</span>
<span id="cb298-3"><a href="multi-omics-integration.html#cb298-3" tabindex="-1"></a>           <span class="at">cell_color =</span> <span class="st">&quot;leiden_clus&quot;</span>,</span>
<span id="cb298-4"><a href="multi-omics-integration.html#cb298-4" tabindex="-1"></a>           <span class="at">point_size =</span> <span class="dv">3</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-416"></span>
<img src="img/03_session3/2-spatPlot2D.png" alt="Spatial distribution of RNA-based Leiden clusters." width="80%" />
<p class="caption">
Figure 14.5: Spatial distribution of RNA-based Leiden clusters.
</p>
</div>
</div>
<div id="protein-processing" class="section level2 hasAnchor" number="14.7">
<h2><span class="header-section-number">14.7</span> Protein processing<a href="multi-omics-integration.html#protein-processing" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ul>
<li>Run the Filtering, normalization, and statistics steps for the protein modality.</li>
</ul>
<div class="sourceCode" id="cb299"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb299-1"><a href="multi-omics-integration.html#cb299-1" tabindex="-1"></a>visium_tonsil <span class="ot">&lt;-</span> <span class="fu">filterGiotto</span>(<span class="at">gobject =</span> visium_tonsil, </span>
<span id="cb299-2"><a href="multi-omics-integration.html#cb299-2" tabindex="-1"></a>                              <span class="at">spat_unit =</span> <span class="st">&quot;cell&quot;</span>,</span>
<span id="cb299-3"><a href="multi-omics-integration.html#cb299-3" tabindex="-1"></a>                              <span class="at">feat_type =</span> <span class="st">&quot;protein&quot;</span>,</span>
<span id="cb299-4"><a href="multi-omics-integration.html#cb299-4" tabindex="-1"></a>                              <span class="at">expression_threshold =</span> <span class="dv">1</span>,</span>
<span id="cb299-5"><a href="multi-omics-integration.html#cb299-5" tabindex="-1"></a>                              <span class="at">feat_det_in_min_cells =</span> <span class="dv">50</span>, </span>
<span id="cb299-6"><a href="multi-omics-integration.html#cb299-6" tabindex="-1"></a>                              <span class="at">min_det_feats_per_cell =</span> <span class="dv">1</span>,</span>
<span id="cb299-7"><a href="multi-omics-integration.html#cb299-7" tabindex="-1"></a>                              <span class="at">expression_values =</span> <span class="st">&quot;raw&quot;</span>, </span>
<span id="cb299-8"><a href="multi-omics-integration.html#cb299-8" tabindex="-1"></a>                              <span class="at">verbose =</span> <span class="cn">TRUE</span>)</span>
<span id="cb299-9"><a href="multi-omics-integration.html#cb299-9" tabindex="-1"></a></span>
<span id="cb299-10"><a href="multi-omics-integration.html#cb299-10" tabindex="-1"></a>visium_tonsil <span class="ot">&lt;-</span> <span class="fu">normalizeGiotto</span>(<span class="at">gobject =</span> visium_tonsil,</span>
<span id="cb299-11"><a href="multi-omics-integration.html#cb299-11" tabindex="-1"></a>                                 <span class="at">spat_unit =</span> <span class="st">&quot;cell&quot;</span>, </span>
<span id="cb299-12"><a href="multi-omics-integration.html#cb299-12" tabindex="-1"></a>                                 <span class="at">feat_type =</span> <span class="st">&quot;protein&quot;</span>, </span>
<span id="cb299-13"><a href="multi-omics-integration.html#cb299-13" tabindex="-1"></a>                                 <span class="at">scalefactor =</span> <span class="dv">6000</span>, </span>
<span id="cb299-14"><a href="multi-omics-integration.html#cb299-14" tabindex="-1"></a>                                 <span class="at">verbose =</span> <span class="cn">TRUE</span>)</span>
<span id="cb299-15"><a href="multi-omics-integration.html#cb299-15" tabindex="-1"></a></span>
<span id="cb299-16"><a href="multi-omics-integration.html#cb299-16" tabindex="-1"></a>visium_tonsil <span class="ot">&lt;-</span> <span class="fu">addStatistics</span>(<span class="at">gobject =</span> visium_tonsil,</span>
<span id="cb299-17"><a href="multi-omics-integration.html#cb299-17" tabindex="-1"></a>                               <span class="at">spat_unit =</span> <span class="st">&quot;cell&quot;</span>, </span>
<span id="cb299-18"><a href="multi-omics-integration.html#cb299-18" tabindex="-1"></a>                               <span class="at">feat_type =</span> <span class="st">&quot;protein&quot;</span>)</span></code></pre></div>
<ul>
<li>Dimension reduction</li>
</ul>
<p>Calculate the principal components using all the proteins available in the dataset.</p>
<div class="sourceCode" id="cb300"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb300-1"><a href="multi-omics-integration.html#cb300-1" tabindex="-1"></a>visium_tonsil <span class="ot">&lt;-</span> <span class="fu">runPCA</span>(<span class="at">gobject =</span> visium_tonsil,</span>
<span id="cb300-2"><a href="multi-omics-integration.html#cb300-2" tabindex="-1"></a>                        <span class="at">spat_unit =</span> <span class="st">&quot;cell&quot;</span>,</span>
<span id="cb300-3"><a href="multi-omics-integration.html#cb300-3" tabindex="-1"></a>                        <span class="at">feat_type =</span> <span class="st">&quot;protein&quot;</span>)</span></code></pre></div>
<ul>
<li>Clustering</li>
</ul>
<p>Calculate the UMAP, tSNE, and shared nearest neighbors network using the first 10 principal components for the Protein modality.</p>
<div class="sourceCode" id="cb301"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb301-1"><a href="multi-omics-integration.html#cb301-1" tabindex="-1"></a>visium_tonsil <span class="ot">&lt;-</span> <span class="fu">runUMAP</span>(visium_tonsil,</span>
<span id="cb301-2"><a href="multi-omics-integration.html#cb301-2" tabindex="-1"></a>                         <span class="at">spat_unit =</span> <span class="st">&quot;cell&quot;</span>, </span>
<span id="cb301-3"><a href="multi-omics-integration.html#cb301-3" tabindex="-1"></a>                         <span class="at">feat_type =</span> <span class="st">&quot;protein&quot;</span>, </span>
<span id="cb301-4"><a href="multi-omics-integration.html#cb301-4" tabindex="-1"></a>                         <span class="at">dimensions_to_use =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>)</span>
<span id="cb301-5"><a href="multi-omics-integration.html#cb301-5" tabindex="-1"></a></span>
<span id="cb301-6"><a href="multi-omics-integration.html#cb301-6" tabindex="-1"></a>visium_tonsil <span class="ot">&lt;-</span> <span class="fu">runtSNE</span>(visium_tonsil, </span>
<span id="cb301-7"><a href="multi-omics-integration.html#cb301-7" tabindex="-1"></a>                         <span class="at">spat_unit =</span> <span class="st">&quot;cell&quot;</span>,</span>
<span id="cb301-8"><a href="multi-omics-integration.html#cb301-8" tabindex="-1"></a>                         <span class="at">feat_type =</span> <span class="st">&quot;protein&quot;</span>,</span>
<span id="cb301-9"><a href="multi-omics-integration.html#cb301-9" tabindex="-1"></a>                         <span class="at">dimensions_to_use =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>)</span>
<span id="cb301-10"><a href="multi-omics-integration.html#cb301-10" tabindex="-1"></a></span>
<span id="cb301-11"><a href="multi-omics-integration.html#cb301-11" tabindex="-1"></a>visium_tonsil <span class="ot">&lt;-</span> <span class="fu">createNearestNetwork</span>(<span class="at">gobject =</span> visium_tonsil,</span>
<span id="cb301-12"><a href="multi-omics-integration.html#cb301-12" tabindex="-1"></a>                                      <span class="at">spat_unit =</span> <span class="st">&quot;cell&quot;</span>,</span>
<span id="cb301-13"><a href="multi-omics-integration.html#cb301-13" tabindex="-1"></a>                                      <span class="at">feat_type =</span> <span class="st">&quot;protein&quot;</span>,</span>
<span id="cb301-14"><a href="multi-omics-integration.html#cb301-14" tabindex="-1"></a>                                      <span class="at">dimensions_to_use =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>,</span>
<span id="cb301-15"><a href="multi-omics-integration.html#cb301-15" tabindex="-1"></a>                                      <span class="at">k =</span> <span class="dv">30</span>)</span></code></pre></div>
<p>Calculate the Protein-based Leiden clusters.</p>
<div class="sourceCode" id="cb302"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb302-1"><a href="multi-omics-integration.html#cb302-1" tabindex="-1"></a>visium_tonsil <span class="ot">&lt;-</span> <span class="fu">doLeidenCluster</span>(<span class="at">gobject =</span> visium_tonsil,</span>
<span id="cb302-2"><a href="multi-omics-integration.html#cb302-2" tabindex="-1"></a>                                 <span class="at">spat_unit =</span> <span class="st">&quot;cell&quot;</span>,</span>
<span id="cb302-3"><a href="multi-omics-integration.html#cb302-3" tabindex="-1"></a>                                 <span class="at">feat_type =</span> <span class="st">&quot;protein&quot;</span>,</span>
<span id="cb302-4"><a href="multi-omics-integration.html#cb302-4" tabindex="-1"></a>                                 <span class="at">resolution =</span> <span class="dv">1</span>,</span>
<span id="cb302-5"><a href="multi-omics-integration.html#cb302-5" tabindex="-1"></a>                                 <span class="at">n_iterations =</span> <span class="dv">1000</span>)</span></code></pre></div>
<ul>
<li>Visualization</li>
</ul>
<p>Plot the Protein UMAP and color the spots using the Protein-based Leiden clusters.</p>
<div class="sourceCode" id="cb303"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb303-1"><a href="multi-omics-integration.html#cb303-1" tabindex="-1"></a><span class="fu">plotUMAP</span>(<span class="at">gobject =</span> visium_tonsil,</span>
<span id="cb303-2"><a href="multi-omics-integration.html#cb303-2" tabindex="-1"></a>         <span class="at">spat_unit =</span> <span class="st">&quot;cell&quot;</span>, </span>
<span id="cb303-3"><a href="multi-omics-integration.html#cb303-3" tabindex="-1"></a>         <span class="at">feat_type =</span> <span class="st">&quot;protein&quot;</span>, </span>
<span id="cb303-4"><a href="multi-omics-integration.html#cb303-4" tabindex="-1"></a>         <span class="at">cell_color =</span> <span class="st">&quot;leiden_clus&quot;</span>, </span>
<span id="cb303-5"><a href="multi-omics-integration.html#cb303-5" tabindex="-1"></a>         <span class="at">show_NN_network =</span> <span class="cn">TRUE</span>, </span>
<span id="cb303-6"><a href="multi-omics-integration.html#cb303-6" tabindex="-1"></a>         <span class="at">point_size =</span> <span class="dv">2</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-422"></span>
<img src="img/03_session3/3-UMAP.png" alt="Protein UMAP, color indicates the Protein-based Leiden clusters." width="80%" />
<p class="caption">
Figure 14.6: Protein UMAP, color indicates the Protein-based Leiden clusters.
</p>
</div>
<p>Plot the spatial distribution of the Protein-based Leiden clusters.</p>
<div class="sourceCode" id="cb304"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb304-1"><a href="multi-omics-integration.html#cb304-1" tabindex="-1"></a><span class="fu">spatPlot2D</span>(<span class="at">gobject =</span> visium_tonsil,</span>
<span id="cb304-2"><a href="multi-omics-integration.html#cb304-2" tabindex="-1"></a>           <span class="at">spat_unit =</span> <span class="st">&quot;cell&quot;</span>,</span>
<span id="cb304-3"><a href="multi-omics-integration.html#cb304-3" tabindex="-1"></a>           <span class="at">feat_type =</span> <span class="st">&quot;protein&quot;</span>,</span>
<span id="cb304-4"><a href="multi-omics-integration.html#cb304-4" tabindex="-1"></a>           <span class="at">show_image =</span> <span class="cn">TRUE</span>,</span>
<span id="cb304-5"><a href="multi-omics-integration.html#cb304-5" tabindex="-1"></a>           <span class="at">cell_color =</span> <span class="st">&quot;leiden_clus&quot;</span>,</span>
<span id="cb304-6"><a href="multi-omics-integration.html#cb304-6" tabindex="-1"></a>           <span class="at">point_size =</span> <span class="dv">3</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-424"></span>
<img src="img/03_session3/4-spatPlot2D.png" alt="Spatial distribution of Protein-based Leiden clusters." width="80%" />
<p class="caption">
Figure 14.7: Spatial distribution of Protein-based Leiden clusters.
</p>
</div>
</div>
<div id="multi-omics-integration-1" class="section level2 hasAnchor" number="14.8">
<h2><span class="header-section-number">14.8</span> Multi-omics integration<a href="multi-omics-integration.html#multi-omics-integration-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ul>
<li>Calculate kNN</li>
</ul>
<p>Calculate the k nearest neighbors network for each modality (RNA and Protein), using the first 10 principal components of each feature type.</p>
<div class="sourceCode" id="cb305"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb305-1"><a href="multi-omics-integration.html#cb305-1" tabindex="-1"></a><span class="do">## RNA modality</span></span>
<span id="cb305-2"><a href="multi-omics-integration.html#cb305-2" tabindex="-1"></a>visium_tonsil <span class="ot">&lt;-</span> <span class="fu">createNearestNetwork</span>(<span class="at">gobject =</span> visium_tonsil,</span>
<span id="cb305-3"><a href="multi-omics-integration.html#cb305-3" tabindex="-1"></a>                                      <span class="at">type =</span> <span class="st">&quot;kNN&quot;</span>,</span>
<span id="cb305-4"><a href="multi-omics-integration.html#cb305-4" tabindex="-1"></a>                                      <span class="at">dimensions_to_use =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>,</span>
<span id="cb305-5"><a href="multi-omics-integration.html#cb305-5" tabindex="-1"></a>                                      <span class="at">k =</span> <span class="dv">20</span>)</span>
<span id="cb305-6"><a href="multi-omics-integration.html#cb305-6" tabindex="-1"></a></span>
<span id="cb305-7"><a href="multi-omics-integration.html#cb305-7" tabindex="-1"></a><span class="do">## Protein modality</span></span>
<span id="cb305-8"><a href="multi-omics-integration.html#cb305-8" tabindex="-1"></a>visium_tonsil <span class="ot">&lt;-</span> <span class="fu">createNearestNetwork</span>(<span class="at">gobject =</span> visium_tonsil,</span>
<span id="cb305-9"><a href="multi-omics-integration.html#cb305-9" tabindex="-1"></a>                                      <span class="at">spat_unit =</span> <span class="st">&quot;cell&quot;</span>,</span>
<span id="cb305-10"><a href="multi-omics-integration.html#cb305-10" tabindex="-1"></a>                                      <span class="at">feat_type =</span> <span class="st">&quot;protein&quot;</span>,</span>
<span id="cb305-11"><a href="multi-omics-integration.html#cb305-11" tabindex="-1"></a>                                      <span class="at">type =</span> <span class="st">&quot;kNN&quot;</span>,</span>
<span id="cb305-12"><a href="multi-omics-integration.html#cb305-12" tabindex="-1"></a>                                      <span class="at">dimensions_to_use =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>,</span>
<span id="cb305-13"><a href="multi-omics-integration.html#cb305-13" tabindex="-1"></a>                                      <span class="at">k =</span> <span class="dv">20</span>)</span></code></pre></div>
<ul>
<li>Run WNN</li>
</ul>
<p>Run the Weighted Nearest Neighbor analysis to weight the contribution of each feature type per spot. The results will be saved in the multiomics slot of the giotto object.</p>
<div class="sourceCode" id="cb306"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb306-1"><a href="multi-omics-integration.html#cb306-1" tabindex="-1"></a>visium_tonsil <span class="ot">&lt;-</span> <span class="fu">runWNN</span>(visium_tonsil,</span>
<span id="cb306-2"><a href="multi-omics-integration.html#cb306-2" tabindex="-1"></a>                        <span class="at">spat_unit =</span> <span class="st">&quot;cell&quot;</span>,</span>
<span id="cb306-3"><a href="multi-omics-integration.html#cb306-3" tabindex="-1"></a>                        <span class="at">modality_1 =</span> <span class="st">&quot;rna&quot;</span>,</span>
<span id="cb306-4"><a href="multi-omics-integration.html#cb306-4" tabindex="-1"></a>                        <span class="at">modality_2 =</span> <span class="st">&quot;protein&quot;</span>,</span>
<span id="cb306-5"><a href="multi-omics-integration.html#cb306-5" tabindex="-1"></a>                        <span class="at">pca_name_modality_1 =</span> <span class="st">&quot;pca&quot;</span>,</span>
<span id="cb306-6"><a href="multi-omics-integration.html#cb306-6" tabindex="-1"></a>                        <span class="at">pca_name_modality_2 =</span> <span class="st">&quot;protein.pca&quot;</span>,</span>
<span id="cb306-7"><a href="multi-omics-integration.html#cb306-7" tabindex="-1"></a>                        <span class="at">k =</span> <span class="dv">20</span>,</span>
<span id="cb306-8"><a href="multi-omics-integration.html#cb306-8" tabindex="-1"></a>                        <span class="at">integrated_feat_type =</span> <span class="cn">NULL</span>,</span>
<span id="cb306-9"><a href="multi-omics-integration.html#cb306-9" tabindex="-1"></a>                        <span class="at">matrix_result_name =</span> <span class="cn">NULL</span>,</span>
<span id="cb306-10"><a href="multi-omics-integration.html#cb306-10" tabindex="-1"></a>                        <span class="at">w_name_modality_1 =</span> <span class="cn">NULL</span>,</span>
<span id="cb306-11"><a href="multi-omics-integration.html#cb306-11" tabindex="-1"></a>                        <span class="at">w_name_modality_2 =</span> <span class="cn">NULL</span>,</span>
<span id="cb306-12"><a href="multi-omics-integration.html#cb306-12" tabindex="-1"></a>                        <span class="at">verbose =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<ul>
<li>Run Integrated umap</li>
</ul>
<p>Calculate the UMAP using the weights of each feature per spot.</p>
<div class="sourceCode" id="cb307"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb307-1"><a href="multi-omics-integration.html#cb307-1" tabindex="-1"></a>visium_tonsil <span class="ot">&lt;-</span> <span class="fu">runIntegratedUMAP</span>(visium_tonsil,</span>
<span id="cb307-2"><a href="multi-omics-integration.html#cb307-2" tabindex="-1"></a>                                   <span class="at">modality1 =</span> <span class="st">&quot;rna&quot;</span>,</span>
<span id="cb307-3"><a href="multi-omics-integration.html#cb307-3" tabindex="-1"></a>                                   <span class="at">modality2 =</span> <span class="st">&quot;protein&quot;</span>,</span>
<span id="cb307-4"><a href="multi-omics-integration.html#cb307-4" tabindex="-1"></a>                                   <span class="at">spread =</span> <span class="dv">7</span>,</span>
<span id="cb307-5"><a href="multi-omics-integration.html#cb307-5" tabindex="-1"></a>                                   <span class="at">min_dist =</span> <span class="dv">1</span>,</span>
<span id="cb307-6"><a href="multi-omics-integration.html#cb307-6" tabindex="-1"></a>                                   <span class="at">force =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<ul>
<li>Calculate integrated clusters</li>
</ul>
<p>Calculate the multiomics-based Leiden clusters using the weights of each feature per spot.</p>
<div class="sourceCode" id="cb308"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb308-1"><a href="multi-omics-integration.html#cb308-1" tabindex="-1"></a>visium_tonsil <span class="ot">&lt;-</span> <span class="fu">doLeidenCluster</span>(<span class="at">gobject =</span> visium_tonsil,</span>
<span id="cb308-2"><a href="multi-omics-integration.html#cb308-2" tabindex="-1"></a>                                 <span class="at">spat_unit =</span> <span class="st">&quot;cell&quot;</span>,</span>
<span id="cb308-3"><a href="multi-omics-integration.html#cb308-3" tabindex="-1"></a>                                 <span class="at">feat_type =</span> <span class="st">&quot;rna&quot;</span>,</span>
<span id="cb308-4"><a href="multi-omics-integration.html#cb308-4" tabindex="-1"></a>                                 <span class="at">nn_network_to_use =</span> <span class="st">&quot;kNN&quot;</span>,</span>
<span id="cb308-5"><a href="multi-omics-integration.html#cb308-5" tabindex="-1"></a>                                 <span class="at">network_name =</span> <span class="st">&quot;integrated_kNN&quot;</span>,</span>
<span id="cb308-6"><a href="multi-omics-integration.html#cb308-6" tabindex="-1"></a>                                 <span class="at">name =</span> <span class="st">&quot;integrated_leiden_clus&quot;</span>,</span>
<span id="cb308-7"><a href="multi-omics-integration.html#cb308-7" tabindex="-1"></a>                                 <span class="at">resolution =</span> <span class="dv">1</span>)</span></code></pre></div>
<ul>
<li>Visualize the integrated UMAP</li>
</ul>
<p>Plot the integrated UMAP and color the spots using the integrated Leiden clusters.</p>
<div class="sourceCode" id="cb309"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb309-1"><a href="multi-omics-integration.html#cb309-1" tabindex="-1"></a><span class="fu">plotUMAP</span>(<span class="at">gobject =</span> visium_tonsil,</span>
<span id="cb309-2"><a href="multi-omics-integration.html#cb309-2" tabindex="-1"></a>         <span class="at">spat_unit =</span> <span class="st">&quot;cell&quot;</span>,</span>
<span id="cb309-3"><a href="multi-omics-integration.html#cb309-3" tabindex="-1"></a>         <span class="at">feat_type =</span> <span class="st">&quot;rna&quot;</span>,</span>
<span id="cb309-4"><a href="multi-omics-integration.html#cb309-4" tabindex="-1"></a>         <span class="at">cell_color =</span> <span class="st">&quot;integrated_leiden_clus&quot;</span>,</span>
<span id="cb309-5"><a href="multi-omics-integration.html#cb309-5" tabindex="-1"></a>         <span class="at">dim_reduction_name =</span> <span class="st">&quot;integrated.umap&quot;</span>,</span>
<span id="cb309-6"><a href="multi-omics-integration.html#cb309-6" tabindex="-1"></a>         <span class="at">point_size =</span> <span class="dv">2</span>,</span>
<span id="cb309-7"><a href="multi-omics-integration.html#cb309-7" tabindex="-1"></a>         <span class="at">title =</span> <span class="st">&quot;Integrated UMAP using Integrated Leiden clusters&quot;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-430"></span>
<img src="img/03_session3/5-UMAP.png" alt="Integrated UMAP. Color represents the integrated Leiden clusters." width="80%" />
<p class="caption">
Figure 14.8: Integrated UMAP. Color represents the integrated Leiden clusters.
</p>
</div>
<ul>
<li>Visualize spatial plot with integrated clusters</li>
</ul>
<p>Plot the spatial distribution of the integrated Leiden clusters.</p>
<div class="sourceCode" id="cb310"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb310-1"><a href="multi-omics-integration.html#cb310-1" tabindex="-1"></a><span class="fu">spatPlot2D</span>(visium_tonsil,</span>
<span id="cb310-2"><a href="multi-omics-integration.html#cb310-2" tabindex="-1"></a>           <span class="at">spat_unit =</span> <span class="st">&quot;cell&quot;</span>,</span>
<span id="cb310-3"><a href="multi-omics-integration.html#cb310-3" tabindex="-1"></a>           <span class="at">feat_type =</span> <span class="st">&quot;rna&quot;</span>,</span>
<span id="cb310-4"><a href="multi-omics-integration.html#cb310-4" tabindex="-1"></a>           <span class="at">cell_color =</span> <span class="st">&quot;integrated_leiden_clus&quot;</span>,</span>
<span id="cb310-5"><a href="multi-omics-integration.html#cb310-5" tabindex="-1"></a>           <span class="at">point_size =</span> <span class="dv">3</span>,</span>
<span id="cb310-6"><a href="multi-omics-integration.html#cb310-6" tabindex="-1"></a>           <span class="at">show_image =</span> <span class="cn">TRUE</span>,</span>
<span id="cb310-7"><a href="multi-omics-integration.html#cb310-7" tabindex="-1"></a>           <span class="at">title =</span> <span class="st">&quot;Integrated Leiden clustering&quot;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-432"></span>
<img src="img/03_session3/6-spatPlot2D.png" alt="Spatial distribution of the integrated Leiden clusters." width="80%" />
<p class="caption">
Figure 14.9: Spatial distribution of the integrated Leiden clusters.
</p>
</div>
</div>
<div id="session-info-2" class="section level2 hasAnchor" number="14.9">
<h2><span class="header-section-number">14.9</span> Session info<a href="multi-omics-integration.html#session-info-2" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb311"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb311-1"><a href="multi-omics-integration.html#cb311-1" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code></pre></div>
<div class="sourceCode" id="cb312"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb312-1"><a href="multi-omics-integration.html#cb312-1" tabindex="-1"></a>R version <span class="dv">4</span>.<span class="fl">4.1</span> (<span class="dv">2024-06-14</span>)</span>
<span id="cb312-2"><a href="multi-omics-integration.html#cb312-2" tabindex="-1"></a>Platform<span class="sc">:</span> aarch64<span class="sc">-</span>apple<span class="sc">-</span>darwin20</span>
<span id="cb312-3"><a href="multi-omics-integration.html#cb312-3" tabindex="-1"></a>Running under<span class="sc">:</span> macOS Sonoma <span class="fl">14.5</span></span>
<span id="cb312-4"><a href="multi-omics-integration.html#cb312-4" tabindex="-1"></a></span>
<span id="cb312-5"><a href="multi-omics-integration.html#cb312-5" tabindex="-1"></a>Matrix products<span class="sc">:</span> default</span>
<span id="cb312-6"><a href="multi-omics-integration.html#cb312-6" tabindex="-1"></a>BLAS<span class="sc">:</span>   <span class="er">/</span>System<span class="sc">/</span>Library<span class="sc">/</span>Frameworks<span class="sc">/</span>Accelerate.framework<span class="sc">/</span>Versions<span class="sc">/</span>A<span class="sc">/</span>Frameworks<span class="sc">/</span>vecLib.framework<span class="sc">/</span>Versions<span class="sc">/</span>A<span class="sc">/</span>libBLAS.dylib </span>
<span id="cb312-7"><a href="multi-omics-integration.html#cb312-7" tabindex="-1"></a>LAPACK<span class="sc">:</span> <span class="er">/</span>Library<span class="sc">/</span>Frameworks<span class="sc">/</span>R.framework<span class="sc">/</span>Versions<span class="sc">/</span><span class="fl">4.4</span><span class="sc">-</span>arm64<span class="sc">/</span>Resources<span class="sc">/</span>lib<span class="sc">/</span>libRlapack.dylib;  LAPACK version <span class="dv">3</span>.<span class="fl">12.0</span></span>
<span id="cb312-8"><a href="multi-omics-integration.html#cb312-8" tabindex="-1"></a></span>
<span id="cb312-9"><a href="multi-omics-integration.html#cb312-9" tabindex="-1"></a>locale<span class="sc">:</span></span>
<span id="cb312-10"><a href="multi-omics-integration.html#cb312-10" tabindex="-1"></a>[<span class="dv">1</span>] en_US.UTF<span class="dv">-8</span><span class="sc">/</span>en_US.UTF<span class="dv">-8</span><span class="sc">/</span>en_US.UTF<span class="dv">-8</span><span class="sc">/</span>C<span class="sc">/</span>en_US.UTF<span class="dv">-8</span><span class="sc">/</span>en_US.UTF<span class="dv">-8</span></span>
<span id="cb312-11"><a href="multi-omics-integration.html#cb312-11" tabindex="-1"></a></span>
<span id="cb312-12"><a href="multi-omics-integration.html#cb312-12" tabindex="-1"></a>time zone<span class="sc">:</span> America<span class="sc">/</span>New_York</span>
<span id="cb312-13"><a href="multi-omics-integration.html#cb312-13" tabindex="-1"></a>tzcode source<span class="sc">:</span> internal</span>
<span id="cb312-14"><a href="multi-omics-integration.html#cb312-14" tabindex="-1"></a></span>
<span id="cb312-15"><a href="multi-omics-integration.html#cb312-15" tabindex="-1"></a>attached base packages<span class="sc">:</span></span>
<span id="cb312-16"><a href="multi-omics-integration.html#cb312-16" tabindex="-1"></a>[<span class="dv">1</span>] stats     graphics  grDevices utils     datasets  methods   base     </span>
<span id="cb312-17"><a href="multi-omics-integration.html#cb312-17" tabindex="-1"></a></span>
<span id="cb312-18"><a href="multi-omics-integration.html#cb312-18" tabindex="-1"></a>other attached packages<span class="sc">:</span></span>
<span id="cb312-19"><a href="multi-omics-integration.html#cb312-19" tabindex="-1"></a>[<span class="dv">1</span>] Giotto_4.<span class="fl">1.0</span>      GiottoClass_0.<span class="fl">3.3</span></span>
<span id="cb312-20"><a href="multi-omics-integration.html#cb312-20" tabindex="-1"></a></span>
<span id="cb312-21"><a href="multi-omics-integration.html#cb312-21" tabindex="-1"></a>loaded via a <span class="fu">namespace</span> (and not attached)<span class="sc">:</span></span>
<span id="cb312-22"><a href="multi-omics-integration.html#cb312-22" tabindex="-1"></a>  [<span class="dv">1</span>] colorRamp2_0.<span class="fl">1.0</span>            deldir_2<span class="fl">.0</span><span class="dv">-4</span>               </span>
<span id="cb312-23"><a href="multi-omics-integration.html#cb312-23" tabindex="-1"></a>  [<span class="dv">3</span>] rlang_1.<span class="fl">1.4</span>                 magrittr_2.<span class="fl">0.3</span>             </span>
<span id="cb312-24"><a href="multi-omics-integration.html#cb312-24" tabindex="-1"></a>  [<span class="dv">5</span>] RcppAnnoy_0.<span class="fl">0.22</span>            GiottoUtils_0.<span class="fl">1.10</span>         </span>
<span id="cb312-25"><a href="multi-omics-integration.html#cb312-25" tabindex="-1"></a>  [<span class="dv">7</span>] matrixStats_1.<span class="fl">3.0</span>           compiler_4.<span class="fl">4.1</span>             </span>
<span id="cb312-26"><a href="multi-omics-integration.html#cb312-26" tabindex="-1"></a>  [<span class="dv">9</span>] png_0<span class="fl">.1</span><span class="dv">-8</span>                   systemfonts_1.<span class="fl">1.0</span>          </span>
<span id="cb312-27"><a href="multi-omics-integration.html#cb312-27" tabindex="-1"></a> [<span class="dv">11</span>] vctrs_0.<span class="fl">6.5</span>                 reshape2_1.<span class="fl">4.4</span>             </span>
<span id="cb312-28"><a href="multi-omics-integration.html#cb312-28" tabindex="-1"></a> [<span class="dv">13</span>] stringr_1.<span class="fl">5.1</span>               pkgconfig_2.<span class="fl">0.3</span>            </span>
<span id="cb312-29"><a href="multi-omics-integration.html#cb312-29" tabindex="-1"></a> [<span class="dv">15</span>] SpatialExperiment_1.<span class="fl">14.0</span>    crayon_1.<span class="fl">5.3</span>               </span>
<span id="cb312-30"><a href="multi-omics-integration.html#cb312-30" tabindex="-1"></a> [<span class="dv">17</span>] fastmap_1.<span class="fl">2.0</span>               backports_1.<span class="fl">5.0</span>            </span>
<span id="cb312-31"><a href="multi-omics-integration.html#cb312-31" tabindex="-1"></a> [<span class="dv">19</span>] magick_2.<span class="fl">8.4</span>                XVector_0.<span class="fl">44.0</span>             </span>
<span id="cb312-32"><a href="multi-omics-integration.html#cb312-32" tabindex="-1"></a> [<span class="dv">21</span>] labeling_0.<span class="fl">4.3</span>              utf8_1.<span class="fl">2.4</span>                 </span>
<span id="cb312-33"><a href="multi-omics-integration.html#cb312-33" tabindex="-1"></a> [<span class="dv">23</span>] rmarkdown_2<span class="fl">.27</span>              UCSC.utils_1.<span class="fl">0.0</span>           </span>
<span id="cb312-34"><a href="multi-omics-integration.html#cb312-34" tabindex="-1"></a> [<span class="dv">25</span>] ragg_1.<span class="fl">3.2</span>                  purrr_1.<span class="fl">0.2</span>                </span>
<span id="cb312-35"><a href="multi-omics-integration.html#cb312-35" tabindex="-1"></a> [<span class="dv">27</span>] xfun_0<span class="fl">.46</span>                   beachmat_2.<span class="fl">20.0</span>            </span>
<span id="cb312-36"><a href="multi-omics-integration.html#cb312-36" tabindex="-1"></a> [<span class="dv">29</span>] zlibbioc_1.<span class="fl">50.0</span>             GenomeInfoDb_1.<span class="fl">40.1</span>        </span>
<span id="cb312-37"><a href="multi-omics-integration.html#cb312-37" tabindex="-1"></a> [<span class="dv">31</span>] jsonlite_1.<span class="fl">8.8</span>              DelayedArray_0.<span class="fl">30.1</span>        </span>
<span id="cb312-38"><a href="multi-omics-integration.html#cb312-38" tabindex="-1"></a> [<span class="dv">33</span>] BiocParallel_1.<span class="fl">38.0</span>         terra_1<span class="fl">.7</span><span class="dv">-78</span>               </span>
<span id="cb312-39"><a href="multi-omics-integration.html#cb312-39" tabindex="-1"></a> [<span class="dv">35</span>] irlba_2.<span class="dv">3</span>.<span class="fl">5.1</span>               parallel_4.<span class="fl">4.1</span>             </span>
<span id="cb312-40"><a href="multi-omics-integration.html#cb312-40" tabindex="-1"></a> [<span class="dv">37</span>] R6_2.<span class="fl">5.1</span>                    stringi_1.<span class="fl">8.4</span>              </span>
<span id="cb312-41"><a href="multi-omics-integration.html#cb312-41" tabindex="-1"></a> [<span class="dv">39</span>] RColorBrewer_1<span class="fl">.1</span><span class="dv">-3</span>          reticulate_1.<span class="fl">38.0</span>          </span>
<span id="cb312-42"><a href="multi-omics-integration.html#cb312-42" tabindex="-1"></a> [<span class="dv">41</span>] parallelly_1.<span class="fl">37.1</span>           GenomicRanges_1.<span class="fl">56.1</span>       </span>
<span id="cb312-43"><a href="multi-omics-integration.html#cb312-43" tabindex="-1"></a> [<span class="dv">43</span>] scattermore_1<span class="fl">.2</span>             Rcpp_1.<span class="fl">0.13</span>                </span>
<span id="cb312-44"><a href="multi-omics-integration.html#cb312-44" tabindex="-1"></a> [<span class="dv">45</span>] bookdown_0<span class="fl">.40</span>               SummarizedExperiment_1.<span class="fl">34.0</span></span>
<span id="cb312-45"><a href="multi-omics-integration.html#cb312-45" tabindex="-1"></a> [<span class="dv">47</span>] knitr_1<span class="fl">.48</span>                  future.apply_1.<span class="fl">11.2</span>        </span>
<span id="cb312-46"><a href="multi-omics-integration.html#cb312-46" tabindex="-1"></a> [<span class="dv">49</span>] R.utils_2.<span class="fl">12.3</span>              IRanges_2.<span class="fl">38.1</span>             </span>
<span id="cb312-47"><a href="multi-omics-integration.html#cb312-47" tabindex="-1"></a> [<span class="dv">51</span>] Matrix_1<span class="fl">.7</span><span class="dv">-0</span>                igraph_2.<span class="fl">0.3</span>               </span>
<span id="cb312-48"><a href="multi-omics-integration.html#cb312-48" tabindex="-1"></a> [<span class="dv">53</span>] tidyselect_1.<span class="fl">2.1</span>            rstudioapi_0.<span class="fl">16.0</span>          </span>
<span id="cb312-49"><a href="multi-omics-integration.html#cb312-49" tabindex="-1"></a> [<span class="dv">55</span>] abind_1<span class="fl">.4</span><span class="dv">-5</span>                 yaml_2.<span class="fl">3.9</span>                 </span>
<span id="cb312-50"><a href="multi-omics-integration.html#cb312-50" tabindex="-1"></a> [<span class="dv">57</span>] codetools_0<span class="fl">.2</span><span class="dv">-20</span>            listenv_0.<span class="fl">9.1</span>              </span>
<span id="cb312-51"><a href="multi-omics-integration.html#cb312-51" tabindex="-1"></a> [<span class="dv">59</span>] lattice_0<span class="fl">.22</span><span class="dv">-6</span>              tibble_3.<span class="fl">2.1</span>               </span>
<span id="cb312-52"><a href="multi-omics-integration.html#cb312-52" tabindex="-1"></a> [<span class="dv">61</span>] plyr_1.<span class="fl">8.9</span>                  Biobase_2.<span class="fl">64.0</span>             </span>
<span id="cb312-53"><a href="multi-omics-integration.html#cb312-53" tabindex="-1"></a> [<span class="dv">63</span>] withr_3.<span class="fl">0.0</span>                 Rtsne_0<span class="fl">.17</span>                 </span>
<span id="cb312-54"><a href="multi-omics-integration.html#cb312-54" tabindex="-1"></a> [<span class="dv">65</span>] evaluate_0.<span class="fl">24.0</span>             future_1.<span class="fl">33.2</span>              </span>
<span id="cb312-55"><a href="multi-omics-integration.html#cb312-55" tabindex="-1"></a> [<span class="dv">67</span>] pillar_1.<span class="fl">9.0</span>                MatrixGenerics_1.<span class="fl">16.0</span>      </span>
<span id="cb312-56"><a href="multi-omics-integration.html#cb312-56" tabindex="-1"></a> [<span class="dv">69</span>] checkmate_2.<span class="fl">3.1</span>             stats4_4.<span class="fl">4.1</span>               </span>
<span id="cb312-57"><a href="multi-omics-integration.html#cb312-57" tabindex="-1"></a> [<span class="dv">71</span>] plotly_4.<span class="fl">10.4</span>               generics_0.<span class="fl">1.3</span>             </span>
<span id="cb312-58"><a href="multi-omics-integration.html#cb312-58" tabindex="-1"></a> [<span class="dv">73</span>] dbscan_1<span class="fl">.2</span><span class="dv">-0</span>                sp_2<span class="fl">.1</span><span class="dv">-4</span>                   </span>
<span id="cb312-59"><a href="multi-omics-integration.html#cb312-59" tabindex="-1"></a> [<span class="dv">75</span>] S4Vectors_0.<span class="fl">42.1</span>            ggplot2_3.<span class="fl">5.1</span>              </span>
<span id="cb312-60"><a href="multi-omics-integration.html#cb312-60" tabindex="-1"></a> [<span class="dv">77</span>] munsell_0.<span class="fl">5.1</span>               scales_1.<span class="fl">3.0</span>               </span>
<span id="cb312-61"><a href="multi-omics-integration.html#cb312-61" tabindex="-1"></a> [<span class="dv">79</span>] globals_0.<span class="fl">16.3</span>              gtools_3.<span class="fl">9.5</span>               </span>
<span id="cb312-62"><a href="multi-omics-integration.html#cb312-62" tabindex="-1"></a> [<span class="dv">81</span>] glue_1.<span class="fl">7.0</span>                  lazyeval_0.<span class="fl">2.2</span>             </span>
<span id="cb312-63"><a href="multi-omics-integration.html#cb312-63" tabindex="-1"></a> [<span class="dv">83</span>] tools_4.<span class="fl">4.1</span>                 GiottoVisuals_0.<span class="fl">2.4</span>        </span>
<span id="cb312-64"><a href="multi-omics-integration.html#cb312-64" tabindex="-1"></a> [<span class="dv">85</span>] data.table_1.<span class="fl">15.4</span>           ScaledMatrix_1.<span class="fl">12.0</span>        </span>
<span id="cb312-65"><a href="multi-omics-integration.html#cb312-65" tabindex="-1"></a> [<span class="dv">87</span>] cowplot_1.<span class="fl">1.3</span>               grid_4.<span class="fl">4.1</span>                 </span>
<span id="cb312-66"><a href="multi-omics-integration.html#cb312-66" tabindex="-1"></a> [<span class="dv">89</span>] tidyr_1.<span class="fl">3.1</span>                 colorspace_2<span class="fl">.1</span><span class="dv">-0</span>           </span>
<span id="cb312-67"><a href="multi-omics-integration.html#cb312-67" tabindex="-1"></a> [<span class="dv">91</span>] SingleCellExperiment_1.<span class="fl">26.0</span> GenomeInfoDbData_1.<span class="fl">2.12</span>    </span>
<span id="cb312-68"><a href="multi-omics-integration.html#cb312-68" tabindex="-1"></a> [<span class="dv">93</span>] BiocSingular_1.<span class="fl">20.0</span>         rsvd_1.<span class="fl">0.5</span>                 </span>
<span id="cb312-69"><a href="multi-omics-integration.html#cb312-69" tabindex="-1"></a> [<span class="dv">95</span>] cli_3.<span class="fl">6.3</span>                   textshaping_0.<span class="fl">4.0</span>          </span>
<span id="cb312-70"><a href="multi-omics-integration.html#cb312-70" tabindex="-1"></a> [<span class="dv">97</span>] fansi_1.<span class="fl">0.6</span>                 S4Arrays_1.<span class="fl">4.1</span>             </span>
<span id="cb312-71"><a href="multi-omics-integration.html#cb312-71" tabindex="-1"></a> [<span class="dv">99</span>] viridisLite_0.<span class="fl">4.2</span>           dplyr_1.<span class="fl">1.4</span>                </span>
<span id="cb312-72"><a href="multi-omics-integration.html#cb312-72" tabindex="-1"></a>[<span class="dv">101</span>] uwot_0.<span class="fl">2.2</span>                  gtable_0.<span class="fl">3.5</span>               </span>
<span id="cb312-73"><a href="multi-omics-integration.html#cb312-73" tabindex="-1"></a>[<span class="dv">103</span>] R.methodsS3_1.<span class="fl">8.2</span>           digest_0.<span class="fl">6.36</span>              </span>
<span id="cb312-74"><a href="multi-omics-integration.html#cb312-74" tabindex="-1"></a>[<span class="dv">105</span>] BiocGenerics_0.<span class="fl">50.0</span>         SparseArray_1.<span class="fl">4.8</span>          </span>
<span id="cb312-75"><a href="multi-omics-integration.html#cb312-75" tabindex="-1"></a>[<span class="dv">107</span>] ggrepel_0.<span class="fl">9.5</span>               farver_2.<span class="fl">1.2</span>               </span>
<span id="cb312-76"><a href="multi-omics-integration.html#cb312-76" tabindex="-1"></a>[<span class="dv">109</span>] rjson_0.<span class="fl">2.21</span>                htmlwidgets_1.<span class="fl">6.4</span>          </span>
<span id="cb312-77"><a href="multi-omics-integration.html#cb312-77" tabindex="-1"></a>[<span class="dv">111</span>] htmltools_0.<span class="dv">5</span>.<span class="fl">8.1</span>           R.oo_1.<span class="fl">26.0</span>                </span>
<span id="cb312-78"><a href="multi-omics-integration.html#cb312-78" tabindex="-1"></a>[<span class="dv">113</span>] lifecycle_1.<span class="fl">0.4</span>             httr_1.<span class="fl">4.7</span>  </span></code></pre></div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="spatial-multi-modal-analysis.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="interoperability-with-other-frameworks.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
