<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>15 Multi-omics integration | Workshop: Spatial multi-omics data analysis with Giotto Suite</title>
  <meta name="description" content="15 Multi-omics integration | Workshop: Spatial multi-omics data analysis with Giotto Suite" />
  <meta name="generator" content="bookdown 0.39 and GitBook 2.6.7" />

  <meta property="og:title" content="15 Multi-omics integration | Workshop: Spatial multi-omics data analysis with Giotto Suite" />
  <meta property="og:type" content="book" />
  
  
  <meta name="github-repo" content="drieslab/giotto_workshop_2024" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="15 Multi-omics integration | Workshop: Spatial multi-omics data analysis with Giotto Suite" />
  
  
  

<meta name="author" content="Ruben Dries, Jiaji George Chen, Joselyn Cristina ChÃ¡vez-Fuentes, Junxiang Xu ,Edward Ruiz, Jeff Sheridan, Iqra Amin, Wen Wang" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="spatial-multi-modal-analysis.html"/>
<link rel="next" href="interoperability-with-other-frameworks.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Giotto Suite Workshop 2024</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#instructors"><i class="fa fa-check"></i><b>1.1</b> Instructors</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#topics-and-schedule"><i class="fa fa-check"></i><b>1.2</b> Topics and Schedule:</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#license"><i class="fa fa-check"></i><b>1.3</b> License</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="datasets-to-download.html"><a href="datasets-to-download.html"><i class="fa fa-check"></i><b>2</b> Datasets to download</a>
<ul>
<li class="chapter" data-level="2.1" data-path="datasets-to-download.html"><a href="datasets-to-download.html#zenodo-repository"><i class="fa fa-check"></i><b>2.1</b> Zenodo repository</a></li>
<li class="chapter" data-level="2.2" data-path="datasets-to-download.html"><a href="datasets-to-download.html#x-genomics-visium-mouse-brain-section-coronal-dataset"><i class="fa fa-check"></i><b>2.2</b> 10X Genomics Visium Mouse Brain Section (Coronal) dataset</a></li>
<li class="chapter" data-level="2.3" data-path="datasets-to-download.html"><a href="datasets-to-download.html#x-genomics-visium-hd-ffpe-human-colon-cancer"><i class="fa fa-check"></i><b>2.3</b> 10X Genomics Visium HD: FFPE Human Colon Cancer</a></li>
<li class="chapter" data-level="2.4" data-path="datasets-to-download.html"><a href="datasets-to-download.html#x-genomics-multi-modal-dataset"><i class="fa fa-check"></i><b>2.4</b> 10X Genomics multi-modal dataset</a></li>
<li class="chapter" data-level="2.5" data-path="datasets-to-download.html"><a href="datasets-to-download.html#x-genomics-multi-omics-visium-cytassist-human-tonsil-dataset"><i class="fa fa-check"></i><b>2.5</b> 10X Genomics multi-omics Visium CytAssist Human Tonsil dataset</a></li>
<li class="chapter" data-level="2.6" data-path="datasets-to-download.html"><a href="datasets-to-download.html#x-genomics-human-prostate-cancer-adenocarcinoma-with-invasive-carcinoma-ffpe"><i class="fa fa-check"></i><b>2.6</b> 10X Genomics Human Prostate Cancer Adenocarcinoma with Invasive Carcinoma (FFPE)</a></li>
<li class="chapter" data-level="2.7" data-path="datasets-to-download.html"><a href="datasets-to-download.html#x-genomics-normal-human-prostate-ffpe"><i class="fa fa-check"></i><b>2.7</b> 10X Genomics Normal Human Prostate (FFPE)</a></li>
<li class="chapter" data-level="2.8" data-path="datasets-to-download.html"><a href="datasets-to-download.html#xenium"><i class="fa fa-check"></i><b>2.8</b> Xenium</a></li>
<li class="chapter" data-level="2.9" data-path="datasets-to-download.html"><a href="datasets-to-download.html#merfish-cortex-dataset"><i class="fa fa-check"></i><b>2.9</b> MERFISH cortex dataset</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="needed-packages.html"><a href="needed-packages.html"><i class="fa fa-check"></i><b>3</b> Needed packages</a>
<ul>
<li class="chapter" data-level="3.1" data-path="needed-packages.html"><a href="needed-packages.html#cran-dependencies"><i class="fa fa-check"></i><b>3.1</b> CRAN dependencies:</a></li>
<li class="chapter" data-level="3.2" data-path="needed-packages.html"><a href="needed-packages.html#terra-installation"><i class="fa fa-check"></i><b>3.2</b> <em>terra</em> installation</a></li>
<li class="chapter" data-level="3.3" data-path="needed-packages.html"><a href="needed-packages.html#matrix-installation"><i class="fa fa-check"></i><b>3.3</b> <em>Matrix</em> installation</a></li>
<li class="chapter" data-level="3.4" data-path="needed-packages.html"><a href="needed-packages.html#giotto-installation"><i class="fa fa-check"></i><b>3.4</b> <em>Giotto</em> installation</a></li>
<li class="chapter" data-level="3.5" data-path="needed-packages.html"><a href="needed-packages.html#irlba-install"><i class="fa fa-check"></i><b>3.5</b> <em>irlba</em> install</a></li>
<li class="chapter" data-level="3.6" data-path="needed-packages.html"><a href="needed-packages.html#arrow-install"><i class="fa fa-check"></i><b>3.6</b> <em>arrow</em> install</a></li>
<li class="chapter" data-level="3.7" data-path="needed-packages.html"><a href="needed-packages.html#bioconductor-dependencies"><i class="fa fa-check"></i><b>3.7</b> Bioconductor dependencies:</a></li>
<li class="chapter" data-level="3.8" data-path="needed-packages.html"><a href="needed-packages.html#cran-packages"><i class="fa fa-check"></i><b>3.8</b> CRAN packages:</a></li>
<li class="chapter" data-level="3.9" data-path="needed-packages.html"><a href="needed-packages.html#packages-from-github"><i class="fa fa-check"></i><b>3.9</b> Packages from GitHub</a></li>
<li class="chapter" data-level="3.10" data-path="needed-packages.html"><a href="needed-packages.html#python-environments"><i class="fa fa-check"></i><b>3.10</b> Python environments</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html"><i class="fa fa-check"></i><b>4</b> Spatial omics technologies</a>
<ul>
<li class="chapter" data-level="4.1" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#presentation"><i class="fa fa-check"></i><b>4.1</b> Presentation</a></li>
<li class="chapter" data-level="4.2" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#short-summary"><i class="fa fa-check"></i><b>4.2</b> Short summary</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#why-do-we-need-spatial-omics-technologies"><i class="fa fa-check"></i><b>4.2.1</b> Why do we need spatial omics technologies?</a></li>
<li class="chapter" data-level="4.2.2" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#what-is-spatial-omics"><i class="fa fa-check"></i><b>4.2.2</b> What is spatial omics?</a></li>
<li class="chapter" data-level="4.2.3" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#what-are-the-main-spatial-omics-technologies"><i class="fa fa-check"></i><b>4.2.3</b> What are the main spatial omics technologies?</a></li>
<li class="chapter" data-level="4.2.4" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#other-spatial-omics-atac-seq-cuttag-lipidomics-etc"><i class="fa fa-check"></i><b>4.2.4</b> Other Spatial omics: ATAC-seq, CUT&amp;Tag, lipidomics, etc</a></li>
<li class="chapter" data-level="4.2.5" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#what-are-the-different-types-of-spatial-downstream-analyses"><i class="fa fa-check"></i><b>4.2.5</b> What are the different types of spatial downstream analyses?</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html"><i class="fa fa-check"></i><b>5</b> Introduction to the Giotto package</a>
<ul>
<li class="chapter" data-level="5.1" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html#presentation-1"><i class="fa fa-check"></i><b>5.1</b> Presentation</a></li>
<li class="chapter" data-level="5.2" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html#ecosystem"><i class="fa fa-check"></i><b>5.2</b> Ecosystem</a></li>
<li class="chapter" data-level="5.3" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html#installation-python-environment"><i class="fa fa-check"></i><b>5.3</b> Installation + python environment</a></li>
<li class="chapter" data-level="5.4" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html#giotto-instructions"><i class="fa fa-check"></i><b>5.4</b> Giotto instructions</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html"><i class="fa fa-check"></i><b>6</b> Data formatting and Pre-processing</a>
<ul>
<li class="chapter" data-level="6.1" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#data-formats"><i class="fa fa-check"></i><b>6.1</b> Data formats</a></li>
<li class="chapter" data-level="6.2" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#pre-processing"><i class="fa fa-check"></i><b>6.2</b> Pre-processing</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html"><i class="fa fa-check"></i><b>7</b> Creating a Giotto object</a>
<ul>
<li class="chapter" data-level="7.0.1" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#used-packages"><i class="fa fa-check"></i><b>7.0.1</b> Used packages</a></li>
<li class="chapter" data-level="7.1" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#overview"><i class="fa fa-check"></i><b>7.1</b> Overview</a></li>
<li class="chapter" data-level="7.2" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#from-matrix-locations"><i class="fa fa-check"></i><b>7.2</b> From matrix + locations</a></li>
<li class="chapter" data-level="7.3" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#from-subcellular-raw-data-transcripts-or-images-polygons"><i class="fa fa-check"></i><b>7.3</b> From subcellular raw data (transcripts or images) + polygons</a></li>
<li class="chapter" data-level="7.4" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#from-piece-wise"><i class="fa fa-check"></i><b>7.4</b> From piece-wise</a></li>
<li class="chapter" data-level="7.5" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#using-convenience-functions-for-popular-technologies-vizgen-xenium-cosmx"><i class="fa fa-check"></i><b>7.5</b> Using convenience functions for popular technologies (Vizgen, Xenium, CosMx, â¦)</a></li>
<li class="chapter" data-level="7.6" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#spatial-plots"><i class="fa fa-check"></i><b>7.6</b> Spatial plots</a></li>
<li class="chapter" data-level="7.7" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#subsetting"><i class="fa fa-check"></i><b>7.7</b> Subsetting</a></li>
<li class="chapter" data-level="7.8" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#mini-objects-giottodata"><i class="fa fa-check"></i><b>7.8</b> Mini objects &amp; GiottoData</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="visium-part-i.html"><a href="visium-part-i.html"><i class="fa fa-check"></i><b>8</b> Visium Part I</a>
<ul>
<li class="chapter" data-level="8.1" data-path="visium-part-i.html"><a href="visium-part-i.html#introduction-to-the-spatial-dataset"><i class="fa fa-check"></i><b>8.1</b> Introduction to the spatial dataset</a></li>
<li class="chapter" data-level="8.2" data-path="visium-part-i.html"><a href="visium-part-i.html#download-dataset"><i class="fa fa-check"></i><b>8.2</b> Download dataset</a></li>
<li class="chapter" data-level="8.3" data-path="visium-part-i.html"><a href="visium-part-i.html#create-the-giotto-object"><i class="fa fa-check"></i><b>8.3</b> Create the Giotto object</a></li>
<li class="chapter" data-level="8.4" data-path="visium-part-i.html"><a href="visium-part-i.html#subset-on-spots-that-were-covered-by-tissue"><i class="fa fa-check"></i><b>8.4</b> Subset on spots that were covered by tissue</a></li>
<li class="chapter" data-level="8.5" data-path="visium-part-i.html"><a href="visium-part-i.html#quality-control"><i class="fa fa-check"></i><b>8.5</b> Quality control</a></li>
<li class="chapter" data-level="8.6" data-path="visium-part-i.html"><a href="visium-part-i.html#filtering"><i class="fa fa-check"></i><b>8.6</b> Filtering</a></li>
<li class="chapter" data-level="8.7" data-path="visium-part-i.html"><a href="visium-part-i.html#normalization"><i class="fa fa-check"></i><b>8.7</b> Normalization</a></li>
<li class="chapter" data-level="8.8" data-path="visium-part-i.html"><a href="visium-part-i.html#feature-selection"><i class="fa fa-check"></i><b>8.8</b> Feature selection</a>
<ul>
<li class="chapter" data-level="8.8.1" data-path="visium-part-i.html"><a href="visium-part-i.html#highly-variable-features"><i class="fa fa-check"></i><b>8.8.1</b> Highly Variable Features:</a></li>
</ul></li>
<li class="chapter" data-level="8.9" data-path="visium-part-i.html"><a href="visium-part-i.html#dimension-reduction"><i class="fa fa-check"></i><b>8.9</b> Dimension Reduction</a>
<ul>
<li class="chapter" data-level="8.9.1" data-path="visium-part-i.html"><a href="visium-part-i.html#pca"><i class="fa fa-check"></i><b>8.9.1</b> PCA</a></li>
<li class="chapter" data-level="8.9.2" data-path="visium-part-i.html"><a href="visium-part-i.html#umap"><i class="fa fa-check"></i><b>8.9.2</b> UMAP</a></li>
<li class="chapter" data-level="8.9.3" data-path="visium-part-i.html"><a href="visium-part-i.html#t-sne"><i class="fa fa-check"></i><b>8.9.3</b> t-SNE</a></li>
</ul></li>
<li class="chapter" data-level="8.10" data-path="visium-part-i.html"><a href="visium-part-i.html#clustering"><i class="fa fa-check"></i><b>8.10</b> Clustering</a>
<ul>
<li class="chapter" data-level="8.10.1" data-path="visium-part-i.html"><a href="visium-part-i.html#calculate-leiden-clustering"><i class="fa fa-check"></i><b>8.10.1</b> Calculate Leiden clustering</a></li>
<li class="chapter" data-level="8.10.2" data-path="visium-part-i.html"><a href="visium-part-i.html#calculate-louvain-clustering"><i class="fa fa-check"></i><b>8.10.2</b> Calculate Louvain clustering</a></li>
</ul></li>
<li class="chapter" data-level="8.11" data-path="visium-part-i.html"><a href="visium-part-i.html#save-the-object"><i class="fa fa-check"></i><b>8.11</b> Save the object</a></li>
<li class="chapter" data-level="8.12" data-path="visium-part-i.html"><a href="visium-part-i.html#session-info"><i class="fa fa-check"></i><b>8.12</b> Session info</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="visium-part-ii.html"><a href="visium-part-ii.html"><i class="fa fa-check"></i><b>9</b> Visium Part II</a>
<ul>
<li class="chapter" data-level="9.1" data-path="visium-part-ii.html"><a href="visium-part-ii.html#load-the-object"><i class="fa fa-check"></i><b>9.1</b> Load the object</a></li>
<li class="chapter" data-level="9.2" data-path="visium-part-ii.html"><a href="visium-part-ii.html#differential-expression"><i class="fa fa-check"></i><b>9.2</b> Differential expression</a>
<ul>
<li class="chapter" data-level="9.2.1" data-path="visium-part-ii.html"><a href="visium-part-ii.html#gini-markers"><i class="fa fa-check"></i><b>9.2.1</b> Gini markers</a></li>
<li class="chapter" data-level="9.2.2" data-path="visium-part-ii.html"><a href="visium-part-ii.html#scran-markers"><i class="fa fa-check"></i><b>9.2.2</b> Scran markers</a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="visium-part-ii.html"><a href="visium-part-ii.html#enrichment-deconvolution"><i class="fa fa-check"></i><b>9.3</b> Enrichment &amp; Deconvolution</a>
<ul>
<li class="chapter" data-level="9.3.1" data-path="visium-part-ii.html"><a href="visium-part-ii.html#pagerank"><i class="fa fa-check"></i><b>9.3.1</b> PAGE/Rank</a></li>
<li class="chapter" data-level="9.3.2" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatialdwls"><i class="fa fa-check"></i><b>9.3.2</b> SpatialDWLS</a></li>
</ul></li>
<li class="chapter" data-level="9.4" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatial-expression-patterns"><i class="fa fa-check"></i><b>9.4</b> Spatial expression patterns</a>
<ul>
<li class="chapter" data-level="9.4.1" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatial-variable-genes"><i class="fa fa-check"></i><b>9.4.1</b> Spatial variable genes</a></li>
<li class="chapter" data-level="9.4.2" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatial-co-expression-modules"><i class="fa fa-check"></i><b>9.4.2</b> Spatial co-expression modules</a></li>
</ul></li>
<li class="chapter" data-level="9.5" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatially-informed-clusters"><i class="fa fa-check"></i><b>9.5</b> Spatially informed clusters</a></li>
<li class="chapter" data-level="9.6" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatial-domains-hmrf"><i class="fa fa-check"></i><b>9.6</b> Spatial domains HMRF</a></li>
<li class="chapter" data-level="9.7" data-path="visium-part-ii.html"><a href="visium-part-ii.html#interactive-tools"><i class="fa fa-check"></i><b>9.7</b> Interactive tools</a></li>
<li class="chapter" data-level="9.8" data-path="visium-part-ii.html"><a href="visium-part-ii.html#session-info-1"><i class="fa fa-check"></i><b>9.8</b> Session info</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="visium-hd.html"><a href="visium-hd.html"><i class="fa fa-check"></i><b>10</b> Visium HD</a>
<ul>
<li class="chapter" data-level="10.1" data-path="visium-hd.html"><a href="visium-hd.html#objective"><i class="fa fa-check"></i><b>10.1</b> Objective</a></li>
<li class="chapter" data-level="10.2" data-path="visium-hd.html"><a href="visium-hd.html#background"><i class="fa fa-check"></i><b>10.2</b> Background</a>
<ul>
<li class="chapter" data-level="10.2.1" data-path="visium-hd.html"><a href="visium-hd.html#visium-hd-technology"><i class="fa fa-check"></i><b>10.2.1</b> Visium HD Technology</a></li>
<li class="chapter" data-level="10.2.2" data-path="visium-hd.html"><a href="visium-hd.html#colorectal-cancer-sample"><i class="fa fa-check"></i><b>10.2.2</b> Colorectal Cancer Sample</a></li>
</ul></li>
<li class="chapter" data-level="10.3" data-path="visium-hd.html"><a href="visium-hd.html#data-ingestion"><i class="fa fa-check"></i><b>10.3</b> Data Ingestion</a>
<ul>
<li class="chapter" data-level="10.3.1" data-path="visium-hd.html"><a href="visium-hd.html#visium-hd-output-data-format"><i class="fa fa-check"></i><b>10.3.1</b> Visium HD output data format</a></li>
<li class="chapter" data-level="10.3.2" data-path="visium-hd.html"><a href="visium-hd.html#read-in-raw-data"><i class="fa fa-check"></i><b>10.3.2</b> Read in raw data</a></li>
<li class="chapter" data-level="10.3.3" data-path="visium-hd.html"><a href="visium-hd.html#giotto-visium-hd-convenience-function"><i class="fa fa-check"></i><b>10.3.3</b> Giotto Visium HD convenience function</a></li>
</ul></li>
<li class="chapter" data-level="10.4" data-path="visium-hd.html"><a href="visium-hd.html#tiling-and-aggregation"><i class="fa fa-check"></i><b>10.4</b> Tiling and aggregation</a></li>
<li class="chapter" data-level="10.5" data-path="visium-hd.html"><a href="visium-hd.html#scalability-and-projection-functions"><i class="fa fa-check"></i><b>10.5</b> Scalability and projection functions</a></li>
<li class="chapter" data-level="10.6" data-path="visium-hd.html"><a href="visium-hd.html#spatial-expression-patterns-1"><i class="fa fa-check"></i><b>10.6</b> Spatial expression patterns</a></li>
<li class="chapter" data-level="10.7" data-path="visium-hd.html"><a href="visium-hd.html#spatial-co-expression-modules-1"><i class="fa fa-check"></i><b>10.7</b> Spatial co-expression modules</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="xenium-1.html"><a href="xenium-1.html"><i class="fa fa-check"></i><b>11</b> Xenium</a>
<ul>
<li class="chapter" data-level="11.1" data-path="xenium-1.html"><a href="xenium-1.html#introduction-to-spatial-dataset"><i class="fa fa-check"></i><b>11.1</b> Introduction to spatial dataset</a></li>
<li class="chapter" data-level="11.2" data-path="xenium-1.html"><a href="xenium-1.html#additional-package-requirements"><i class="fa fa-check"></i><b>11.2</b> Additional package requirements</a></li>
<li class="chapter" data-level="11.3" data-path="xenium-1.html"><a href="xenium-1.html#introduce-mini-object"><i class="fa fa-check"></i><b>11.3</b> Introduce mini object</a></li>
<li class="chapter" data-level="11.4" data-path="xenium-1.html"><a href="xenium-1.html#read-in-raw-data-1"><i class="fa fa-check"></i><b>11.4</b> Read in raw data</a></li>
<li class="chapter" data-level="11.5" data-path="xenium-1.html"><a href="xenium-1.html#overlap-txs-polygons"><i class="fa fa-check"></i><b>11.5</b> Overlap txs &amp; polygons</a></li>
<li class="chapter" data-level="11.6" data-path="xenium-1.html"><a href="xenium-1.html#transcript-enrichment-gsea"><i class="fa fa-check"></i><b>11.6</b> Transcript enrichment GSEA</a></li>
<li class="chapter" data-level="11.7" data-path="xenium-1.html"><a href="xenium-1.html#spatial-location-analysis"><i class="fa fa-check"></i><b>11.7</b> Spatial location analysis</a></li>
<li class="chapter" data-level="11.8" data-path="xenium-1.html"><a href="xenium-1.html#visualizations"><i class="fa fa-check"></i><b>11.8</b> Visualizations</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="spatial-proteomics-multiplex-if.html"><a href="spatial-proteomics-multiplex-if.html"><i class="fa fa-check"></i><b>12</b> Spatial proteomics: multiplex IF</a>
<ul>
<li class="chapter" data-level="12.1" data-path="spatial-proteomics-multiplex-if.html"><a href="spatial-proteomics-multiplex-if.html#read-in-raw-data-2"><i class="fa fa-check"></i><b>12.1</b> Read in raw data</a></li>
<li class="chapter" data-level="12.2" data-path="spatial-proteomics-multiplex-if.html"><a href="spatial-proteomics-multiplex-if.html#overlap-intensity-workflows"><i class="fa fa-check"></i><b>12.2</b> Overlap intensity &amp; workflows</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html"><i class="fa fa-check"></i><b>13</b> Working with multiple samples</a>
<ul>
<li class="chapter" data-level="13.1" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#objective-1"><i class="fa fa-check"></i><b>13.1</b> Objective</a></li>
<li class="chapter" data-level="13.2" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#background-1"><i class="fa fa-check"></i><b>13.2</b> Background</a>
<ul>
<li class="chapter" data-level="13.2.1" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#dataset"><i class="fa fa-check"></i><b>13.2.1</b> Dataset</a></li>
<li class="chapter" data-level="13.2.2" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#visium-technology"><i class="fa fa-check"></i><b>13.2.2</b> Visium technology</a></li>
</ul></li>
<li class="chapter" data-level="13.3" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#create-individual-giotto-objects"><i class="fa fa-check"></i><b>13.3</b> Create individual giotto objects</a>
<ul>
<li class="chapter" data-level="13.3.1" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#download-the-data"><i class="fa fa-check"></i><b>13.3.1</b> Download the data</a></li>
<li class="chapter" data-level="13.3.2" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#create-giotto-instructions"><i class="fa fa-check"></i><b>13.3.2</b> create giotto instructions</a></li>
<li class="chapter" data-level="13.3.3" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#load-visium-data-into-giotto"><i class="fa fa-check"></i><b>13.3.3</b> load visium data into Giotto</a></li>
<li class="chapter" data-level="13.3.4" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#healthy-prostate-tissue-coverage"><i class="fa fa-check"></i><b>13.3.4</b> Healthy prostate tissue coverage</a></li>
<li class="chapter" data-level="13.3.5" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#adenocarcinoma-prostate-tissue-coverage"><i class="fa fa-check"></i><b>13.3.5</b> Adenocarcinoma prostate tissue coverage</a></li>
<li class="chapter" data-level="13.3.6" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#showing-the-data-strucutre-for-the-inidividual-objects"><i class="fa fa-check"></i><b>13.3.6</b> Showing the data strucutre for the inidividual objects</a></li>
</ul></li>
<li class="chapter" data-level="13.4" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#join-giotto-objects"><i class="fa fa-check"></i><b>13.4</b> Join Giotto Objects</a></li>
<li class="chapter" data-level="13.5" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#visualizing-combined-datasets"><i class="fa fa-check"></i><b>13.5</b> Visualizing combined datasets</a>
<ul>
<li class="chapter" data-level="13.5.1" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#vizualizing-in-the-same-plot"><i class="fa fa-check"></i><b>13.5.1</b> Vizualizing in the same plot</a></li>
<li class="chapter" data-level="13.5.2" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#vizualizing-on-separate-plots"><i class="fa fa-check"></i><b>13.5.2</b> Vizualizing on separate plots</a></li>
</ul></li>
<li class="chapter" data-level="13.6" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#splitting-combined-dataset"><i class="fa fa-check"></i><b>13.6</b> Splitting combined dataset</a></li>
<li class="chapter" data-level="13.7" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#analyzing-joined-objects"><i class="fa fa-check"></i><b>13.7</b> Analyzing joined objects</a>
<ul>
<li class="chapter" data-level="13.7.1" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#normalization-and-adding-statistics"><i class="fa fa-check"></i><b>13.7.1</b> Normalization and adding statistics</a></li>
<li class="chapter" data-level="13.7.2" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#clustering-the-datasets"><i class="fa fa-check"></i><b>13.7.2</b> Clustering the datasets</a></li>
<li class="chapter" data-level="13.7.3" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#vizualizing-spatial-location-of-clusters"><i class="fa fa-check"></i><b>13.7.3</b> Vizualizing spatial location of clusters</a></li>
<li class="chapter" data-level="13.7.4" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#vizualizing-tissue-contribution-to-clusters"><i class="fa fa-check"></i><b>13.7.4</b> Vizualizing tissue contribution to clusters</a></li>
</ul></li>
<li class="chapter" data-level="13.8" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#perform-harmony-and-default-workflows"><i class="fa fa-check"></i><b>13.8</b> Perform Harmony and default workflows</a>
<ul>
<li class="chapter" data-level="13.8.1" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#clustering-harmonized-object"><i class="fa fa-check"></i><b>13.8.1</b> Clustering harmonized object</a></li>
<li class="chapter" data-level="13.8.2" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#vizualizing-the-tissue-contribution"><i class="fa fa-check"></i><b>13.8.2</b> Vizualizing the tissue contribution</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="14" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html"><i class="fa fa-check"></i><b>14</b> Spatial multi-modal analysis</a>
<ul>
<li class="chapter" data-level="14.1" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#overview-1"><i class="fa fa-check"></i><b>14.1</b> Overview</a></li>
<li class="chapter" data-level="14.2" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#spatial-manipulation"><i class="fa fa-check"></i><b>14.2</b> Spatial manipulation</a>
<ul>
<li class="chapter" data-level="14.2.1" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#spatial-transforms"><i class="fa fa-check"></i><b>14.2.1</b> Spatial transforms:</a></li>
<li class="chapter" data-level="14.2.2" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#spatial-utilities"><i class="fa fa-check"></i><b>14.2.2</b> Spatial utilities:</a></li>
<li class="chapter" data-level="14.2.3" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#spatial-classes"><i class="fa fa-check"></i><b>14.2.3</b> Spatial classes:</a></li>
</ul></li>
<li class="chapter" data-level="14.3" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#image-transforms"><i class="fa fa-check"></i><b>14.3</b> Image transforms</a></li>
<li class="chapter" data-level="14.4" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#co-registration-of-datasets"><i class="fa fa-check"></i><b>14.4</b> Co-registration of datasets</a></li>
<li class="chapter" data-level="14.5" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#examples-in-giotto-suite-manuscript"><i class="fa fa-check"></i><b>14.5</b> Examples in giotto suite manuscript</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html"><i class="fa fa-check"></i><b>15</b> Multi-omics integration</a>
<ul>
<li class="chapter" data-level="15.1" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#introduction-to-the-spatial-dataset-1"><i class="fa fa-check"></i><b>15.1</b> Introduction to the spatial dataset</a></li>
<li class="chapter" data-level="15.2" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#download-dataset-1"><i class="fa fa-check"></i><b>15.2</b> Download dataset</a></li>
<li class="chapter" data-level="15.3" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#create-the-giotto-object-1"><i class="fa fa-check"></i><b>15.3</b> Create the Giotto object</a></li>
<li class="chapter" data-level="15.4" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#subset-on-spots-that-were-covered-by-tissue-1"><i class="fa fa-check"></i><b>15.4</b> Subset on spots that were covered by tissue</a></li>
<li class="chapter" data-level="15.5" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#rna-processing"><i class="fa fa-check"></i><b>15.5</b> RNA processing</a></li>
<li class="chapter" data-level="15.6" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#protein-processing"><i class="fa fa-check"></i><b>15.6</b> Protein processing</a></li>
<li class="chapter" data-level="15.7" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#multi-omics-integration-1"><i class="fa fa-check"></i><b>15.7</b> Multi-omics integration</a></li>
<li class="chapter" data-level="15.8" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#session-info-2"><i class="fa fa-check"></i><b>15.8</b> Session info</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html"><i class="fa fa-check"></i><b>16</b> Interoperability with other frameworks</a>
<ul>
<li class="chapter" data-level="16.1" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#load-giotto-object"><i class="fa fa-check"></i><b>16.1</b> Load Giotto object</a></li>
<li class="chapter" data-level="16.2" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#seurat"><i class="fa fa-check"></i><b>16.2</b> Seurat</a>
<ul>
<li class="chapter" data-level="16.2.1" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#conversion-of-giotto-object-to-seurat-object"><i class="fa fa-check"></i><b>16.2.1</b> Conversion of Giotto Object to Seurat Object</a></li>
<li class="chapter" data-level="16.2.2" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#conversion-of-seurat-object-back-to-giotto-object"><i class="fa fa-check"></i><b>16.2.2</b> Conversion of Seurat object Back to Giotto Object</a></li>
</ul></li>
<li class="chapter" data-level="16.3" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#spatialexperiment"><i class="fa fa-check"></i><b>16.3</b> SpatialExperiment</a>
<ul>
<li class="chapter" data-level="16.3.1" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#convert-giotto-object-to-spatialexperiment-object"><i class="fa fa-check"></i><b>16.3.1</b> Convert Giotto Object to SpatialExperiment Object</a></li>
<li class="chapter" data-level="16.3.2" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#conversion-of-spatialexperiment-object-back-to-giotto"><i class="fa fa-check"></i><b>16.3.2</b> Conversion of SpatialExperiment object back to Giotto</a></li>
</ul></li>
<li class="chapter" data-level="16.4" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#anndata"><i class="fa fa-check"></i><b>16.4</b> AnnData</a>
<ul>
<li class="chapter" data-level="16.4.1" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#load-required-libraries"><i class="fa fa-check"></i><b>16.4.1</b> Load Required Libraries</a></li>
<li class="chapter" data-level="16.4.2" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#specify-path-for-results"><i class="fa fa-check"></i><b>16.4.2</b> Specify Path for Results</a></li>
<li class="chapter" data-level="16.4.3" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#giotto-to-anndata"><i class="fa fa-check"></i><b>16.4.3</b> Giotto To AnnData</a></li>
<li class="chapter" data-level="16.4.4" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#convert-anndata-to-giotto"><i class="fa fa-check"></i><b>16.4.4</b> Convert AnnData to Giotto</a></li>
</ul></li>
<li class="chapter" data-level="16.5" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#create-mini-vizgen-object"><i class="fa fa-check"></i><b>16.5</b> Create mini Vizgen object</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html"><i class="fa fa-check"></i><b>17</b> Interoperability with isolated tools</a>
<ul>
<li class="chapter" data-level="17.1" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#spatial-niche-trajectory-analysis"><i class="fa fa-check"></i><b>17.1</b> Spatial niche trajectory analysis</a>
<ul>
<li class="chapter" data-level="17.1.1" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#dataset-download"><i class="fa fa-check"></i><b>17.1.1</b> Dataset download</a></li>
<li class="chapter" data-level="17.1.2" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#create-the-giotto-object-2"><i class="fa fa-check"></i><b>17.1.2</b> Create the Giotto object</a></li>
<li class="chapter" data-level="17.1.3" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#visualization"><i class="fa fa-check"></i><b>17.1.3</b> visualization</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="18" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html"><i class="fa fa-check"></i><b>18</b> Interactivity with the R/Spatial ecosystem</a>
<ul>
<li class="chapter" data-level="18.1" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#kriging"><i class="fa fa-check"></i><b>18.1</b> Kriging</a>
<ul>
<li class="chapter" data-level="18.1.1" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#visium-technology-1"><i class="fa fa-check"></i><b>18.1.1</b> Visium technology</a></li>
<li class="chapter" data-level="18.1.2" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#dataset-1"><i class="fa fa-check"></i><b>18.1.2</b> Dataset</a></li>
<li class="chapter" data-level="18.1.3" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#generating-a-geojson-file-of-nuclei-location"><i class="fa fa-check"></i><b>18.1.3</b> Generating a geojson file of nuclei location</a></li>
</ul></li>
<li class="chapter" data-level="18.2" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#downloading-the-dataset"><i class="fa fa-check"></i><b>18.2</b> Downloading the dataset</a></li>
<li class="chapter" data-level="18.3" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#importing-visium-data"><i class="fa fa-check"></i><b>18.3</b> Importing visium data</a>
<ul>
<li class="chapter" data-level="18.3.1" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#identifying-spatially-organized-features"><i class="fa fa-check"></i><b>18.3.1</b> Identifying spatially organized features</a></li>
</ul></li>
<li class="chapter" data-level="18.4" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#adding-cell-polygons-to-giotto-object"><i class="fa fa-check"></i><b>18.4</b> Adding cell polygons to Giotto object</a>
<ul>
<li class="chapter" data-level="18.4.1" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#read-in-the-poly-information"><i class="fa fa-check"></i><b>18.4.1</b> Read in the poly information</a></li>
<li class="chapter" data-level="18.4.2" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#vizualizing-polygons"><i class="fa fa-check"></i><b>18.4.2</b> Vizualizing polygons</a></li>
<li class="chapter" data-level="18.4.3" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#showing-giotto-object-prior-to-polygon-addition"><i class="fa fa-check"></i><b>18.4.3</b> Showing Giotto object prior to polygon addition</a></li>
<li class="chapter" data-level="18.4.4" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#adding-polygons-to-giotto-object"><i class="fa fa-check"></i><b>18.4.4</b> Adding polygons to giotto object</a></li>
<li class="chapter" data-level="18.4.5" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#check-polygon-information"><i class="fa fa-check"></i><b>18.4.5</b> Check polygon information</a></li>
</ul></li>
<li class="chapter" data-level="18.5" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#performing-kriging"><i class="fa fa-check"></i><b>18.5</b> Performing kriging</a>
<ul>
<li class="chapter" data-level="18.5.1" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#interpolating-features"><i class="fa fa-check"></i><b>18.5.1</b> Interpolating features</a></li>
<li class="chapter" data-level="18.5.2" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#expression-overlap"><i class="fa fa-check"></i><b>18.5.2</b> Expression overlap</a></li>
</ul></li>
<li class="chapter" data-level="18.6" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#reading-in-larger-dataset"><i class="fa fa-check"></i><b>18.6</b> Reading in larger dataset</a></li>
<li class="chapter" data-level="18.7" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#analyzing-interpolated-features"><i class="fa fa-check"></i><b>18.7</b> Analyzing interpolated features</a>
<ul>
<li class="chapter" data-level="18.7.1" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#filter-and-normalization"><i class="fa fa-check"></i><b>18.7.1</b> Filter and normalization</a></li>
<li class="chapter" data-level="18.7.2" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#visualizing-gene-expression-from-interpolated-expression"><i class="fa fa-check"></i><b>18.7.2</b> Visualizing gene expression from interpolated expression</a></li>
<li class="chapter" data-level="18.7.3" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#run-pca"><i class="fa fa-check"></i><b>18.7.3</b> Run PCA</a></li>
<li class="chapter" data-level="18.7.4" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#clustering-1"><i class="fa fa-check"></i><b>18.7.4</b> Clustering</a></li>
<li class="chapter" data-level="18.7.5" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#visualizing-clustering"><i class="fa fa-check"></i><b>18.7.5</b> Visualizing clustering</a></li>
<li class="chapter" data-level="18.7.6" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#cropping-objects"><i class="fa fa-check"></i><b>18.7.6</b> Cropping objects</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="19" data-path="contributing-to-giotto.html"><a href="contributing-to-giotto.html"><i class="fa fa-check"></i><b>19</b> Contributing to Giotto</a>
<ul>
<li class="chapter" data-level="19.1" data-path="contributing-to-giotto.html"><a href="contributing-to-giotto.html#contribution-guideline"><i class="fa fa-check"></i><b>19.1</b> Contribution guideline</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Workshop: Spatial multi-omics data analysis with Giotto Suite</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="multi-omics-integration" class="section level1 hasAnchor" number="15">
<h1><span class="header-section-number">15</span> Multi-omics integration<a href="multi-omics-integration.html#multi-omics-integration" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Joselyn Cristina ChÃ¡vez Fuentes</p>
<p>August 7th 2024</p>
<div id="introduction-to-the-spatial-dataset-1" class="section level2 hasAnchor" number="15.1">
<h2><span class="header-section-number">15.1</span> Introduction to the spatial dataset<a href="multi-omics-integration.html#introduction-to-the-spatial-dataset-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The <a href="https://www.10xgenomics.com/resources/datasets/gene-and-protein-expression-library-of-human-glioblastoma-cytassist-ffpe-2-standard/">Human glioblastoma (FFPE) dataset</a> was obtained from 10X Genomics. The tissue was sectioned as described in visium_glioblastoma CytAssist Spatial Gene Expression for FFPE â Tissue Preparation Guide Demonstrated Protocol (CG000518). 5 Âµm tissue sections were placed on Superfrost glass slides, then IF stained following deparaffinization, then hard coverslipped. Sections were imaged, decoverslipped, followed by Demonstrated Protocol (CG000494).</p>
<p>More information about this dataset can be found <a href="https://www.10xgenomics.com/resources/datasets/gene-and-protein-expression-library-of-human-glioblastoma-cytassist-ffpe-2-standard">here</a>.</p>
</div>
<div id="download-dataset-1" class="section level2 hasAnchor" number="15.2">
<h2><span class="header-section-number">15.2</span> Download dataset<a href="multi-omics-integration.html#download-dataset-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>You need to download the expression matrix and spatial information by running these commands:</p>
<div class="sourceCode" id="cb140"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb140-1"><a href="multi-omics-integration.html#cb140-1" tabindex="-1"></a><span class="fu">dir.create</span>(<span class="st">&quot;data&quot;</span>)</span>
<span id="cb140-2"><a href="multi-omics-integration.html#cb140-2" tabindex="-1"></a></span>
<span id="cb140-3"><a href="multi-omics-integration.html#cb140-3" tabindex="-1"></a><span class="fu">download.file</span>(<span class="at">url =</span> <span class="st">&quot;https://cf.10xgenomics.com/samples/spatial-exp/2.1.0/CytAssist_FFPE_Protein_Expression_Human_Glioblastoma/CytAssist_FFPE_Protein_Expression_Human_Glioblastoma_raw_feature_bc_matrix.tar.gz&quot;</span>,</span>
<span id="cb140-4"><a href="multi-omics-integration.html#cb140-4" tabindex="-1"></a>              <span class="at">destfile =</span> <span class="st">&quot;data/CytAssist_FFPE_Protein_Expression_Human_Glioblastoma_raw_feature_bc_matrix.tar.gz&quot;</span>)</span>
<span id="cb140-5"><a href="multi-omics-integration.html#cb140-5" tabindex="-1"></a></span>
<span id="cb140-6"><a href="multi-omics-integration.html#cb140-6" tabindex="-1"></a><span class="fu">download.file</span>(<span class="at">url =</span> <span class="st">&quot;https://cf.10xgenomics.com/samples/spatial-exp/2.1.0/CytAssist_FFPE_Protein_Expression_Human_Glioblastoma/CytAssist_FFPE_Protein_Expression_Human_Glioblastoma_spatial.tar.gz&quot;</span>,</span>
<span id="cb140-7"><a href="multi-omics-integration.html#cb140-7" tabindex="-1"></a>              <span class="at">destfile =</span> <span class="st">&quot;data/CytAssist_FFPE_Protein_Expression_Human_Glioblastoma_spatial.tar.gz&quot;</span>)</span></code></pre></div>
<p>After downloading, unzip the gz files. You should get the âraw_feature_bc_matrixâ and âspatialâ folders inside âdata/â.</p>
</div>
<div id="create-the-giotto-object-1" class="section level2 hasAnchor" number="15.3">
<h2><span class="header-section-number">15.3</span> Create the Giotto object<a href="multi-omics-integration.html#create-the-giotto-object-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The minimum requirements are:</p>
<ul>
<li>matrix with expression information (or the path to)</li>
<li>x,y(,z) coordinates for cells or spots (or the path to)</li>
</ul>
<p>createGiottoVisiumObject() will automatically detect both RNA and Protein modalities in the expression matrix and will create a multi-omics Giotto object.</p>
<div class="sourceCode" id="cb141"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb141-1"><a href="multi-omics-integration.html#cb141-1" tabindex="-1"></a><span class="fu">library</span>(Giotto)</span>
<span id="cb141-2"><a href="multi-omics-integration.html#cb141-2" tabindex="-1"></a></span>
<span id="cb141-3"><a href="multi-omics-integration.html#cb141-3" tabindex="-1"></a><span class="do">## Set instructions</span></span>
<span id="cb141-4"><a href="multi-omics-integration.html#cb141-4" tabindex="-1"></a>results_folder <span class="ot">&lt;-</span> <span class="st">&quot;results/&quot;</span></span>
<span id="cb141-5"><a href="multi-omics-integration.html#cb141-5" tabindex="-1"></a></span>
<span id="cb141-6"><a href="multi-omics-integration.html#cb141-6" tabindex="-1"></a>python_path <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb141-7"><a href="multi-omics-integration.html#cb141-7" tabindex="-1"></a></span>
<span id="cb141-8"><a href="multi-omics-integration.html#cb141-8" tabindex="-1"></a>instructions <span class="ot">&lt;-</span> <span class="fu">createGiottoInstructions</span>(</span>
<span id="cb141-9"><a href="multi-omics-integration.html#cb141-9" tabindex="-1"></a>    <span class="at">save_dir =</span> results_folder,</span>
<span id="cb141-10"><a href="multi-omics-integration.html#cb141-10" tabindex="-1"></a>    <span class="at">save_plot =</span> <span class="cn">TRUE</span>,</span>
<span id="cb141-11"><a href="multi-omics-integration.html#cb141-11" tabindex="-1"></a>    <span class="at">show_plot =</span> <span class="cn">FALSE</span>,</span>
<span id="cb141-12"><a href="multi-omics-integration.html#cb141-12" tabindex="-1"></a>    <span class="at">return_plot =</span> <span class="cn">FALSE</span>,</span>
<span id="cb141-13"><a href="multi-omics-integration.html#cb141-13" tabindex="-1"></a>    <span class="at">python_path =</span> python_path</span>
<span id="cb141-14"><a href="multi-omics-integration.html#cb141-14" tabindex="-1"></a>)</span>
<span id="cb141-15"><a href="multi-omics-integration.html#cb141-15" tabindex="-1"></a></span>
<span id="cb141-16"><a href="multi-omics-integration.html#cb141-16" tabindex="-1"></a><span class="co"># Provide the path to the visium_glioblastoma folder</span></span>
<span id="cb141-17"><a href="multi-omics-integration.html#cb141-17" tabindex="-1"></a>data_path <span class="ot">&lt;-</span> <span class="st">&quot;data&quot;</span></span>
<span id="cb141-18"><a href="multi-omics-integration.html#cb141-18" tabindex="-1"></a></span>
<span id="cb141-19"><a href="multi-omics-integration.html#cb141-19" tabindex="-1"></a><span class="co"># Create object directly from the visium_glioblastoma folder</span></span>
<span id="cb141-20"><a href="multi-omics-integration.html#cb141-20" tabindex="-1"></a>visium_glioblastoma <span class="ot">&lt;-</span> <span class="fu">createGiottoVisiumObject</span>(</span>
<span id="cb141-21"><a href="multi-omics-integration.html#cb141-21" tabindex="-1"></a>    <span class="at">visium_dir =</span> data_path,</span>
<span id="cb141-22"><a href="multi-omics-integration.html#cb141-22" tabindex="-1"></a>    <span class="at">expr_data =</span> <span class="st">&quot;raw&quot;</span>,</span>
<span id="cb141-23"><a href="multi-omics-integration.html#cb141-23" tabindex="-1"></a>    <span class="at">png_name =</span> <span class="st">&quot;tissue_lowres_image.png&quot;</span>,</span>
<span id="cb141-24"><a href="multi-omics-integration.html#cb141-24" tabindex="-1"></a>    <span class="at">gene_column_index =</span> <span class="dv">2</span>,</span>
<span id="cb141-25"><a href="multi-omics-integration.html#cb141-25" tabindex="-1"></a>    <span class="at">instructions =</span> instructions</span>
<span id="cb141-26"><a href="multi-omics-integration.html#cb141-26" tabindex="-1"></a>)</span></code></pre></div>
<ul>
<li>Print the information of the object, note that both rna and protein are listed in the expression slot</li>
</ul>
<div class="sourceCode" id="cb142"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb142-1"><a href="multi-omics-integration.html#cb142-1" tabindex="-1"></a>visium_glioblastoma</span></code></pre></div>
</div>
<div id="subset-on-spots-that-were-covered-by-tissue-1" class="section level2 hasAnchor" number="15.4">
<h2><span class="header-section-number">15.4</span> Subset on spots that were covered by tissue<a href="multi-omics-integration.html#subset-on-spots-that-were-covered-by-tissue-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb143"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb143-1"><a href="multi-omics-integration.html#cb143-1" tabindex="-1"></a><span class="fu">spatPlot2D</span>(</span>
<span id="cb143-2"><a href="multi-omics-integration.html#cb143-2" tabindex="-1"></a>    <span class="at">gobject =</span> visium_glioblastoma,</span>
<span id="cb143-3"><a href="multi-omics-integration.html#cb143-3" tabindex="-1"></a>    <span class="at">cell_color =</span> <span class="st">&quot;in_tissue&quot;</span>,</span>
<span id="cb143-4"><a href="multi-omics-integration.html#cb143-4" tabindex="-1"></a>    <span class="at">point_size =</span> <span class="dv">2</span>,</span>
<span id="cb143-5"><a href="multi-omics-integration.html#cb143-5" tabindex="-1"></a>    <span class="at">cell_color_code =</span> <span class="fu">c</span>(<span class="st">&quot;0&quot;</span> <span class="ot">=</span> <span class="st">&quot;lightgrey&quot;</span>, <span class="st">&quot;1&quot;</span> <span class="ot">=</span> <span class="st">&quot;blue&quot;</span>),</span>
<span id="cb143-6"><a href="multi-omics-integration.html#cb143-6" tabindex="-1"></a>    <span class="at">show_image =</span> <span class="cn">TRUE</span>,</span>
<span id="cb143-7"><a href="multi-omics-integration.html#cb143-7" tabindex="-1"></a>    <span class="at">image_name =</span> <span class="st">&quot;image&quot;</span></span>
<span id="cb143-8"><a href="multi-omics-integration.html#cb143-8" tabindex="-1"></a>)</span></code></pre></div>
<p><img src="img/03_session3/0-spatPlot2D.png" width="80%" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb144"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb144-1"><a href="multi-omics-integration.html#cb144-1" tabindex="-1"></a>metadata <span class="ot">&lt;-</span> <span class="fu">getCellMetadata</span>(<span class="at">gobject =</span> visium_glioblastoma,</span>
<span id="cb144-2"><a href="multi-omics-integration.html#cb144-2" tabindex="-1"></a>                            <span class="at">output =</span> <span class="st">&quot;data.table&quot;</span>)</span>
<span id="cb144-3"><a href="multi-omics-integration.html#cb144-3" tabindex="-1"></a></span>
<span id="cb144-4"><a href="multi-omics-integration.html#cb144-4" tabindex="-1"></a>in_tissue_barcodes <span class="ot">&lt;-</span> metadata[in_tissue <span class="sc">==</span> <span class="dv">1</span>]<span class="sc">$</span>cell_ID</span>
<span id="cb144-5"><a href="multi-omics-integration.html#cb144-5" tabindex="-1"></a></span>
<span id="cb144-6"><a href="multi-omics-integration.html#cb144-6" tabindex="-1"></a>visium_glioblastoma <span class="ot">&lt;-</span> <span class="fu">subsetGiotto</span>(visium_glioblastoma, </span>
<span id="cb144-7"><a href="multi-omics-integration.html#cb144-7" tabindex="-1"></a>                                    <span class="at">cell_ids =</span> in_tissue_barcodes)</span></code></pre></div>
</div>
<div id="rna-processing" class="section level2 hasAnchor" number="15.5">
<h2><span class="header-section-number">15.5</span> RNA processing<a href="multi-omics-integration.html#rna-processing" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ul>
<li>Filtering, normalization, and statistics</li>
</ul>
<div class="sourceCode" id="cb145"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb145-1"><a href="multi-omics-integration.html#cb145-1" tabindex="-1"></a>visium_glioblastoma <span class="ot">&lt;-</span> <span class="fu">filterGiotto</span>(</span>
<span id="cb145-2"><a href="multi-omics-integration.html#cb145-2" tabindex="-1"></a>    <span class="at">gobject =</span> visium_glioblastoma,</span>
<span id="cb145-3"><a href="multi-omics-integration.html#cb145-3" tabindex="-1"></a>    <span class="at">expression_threshold =</span> <span class="dv">1</span>,</span>
<span id="cb145-4"><a href="multi-omics-integration.html#cb145-4" tabindex="-1"></a>    <span class="at">feat_det_in_min_cells =</span> <span class="dv">50</span>,</span>
<span id="cb145-5"><a href="multi-omics-integration.html#cb145-5" tabindex="-1"></a>    <span class="at">min_det_feats_per_cell =</span> <span class="dv">1000</span>,</span>
<span id="cb145-6"><a href="multi-omics-integration.html#cb145-6" tabindex="-1"></a>    <span class="at">expression_values =</span> <span class="st">&quot;raw&quot;</span>,</span>
<span id="cb145-7"><a href="multi-omics-integration.html#cb145-7" tabindex="-1"></a>    <span class="at">verbose =</span> <span class="cn">TRUE</span>)</span>
<span id="cb145-8"><a href="multi-omics-integration.html#cb145-8" tabindex="-1"></a></span>
<span id="cb145-9"><a href="multi-omics-integration.html#cb145-9" tabindex="-1"></a>visium_glioblastoma <span class="ot">&lt;-</span> <span class="fu">normalizeGiotto</span>(<span class="at">gobject =</span> visium_glioblastoma,</span>
<span id="cb145-10"><a href="multi-omics-integration.html#cb145-10" tabindex="-1"></a>                                       <span class="at">scalefactor =</span> <span class="dv">6000</span>,</span>
<span id="cb145-11"><a href="multi-omics-integration.html#cb145-11" tabindex="-1"></a>                                       <span class="at">verbose =</span> <span class="cn">TRUE</span>)</span>
<span id="cb145-12"><a href="multi-omics-integration.html#cb145-12" tabindex="-1"></a></span>
<span id="cb145-13"><a href="multi-omics-integration.html#cb145-13" tabindex="-1"></a>visium_glioblastoma <span class="ot">&lt;-</span> <span class="fu">addStatistics</span>(<span class="at">gobject =</span> visium_glioblastoma)</span></code></pre></div>
<ul>
<li>Dimension reduction</li>
</ul>
<div class="sourceCode" id="cb146"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb146-1"><a href="multi-omics-integration.html#cb146-1" tabindex="-1"></a>visium_glioblastoma <span class="ot">&lt;-</span> <span class="fu">calculateHVF</span>(<span class="at">gobject =</span> visium_glioblastoma)</span>
<span id="cb146-2"><a href="multi-omics-integration.html#cb146-2" tabindex="-1"></a></span>
<span id="cb146-3"><a href="multi-omics-integration.html#cb146-3" tabindex="-1"></a>visium_glioblastoma <span class="ot">&lt;-</span> <span class="fu">runPCA</span>(<span class="at">gobject =</span> visium_glioblastoma)</span></code></pre></div>
<ul>
<li>Clustering</li>
</ul>
<div class="sourceCode" id="cb147"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb147-1"><a href="multi-omics-integration.html#cb147-1" tabindex="-1"></a>visium_glioblastoma <span class="ot">&lt;-</span> <span class="fu">runUMAP</span>(visium_glioblastoma,</span>
<span id="cb147-2"><a href="multi-omics-integration.html#cb147-2" tabindex="-1"></a>                               <span class="at">dimensions_to_use =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>)</span>
<span id="cb147-3"><a href="multi-omics-integration.html#cb147-3" tabindex="-1"></a></span>
<span id="cb147-4"><a href="multi-omics-integration.html#cb147-4" tabindex="-1"></a>visium_glioblastoma <span class="ot">&lt;-</span> <span class="fu">createNearestNetwork</span>(<span class="at">gobject =</span> visium_glioblastoma,</span>
<span id="cb147-5"><a href="multi-omics-integration.html#cb147-5" tabindex="-1"></a>                                            <span class="at">dimensions_to_use =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>,</span>
<span id="cb147-6"><a href="multi-omics-integration.html#cb147-6" tabindex="-1"></a>                                            <span class="at">k =</span> <span class="dv">30</span>)</span>
<span id="cb147-7"><a href="multi-omics-integration.html#cb147-7" tabindex="-1"></a></span>
<span id="cb147-8"><a href="multi-omics-integration.html#cb147-8" tabindex="-1"></a>visium_glioblastoma <span class="ot">&lt;-</span> <span class="fu">doLeidenCluster</span>(<span class="at">gobject =</span> visium_glioblastoma,</span>
<span id="cb147-9"><a href="multi-omics-integration.html#cb147-9" tabindex="-1"></a>                                       <span class="at">resolution =</span> <span class="dv">1</span>,</span>
<span id="cb147-10"><a href="multi-omics-integration.html#cb147-10" tabindex="-1"></a>                                       <span class="at">n_iterations =</span> <span class="dv">1000</span>)</span></code></pre></div>
<ul>
<li>Visualization</li>
</ul>
<div class="sourceCode" id="cb148"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb148-1"><a href="multi-omics-integration.html#cb148-1" tabindex="-1"></a><span class="fu">plotUMAP</span>(<span class="at">gobject =</span> visium_glioblastoma,</span>
<span id="cb148-2"><a href="multi-omics-integration.html#cb148-2" tabindex="-1"></a>         <span class="at">cell_color =</span> <span class="st">&quot;leiden_clus&quot;</span>,</span>
<span id="cb148-3"><a href="multi-omics-integration.html#cb148-3" tabindex="-1"></a>         <span class="at">show_NN_network =</span> <span class="cn">TRUE</span>,</span>
<span id="cb148-4"><a href="multi-omics-integration.html#cb148-4" tabindex="-1"></a>         <span class="at">point_size =</span> <span class="dv">2</span>)</span></code></pre></div>
<p><img src="img/03_session3/1-UMAP.png" width="80%" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb149"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb149-1"><a href="multi-omics-integration.html#cb149-1" tabindex="-1"></a><span class="fu">spatPlot2D</span>(<span class="at">gobject =</span> visium_glioblastoma,</span>
<span id="cb149-2"><a href="multi-omics-integration.html#cb149-2" tabindex="-1"></a>           <span class="at">show_image =</span> <span class="cn">FALSE</span>,</span>
<span id="cb149-3"><a href="multi-omics-integration.html#cb149-3" tabindex="-1"></a>           <span class="at">cell_color =</span> <span class="st">&quot;leiden_clus&quot;</span>,</span>
<span id="cb149-4"><a href="multi-omics-integration.html#cb149-4" tabindex="-1"></a>           <span class="at">point_size =</span> <span class="dv">2</span>)</span></code></pre></div>
<p><img src="img/03_session3/2-spatPlot2D.png" width="80%" style="display: block; margin: auto;" /></p>
</div>
<div id="protein-processing" class="section level2 hasAnchor" number="15.6">
<h2><span class="header-section-number">15.6</span> Protein processing<a href="multi-omics-integration.html#protein-processing" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ul>
<li>Filtering, normalization, and statistics</li>
</ul>
<div class="sourceCode" id="cb150"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb150-1"><a href="multi-omics-integration.html#cb150-1" tabindex="-1"></a>visium_glioblastoma <span class="ot">&lt;-</span> <span class="fu">filterGiotto</span>(<span class="at">gobject =</span> visium_glioblastoma, </span>
<span id="cb150-2"><a href="multi-omics-integration.html#cb150-2" tabindex="-1"></a>                                    <span class="at">spat_unit =</span> <span class="st">&quot;cell&quot;</span>,</span>
<span id="cb150-3"><a href="multi-omics-integration.html#cb150-3" tabindex="-1"></a>                                    <span class="at">feat_type =</span> <span class="st">&quot;protein&quot;</span>,</span>
<span id="cb150-4"><a href="multi-omics-integration.html#cb150-4" tabindex="-1"></a>                                    <span class="at">expression_threshold =</span> <span class="dv">1</span>,</span>
<span id="cb150-5"><a href="multi-omics-integration.html#cb150-5" tabindex="-1"></a>                                    <span class="at">feat_det_in_min_cells =</span> <span class="dv">50</span>, </span>
<span id="cb150-6"><a href="multi-omics-integration.html#cb150-6" tabindex="-1"></a>                                    <span class="at">min_det_feats_per_cell =</span> <span class="dv">1</span>,</span>
<span id="cb150-7"><a href="multi-omics-integration.html#cb150-7" tabindex="-1"></a>                                    <span class="at">expression_values =</span> <span class="st">&quot;raw&quot;</span>, </span>
<span id="cb150-8"><a href="multi-omics-integration.html#cb150-8" tabindex="-1"></a>                                    <span class="at">verbose =</span> <span class="cn">TRUE</span>)</span>
<span id="cb150-9"><a href="multi-omics-integration.html#cb150-9" tabindex="-1"></a></span>
<span id="cb150-10"><a href="multi-omics-integration.html#cb150-10" tabindex="-1"></a>visium_glioblastoma <span class="ot">&lt;-</span> <span class="fu">normalizeGiotto</span>(<span class="at">gobject =</span> visium_glioblastoma,</span>
<span id="cb150-11"><a href="multi-omics-integration.html#cb150-11" tabindex="-1"></a>                                       <span class="at">spat_unit =</span> <span class="st">&quot;cell&quot;</span>, </span>
<span id="cb150-12"><a href="multi-omics-integration.html#cb150-12" tabindex="-1"></a>                                       <span class="at">feat_type =</span> <span class="st">&quot;protein&quot;</span>, </span>
<span id="cb150-13"><a href="multi-omics-integration.html#cb150-13" tabindex="-1"></a>                                       <span class="at">scalefactor =</span> <span class="dv">6000</span>, </span>
<span id="cb150-14"><a href="multi-omics-integration.html#cb150-14" tabindex="-1"></a>                                       <span class="at">verbose =</span> <span class="cn">TRUE</span>)</span>
<span id="cb150-15"><a href="multi-omics-integration.html#cb150-15" tabindex="-1"></a></span>
<span id="cb150-16"><a href="multi-omics-integration.html#cb150-16" tabindex="-1"></a>visium_glioblastoma <span class="ot">&lt;-</span> <span class="fu">addStatistics</span>(<span class="at">gobject =</span> visium_glioblastoma,</span>
<span id="cb150-17"><a href="multi-omics-integration.html#cb150-17" tabindex="-1"></a>                                     <span class="at">spat_unit =</span> <span class="st">&quot;cell&quot;</span>, </span>
<span id="cb150-18"><a href="multi-omics-integration.html#cb150-18" tabindex="-1"></a>                                     <span class="at">feat_type =</span> <span class="st">&quot;protein&quot;</span>)</span></code></pre></div>
<ul>
<li>Dimension reduction</li>
</ul>
<div class="sourceCode" id="cb151"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb151-1"><a href="multi-omics-integration.html#cb151-1" tabindex="-1"></a>visium_glioblastoma <span class="ot">&lt;-</span> <span class="fu">runPCA</span>(<span class="at">gobject =</span> visium_glioblastoma,</span>
<span id="cb151-2"><a href="multi-omics-integration.html#cb151-2" tabindex="-1"></a>                              <span class="at">spat_unit =</span> <span class="st">&quot;cell&quot;</span>,</span>
<span id="cb151-3"><a href="multi-omics-integration.html#cb151-3" tabindex="-1"></a>                              <span class="at">feat_type =</span> <span class="st">&quot;protein&quot;</span>)</span></code></pre></div>
<ul>
<li>Clustering</li>
</ul>
<div class="sourceCode" id="cb152"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb152-1"><a href="multi-omics-integration.html#cb152-1" tabindex="-1"></a>visium_glioblastoma <span class="ot">&lt;-</span> <span class="fu">runUMAP</span>(visium_glioblastoma,</span>
<span id="cb152-2"><a href="multi-omics-integration.html#cb152-2" tabindex="-1"></a>                               <span class="at">spat_unit =</span> <span class="st">&quot;cell&quot;</span>, </span>
<span id="cb152-3"><a href="multi-omics-integration.html#cb152-3" tabindex="-1"></a>                               <span class="at">feat_type =</span> <span class="st">&quot;protein&quot;</span>, </span>
<span id="cb152-4"><a href="multi-omics-integration.html#cb152-4" tabindex="-1"></a>                               <span class="at">dimensions_to_use =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>)</span>
<span id="cb152-5"><a href="multi-omics-integration.html#cb152-5" tabindex="-1"></a></span>
<span id="cb152-6"><a href="multi-omics-integration.html#cb152-6" tabindex="-1"></a>visium_glioblastoma <span class="ot">&lt;-</span> <span class="fu">createNearestNetwork</span>(<span class="at">gobject =</span> visium_glioblastoma,</span>
<span id="cb152-7"><a href="multi-omics-integration.html#cb152-7" tabindex="-1"></a>                                            <span class="at">spat_unit =</span> <span class="st">&quot;cell&quot;</span>,</span>
<span id="cb152-8"><a href="multi-omics-integration.html#cb152-8" tabindex="-1"></a>                                            <span class="at">feat_type =</span> <span class="st">&quot;protein&quot;</span>,</span>
<span id="cb152-9"><a href="multi-omics-integration.html#cb152-9" tabindex="-1"></a>                                            <span class="at">dimensions_to_use =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>,</span>
<span id="cb152-10"><a href="multi-omics-integration.html#cb152-10" tabindex="-1"></a>                                            <span class="at">k =</span> <span class="dv">30</span>)</span>
<span id="cb152-11"><a href="multi-omics-integration.html#cb152-11" tabindex="-1"></a></span>
<span id="cb152-12"><a href="multi-omics-integration.html#cb152-12" tabindex="-1"></a>visium_glioblastoma <span class="ot">&lt;-</span> <span class="fu">doLeidenCluster</span>(<span class="at">gobject =</span> visium_glioblastoma,</span>
<span id="cb152-13"><a href="multi-omics-integration.html#cb152-13" tabindex="-1"></a>                                       <span class="at">spat_unit =</span> <span class="st">&quot;cell&quot;</span>,</span>
<span id="cb152-14"><a href="multi-omics-integration.html#cb152-14" tabindex="-1"></a>                                       <span class="at">feat_type =</span> <span class="st">&quot;protein&quot;</span>,</span>
<span id="cb152-15"><a href="multi-omics-integration.html#cb152-15" tabindex="-1"></a>                                       <span class="at">resolution =</span> <span class="dv">1</span>,</span>
<span id="cb152-16"><a href="multi-omics-integration.html#cb152-16" tabindex="-1"></a>                                       <span class="at">n_iterations =</span> <span class="dv">1000</span>)</span></code></pre></div>
<ul>
<li>Visualization</li>
</ul>
<div class="sourceCode" id="cb153"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb153-1"><a href="multi-omics-integration.html#cb153-1" tabindex="-1"></a><span class="fu">plotUMAP</span>(<span class="at">gobject =</span> visium_glioblastoma,</span>
<span id="cb153-2"><a href="multi-omics-integration.html#cb153-2" tabindex="-1"></a>         <span class="at">spat_unit =</span> <span class="st">&quot;cell&quot;</span>, </span>
<span id="cb153-3"><a href="multi-omics-integration.html#cb153-3" tabindex="-1"></a>         <span class="at">feat_type =</span> <span class="st">&quot;protein&quot;</span>, </span>
<span id="cb153-4"><a href="multi-omics-integration.html#cb153-4" tabindex="-1"></a>         <span class="at">cell_color =</span> <span class="st">&quot;leiden_clus&quot;</span>, </span>
<span id="cb153-5"><a href="multi-omics-integration.html#cb153-5" tabindex="-1"></a>         <span class="at">show_NN_network =</span> <span class="cn">TRUE</span>, </span>
<span id="cb153-6"><a href="multi-omics-integration.html#cb153-6" tabindex="-1"></a>         <span class="at">point_size =</span> <span class="dv">2</span>)</span></code></pre></div>
<p><img src="img/03_session3/3-UMAP.png" width="80%" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb154"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb154-1"><a href="multi-omics-integration.html#cb154-1" tabindex="-1"></a><span class="fu">spatPlot2D</span>(<span class="at">gobject =</span> visium_glioblastoma,</span>
<span id="cb154-2"><a href="multi-omics-integration.html#cb154-2" tabindex="-1"></a>           <span class="at">spat_unit =</span> <span class="st">&quot;cell&quot;</span>,</span>
<span id="cb154-3"><a href="multi-omics-integration.html#cb154-3" tabindex="-1"></a>           <span class="at">feat_type =</span> <span class="st">&quot;protein&quot;</span>,</span>
<span id="cb154-4"><a href="multi-omics-integration.html#cb154-4" tabindex="-1"></a>           <span class="at">show_image =</span> <span class="cn">FALSE</span>,</span>
<span id="cb154-5"><a href="multi-omics-integration.html#cb154-5" tabindex="-1"></a>           <span class="at">cell_color =</span> <span class="st">&quot;leiden_clus&quot;</span>,</span>
<span id="cb154-6"><a href="multi-omics-integration.html#cb154-6" tabindex="-1"></a>           <span class="at">point_size =</span> <span class="dv">2</span>)</span></code></pre></div>
<p><img src="img/03_session3/4-spatPlot2D.png" width="80%" style="display: block; margin: auto;" /></p>
</div>
<div id="multi-omics-integration-1" class="section level2 hasAnchor" number="15.7">
<h2><span class="header-section-number">15.7</span> Multi-omics integration<a href="multi-omics-integration.html#multi-omics-integration-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ul>
<li>Calculate kNN</li>
</ul>
<div class="sourceCode" id="cb155"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb155-1"><a href="multi-omics-integration.html#cb155-1" tabindex="-1"></a><span class="do">## RNA modality</span></span>
<span id="cb155-2"><a href="multi-omics-integration.html#cb155-2" tabindex="-1"></a>visium_glioblastoma <span class="ot">&lt;-</span> <span class="fu">createNearestNetwork</span>(<span class="at">gobject =</span> visium_glioblastoma,</span>
<span id="cb155-3"><a href="multi-omics-integration.html#cb155-3" tabindex="-1"></a>                                            <span class="at">type =</span> <span class="st">&quot;kNN&quot;</span>,</span>
<span id="cb155-4"><a href="multi-omics-integration.html#cb155-4" tabindex="-1"></a>                                            <span class="at">dimensions_to_use =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>,</span>
<span id="cb155-5"><a href="multi-omics-integration.html#cb155-5" tabindex="-1"></a>                                            <span class="at">k =</span> <span class="dv">20</span>)</span>
<span id="cb155-6"><a href="multi-omics-integration.html#cb155-6" tabindex="-1"></a></span>
<span id="cb155-7"><a href="multi-omics-integration.html#cb155-7" tabindex="-1"></a><span class="do">## Protein modality</span></span>
<span id="cb155-8"><a href="multi-omics-integration.html#cb155-8" tabindex="-1"></a>visium_glioblastoma <span class="ot">&lt;-</span> <span class="fu">createNearestNetwork</span>(<span class="at">gobject =</span> visium_glioblastoma,</span>
<span id="cb155-9"><a href="multi-omics-integration.html#cb155-9" tabindex="-1"></a>                                            <span class="at">spat_unit =</span> <span class="st">&quot;cell&quot;</span>,</span>
<span id="cb155-10"><a href="multi-omics-integration.html#cb155-10" tabindex="-1"></a>                                            <span class="at">feat_type =</span> <span class="st">&quot;protein&quot;</span>,</span>
<span id="cb155-11"><a href="multi-omics-integration.html#cb155-11" tabindex="-1"></a>                                            <span class="at">type =</span> <span class="st">&quot;kNN&quot;</span>,</span>
<span id="cb155-12"><a href="multi-omics-integration.html#cb155-12" tabindex="-1"></a>                                            <span class="at">dimensions_to_use =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>,</span>
<span id="cb155-13"><a href="multi-omics-integration.html#cb155-13" tabindex="-1"></a>                                            <span class="at">k =</span> <span class="dv">20</span>)</span></code></pre></div>
<ul>
<li>Run WNN</li>
</ul>
<div class="sourceCode" id="cb156"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb156-1"><a href="multi-omics-integration.html#cb156-1" tabindex="-1"></a>visium_glioblastoma <span class="ot">&lt;-</span> <span class="fu">runWNN</span>(visium_glioblastoma,</span>
<span id="cb156-2"><a href="multi-omics-integration.html#cb156-2" tabindex="-1"></a>                              <span class="at">spat_unit =</span> <span class="st">&quot;cell&quot;</span>,</span>
<span id="cb156-3"><a href="multi-omics-integration.html#cb156-3" tabindex="-1"></a>                              <span class="at">modality_1 =</span> <span class="st">&quot;rna&quot;</span>,</span>
<span id="cb156-4"><a href="multi-omics-integration.html#cb156-4" tabindex="-1"></a>                              <span class="at">modality_2 =</span> <span class="st">&quot;protein&quot;</span>,</span>
<span id="cb156-5"><a href="multi-omics-integration.html#cb156-5" tabindex="-1"></a>                              <span class="at">pca_name_modality_1 =</span> <span class="st">&quot;pca&quot;</span>,</span>
<span id="cb156-6"><a href="multi-omics-integration.html#cb156-6" tabindex="-1"></a>                              <span class="at">pca_name_modality_2 =</span> <span class="st">&quot;protein.pca&quot;</span>,</span>
<span id="cb156-7"><a href="multi-omics-integration.html#cb156-7" tabindex="-1"></a>                              <span class="at">k =</span> <span class="dv">20</span>,</span>
<span id="cb156-8"><a href="multi-omics-integration.html#cb156-8" tabindex="-1"></a>                              <span class="at">integrated_feat_type =</span> <span class="cn">NULL</span>,</span>
<span id="cb156-9"><a href="multi-omics-integration.html#cb156-9" tabindex="-1"></a>                              <span class="at">matrix_result_name =</span> <span class="cn">NULL</span>,</span>
<span id="cb156-10"><a href="multi-omics-integration.html#cb156-10" tabindex="-1"></a>                              <span class="at">w_name_modality_1 =</span> <span class="cn">NULL</span>,</span>
<span id="cb156-11"><a href="multi-omics-integration.html#cb156-11" tabindex="-1"></a>                              <span class="at">w_name_modality_2 =</span> <span class="cn">NULL</span>,</span>
<span id="cb156-12"><a href="multi-omics-integration.html#cb156-12" tabindex="-1"></a>                              <span class="at">verbose =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<ul>
<li>Run Integrated umap</li>
</ul>
<div class="sourceCode" id="cb157"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb157-1"><a href="multi-omics-integration.html#cb157-1" tabindex="-1"></a>visium_glioblastoma <span class="ot">&lt;-</span> <span class="fu">runIntegratedUMAP</span>(visium_glioblastoma,</span>
<span id="cb157-2"><a href="multi-omics-integration.html#cb157-2" tabindex="-1"></a>                                         <span class="at">modality1 =</span> <span class="st">&quot;rna&quot;</span>,</span>
<span id="cb157-3"><a href="multi-omics-integration.html#cb157-3" tabindex="-1"></a>                                         <span class="at">modality2 =</span> <span class="st">&quot;protein&quot;</span>,</span>
<span id="cb157-4"><a href="multi-omics-integration.html#cb157-4" tabindex="-1"></a>                                         <span class="at">spread =</span> <span class="dv">5</span>,</span>
<span id="cb157-5"><a href="multi-omics-integration.html#cb157-5" tabindex="-1"></a>                                         <span class="at">min_dist =</span> <span class="fl">0.5</span>,</span>
<span id="cb157-6"><a href="multi-omics-integration.html#cb157-6" tabindex="-1"></a>                                         <span class="at">force =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<ul>
<li>Calculate integrated clusters</li>
</ul>
<div class="sourceCode" id="cb158"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb158-1"><a href="multi-omics-integration.html#cb158-1" tabindex="-1"></a>visium_glioblastoma <span class="ot">&lt;-</span> <span class="fu">doLeidenCluster</span>(<span class="at">gobject =</span> visium_glioblastoma,</span>
<span id="cb158-2"><a href="multi-omics-integration.html#cb158-2" tabindex="-1"></a>                                       <span class="at">spat_unit =</span> <span class="st">&quot;cell&quot;</span>,</span>
<span id="cb158-3"><a href="multi-omics-integration.html#cb158-3" tabindex="-1"></a>                                       <span class="at">feat_type =</span> <span class="st">&quot;rna&quot;</span>,</span>
<span id="cb158-4"><a href="multi-omics-integration.html#cb158-4" tabindex="-1"></a>                                       <span class="at">nn_network_to_use =</span> <span class="st">&quot;kNN&quot;</span>,</span>
<span id="cb158-5"><a href="multi-omics-integration.html#cb158-5" tabindex="-1"></a>                                       <span class="at">network_name =</span> <span class="st">&quot;integrated_kNN&quot;</span>,</span>
<span id="cb158-6"><a href="multi-omics-integration.html#cb158-6" tabindex="-1"></a>                                       <span class="at">name =</span> <span class="st">&quot;integrated_leiden_clus&quot;</span>,</span>
<span id="cb158-7"><a href="multi-omics-integration.html#cb158-7" tabindex="-1"></a>                                       <span class="at">resolution =</span> <span class="dv">1</span>)</span></code></pre></div>
<ul>
<li>Visualize the integrated umap</li>
</ul>
<div class="sourceCode" id="cb159"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb159-1"><a href="multi-omics-integration.html#cb159-1" tabindex="-1"></a><span class="fu">plotUMAP</span>(<span class="at">gobject =</span> visium_glioblastoma,</span>
<span id="cb159-2"><a href="multi-omics-integration.html#cb159-2" tabindex="-1"></a>         <span class="at">spat_unit =</span> <span class="st">&quot;cell&quot;</span>,</span>
<span id="cb159-3"><a href="multi-omics-integration.html#cb159-3" tabindex="-1"></a>         <span class="at">feat_type =</span> <span class="st">&quot;rna&quot;</span>,</span>
<span id="cb159-4"><a href="multi-omics-integration.html#cb159-4" tabindex="-1"></a>         <span class="at">cell_color =</span> <span class="st">&quot;integrated_leiden_clus&quot;</span>,</span>
<span id="cb159-5"><a href="multi-omics-integration.html#cb159-5" tabindex="-1"></a>         <span class="at">dim_reduction_name =</span> <span class="st">&quot;integrated.umap&quot;</span>,</span>
<span id="cb159-6"><a href="multi-omics-integration.html#cb159-6" tabindex="-1"></a>         <span class="at">point_size =</span> <span class="fl">1.5</span>,</span>
<span id="cb159-7"><a href="multi-omics-integration.html#cb159-7" tabindex="-1"></a>         <span class="at">title =</span> <span class="st">&quot;Integrated UMAP using Integrated Leiden clusters&quot;</span>,</span>
<span id="cb159-8"><a href="multi-omics-integration.html#cb159-8" tabindex="-1"></a>         <span class="at">axis_title =</span> <span class="dv">12</span>,</span>
<span id="cb159-9"><a href="multi-omics-integration.html#cb159-9" tabindex="-1"></a>         <span class="at">axis_text =</span> <span class="dv">10</span> )</span></code></pre></div>
<p><img src="img/03_session3/5-UMAP.png" width="80%" style="display: block; margin: auto;" /></p>
<ul>
<li>Visualize spatial plot with integrated clusters</li>
</ul>
<div class="sourceCode" id="cb160"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb160-1"><a href="multi-omics-integration.html#cb160-1" tabindex="-1"></a><span class="fu">spatPlot2D</span>(visium_glioblastoma,</span>
<span id="cb160-2"><a href="multi-omics-integration.html#cb160-2" tabindex="-1"></a>           <span class="at">spat_unit =</span> <span class="st">&quot;cell&quot;</span>,</span>
<span id="cb160-3"><a href="multi-omics-integration.html#cb160-3" tabindex="-1"></a>           <span class="at">feat_type =</span> <span class="st">&quot;rna&quot;</span>,</span>
<span id="cb160-4"><a href="multi-omics-integration.html#cb160-4" tabindex="-1"></a>           <span class="at">cell_color =</span> <span class="st">&quot;integrated_leiden_clus&quot;</span>,</span>
<span id="cb160-5"><a href="multi-omics-integration.html#cb160-5" tabindex="-1"></a>           <span class="at">point_size =</span> <span class="dv">2</span>,</span>
<span id="cb160-6"><a href="multi-omics-integration.html#cb160-6" tabindex="-1"></a>           <span class="at">show_image =</span> <span class="cn">FALSE</span>,</span>
<span id="cb160-7"><a href="multi-omics-integration.html#cb160-7" tabindex="-1"></a>           <span class="at">title =</span> <span class="st">&quot;Integrated Leiden clustering&quot;</span>)</span></code></pre></div>
<p><img src="img/03_session3/6-spatPlot2D.png" width="80%" style="display: block; margin: auto;" /></p>
</div>
<div id="session-info-2" class="section level2 hasAnchor" number="15.8">
<h2><span class="header-section-number">15.8</span> Session info<a href="multi-omics-integration.html#session-info-2" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb161"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb161-1"><a href="multi-omics-integration.html#cb161-1" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code></pre></div>
<div class="sourceCode" id="cb162"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb162-1"><a href="multi-omics-integration.html#cb162-1" tabindex="-1"></a>R version <span class="dv">4</span>.<span class="fl">4.1</span> (<span class="dv">2024-06-14</span>)</span>
<span id="cb162-2"><a href="multi-omics-integration.html#cb162-2" tabindex="-1"></a>Platform<span class="sc">:</span> aarch64<span class="sc">-</span>apple<span class="sc">-</span>darwin20</span>
<span id="cb162-3"><a href="multi-omics-integration.html#cb162-3" tabindex="-1"></a>Running under<span class="sc">:</span> macOS Sonoma <span class="fl">14.5</span></span>
<span id="cb162-4"><a href="multi-omics-integration.html#cb162-4" tabindex="-1"></a></span>
<span id="cb162-5"><a href="multi-omics-integration.html#cb162-5" tabindex="-1"></a>Matrix products<span class="sc">:</span> default</span>
<span id="cb162-6"><a href="multi-omics-integration.html#cb162-6" tabindex="-1"></a>BLAS<span class="sc">:</span>   <span class="er">/</span>System<span class="sc">/</span>Library<span class="sc">/</span>Frameworks<span class="sc">/</span>Accelerate.framework<span class="sc">/</span>Versions<span class="sc">/</span>A<span class="sc">/</span>Frameworks<span class="sc">/</span>vecLib.framework<span class="sc">/</span>Versions<span class="sc">/</span>A<span class="sc">/</span>libBLAS.dylib </span>
<span id="cb162-7"><a href="multi-omics-integration.html#cb162-7" tabindex="-1"></a>LAPACK<span class="sc">:</span> <span class="er">/</span>Library<span class="sc">/</span>Frameworks<span class="sc">/</span>R.framework<span class="sc">/</span>Versions<span class="sc">/</span><span class="fl">4.4</span><span class="sc">-</span>arm64<span class="sc">/</span>Resources<span class="sc">/</span>lib<span class="sc">/</span>libRlapack.dylib;  LAPACK version <span class="dv">3</span>.<span class="fl">12.0</span></span>
<span id="cb162-8"><a href="multi-omics-integration.html#cb162-8" tabindex="-1"></a></span>
<span id="cb162-9"><a href="multi-omics-integration.html#cb162-9" tabindex="-1"></a>locale<span class="sc">:</span></span>
<span id="cb162-10"><a href="multi-omics-integration.html#cb162-10" tabindex="-1"></a>[<span class="dv">1</span>] en_US.UTF<span class="dv">-8</span><span class="sc">/</span>en_US.UTF<span class="dv">-8</span><span class="sc">/</span>en_US.UTF<span class="dv">-8</span><span class="sc">/</span>C<span class="sc">/</span>en_US.UTF<span class="dv">-8</span><span class="sc">/</span>en_US.UTF<span class="dv">-8</span></span>
<span id="cb162-11"><a href="multi-omics-integration.html#cb162-11" tabindex="-1"></a></span>
<span id="cb162-12"><a href="multi-omics-integration.html#cb162-12" tabindex="-1"></a>time zone<span class="sc">:</span> America<span class="sc">/</span>New_York</span>
<span id="cb162-13"><a href="multi-omics-integration.html#cb162-13" tabindex="-1"></a>tzcode source<span class="sc">:</span> internal</span>
<span id="cb162-14"><a href="multi-omics-integration.html#cb162-14" tabindex="-1"></a></span>
<span id="cb162-15"><a href="multi-omics-integration.html#cb162-15" tabindex="-1"></a>attached base packages<span class="sc">:</span></span>
<span id="cb162-16"><a href="multi-omics-integration.html#cb162-16" tabindex="-1"></a>[<span class="dv">1</span>] stats     graphics  grDevices utils     datasets  methods   base     </span>
<span id="cb162-17"><a href="multi-omics-integration.html#cb162-17" tabindex="-1"></a></span>
<span id="cb162-18"><a href="multi-omics-integration.html#cb162-18" tabindex="-1"></a>other attached packages<span class="sc">:</span></span>
<span id="cb162-19"><a href="multi-omics-integration.html#cb162-19" tabindex="-1"></a>[<span class="dv">1</span>] Giotto_4.<span class="fl">0.8</span>      GiottoClass_0.<span class="fl">3.1</span></span>
<span id="cb162-20"><a href="multi-omics-integration.html#cb162-20" tabindex="-1"></a></span>
<span id="cb162-21"><a href="multi-omics-integration.html#cb162-21" tabindex="-1"></a>loaded via a <span class="fu">namespace</span> (and not attached)<span class="sc">:</span></span>
<span id="cb162-22"><a href="multi-omics-integration.html#cb162-22" tabindex="-1"></a>  [<span class="dv">1</span>] colorRamp2_0.<span class="fl">1.0</span>      deldir_2<span class="fl">.0</span><span class="dv">-4</span>          rlang_1.<span class="fl">1.4</span>          </span>
<span id="cb162-23"><a href="multi-omics-integration.html#cb162-23" tabindex="-1"></a>  [<span class="dv">4</span>] magrittr_2.<span class="fl">0.3</span>        RcppAnnoy_0.<span class="fl">0.22</span>      GiottoUtils_0.<span class="fl">1.8</span>    </span>
<span id="cb162-24"><a href="multi-omics-integration.html#cb162-24" tabindex="-1"></a>  [<span class="dv">7</span>] matrixStats_1.<span class="fl">3.0</span>     compiler_4.<span class="fl">4.1</span>        png_0<span class="fl">.1</span><span class="dv">-8</span>            </span>
<span id="cb162-25"><a href="multi-omics-integration.html#cb162-25" tabindex="-1"></a> [<span class="dv">10</span>] systemfonts_1.<span class="fl">1.0</span>     vctrs_0.<span class="fl">6.5</span>           reshape2_1.<span class="fl">4.4</span>       </span>
<span id="cb162-26"><a href="multi-omics-integration.html#cb162-26" tabindex="-1"></a> [<span class="dv">13</span>] stringr_1.<span class="fl">5.1</span>         pkgconfig_2.<span class="fl">0.3</span>       crayon_1.<span class="fl">5.3</span>         </span>
<span id="cb162-27"><a href="multi-omics-integration.html#cb162-27" tabindex="-1"></a> [<span class="dv">16</span>] fastmap_1.<span class="fl">2.0</span>         backports_1.<span class="fl">5.0</span>       magick_2.<span class="fl">8.4</span>         </span>
<span id="cb162-28"><a href="multi-omics-integration.html#cb162-28" tabindex="-1"></a> [<span class="dv">19</span>] XVector_0.<span class="fl">44.0</span>        labeling_0.<span class="fl">4.3</span>        utf8_1.<span class="fl">2.4</span>           </span>
<span id="cb162-29"><a href="multi-omics-integration.html#cb162-29" tabindex="-1"></a> [<span class="dv">22</span>] rmarkdown_2<span class="fl">.27</span>        ragg_1.<span class="fl">3.2</span>            purrr_1.<span class="fl">0.2</span>          </span>
<span id="cb162-30"><a href="multi-omics-integration.html#cb162-30" tabindex="-1"></a> [<span class="dv">25</span>] xfun_0<span class="fl">.46</span>             zlibbioc_1.<span class="fl">50.0</span>       beachmat_2.<span class="fl">20.0</span>      </span>
<span id="cb162-31"><a href="multi-omics-integration.html#cb162-31" tabindex="-1"></a> [<span class="dv">28</span>] jsonlite_1.<span class="fl">8.8</span>        DelayedArray_0.<span class="fl">30.1</span>   BiocParallel_1.<span class="fl">38.0</span>  </span>
<span id="cb162-32"><a href="multi-omics-integration.html#cb162-32" tabindex="-1"></a> [<span class="dv">31</span>] terra_1<span class="fl">.7</span><span class="dv">-78</span>          irlba_2.<span class="dv">3</span>.<span class="fl">5.1</span>         parallel_4.<span class="fl">4.1</span>       </span>
<span id="cb162-33"><a href="multi-omics-integration.html#cb162-33" tabindex="-1"></a> [<span class="dv">34</span>] R6_2.<span class="fl">5.1</span>              stringi_1.<span class="fl">8.4</span>         RColorBrewer_1<span class="fl">.1</span><span class="dv">-3</span>   </span>
<span id="cb162-34"><a href="multi-omics-integration.html#cb162-34" tabindex="-1"></a> [<span class="dv">37</span>] reticulate_1.<span class="fl">38.0</span>     parallelly_1.<span class="fl">37.1</span>     scattermore_1<span class="fl">.2</span>      </span>
<span id="cb162-35"><a href="multi-omics-integration.html#cb162-35" tabindex="-1"></a> [<span class="dv">40</span>] Rcpp_1.<span class="fl">0.13</span>           bookdown_0<span class="fl">.40</span>         knitr_1<span class="fl">.48</span>           </span>
<span id="cb162-36"><a href="multi-omics-integration.html#cb162-36" tabindex="-1"></a> [<span class="dv">43</span>] future.apply_1.<span class="fl">11.2</span>   R.utils_2.<span class="fl">12.3</span>        IRanges_2.<span class="fl">38.1</span>       </span>
<span id="cb162-37"><a href="multi-omics-integration.html#cb162-37" tabindex="-1"></a> [<span class="dv">46</span>] Matrix_1<span class="fl">.7</span><span class="dv">-0</span>          igraph_2.<span class="fl">0.3</span>          tidyselect_1.<span class="fl">2.1</span>     </span>
<span id="cb162-38"><a href="multi-omics-integration.html#cb162-38" tabindex="-1"></a> [<span class="dv">49</span>] rstudioapi_0.<span class="fl">16.0</span>     abind_1<span class="fl">.4</span><span class="dv">-5</span>           yaml_2.<span class="fl">3.9</span>           </span>
<span id="cb162-39"><a href="multi-omics-integration.html#cb162-39" tabindex="-1"></a> [<span class="dv">52</span>] codetools_0<span class="fl">.2</span><span class="dv">-20</span>      listenv_0.<span class="fl">9.1</span>         lattice_0<span class="fl">.22</span><span class="dv">-6</span>       </span>
<span id="cb162-40"><a href="multi-omics-integration.html#cb162-40" tabindex="-1"></a> [<span class="dv">55</span>] tibble_3.<span class="fl">2.1</span>          plyr_1.<span class="fl">8.9</span>            withr_3.<span class="fl">0.0</span>          </span>
<span id="cb162-41"><a href="multi-omics-integration.html#cb162-41" tabindex="-1"></a> [<span class="dv">58</span>] evaluate_0.<span class="fl">24.0</span>       future_1.<span class="fl">33.2</span>         pillar_1.<span class="fl">9.0</span>         </span>
<span id="cb162-42"><a href="multi-omics-integration.html#cb162-42" tabindex="-1"></a> [<span class="dv">61</span>] MatrixGenerics_1.<span class="fl">16.0</span> checkmate_2.<span class="fl">3.1</span>       stats4_4.<span class="fl">4.1</span>         </span>
<span id="cb162-43"><a href="multi-omics-integration.html#cb162-43" tabindex="-1"></a> [<span class="dv">64</span>] plotly_4.<span class="fl">10.4</span>         generics_0.<span class="fl">1.3</span>        dbscan_1<span class="fl">.2</span><span class="dv">-0</span>         </span>
<span id="cb162-44"><a href="multi-omics-integration.html#cb162-44" tabindex="-1"></a> [<span class="dv">67</span>] sp_2<span class="fl">.1</span><span class="dv">-4</span>              S4Vectors_0.<span class="fl">42.1</span>      ggplot2_3.<span class="fl">5.1</span>        </span>
<span id="cb162-45"><a href="multi-omics-integration.html#cb162-45" tabindex="-1"></a> [<span class="dv">70</span>] munsell_0.<span class="fl">5.1</span>         scales_1.<span class="fl">3.0</span>          gtools_3.<span class="fl">9.5</span>         </span>
<span id="cb162-46"><a href="multi-omics-integration.html#cb162-46" tabindex="-1"></a> [<span class="dv">73</span>] globals_0.<span class="fl">16.3</span>        glue_1.<span class="fl">7.0</span>            lazyeval_0.<span class="fl">2.2</span>       </span>
<span id="cb162-47"><a href="multi-omics-integration.html#cb162-47" tabindex="-1"></a> [<span class="dv">76</span>] tools_4.<span class="fl">4.1</span>           GiottoVisuals_0.<span class="fl">2.3</span>   data.table_1.<span class="fl">15.4</span>    </span>
<span id="cb162-48"><a href="multi-omics-integration.html#cb162-48" tabindex="-1"></a> [<span class="dv">79</span>] ScaledMatrix_1.<span class="fl">12.0</span>   cowplot_1.<span class="fl">1.3</span>         grid_4.<span class="fl">4.1</span>           </span>
<span id="cb162-49"><a href="multi-omics-integration.html#cb162-49" tabindex="-1"></a> [<span class="dv">82</span>] tidyr_1.<span class="fl">3.1</span>           colorspace_2<span class="fl">.1</span><span class="dv">-0</span>      BiocSingular_1.<span class="fl">20.0</span>  </span>
<span id="cb162-50"><a href="multi-omics-integration.html#cb162-50" tabindex="-1"></a> [<span class="dv">85</span>] cli_3.<span class="fl">6.3</span>             rsvd_1.<span class="fl">0.5</span>            textshaping_0.<span class="fl">4.0</span>    </span>
<span id="cb162-51"><a href="multi-omics-integration.html#cb162-51" tabindex="-1"></a> [<span class="dv">88</span>] fansi_1.<span class="fl">0.6</span>           S4Arrays_1.<span class="fl">4.1</span>        viridisLite_0.<span class="fl">4.2</span>    </span>
<span id="cb162-52"><a href="multi-omics-integration.html#cb162-52" tabindex="-1"></a> [<span class="dv">91</span>] dplyr_1.<span class="fl">1.4</span>           uwot_0.<span class="fl">2.2</span>            gtable_0.<span class="fl">3.5</span>         </span>
<span id="cb162-53"><a href="multi-omics-integration.html#cb162-53" tabindex="-1"></a> [<span class="dv">94</span>] R.methodsS3_1.<span class="fl">8.2</span>     digest_0.<span class="fl">6.36</span>         BiocGenerics_0.<span class="fl">50.0</span>  </span>
<span id="cb162-54"><a href="multi-omics-integration.html#cb162-54" tabindex="-1"></a> [<span class="dv">97</span>] SparseArray_1.<span class="fl">4.8</span>     ggrepel_0.<span class="fl">9.5</span>         htmlwidgets_1.<span class="fl">6.4</span>    </span>
<span id="cb162-55"><a href="multi-omics-integration.html#cb162-55" tabindex="-1"></a>[<span class="dv">100</span>] farver_2.<span class="fl">1.2</span>          htmltools_0.<span class="dv">5</span>.<span class="fl">8.1</span>     R.oo_1.<span class="fl">26.0</span>          </span>
<span id="cb162-56"><a href="multi-omics-integration.html#cb162-56" tabindex="-1"></a>[<span class="dv">103</span>] lifecycle_1.<span class="fl">0.4</span>       httr_1.<span class="fl">4.7</span></span></code></pre></div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="spatial-multi-modal-analysis.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="interoperability-with-other-frameworks.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
