<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>4 Spatial omics technologies | Workshop: Spatial multi-omics data analysis with Giotto Suite</title>
  <meta name="description" content="4 Spatial omics technologies | Workshop: Spatial multi-omics data analysis with Giotto Suite" />
  <meta name="generator" content="bookdown 0.39 and GitBook 2.6.7" />

  <meta property="og:title" content="4 Spatial omics technologies | Workshop: Spatial multi-omics data analysis with Giotto Suite" />
  <meta property="og:type" content="book" />
  
  
  <meta name="github-repo" content="drieslab/giotto_workshop_2024" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="4 Spatial omics technologies | Workshop: Spatial multi-omics data analysis with Giotto Suite" />
  
  
  

<meta name="author" content="Ruben Dries, Jiaji George Chen, Joselyn Cristina Chávez-Fuentes, Junxiang Xu ,Edward Ruiz, Jeff Sheridan, Iqra Amin, Wen Wang" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="needed-packages.html"/>
<link rel="next" href="introduction-to-the-giotto-package.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Giotto Suite Workshop 2024</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#instructors"><i class="fa fa-check"></i><b>1.1</b> Instructors</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#topics-and-schedule"><i class="fa fa-check"></i><b>1.2</b> Topics and Schedule:</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#license"><i class="fa fa-check"></i><b>1.3</b> License</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="datasets-to-download.html"><a href="datasets-to-download.html"><i class="fa fa-check"></i><b>2</b> Datasets to download</a>
<ul>
<li class="chapter" data-level="2.1" data-path="datasets-to-download.html"><a href="datasets-to-download.html#zenodo-repository"><i class="fa fa-check"></i><b>2.1</b> Zenodo repository</a></li>
<li class="chapter" data-level="2.2" data-path="datasets-to-download.html"><a href="datasets-to-download.html#x-genomics-visium-mouse-brain-section-coronal-dataset"><i class="fa fa-check"></i><b>2.2</b> 10X Genomics Visium Mouse Brain Section (Coronal) dataset</a></li>
<li class="chapter" data-level="2.3" data-path="datasets-to-download.html"><a href="datasets-to-download.html#x-genomics-visium-hd-ffpe-human-colon-cancer"><i class="fa fa-check"></i><b>2.3</b> 10X Genomics Visium HD: FFPE Human Colon Cancer</a></li>
<li class="chapter" data-level="2.4" data-path="datasets-to-download.html"><a href="datasets-to-download.html#x-genomics-multi-modal-dataset"><i class="fa fa-check"></i><b>2.4</b> 10X Genomics multi-modal dataset</a></li>
<li class="chapter" data-level="2.5" data-path="datasets-to-download.html"><a href="datasets-to-download.html#x-genomics-multi-omics-visium-cytassist-human-tonsil-dataset"><i class="fa fa-check"></i><b>2.5</b> 10X Genomics multi-omics Visium CytAssist Human Tonsil dataset</a></li>
<li class="chapter" data-level="2.6" data-path="datasets-to-download.html"><a href="datasets-to-download.html#x-genomics-human-prostate-cancer-adenocarcinoma-with-invasive-carcinoma-ffpe"><i class="fa fa-check"></i><b>2.6</b> 10X Genomics Human Prostate Cancer Adenocarcinoma with Invasive Carcinoma (FFPE)</a></li>
<li class="chapter" data-level="2.7" data-path="datasets-to-download.html"><a href="datasets-to-download.html#x-genomics-normal-human-prostate-ffpe"><i class="fa fa-check"></i><b>2.7</b> 10X Genomics Normal Human Prostate (FFPE)</a></li>
<li class="chapter" data-level="2.8" data-path="datasets-to-download.html"><a href="datasets-to-download.html#xenium"><i class="fa fa-check"></i><b>2.8</b> Xenium</a></li>
<li class="chapter" data-level="2.9" data-path="datasets-to-download.html"><a href="datasets-to-download.html#merfish-cortex-dataset"><i class="fa fa-check"></i><b>2.9</b> MERFISH cortex dataset</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="needed-packages.html"><a href="needed-packages.html"><i class="fa fa-check"></i><b>3</b> Needed packages</a>
<ul>
<li class="chapter" data-level="3.1" data-path="needed-packages.html"><a href="needed-packages.html#cran-dependencies"><i class="fa fa-check"></i><b>3.1</b> CRAN dependencies:</a></li>
<li class="chapter" data-level="3.2" data-path="needed-packages.html"><a href="needed-packages.html#terra-installation"><i class="fa fa-check"></i><b>3.2</b> <em>terra</em> installation</a></li>
<li class="chapter" data-level="3.3" data-path="needed-packages.html"><a href="needed-packages.html#matrix-installation"><i class="fa fa-check"></i><b>3.3</b> <em>Matrix</em> installation</a></li>
<li class="chapter" data-level="3.4" data-path="needed-packages.html"><a href="needed-packages.html#giotto-installation"><i class="fa fa-check"></i><b>3.4</b> <em>Giotto</em> installation</a></li>
<li class="chapter" data-level="3.5" data-path="needed-packages.html"><a href="needed-packages.html#irlba-install"><i class="fa fa-check"></i><b>3.5</b> <em>irlba</em> install</a></li>
<li class="chapter" data-level="3.6" data-path="needed-packages.html"><a href="needed-packages.html#arrow-install"><i class="fa fa-check"></i><b>3.6</b> <em>arrow</em> install</a></li>
<li class="chapter" data-level="3.7" data-path="needed-packages.html"><a href="needed-packages.html#bioconductor-dependencies"><i class="fa fa-check"></i><b>3.7</b> Bioconductor dependencies:</a></li>
<li class="chapter" data-level="3.8" data-path="needed-packages.html"><a href="needed-packages.html#cran-packages"><i class="fa fa-check"></i><b>3.8</b> CRAN packages:</a></li>
<li class="chapter" data-level="3.9" data-path="needed-packages.html"><a href="needed-packages.html#packages-from-github"><i class="fa fa-check"></i><b>3.9</b> Packages from GitHub</a></li>
<li class="chapter" data-level="3.10" data-path="needed-packages.html"><a href="needed-packages.html#python-environments"><i class="fa fa-check"></i><b>3.10</b> Python environments</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html"><i class="fa fa-check"></i><b>4</b> Spatial omics technologies</a>
<ul>
<li class="chapter" data-level="4.1" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#presentation"><i class="fa fa-check"></i><b>4.1</b> Presentation</a></li>
<li class="chapter" data-level="4.2" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#short-summary"><i class="fa fa-check"></i><b>4.2</b> Short summary</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#why-do-we-need-spatial-omics-technologies"><i class="fa fa-check"></i><b>4.2.1</b> Why do we need spatial omics technologies?</a></li>
<li class="chapter" data-level="4.2.2" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#what-is-spatial-omics"><i class="fa fa-check"></i><b>4.2.2</b> What is spatial omics?</a></li>
<li class="chapter" data-level="4.2.3" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#what-are-the-main-spatial-omics-technologies"><i class="fa fa-check"></i><b>4.2.3</b> What are the main spatial omics technologies?</a></li>
<li class="chapter" data-level="4.2.4" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#other-spatial-omics-atac-seq-cuttag-lipidomics-etc"><i class="fa fa-check"></i><b>4.2.4</b> Other Spatial omics: ATAC-seq, CUT&amp;Tag, lipidomics, etc</a></li>
<li class="chapter" data-level="4.2.5" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#what-are-the-different-types-of-spatial-downstream-analyses"><i class="fa fa-check"></i><b>4.2.5</b> What are the different types of spatial downstream analyses?</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html"><i class="fa fa-check"></i><b>5</b> Introduction to the Giotto package</a>
<ul>
<li class="chapter" data-level="5.1" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html#presentation-1"><i class="fa fa-check"></i><b>5.1</b> Presentation</a></li>
<li class="chapter" data-level="5.2" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html#ecosystem"><i class="fa fa-check"></i><b>5.2</b> Ecosystem</a></li>
<li class="chapter" data-level="5.3" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html#installation-python-environment"><i class="fa fa-check"></i><b>5.3</b> Installation + python environment</a></li>
<li class="chapter" data-level="5.4" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html#giotto-instructions"><i class="fa fa-check"></i><b>5.4</b> Giotto instructions</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html"><i class="fa fa-check"></i><b>6</b> Data formatting and Pre-processing</a>
<ul>
<li class="chapter" data-level="6.1" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#data-formats"><i class="fa fa-check"></i><b>6.1</b> Data formats</a></li>
<li class="chapter" data-level="6.2" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#pre-processing"><i class="fa fa-check"></i><b>6.2</b> Pre-processing</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html"><i class="fa fa-check"></i><b>7</b> Creating a Giotto object</a>
<ul>
<li class="chapter" data-level="7.0.1" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#used-packages"><i class="fa fa-check"></i><b>7.0.1</b> Used packages</a></li>
<li class="chapter" data-level="7.1" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#overview"><i class="fa fa-check"></i><b>7.1</b> Overview</a></li>
<li class="chapter" data-level="7.2" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#from-matrix-locations"><i class="fa fa-check"></i><b>7.2</b> From matrix + locations</a></li>
<li class="chapter" data-level="7.3" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#from-subcellular-raw-data-transcripts-or-images-polygons"><i class="fa fa-check"></i><b>7.3</b> From subcellular raw data (transcripts or images) + polygons</a></li>
<li class="chapter" data-level="7.4" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#from-piece-wise"><i class="fa fa-check"></i><b>7.4</b> From piece-wise</a></li>
<li class="chapter" data-level="7.5" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#using-convenience-functions-for-popular-technologies-vizgen-xenium-cosmx"><i class="fa fa-check"></i><b>7.5</b> Using convenience functions for popular technologies (Vizgen, Xenium, CosMx, …)</a></li>
<li class="chapter" data-level="7.6" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#spatial-plots"><i class="fa fa-check"></i><b>7.6</b> Spatial plots</a></li>
<li class="chapter" data-level="7.7" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#subsetting"><i class="fa fa-check"></i><b>7.7</b> Subsetting</a></li>
<li class="chapter" data-level="7.8" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#mini-objects-giottodata"><i class="fa fa-check"></i><b>7.8</b> Mini objects &amp; GiottoData</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="visium-part-i.html"><a href="visium-part-i.html"><i class="fa fa-check"></i><b>8</b> Visium Part I</a>
<ul>
<li class="chapter" data-level="8.1" data-path="visium-part-i.html"><a href="visium-part-i.html#introduction-to-the-spatial-dataset"><i class="fa fa-check"></i><b>8.1</b> Introduction to the spatial dataset</a></li>
<li class="chapter" data-level="8.2" data-path="visium-part-i.html"><a href="visium-part-i.html#download-dataset"><i class="fa fa-check"></i><b>8.2</b> Download dataset</a></li>
<li class="chapter" data-level="8.3" data-path="visium-part-i.html"><a href="visium-part-i.html#create-the-giotto-object"><i class="fa fa-check"></i><b>8.3</b> Create the Giotto object</a></li>
<li class="chapter" data-level="8.4" data-path="visium-part-i.html"><a href="visium-part-i.html#subset-on-spots-that-were-covered-by-tissue"><i class="fa fa-check"></i><b>8.4</b> Subset on spots that were covered by tissue</a></li>
<li class="chapter" data-level="8.5" data-path="visium-part-i.html"><a href="visium-part-i.html#quality-control"><i class="fa fa-check"></i><b>8.5</b> Quality control</a></li>
<li class="chapter" data-level="8.6" data-path="visium-part-i.html"><a href="visium-part-i.html#filtering"><i class="fa fa-check"></i><b>8.6</b> Filtering</a></li>
<li class="chapter" data-level="8.7" data-path="visium-part-i.html"><a href="visium-part-i.html#normalization"><i class="fa fa-check"></i><b>8.7</b> Normalization</a></li>
<li class="chapter" data-level="8.8" data-path="visium-part-i.html"><a href="visium-part-i.html#feature-selection"><i class="fa fa-check"></i><b>8.8</b> Feature selection</a>
<ul>
<li class="chapter" data-level="8.8.1" data-path="visium-part-i.html"><a href="visium-part-i.html#highly-variable-features"><i class="fa fa-check"></i><b>8.8.1</b> Highly Variable Features:</a></li>
</ul></li>
<li class="chapter" data-level="8.9" data-path="visium-part-i.html"><a href="visium-part-i.html#dimension-reduction"><i class="fa fa-check"></i><b>8.9</b> Dimension Reduction</a>
<ul>
<li class="chapter" data-level="8.9.1" data-path="visium-part-i.html"><a href="visium-part-i.html#pca"><i class="fa fa-check"></i><b>8.9.1</b> PCA</a></li>
<li class="chapter" data-level="8.9.2" data-path="visium-part-i.html"><a href="visium-part-i.html#umap"><i class="fa fa-check"></i><b>8.9.2</b> UMAP</a></li>
<li class="chapter" data-level="8.9.3" data-path="visium-part-i.html"><a href="visium-part-i.html#t-sne"><i class="fa fa-check"></i><b>8.9.3</b> t-SNE</a></li>
</ul></li>
<li class="chapter" data-level="8.10" data-path="visium-part-i.html"><a href="visium-part-i.html#clustering"><i class="fa fa-check"></i><b>8.10</b> Clustering</a>
<ul>
<li class="chapter" data-level="8.10.1" data-path="visium-part-i.html"><a href="visium-part-i.html#calculate-leiden-clustering"><i class="fa fa-check"></i><b>8.10.1</b> Calculate Leiden clustering</a></li>
<li class="chapter" data-level="8.10.2" data-path="visium-part-i.html"><a href="visium-part-i.html#calculate-louvain-clustering"><i class="fa fa-check"></i><b>8.10.2</b> Calculate Louvain clustering</a></li>
</ul></li>
<li class="chapter" data-level="8.11" data-path="visium-part-i.html"><a href="visium-part-i.html#save-the-object"><i class="fa fa-check"></i><b>8.11</b> Save the object</a></li>
<li class="chapter" data-level="8.12" data-path="visium-part-i.html"><a href="visium-part-i.html#session-info"><i class="fa fa-check"></i><b>8.12</b> Session info</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="visium-part-ii.html"><a href="visium-part-ii.html"><i class="fa fa-check"></i><b>9</b> Visium Part II</a>
<ul>
<li class="chapter" data-level="9.1" data-path="visium-part-ii.html"><a href="visium-part-ii.html#load-the-object"><i class="fa fa-check"></i><b>9.1</b> Load the object</a></li>
<li class="chapter" data-level="9.2" data-path="visium-part-ii.html"><a href="visium-part-ii.html#differential-expression"><i class="fa fa-check"></i><b>9.2</b> Differential expression</a>
<ul>
<li class="chapter" data-level="9.2.1" data-path="visium-part-ii.html"><a href="visium-part-ii.html#gini-markers"><i class="fa fa-check"></i><b>9.2.1</b> Gini markers</a></li>
<li class="chapter" data-level="9.2.2" data-path="visium-part-ii.html"><a href="visium-part-ii.html#scran-markers"><i class="fa fa-check"></i><b>9.2.2</b> Scran markers</a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="visium-part-ii.html"><a href="visium-part-ii.html#enrichment-deconvolution"><i class="fa fa-check"></i><b>9.3</b> Enrichment &amp; Deconvolution</a>
<ul>
<li class="chapter" data-level="9.3.1" data-path="visium-part-ii.html"><a href="visium-part-ii.html#pagerank"><i class="fa fa-check"></i><b>9.3.1</b> PAGE/Rank</a></li>
<li class="chapter" data-level="9.3.2" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatialdwls"><i class="fa fa-check"></i><b>9.3.2</b> SpatialDWLS</a></li>
</ul></li>
<li class="chapter" data-level="9.4" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatial-expression-patterns"><i class="fa fa-check"></i><b>9.4</b> Spatial expression patterns</a>
<ul>
<li class="chapter" data-level="9.4.1" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatial-variable-genes"><i class="fa fa-check"></i><b>9.4.1</b> Spatial variable genes</a></li>
<li class="chapter" data-level="9.4.2" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatial-co-expression-modules"><i class="fa fa-check"></i><b>9.4.2</b> Spatial co-expression modules</a></li>
</ul></li>
<li class="chapter" data-level="9.5" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatially-informed-clusters"><i class="fa fa-check"></i><b>9.5</b> Spatially informed clusters</a></li>
<li class="chapter" data-level="9.6" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatial-domains-hmrf"><i class="fa fa-check"></i><b>9.6</b> Spatial domains HMRF</a></li>
<li class="chapter" data-level="9.7" data-path="visium-part-ii.html"><a href="visium-part-ii.html#interactive-tools"><i class="fa fa-check"></i><b>9.7</b> Interactive tools</a></li>
<li class="chapter" data-level="9.8" data-path="visium-part-ii.html"><a href="visium-part-ii.html#session-info-1"><i class="fa fa-check"></i><b>9.8</b> Session info</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="visium-hd.html"><a href="visium-hd.html"><i class="fa fa-check"></i><b>10</b> Visium HD</a>
<ul>
<li class="chapter" data-level="10.1" data-path="visium-hd.html"><a href="visium-hd.html#objective"><i class="fa fa-check"></i><b>10.1</b> Objective</a></li>
<li class="chapter" data-level="10.2" data-path="visium-hd.html"><a href="visium-hd.html#background"><i class="fa fa-check"></i><b>10.2</b> Background</a>
<ul>
<li class="chapter" data-level="10.2.1" data-path="visium-hd.html"><a href="visium-hd.html#visium-hd-technology"><i class="fa fa-check"></i><b>10.2.1</b> Visium HD Technology</a></li>
<li class="chapter" data-level="10.2.2" data-path="visium-hd.html"><a href="visium-hd.html#colorectal-cancer-sample"><i class="fa fa-check"></i><b>10.2.2</b> Colorectal Cancer Sample</a></li>
</ul></li>
<li class="chapter" data-level="10.3" data-path="visium-hd.html"><a href="visium-hd.html#data-ingestion"><i class="fa fa-check"></i><b>10.3</b> Data Ingestion</a>
<ul>
<li class="chapter" data-level="10.3.1" data-path="visium-hd.html"><a href="visium-hd.html#visium-hd-output-data-format"><i class="fa fa-check"></i><b>10.3.1</b> Visium HD output data format</a></li>
<li class="chapter" data-level="10.3.2" data-path="visium-hd.html"><a href="visium-hd.html#read-in-raw-data"><i class="fa fa-check"></i><b>10.3.2</b> Read in raw data</a></li>
<li class="chapter" data-level="10.3.3" data-path="visium-hd.html"><a href="visium-hd.html#giotto-visium-hd-convenience-function"><i class="fa fa-check"></i><b>10.3.3</b> Giotto Visium HD convenience function</a></li>
</ul></li>
<li class="chapter" data-level="10.4" data-path="visium-hd.html"><a href="visium-hd.html#tiling-and-aggregation"><i class="fa fa-check"></i><b>10.4</b> Tiling and aggregation</a></li>
<li class="chapter" data-level="10.5" data-path="visium-hd.html"><a href="visium-hd.html#scalability-and-projection-functions"><i class="fa fa-check"></i><b>10.5</b> Scalability and projection functions</a></li>
<li class="chapter" data-level="10.6" data-path="visium-hd.html"><a href="visium-hd.html#spatial-expression-patterns-1"><i class="fa fa-check"></i><b>10.6</b> Spatial expression patterns</a></li>
<li class="chapter" data-level="10.7" data-path="visium-hd.html"><a href="visium-hd.html#spatial-co-expression-modules-1"><i class="fa fa-check"></i><b>10.7</b> Spatial co-expression modules</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="xenium-1.html"><a href="xenium-1.html"><i class="fa fa-check"></i><b>11</b> Xenium</a>
<ul>
<li class="chapter" data-level="11.1" data-path="xenium-1.html"><a href="xenium-1.html#introduction-to-spatial-dataset"><i class="fa fa-check"></i><b>11.1</b> Introduction to spatial dataset</a></li>
<li class="chapter" data-level="11.2" data-path="xenium-1.html"><a href="xenium-1.html#additional-package-requirements"><i class="fa fa-check"></i><b>11.2</b> Additional package requirements</a></li>
<li class="chapter" data-level="11.3" data-path="xenium-1.html"><a href="xenium-1.html#introduce-mini-object"><i class="fa fa-check"></i><b>11.3</b> Introduce mini object</a></li>
<li class="chapter" data-level="11.4" data-path="xenium-1.html"><a href="xenium-1.html#read-in-raw-data-1"><i class="fa fa-check"></i><b>11.4</b> Read in raw data</a></li>
<li class="chapter" data-level="11.5" data-path="xenium-1.html"><a href="xenium-1.html#overlap-txs-polygons"><i class="fa fa-check"></i><b>11.5</b> Overlap txs &amp; polygons</a></li>
<li class="chapter" data-level="11.6" data-path="xenium-1.html"><a href="xenium-1.html#transcript-enrichment-gsea"><i class="fa fa-check"></i><b>11.6</b> Transcript enrichment GSEA</a></li>
<li class="chapter" data-level="11.7" data-path="xenium-1.html"><a href="xenium-1.html#spatial-location-analysis"><i class="fa fa-check"></i><b>11.7</b> Spatial location analysis</a></li>
<li class="chapter" data-level="11.8" data-path="xenium-1.html"><a href="xenium-1.html#visualizations"><i class="fa fa-check"></i><b>11.8</b> Visualizations</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="spatial-proteomics-multiplex-if.html"><a href="spatial-proteomics-multiplex-if.html"><i class="fa fa-check"></i><b>12</b> Spatial proteomics: multiplex IF</a>
<ul>
<li class="chapter" data-level="12.1" data-path="spatial-proteomics-multiplex-if.html"><a href="spatial-proteomics-multiplex-if.html#read-in-raw-data-2"><i class="fa fa-check"></i><b>12.1</b> Read in raw data</a></li>
<li class="chapter" data-level="12.2" data-path="spatial-proteomics-multiplex-if.html"><a href="spatial-proteomics-multiplex-if.html#overlap-intensity-workflows"><i class="fa fa-check"></i><b>12.2</b> Overlap intensity &amp; workflows</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html"><i class="fa fa-check"></i><b>13</b> Working with multiple samples</a>
<ul>
<li class="chapter" data-level="13.1" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#objective-1"><i class="fa fa-check"></i><b>13.1</b> Objective</a></li>
<li class="chapter" data-level="13.2" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#background-1"><i class="fa fa-check"></i><b>13.2</b> Background</a>
<ul>
<li class="chapter" data-level="13.2.1" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#dataset"><i class="fa fa-check"></i><b>13.2.1</b> Dataset</a></li>
<li class="chapter" data-level="13.2.2" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#visium-technology"><i class="fa fa-check"></i><b>13.2.2</b> Visium technology</a></li>
</ul></li>
<li class="chapter" data-level="13.3" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#create-individual-giotto-objects"><i class="fa fa-check"></i><b>13.3</b> Create individual giotto objects</a>
<ul>
<li class="chapter" data-level="13.3.1" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#download-the-data"><i class="fa fa-check"></i><b>13.3.1</b> Download the data</a></li>
<li class="chapter" data-level="13.3.2" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#create-giotto-instructions"><i class="fa fa-check"></i><b>13.3.2</b> create giotto instructions</a></li>
<li class="chapter" data-level="13.3.3" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#load-visium-data-into-giotto"><i class="fa fa-check"></i><b>13.3.3</b> load visium data into Giotto</a></li>
<li class="chapter" data-level="13.3.4" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#healthy-prostate-tissue-coverage"><i class="fa fa-check"></i><b>13.3.4</b> Healthy prostate tissue coverage</a></li>
<li class="chapter" data-level="13.3.5" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#adenocarcinoma-prostate-tissue-coverage"><i class="fa fa-check"></i><b>13.3.5</b> Adenocarcinoma prostate tissue coverage</a></li>
<li class="chapter" data-level="13.3.6" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#showing-the-data-strucutre-for-the-inidividual-objects"><i class="fa fa-check"></i><b>13.3.6</b> Showing the data strucutre for the inidividual objects</a></li>
</ul></li>
<li class="chapter" data-level="13.4" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#join-giotto-objects"><i class="fa fa-check"></i><b>13.4</b> Join Giotto Objects</a></li>
<li class="chapter" data-level="13.5" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#visualizing-combined-datasets"><i class="fa fa-check"></i><b>13.5</b> Visualizing combined datasets</a>
<ul>
<li class="chapter" data-level="13.5.1" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#vizualizing-in-the-same-plot"><i class="fa fa-check"></i><b>13.5.1</b> Vizualizing in the same plot</a></li>
<li class="chapter" data-level="13.5.2" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#vizualizing-on-separate-plots"><i class="fa fa-check"></i><b>13.5.2</b> Vizualizing on separate plots</a></li>
</ul></li>
<li class="chapter" data-level="13.6" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#splitting-combined-dataset"><i class="fa fa-check"></i><b>13.6</b> Splitting combined dataset</a></li>
<li class="chapter" data-level="13.7" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#analyzing-joined-objects"><i class="fa fa-check"></i><b>13.7</b> Analyzing joined objects</a>
<ul>
<li class="chapter" data-level="13.7.1" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#normalization-and-adding-statistics"><i class="fa fa-check"></i><b>13.7.1</b> Normalization and adding statistics</a></li>
<li class="chapter" data-level="13.7.2" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#clustering-the-datasets"><i class="fa fa-check"></i><b>13.7.2</b> Clustering the datasets</a></li>
<li class="chapter" data-level="13.7.3" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#vizualizing-spatial-location-of-clusters"><i class="fa fa-check"></i><b>13.7.3</b> Vizualizing spatial location of clusters</a></li>
<li class="chapter" data-level="13.7.4" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#vizualizing-tissue-contribution-to-clusters"><i class="fa fa-check"></i><b>13.7.4</b> Vizualizing tissue contribution to clusters</a></li>
</ul></li>
<li class="chapter" data-level="13.8" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#perform-harmony-and-default-workflows"><i class="fa fa-check"></i><b>13.8</b> Perform Harmony and default workflows</a>
<ul>
<li class="chapter" data-level="13.8.1" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#clustering-harmonized-object"><i class="fa fa-check"></i><b>13.8.1</b> Clustering harmonized object</a></li>
<li class="chapter" data-level="13.8.2" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#vizualizing-the-tissue-contribution"><i class="fa fa-check"></i><b>13.8.2</b> Vizualizing the tissue contribution</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="14" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html"><i class="fa fa-check"></i><b>14</b> Spatial multi-modal analysis</a>
<ul>
<li class="chapter" data-level="14.1" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#overview-1"><i class="fa fa-check"></i><b>14.1</b> Overview</a></li>
<li class="chapter" data-level="14.2" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#spatial-manipulation"><i class="fa fa-check"></i><b>14.2</b> Spatial manipulation</a>
<ul>
<li class="chapter" data-level="14.2.1" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#spatial-transforms"><i class="fa fa-check"></i><b>14.2.1</b> Spatial transforms:</a></li>
<li class="chapter" data-level="14.2.2" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#spatial-utilities"><i class="fa fa-check"></i><b>14.2.2</b> Spatial utilities:</a></li>
<li class="chapter" data-level="14.2.3" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#spatial-classes"><i class="fa fa-check"></i><b>14.2.3</b> Spatial classes:</a></li>
</ul></li>
<li class="chapter" data-level="14.3" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#image-transforms"><i class="fa fa-check"></i><b>14.3</b> Image transforms</a></li>
<li class="chapter" data-level="14.4" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#co-registration-of-datasets"><i class="fa fa-check"></i><b>14.4</b> Co-registration of datasets</a></li>
<li class="chapter" data-level="14.5" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#examples-in-giotto-suite-manuscript"><i class="fa fa-check"></i><b>14.5</b> Examples in giotto suite manuscript</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html"><i class="fa fa-check"></i><b>15</b> Multi-omics integration</a>
<ul>
<li class="chapter" data-level="15.1" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#introduction-to-the-spatial-dataset-1"><i class="fa fa-check"></i><b>15.1</b> Introduction to the spatial dataset</a></li>
<li class="chapter" data-level="15.2" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#download-dataset-1"><i class="fa fa-check"></i><b>15.2</b> Download dataset</a></li>
<li class="chapter" data-level="15.3" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#create-the-giotto-object-1"><i class="fa fa-check"></i><b>15.3</b> Create the Giotto object</a></li>
<li class="chapter" data-level="15.4" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#subset-on-spots-that-were-covered-by-tissue-1"><i class="fa fa-check"></i><b>15.4</b> Subset on spots that were covered by tissue</a></li>
<li class="chapter" data-level="15.5" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#rna-processing"><i class="fa fa-check"></i><b>15.5</b> RNA processing</a></li>
<li class="chapter" data-level="15.6" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#protein-processing"><i class="fa fa-check"></i><b>15.6</b> Protein processing</a></li>
<li class="chapter" data-level="15.7" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#multi-omics-integration-1"><i class="fa fa-check"></i><b>15.7</b> Multi-omics integration</a></li>
<li class="chapter" data-level="15.8" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#session-info-2"><i class="fa fa-check"></i><b>15.8</b> Session info</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html"><i class="fa fa-check"></i><b>16</b> Interoperability with other frameworks</a>
<ul>
<li class="chapter" data-level="16.1" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#load-giotto-object"><i class="fa fa-check"></i><b>16.1</b> Load Giotto object</a></li>
<li class="chapter" data-level="16.2" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#seurat"><i class="fa fa-check"></i><b>16.2</b> Seurat</a>
<ul>
<li class="chapter" data-level="16.2.1" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#conversion-of-giotto-object-to-seurat-object"><i class="fa fa-check"></i><b>16.2.1</b> Conversion of Giotto Object to Seurat Object</a></li>
<li class="chapter" data-level="16.2.2" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#conversion-of-seurat-object-back-to-giotto-object"><i class="fa fa-check"></i><b>16.2.2</b> Conversion of Seurat object Back to Giotto Object</a></li>
</ul></li>
<li class="chapter" data-level="16.3" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#spatialexperiment"><i class="fa fa-check"></i><b>16.3</b> SpatialExperiment</a>
<ul>
<li class="chapter" data-level="16.3.1" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#convert-giotto-object-to-spatialexperiment-object"><i class="fa fa-check"></i><b>16.3.1</b> Convert Giotto Object to SpatialExperiment Object</a></li>
<li class="chapter" data-level="16.3.2" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#conversion-of-spatialexperiment-object-back-to-giotto"><i class="fa fa-check"></i><b>16.3.2</b> Conversion of SpatialExperiment object back to Giotto</a></li>
</ul></li>
<li class="chapter" data-level="16.4" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#anndata"><i class="fa fa-check"></i><b>16.4</b> AnnData</a>
<ul>
<li class="chapter" data-level="16.4.1" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#load-required-libraries"><i class="fa fa-check"></i><b>16.4.1</b> Load Required Libraries</a></li>
<li class="chapter" data-level="16.4.2" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#specify-path-for-results"><i class="fa fa-check"></i><b>16.4.2</b> Specify Path for Results</a></li>
<li class="chapter" data-level="16.4.3" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#giotto-to-anndata"><i class="fa fa-check"></i><b>16.4.3</b> Giotto To AnnData</a></li>
<li class="chapter" data-level="16.4.4" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#convert-anndata-to-giotto"><i class="fa fa-check"></i><b>16.4.4</b> Convert AnnData to Giotto</a></li>
</ul></li>
<li class="chapter" data-level="16.5" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#create-mini-vizgen-object"><i class="fa fa-check"></i><b>16.5</b> Create mini Vizgen object</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html"><i class="fa fa-check"></i><b>17</b> Interoperability with isolated tools</a>
<ul>
<li class="chapter" data-level="17.1" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#spatial-niche-trajectory-analysis"><i class="fa fa-check"></i><b>17.1</b> Spatial niche trajectory analysis</a>
<ul>
<li class="chapter" data-level="17.1.1" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#dataset-download"><i class="fa fa-check"></i><b>17.1.1</b> Dataset download</a></li>
<li class="chapter" data-level="17.1.2" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#create-the-giotto-object-2"><i class="fa fa-check"></i><b>17.1.2</b> Create the Giotto object</a></li>
<li class="chapter" data-level="17.1.3" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#visualization"><i class="fa fa-check"></i><b>17.1.3</b> visualization</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="18" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html"><i class="fa fa-check"></i><b>18</b> Interactivity with the R/Spatial ecosystem</a>
<ul>
<li class="chapter" data-level="18.1" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#kriging"><i class="fa fa-check"></i><b>18.1</b> Kriging</a>
<ul>
<li class="chapter" data-level="18.1.1" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#visium-technology-1"><i class="fa fa-check"></i><b>18.1.1</b> Visium technology</a></li>
<li class="chapter" data-level="18.1.2" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#dataset-1"><i class="fa fa-check"></i><b>18.1.2</b> Dataset</a></li>
<li class="chapter" data-level="18.1.3" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#generating-a-geojson-file-of-nuclei-location"><i class="fa fa-check"></i><b>18.1.3</b> Generating a geojson file of nuclei location</a></li>
</ul></li>
<li class="chapter" data-level="18.2" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#downloading-the-dataset"><i class="fa fa-check"></i><b>18.2</b> Downloading the dataset</a></li>
<li class="chapter" data-level="18.3" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#importing-visium-data"><i class="fa fa-check"></i><b>18.3</b> Importing visium data</a>
<ul>
<li class="chapter" data-level="18.3.1" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#identifying-spatially-organized-features"><i class="fa fa-check"></i><b>18.3.1</b> Identifying spatially organized features</a></li>
</ul></li>
<li class="chapter" data-level="18.4" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#adding-cell-polygons-to-giotto-object"><i class="fa fa-check"></i><b>18.4</b> Adding cell polygons to Giotto object</a>
<ul>
<li class="chapter" data-level="18.4.1" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#read-in-the-poly-information"><i class="fa fa-check"></i><b>18.4.1</b> Read in the poly information</a></li>
<li class="chapter" data-level="18.4.2" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#vizualizing-polygons"><i class="fa fa-check"></i><b>18.4.2</b> Vizualizing polygons</a></li>
<li class="chapter" data-level="18.4.3" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#showing-giotto-object-prior-to-polygon-addition"><i class="fa fa-check"></i><b>18.4.3</b> Showing Giotto object prior to polygon addition</a></li>
<li class="chapter" data-level="18.4.4" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#adding-polygons-to-giotto-object"><i class="fa fa-check"></i><b>18.4.4</b> Adding polygons to giotto object</a></li>
<li class="chapter" data-level="18.4.5" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#check-polygon-information"><i class="fa fa-check"></i><b>18.4.5</b> Check polygon information</a></li>
</ul></li>
<li class="chapter" data-level="18.5" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#performing-kriging"><i class="fa fa-check"></i><b>18.5</b> Performing kriging</a>
<ul>
<li class="chapter" data-level="18.5.1" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#interpolating-features"><i class="fa fa-check"></i><b>18.5.1</b> Interpolating features</a></li>
<li class="chapter" data-level="18.5.2" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#expression-overlap"><i class="fa fa-check"></i><b>18.5.2</b> Expression overlap</a></li>
</ul></li>
<li class="chapter" data-level="18.6" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#reading-in-larger-dataset"><i class="fa fa-check"></i><b>18.6</b> Reading in larger dataset</a></li>
<li class="chapter" data-level="18.7" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#analyzing-interpolated-features"><i class="fa fa-check"></i><b>18.7</b> Analyzing interpolated features</a>
<ul>
<li class="chapter" data-level="18.7.1" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#filter-and-normalization"><i class="fa fa-check"></i><b>18.7.1</b> Filter and normalization</a></li>
<li class="chapter" data-level="18.7.2" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#visualizing-gene-expression-from-interpolated-expression"><i class="fa fa-check"></i><b>18.7.2</b> Visualizing gene expression from interpolated expression</a></li>
<li class="chapter" data-level="18.7.3" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#run-pca"><i class="fa fa-check"></i><b>18.7.3</b> Run PCA</a></li>
<li class="chapter" data-level="18.7.4" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#clustering-1"><i class="fa fa-check"></i><b>18.7.4</b> Clustering</a></li>
<li class="chapter" data-level="18.7.5" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#visualizing-clustering"><i class="fa fa-check"></i><b>18.7.5</b> Visualizing clustering</a></li>
<li class="chapter" data-level="18.7.6" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#cropping-objects"><i class="fa fa-check"></i><b>18.7.6</b> Cropping objects</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="19" data-path="contributing-to-giotto.html"><a href="contributing-to-giotto.html"><i class="fa fa-check"></i><b>19</b> Contributing to Giotto</a>
<ul>
<li class="chapter" data-level="19.1" data-path="contributing-to-giotto.html"><a href="contributing-to-giotto.html#contribution-guideline"><i class="fa fa-check"></i><b>19.1</b> Contribution guideline</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Workshop: Spatial multi-omics data analysis with Giotto Suite</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="spatial-omics-technologies" class="section level1 hasAnchor" number="4">
<h1><span class="header-section-number">4</span> Spatial omics technologies<a href="spatial-omics-technologies.html#spatial-omics-technologies" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Ruben Dries</p>
<p>August 5th 2024</p>
<div id="presentation" class="section level2 hasAnchor" number="4.1">
<h2><span class="header-section-number">4.1</span> Presentation<a href="spatial-omics-technologies.html#presentation" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><a href="https://drieslab.github.io/giotto_workshop_2024/spatial_omics_technologies_GS_workshop.pdf"><embed src="https://drieslab.github.io/giotto_workshop_2024/spatial_omics_technologies_GS_workshop.pdf" width="800px" height="460px" type="application/pdf" /></a></p>
<p><br></p>
</div>
<div id="short-summary" class="section level2 hasAnchor" number="4.2">
<h2><span class="header-section-number">4.2</span> Short summary<a href="spatial-omics-technologies.html#short-summary" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="why-do-we-need-spatial-omics-technologies" class="section level3 hasAnchor" number="4.2.1">
<h3><span class="header-section-number">4.2.1</span> Why do we need spatial omics technologies?<a href="spatial-omics-technologies.html#why-do-we-need-spatial-omics-technologies" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Spatial omics allows us to examine the role of one or more cells within its normal context.
This spatial context is typically organized at multiple length scales, and considers
both adjacent neighboring cells and larger levels of tissue organization.</p>
<p><img src="img/01_session1/0-spatial_vs_singlecell.png" /></p>
</div>
<div id="what-is-spatial-omics" class="section level3 hasAnchor" number="4.2.2">
<h3><span class="header-section-number">4.2.2</span> What is spatial omics?<a href="spatial-omics-technologies.html#what-is-spatial-omics" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Spatial omics is typically a combination of spatial sequencing and/or imaging
together with understanding the obtained results through spatial data science.</p>
<p><img src="img/01_session1/1-spatial_combo.png" /></p>
<p><br></p>
</div>
<div id="what-are-the-main-spatial-omics-technologies" class="section level3 hasAnchor" number="4.2.3">
<h3><span class="header-section-number">4.2.3</span> What are the main spatial omics technologies?<a href="spatial-omics-technologies.html#what-are-the-main-spatial-omics-technologies" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The large majority - and most popular or accessible - spatial technologies are:<br />
- spatial antibody-multiplex proteomics<br />
- spatial multiplex <strong>in situ hybridization</strong> (ISH)-based transcriptomics<br />
- spatial sequencing-based transcriptomics</p>
<p><img src="img/01_session1/2-spatial_technologies_web.png" /></p>
</div>
<div id="other-spatial-omics-atac-seq-cuttag-lipidomics-etc" class="section level3 hasAnchor" number="4.2.4">
<h3><span class="header-section-number">4.2.4</span> Other Spatial omics: ATAC-seq, CUT&amp;Tag, lipidomics, etc<a href="spatial-omics-technologies.html#other-spatial-omics-atac-seq-cuttag-lipidomics-etc" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>A growing number of other spatial technologies exist that profile different types
of molecular analytes. One example is using a deterministic barcoding approach
(Rong Fan’s group) to explore open (ATAC-seq) or modified (CUT&amp;Tag) chromatin in
a spatially aware manner.</p>
<p><img src="img/01_session1/3-spatial_other_technologies.png" /></p>
<p><br></p>
</div>
<div id="what-are-the-different-types-of-spatial-downstream-analyses" class="section level3 hasAnchor" number="4.2.5">
<h3><span class="header-section-number">4.2.5</span> What are the different types of spatial downstream analyses?<a href="spatial-omics-technologies.html#what-are-the-different-types-of-spatial-downstream-analyses" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>There exist a large and diverse amount of different downstream spatial data analyses
that use different available data types and formats as input.</p>
<p><img src="img/01_session1/4-spatial_downstream_analyses.png" /></p>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="needed-packages.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="introduction-to-the-giotto-package.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
