<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>13 Spatial multi-modal analysis | Workshop: Spatial multi-omics data analysis with Giotto Suite</title>
  <meta name="description" content="13 Spatial multi-modal analysis | Workshop: Spatial multi-omics data analysis with Giotto Suite" />
  <meta name="generator" content="bookdown 0.39 and GitBook 2.6.7" />

  <meta property="og:title" content="13 Spatial multi-modal analysis | Workshop: Spatial multi-omics data analysis with Giotto Suite" />
  <meta property="og:type" content="book" />
  
  
  <meta name="github-repo" content="drieslab/giotto_workshop_2024" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="13 Spatial multi-modal analysis | Workshop: Spatial multi-omics data analysis with Giotto Suite" />
  
  
  

<meta name="author" content="Ruben Dries, Jiaji George Chen, Joselyn Cristina ChÃ¡vez-Fuentes, Junxiang Xu ,Edward Ruiz, Jeff Sheridan, Iqra Amin, Wen Wang" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="working-with-multiple-samples.html"/>
<link rel="next" href="multi-omics-integration.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Giotto Suite Workshop 2024</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#instructors"><i class="fa fa-check"></i><b>1.1</b> Instructors</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#topics-and-schedule"><i class="fa fa-check"></i><b>1.2</b> Topics and Schedule:</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#license"><i class="fa fa-check"></i><b>1.3</b> License</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="datasets-packages.html"><a href="datasets-packages.html"><i class="fa fa-check"></i><b>2</b> Datasets &amp; Packages</a>
<ul>
<li class="chapter" data-level="2.1" data-path="datasets-packages.html"><a href="datasets-packages.html#datasets-to-download"><i class="fa fa-check"></i><b>2.1</b> Datasets to download</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="datasets-packages.html"><a href="datasets-packages.html#zenodo-repository"><i class="fa fa-check"></i><b>2.1.1</b> Zenodo repository</a></li>
<li class="chapter" data-level="2.1.2" data-path="datasets-packages.html"><a href="datasets-packages.html#x-genomics-visium-mouse-brain-section-coronal-dataset"><i class="fa fa-check"></i><b>2.1.2</b> 10X Genomics Visium Mouse Brain Section (Coronal) dataset</a></li>
<li class="chapter" data-level="2.1.3" data-path="datasets-packages.html"><a href="datasets-packages.html#x-genomics-visium-hd-ffpe-human-colon-cancer"><i class="fa fa-check"></i><b>2.1.3</b> 10X Genomics Visium HD: FFPE Human Colon Cancer</a></li>
<li class="chapter" data-level="2.1.4" data-path="datasets-packages.html"><a href="datasets-packages.html#x-genomics-multi-modal-dataset"><i class="fa fa-check"></i><b>2.1.4</b> 10X Genomics multi-modal dataset</a></li>
<li class="chapter" data-level="2.1.5" data-path="datasets-packages.html"><a href="datasets-packages.html#x-genomics-multi-omics-visium-cytassist-human-tonsil-dataset"><i class="fa fa-check"></i><b>2.1.5</b> 10X Genomics multi-omics Visium CytAssist Human Tonsil dataset</a></li>
<li class="chapter" data-level="2.1.6" data-path="datasets-packages.html"><a href="datasets-packages.html#x-genomics-human-prostate-cancer-adenocarcinoma-with-invasive-carcinoma-ffpe"><i class="fa fa-check"></i><b>2.1.6</b> 10X Genomics Human Prostate Cancer Adenocarcinoma with Invasive Carcinoma (FFPE)</a></li>
<li class="chapter" data-level="2.1.7" data-path="datasets-packages.html"><a href="datasets-packages.html#x-genomics-normal-human-prostate-ffpe"><i class="fa fa-check"></i><b>2.1.7</b> 10X Genomics Normal Human Prostate (FFPE)</a></li>
<li class="chapter" data-level="2.1.8" data-path="datasets-packages.html"><a href="datasets-packages.html#xenium"><i class="fa fa-check"></i><b>2.1.8</b> Xenium</a></li>
<li class="chapter" data-level="2.1.9" data-path="datasets-packages.html"><a href="datasets-packages.html#merfish-cortex-dataset"><i class="fa fa-check"></i><b>2.1.9</b> MERFISH cortex dataset</a></li>
<li class="chapter" data-level="2.1.10" data-path="datasets-packages.html"><a href="datasets-packages.html#lunaphore-if-dataset"><i class="fa fa-check"></i><b>2.1.10</b> Lunaphore IF dataset</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="datasets-packages.html"><a href="datasets-packages.html#needed-packages"><i class="fa fa-check"></i><b>2.2</b> Needed packages</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="datasets-packages.html"><a href="datasets-packages.html#cran-dependencies"><i class="fa fa-check"></i><b>2.2.1</b> CRAN dependencies:</a></li>
<li class="chapter" data-level="2.2.2" data-path="datasets-packages.html"><a href="datasets-packages.html#terra-installation"><i class="fa fa-check"></i><b>2.2.2</b> <em>terra</em> installation</a></li>
<li class="chapter" data-level="2.2.3" data-path="datasets-packages.html"><a href="datasets-packages.html#matrix-installation"><i class="fa fa-check"></i><b>2.2.3</b> <em>Matrix</em> installation</a></li>
<li class="chapter" data-level="2.2.4" data-path="datasets-packages.html"><a href="datasets-packages.html#rtools-installation"><i class="fa fa-check"></i><b>2.2.4</b> <em>Rtools</em> installation</a></li>
<li class="chapter" data-level="2.2.5" data-path="datasets-packages.html"><a href="datasets-packages.html#giotto-installation"><i class="fa fa-check"></i><b>2.2.5</b> <em>Giotto</em> installation</a></li>
<li class="chapter" data-level="2.2.6" data-path="datasets-packages.html"><a href="datasets-packages.html#irlba-install"><i class="fa fa-check"></i><b>2.2.6</b> <em>irlba</em> install</a></li>
<li class="chapter" data-level="2.2.7" data-path="datasets-packages.html"><a href="datasets-packages.html#arrow-install"><i class="fa fa-check"></i><b>2.2.7</b> <em>arrow</em> install</a></li>
<li class="chapter" data-level="2.2.8" data-path="datasets-packages.html"><a href="datasets-packages.html#bioconductor-dependencies"><i class="fa fa-check"></i><b>2.2.8</b> Bioconductor dependencies:</a></li>
<li class="chapter" data-level="2.2.9" data-path="datasets-packages.html"><a href="datasets-packages.html#cran-packages"><i class="fa fa-check"></i><b>2.2.9</b> CRAN packages:</a></li>
<li class="chapter" data-level="2.2.10" data-path="datasets-packages.html"><a href="datasets-packages.html#packages-from-github"><i class="fa fa-check"></i><b>2.2.10</b> Packages from GitHub</a></li>
<li class="chapter" data-level="2.2.11" data-path="datasets-packages.html"><a href="datasets-packages.html#python-environments"><i class="fa fa-check"></i><b>2.2.11</b> Python environments</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html"><i class="fa fa-check"></i><b>3</b> Spatial omics technologies</a>
<ul>
<li class="chapter" data-level="3.1" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#presentation"><i class="fa fa-check"></i><b>3.1</b> Presentation</a></li>
<li class="chapter" data-level="3.2" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#short-summary"><i class="fa fa-check"></i><b>3.2</b> Short summary</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#why-do-we-need-spatial-omics-technologies"><i class="fa fa-check"></i><b>3.2.1</b> Why do we need spatial omics technologies?</a></li>
<li class="chapter" data-level="3.2.2" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#what-is-spatial-omics"><i class="fa fa-check"></i><b>3.2.2</b> What is spatial omics?</a></li>
<li class="chapter" data-level="3.2.3" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#what-are-the-main-spatial-omics-technologies"><i class="fa fa-check"></i><b>3.2.3</b> What are the main spatial omics technologies?</a></li>
<li class="chapter" data-level="3.2.4" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#other-spatial-omics-atac-seq-cuttag-lipidomics-etc"><i class="fa fa-check"></i><b>3.2.4</b> Other Spatial omics: ATAC-seq, CUT&amp;Tag, lipidomics, etc</a></li>
<li class="chapter" data-level="3.2.5" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#what-are-the-different-types-of-spatial-downstream-analyses"><i class="fa fa-check"></i><b>3.2.5</b> What are the different types of spatial downstream analyses?</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html"><i class="fa fa-check"></i><b>4</b> Introduction to the Giotto package</a>
<ul>
<li class="chapter" data-level="4.1" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html#presentation-1"><i class="fa fa-check"></i><b>4.1</b> Presentation</a></li>
<li class="chapter" data-level="4.2" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html#ecosystem"><i class="fa fa-check"></i><b>4.2</b> Ecosystem</a></li>
<li class="chapter" data-level="4.3" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html#installation-python-environment"><i class="fa fa-check"></i><b>4.3</b> Installation + python environment</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html#giotto-installation-1"><i class="fa fa-check"></i><b>4.3.1</b> Giotto installation</a></li>
<li class="chapter" data-level="4.3.2" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html#python-environment"><i class="fa fa-check"></i><b>4.3.2</b> Python environment</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html#giotto-instructions"><i class="fa fa-check"></i><b>4.4</b> Giotto instructions</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html"><i class="fa fa-check"></i><b>5</b> Data formatting and Pre-processing</a>
<ul>
<li class="chapter" data-level="5.1" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#data-formats"><i class="fa fa-check"></i><b>5.1</b> Data formats</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#general-formats"><i class="fa fa-check"></i><b>5.1.1</b> General formats</a></li>
<li class="chapter" data-level="5.1.2" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#matrix-formats"><i class="fa fa-check"></i><b>5.1.2</b> Matrix formats</a></li>
<li class="chapter" data-level="5.1.3" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#tabular-formats"><i class="fa fa-check"></i><b>5.1.3</b> Tabular formats</a></li>
<li class="chapter" data-level="5.1.4" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#spatial-formats"><i class="fa fa-check"></i><b>5.1.4</b> Spatial formats</a></li>
<li class="chapter" data-level="5.1.5" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#mask-files"><i class="fa fa-check"></i><b>5.1.5</b> Mask files</a></li>
<li class="chapter" data-level="5.1.6" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#images"><i class="fa fa-check"></i><b>5.1.6</b> images</a></li>
<li class="chapter" data-level="5.1.7" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#jsonlike-formats"><i class="fa fa-check"></i><b>5.1.7</b> jsonlike formats</a></li>
<li class="chapter" data-level="5.1.8" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#hierarchical-formats"><i class="fa fa-check"></i><b>5.1.8</b> Hierarchical formats</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#pre-processing"><i class="fa fa-check"></i><b>5.2</b> Pre-processing</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#expression-matrix"><i class="fa fa-check"></i><b>5.2.1</b> Expression matrix</a></li>
<li class="chapter" data-level="5.2.2" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#spatial-locations"><i class="fa fa-check"></i><b>5.2.2</b> Spatial locations</a></li>
<li class="chapter" data-level="5.2.3" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#giottopoints"><i class="fa fa-check"></i><b>5.2.3</b> giottoPoints</a></li>
<li class="chapter" data-level="5.2.4" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#giottopolygon"><i class="fa fa-check"></i><b>5.2.4</b> giottoPolygon</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#subobject-utility-functions"><i class="fa fa-check"></i><b>5.3</b> Subobject utility functions</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#ids"><i class="fa fa-check"></i><b>5.3.1</b> IDs</a></li>
<li class="chapter" data-level="5.3.2" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#bracket-subsetting-and-extraction"><i class="fa fa-check"></i><b>5.3.2</b> Bracket subsetting and extraction</a></li>
<li class="chapter" data-level="5.3.3" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#nesting-metadata-generics"><i class="fa fa-check"></i><b>5.3.3</b> Nesting metadata generics</a></li>
<li class="chapter" data-level="5.3.4" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#appending-to-a-giotto-object"><i class="fa fa-check"></i><b>5.3.4</b> Appending to a giotto object</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html"><i class="fa fa-check"></i><b>6</b> Creating a Giotto object</a>
<ul>
<li class="chapter" data-level="6.1" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#overview"><i class="fa fa-check"></i><b>6.1</b> Overview</a></li>
<li class="chapter" data-level="6.2" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#giottodata-modular-package"><i class="fa fa-check"></i><b>6.2</b> GiottoData modular package</a></li>
<li class="chapter" data-level="6.3" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#from-matrix-locations"><i class="fa fa-check"></i><b>6.3</b> From matrix + locations</a></li>
<li class="chapter" data-level="6.4" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#from-subcellular-raw-data-transcripts-or-images-polygons"><i class="fa fa-check"></i><b>6.4</b> From subcellular raw data (transcripts or images) + polygons</a></li>
<li class="chapter" data-level="6.5" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#from-piece-wise"><i class="fa fa-check"></i><b>6.5</b> From piece-wise</a></li>
<li class="chapter" data-level="6.6" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#using-convenience-functions-for-popular-technologies-vizgen-xenium-cosmx"><i class="fa fa-check"></i><b>6.6</b> Using convenience functions for popular technologies (Vizgen, Xenium, CosMx, â¦)</a></li>
<li class="chapter" data-level="6.7" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#plotting"><i class="fa fa-check"></i><b>6.7</b> Plotting</a>
<ul>
<li class="chapter" data-level="6.7.1" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#subobject-plotting"><i class="fa fa-check"></i><b>6.7.1</b> Subobject plotting</a></li>
<li class="chapter" data-level="6.7.2" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#additive-subobject-plotting"><i class="fa fa-check"></i><b>6.7.2</b> Additive subobject plotting</a></li>
<li class="chapter" data-level="6.7.3" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#giotto-object-plotting"><i class="fa fa-check"></i><b>6.7.3</b> Giotto object plotting</a></li>
</ul></li>
<li class="chapter" data-level="6.8" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#subsetting"><i class="fa fa-check"></i><b>6.8</b> Subsetting</a></li>
<li class="chapter" data-level="6.9" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#giottodata-mini-objects"><i class="fa fa-check"></i><b>6.9</b> <em>GiottoData</em> Mini objects</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="visium-part-i.html"><a href="visium-part-i.html"><i class="fa fa-check"></i><b>7</b> Visium Part I</a>
<ul>
<li class="chapter" data-level="7.1" data-path="visium-part-i.html"><a href="visium-part-i.html#the-visium-technology"><i class="fa fa-check"></i><b>7.1</b> The Visium technology</a></li>
<li class="chapter" data-level="7.2" data-path="visium-part-i.html"><a href="visium-part-i.html#introduction-to-the-spatial-dataset"><i class="fa fa-check"></i><b>7.2</b> Introduction to the spatial dataset</a></li>
<li class="chapter" data-level="7.3" data-path="visium-part-i.html"><a href="visium-part-i.html#download-dataset"><i class="fa fa-check"></i><b>7.3</b> Download dataset</a></li>
<li class="chapter" data-level="7.4" data-path="visium-part-i.html"><a href="visium-part-i.html#create-the-giotto-object"><i class="fa fa-check"></i><b>7.4</b> Create the Giotto object</a></li>
<li class="chapter" data-level="7.5" data-path="visium-part-i.html"><a href="visium-part-i.html#subset-on-spots-that-were-covered-by-tissue"><i class="fa fa-check"></i><b>7.5</b> Subset on spots that were covered by tissue</a></li>
<li class="chapter" data-level="7.6" data-path="visium-part-i.html"><a href="visium-part-i.html#quality-control"><i class="fa fa-check"></i><b>7.6</b> Quality control</a></li>
<li class="chapter" data-level="7.7" data-path="visium-part-i.html"><a href="visium-part-i.html#filtering"><i class="fa fa-check"></i><b>7.7</b> Filtering</a></li>
<li class="chapter" data-level="7.8" data-path="visium-part-i.html"><a href="visium-part-i.html#normalization"><i class="fa fa-check"></i><b>7.8</b> Normalization</a></li>
<li class="chapter" data-level="7.9" data-path="visium-part-i.html"><a href="visium-part-i.html#feature-selection"><i class="fa fa-check"></i><b>7.9</b> Feature selection</a>
<ul>
<li class="chapter" data-level="7.9.1" data-path="visium-part-i.html"><a href="visium-part-i.html#highly-variable-features"><i class="fa fa-check"></i><b>7.9.1</b> Highly Variable Features:</a></li>
</ul></li>
<li class="chapter" data-level="7.10" data-path="visium-part-i.html"><a href="visium-part-i.html#dimension-reduction"><i class="fa fa-check"></i><b>7.10</b> Dimension Reduction</a>
<ul>
<li class="chapter" data-level="7.10.1" data-path="visium-part-i.html"><a href="visium-part-i.html#pca"><i class="fa fa-check"></i><b>7.10.1</b> PCA</a></li>
<li class="chapter" data-level="7.10.2" data-path="visium-part-i.html"><a href="visium-part-i.html#umap"><i class="fa fa-check"></i><b>7.10.2</b> UMAP</a></li>
<li class="chapter" data-level="7.10.3" data-path="visium-part-i.html"><a href="visium-part-i.html#t-sne"><i class="fa fa-check"></i><b>7.10.3</b> t-SNE</a></li>
</ul></li>
<li class="chapter" data-level="7.11" data-path="visium-part-i.html"><a href="visium-part-i.html#clustering"><i class="fa fa-check"></i><b>7.11</b> Clustering</a>
<ul>
<li class="chapter" data-level="7.11.1" data-path="visium-part-i.html"><a href="visium-part-i.html#calculate-leiden-clustering"><i class="fa fa-check"></i><b>7.11.1</b> Calculate Leiden clustering</a></li>
<li class="chapter" data-level="7.11.2" data-path="visium-part-i.html"><a href="visium-part-i.html#calculate-louvain-clustering"><i class="fa fa-check"></i><b>7.11.2</b> Calculate Louvain clustering</a></li>
</ul></li>
<li class="chapter" data-level="7.12" data-path="visium-part-i.html"><a href="visium-part-i.html#save-the-object"><i class="fa fa-check"></i><b>7.12</b> Save the object</a></li>
<li class="chapter" data-level="7.13" data-path="visium-part-i.html"><a href="visium-part-i.html#session-info"><i class="fa fa-check"></i><b>7.13</b> Session info</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="visium-part-ii.html"><a href="visium-part-ii.html"><i class="fa fa-check"></i><b>8</b> Visium Part II</a>
<ul>
<li class="chapter" data-level="8.1" data-path="visium-part-ii.html"><a href="visium-part-ii.html#load-the-object"><i class="fa fa-check"></i><b>8.1</b> Load the object</a></li>
<li class="chapter" data-level="8.2" data-path="visium-part-ii.html"><a href="visium-part-ii.html#differential-expression"><i class="fa fa-check"></i><b>8.2</b> Differential expression</a>
<ul>
<li class="chapter" data-level="8.2.1" data-path="visium-part-ii.html"><a href="visium-part-ii.html#gini-markers"><i class="fa fa-check"></i><b>8.2.1</b> Gini markers</a></li>
<li class="chapter" data-level="8.2.2" data-path="visium-part-ii.html"><a href="visium-part-ii.html#scran-markers"><i class="fa fa-check"></i><b>8.2.2</b> Scran markers</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="visium-part-ii.html"><a href="visium-part-ii.html#enrichment-deconvolution"><i class="fa fa-check"></i><b>8.3</b> Enrichment &amp; Deconvolution</a>
<ul>
<li class="chapter" data-level="8.3.1" data-path="visium-part-ii.html"><a href="visium-part-ii.html#pagerank"><i class="fa fa-check"></i><b>8.3.1</b> PAGE/Rank</a></li>
<li class="chapter" data-level="8.3.2" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatialdwls"><i class="fa fa-check"></i><b>8.3.2</b> SpatialDWLS</a></li>
</ul></li>
<li class="chapter" data-level="8.4" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatial-expression-patterns"><i class="fa fa-check"></i><b>8.4</b> Spatial expression patterns</a>
<ul>
<li class="chapter" data-level="8.4.1" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatial-variable-genes"><i class="fa fa-check"></i><b>8.4.1</b> Spatial variable genes</a></li>
<li class="chapter" data-level="8.4.2" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatial-co-expression-modules"><i class="fa fa-check"></i><b>8.4.2</b> Spatial co-expression modules</a></li>
</ul></li>
<li class="chapter" data-level="8.5" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatially-informed-clusters"><i class="fa fa-check"></i><b>8.5</b> Spatially informed clusters</a></li>
<li class="chapter" data-level="8.6" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatial-domains-hmrf"><i class="fa fa-check"></i><b>8.6</b> Spatial domains HMRF</a></li>
<li class="chapter" data-level="8.7" data-path="visium-part-ii.html"><a href="visium-part-ii.html#interactive-tools"><i class="fa fa-check"></i><b>8.7</b> Interactive tools</a></li>
<li class="chapter" data-level="8.8" data-path="visium-part-ii.html"><a href="visium-part-ii.html#save-the-object-1"><i class="fa fa-check"></i><b>8.8</b> Save the object</a></li>
<li class="chapter" data-level="8.9" data-path="visium-part-ii.html"><a href="visium-part-ii.html#session-info-1"><i class="fa fa-check"></i><b>8.9</b> Session info</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="visium-hd.html"><a href="visium-hd.html"><i class="fa fa-check"></i><b>9</b> Visium HD</a>
<ul>
<li class="chapter" data-level="9.1" data-path="visium-hd.html"><a href="visium-hd.html#objective"><i class="fa fa-check"></i><b>9.1</b> Objective</a></li>
<li class="chapter" data-level="9.2" data-path="visium-hd.html"><a href="visium-hd.html#background"><i class="fa fa-check"></i><b>9.2</b> Background</a>
<ul>
<li class="chapter" data-level="9.2.1" data-path="visium-hd.html"><a href="visium-hd.html#visium-hd-technology"><i class="fa fa-check"></i><b>9.2.1</b> Visium HD Technology</a></li>
<li class="chapter" data-level="9.2.2" data-path="visium-hd.html"><a href="visium-hd.html#colorectal-cancer-sample"><i class="fa fa-check"></i><b>9.2.2</b> Colorectal Cancer Sample</a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="visium-hd.html"><a href="visium-hd.html#data-ingestion"><i class="fa fa-check"></i><b>9.3</b> Data Ingestion</a>
<ul>
<li class="chapter" data-level="9.3.1" data-path="visium-hd.html"><a href="visium-hd.html#visium-hd-output-data-format"><i class="fa fa-check"></i><b>9.3.1</b> Visium HD output data format</a></li>
<li class="chapter" data-level="9.3.2" data-path="visium-hd.html"><a href="visium-hd.html#mini-visium-hd-dataset"><i class="fa fa-check"></i><b>9.3.2</b> Mini Visium HD dataset</a></li>
<li class="chapter" data-level="9.3.3" data-path="visium-hd.html"><a href="visium-hd.html#giotto-visium-hd-convenience-function"><i class="fa fa-check"></i><b>9.3.3</b> Giotto Visium HD convenience function</a></li>
<li class="chapter" data-level="9.3.4" data-path="visium-hd.html"><a href="visium-hd.html#read-in-data-manually"><i class="fa fa-check"></i><b>9.3.4</b> Read in data manually</a></li>
</ul></li>
<li class="chapter" data-level="9.4" data-path="visium-hd.html"><a href="visium-hd.html#giotto-object-at-hexbin-400"><i class="fa fa-check"></i><b>9.4</b> Giotto object at Hexbin 400</a>
<ul>
<li class="chapter" data-level="9.4.1" data-path="visium-hd.html"><a href="visium-hd.html#create-giotto-points"><i class="fa fa-check"></i><b>9.4.1</b> create giotto points</a></li>
<li class="chapter" data-level="9.4.2" data-path="visium-hd.html"><a href="visium-hd.html#create-giotto-polygons"><i class="fa fa-check"></i><b>9.4.2</b> create giotto polygons</a></li>
<li class="chapter" data-level="9.4.3" data-path="visium-hd.html"><a href="visium-hd.html#combine-giotto-points-and-polygons-to-create-giotto-object"><i class="fa fa-check"></i><b>9.4.3</b> combine Giotto points and polygons to create Giotto object</a></li>
<li class="chapter" data-level="9.4.4" data-path="visium-hd.html"><a href="visium-hd.html#process-giotto-object"><i class="fa fa-check"></i><b>9.4.4</b> Process Giotto object</a></li>
<li class="chapter" data-level="9.4.5" data-path="visium-hd.html"><a href="visium-hd.html#dimension-reduction-clustering"><i class="fa fa-check"></i><b>9.4.5</b> Dimension reduction + clustering</a></li>
</ul></li>
<li class="chapter" data-level="9.5" data-path="visium-hd.html"><a href="visium-hd.html#hexbin-100"><i class="fa fa-check"></i><b>9.5</b> Hexbin 100</a>
<ul>
<li class="chapter" data-level="9.5.1" data-path="visium-hd.html"><a href="visium-hd.html#standard-subcellular-pipeline"><i class="fa fa-check"></i><b>9.5.1</b> Standard subcellular pipeline</a></li>
<li class="chapter" data-level="9.5.2" data-path="visium-hd.html"><a href="visium-hd.html#spatial-expression-patterns-1"><i class="fa fa-check"></i><b>9.5.2</b> Spatial expression patterns</a></li>
</ul></li>
<li class="chapter" data-level="9.6" data-path="visium-hd.html"><a href="visium-hd.html#hexbin-25"><i class="fa fa-check"></i><b>9.6</b> Hexbin 25</a>
<ul>
<li class="chapter" data-level="9.6.1" data-path="visium-hd.html"><a href="visium-hd.html#subcellular-workflow-and-projection-functions"><i class="fa fa-check"></i><b>9.6.1</b> Subcellular workflow and projection functions</a></li>
</ul></li>
<li class="chapter" data-level="9.7" data-path="visium-hd.html"><a href="visium-hd.html#database-backend---work-in-progress-but-coming-soon"><i class="fa fa-check"></i><b>9.7</b> Database backend - Work in progress, but coming soon!</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="xenium-1.html"><a href="xenium-1.html"><i class="fa fa-check"></i><b>10</b> Xenium</a>
<ul>
<li class="chapter" data-level="10.1" data-path="xenium-1.html"><a href="xenium-1.html#introduction-to-spatial-dataset"><i class="fa fa-check"></i><b>10.1</b> Introduction to spatial dataset</a>
<ul>
<li class="chapter" data-level="10.1.1" data-path="xenium-1.html"><a href="xenium-1.html#output-directory-structure"><i class="fa fa-check"></i><b>10.1.1</b> Output directory structure</a></li>
<li class="chapter" data-level="10.1.2" data-path="xenium-1.html"><a href="xenium-1.html#mini-xenium-dataset"><i class="fa fa-check"></i><b>10.1.2</b> Mini Xenium Dataset</a></li>
</ul></li>
<li class="chapter" data-level="10.2" data-path="xenium-1.html"><a href="xenium-1.html#data-prep"><i class="fa fa-check"></i><b>10.2</b> Data prep</a>
<ul>
<li class="chapter" data-level="10.2.1" data-path="xenium-1.html"><a href="xenium-1.html#image-conversion-may-change"><i class="fa fa-check"></i><b>10.2.1</b> Image conversion (may change)</a></li>
</ul></li>
<li class="chapter" data-level="10.3" data-path="xenium-1.html"><a href="xenium-1.html#convenience-function"><i class="fa fa-check"></i><b>10.3</b> Convenience function</a>
<ul>
<li class="chapter" data-level="10.3.1" data-path="xenium-1.html"><a href="xenium-1.html#specific-filepaths"><i class="fa fa-check"></i><b>10.3.1</b> Specific filepaths</a></li>
<li class="chapter" data-level="10.3.2" data-path="xenium-1.html"><a href="xenium-1.html#quality-value"><i class="fa fa-check"></i><b>10.3.2</b> Quality value</a></li>
<li class="chapter" data-level="10.3.3" data-path="xenium-1.html"><a href="xenium-1.html#transcript-type-splitting"><i class="fa fa-check"></i><b>10.3.3</b> Transcript type splitting</a></li>
<li class="chapter" data-level="10.3.4" data-path="xenium-1.html"><a href="xenium-1.html#centroids-calculation"><i class="fa fa-check"></i><b>10.3.4</b> Centroids calculation</a></li>
<li class="chapter" data-level="10.3.5" data-path="xenium-1.html"><a href="xenium-1.html#simple-visualization"><i class="fa fa-check"></i><b>10.3.5</b> Simple visualization</a></li>
</ul></li>
<li class="chapter" data-level="10.4" data-path="xenium-1.html"><a href="xenium-1.html#piecewise-loading"><i class="fa fa-check"></i><b>10.4</b> Piecewise loading</a>
<ul>
<li class="chapter" data-level="10.4.1" data-path="xenium-1.html"><a href="xenium-1.html#load-giottopoints-transcripts"><i class="fa fa-check"></i><b>10.4.1</b> load giottoPoints transcripts</a></li>
<li class="chapter" data-level="10.4.2" data-path="xenium-1.html"><a href="xenium-1.html#optional-loading-pre-aggregated-data"><i class="fa fa-check"></i><b>10.4.2</b> (optional) Loading pre-aggregated data</a></li>
</ul></li>
<li class="chapter" data-level="10.5" data-path="xenium-1.html"><a href="xenium-1.html#xenium-images"><i class="fa fa-check"></i><b>10.5</b> Xenium Images</a>
<ul>
<li class="chapter" data-level="10.5.1" data-path="xenium-1.html"><a href="xenium-1.html#image-metadata"><i class="fa fa-check"></i><b>10.5.1</b> Image metadata</a></li>
<li class="chapter" data-level="10.5.2" data-path="xenium-1.html"><a href="xenium-1.html#image-loading"><i class="fa fa-check"></i><b>10.5.2</b> Image loading</a></li>
</ul></li>
<li class="chapter" data-level="10.6" data-path="xenium-1.html"><a href="xenium-1.html#spatial-aggregation"><i class="fa fa-check"></i><b>10.6</b> Spatial aggregation</a></li>
<li class="chapter" data-level="10.7" data-path="xenium-1.html"><a href="xenium-1.html#aggregate-analyses-workflow"><i class="fa fa-check"></i><b>10.7</b> Aggregate analyses workflow</a>
<ul>
<li class="chapter" data-level="10.7.1" data-path="xenium-1.html"><a href="xenium-1.html#filtering-1"><i class="fa fa-check"></i><b>10.7.1</b> Filtering</a></li>
<li class="chapter" data-level="10.7.2" data-path="xenium-1.html"><a href="xenium-1.html#normalization-1"><i class="fa fa-check"></i><b>10.7.2</b> Normalization</a></li>
<li class="chapter" data-level="10.7.3" data-path="xenium-1.html"><a href="xenium-1.html#dimension-reduction-1"><i class="fa fa-check"></i><b>10.7.3</b> Dimension Reduction</a></li>
<li class="chapter" data-level="10.7.4" data-path="xenium-1.html"><a href="xenium-1.html#clustering-1"><i class="fa fa-check"></i><b>10.7.4</b> Clustering</a></li>
</ul></li>
<li class="chapter" data-level="10.8" data-path="xenium-1.html"><a href="xenium-1.html#niche-clustering"><i class="fa fa-check"></i><b>10.8</b> Niche clustering</a>
<ul>
<li class="chapter" data-level="10.8.1" data-path="xenium-1.html"><a href="xenium-1.html#spatial-network"><i class="fa fa-check"></i><b>10.8.1</b> Spatial network</a></li>
<li class="chapter" data-level="10.8.2" data-path="xenium-1.html"><a href="xenium-1.html#niche-calculation"><i class="fa fa-check"></i><b>10.8.2</b> Niche calculation</a></li>
<li class="chapter" data-level="10.8.3" data-path="xenium-1.html"><a href="xenium-1.html#kmeans-clustering-based-on-niche-signature"><i class="fa fa-check"></i><b>10.8.3</b> kmeans clustering based on niche signature</a></li>
</ul></li>
<li class="chapter" data-level="10.9" data-path="xenium-1.html"><a href="xenium-1.html#cell-proximity-enrichment"><i class="fa fa-check"></i><b>10.9</b> Cell proximity enrichment</a></li>
<li class="chapter" data-level="10.10" data-path="xenium-1.html"><a href="xenium-1.html#pseudovisium"><i class="fa fa-check"></i><b>10.10</b> Pseudovisium</a>
<ul>
<li class="chapter" data-level="10.10.1" data-path="xenium-1.html"><a href="xenium-1.html#pseudovisium-aggregation-and-workflow"><i class="fa fa-check"></i><b>10.10.1</b> Pseudovisium aggregation and workflow</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html"><i class="fa fa-check"></i><b>11</b> Spatial proteomics: Multiplexed Immunofluorescence</a>
<ul>
<li class="chapter" data-level="11.1" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#spatial-proteomics-technologies"><i class="fa fa-check"></i><b>11.1</b> Spatial Proteomics Technologies</a></li>
<li class="chapter" data-level="11.2" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#raw-data-type-coming-out-of-different-technologies"><i class="fa fa-check"></i><b>11.2</b> Raw data type coming out of different technologies</a>
<ul>
<li class="chapter" data-level="11.2.1" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#use-ome.tiff-as-an-example-output-data-to-begin-with"><i class="fa fa-check"></i><b>11.2.1</b> Use ome.tiff as an example output data to begin with</a></li>
<li class="chapter" data-level="11.2.2" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#use-single-channel-images-as-an-example-output-data-to-begin-with"><i class="fa fa-check"></i><b>11.2.2</b> Use single channel images as an example output data to begin with</a></li>
</ul></li>
<li class="chapter" data-level="11.3" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#cell-segmentation-file-to-get-single-cell-level-protein-expression"><i class="fa fa-check"></i><b>11.3</b> Cell Segmentation file to get single cell level protein expression</a>
<ul>
<li class="chapter" data-level="11.3.1" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#using-segmentation-output-file-from-deepcellmesmer-as-an-example."><i class="fa fa-check"></i><b>11.3.1</b> Using segmentation output file from DeepCell(mesmer) as an example.</a></li>
<li class="chapter" data-level="11.3.2" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#using-giotto-wrapper-of-cellpose-to-perform-segmentation"><i class="fa fa-check"></i><b>11.3.2</b> Using Giotto wrapper of Cellpose to perform segmentation</a></li>
</ul></li>
<li class="chapter" data-level="11.4" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#create-a-giotto-object-using-list-of-gitto-large-images-and-polygons"><i class="fa fa-check"></i><b>11.4</b> Create a Giotto Object using list of gitto large images and polygons</a>
<ul>
<li class="chapter" data-level="11.4.1" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#overlap-to-matrix"><i class="fa fa-check"></i><b>11.4.1</b> Overlap to matrix</a></li>
<li class="chapter" data-level="11.4.2" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#manipulate-expression-information"><i class="fa fa-check"></i><b>11.4.2</b> Manipulate Expression information</a></li>
<li class="chapter" data-level="11.4.3" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#rescale-polygons"><i class="fa fa-check"></i><b>11.4.3</b> Rescale polygons</a></li>
<li class="chapter" data-level="11.4.4" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#perform-clustering-and-differential-expression"><i class="fa fa-check"></i><b>11.4.4</b> Perform clustering and differential expression</a></li>
<li class="chapter" data-level="11.4.5" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#give-the-cluster-an-annotation-based-on-expression-values"><i class="fa fa-check"></i><b>11.4.5</b> Give the cluster an annotation based on expression values</a></li>
<li class="chapter" data-level="11.4.6" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#spatial-network-1"><i class="fa fa-check"></i><b>11.4.6</b> spatial network</a></li>
<li class="chapter" data-level="11.4.7" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#cell-neighborhood-cell-typecell-type-interactions"><i class="fa fa-check"></i><b>11.4.7</b> Cell Neighborhood: Cell-Type/Cell-Type Interactions</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="12" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html"><i class="fa fa-check"></i><b>12</b> Working with multiple samples</a>
<ul>
<li class="chapter" data-level="12.1" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#objective-1"><i class="fa fa-check"></i><b>12.1</b> Objective</a></li>
<li class="chapter" data-level="12.2" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#background-1"><i class="fa fa-check"></i><b>12.2</b> Background</a>
<ul>
<li class="chapter" data-level="12.2.1" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#dataset"><i class="fa fa-check"></i><b>12.2.1</b> Dataset</a></li>
<li class="chapter" data-level="12.2.2" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#visium-technology"><i class="fa fa-check"></i><b>12.2.2</b> Visium technology</a></li>
</ul></li>
<li class="chapter" data-level="12.3" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#create-individual-giotto-objects"><i class="fa fa-check"></i><b>12.3</b> Create individual giotto objects</a>
<ul>
<li class="chapter" data-level="12.3.1" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#download-the-data"><i class="fa fa-check"></i><b>12.3.1</b> Download the data</a></li>
</ul></li>
<li class="chapter" data-level="12.4" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#extracting-the-downloaded-files"><i class="fa fa-check"></i><b>12.4</b> Extracting the downloaded files</a>
<ul>
<li class="chapter" data-level="12.4.1" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#create-giotto-instructions"><i class="fa fa-check"></i><b>12.4.1</b> Create giotto instructions</a></li>
<li class="chapter" data-level="12.4.2" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#load-visium-data-into-giotto"><i class="fa fa-check"></i><b>12.4.2</b> Load visium data into Giotto</a></li>
<li class="chapter" data-level="12.4.3" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#healthy-prostate-tissue-coverage"><i class="fa fa-check"></i><b>12.4.3</b> Healthy prostate tissue coverage</a></li>
<li class="chapter" data-level="12.4.4" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#adenocarcinoma-prostate-tissue-coverage"><i class="fa fa-check"></i><b>12.4.4</b> Adenocarcinoma prostate tissue coverage</a></li>
<li class="chapter" data-level="12.4.5" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#showing-the-data-strucutre-for-the-inidividual-objects"><i class="fa fa-check"></i><b>12.4.5</b> Showing the data strucutre for the inidividual objects</a></li>
</ul></li>
<li class="chapter" data-level="12.5" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#join-giotto-objects"><i class="fa fa-check"></i><b>12.5</b> Join Giotto Objects</a></li>
<li class="chapter" data-level="12.6" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#visualizing-combined-datasets"><i class="fa fa-check"></i><b>12.6</b> Visualizing combined datasets</a>
<ul>
<li class="chapter" data-level="12.6.1" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#vizualizing-in-the-same-plot"><i class="fa fa-check"></i><b>12.6.1</b> Vizualizing in the same plot</a></li>
<li class="chapter" data-level="12.6.2" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#vizualizing-on-separate-plots"><i class="fa fa-check"></i><b>12.6.2</b> Vizualizing on separate plots</a></li>
</ul></li>
<li class="chapter" data-level="12.7" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#splitting-combined-dataset"><i class="fa fa-check"></i><b>12.7</b> Splitting combined dataset</a></li>
<li class="chapter" data-level="12.8" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#analyzing-joined-objects"><i class="fa fa-check"></i><b>12.8</b> Analyzing joined objects</a>
<ul>
<li class="chapter" data-level="12.8.1" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#normalization-and-adding-statistics"><i class="fa fa-check"></i><b>12.8.1</b> Normalization and adding statistics</a></li>
<li class="chapter" data-level="12.8.2" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#clustering-the-datasets"><i class="fa fa-check"></i><b>12.8.2</b> Clustering the datasets</a></li>
<li class="chapter" data-level="12.8.3" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#vizualizing-spatial-location-of-clusters"><i class="fa fa-check"></i><b>12.8.3</b> Vizualizing spatial location of clusters</a></li>
<li class="chapter" data-level="12.8.4" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#vizualizing-tissue-contribution-to-clusters"><i class="fa fa-check"></i><b>12.8.4</b> Vizualizing tissue contribution to clusters</a></li>
</ul></li>
<li class="chapter" data-level="12.9" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#perform-harmony-and-default-workflows"><i class="fa fa-check"></i><b>12.9</b> Perform Harmony and default workflows</a>
<ul>
<li class="chapter" data-level="12.9.1" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#clustering-harmonized-object"><i class="fa fa-check"></i><b>12.9.1</b> Clustering harmonized object</a></li>
<li class="chapter" data-level="12.9.2" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#vizualizing-the-tissue-contribution"><i class="fa fa-check"></i><b>12.9.2</b> Vizualizing the tissue contribution</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="13" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html"><i class="fa fa-check"></i><b>13</b> Spatial multi-modal analysis</a>
<ul>
<li class="chapter" data-level="13.1" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#overview-1"><i class="fa fa-check"></i><b>13.1</b> Overview</a></li>
<li class="chapter" data-level="13.2" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#spatial-manipulation"><i class="fa fa-check"></i><b>13.2</b> Spatial manipulation</a>
<ul>
<li class="chapter" data-level="13.2.1" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#spatial-transforms"><i class="fa fa-check"></i><b>13.2.1</b> Spatial transforms:</a></li>
<li class="chapter" data-level="13.2.2" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#spatial-utilities"><i class="fa fa-check"></i><b>13.2.2</b> Spatial utilities:</a></li>
<li class="chapter" data-level="13.2.3" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#spatial-classes"><i class="fa fa-check"></i><b>13.2.3</b> Spatial classes:</a></li>
</ul></li>
<li class="chapter" data-level="13.3" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#examples-of-the-simple-transforms-with-a-giottopolygon"><i class="fa fa-check"></i><b>13.3</b> Examples of the simple transforms with a giottoPolygon</a></li>
<li class="chapter" data-level="13.4" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#affine-transforms"><i class="fa fa-check"></i><b>13.4</b> Affine transforms</a></li>
<li class="chapter" data-level="13.5" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#image-transforms"><i class="fa fa-check"></i><b>13.5</b> Image transforms</a></li>
<li class="chapter" data-level="13.6" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#the-practical-usage-of-multi-modality-co-registration"><i class="fa fa-check"></i><b>13.6</b> The practical usage of multi-modality co-registration</a>
<ul>
<li class="chapter" data-level="13.6.1" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#example-dataset-xenium-breast-cancer-pre-release-pack"><i class="fa fa-check"></i><b>13.6.1</b> Example dataset: Xenium Breast Cancer pre-release pack</a></li>
<li class="chapter" data-level="13.6.2" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#target-coordinate-system"><i class="fa fa-check"></i><b>13.6.2</b> Target Coordinate system</a></li>
<li class="chapter" data-level="13.6.3" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#visium-to-register"><i class="fa fa-check"></i><b>13.6.3</b> Visium to register</a></li>
<li class="chapter" data-level="13.6.4" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#register-post-xenium-he-and-if-image"><i class="fa fa-check"></i><b>13.6.4</b> Register post-Xenium H&amp;E and IF image</a></li>
<li class="chapter" data-level="13.6.5" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#automatic-alignment-via-sift-feature-descriptor-matching-and-affine-transformation"><i class="fa fa-check"></i><b>13.6.5</b> Automatic alignment via SIFT feature descriptor matching and affine transformation</a></li>
<li class="chapter" data-level="13.6.6" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#final-notes"><i class="fa fa-check"></i><b>13.6.6</b> Final Notes</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="14" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html"><i class="fa fa-check"></i><b>14</b> Multi-omics integration</a>
<ul>
<li class="chapter" data-level="14.1" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#the-cytassist-technology"><i class="fa fa-check"></i><b>14.1</b> The CytAssist technology</a></li>
<li class="chapter" data-level="14.2" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#introduction-to-the-spatial-dataset-1"><i class="fa fa-check"></i><b>14.2</b> Introduction to the spatial dataset</a></li>
<li class="chapter" data-level="14.3" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#download-dataset-1"><i class="fa fa-check"></i><b>14.3</b> Download dataset</a></li>
<li class="chapter" data-level="14.4" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#create-the-giotto-object-1"><i class="fa fa-check"></i><b>14.4</b> Create the Giotto object</a></li>
<li class="chapter" data-level="14.5" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#subset-on-spots-that-were-covered-by-tissue-1"><i class="fa fa-check"></i><b>14.5</b> Subset on spots that were covered by tissue</a></li>
<li class="chapter" data-level="14.6" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#rna-processing"><i class="fa fa-check"></i><b>14.6</b> RNA processing</a></li>
<li class="chapter" data-level="14.7" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#protein-processing"><i class="fa fa-check"></i><b>14.7</b> Protein processing</a></li>
<li class="chapter" data-level="14.8" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#multi-omics-integration-1"><i class="fa fa-check"></i><b>14.8</b> Multi-omics integration</a></li>
<li class="chapter" data-level="14.9" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#session-info-2"><i class="fa fa-check"></i><b>14.9</b> Session info</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html"><i class="fa fa-check"></i><b>15</b> Interoperability with other frameworks</a>
<ul>
<li class="chapter" data-level="15.1" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#load-giotto-object"><i class="fa fa-check"></i><b>15.1</b> Load Giotto object</a></li>
<li class="chapter" data-level="15.2" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#seurat"><i class="fa fa-check"></i><b>15.2</b> Seurat</a>
<ul>
<li class="chapter" data-level="15.2.1" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#conversion-of-giotto-object-to-seurat-object"><i class="fa fa-check"></i><b>15.2.1</b> Conversion of Giotto Object to Seurat Object</a></li>
<li class="chapter" data-level="15.2.2" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#conversion-of-seurat-object-back-to-giotto-object"><i class="fa fa-check"></i><b>15.2.2</b> Conversion of Seurat object Back to Giotto Object</a></li>
</ul></li>
<li class="chapter" data-level="15.3" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#spatialexperiment"><i class="fa fa-check"></i><b>15.3</b> SpatialExperiment</a>
<ul>
<li class="chapter" data-level="15.3.1" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#convert-giotto-object-to-spatialexperiment-object"><i class="fa fa-check"></i><b>15.3.1</b> Convert Giotto Object to SpatialExperiment Object</a></li>
<li class="chapter" data-level="15.3.2" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#conversion-of-spatialexperiment-object-back-to-giotto"><i class="fa fa-check"></i><b>15.3.2</b> Conversion of SpatialExperiment object back to Giotto</a></li>
</ul></li>
<li class="chapter" data-level="15.4" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#anndata"><i class="fa fa-check"></i><b>15.4</b> AnnData</a>
<ul>
<li class="chapter" data-level="15.4.1" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#load-required-libraries"><i class="fa fa-check"></i><b>15.4.1</b> Load Required Libraries</a></li>
<li class="chapter" data-level="15.4.2" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#specify-path-for-results"><i class="fa fa-check"></i><b>15.4.2</b> Specify Path for Results</a></li>
<li class="chapter" data-level="15.4.3" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#giotto-to-anndata"><i class="fa fa-check"></i><b>15.4.3</b> Giotto To AnnData</a></li>
<li class="chapter" data-level="15.4.4" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#convert-anndata-to-giotto"><i class="fa fa-check"></i><b>15.4.4</b> Convert AnnData to Giotto</a></li>
</ul></li>
<li class="chapter" data-level="15.5" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#create-mini-vizgen-object"><i class="fa fa-check"></i><b>15.5</b> Create mini Vizgen object</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html"><i class="fa fa-check"></i><b>16</b> Interoperability with isolated tools</a>
<ul>
<li class="chapter" data-level="16.1" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#spatial-niche-trajectory-analysis-ontrac"><i class="fa fa-check"></i><b>16.1</b> Spatial niche trajectory analysis (ONTraC)</a>
<ul>
<li class="chapter" data-level="16.1.1" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#introduction-to-ontrac"><i class="fa fa-check"></i><b>16.1.1</b> Introduction to ONTraC</a></li>
<li class="chapter" data-level="16.1.2" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#introduction-to-merfish"><i class="fa fa-check"></i><b>16.1.2</b> Introduction to MERFISH</a></li>
<li class="chapter" data-level="16.1.3" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#settings"><i class="fa fa-check"></i><b>16.1.3</b> Settings</a></li>
<li class="chapter" data-level="16.1.4" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#dataset-1"><i class="fa fa-check"></i><b>16.1.4</b> Dataset</a></li>
<li class="chapter" data-level="16.1.5" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#create-the-giotto-object-2"><i class="fa fa-check"></i><b>16.1.5</b> Create the Giotto object</a></li>
<li class="chapter" data-level="16.1.6" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#perform-spatial-niche-trajectory-analysis-using-ontrac"><i class="fa fa-check"></i><b>16.1.6</b> Perform spatial niche trajectory analysis using ONTraC</a></li>
<li class="chapter" data-level="16.1.7" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#visualization"><i class="fa fa-check"></i><b>16.1.7</b> Visualization</a></li>
</ul></li>
<li class="chapter" data-level="16.2" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#session-info-3"><i class="fa fa-check"></i><b>16.2</b> Session info</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html"><i class="fa fa-check"></i><b>17</b> Interactivity with the R/Spatial ecosystem</a>
<ul>
<li class="chapter" data-level="17.1" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#kriging"><i class="fa fa-check"></i><b>17.1</b> Kriging</a>
<ul>
<li class="chapter" data-level="17.1.1" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#visium-technology-1"><i class="fa fa-check"></i><b>17.1.1</b> Visium technology</a></li>
<li class="chapter" data-level="17.1.2" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#dataset-2"><i class="fa fa-check"></i><b>17.1.2</b> Dataset</a></li>
<li class="chapter" data-level="17.1.3" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#generating-a-geojson-file-of-nuclei-location"><i class="fa fa-check"></i><b>17.1.3</b> Generating a geojson file of nuclei location</a></li>
</ul></li>
<li class="chapter" data-level="17.2" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#downloading-the-dataset"><i class="fa fa-check"></i><b>17.2</b> Downloading the dataset</a></li>
<li class="chapter" data-level="17.3" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#extracting-the-files"><i class="fa fa-check"></i><b>17.3</b> Extracting the files</a></li>
<li class="chapter" data-level="17.4" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#downloading-giotto-object-and-nuclei-segmentation"><i class="fa fa-check"></i><b>17.4</b> Downloading giotto object and nuclei segmentation</a></li>
<li class="chapter" data-level="17.5" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#importing-visium-data"><i class="fa fa-check"></i><b>17.5</b> Importing visium data</a>
<ul>
<li class="chapter" data-level="17.5.1" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#identifying-spatially-organized-features"><i class="fa fa-check"></i><b>17.5.1</b> Identifying spatially organized features</a></li>
</ul></li>
<li class="chapter" data-level="17.6" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#adding-cell-polygons-to-giotto-object"><i class="fa fa-check"></i><b>17.6</b> Adding cell polygons to Giotto object</a>
<ul>
<li class="chapter" data-level="17.6.1" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#read-in-the-poly-information"><i class="fa fa-check"></i><b>17.6.1</b> Read in the poly information</a></li>
<li class="chapter" data-level="17.6.2" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#vizualizing-polygons"><i class="fa fa-check"></i><b>17.6.2</b> Vizualizing polygons</a></li>
<li class="chapter" data-level="17.6.3" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#showing-giotto-object-prior-to-polygon-addition"><i class="fa fa-check"></i><b>17.6.3</b> Showing Giotto object prior to polygon addition</a></li>
<li class="chapter" data-level="17.6.4" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#adding-polygons-to-giotto-object"><i class="fa fa-check"></i><b>17.6.4</b> Adding polygons to giotto object</a></li>
<li class="chapter" data-level="17.6.5" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#check-polygon-information"><i class="fa fa-check"></i><b>17.6.5</b> Check polygon information</a></li>
</ul></li>
<li class="chapter" data-level="17.7" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#performing-kriging"><i class="fa fa-check"></i><b>17.7</b> Performing kriging</a>
<ul>
<li class="chapter" data-level="17.7.1" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#interpolating-features"><i class="fa fa-check"></i><b>17.7.1</b> Interpolating features</a></li>
<li class="chapter" data-level="17.7.2" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#expression-overlap"><i class="fa fa-check"></i><b>17.7.2</b> Expression overlap</a></li>
</ul></li>
<li class="chapter" data-level="17.8" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#reading-in-larger-dataset"><i class="fa fa-check"></i><b>17.8</b> Reading in larger dataset</a></li>
<li class="chapter" data-level="17.9" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#analyzing-interpolated-features"><i class="fa fa-check"></i><b>17.9</b> Analyzing interpolated features</a>
<ul>
<li class="chapter" data-level="17.9.1" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#filter-and-normalization"><i class="fa fa-check"></i><b>17.9.1</b> Filter and normalization</a></li>
<li class="chapter" data-level="17.9.2" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#visualizing-gene-expression-from-interpolated-expression"><i class="fa fa-check"></i><b>17.9.2</b> Visualizing gene expression from interpolated expression</a></li>
<li class="chapter" data-level="17.9.3" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#run-pca"><i class="fa fa-check"></i><b>17.9.3</b> Run PCA</a></li>
<li class="chapter" data-level="17.9.4" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#clustering-2"><i class="fa fa-check"></i><b>17.9.4</b> Clustering</a></li>
<li class="chapter" data-level="17.9.5" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#visualizing-clustering"><i class="fa fa-check"></i><b>17.9.5</b> Visualizing clustering</a></li>
<li class="chapter" data-level="17.9.6" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#cropping-objects"><i class="fa fa-check"></i><b>17.9.6</b> Cropping objects</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="18" data-path="contributing-to-giotto.html"><a href="contributing-to-giotto.html"><i class="fa fa-check"></i><b>18</b> Contributing to Giotto</a>
<ul>
<li class="chapter" data-level="18.1" data-path="contributing-to-giotto.html"><a href="contributing-to-giotto.html#contribution-guideline"><i class="fa fa-check"></i><b>18.1</b> Contribution guideline</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Workshop: Spatial multi-omics data analysis with Giotto Suite</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="spatial-multi-modal-analysis" class="section level1 hasAnchor" number="13">
<h1><span class="header-section-number">13</span> Spatial multi-modal analysis<a href="spatial-multi-modal-analysis.html#spatial-multi-modal-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>George Chen</p>
<p>Junxiang Xu</p>
<p>August 7th 2024</p>
<div id="overview-1" class="section level2 hasAnchor" number="13.1">
<h2><span class="header-section-number">13.1</span> Overview<a href="spatial-multi-modal-analysis.html#overview-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Spatial multimodal datasets are created when there is more than one modality available for a single piece of tissue. One way that these datasets can be assembled is by performing multiple spatial assays on closely adjacent tissue sections or ideally the same section. However, for these datasets, in addition to the usual expression space integration, we must also first spatially align them.</p>
</div>
<div id="spatial-manipulation" class="section level2 hasAnchor" number="13.2">
<h2><span class="header-section-number">13.2</span> Spatial manipulation<a href="spatial-multi-modal-analysis.html#spatial-manipulation" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Performing spatial analyses across any two sections of tissue from the same block requires that data to be spatially aligned into a common coordinate space. Minute differences during the sectioning process from the cutting motion to how long an FFPE section was floated can result in even neighboring sections being distorted when compared side-by-side.</p>
<p>These differences make it difficult to assemble multislice and/or cross platform multimodal datasets into a cohesive 3D volume. The solution for this is to perform registration across either the dataset images or expression information. Based on the registration results, both the raster images and vector feature and polygon information can be aligned into a continuous whole.</p>
<p>Ideally this registration will be a free deformation based on sets of control points or a deformation matrix, however affine transforms already provide a good approximation. In either case, <em>the transform or deformation applied must work in the same way across both raster and vector information.</em></p>
<p>Giotto provides spatial classes and methods for easy manipulation of data with 2D affine transformations. These functionalities are all available from <em>GiottoClass</em>.</p>
<div id="spatial-transforms" class="section level3 hasAnchor" number="13.2.1">
<h3><span class="header-section-number">13.2.1</span> Spatial transforms:<a href="spatial-multi-modal-analysis.html#spatial-transforms" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>We support simple transformations and more complex affine transformations which can be used to combine and encode more than one simple transform.</p>
<ul>
<li><code>spatShift()</code> - translations</li>
<li><code>spin()</code> - rotations (degrees)</li>
<li><code>rescale()</code> - scaling</li>
<li><code>flip()</code> - flip vertical or horizontal across arbitrary lines</li>
<li><code>t()</code> - transpose</li>
<li><code>shear()</code> - shear transform</li>
<li><code>affine()</code> - affine matrix transform</li>
</ul>
</div>
<div id="spatial-utilities" class="section level3 hasAnchor" number="13.2.2">
<h3><span class="header-section-number">13.2.2</span> Spatial utilities:<a href="spatial-multi-modal-analysis.html#spatial-utilities" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Helpful functions for use alongside these spatial transforms are <code>ext()</code> for finding the spatial bounding box of where your data object is, <code>crop()</code> for cutting out a spatial region of the data, and <code>plot()</code> for <em>terra</em>/<em>base</em> plots of the data.</p>
<ul>
<li><code>ext()</code> - spatial extent or bounding box</li>
<li><code>crop()</code> - cut out a spatial region of the data</li>
<li><code>plot()</code> - plot a spatial object</li>
</ul>
</div>
<div id="spatial-classes" class="section level3 hasAnchor" number="13.2.3">
<h3><span class="header-section-number">13.2.3</span> Spatial classes:<a href="spatial-multi-modal-analysis.html#spatial-classes" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Giottoâs spatial subobjects respond to the above functions. The Giotto object itself can also be affine transformed.</p>
<ul>
<li><code>spatLocsObj</code> - xy centroids</li>
<li><code>spatialNetworkObj</code> - spatial networks between centroids</li>
<li><code>giottoPoints</code> - xy feature point detections</li>
<li><code>giottoPolygon</code> - spatial polygons</li>
<li><code>giottoImage</code> (mostly deprecated) - <em>magick</em>-based images</li>
<li><code>giottoLargeImage</code>/<code>giottoAffineImage</code> - <em>terra</em>-based images</li>
<li><code>affine2d</code> - affine matrix container</li>
<li><code>giotto</code> - giotto analysis object</li>
</ul>
<div class="sourceCode" id="cb314"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb314-1"><a href="spatial-multi-modal-analysis.html#cb314-1" tabindex="-1"></a><span class="co"># load in data</span></span>
<span id="cb314-2"><a href="spatial-multi-modal-analysis.html#cb314-2" tabindex="-1"></a><span class="fu">library</span>(Giotto)</span>
<span id="cb314-3"><a href="spatial-multi-modal-analysis.html#cb314-3" tabindex="-1"></a></span>
<span id="cb314-4"><a href="spatial-multi-modal-analysis.html#cb314-4" tabindex="-1"></a>g <span class="ot">&lt;-</span> GiottoData<span class="sc">::</span><span class="fu">loadGiottoMini</span>(<span class="st">&quot;vizgen&quot;</span>)</span>
<span id="cb314-5"><a href="spatial-multi-modal-analysis.html#cb314-5" tabindex="-1"></a></span>
<span id="cb314-6"><a href="spatial-multi-modal-analysis.html#cb314-6" tabindex="-1"></a><span class="fu">activeSpatUnit</span>(g) <span class="ot">&lt;-</span> <span class="st">&quot;aggregate&quot;</span></span>
<span id="cb314-7"><a href="spatial-multi-modal-analysis.html#cb314-7" tabindex="-1"></a></span>
<span id="cb314-8"><a href="spatial-multi-modal-analysis.html#cb314-8" tabindex="-1"></a>gpoly <span class="ot">&lt;-</span> <span class="fu">getPolygonInfo</span>(g, <span class="at">return_giottoPolygon =</span> <span class="cn">TRUE</span>)</span>
<span id="cb314-9"><a href="spatial-multi-modal-analysis.html#cb314-9" tabindex="-1"></a></span>
<span id="cb314-10"><a href="spatial-multi-modal-analysis.html#cb314-10" tabindex="-1"></a>gimg <span class="ot">&lt;-</span> <span class="fu">getGiottoImage</span>(g)</span></code></pre></div>
</div>
</div>
<div id="examples-of-the-simple-transforms-with-a-giottopolygon" class="section level2 hasAnchor" number="13.3">
<h2><span class="header-section-number">13.3</span> Examples of the simple transforms with a giottoPolygon<a href="spatial-multi-modal-analysis.html#examples-of-the-simple-transforms-with-a-giottopolygon" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb315"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb315-1"><a href="spatial-multi-modal-analysis.html#cb315-1" tabindex="-1"></a>rain <span class="ot">&lt;-</span> <span class="fu">rainbow</span>(<span class="fu">nrow</span>(gpoly))</span>
<span id="cb315-2"><a href="spatial-multi-modal-analysis.html#cb315-2" tabindex="-1"></a></span>
<span id="cb315-3"><a href="spatial-multi-modal-analysis.html#cb315-3" tabindex="-1"></a>line_width <span class="ot">&lt;-</span> <span class="fl">0.3</span></span>
<span id="cb315-4"><a href="spatial-multi-modal-analysis.html#cb315-4" tabindex="-1"></a></span>
<span id="cb315-5"><a href="spatial-multi-modal-analysis.html#cb315-5" tabindex="-1"></a><span class="co"># par to setup the grid plotting layout</span></span>
<span id="cb315-6"><a href="spatial-multi-modal-analysis.html#cb315-6" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">par</span>(<span class="at">no.readonly =</span> <span class="cn">TRUE</span>)</span>
<span id="cb315-7"><a href="spatial-multi-modal-analysis.html#cb315-7" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow=</span><span class="fu">c</span>(<span class="dv">3</span>,<span class="dv">3</span>))</span>
<span id="cb315-8"><a href="spatial-multi-modal-analysis.html#cb315-8" tabindex="-1"></a>gpoly <span class="sc">|&gt;</span> </span>
<span id="cb315-9"><a href="spatial-multi-modal-analysis.html#cb315-9" tabindex="-1"></a>    <span class="fu">plot</span>(<span class="at">main =</span> <span class="st">&quot;no transform&quot;</span>, <span class="at">col =</span> rain, <span class="at">lwd =</span> line_width)</span>
<span id="cb315-10"><a href="spatial-multi-modal-analysis.html#cb315-10" tabindex="-1"></a></span>
<span id="cb315-11"><a href="spatial-multi-modal-analysis.html#cb315-11" tabindex="-1"></a>gpoly <span class="sc">|&gt;</span> <span class="fu">spatShift</span>(<span class="at">dx =</span> <span class="dv">1000</span>) <span class="sc">|&gt;</span> </span>
<span id="cb315-12"><a href="spatial-multi-modal-analysis.html#cb315-12" tabindex="-1"></a>    <span class="fu">plot</span>(<span class="at">main =</span> <span class="st">&quot;spatShift(dx = 1000)&quot;</span>, <span class="at">col =</span> rain, <span class="at">lwd =</span> line_width)</span>
<span id="cb315-13"><a href="spatial-multi-modal-analysis.html#cb315-13" tabindex="-1"></a></span>
<span id="cb315-14"><a href="spatial-multi-modal-analysis.html#cb315-14" tabindex="-1"></a>gpoly <span class="sc">|&gt;</span> <span class="fu">spin</span>(<span class="dv">45</span>) <span class="sc">|&gt;</span> </span>
<span id="cb315-15"><a href="spatial-multi-modal-analysis.html#cb315-15" tabindex="-1"></a>    <span class="fu">plot</span>(<span class="at">main =</span> <span class="st">&quot;spin(45)&quot;</span>, <span class="at">col =</span> rain, <span class="at">lwd =</span> line_width)</span>
<span id="cb315-16"><a href="spatial-multi-modal-analysis.html#cb315-16" tabindex="-1"></a></span>
<span id="cb315-17"><a href="spatial-multi-modal-analysis.html#cb315-17" tabindex="-1"></a>gpoly <span class="sc">|&gt;</span> <span class="fu">rescale</span>(<span class="at">fx =</span> <span class="dv">10</span>, <span class="at">fy =</span> <span class="dv">6</span>) <span class="sc">|&gt;</span> </span>
<span id="cb315-18"><a href="spatial-multi-modal-analysis.html#cb315-18" tabindex="-1"></a>    <span class="fu">plot</span>(<span class="at">main =</span> <span class="st">&quot;rescale(fx = 10, fy = 6)&quot;</span>, <span class="at">col =</span> rain, <span class="at">lwd =</span> line_width)</span>
<span id="cb315-19"><a href="spatial-multi-modal-analysis.html#cb315-19" tabindex="-1"></a></span>
<span id="cb315-20"><a href="spatial-multi-modal-analysis.html#cb315-20" tabindex="-1"></a>gpoly <span class="sc">|&gt;</span> <span class="fu">flip</span>(<span class="at">direction =</span> <span class="st">&quot;vertical&quot;</span>) <span class="sc">|&gt;</span> </span>
<span id="cb315-21"><a href="spatial-multi-modal-analysis.html#cb315-21" tabindex="-1"></a>    <span class="fu">plot</span>(<span class="at">main =</span> <span class="st">&quot;flip()&quot;</span>, <span class="at">col =</span> rain, <span class="at">lwd =</span> line_width)</span>
<span id="cb315-22"><a href="spatial-multi-modal-analysis.html#cb315-22" tabindex="-1"></a></span>
<span id="cb315-23"><a href="spatial-multi-modal-analysis.html#cb315-23" tabindex="-1"></a>gpoly <span class="sc">|&gt;</span> <span class="fu">t</span>() <span class="sc">|&gt;</span> </span>
<span id="cb315-24"><a href="spatial-multi-modal-analysis.html#cb315-24" tabindex="-1"></a>    <span class="fu">plot</span>(<span class="at">main =</span> <span class="st">&quot;t()&quot;</span>, <span class="at">col =</span> rain, <span class="at">lwd =</span> line_width)</span>
<span id="cb315-25"><a href="spatial-multi-modal-analysis.html#cb315-25" tabindex="-1"></a></span>
<span id="cb315-26"><a href="spatial-multi-modal-analysis.html#cb315-26" tabindex="-1"></a>gpoly <span class="sc">|&gt;</span> <span class="fu">shear</span>(<span class="at">fx =</span> <span class="fl">0.5</span>) <span class="sc">|&gt;</span> </span>
<span id="cb315-27"><a href="spatial-multi-modal-analysis.html#cb315-27" tabindex="-1"></a>    <span class="fu">plot</span>(<span class="at">main =</span> <span class="st">&quot;shear(fx = 0.5)&quot;</span>, <span class="at">col =</span> rain, <span class="at">lwd =</span> line_width)</span>
<span id="cb315-28"><a href="spatial-multi-modal-analysis.html#cb315-28" tabindex="-1"></a><span class="fu">par</span>(p)</span></code></pre></div>
<p><img src="img/03_session2/transforms.png" /><!-- --></p>
</div>
<div id="affine-transforms" class="section level2 hasAnchor" number="13.4">
<h2><span class="header-section-number">13.4</span> Affine transforms<a href="spatial-multi-modal-analysis.html#affine-transforms" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The above transforms are all simple to understand in how they work, but you can imagine that performing them in sequence on your dataset can be computationally expensive.</p>
<p>Luckily, the above operations are all affine transformation, and they can be condensed into a single step. Affine transforms where the x and y values undergo a linear transform. These transforms in 2D, can all be represented as a 2x2 matrix or 2x3 if the xy translation values are included.</p>
<p><img src="img/03_session2/matrix.png" /><!-- --></p>
<p>To perform the linear transform, the xy coordinates just need to be matrix multiplied by the 2x2 affine matrix. The resulting values should then be added to the translate values.</p>
<p><img src="img/03_session2/apply_matrix.png" /><!-- --></p>
<p>Due to the nature of matrix multiplication, you can simply multiply the affine matrices with each other and when the xy coordinates are multiplied by the resulting matrix, it performs both linear transforms in the same step.</p>
<p><em>Giotto</em> provides a utility <code>affine2d</code> S4 class that can be created from any affine matrix and responds to the affine transform functions to simplify this accumulation of simple transforms.</p>
<p>Once done, the <code>affine2d</code> can be applied to spatial objects in a single step using <code>affine()</code> in the same way that you would use a matrix.</p>
<div class="sourceCode" id="cb316"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb316-1"><a href="spatial-multi-modal-analysis.html#cb316-1" tabindex="-1"></a><span class="co"># create affine2d</span></span>
<span id="cb316-2"><a href="spatial-multi-modal-analysis.html#cb316-2" tabindex="-1"></a>aff <span class="ot">&lt;-</span> <span class="fu">affine</span>() <span class="co"># when called without params, this is the same as affine(diag(c(1, 1)))</span></span></code></pre></div>
<p>The <code>affine2d</code> object also has an anchor spatial extent, which is used in calculations of the translation values. <code>affine2d</code> generates with a default extent, but a specific one matching that of the object you are manipulating (such as that of the giottoPolygon) should be set.</p>
<div class="sourceCode" id="cb317"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb317-1"><a href="spatial-multi-modal-analysis.html#cb317-1" tabindex="-1"></a>aff<span class="sc">@</span>anchor <span class="ot">&lt;-</span> <span class="fu">ext</span>(gpoly)</span>
<span id="cb317-2"><a href="spatial-multi-modal-analysis.html#cb317-2" tabindex="-1"></a>aff <span class="ot">&lt;-</span> <span class="fu">initialize</span>(aff)</span></code></pre></div>
<div class="sourceCode" id="cb318"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb318-1"><a href="spatial-multi-modal-analysis.html#cb318-1" tabindex="-1"></a><span class="co"># append several simple transforms</span></span>
<span id="cb318-2"><a href="spatial-multi-modal-analysis.html#cb318-2" tabindex="-1"></a>aff <span class="ot">&lt;-</span> aff <span class="sc">|&gt;</span> </span>
<span id="cb318-3"><a href="spatial-multi-modal-analysis.html#cb318-3" tabindex="-1"></a>    <span class="fu">spatShift</span>(<span class="at">dx =</span> <span class="dv">1000</span>) <span class="sc">|&gt;</span></span>
<span id="cb318-4"><a href="spatial-multi-modal-analysis.html#cb318-4" tabindex="-1"></a>    <span class="fu">spin</span>(<span class="dv">45</span>, <span class="at">x0 =</span> <span class="dv">0</span>, <span class="at">y0 =</span> <span class="dv">0</span>) <span class="sc">|&gt;</span> <span class="co"># without the x0, y0 params, the extent center is used</span></span>
<span id="cb318-5"><a href="spatial-multi-modal-analysis.html#cb318-5" tabindex="-1"></a>    <span class="fu">rescale</span>(<span class="dv">10</span>, <span class="at">x0 =</span> <span class="dv">0</span>, <span class="at">y0 =</span> <span class="dv">0</span>) <span class="sc">|&gt;</span> <span class="co"># without the x0, y0 params, the extent center is used</span></span>
<span id="cb318-6"><a href="spatial-multi-modal-analysis.html#cb318-6" tabindex="-1"></a>    <span class="fu">flip</span>(<span class="at">direction =</span> <span class="st">&quot;vertical&quot;</span>) <span class="sc">|&gt;</span></span>
<span id="cb318-7"><a href="spatial-multi-modal-analysis.html#cb318-7" tabindex="-1"></a>    <span class="fu">t</span>() <span class="sc">|&gt;</span></span>
<span id="cb318-8"><a href="spatial-multi-modal-analysis.html#cb318-8" tabindex="-1"></a>    <span class="fu">shear</span>(<span class="at">fx =</span> <span class="fl">0.5</span>)</span>
<span id="cb318-9"><a href="spatial-multi-modal-analysis.html#cb318-9" tabindex="-1"></a><span class="fu">force</span>(aff)</span></code></pre></div>
<pre><code>&lt;affine2d&gt;
anchor    : 6399.24384990901, 6903.24298517207, -5152.38959073896, -4694.86823300896  (xmin, xmax, ymin, ymax)
rotate    : -0.785398163397448  (rad)
shear     : 0.5, 0  (x, y)
scale     : 10, 10  (x, y)
translate : 963.028150700062, 7071.06781186548  (x, y)</code></pre>
<p>The show() function displays some information about the stored affine transform, including a set of decomposed simple transformations.</p>
<p>You can then plot the affine object and see a projection of the spatial transform where blue is the starting position and red is the end.</p>
<div class="sourceCode" id="cb320"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb320-1"><a href="spatial-multi-modal-analysis.html#cb320-1" tabindex="-1"></a><span class="fu">plot</span>(aff)</span></code></pre></div>
<p><img src="img/03_session2/affplot.png" /><!-- --></p>
<p>We can then apply the affine transforms to the <code>giottoPolygon</code> to see that it indeed in the location and orientation that the projection suggests.</p>
<div class="sourceCode" id="cb321"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb321-1"><a href="spatial-multi-modal-analysis.html#cb321-1" tabindex="-1"></a>gpoly <span class="sc">|&gt;</span> <span class="fu">affine</span>(aff) <span class="sc">|&gt;</span> </span>
<span id="cb321-2"><a href="spatial-multi-modal-analysis.html#cb321-2" tabindex="-1"></a>    <span class="fu">plot</span>(<span class="at">main =</span> <span class="st">&quot;affine()&quot;</span>, <span class="at">col =</span> rain, <span class="at">lwd =</span> line_width)</span></code></pre></div>
<p><img src="img/03_session2/affine.png" /><!-- --></p>
</div>
<div id="image-transforms" class="section level2 hasAnchor" number="13.5">
<h2><span class="header-section-number">13.5</span> Image transforms<a href="spatial-multi-modal-analysis.html#image-transforms" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Giotto uses <code>giottoLargeImages</code> as the core image class which is based on <em>terra</em> <code>SpatRaster</code>. Images are not loaded into memory when the object is generated and instead an amount of regular sampling appropriate to the zoom level requested is performed at time of plotting.</p>
<p><code>spatShift()</code> and <code>rescale()</code> operations are supported by <em>terra</em> <code>SpatRaster</code>, and we inherit those functionalities. <code>spin()</code>, <code>flip()</code>, <code>t()</code>, <code>shear()</code>, <code>affine()</code> operations will coerce <code>giottoLargeImage</code> to <code>giottoAffineImage</code>, which is much the same, except it contains an <code>affine2d</code> object that tracks spatial manipulations performed, so that they can be applied through <code>magick::image_distort()</code> processing after sampled values are pulled into memory. <code>giottoAffineImage</code> also has alternative <code>ext()</code> and <code>crop()</code> methods so that those operations respect both the expected post-affine space and un-transformed source image.</p>
<div class="sourceCode" id="cb322"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb322-1"><a href="spatial-multi-modal-analysis.html#cb322-1" tabindex="-1"></a><span class="co"># affine transform of image info matches with polygon info</span></span>
<span id="cb322-2"><a href="spatial-multi-modal-analysis.html#cb322-2" tabindex="-1"></a>gimg <span class="sc">|&gt;</span> <span class="fu">affine</span>(aff) <span class="sc">|&gt;</span> <span class="fu">plot</span>()</span>
<span id="cb322-3"><a href="spatial-multi-modal-analysis.html#cb322-3" tabindex="-1"></a>gpoly <span class="sc">|&gt;</span> <span class="fu">affine</span>(aff) <span class="sc">|&gt;</span> </span>
<span id="cb322-4"><a href="spatial-multi-modal-analysis.html#cb322-4" tabindex="-1"></a>    <span class="fu">plot</span>(<span class="at">add =</span> <span class="cn">TRUE</span>, </span>
<span id="cb322-5"><a href="spatial-multi-modal-analysis.html#cb322-5" tabindex="-1"></a>         <span class="at">border =</span> <span class="st">&quot;cyan&quot;</span>,</span>
<span id="cb322-6"><a href="spatial-multi-modal-analysis.html#cb322-6" tabindex="-1"></a>         <span class="at">lwd =</span> <span class="fl">0.3</span>)</span></code></pre></div>
<p><img src="img/03_session2/polys_and_image.png" /><!-- --></p>
<div class="sourceCode" id="cb323"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb323-1"><a href="spatial-multi-modal-analysis.html#cb323-1" tabindex="-1"></a><span class="co"># affine of the giotto object</span></span>
<span id="cb323-2"><a href="spatial-multi-modal-analysis.html#cb323-2" tabindex="-1"></a>g <span class="sc">|&gt;</span> <span class="fu">affine</span>(aff) <span class="sc">|&gt;</span> </span>
<span id="cb323-3"><a href="spatial-multi-modal-analysis.html#cb323-3" tabindex="-1"></a>    <span class="fu">spatInSituPlotPoints</span>(</span>
<span id="cb323-4"><a href="spatial-multi-modal-analysis.html#cb323-4" tabindex="-1"></a>        <span class="at">show_image =</span> <span class="cn">TRUE</span>, </span>
<span id="cb323-5"><a href="spatial-multi-modal-analysis.html#cb323-5" tabindex="-1"></a>        <span class="at">feats =</span> <span class="fu">list</span>(<span class="at">rna =</span> <span class="fu">c</span>(<span class="st">&quot;Adgrl1&quot;</span>, <span class="st">&quot;Gfap&quot;</span>, <span class="st">&quot;Ntrk3&quot;</span>, <span class="st">&quot;Slc17a7&quot;</span>)),</span>
<span id="cb323-6"><a href="spatial-multi-modal-analysis.html#cb323-6" tabindex="-1"></a>        <span class="at">feats_color_code =</span> <span class="fu">rainbow</span>(<span class="dv">4</span>),</span>
<span id="cb323-7"><a href="spatial-multi-modal-analysis.html#cb323-7" tabindex="-1"></a>        <span class="at">polygon_color =</span> <span class="st">&quot;cyan&quot;</span>,</span>
<span id="cb323-8"><a href="spatial-multi-modal-analysis.html#cb323-8" tabindex="-1"></a>        <span class="at">polygon_line_size =</span> <span class="fl">0.1</span>,</span>
<span id="cb323-9"><a href="spatial-multi-modal-analysis.html#cb323-9" tabindex="-1"></a>        <span class="at">point_size =</span> <span class="fl">0.1</span>,</span>
<span id="cb323-10"><a href="spatial-multi-modal-analysis.html#cb323-10" tabindex="-1"></a>        <span class="at">use_overlap =</span> <span class="cn">FALSE</span></span>
<span id="cb323-11"><a href="spatial-multi-modal-analysis.html#cb323-11" tabindex="-1"></a>    )</span></code></pre></div>
<p><img src="img/03_session2/gobj_affine.png" /><!-- --></p>
<p>Currently <em>giotto</em> image objects are not fully compatible with .ome.tif files. <em>terra</em> which relies on <em>gdal</em> drivers for image loading will find that the Gtiff driver opens some .ome.tif images, but fails when certain compressions (notably JP2000 as used by 10x for their single-channel stains) are used.</p>
</div>
<div id="the-practical-usage-of-multi-modality-co-registration" class="section level2 hasAnchor" number="13.6">
<h2><span class="header-section-number">13.6</span> The practical usage of multi-modality co-registration<a href="spatial-multi-modal-analysis.html#the-practical-usage-of-multi-modality-co-registration" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="example-dataset-xenium-breast-cancer-pre-release-pack" class="section level3 hasAnchor" number="13.6.1">
<h3><span class="header-section-number">13.6.1</span> Example dataset: Xenium Breast Cancer pre-release pack<a href="spatial-multi-modal-analysis.html#example-dataset-xenium-breast-cancer-pre-release-pack" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>10X Genomics Released a comprehensive dataset on 2022. To capture spatial structure by complementing different spatial resolutions and modalities across different assays, they provided a dataset with Xenium in situ transcriptomics data, together with Visium on closely adjacent sections. Additional IF staining was also performed on the Xenium slides. For more information, please refer to the pre-release dataset <a href="https://www.10xgenomics.com/products/xenium-in-situ/preview-dataset-human-breast">page</a> as well as the <a href="https://www.nature.com/articles/s41467-023-43458-x">publication</a>.</p>
<ul>
<li>Visium
<ul>
<li>H&amp;E Histology</li>
<li>55um spot level expression with transcriptome coverage</li>
</ul></li>
<li>Xenium
<ul>
<li>H&amp;E Histology</li>
<li>IF image staining DAPI, HER2 and CD20</li>
<li>in situ transcripts</li>
<li>cooresponding centroid locations</li>
</ul></li>
</ul>
<p>The goal of creating this multi-modal dataset is to register all the modalities listed above to the same coordinate system as Xenium in situ transcripts as the coordinate represents a certain micron distance.</p>
<div class="sourceCode" id="cb324"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb324-1"><a href="spatial-multi-modal-analysis.html#cb324-1" tabindex="-1"></a><span class="fu">library</span>(Giotto)</span>
<span id="cb324-2"><a href="spatial-multi-modal-analysis.html#cb324-2" tabindex="-1"></a>instrs <span class="ot">&lt;-</span> <span class="fu">createGiottoInstructions</span>(<span class="at">save_dir =</span> <span class="fu">file.path</span>(<span class="fu">getwd</span>(),<span class="st">&#39;/img/03_session2/&#39;</span>),</span>
<span id="cb324-3"><a href="spatial-multi-modal-analysis.html#cb324-3" tabindex="-1"></a>                                  <span class="at">save_plot =</span> <span class="cn">TRUE</span>,</span>
<span id="cb324-4"><a href="spatial-multi-modal-analysis.html#cb324-4" tabindex="-1"></a>                                  <span class="at">show_plot =</span> <span class="cn">TRUE</span>)</span>
<span id="cb324-5"><a href="spatial-multi-modal-analysis.html#cb324-5" tabindex="-1"></a><span class="fu">options</span>(<span class="at">timeout =</span> <span class="dv">999999</span>)</span>
<span id="cb324-6"><a href="spatial-multi-modal-analysis.html#cb324-6" tabindex="-1"></a>download_dir <span class="ot">&lt;-</span><span class="fu">file.path</span>(<span class="fu">getwd</span>(),<span class="st">&#39;/data/03_session2/&#39;</span>)</span>
<span id="cb324-7"><a href="spatial-multi-modal-analysis.html#cb324-7" tabindex="-1"></a>destfile <span class="ot">&lt;-</span> <span class="fu">file.path</span>(download_dir,<span class="st">&#39;Multimodal_registration.zip&#39;</span>)</span>
<span id="cb324-8"><a href="spatial-multi-modal-analysis.html#cb324-8" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">dir.exists</span>(download_dir)) { <span class="fu">dir.create</span>(download_dir, <span class="at">recursive =</span> <span class="cn">TRUE</span>) }</span>
<span id="cb324-9"><a href="spatial-multi-modal-analysis.html#cb324-9" tabindex="-1"></a><span class="fu">download.file</span>(<span class="st">&#39;https://zenodo.org/records/13208139/files/Multimodal_registration.zip?download=1&#39;</span>, <span class="at">destfile =</span> destfile)</span>
<span id="cb324-10"><a href="spatial-multi-modal-analysis.html#cb324-10" tabindex="-1"></a><span class="fu">unzip</span>(<span class="fu">paste0</span>(download_dir,<span class="st">&#39;/Multimodal_registration.zip&#39;</span>), <span class="at">exdir =</span> download_dir)</span>
<span id="cb324-11"><a href="spatial-multi-modal-analysis.html#cb324-11" tabindex="-1"></a>Xenium_dir <span class="ot">&lt;-</span> <span class="fu">paste0</span>(download_dir,<span class="st">&#39;/Xenium/&#39;</span>)</span>
<span id="cb324-12"><a href="spatial-multi-modal-analysis.html#cb324-12" tabindex="-1"></a>Visium_dir <span class="ot">&lt;-</span> <span class="fu">paste0</span>(download_dir,<span class="st">&#39;/Visium/&#39;</span>)</span></code></pre></div>
</div>
<div id="target-coordinate-system" class="section level3 hasAnchor" number="13.6.2">
<h3><span class="header-section-number">13.6.2</span> Target Coordinate system<a href="spatial-multi-modal-analysis.html#target-coordinate-system" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Xenium transcripts, polygon information and corresponding centroids are output from the Xenium instrument and are in the same coordinate system from the raw output. We can start with checking the centroid information as a representation of the target coordinate system.</p>
<div class="sourceCode" id="cb325"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb325-1"><a href="spatial-multi-modal-analysis.html#cb325-1" tabindex="-1"></a>xen_cell_df <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="fu">paste0</span>(Xenium_dir,<span class="st">&quot;/cells.csv.gz&quot;</span>))</span>
<span id="cb325-2"><a href="spatial-multi-modal-analysis.html#cb325-2" tabindex="-1"></a>xen_cell_pl <span class="ot">&lt;-</span> ggplot2<span class="sc">::</span><span class="fu">ggplot</span>() <span class="sc">+</span> ggplot2<span class="sc">::</span><span class="fu">geom_point</span>(<span class="at">data =</span> xen_cell_df, ggplot2<span class="sc">::</span><span class="fu">aes</span>(<span class="at">x =</span> x_centroid , <span class="at">y =</span> y_centroid),<span class="at">size =</span> <span class="fl">1e-150</span>,,<span class="at">color =</span> <span class="st">&#39;orange&#39;</span>) <span class="sc">+</span> ggplot2<span class="sc">::</span><span class="fu">theme_classic</span>()</span>
<span id="cb325-3"><a href="spatial-multi-modal-analysis.html#cb325-3" tabindex="-1"></a>xen_cell_pl</span></code></pre></div>
<p><img src="img/03_session2/Xen_cell_centroid.png" width="80%" style="display: block; margin: auto;" /></p>
</div>
<div id="visium-to-register" class="section level3 hasAnchor" number="13.6.3">
<h3><span class="header-section-number">13.6.3</span> Visium to register<a href="spatial-multi-modal-analysis.html#visium-to-register" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Load the Visium directory using the Giotto convenience function, note that here we are using the âtissue_hires_image.pngâ as a image to plot. Using the convenience function, hires image and scale factor stored in the <a href="https://www.10xgenomics.com/support/software/space-ranger/latest/analysis/outputs/output-overview">spaceranger</a> will be used for automatic alignment while the Giotto Visium Object creation. SpatPlot2D provided by Giotto will random sample pixels from the image you provide, thus providing microscopic image as the image input for createGiottoVisiumObject() will improve the visual performance for downstream registration</p>
<div class="sourceCode" id="cb326"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb326-1"><a href="spatial-multi-modal-analysis.html#cb326-1" tabindex="-1"></a>G_visium <span class="ot">&lt;-</span> <span class="fu">createGiottoVisiumObject</span>(<span class="at">visium_dir =</span> Visium_dir,</span>
<span id="cb326-2"><a href="spatial-multi-modal-analysis.html#cb326-2" tabindex="-1"></a>                                     <span class="at">gene_column_index =</span> <span class="dv">2</span>,</span>
<span id="cb326-3"><a href="spatial-multi-modal-analysis.html#cb326-3" tabindex="-1"></a>                                     <span class="at">png_name =</span> <span class="st">&#39;tissue_hires_image.png&#39;</span>,</span>
<span id="cb326-4"><a href="spatial-multi-modal-analysis.html#cb326-4" tabindex="-1"></a>                                     <span class="at">instructions =</span> <span class="cn">NULL</span>)</span>
<span id="cb326-5"><a href="spatial-multi-modal-analysis.html#cb326-5" tabindex="-1"></a><span class="co"># In the meantime, calculate statistics for easier plot showing</span></span>
<span id="cb326-6"><a href="spatial-multi-modal-analysis.html#cb326-6" tabindex="-1"></a>G_visium <span class="ot">&lt;-</span> <span class="fu">normalizeGiotto</span>(G_visium)</span>
<span id="cb326-7"><a href="spatial-multi-modal-analysis.html#cb326-7" tabindex="-1"></a>G_visium <span class="ot">&lt;-</span> <span class="fu">addStatistics</span>(G_visium)</span>
<span id="cb326-8"><a href="spatial-multi-modal-analysis.html#cb326-8" tabindex="-1"></a>V_origin <span class="ot">&lt;-</span> <span class="fu">spatPlot2D</span>(G_visium,<span class="at">show_image =</span> T,<span class="at">point_size =</span> <span class="dv">0</span>,<span class="at">return_plot =</span> T)</span>
<span id="cb326-9"><a href="spatial-multi-modal-analysis.html#cb326-9" tabindex="-1"></a>V_origin</span></code></pre></div>
<p><img src="img/03_session2/Visium_SpatPlot_Original.png" width="80%" style="display: block; margin: auto;" /></p>
<p>The Visium Object needs to be transformed to the same orientation as target coordinate system, so we perform the first transform.</p>
<div class="sourceCode" id="cb327"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb327-1"><a href="spatial-multi-modal-analysis.html#cb327-1" tabindex="-1"></a><span class="co"># create affine2d</span></span>
<span id="cb327-2"><a href="spatial-multi-modal-analysis.html#cb327-2" tabindex="-1"></a>aff <span class="ot">&lt;-</span> <span class="fu">affine</span>(<span class="fu">diag</span>(<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">1</span>)))</span>
<span id="cb327-3"><a href="spatial-multi-modal-analysis.html#cb327-3" tabindex="-1"></a>aff <span class="ot">&lt;-</span> aff <span class="sc">|&gt;</span> </span>
<span id="cb327-4"><a href="spatial-multi-modal-analysis.html#cb327-4" tabindex="-1"></a>    <span class="fu">spin</span>(<span class="dv">90</span>) <span class="sc">|&gt;</span></span>
<span id="cb327-5"><a href="spatial-multi-modal-analysis.html#cb327-5" tabindex="-1"></a>    <span class="fu">flip</span>(<span class="at">direction =</span> <span class="st">&quot;horizontal&quot;</span>)</span>
<span id="cb327-6"><a href="spatial-multi-modal-analysis.html#cb327-6" tabindex="-1"></a><span class="fu">force</span>(aff)</span>
<span id="cb327-7"><a href="spatial-multi-modal-analysis.html#cb327-7" tabindex="-1"></a></span>
<span id="cb327-8"><a href="spatial-multi-modal-analysis.html#cb327-8" tabindex="-1"></a><span class="co"># Apply the transform</span></span>
<span id="cb327-9"><a href="spatial-multi-modal-analysis.html#cb327-9" tabindex="-1"></a>V_tansformed <span class="ot">&lt;-</span> <span class="fu">affine</span>(G_visium,aff)</span>
<span id="cb327-10"><a href="spatial-multi-modal-analysis.html#cb327-10" tabindex="-1"></a>spatplot_to_register <span class="ot">&lt;-</span> <span class="fu">spatPlot2D</span>(V_tansformed,<span class="at">show_image =</span> T,<span class="at">point_size =</span> <span class="dv">0</span>,<span class="at">return_plot =</span> T)</span>
<span id="cb327-11"><a href="spatial-multi-modal-analysis.html#cb327-11" tabindex="-1"></a>spatplot_to_register</span></code></pre></div>
<p><img src="img/03_session2/spatplot_to_regsiter.png" width="80%" style="display: block; margin: auto;" /></p>
<p>Landmarks are considered to be a set of points that are defining same location from two different resources. They are very helpful to be used as anchors to create affine transformtion. For example, after the affine transformation source landmarks should be as close to target landmarks as possible. Since images from different modalities can share similar morphology, the easiest way is to pin landmarks at the morphological identities shared between images.
Giotto provides a interactive landmark selection tool to pin landmarks, two input plots can be generated from a ggplot object, a GiottoLargeImage object, or a path to a image you want to register for. Note that if you directly provide image path, you will need to create a separate GiottoLargeImage to perform transformation, and make sure the GiottoLargeImage has the same coordinate system as shown in the shiny app.</p>
<div class="sourceCode" id="cb328"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb328-1"><a href="spatial-multi-modal-analysis.html#cb328-1" tabindex="-1"></a>landmarks <span class="ot">&lt;-</span> <span class="fu">interactiveLandmarkSelection</span>(spatplot_to_register, xen_cell_pl)</span></code></pre></div>
<p><img src="img/03_session2/landmark_interactive.png" width="80%" style="display: block; margin: auto;" /></p>
<p>Now, use the landmarks to estimate the transformation matrix needed, and to register the Giotto Visium Object to the target coordinate system. For reproducibility purpose, the landmarks used in the chunck below will be loaded from saved result.</p>
<div class="sourceCode" id="cb329"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb329-1"><a href="spatial-multi-modal-analysis.html#cb329-1" tabindex="-1"></a>landmarks<span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="fu">paste0</span>(Xenium_dir,<span class="st">&#39;/Visium_to_Xen_Landmarks.rds&#39;</span>))</span>
<span id="cb329-2"><a href="spatial-multi-modal-analysis.html#cb329-2" tabindex="-1"></a>affine_mtx <span class="ot">&lt;-</span> <span class="fu">calculateAffineMatrixFromLandmarks</span>(landmarks[[<span class="dv">1</span>]],landmarks[[<span class="dv">2</span>]])</span>
<span id="cb329-3"><a href="spatial-multi-modal-analysis.html#cb329-3" tabindex="-1"></a></span>
<span id="cb329-4"><a href="spatial-multi-modal-analysis.html#cb329-4" tabindex="-1"></a>V_final <span class="ot">&lt;-</span> <span class="fu">affine</span>(G_visium,affine_mtx <span class="sc">%*%</span> aff<span class="sc">@</span>affine)</span>
<span id="cb329-5"><a href="spatial-multi-modal-analysis.html#cb329-5" tabindex="-1"></a></span>
<span id="cb329-6"><a href="spatial-multi-modal-analysis.html#cb329-6" tabindex="-1"></a>spatplot_final <span class="ot">&lt;-</span> <span class="fu">spatPlot2D</span>(V_final,<span class="at">show_image =</span> T,<span class="at">point_size =</span> <span class="dv">0</span>,<span class="at">show_plot =</span> F) </span>
<span id="cb329-7"><a href="spatial-multi-modal-analysis.html#cb329-7" tabindex="-1"></a>spatplot_final <span class="sc">+</span> ggplot2<span class="sc">::</span><span class="fu">geom_point</span>(<span class="at">data =</span> xen_cell_df, ggplot2<span class="sc">::</span><span class="fu">aes</span>(<span class="at">x =</span> x_centroid , <span class="at">y =</span> y_centroid),<span class="at">size =</span> <span class="fl">1e-150</span>,,<span class="at">color =</span> <span class="st">&#39;orange&#39;</span>) <span class="sc">+</span> ggplot2<span class="sc">::</span><span class="fu">theme_classic</span>()</span></code></pre></div>
<p><img src="img/03_session2/Visium_resgister_check.png" width="80%" style="display: block; margin: auto;" /></p>
<div id="create-pseudo-visium-dataset-for-comparison" class="section level4 hasAnchor" number="13.6.3.1">
<h4><span class="header-section-number">13.6.3.1</span> Create Pseudo Visium dataset for comparison<a href="spatial-multi-modal-analysis.html#create-pseudo-visium-dataset-for-comparison" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Giotto provides a way to create different shapes on certain locations, we can use that to create a pseudo-visium polygons to aggregate transcripts or image intensities. To do that, we will need the centroid locations, which can be get using getSpatialLocations(). and also the radius information to create circles. We know that Visium certer to center distance is 100um and spot diameter is 55um, thus we can estimate the radius from certer to center distance. And we can use a spatial network created by nearest neighbor = 2 to capture the distance.</p>
<div class="sourceCode" id="cb330"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb330-1"><a href="spatial-multi-modal-analysis.html#cb330-1" tabindex="-1"></a>V_final <span class="ot">&lt;-</span> <span class="fu">createSpatialNetwork</span>(V_final, <span class="at">k =</span> <span class="dv">1</span>)</span>
<span id="cb330-2"><a href="spatial-multi-modal-analysis.html#cb330-2" tabindex="-1"></a>spat_network <span class="ot">&lt;-</span> <span class="fu">getSpatialNetwork</span>(V_final,<span class="at">output =</span> <span class="st">&#39;networkDT&#39;</span>)</span>
<span id="cb330-3"><a href="spatial-multi-modal-analysis.html#cb330-3" tabindex="-1"></a><span class="fu">spatPlot2D</span>(V_final, </span>
<span id="cb330-4"><a href="spatial-multi-modal-analysis.html#cb330-4" tabindex="-1"></a>           <span class="at">show_network =</span> T,</span>
<span id="cb330-5"><a href="spatial-multi-modal-analysis.html#cb330-5" tabindex="-1"></a>           <span class="at">network_color =</span> <span class="st">&#39;blue&#39;</span>,</span>
<span id="cb330-6"><a href="spatial-multi-modal-analysis.html#cb330-6" tabindex="-1"></a>           <span class="at">point_size =</span> <span class="dv">1</span>)</span>
<span id="cb330-7"><a href="spatial-multi-modal-analysis.html#cb330-7" tabindex="-1"></a>center_to_center <span class="ot">&lt;-</span> <span class="fu">min</span>(spat_network<span class="sc">$</span>distance)</span>
<span id="cb330-8"><a href="spatial-multi-modal-analysis.html#cb330-8" tabindex="-1"></a>radius <span class="ot">&lt;-</span> center_to_center<span class="sc">*</span><span class="dv">55</span><span class="sc">/</span><span class="dv">200</span></span></code></pre></div>
<p><img src="img/03_session2/Spatial_network.png" width="80%" style="display: block; margin: auto;" /></p>
<p>Now we get the Pseudo Visium polygons</p>
<div class="sourceCode" id="cb331"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb331-1"><a href="spatial-multi-modal-analysis.html#cb331-1" tabindex="-1"></a>Visium_centroid <span class="ot">&lt;-</span> <span class="fu">getSpatialLocations</span>(V_final,<span class="at">output =</span> <span class="st">&#39;data.table&#39;</span>)</span>
<span id="cb331-2"><a href="spatial-multi-modal-analysis.html#cb331-2" tabindex="-1"></a>stamp_dt <span class="ot">&lt;-</span> <span class="fu">circleVertices</span>(<span class="at">radius =</span> radius, <span class="at">npoints =</span> <span class="dv">100</span>)</span>
<span id="cb331-3"><a href="spatial-multi-modal-analysis.html#cb331-3" tabindex="-1"></a>pseudo_visium_dt <span class="ot">&lt;-</span> <span class="fu">polyStamp</span>(stamp_dt, Visium_centroid)</span>
<span id="cb331-4"><a href="spatial-multi-modal-analysis.html#cb331-4" tabindex="-1"></a>pseudo_visium_poly <span class="ot">&lt;-</span> <span class="fu">createGiottoPolygonsFromDfr</span>(pseudo_visium_dt,<span class="at">calc_centroids =</span> T)</span>
<span id="cb331-5"><a href="spatial-multi-modal-analysis.html#cb331-5" tabindex="-1"></a><span class="fu">plot</span>(pseudo_visium_poly)</span></code></pre></div>
<p><img src="img/03_session2/Visium_polygon.png" width="80%" style="display: block; margin: auto;" /></p>
<p>Create Xenium object with pseudo Visium polygon. To save run time, example shown here only have MS4A1 and ERBB2 genes to create Giotto points</p>
<div class="sourceCode" id="cb332"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb332-1"><a href="spatial-multi-modal-analysis.html#cb332-1" tabindex="-1"></a>xen_transcripts <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="fu">paste0</span>(Xenium_dir,<span class="st">&#39;/Xen_2_genes.csv.gz&#39;</span>))</span>
<span id="cb332-2"><a href="spatial-multi-modal-analysis.html#cb332-2" tabindex="-1"></a>gpoints <span class="ot">&lt;-</span> <span class="fu">createGiottoPoints</span>(xen_transcripts)</span>
<span id="cb332-3"><a href="spatial-multi-modal-analysis.html#cb332-3" tabindex="-1"></a>Xen_obj <span class="ot">&lt;-</span><span class="fu">createGiottoObjectSubcellular</span>(<span class="at">gpoints =</span> <span class="fu">list</span>(<span class="st">&#39;rna&#39;</span> <span class="ot">=</span> gpoints),</span>
<span id="cb332-4"><a href="spatial-multi-modal-analysis.html#cb332-4" tabindex="-1"></a>                                        <span class="at">gpolygons =</span> <span class="fu">list</span>(<span class="st">&#39;visium&#39;</span> <span class="ot">=</span> pseudo_visium_poly))</span></code></pre></div>
<p>Get gene expression information by overlapping polygon to points</p>
<div class="sourceCode" id="cb333"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb333-1"><a href="spatial-multi-modal-analysis.html#cb333-1" tabindex="-1"></a>Xen_obj <span class="ot">&lt;-</span> <span class="fu">calculateOverlap</span>(Xen_obj,</span>
<span id="cb333-2"><a href="spatial-multi-modal-analysis.html#cb333-2" tabindex="-1"></a>                           <span class="at">feat_info =</span> <span class="st">&#39;rna&#39;</span>,</span>
<span id="cb333-3"><a href="spatial-multi-modal-analysis.html#cb333-3" tabindex="-1"></a>                           <span class="at">spatial_info =</span> <span class="st">&#39;visium&#39;</span>)</span>
<span id="cb333-4"><a href="spatial-multi-modal-analysis.html#cb333-4" tabindex="-1"></a></span>
<span id="cb333-5"><a href="spatial-multi-modal-analysis.html#cb333-5" tabindex="-1"></a>Xen_obj <span class="ot">&lt;-</span> <span class="fu">overlapToMatrix</span>(<span class="at">x =</span> Xen_obj,</span>
<span id="cb333-6"><a href="spatial-multi-modal-analysis.html#cb333-6" tabindex="-1"></a>                          <span class="at">type =</span> <span class="st">&quot;point&quot;</span>, </span>
<span id="cb333-7"><a href="spatial-multi-modal-analysis.html#cb333-7" tabindex="-1"></a>                          <span class="at">poly_info =</span> <span class="st">&quot;visium&quot;</span>, </span>
<span id="cb333-8"><a href="spatial-multi-modal-analysis.html#cb333-8" tabindex="-1"></a>                          <span class="at">feat_info =</span> <span class="st">&quot;rna&quot;</span>,</span>
<span id="cb333-9"><a href="spatial-multi-modal-analysis.html#cb333-9" tabindex="-1"></a>                          <span class="at">aggr_function =</span> <span class="st">&quot;sum&quot;</span>)</span></code></pre></div>
<p>Manipulate the expression for plotting</p>
<div class="sourceCode" id="cb334"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb334-1"><a href="spatial-multi-modal-analysis.html#cb334-1" tabindex="-1"></a>Xen_obj <span class="ot">&lt;-</span> <span class="fu">filterGiotto</span>(Xen_obj,</span>
<span id="cb334-2"><a href="spatial-multi-modal-analysis.html#cb334-2" tabindex="-1"></a>                        <span class="at">feat_type =</span> <span class="st">&#39;rna&#39;</span>,</span>
<span id="cb334-3"><a href="spatial-multi-modal-analysis.html#cb334-3" tabindex="-1"></a>                        <span class="at">spat_unit =</span> <span class="st">&#39;visium&#39;</span>,</span>
<span id="cb334-4"><a href="spatial-multi-modal-analysis.html#cb334-4" tabindex="-1"></a>                        <span class="at">expression_threshold =</span> <span class="dv">1</span>, </span>
<span id="cb334-5"><a href="spatial-multi-modal-analysis.html#cb334-5" tabindex="-1"></a>                        <span class="at">feat_det_in_min_cells =</span> <span class="dv">0</span>,</span>
<span id="cb334-6"><a href="spatial-multi-modal-analysis.html#cb334-6" tabindex="-1"></a>                        <span class="at">min_det_feats_per_cell =</span> <span class="dv">1</span>)</span>
<span id="cb334-7"><a href="spatial-multi-modal-analysis.html#cb334-7" tabindex="-1"></a>tmp_exprs <span class="ot">&lt;-</span> <span class="fu">getExpression</span>(Xen_obj,</span>
<span id="cb334-8"><a href="spatial-multi-modal-analysis.html#cb334-8" tabindex="-1"></a>                       <span class="at">feat_type =</span> <span class="st">&#39;rna&#39;</span>,</span>
<span id="cb334-9"><a href="spatial-multi-modal-analysis.html#cb334-9" tabindex="-1"></a>                       <span class="at">spat_unit =</span> <span class="st">&#39;visium&#39;</span>,</span>
<span id="cb334-10"><a href="spatial-multi-modal-analysis.html#cb334-10" tabindex="-1"></a>                       <span class="at">output =</span> <span class="st">&#39;matrix&#39;</span>)</span>
<span id="cb334-11"><a href="spatial-multi-modal-analysis.html#cb334-11" tabindex="-1"></a>Xen_obj <span class="ot">&lt;-</span> <span class="fu">setExpression</span>(Xen_obj,</span>
<span id="cb334-12"><a href="spatial-multi-modal-analysis.html#cb334-12" tabindex="-1"></a>                         <span class="at">x =</span> <span class="fu">createExprObj</span>(<span class="fu">log</span>(tmp_exprs<span class="sc">+</span><span class="dv">1</span>)),</span>
<span id="cb334-13"><a href="spatial-multi-modal-analysis.html#cb334-13" tabindex="-1"></a>                         <span class="at">feat_type =</span> <span class="st">&#39;rna&#39;</span>,</span>
<span id="cb334-14"><a href="spatial-multi-modal-analysis.html#cb334-14" tabindex="-1"></a>                         <span class="at">spat_unit =</span> <span class="st">&#39;visium&#39;</span>,</span>
<span id="cb334-15"><a href="spatial-multi-modal-analysis.html#cb334-15" tabindex="-1"></a>                         <span class="at">name =</span> <span class="st">&#39;plot&#39;</span>)</span>
<span id="cb334-16"><a href="spatial-multi-modal-analysis.html#cb334-16" tabindex="-1"></a></span>
<span id="cb334-17"><a href="spatial-multi-modal-analysis.html#cb334-17" tabindex="-1"></a><span class="fu">spatFeatPlot2D</span>(Xen_obj,</span>
<span id="cb334-18"><a href="spatial-multi-modal-analysis.html#cb334-18" tabindex="-1"></a>               <span class="at">point_size =</span> <span class="fl">3.5</span>,</span>
<span id="cb334-19"><a href="spatial-multi-modal-analysis.html#cb334-19" tabindex="-1"></a>               <span class="at">expression_values =</span> <span class="st">&#39;plot&#39;</span>,</span>
<span id="cb334-20"><a href="spatial-multi-modal-analysis.html#cb334-20" tabindex="-1"></a>               <span class="at">show_image =</span> F,</span>
<span id="cb334-21"><a href="spatial-multi-modal-analysis.html#cb334-21" tabindex="-1"></a>               <span class="at">feats =</span> <span class="st">&#39;ERBB2&#39;</span>)</span></code></pre></div>
<p><img src="img/03_session2/PsuedoVisium_spatfeatplot.png" width="80%" style="display: block; margin: auto;" /></p>
<p>Subset the registered Visium and plot same gene</p>
<div class="sourceCode" id="cb335"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb335-1"><a href="spatial-multi-modal-analysis.html#cb335-1" tabindex="-1"></a><span class="co">#get the extent of giotto points, xmin, xmax, ymin, ymax</span></span>
<span id="cb335-2"><a href="spatial-multi-modal-analysis.html#cb335-2" tabindex="-1"></a>subset_extent <span class="ot">&lt;-</span> <span class="fu">ext</span>(gpoints<span class="sc">@</span>spatVector)</span>
<span id="cb335-3"><a href="spatial-multi-modal-analysis.html#cb335-3" tabindex="-1"></a>sub_visium <span class="ot">&lt;-</span> <span class="fu">subsetGiottoLocs</span>(V_final,</span>
<span id="cb335-4"><a href="spatial-multi-modal-analysis.html#cb335-4" tabindex="-1"></a>                               <span class="at">x_min =</span> subset_extent[<span class="dv">1</span>],</span>
<span id="cb335-5"><a href="spatial-multi-modal-analysis.html#cb335-5" tabindex="-1"></a>                               <span class="at">x_max =</span> subset_extent[<span class="dv">2</span>],</span>
<span id="cb335-6"><a href="spatial-multi-modal-analysis.html#cb335-6" tabindex="-1"></a>                               <span class="at">y_min =</span> subset_extent[<span class="dv">3</span>],</span>
<span id="cb335-7"><a href="spatial-multi-modal-analysis.html#cb335-7" tabindex="-1"></a>                               <span class="at">y_max =</span> subset_extent[<span class="dv">4</span>])</span>
<span id="cb335-8"><a href="spatial-multi-modal-analysis.html#cb335-8" tabindex="-1"></a><span class="fu">spatFeatPlot2D</span>(sub_visium,</span>
<span id="cb335-9"><a href="spatial-multi-modal-analysis.html#cb335-9" tabindex="-1"></a>               <span class="at">point_size =</span> <span class="dv">2</span>,</span>
<span id="cb335-10"><a href="spatial-multi-modal-analysis.html#cb335-10" tabindex="-1"></a>               <span class="at">expression_values =</span> <span class="st">&#39;scaled&#39;</span>,</span>
<span id="cb335-11"><a href="spatial-multi-modal-analysis.html#cb335-11" tabindex="-1"></a>               <span class="at">show_image =</span> F,</span>
<span id="cb335-12"><a href="spatial-multi-modal-analysis.html#cb335-12" tabindex="-1"></a>               <span class="at">feats =</span> <span class="st">&#39;ERBB2&#39;</span>)</span></code></pre></div>
<p><img src="img/03_session2/Visium_spatfeatplot.png" width="80%" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="register-post-xenium-he-and-if-image" class="section level3 hasAnchor" number="13.6.4">
<h3><span class="header-section-number">13.6.4</span> Register post-Xenium H&amp;E and IF image<a href="spatial-multi-modal-analysis.html#register-post-xenium-he-and-if-image" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>For Xenium instrument output, Giotto provide a convenience function to load the output from the Xenium ranger output. Note that 10X created the affine image alignment file by applying rotation, scale at (0,0) of the top left corner and translation last. Thus, it will look different than the affine matrix created from landmarks above. In this example, we used a 0.05X compressed ometiff and the alignment file is also create by first rescale at 20X, then apply the affine matrix provided by 10X Genomics.</p>
<div class="sourceCode" id="cb336"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb336-1"><a href="spatial-multi-modal-analysis.html#cb336-1" tabindex="-1"></a>HE_xen <span class="ot">&lt;-</span> <span class="fu">read10xAffineImage</span>(<span class="at">file =</span> <span class="fu">paste0</span>(Xenium_dir, <span class="st">&quot;/HE_ome_compressed.tiff&quot;</span>),</span>
<span id="cb336-2"><a href="spatial-multi-modal-analysis.html#cb336-2" tabindex="-1"></a>                           <span class="at">imagealignment_path =</span> <span class="fu">paste0</span>(Xenium_dir,<span class="st">&quot;/Xenium_he_imagealignment.csv&quot;</span>),</span>
<span id="cb336-3"><a href="spatial-multi-modal-analysis.html#cb336-3" tabindex="-1"></a>                           <span class="at">micron =</span> <span class="fl">0.2125</span>)</span>
<span id="cb336-4"><a href="spatial-multi-modal-analysis.html#cb336-4" tabindex="-1"></a><span class="fu">plot</span>(HE_xen)</span></code></pre></div>
<p><img src="img/03_session2/HE_register_1.png" width="80%" style="display: block; margin: auto;" /></p>
<p>The image is still on the top left corner, so we flip the image to make it align with the target coordinate system. We can also save the transformed image raster by re-sample all pixel from the original image, and write it to a file on disk for future use.</p>
<div class="sourceCode" id="cb337"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb337-1"><a href="spatial-multi-modal-analysis.html#cb337-1" tabindex="-1"></a>HE_xen <span class="ot">&lt;-</span> HE_xen <span class="sc">|&gt;</span> <span class="fu">flip</span>(<span class="at">direction =</span> <span class="st">&quot;vertical&quot;</span>)</span>
<span id="cb337-2"><a href="spatial-multi-modal-analysis.html#cb337-2" tabindex="-1"></a>gimg_rast <span class="ot">&lt;-</span> HE_xen<span class="sc">@</span>funs<span class="sc">$</span><span class="fu">realize_magick</span>(<span class="at">size =</span> <span class="fu">prod</span>(<span class="fu">dim</span>(HE_xen)))</span>
<span id="cb337-3"><a href="spatial-multi-modal-analysis.html#cb337-3" tabindex="-1"></a><span class="fu">plot</span>(gimg_rast)</span>
<span id="cb337-4"><a href="spatial-multi-modal-analysis.html#cb337-4" tabindex="-1"></a><span class="co">#terra::writeRaster(gimg_rast@raster_object, filename = output,  gdal = &quot;COG&quot; # save as GeoTIFF with extent info)</span></span></code></pre></div>
<p><img src="img/03_session2/HE_register_2.png" width="80%" style="display: block; margin: auto;" /></p>
<p>Now we can check the registration results. GiottoVisuals provide a function to plot a giottoLargeImage to a ggplot object in order to plot additional layers of ggplots</p>
<div class="sourceCode" id="cb338"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb338-1"><a href="spatial-multi-modal-analysis.html#cb338-1" tabindex="-1"></a>gg <span class="ot">&lt;-</span> ggplot2<span class="sc">::</span><span class="fu">ggplot</span>()</span>
<span id="cb338-2"><a href="spatial-multi-modal-analysis.html#cb338-2" tabindex="-1"></a>pl <span class="ot">&lt;-</span> GiottoVisuals<span class="sc">::</span><span class="fu">gg_annotation_raster</span>(gg,gimg_rast)</span>
<span id="cb338-3"><a href="spatial-multi-modal-analysis.html#cb338-3" tabindex="-1"></a>pl <span class="sc">+</span> ggplot2<span class="sc">::</span><span class="fu">geom_smooth</span>() <span class="sc">+</span> </span>
<span id="cb338-4"><a href="spatial-multi-modal-analysis.html#cb338-4" tabindex="-1"></a>     ggplot2<span class="sc">::</span><span class="fu">geom_point</span>(<span class="at">data =</span> xen_cell_df, ggplot2<span class="sc">::</span><span class="fu">aes</span>(<span class="at">x =</span> x_centroid , <span class="at">y =</span> y_centroid),<span class="at">size =</span> <span class="fl">1e-150</span>,,<span class="at">color =</span> <span class="st">&#39;orange&#39;</span>) <span class="sc">+</span> ggplot2<span class="sc">::</span><span class="fu">theme_classic</span>()</span></code></pre></div>
<p><img src="img/03_session2/HE_register_3.png" width="80%" style="display: block; margin: auto;" /></p>
<div id="add-registered-image-information-and-compare-rna-vs-protein-expression" class="section level4 hasAnchor" number="13.6.4.1">
<h4><span class="header-section-number">13.6.4.1</span> Add registered image information and compare RNA vs protein expression<a href="spatial-multi-modal-analysis.html#add-registered-image-information-and-compare-rna-vs-protein-expression" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>With the strategy described above, affine transformed image can be saved and used for quantitive analysis. Here, we can use the same strategy as dealing with spatial proteomics data for IF</p>
<div class="sourceCode" id="cb339"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb339-1"><a href="spatial-multi-modal-analysis.html#cb339-1" tabindex="-1"></a>CD20_gimg <span class="ot">&lt;-</span> <span class="fu">createGiottoLargeImage</span>(<span class="fu">paste0</span>(Xenium_dir,<span class="st">&#39;/CD20_registered.tiff&#39;</span>), <span class="at">use_rast_ext =</span> T,<span class="at">name =</span> <span class="st">&#39;CD20&#39;</span>)</span>
<span id="cb339-2"><a href="spatial-multi-modal-analysis.html#cb339-2" tabindex="-1"></a>HER2_gimg <span class="ot">&lt;-</span> <span class="fu">createGiottoLargeImage</span>(<span class="fu">paste0</span>(Xenium_dir,<span class="st">&#39;/HER2_registered.tiff&#39;</span>), <span class="at">use_rast_ext =</span> T,<span class="at">name =</span> <span class="st">&#39;HER2&#39;</span>)</span>
<span id="cb339-3"><a href="spatial-multi-modal-analysis.html#cb339-3" tabindex="-1"></a>Xen_obj <span class="ot">&lt;-</span> <span class="fu">addGiottoLargeImage</span>(<span class="at">gobject =</span> Xen_obj,</span>
<span id="cb339-4"><a href="spatial-multi-modal-analysis.html#cb339-4" tabindex="-1"></a>                               <span class="at">largeImages =</span> <span class="fu">list</span>(<span class="st">&#39;CD20&#39;</span> <span class="ot">=</span> CD20_gimg,<span class="st">&#39;HER2&#39;</span> <span class="ot">=</span> HER2_gimg))</span></code></pre></div>
<p>Get the cell polygons, as Xenium and IF are both subcellular resolution</p>
<div class="sourceCode" id="cb340"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb340-1"><a href="spatial-multi-modal-analysis.html#cb340-1" tabindex="-1"></a>cellpoly_dt <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="fu">paste0</span>(Xenium_dir,<span class="st">&#39;/cell_boundaries.csv.gz&#39;</span>))</span>
<span id="cb340-2"><a href="spatial-multi-modal-analysis.html#cb340-2" tabindex="-1"></a><span class="fu">colnames</span>(cellpoly_dt) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&#39;poly_ID&#39;</span>,<span class="st">&#39;x&#39;</span>,<span class="st">&#39;y&#39;</span>)</span>
<span id="cb340-3"><a href="spatial-multi-modal-analysis.html#cb340-3" tabindex="-1"></a>cellpoly <span class="ot">&lt;-</span> <span class="fu">createGiottoPolygonsFromDfr</span>(cellpoly_dt)</span>
<span id="cb340-4"><a href="spatial-multi-modal-analysis.html#cb340-4" tabindex="-1"></a>Xen_obj <span class="ot">&lt;-</span> <span class="fu">addGiottoPolygons</span>(Xen_obj,<span class="at">gpolygons =</span> <span class="fu">list</span>(<span class="st">&#39;cell&#39;</span> <span class="ot">=</span> cellpoly))</span></code></pre></div>
<p>Compute the gene expression matrix by overlay the cell polygons and giotto points.</p>
<div class="sourceCode" id="cb341"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb341-1"><a href="spatial-multi-modal-analysis.html#cb341-1" tabindex="-1"></a>Xen_obj <span class="ot">&lt;-</span> <span class="fu">calculateOverlap</span>(Xen_obj,</span>
<span id="cb341-2"><a href="spatial-multi-modal-analysis.html#cb341-2" tabindex="-1"></a>                           <span class="at">feat_info =</span> <span class="st">&#39;rna&#39;</span>,</span>
<span id="cb341-3"><a href="spatial-multi-modal-analysis.html#cb341-3" tabindex="-1"></a>                           <span class="at">spatial_info =</span> <span class="st">&#39;cell&#39;</span>)</span>
<span id="cb341-4"><a href="spatial-multi-modal-analysis.html#cb341-4" tabindex="-1"></a></span>
<span id="cb341-5"><a href="spatial-multi-modal-analysis.html#cb341-5" tabindex="-1"></a>Xen_obj <span class="ot">&lt;-</span> <span class="fu">overlapToMatrix</span>(<span class="at">x =</span> Xen_obj,</span>
<span id="cb341-6"><a href="spatial-multi-modal-analysis.html#cb341-6" tabindex="-1"></a>                          <span class="at">type =</span> <span class="st">&quot;point&quot;</span>, </span>
<span id="cb341-7"><a href="spatial-multi-modal-analysis.html#cb341-7" tabindex="-1"></a>                          <span class="at">poly_info =</span> <span class="st">&quot;cell&quot;</span>, </span>
<span id="cb341-8"><a href="spatial-multi-modal-analysis.html#cb341-8" tabindex="-1"></a>                          <span class="at">feat_info =</span> <span class="st">&quot;rna&quot;</span>,</span>
<span id="cb341-9"><a href="spatial-multi-modal-analysis.html#cb341-9" tabindex="-1"></a>                          <span class="at">aggr_function =</span> <span class="st">&quot;sum&quot;</span>)</span>
<span id="cb341-10"><a href="spatial-multi-modal-analysis.html#cb341-10" tabindex="-1"></a>tmp_exprs <span class="ot">&lt;-</span> <span class="fu">getExpression</span>(Xen_obj,</span>
<span id="cb341-11"><a href="spatial-multi-modal-analysis.html#cb341-11" tabindex="-1"></a>                       <span class="at">feat_type =</span> <span class="st">&#39;rna&#39;</span>,</span>
<span id="cb341-12"><a href="spatial-multi-modal-analysis.html#cb341-12" tabindex="-1"></a>                       <span class="at">spat_unit =</span> <span class="st">&#39;cell&#39;</span>,</span>
<span id="cb341-13"><a href="spatial-multi-modal-analysis.html#cb341-13" tabindex="-1"></a>                       <span class="at">output =</span> <span class="st">&#39;matrix&#39;</span>)</span>
<span id="cb341-14"><a href="spatial-multi-modal-analysis.html#cb341-14" tabindex="-1"></a>Xen_obj <span class="ot">&lt;-</span> <span class="fu">setExpression</span>(Xen_obj,</span>
<span id="cb341-15"><a href="spatial-multi-modal-analysis.html#cb341-15" tabindex="-1"></a>                         <span class="at">x =</span> <span class="fu">createExprObj</span>(<span class="fu">log</span>(tmp_exprs<span class="sc">+</span><span class="dv">1</span>)),</span>
<span id="cb341-16"><a href="spatial-multi-modal-analysis.html#cb341-16" tabindex="-1"></a>                         <span class="at">feat_type =</span> <span class="st">&#39;rna&#39;</span>,</span>
<span id="cb341-17"><a href="spatial-multi-modal-analysis.html#cb341-17" tabindex="-1"></a>                         <span class="at">spat_unit =</span> <span class="st">&#39;cell&#39;</span>,</span>
<span id="cb341-18"><a href="spatial-multi-modal-analysis.html#cb341-18" tabindex="-1"></a>                         <span class="at">name =</span> <span class="st">&#39;plot&#39;</span>)</span>
<span id="cb341-19"><a href="spatial-multi-modal-analysis.html#cb341-19" tabindex="-1"></a><span class="fu">spatFeatPlot2D</span>(Xen_obj,</span>
<span id="cb341-20"><a href="spatial-multi-modal-analysis.html#cb341-20" tabindex="-1"></a>               <span class="at">feat_type =</span> <span class="st">&#39;rna&#39;</span>,</span>
<span id="cb341-21"><a href="spatial-multi-modal-analysis.html#cb341-21" tabindex="-1"></a>               <span class="at">expression_values =</span> <span class="st">&#39;plot&#39;</span>,</span>
<span id="cb341-22"><a href="spatial-multi-modal-analysis.html#cb341-22" tabindex="-1"></a>               <span class="at">spat_unit =</span> <span class="st">&#39;cell&#39;</span>,</span>
<span id="cb341-23"><a href="spatial-multi-modal-analysis.html#cb341-23" tabindex="-1"></a>               <span class="at">feats =</span> <span class="st">&#39;ERBB2&#39;</span>,</span>
<span id="cb341-24"><a href="spatial-multi-modal-analysis.html#cb341-24" tabindex="-1"></a>               <span class="at">point_size =</span> <span class="fl">0.05</span>)</span></code></pre></div>
<p><img src="img/03_session2/ERBB2_spatplot.png" width="80%" style="display: block; margin: auto;" /></p>
<p>Now we overlay the HER2 expression from the raster image with the cell polygons.</p>
<div class="sourceCode" id="cb342"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb342-1"><a href="spatial-multi-modal-analysis.html#cb342-1" tabindex="-1"></a>Xen_obj <span class="ot">&lt;-</span> <span class="fu">calculateOverlap</span>(Xen_obj,</span>
<span id="cb342-2"><a href="spatial-multi-modal-analysis.html#cb342-2" tabindex="-1"></a>                            <span class="at">spatial_info =</span> <span class="st">&#39;cell&#39;</span>,</span>
<span id="cb342-3"><a href="spatial-multi-modal-analysis.html#cb342-3" tabindex="-1"></a>                            <span class="at">image_names =</span> <span class="fu">c</span>(<span class="st">&#39;HER2&#39;</span>,<span class="st">&#39;CD20&#39;</span>))</span>
<span id="cb342-4"><a href="spatial-multi-modal-analysis.html#cb342-4" tabindex="-1"></a></span>
<span id="cb342-5"><a href="spatial-multi-modal-analysis.html#cb342-5" tabindex="-1"></a>Xen_obj <span class="ot">&lt;-</span> <span class="fu">overlapToMatrix</span>(<span class="at">x =</span> Xen_obj,</span>
<span id="cb342-6"><a href="spatial-multi-modal-analysis.html#cb342-6" tabindex="-1"></a>                          <span class="at">type =</span> <span class="st">&quot;intensity&quot;</span>, </span>
<span id="cb342-7"><a href="spatial-multi-modal-analysis.html#cb342-7" tabindex="-1"></a>                          <span class="at">poly_info =</span> <span class="st">&quot;cell&quot;</span>, </span>
<span id="cb342-8"><a href="spatial-multi-modal-analysis.html#cb342-8" tabindex="-1"></a>                          <span class="at">feat_info =</span> <span class="st">&quot;protein&quot;</span>,</span>
<span id="cb342-9"><a href="spatial-multi-modal-analysis.html#cb342-9" tabindex="-1"></a>                          <span class="at">aggr_function =</span> <span class="st">&quot;sum&quot;</span>)</span>
<span id="cb342-10"><a href="spatial-multi-modal-analysis.html#cb342-10" tabindex="-1"></a></span>
<span id="cb342-11"><a href="spatial-multi-modal-analysis.html#cb342-11" tabindex="-1"></a>tmp_exprs <span class="ot">&lt;-</span> <span class="fu">getExpression</span>(Xen_obj,</span>
<span id="cb342-12"><a href="spatial-multi-modal-analysis.html#cb342-12" tabindex="-1"></a>                       <span class="at">feat_type =</span> <span class="st">&#39;protein&#39;</span>,</span>
<span id="cb342-13"><a href="spatial-multi-modal-analysis.html#cb342-13" tabindex="-1"></a>                       <span class="at">spat_unit =</span> <span class="st">&#39;cell&#39;</span>,</span>
<span id="cb342-14"><a href="spatial-multi-modal-analysis.html#cb342-14" tabindex="-1"></a>                       <span class="at">output =</span> <span class="st">&#39;matrix&#39;</span>)</span>
<span id="cb342-15"><a href="spatial-multi-modal-analysis.html#cb342-15" tabindex="-1"></a>Xen_obj <span class="ot">&lt;-</span> <span class="fu">setExpression</span>(Xen_obj,</span>
<span id="cb342-16"><a href="spatial-multi-modal-analysis.html#cb342-16" tabindex="-1"></a>                         <span class="at">x =</span> <span class="fu">createExprObj</span>(<span class="fu">log</span>(tmp_exprs<span class="sc">+</span><span class="dv">1</span>)),</span>
<span id="cb342-17"><a href="spatial-multi-modal-analysis.html#cb342-17" tabindex="-1"></a>                         <span class="at">feat_type =</span> <span class="st">&#39;protein&#39;</span>,</span>
<span id="cb342-18"><a href="spatial-multi-modal-analysis.html#cb342-18" tabindex="-1"></a>                         <span class="at">spat_unit =</span> <span class="st">&#39;cell&#39;</span>,</span>
<span id="cb342-19"><a href="spatial-multi-modal-analysis.html#cb342-19" tabindex="-1"></a>                         <span class="at">name =</span> <span class="st">&#39;plot&#39;</span>)</span>
<span id="cb342-20"><a href="spatial-multi-modal-analysis.html#cb342-20" tabindex="-1"></a><span class="fu">spatFeatPlot2D</span>(Xen_obj,</span>
<span id="cb342-21"><a href="spatial-multi-modal-analysis.html#cb342-21" tabindex="-1"></a>               <span class="at">feat_type =</span> <span class="st">&#39;protein&#39;</span>,</span>
<span id="cb342-22"><a href="spatial-multi-modal-analysis.html#cb342-22" tabindex="-1"></a>               <span class="at">expression_values =</span> <span class="st">&#39;plot&#39;</span>,</span>
<span id="cb342-23"><a href="spatial-multi-modal-analysis.html#cb342-23" tabindex="-1"></a>               <span class="at">spat_unit =</span> <span class="st">&#39;cell&#39;</span>,</span>
<span id="cb342-24"><a href="spatial-multi-modal-analysis.html#cb342-24" tabindex="-1"></a>               <span class="at">feats =</span> <span class="st">&#39;HER2&#39;</span>,</span>
<span id="cb342-25"><a href="spatial-multi-modal-analysis.html#cb342-25" tabindex="-1"></a>               <span class="at">point_size =</span> <span class="fl">0.05</span>)</span></code></pre></div>
<p><img src="img/03_session2/HER2_spatplot.png" width="80%" style="display: block; margin: auto;" />
We can also overlay the protein expression to Visium spots</p>
<div class="sourceCode" id="cb343"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb343-1"><a href="spatial-multi-modal-analysis.html#cb343-1" tabindex="-1"></a>Xen_obj <span class="ot">&lt;-</span> <span class="fu">calculateOverlap</span>(Xen_obj,</span>
<span id="cb343-2"><a href="spatial-multi-modal-analysis.html#cb343-2" tabindex="-1"></a>                            <span class="at">spatial_info =</span> <span class="st">&#39;visium&#39;</span>,</span>
<span id="cb343-3"><a href="spatial-multi-modal-analysis.html#cb343-3" tabindex="-1"></a>                            <span class="at">image_names =</span> <span class="fu">c</span>(<span class="st">&#39;HER2&#39;</span>,<span class="st">&#39;CD20&#39;</span>))</span>
<span id="cb343-4"><a href="spatial-multi-modal-analysis.html#cb343-4" tabindex="-1"></a>Xen_obj <span class="ot">&lt;-</span> <span class="fu">overlapToMatrix</span>(<span class="at">x =</span> Xen_obj,</span>
<span id="cb343-5"><a href="spatial-multi-modal-analysis.html#cb343-5" tabindex="-1"></a>                          <span class="at">type =</span> <span class="st">&quot;intensity&quot;</span>, </span>
<span id="cb343-6"><a href="spatial-multi-modal-analysis.html#cb343-6" tabindex="-1"></a>                          <span class="at">poly_info =</span> <span class="st">&quot;visium&quot;</span>, </span>
<span id="cb343-7"><a href="spatial-multi-modal-analysis.html#cb343-7" tabindex="-1"></a>                          <span class="at">feat_info =</span> <span class="st">&quot;protein&quot;</span>,</span>
<span id="cb343-8"><a href="spatial-multi-modal-analysis.html#cb343-8" tabindex="-1"></a>                          <span class="at">aggr_function =</span> <span class="st">&quot;sum&quot;</span>)</span>
<span id="cb343-9"><a href="spatial-multi-modal-analysis.html#cb343-9" tabindex="-1"></a></span>
<span id="cb343-10"><a href="spatial-multi-modal-analysis.html#cb343-10" tabindex="-1"></a>Xen_obj <span class="ot">&lt;-</span> <span class="fu">filterGiotto</span>(Xen_obj,</span>
<span id="cb343-11"><a href="spatial-multi-modal-analysis.html#cb343-11" tabindex="-1"></a>                        <span class="at">feat_type =</span> <span class="st">&#39;protein&#39;</span>,</span>
<span id="cb343-12"><a href="spatial-multi-modal-analysis.html#cb343-12" tabindex="-1"></a>                        <span class="at">spat_unit =</span> <span class="st">&#39;visium&#39;</span>,</span>
<span id="cb343-13"><a href="spatial-multi-modal-analysis.html#cb343-13" tabindex="-1"></a>                        <span class="at">expression_threshold =</span> <span class="dv">1</span>, </span>
<span id="cb343-14"><a href="spatial-multi-modal-analysis.html#cb343-14" tabindex="-1"></a>                        <span class="at">feat_det_in_min_cells =</span> <span class="dv">0</span>,</span>
<span id="cb343-15"><a href="spatial-multi-modal-analysis.html#cb343-15" tabindex="-1"></a>                        <span class="at">min_det_feats_per_cell =</span> <span class="dv">1</span>)</span>
<span id="cb343-16"><a href="spatial-multi-modal-analysis.html#cb343-16" tabindex="-1"></a></span>
<span id="cb343-17"><a href="spatial-multi-modal-analysis.html#cb343-17" tabindex="-1"></a>tmp_exprs <span class="ot">&lt;-</span> <span class="fu">getExpression</span>(Xen_obj,</span>
<span id="cb343-18"><a href="spatial-multi-modal-analysis.html#cb343-18" tabindex="-1"></a>                       <span class="at">feat_type =</span> <span class="st">&#39;protein&#39;</span>,</span>
<span id="cb343-19"><a href="spatial-multi-modal-analysis.html#cb343-19" tabindex="-1"></a>                       <span class="at">spat_unit =</span> <span class="st">&#39;visium&#39;</span>,</span>
<span id="cb343-20"><a href="spatial-multi-modal-analysis.html#cb343-20" tabindex="-1"></a>                       <span class="at">output =</span> <span class="st">&#39;matrix&#39;</span>)</span>
<span id="cb343-21"><a href="spatial-multi-modal-analysis.html#cb343-21" tabindex="-1"></a>Xen_obj <span class="ot">&lt;-</span> <span class="fu">setExpression</span>(Xen_obj,</span>
<span id="cb343-22"><a href="spatial-multi-modal-analysis.html#cb343-22" tabindex="-1"></a>                         <span class="at">x =</span> <span class="fu">createExprObj</span>(<span class="fu">log</span>(tmp_exprs<span class="sc">+</span><span class="dv">1</span>)),</span>
<span id="cb343-23"><a href="spatial-multi-modal-analysis.html#cb343-23" tabindex="-1"></a>                         <span class="at">feat_type =</span> <span class="st">&#39;protein&#39;</span>,</span>
<span id="cb343-24"><a href="spatial-multi-modal-analysis.html#cb343-24" tabindex="-1"></a>                         <span class="at">spat_unit =</span> <span class="st">&#39;visium&#39;</span>,</span>
<span id="cb343-25"><a href="spatial-multi-modal-analysis.html#cb343-25" tabindex="-1"></a>                         <span class="at">name =</span> <span class="st">&#39;plot&#39;</span>)</span>
<span id="cb343-26"><a href="spatial-multi-modal-analysis.html#cb343-26" tabindex="-1"></a><span class="fu">spatFeatPlot2D</span>(Xen_obj,</span>
<span id="cb343-27"><a href="spatial-multi-modal-analysis.html#cb343-27" tabindex="-1"></a>               <span class="at">feat_type =</span> <span class="st">&#39;protein&#39;</span>,</span>
<span id="cb343-28"><a href="spatial-multi-modal-analysis.html#cb343-28" tabindex="-1"></a>               <span class="at">expression_values =</span> <span class="st">&#39;plot&#39;</span>,</span>
<span id="cb343-29"><a href="spatial-multi-modal-analysis.html#cb343-29" tabindex="-1"></a>               <span class="at">spat_unit =</span> <span class="st">&#39;visium&#39;</span>,</span>
<span id="cb343-30"><a href="spatial-multi-modal-analysis.html#cb343-30" tabindex="-1"></a>               <span class="at">feats =</span> <span class="st">&#39;HER2&#39;</span>,</span>
<span id="cb343-31"><a href="spatial-multi-modal-analysis.html#cb343-31" tabindex="-1"></a>               <span class="at">point_size =</span> <span class="dv">2</span>)</span></code></pre></div>
<p><img src="img/03_session2/HER2_visium.png" width="80%" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="automatic-alignment-via-sift-feature-descriptor-matching-and-affine-transformation" class="section level3 hasAnchor" number="13.6.5">
<h3><span class="header-section-number">13.6.5</span> Automatic alignment via SIFT feature descriptor matching and affine transformation<a href="spatial-multi-modal-analysis.html#automatic-alignment-via-sift-feature-descriptor-matching-and-affine-transformation" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Pin landmarks or use compounded affine transforms to register image usually provides initial registration results. However, recording landmarks or manually combine transformations require a lot of manual effort. It will require too much effort when having a large amount of images to register. As long as accurate landmarks are provided, registration will be easy to automatically perform. Here we provide a wrapper function of Scale invariant feature transform(SIFT). SIFT will first identify the extreme points in different scale spaces from paired images, then use a brutal force way to match the points. The matched points can then be used to estimate the transform and warp the image. The major drawback is once the dimension of the image become bigger, the computing time will increase exponentially.</p>
<p>Here, we provide an example of two compressed images to show the automatic alignment pipeline.</p>
<div class="sourceCode" id="cb344"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb344-1"><a href="spatial-multi-modal-analysis.html#cb344-1" tabindex="-1"></a>HE <span class="ot">&lt;-</span> <span class="fu">createGiottoLargeImage</span>(<span class="fu">paste0</span>(Xenium_dir,<span class="st">&#39;/mini_HE.png&#39;</span>),<span class="at">negative_y =</span> F)</span>
<span id="cb344-2"><a href="spatial-multi-modal-analysis.html#cb344-2" tabindex="-1"></a><span class="fu">plot</span>(HE)</span></code></pre></div>
<p><img src="img/03_session2/mini_HE.png" width="80%" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb345"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb345-1"><a href="spatial-multi-modal-analysis.html#cb345-1" tabindex="-1"></a>IF <span class="ot">&lt;-</span> <span class="fu">createGiottoLargeImage</span>(<span class="fu">paste0</span>(Xenium_dir,<span class="st">&#39;/mini_IF.tif&#39;</span>),<span class="at">negative_y =</span> F,<span class="at">flip_horizontal =</span> T)</span>
<span id="cb345-2"><a href="spatial-multi-modal-analysis.html#cb345-2" tabindex="-1"></a>terra<span class="sc">::</span><span class="fu">plotRGB</span>(IF<span class="sc">@</span>raster_object,<span class="at">r=</span><span class="dv">1</span>, <span class="at">g=</span><span class="dv">2</span>, <span class="at">b=</span><span class="dv">3</span>,, <span class="at">stretch=</span><span class="st">&quot;lin&quot;</span>)</span></code></pre></div>
<p><img src="img/03_session2/mini_IF.png" width="80%" style="display: block; margin: auto;" /></p>
<p>Now, we can use the automated transformation pipeline. Note that we will start with a path to the images, run the preprocessImageToMatrix() first to meet the requirement of estimateAutomatedImageRegistrationWithSIFT() function. The images will be preprocessed to gray scale. And for that purpose, we use the DAPI channel from the miniIF, and set invert = T for mini HE as HE image so that grayscle HE will have higher value for high intensity pixels. The function will output an estimation of the transform.</p>
<div class="sourceCode" id="cb346"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb346-1"><a href="spatial-multi-modal-analysis.html#cb346-1" tabindex="-1"></a>estimation <span class="ot">&lt;-</span> <span class="fu">estimateAutomatedImageRegistrationWithSIFT</span>(<span class="at">x =</span> <span class="fu">preprocessImageToMatrix</span>(<span class="fu">paste0</span>(Xenium_dir,<span class="st">&#39;/mini_IF.tif&#39;</span>),</span>
<span id="cb346-2"><a href="spatial-multi-modal-analysis.html#cb346-2" tabindex="-1"></a>                                                                                     <span class="at">flip_horizontal =</span> T,</span>
<span id="cb346-3"><a href="spatial-multi-modal-analysis.html#cb346-3" tabindex="-1"></a>                                                                                     <span class="at">use_single_channel =</span> T,</span>
<span id="cb346-4"><a href="spatial-multi-modal-analysis.html#cb346-4" tabindex="-1"></a>                                                                                     <span class="at">single_channel_number =</span> <span class="dv">3</span>),</span>
<span id="cb346-5"><a href="spatial-multi-modal-analysis.html#cb346-5" tabindex="-1"></a>                                                         <span class="at">y =</span> <span class="fu">preprocessImageToMatrix</span>(<span class="fu">paste0</span>(Xenium_dir,<span class="st">&#39;/mini_HE.png&#39;</span>),</span>
<span id="cb346-6"><a href="spatial-multi-modal-analysis.html#cb346-6" tabindex="-1"></a>                                                                                     <span class="at">invert =</span> T),</span>
<span id="cb346-7"><a href="spatial-multi-modal-analysis.html#cb346-7" tabindex="-1"></a>                                                         <span class="at">plot_match =</span> T,</span>
<span id="cb346-8"><a href="spatial-multi-modal-analysis.html#cb346-8" tabindex="-1"></a>                                                         <span class="at">max_ratio =</span> <span class="fl">0.5</span>,<span class="at">estimate_fun =</span> <span class="st">&#39;Projective&#39;</span>)</span></code></pre></div>
<p><img src="img/03_session2/SIFT_descriptor_match.png" width="80%" style="display: block; margin: auto;" /></p>
<p>Use the estimation, we can quickly visualize the transformation</p>
<div class="sourceCode" id="cb347"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb347-1"><a href="spatial-multi-modal-analysis.html#cb347-1" tabindex="-1"></a>mtx <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(estimation<span class="sc">$</span>params)</span>
<span id="cb347-2"><a href="spatial-multi-modal-analysis.html#cb347-2" tabindex="-1"></a>transformed <span class="ot">&lt;-</span> <span class="fu">affine</span>(IF, mtx)</span>
<span id="cb347-3"><a href="spatial-multi-modal-analysis.html#cb347-3" tabindex="-1"></a>To_see_overlay <span class="ot">&lt;-</span> transformed<span class="sc">@</span>funs<span class="sc">$</span><span class="fu">realize_magick</span>(<span class="at">size =</span> <span class="fl">2e6</span>)</span>
<span id="cb347-4"><a href="spatial-multi-modal-analysis.html#cb347-4" tabindex="-1"></a><span class="fu">plot</span>(HE)</span>
<span id="cb347-5"><a href="spatial-multi-modal-analysis.html#cb347-5" tabindex="-1"></a><span class="fu">plot</span>(To_see_overlay<span class="sc">@</span>raster_object[[<span class="dv">2</span>]], <span class="at">add=</span><span class="cn">TRUE</span>, <span class="at">alpha=</span><span class="fl">0.5</span>)</span></code></pre></div>
<p><img src="img/03_session2/SIFT_registration_overlay.png" width="80%" style="display: block; margin: auto;" /></p>
</div>
<div id="final-notes" class="section level3 hasAnchor" number="13.6.6">
<h3><span class="header-section-number">13.6.6</span> Final Notes<a href="spatial-multi-modal-analysis.html#final-notes" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Image registration is becoming crucial for spatial multi modal analysis. The methods included here are not the only ways to register images, and either of them may have drawbacks for a good alignment. There are multiple tools coming out for the field with different strategies, including easier landmark detection, deformable transformation as well as matching spatial patterns, etc. Some of them provides transformed images or coordinates that can be directly loaded to Giotto as a multimodal object using a standard pipeline.</p>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="working-with-multiple-samples.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="multi-omics-integration.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
