<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>2 Topics and Schedule: | Workshop: Spatial multi-omics data analysis with Giotto Suite</title>
  <meta name="description" content="2 Topics and Schedule: | Workshop: Spatial multi-omics data analysis with Giotto Suite" />
  <meta name="generator" content="bookdown 0.38 and GitBook 2.6.7" />

  <meta property="og:title" content="2 Topics and Schedule: | Workshop: Spatial multi-omics data analysis with Giotto Suite" />
  <meta property="og:type" content="book" />
  
  
  <meta name="github-repo" content="drieslab/giotto_workshop_2024" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="2 Topics and Schedule: | Workshop: Spatial multi-omics data analysis with Giotto Suite" />
  
  
  

<meta name="author" content="Ruben Dries, Jiaji George Chen, Joselyn Cristina Chávez-Fuentes, Junxiang Xu , Edward Ruiz, Jeff Sheridan, Wen Wang" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="index.html"/>
<link rel="next" href="license.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Instructors</a></li>
<li class="chapter" data-level="2" data-path="topics-and-schedule.html"><a href="topics-and-schedule.html"><i class="fa fa-check"></i><b>2</b> Topics and Schedule:</a></li>
<li class="chapter" data-level="3" data-path="license.html"><a href="license.html"><i class="fa fa-check"></i><b>3</b> License</a></li>
<li class="chapter" data-level="4" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html"><i class="fa fa-check"></i><b>4</b> Spatial omics technologies</a>
<ul>
<li class="chapter" data-level="4.1" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#slides"><i class="fa fa-check"></i><b>4.1</b> Slides</a></li>
<li class="chapter" data-level="4.2" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#spatial-sequencing"><i class="fa fa-check"></i><b>4.2</b> Spatial sequencing</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html"><i class="fa fa-check"></i><b>5</b> Introduction to the Giotto package</a>
<ul>
<li class="chapter" data-level="5.1" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html#slides-1"><i class="fa fa-check"></i><b>5.1</b> Slides</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html"><i class="fa fa-check"></i><b>6</b> Data formatting and Pre-processing</a>
<ul>
<li class="chapter" data-level="6.1" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#creating-a-giotto-object"><i class="fa fa-check"></i><b>6.1</b> Creating a Giotto object</a></li>
<li class="chapter" data-level="6.2" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#spatial-plots"><i class="fa fa-check"></i><b>6.2</b> Spatial plots</a></li>
<li class="chapter" data-level="6.3" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#subsetting"><i class="fa fa-check"></i><b>6.3</b> Subsetting</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="spatial-sequencing-based-technology-visium-part-i.html"><a href="spatial-sequencing-based-technology-visium-part-i.html"><i class="fa fa-check"></i><b>7</b> Spatial sequencing based technology, Visium Part I</a>
<ul>
<li class="chapter" data-level="7.1" data-path="spatial-sequencing-based-technology-visium-part-i.html"><a href="spatial-sequencing-based-technology-visium-part-i.html#introduction-to-spatial-dataset"><i class="fa fa-check"></i><b>7.1</b> Introduction to spatial dataset</a></li>
<li class="chapter" data-level="7.2" data-path="spatial-sequencing-based-technology-visium-part-i.html"><a href="spatial-sequencing-based-technology-visium-part-i.html#create-the-giotto-object"><i class="fa fa-check"></i><b>7.2</b> Create the Giotto object</a></li>
<li class="chapter" data-level="7.3" data-path="spatial-sequencing-based-technology-visium-part-i.html"><a href="spatial-sequencing-based-technology-visium-part-i.html#subset-on-spots-that-were-covered-by-tissue"><i class="fa fa-check"></i><b>7.3</b> Subset on spots that were covered by tissue</a></li>
<li class="chapter" data-level="7.4" data-path="spatial-sequencing-based-technology-visium-part-i.html"><a href="spatial-sequencing-based-technology-visium-part-i.html#quality-control"><i class="fa fa-check"></i><b>7.4</b> Quality control</a></li>
<li class="chapter" data-level="7.5" data-path="spatial-sequencing-based-technology-visium-part-i.html"><a href="spatial-sequencing-based-technology-visium-part-i.html#filtering"><i class="fa fa-check"></i><b>7.5</b> Filtering</a></li>
<li class="chapter" data-level="7.6" data-path="spatial-sequencing-based-technology-visium-part-i.html"><a href="spatial-sequencing-based-technology-visium-part-i.html#normalization"><i class="fa fa-check"></i><b>7.6</b> Normalization</a></li>
<li class="chapter" data-level="7.7" data-path="spatial-sequencing-based-technology-visium-part-i.html"><a href="spatial-sequencing-based-technology-visium-part-i.html#feature-selection"><i class="fa fa-check"></i><b>7.7</b> Feature selection</a>
<ul>
<li class="chapter" data-level="7.7.1" data-path="spatial-sequencing-based-technology-visium-part-i.html"><a href="spatial-sequencing-based-technology-visium-part-i.html#highly-variable-features"><i class="fa fa-check"></i><b>7.7.1</b> Highly Variable Features:</a></li>
<li class="chapter" data-level="7.7.2" data-path="spatial-sequencing-based-technology-visium-part-i.html"><a href="spatial-sequencing-based-technology-visium-part-i.html#spatial-variable-genes"><i class="fa fa-check"></i><b>7.7.2</b> Spatial variable genes</a></li>
</ul></li>
<li class="chapter" data-level="7.8" data-path="spatial-sequencing-based-technology-visium-part-i.html"><a href="spatial-sequencing-based-technology-visium-part-i.html#dimension-reduction"><i class="fa fa-check"></i><b>7.8</b> Dimension Reduction</a></li>
<li class="chapter" data-level="7.9" data-path="spatial-sequencing-based-technology-visium-part-i.html"><a href="spatial-sequencing-based-technology-visium-part-i.html#clustering"><i class="fa fa-check"></i><b>7.9</b> Clustering</a>
<ul>
<li class="chapter" data-level="7.9.1" data-path="spatial-sequencing-based-technology-visium-part-i.html"><a href="spatial-sequencing-based-technology-visium-part-i.html#non-spatial"><i class="fa fa-check"></i><b>7.9.1</b> Non-spatial</a></li>
<li class="chapter" data-level="7.9.2" data-path="spatial-sequencing-based-technology-visium-part-i.html"><a href="spatial-sequencing-based-technology-visium-part-i.html#spatial"><i class="fa fa-check"></i><b>7.9.2</b> Spatial</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="spatial-sequencing-based-technology-visium.html"><a href="spatial-sequencing-based-technology-visium.html"><i class="fa fa-check"></i><b>8</b> Spatial sequencing based technology, Visium</a>
<ul>
<li class="chapter" data-level="8.1" data-path="spatial-sequencing-based-technology-visium.html"><a href="spatial-sequencing-based-technology-visium.html#differential-expression"><i class="fa fa-check"></i><b>8.1</b> Differential expression</a></li>
<li class="chapter" data-level="8.2" data-path="spatial-sequencing-based-technology-visium.html"><a href="spatial-sequencing-based-technology-visium.html#enrichment-deconvolution"><i class="fa fa-check"></i><b>8.2</b> Enrichment &amp; Deconvolution</a></li>
<li class="chapter" data-level="8.3" data-path="spatial-sequencing-based-technology-visium.html"><a href="spatial-sequencing-based-technology-visium.html#interactive-tools"><i class="fa fa-check"></i><b>8.3</b> Interactive tools</a></li>
<li class="chapter" data-level="8.4" data-path="spatial-sequencing-based-technology-visium.html"><a href="spatial-sequencing-based-technology-visium.html#spatial-expression-patterns"><i class="fa fa-check"></i><b>8.4</b> Spatial expression patterns</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Workshop: Spatial multi-omics data analysis with Giotto Suite</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="topics-and-schedule" class="section level1 hasAnchor" number="2">
<h1><span class="header-section-number">2</span> Topics and Schedule:<a href="topics-and-schedule.html#topics-and-schedule" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<ul>
<li>Day 1: Introduction
<ul>
<li>Spatial omics technologies
<ul>
<li>Spatial sequencing</li>
<li>Spatial in situ</li>
<li>Spatial proteomics</li>
<li>spatial other: ATAC-seq, lipidomics, etc</li>
</ul></li>
<li>Introduction to the Giotto package
<ul>
<li>Ecosystem</li>
<li>Installation + python environment</li>
<li>Giotto instructions</li>
</ul></li>
<li>Data formatting and Pre-processing</li>
<li>Creating a Giotto object
<ul>
<li>From matrix + locations</li>
<li>From subcellular raw data (transcripts or images) + polygons</li>
<li>Using convenience functions for popular technologies (Vizgen, Xenium, CosMx, …)</li>
</ul></li>
<li>Spatial plots</li>
<li>Subsetting:
<ul>
<li>Based on IDs</li>
<li>Based on locations</li>
<li>Visualizations</li>
</ul></li>
<li>Introduction to spatial multi-modal dataset (10X Genomics breast cancer) and goal for the next days</li>
<li>Quality control
<ul>
<li>Statistics</li>
</ul></li>
<li>Normalization</li>
<li>Feature selection:
<ul>
<li>Highly Variable Features:
<ul>
<li>loess regression</li>
<li>binned</li>
<li>pearson residuals</li>
</ul></li>
<li>Spatial variable genes</li>
</ul></li>
<li>Dimension Reduction
<ul>
<li>PCA</li>
<li>UMAP/t-SNE</li>
<li>Visualizations</li>
</ul></li>
<li>Clustering
<ul>
<li>Non-spatial
<ul>
<li>k-means</li>
<li>Hierarchical clustering</li>
<li>Leiden/Louvain</li>
</ul></li>
<li>Spatial
<ul>
<li>Spatial variable genes</li>
<li>Spatial co-expression modules</li>
</ul></li>
</ul></li>
</ul></li>
<li>Day 2: Spatial Data Analysis
<ul>
<li>Spatial sequencing based technology: Visium
<ul>
<li>Differential expression</li>
<li>Enrichment &amp; Deconvolution
<ul>
<li>PAGE/Rank</li>
<li>SpatialDWLS</li>
<li>Visualizations</li>
</ul></li>
<li>Interactive tools</li>
<li>Spatial expression patterns
<ul>
<li>Spatial variable genes</li>
<li>Spatial co-expression modules</li>
<li>Spatial HMRF</li>
</ul></li>
</ul></li>
<li>Spatial sequencing based technology: Visium HD
<ul>
<li>Tiling and aggregation</li>
<li>Scalability (duckdb) and projection functions</li>
<li>Spatial expression patterns
<ul>
<li>Spatial co-expression module</li>
</ul></li>
</ul></li>
<li>Spatial in situ technology: Xenium
<ul>
<li>Read in raw data
<ul>
<li>Transcript coordinates</li>
<li>Polygon coordinates</li>
<li>Visualizations</li>
</ul></li>
<li>Overlap txs &amp; polygons
<ul>
<li>Typical aggregated workflow</li>
<li>Feature/molecule specific analysis</li>
<li>Visualizations</li>
</ul></li>
<li>Transcript enrichment GSEA</li>
<li>Spatial location analysis
<ul>
<li>Spatial cell type co-localization analysis</li>
<li>Spatial niche analysis</li>
<li>Spatial niche trajectory analysis</li>
<li>Visualizations</li>
</ul></li>
</ul></li>
<li>Spatial proteomics: multiplex IF
<ul>
<li>Read in raw data
<ul>
<li>Intensity data (IF or any other image)</li>
<li>Polygon coordinates</li>
<li>Visualizations</li>
</ul></li>
<li>Overlap intensity &amp; workflows
<ul>
<li>Typical aggregated workflow</li>
<li>Visualizations</li>
</ul></li>
</ul></li>
</ul></li>
<li>Day 3: Advanced Tutorials
<ul>
<li>Multiple samples
<ul>
<li>Create individual giotto objects</li>
<li>JoinGiottoObjects</li>
<li>Perform Harmony and default workflows</li>
<li>Visualizations</li>
</ul></li>
<li>Spatial multi-modal
<ul>
<li>Co-registration of datasets</li>
<li>Examples in giotto suite manuscript</li>
</ul></li>
<li>Multi-omics integration
<ul>
<li>Example in giotto suite manuscript</li>
</ul></li>
<li>Interoperability w/ other frameworks
<ul>
<li>AnnData/SpatialData</li>
<li>SpatialExperiment</li>
<li>Seurat</li>
</ul></li>
<li>Interoperability w/ isolated tools
<ul>
<li>Spatial niche trajectory analysis</li>
</ul></li>
<li>Interactivity with the R/Spatial ecosystem
<ul>
<li>Kriging</li>
</ul></li>
<li>Contributing to Giotto</li>
</ul></li>
</ul>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="index.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="license.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
