<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>8 Visium Part II | Workshop: Spatial multi-omics data analysis with Giotto Suite</title>
  <meta name="description" content="8 Visium Part II | Workshop: Spatial multi-omics data analysis with Giotto Suite" />
  <meta name="generator" content="bookdown 0.39 and GitBook 2.6.7" />

  <meta property="og:title" content="8 Visium Part II | Workshop: Spatial multi-omics data analysis with Giotto Suite" />
  <meta property="og:type" content="book" />
  
  
  <meta name="github-repo" content="drieslab/giotto_workshop_2024" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="8 Visium Part II | Workshop: Spatial multi-omics data analysis with Giotto Suite" />
  
  
  

<meta name="author" content="Ruben Dries, Jiaji George Chen, Joselyn Cristina Chávez-Fuentes, Junxiang Xu ,Edward Ruiz, Jeff Sheridan, Iqra Amin, Wen Wang" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="visium-part-i.html"/>
<link rel="next" href="visium-hd.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Giotto Suite Workshop 2024</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#instructors"><i class="fa fa-check"></i><b>1.1</b> Instructors</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#topics-and-schedule"><i class="fa fa-check"></i><b>1.2</b> Topics and Schedule:</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#license"><i class="fa fa-check"></i><b>1.3</b> License</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="datasets-packages.html"><a href="datasets-packages.html"><i class="fa fa-check"></i><b>2</b> Datasets &amp; Packages</a>
<ul>
<li class="chapter" data-level="2.1" data-path="datasets-packages.html"><a href="datasets-packages.html#datasets-to-download"><i class="fa fa-check"></i><b>2.1</b> Datasets to download</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="datasets-packages.html"><a href="datasets-packages.html#zenodo-repository"><i class="fa fa-check"></i><b>2.1.1</b> Zenodo repository</a></li>
<li class="chapter" data-level="2.1.2" data-path="datasets-packages.html"><a href="datasets-packages.html#x-genomics-visium-mouse-brain-section-coronal-dataset"><i class="fa fa-check"></i><b>2.1.2</b> 10X Genomics Visium Mouse Brain Section (Coronal) dataset</a></li>
<li class="chapter" data-level="2.1.3" data-path="datasets-packages.html"><a href="datasets-packages.html#x-genomics-visium-hd-ffpe-human-colon-cancer"><i class="fa fa-check"></i><b>2.1.3</b> 10X Genomics Visium HD: FFPE Human Colon Cancer</a></li>
<li class="chapter" data-level="2.1.4" data-path="datasets-packages.html"><a href="datasets-packages.html#x-genomics-multi-modal-dataset"><i class="fa fa-check"></i><b>2.1.4</b> 10X Genomics multi-modal dataset</a></li>
<li class="chapter" data-level="2.1.5" data-path="datasets-packages.html"><a href="datasets-packages.html#x-genomics-multi-omics-visium-cytassist-human-tonsil-dataset"><i class="fa fa-check"></i><b>2.1.5</b> 10X Genomics multi-omics Visium CytAssist Human Tonsil dataset</a></li>
<li class="chapter" data-level="2.1.6" data-path="datasets-packages.html"><a href="datasets-packages.html#x-genomics-human-prostate-cancer-adenocarcinoma-with-invasive-carcinoma-ffpe"><i class="fa fa-check"></i><b>2.1.6</b> 10X Genomics Human Prostate Cancer Adenocarcinoma with Invasive Carcinoma (FFPE)</a></li>
<li class="chapter" data-level="2.1.7" data-path="datasets-packages.html"><a href="datasets-packages.html#x-genomics-normal-human-prostate-ffpe"><i class="fa fa-check"></i><b>2.1.7</b> 10X Genomics Normal Human Prostate (FFPE)</a></li>
<li class="chapter" data-level="2.1.8" data-path="datasets-packages.html"><a href="datasets-packages.html#xenium"><i class="fa fa-check"></i><b>2.1.8</b> Xenium</a></li>
<li class="chapter" data-level="2.1.9" data-path="datasets-packages.html"><a href="datasets-packages.html#merfish-cortex-dataset"><i class="fa fa-check"></i><b>2.1.9</b> MERFISH cortex dataset</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="datasets-packages.html"><a href="datasets-packages.html#needed-packages"><i class="fa fa-check"></i><b>2.2</b> Needed packages</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="datasets-packages.html"><a href="datasets-packages.html#cran-dependencies"><i class="fa fa-check"></i><b>2.2.1</b> CRAN dependencies:</a></li>
<li class="chapter" data-level="2.2.2" data-path="datasets-packages.html"><a href="datasets-packages.html#terra-installation"><i class="fa fa-check"></i><b>2.2.2</b> <em>terra</em> installation</a></li>
<li class="chapter" data-level="2.2.3" data-path="datasets-packages.html"><a href="datasets-packages.html#matrix-installation"><i class="fa fa-check"></i><b>2.2.3</b> <em>Matrix</em> installation</a></li>
<li class="chapter" data-level="2.2.4" data-path="datasets-packages.html"><a href="datasets-packages.html#giotto-installation"><i class="fa fa-check"></i><b>2.2.4</b> <em>Giotto</em> installation</a></li>
<li class="chapter" data-level="2.2.5" data-path="datasets-packages.html"><a href="datasets-packages.html#irlba-install"><i class="fa fa-check"></i><b>2.2.5</b> <em>irlba</em> install</a></li>
<li class="chapter" data-level="2.2.6" data-path="datasets-packages.html"><a href="datasets-packages.html#arrow-install"><i class="fa fa-check"></i><b>2.2.6</b> <em>arrow</em> install</a></li>
<li class="chapter" data-level="2.2.7" data-path="datasets-packages.html"><a href="datasets-packages.html#bioconductor-dependencies"><i class="fa fa-check"></i><b>2.2.7</b> Bioconductor dependencies:</a></li>
<li class="chapter" data-level="2.2.8" data-path="datasets-packages.html"><a href="datasets-packages.html#cran-packages"><i class="fa fa-check"></i><b>2.2.8</b> CRAN packages:</a></li>
<li class="chapter" data-level="2.2.9" data-path="datasets-packages.html"><a href="datasets-packages.html#packages-from-github"><i class="fa fa-check"></i><b>2.2.9</b> Packages from GitHub</a></li>
<li class="chapter" data-level="2.2.10" data-path="datasets-packages.html"><a href="datasets-packages.html#python-environments"><i class="fa fa-check"></i><b>2.2.10</b> Python environments</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html"><i class="fa fa-check"></i><b>3</b> Spatial omics technologies</a>
<ul>
<li class="chapter" data-level="3.1" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#presentation"><i class="fa fa-check"></i><b>3.1</b> Presentation</a></li>
<li class="chapter" data-level="3.2" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#short-summary"><i class="fa fa-check"></i><b>3.2</b> Short summary</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#why-do-we-need-spatial-omics-technologies"><i class="fa fa-check"></i><b>3.2.1</b> Why do we need spatial omics technologies?</a></li>
<li class="chapter" data-level="3.2.2" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#what-is-spatial-omics"><i class="fa fa-check"></i><b>3.2.2</b> What is spatial omics?</a></li>
<li class="chapter" data-level="3.2.3" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#what-are-the-main-spatial-omics-technologies"><i class="fa fa-check"></i><b>3.2.3</b> What are the main spatial omics technologies?</a></li>
<li class="chapter" data-level="3.2.4" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#other-spatial-omics-atac-seq-cuttag-lipidomics-etc"><i class="fa fa-check"></i><b>3.2.4</b> Other Spatial omics: ATAC-seq, CUT&amp;Tag, lipidomics, etc</a></li>
<li class="chapter" data-level="3.2.5" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#what-are-the-different-types-of-spatial-downstream-analyses"><i class="fa fa-check"></i><b>3.2.5</b> What are the different types of spatial downstream analyses?</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html"><i class="fa fa-check"></i><b>4</b> Introduction to the Giotto package</a>
<ul>
<li class="chapter" data-level="4.1" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html#presentation-1"><i class="fa fa-check"></i><b>4.1</b> Presentation</a></li>
<li class="chapter" data-level="4.2" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html#ecosystem"><i class="fa fa-check"></i><b>4.2</b> Ecosystem</a></li>
<li class="chapter" data-level="4.3" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html#installation-python-environment"><i class="fa fa-check"></i><b>4.3</b> Installation + python environment</a></li>
<li class="chapter" data-level="4.4" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html#giotto-instructions"><i class="fa fa-check"></i><b>4.4</b> Giotto instructions</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html"><i class="fa fa-check"></i><b>5</b> Data formatting and Pre-processing</a>
<ul>
<li class="chapter" data-level="5.1" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#data-formats"><i class="fa fa-check"></i><b>5.1</b> Data formats</a></li>
<li class="chapter" data-level="5.2" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#pre-processing"><i class="fa fa-check"></i><b>5.2</b> Pre-processing</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html"><i class="fa fa-check"></i><b>6</b> Creating a Giotto object</a>
<ul>
<li class="chapter" data-level="6.0.1" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#used-packages"><i class="fa fa-check"></i><b>6.0.1</b> Used packages</a></li>
<li class="chapter" data-level="6.1" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#overview"><i class="fa fa-check"></i><b>6.1</b> Overview</a></li>
<li class="chapter" data-level="6.2" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#from-matrix-locations"><i class="fa fa-check"></i><b>6.2</b> From matrix + locations</a></li>
<li class="chapter" data-level="6.3" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#from-subcellular-raw-data-transcripts-or-images-polygons"><i class="fa fa-check"></i><b>6.3</b> From subcellular raw data (transcripts or images) + polygons</a></li>
<li class="chapter" data-level="6.4" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#from-piece-wise"><i class="fa fa-check"></i><b>6.4</b> From piece-wise</a></li>
<li class="chapter" data-level="6.5" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#using-convenience-functions-for-popular-technologies-vizgen-xenium-cosmx"><i class="fa fa-check"></i><b>6.5</b> Using convenience functions for popular technologies (Vizgen, Xenium, CosMx, …)</a></li>
<li class="chapter" data-level="6.6" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#spatial-plots"><i class="fa fa-check"></i><b>6.6</b> Spatial plots</a></li>
<li class="chapter" data-level="6.7" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#subsetting"><i class="fa fa-check"></i><b>6.7</b> Subsetting</a></li>
<li class="chapter" data-level="6.8" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#mini-objects-giottodata"><i class="fa fa-check"></i><b>6.8</b> Mini objects &amp; GiottoData</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="visium-part-i.html"><a href="visium-part-i.html"><i class="fa fa-check"></i><b>7</b> Visium Part I</a>
<ul>
<li class="chapter" data-level="7.1" data-path="visium-part-i.html"><a href="visium-part-i.html#introduction-to-the-spatial-dataset"><i class="fa fa-check"></i><b>7.1</b> Introduction to the spatial dataset</a></li>
<li class="chapter" data-level="7.2" data-path="visium-part-i.html"><a href="visium-part-i.html#download-dataset"><i class="fa fa-check"></i><b>7.2</b> Download dataset</a></li>
<li class="chapter" data-level="7.3" data-path="visium-part-i.html"><a href="visium-part-i.html#create-the-giotto-object"><i class="fa fa-check"></i><b>7.3</b> Create the Giotto object</a></li>
<li class="chapter" data-level="7.4" data-path="visium-part-i.html"><a href="visium-part-i.html#subset-on-spots-that-were-covered-by-tissue"><i class="fa fa-check"></i><b>7.4</b> Subset on spots that were covered by tissue</a></li>
<li class="chapter" data-level="7.5" data-path="visium-part-i.html"><a href="visium-part-i.html#quality-control"><i class="fa fa-check"></i><b>7.5</b> Quality control</a></li>
<li class="chapter" data-level="7.6" data-path="visium-part-i.html"><a href="visium-part-i.html#filtering"><i class="fa fa-check"></i><b>7.6</b> Filtering</a></li>
<li class="chapter" data-level="7.7" data-path="visium-part-i.html"><a href="visium-part-i.html#normalization"><i class="fa fa-check"></i><b>7.7</b> Normalization</a></li>
<li class="chapter" data-level="7.8" data-path="visium-part-i.html"><a href="visium-part-i.html#feature-selection"><i class="fa fa-check"></i><b>7.8</b> Feature selection</a>
<ul>
<li class="chapter" data-level="7.8.1" data-path="visium-part-i.html"><a href="visium-part-i.html#highly-variable-features"><i class="fa fa-check"></i><b>7.8.1</b> Highly Variable Features:</a></li>
</ul></li>
<li class="chapter" data-level="7.9" data-path="visium-part-i.html"><a href="visium-part-i.html#dimension-reduction"><i class="fa fa-check"></i><b>7.9</b> Dimension Reduction</a>
<ul>
<li class="chapter" data-level="7.9.1" data-path="visium-part-i.html"><a href="visium-part-i.html#pca"><i class="fa fa-check"></i><b>7.9.1</b> PCA</a></li>
<li class="chapter" data-level="7.9.2" data-path="visium-part-i.html"><a href="visium-part-i.html#umap"><i class="fa fa-check"></i><b>7.9.2</b> UMAP</a></li>
<li class="chapter" data-level="7.9.3" data-path="visium-part-i.html"><a href="visium-part-i.html#t-sne"><i class="fa fa-check"></i><b>7.9.3</b> t-SNE</a></li>
</ul></li>
<li class="chapter" data-level="7.10" data-path="visium-part-i.html"><a href="visium-part-i.html#clustering"><i class="fa fa-check"></i><b>7.10</b> Clustering</a>
<ul>
<li class="chapter" data-level="7.10.1" data-path="visium-part-i.html"><a href="visium-part-i.html#calculate-leiden-clustering"><i class="fa fa-check"></i><b>7.10.1</b> Calculate Leiden clustering</a></li>
<li class="chapter" data-level="7.10.2" data-path="visium-part-i.html"><a href="visium-part-i.html#calculate-louvain-clustering"><i class="fa fa-check"></i><b>7.10.2</b> Calculate Louvain clustering</a></li>
</ul></li>
<li class="chapter" data-level="7.11" data-path="visium-part-i.html"><a href="visium-part-i.html#save-the-object"><i class="fa fa-check"></i><b>7.11</b> Save the object</a></li>
<li class="chapter" data-level="7.12" data-path="visium-part-i.html"><a href="visium-part-i.html#session-info"><i class="fa fa-check"></i><b>7.12</b> Session info</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="visium-part-ii.html"><a href="visium-part-ii.html"><i class="fa fa-check"></i><b>8</b> Visium Part II</a>
<ul>
<li class="chapter" data-level="8.1" data-path="visium-part-ii.html"><a href="visium-part-ii.html#load-the-object"><i class="fa fa-check"></i><b>8.1</b> Load the object</a></li>
<li class="chapter" data-level="8.2" data-path="visium-part-ii.html"><a href="visium-part-ii.html#differential-expression"><i class="fa fa-check"></i><b>8.2</b> Differential expression</a>
<ul>
<li class="chapter" data-level="8.2.1" data-path="visium-part-ii.html"><a href="visium-part-ii.html#gini-markers"><i class="fa fa-check"></i><b>8.2.1</b> Gini markers</a></li>
<li class="chapter" data-level="8.2.2" data-path="visium-part-ii.html"><a href="visium-part-ii.html#scran-markers"><i class="fa fa-check"></i><b>8.2.2</b> Scran markers</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="visium-part-ii.html"><a href="visium-part-ii.html#enrichment-deconvolution"><i class="fa fa-check"></i><b>8.3</b> Enrichment &amp; Deconvolution</a>
<ul>
<li class="chapter" data-level="8.3.1" data-path="visium-part-ii.html"><a href="visium-part-ii.html#pagerank"><i class="fa fa-check"></i><b>8.3.1</b> PAGE/Rank</a></li>
<li class="chapter" data-level="8.3.2" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatialdwls"><i class="fa fa-check"></i><b>8.3.2</b> SpatialDWLS</a></li>
</ul></li>
<li class="chapter" data-level="8.4" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatial-expression-patterns"><i class="fa fa-check"></i><b>8.4</b> Spatial expression patterns</a>
<ul>
<li class="chapter" data-level="8.4.1" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatial-variable-genes"><i class="fa fa-check"></i><b>8.4.1</b> Spatial variable genes</a></li>
<li class="chapter" data-level="8.4.2" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatial-co-expression-modules"><i class="fa fa-check"></i><b>8.4.2</b> Spatial co-expression modules</a></li>
</ul></li>
<li class="chapter" data-level="8.5" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatially-informed-clusters"><i class="fa fa-check"></i><b>8.5</b> Spatially informed clusters</a></li>
<li class="chapter" data-level="8.6" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatial-domains-hmrf"><i class="fa fa-check"></i><b>8.6</b> Spatial domains HMRF</a></li>
<li class="chapter" data-level="8.7" data-path="visium-part-ii.html"><a href="visium-part-ii.html#interactive-tools"><i class="fa fa-check"></i><b>8.7</b> Interactive tools</a></li>
<li class="chapter" data-level="8.8" data-path="visium-part-ii.html"><a href="visium-part-ii.html#session-info-1"><i class="fa fa-check"></i><b>8.8</b> Session info</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="visium-hd.html"><a href="visium-hd.html"><i class="fa fa-check"></i><b>9</b> Visium HD</a>
<ul>
<li class="chapter" data-level="9.1" data-path="visium-hd.html"><a href="visium-hd.html#objective"><i class="fa fa-check"></i><b>9.1</b> Objective</a></li>
<li class="chapter" data-level="9.2" data-path="visium-hd.html"><a href="visium-hd.html#background"><i class="fa fa-check"></i><b>9.2</b> Background</a>
<ul>
<li class="chapter" data-level="9.2.1" data-path="visium-hd.html"><a href="visium-hd.html#visium-hd-technology"><i class="fa fa-check"></i><b>9.2.1</b> Visium HD Technology</a></li>
<li class="chapter" data-level="9.2.2" data-path="visium-hd.html"><a href="visium-hd.html#colorectal-cancer-sample"><i class="fa fa-check"></i><b>9.2.2</b> Colorectal Cancer Sample</a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="visium-hd.html"><a href="visium-hd.html#data-ingestion"><i class="fa fa-check"></i><b>9.3</b> Data Ingestion</a>
<ul>
<li class="chapter" data-level="9.3.1" data-path="visium-hd.html"><a href="visium-hd.html#visium-hd-output-data-format"><i class="fa fa-check"></i><b>9.3.1</b> Visium HD output data format</a></li>
<li class="chapter" data-level="9.3.2" data-path="visium-hd.html"><a href="visium-hd.html#read-in-raw-data"><i class="fa fa-check"></i><b>9.3.2</b> Read in raw data</a></li>
<li class="chapter" data-level="9.3.3" data-path="visium-hd.html"><a href="visium-hd.html#giotto-visium-hd-convenience-function"><i class="fa fa-check"></i><b>9.3.3</b> Giotto Visium HD convenience function</a></li>
</ul></li>
<li class="chapter" data-level="9.4" data-path="visium-hd.html"><a href="visium-hd.html#tiling-and-aggregation"><i class="fa fa-check"></i><b>9.4</b> Tiling and aggregation</a></li>
<li class="chapter" data-level="9.5" data-path="visium-hd.html"><a href="visium-hd.html#scalability-and-projection-functions"><i class="fa fa-check"></i><b>9.5</b> Scalability and projection functions</a></li>
<li class="chapter" data-level="9.6" data-path="visium-hd.html"><a href="visium-hd.html#spatial-expression-patterns-1"><i class="fa fa-check"></i><b>9.6</b> Spatial expression patterns</a></li>
<li class="chapter" data-level="9.7" data-path="visium-hd.html"><a href="visium-hd.html#spatial-co-expression-modules-1"><i class="fa fa-check"></i><b>9.7</b> Spatial co-expression modules</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="xenium-1.html"><a href="xenium-1.html"><i class="fa fa-check"></i><b>10</b> Xenium</a>
<ul>
<li class="chapter" data-level="10.1" data-path="xenium-1.html"><a href="xenium-1.html#introduction-to-spatial-dataset"><i class="fa fa-check"></i><b>10.1</b> Introduction to spatial dataset</a></li>
<li class="chapter" data-level="10.2" data-path="xenium-1.html"><a href="xenium-1.html#additional-package-requirements"><i class="fa fa-check"></i><b>10.2</b> Additional package requirements</a></li>
<li class="chapter" data-level="10.3" data-path="xenium-1.html"><a href="xenium-1.html#introduce-mini-object"><i class="fa fa-check"></i><b>10.3</b> Introduce mini object</a></li>
<li class="chapter" data-level="10.4" data-path="xenium-1.html"><a href="xenium-1.html#read-in-raw-data-1"><i class="fa fa-check"></i><b>10.4</b> Read in raw data</a></li>
<li class="chapter" data-level="10.5" data-path="xenium-1.html"><a href="xenium-1.html#overlap-txs-polygons"><i class="fa fa-check"></i><b>10.5</b> Overlap txs &amp; polygons</a></li>
<li class="chapter" data-level="10.6" data-path="xenium-1.html"><a href="xenium-1.html#transcript-enrichment-gsea"><i class="fa fa-check"></i><b>10.6</b> Transcript enrichment GSEA</a></li>
<li class="chapter" data-level="10.7" data-path="xenium-1.html"><a href="xenium-1.html#spatial-location-analysis"><i class="fa fa-check"></i><b>10.7</b> Spatial location analysis</a></li>
<li class="chapter" data-level="10.8" data-path="xenium-1.html"><a href="xenium-1.html#visualizations"><i class="fa fa-check"></i><b>10.8</b> Visualizations</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="spatial-proteomics-multiplex-if.html"><a href="spatial-proteomics-multiplex-if.html"><i class="fa fa-check"></i><b>11</b> Spatial proteomics: multiplex IF</a>
<ul>
<li class="chapter" data-level="11.1" data-path="spatial-proteomics-multiplex-if.html"><a href="spatial-proteomics-multiplex-if.html#read-in-raw-data-2"><i class="fa fa-check"></i><b>11.1</b> Read in raw data</a></li>
<li class="chapter" data-level="11.2" data-path="spatial-proteomics-multiplex-if.html"><a href="spatial-proteomics-multiplex-if.html#overlap-intensity-workflows"><i class="fa fa-check"></i><b>11.2</b> Overlap intensity &amp; workflows</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html"><i class="fa fa-check"></i><b>12</b> Working with multiple samples</a>
<ul>
<li class="chapter" data-level="12.1" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#objective-1"><i class="fa fa-check"></i><b>12.1</b> Objective</a></li>
<li class="chapter" data-level="12.2" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#background-1"><i class="fa fa-check"></i><b>12.2</b> Background</a>
<ul>
<li class="chapter" data-level="12.2.1" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#dataset"><i class="fa fa-check"></i><b>12.2.1</b> Dataset</a></li>
<li class="chapter" data-level="12.2.2" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#visium-technology"><i class="fa fa-check"></i><b>12.2.2</b> Visium technology</a></li>
</ul></li>
<li class="chapter" data-level="12.3" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#create-individual-giotto-objects"><i class="fa fa-check"></i><b>12.3</b> Create individual giotto objects</a>
<ul>
<li class="chapter" data-level="12.3.1" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#download-the-data"><i class="fa fa-check"></i><b>12.3.1</b> Download the data</a></li>
<li class="chapter" data-level="12.3.2" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#create-giotto-instructions"><i class="fa fa-check"></i><b>12.3.2</b> create giotto instructions</a></li>
<li class="chapter" data-level="12.3.3" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#load-visium-data-into-giotto"><i class="fa fa-check"></i><b>12.3.3</b> load visium data into Giotto</a></li>
<li class="chapter" data-level="12.3.4" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#healthy-prostate-tissue-coverage"><i class="fa fa-check"></i><b>12.3.4</b> Healthy prostate tissue coverage</a></li>
<li class="chapter" data-level="12.3.5" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#adenocarcinoma-prostate-tissue-coverage"><i class="fa fa-check"></i><b>12.3.5</b> Adenocarcinoma prostate tissue coverage</a></li>
<li class="chapter" data-level="12.3.6" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#showing-the-data-strucutre-for-the-inidividual-objects"><i class="fa fa-check"></i><b>12.3.6</b> Showing the data strucutre for the inidividual objects</a></li>
</ul></li>
<li class="chapter" data-level="12.4" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#join-giotto-objects"><i class="fa fa-check"></i><b>12.4</b> Join Giotto Objects</a></li>
<li class="chapter" data-level="12.5" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#visualizing-combined-datasets"><i class="fa fa-check"></i><b>12.5</b> Visualizing combined datasets</a>
<ul>
<li class="chapter" data-level="12.5.1" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#vizualizing-in-the-same-plot"><i class="fa fa-check"></i><b>12.5.1</b> Vizualizing in the same plot</a></li>
<li class="chapter" data-level="12.5.2" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#vizualizing-on-separate-plots"><i class="fa fa-check"></i><b>12.5.2</b> Vizualizing on separate plots</a></li>
</ul></li>
<li class="chapter" data-level="12.6" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#splitting-combined-dataset"><i class="fa fa-check"></i><b>12.6</b> Splitting combined dataset</a></li>
<li class="chapter" data-level="12.7" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#analyzing-joined-objects"><i class="fa fa-check"></i><b>12.7</b> Analyzing joined objects</a>
<ul>
<li class="chapter" data-level="12.7.1" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#normalization-and-adding-statistics"><i class="fa fa-check"></i><b>12.7.1</b> Normalization and adding statistics</a></li>
<li class="chapter" data-level="12.7.2" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#clustering-the-datasets"><i class="fa fa-check"></i><b>12.7.2</b> Clustering the datasets</a></li>
<li class="chapter" data-level="12.7.3" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#vizualizing-spatial-location-of-clusters"><i class="fa fa-check"></i><b>12.7.3</b> Vizualizing spatial location of clusters</a></li>
<li class="chapter" data-level="12.7.4" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#vizualizing-tissue-contribution-to-clusters"><i class="fa fa-check"></i><b>12.7.4</b> Vizualizing tissue contribution to clusters</a></li>
</ul></li>
<li class="chapter" data-level="12.8" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#perform-harmony-and-default-workflows"><i class="fa fa-check"></i><b>12.8</b> Perform Harmony and default workflows</a>
<ul>
<li class="chapter" data-level="12.8.1" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#clustering-harmonized-object"><i class="fa fa-check"></i><b>12.8.1</b> Clustering harmonized object</a></li>
<li class="chapter" data-level="12.8.2" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#vizualizing-the-tissue-contribution"><i class="fa fa-check"></i><b>12.8.2</b> Vizualizing the tissue contribution</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="13" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html"><i class="fa fa-check"></i><b>13</b> Spatial multi-modal analysis</a>
<ul>
<li class="chapter" data-level="13.1" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#overview-1"><i class="fa fa-check"></i><b>13.1</b> Overview</a></li>
<li class="chapter" data-level="13.2" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#spatial-manipulation"><i class="fa fa-check"></i><b>13.2</b> Spatial manipulation</a>
<ul>
<li class="chapter" data-level="13.2.1" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#spatial-transforms"><i class="fa fa-check"></i><b>13.2.1</b> Spatial transforms:</a></li>
<li class="chapter" data-level="13.2.2" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#spatial-utilities"><i class="fa fa-check"></i><b>13.2.2</b> Spatial utilities:</a></li>
<li class="chapter" data-level="13.2.3" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#spatial-classes"><i class="fa fa-check"></i><b>13.2.3</b> Spatial classes:</a></li>
</ul></li>
<li class="chapter" data-level="13.3" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#examples-of-the-simple-transforms-with-a-giottopolygon"><i class="fa fa-check"></i><b>13.3</b> Examples of the simple transforms with a giottoPolygon</a></li>
<li class="chapter" data-level="13.4" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#affine-transforms"><i class="fa fa-check"></i><b>13.4</b> Affine transforms</a></li>
<li class="chapter" data-level="13.5" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#image-transforms"><i class="fa fa-check"></i><b>13.5</b> Image transforms</a></li>
<li class="chapter" data-level="13.6" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#co-registration-of-datasets"><i class="fa fa-check"></i><b>13.6</b> Co-registration of datasets</a></li>
<li class="chapter" data-level="13.7" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#examples-in-giotto-suite-manuscript"><i class="fa fa-check"></i><b>13.7</b> Examples in giotto suite manuscript</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html"><i class="fa fa-check"></i><b>14</b> Multi-omics integration</a>
<ul>
<li class="chapter" data-level="14.1" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#the-cytassist-technology"><i class="fa fa-check"></i><b>14.1</b> The CytAssist technology</a></li>
<li class="chapter" data-level="14.2" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#introduction-to-the-spatial-dataset-1"><i class="fa fa-check"></i><b>14.2</b> Introduction to the spatial dataset</a></li>
<li class="chapter" data-level="14.3" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#download-dataset-1"><i class="fa fa-check"></i><b>14.3</b> Download dataset</a></li>
<li class="chapter" data-level="14.4" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#create-the-giotto-object-1"><i class="fa fa-check"></i><b>14.4</b> Create the Giotto object</a></li>
<li class="chapter" data-level="14.5" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#subset-on-spots-that-were-covered-by-tissue-1"><i class="fa fa-check"></i><b>14.5</b> Subset on spots that were covered by tissue</a></li>
<li class="chapter" data-level="14.6" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#rna-processing"><i class="fa fa-check"></i><b>14.6</b> RNA processing</a></li>
<li class="chapter" data-level="14.7" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#protein-processing"><i class="fa fa-check"></i><b>14.7</b> Protein processing</a></li>
<li class="chapter" data-level="14.8" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#multi-omics-integration-1"><i class="fa fa-check"></i><b>14.8</b> Multi-omics integration</a></li>
<li class="chapter" data-level="14.9" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#session-info-2"><i class="fa fa-check"></i><b>14.9</b> Session info</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html"><i class="fa fa-check"></i><b>15</b> Interoperability with other frameworks</a>
<ul>
<li class="chapter" data-level="15.1" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#load-giotto-object"><i class="fa fa-check"></i><b>15.1</b> Load Giotto object</a></li>
<li class="chapter" data-level="15.2" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#seurat"><i class="fa fa-check"></i><b>15.2</b> Seurat</a>
<ul>
<li class="chapter" data-level="15.2.1" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#conversion-of-giotto-object-to-seurat-object"><i class="fa fa-check"></i><b>15.2.1</b> Conversion of Giotto Object to Seurat Object</a></li>
<li class="chapter" data-level="15.2.2" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#conversion-of-seurat-object-back-to-giotto-object"><i class="fa fa-check"></i><b>15.2.2</b> Conversion of Seurat object Back to Giotto Object</a></li>
</ul></li>
<li class="chapter" data-level="15.3" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#spatialexperiment"><i class="fa fa-check"></i><b>15.3</b> SpatialExperiment</a>
<ul>
<li class="chapter" data-level="15.3.1" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#convert-giotto-object-to-spatialexperiment-object"><i class="fa fa-check"></i><b>15.3.1</b> Convert Giotto Object to SpatialExperiment Object</a></li>
<li class="chapter" data-level="15.3.2" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#conversion-of-spatialexperiment-object-back-to-giotto"><i class="fa fa-check"></i><b>15.3.2</b> Conversion of SpatialExperiment object back to Giotto</a></li>
</ul></li>
<li class="chapter" data-level="15.4" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#anndata"><i class="fa fa-check"></i><b>15.4</b> AnnData</a>
<ul>
<li class="chapter" data-level="15.4.1" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#load-required-libraries"><i class="fa fa-check"></i><b>15.4.1</b> Load Required Libraries</a></li>
<li class="chapter" data-level="15.4.2" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#specify-path-for-results"><i class="fa fa-check"></i><b>15.4.2</b> Specify Path for Results</a></li>
<li class="chapter" data-level="15.4.3" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#giotto-to-anndata"><i class="fa fa-check"></i><b>15.4.3</b> Giotto To AnnData</a></li>
<li class="chapter" data-level="15.4.4" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#convert-anndata-to-giotto"><i class="fa fa-check"></i><b>15.4.4</b> Convert AnnData to Giotto</a></li>
</ul></li>
<li class="chapter" data-level="15.5" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#create-mini-vizgen-object"><i class="fa fa-check"></i><b>15.5</b> Create mini Vizgen object</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html"><i class="fa fa-check"></i><b>16</b> Interoperability with isolated tools</a>
<ul>
<li class="chapter" data-level="16.1" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#spatial-niche-trajectory-analysis"><i class="fa fa-check"></i><b>16.1</b> Spatial niche trajectory analysis</a>
<ul>
<li class="chapter" data-level="16.1.1" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#dataset-download"><i class="fa fa-check"></i><b>16.1.1</b> Dataset download</a></li>
<li class="chapter" data-level="16.1.2" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#create-the-giotto-object-2"><i class="fa fa-check"></i><b>16.1.2</b> Create the Giotto object</a></li>
<li class="chapter" data-level="16.1.3" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#visualization"><i class="fa fa-check"></i><b>16.1.3</b> visualization</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="17" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html"><i class="fa fa-check"></i><b>17</b> Interactivity with the R/Spatial ecosystem</a>
<ul>
<li class="chapter" data-level="17.1" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#kriging"><i class="fa fa-check"></i><b>17.1</b> Kriging</a>
<ul>
<li class="chapter" data-level="17.1.1" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#visium-technology-1"><i class="fa fa-check"></i><b>17.1.1</b> Visium technology</a></li>
<li class="chapter" data-level="17.1.2" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#dataset-1"><i class="fa fa-check"></i><b>17.1.2</b> Dataset</a></li>
<li class="chapter" data-level="17.1.3" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#generating-a-geojson-file-of-nuclei-location"><i class="fa fa-check"></i><b>17.1.3</b> Generating a geojson file of nuclei location</a></li>
</ul></li>
<li class="chapter" data-level="17.2" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#downloading-the-dataset"><i class="fa fa-check"></i><b>17.2</b> Downloading the dataset</a></li>
<li class="chapter" data-level="17.3" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#importing-visium-data"><i class="fa fa-check"></i><b>17.3</b> Importing visium data</a>
<ul>
<li class="chapter" data-level="17.3.1" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#identifying-spatially-organized-features"><i class="fa fa-check"></i><b>17.3.1</b> Identifying spatially organized features</a></li>
</ul></li>
<li class="chapter" data-level="17.4" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#adding-cell-polygons-to-giotto-object"><i class="fa fa-check"></i><b>17.4</b> Adding cell polygons to Giotto object</a>
<ul>
<li class="chapter" data-level="17.4.1" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#read-in-the-poly-information"><i class="fa fa-check"></i><b>17.4.1</b> Read in the poly information</a></li>
<li class="chapter" data-level="17.4.2" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#vizualizing-polygons"><i class="fa fa-check"></i><b>17.4.2</b> Vizualizing polygons</a></li>
<li class="chapter" data-level="17.4.3" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#showing-giotto-object-prior-to-polygon-addition"><i class="fa fa-check"></i><b>17.4.3</b> Showing Giotto object prior to polygon addition</a></li>
<li class="chapter" data-level="17.4.4" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#adding-polygons-to-giotto-object"><i class="fa fa-check"></i><b>17.4.4</b> Adding polygons to giotto object</a></li>
<li class="chapter" data-level="17.4.5" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#check-polygon-information"><i class="fa fa-check"></i><b>17.4.5</b> Check polygon information</a></li>
</ul></li>
<li class="chapter" data-level="17.5" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#performing-kriging"><i class="fa fa-check"></i><b>17.5</b> Performing kriging</a>
<ul>
<li class="chapter" data-level="17.5.1" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#interpolating-features"><i class="fa fa-check"></i><b>17.5.1</b> Interpolating features</a></li>
<li class="chapter" data-level="17.5.2" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#expression-overlap"><i class="fa fa-check"></i><b>17.5.2</b> Expression overlap</a></li>
</ul></li>
<li class="chapter" data-level="17.6" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#reading-in-larger-dataset"><i class="fa fa-check"></i><b>17.6</b> Reading in larger dataset</a></li>
<li class="chapter" data-level="17.7" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#analyzing-interpolated-features"><i class="fa fa-check"></i><b>17.7</b> Analyzing interpolated features</a>
<ul>
<li class="chapter" data-level="17.7.1" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#filter-and-normalization"><i class="fa fa-check"></i><b>17.7.1</b> Filter and normalization</a></li>
<li class="chapter" data-level="17.7.2" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#visualizing-gene-expression-from-interpolated-expression"><i class="fa fa-check"></i><b>17.7.2</b> Visualizing gene expression from interpolated expression</a></li>
<li class="chapter" data-level="17.7.3" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#run-pca"><i class="fa fa-check"></i><b>17.7.3</b> Run PCA</a></li>
<li class="chapter" data-level="17.7.4" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#clustering-1"><i class="fa fa-check"></i><b>17.7.4</b> Clustering</a></li>
<li class="chapter" data-level="17.7.5" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#visualizing-clustering"><i class="fa fa-check"></i><b>17.7.5</b> Visualizing clustering</a></li>
<li class="chapter" data-level="17.7.6" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#cropping-objects"><i class="fa fa-check"></i><b>17.7.6</b> Cropping objects</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="18" data-path="contributing-to-giotto.html"><a href="contributing-to-giotto.html"><i class="fa fa-check"></i><b>18</b> Contributing to Giotto</a>
<ul>
<li class="chapter" data-level="18.1" data-path="contributing-to-giotto.html"><a href="contributing-to-giotto.html#contribution-guideline"><i class="fa fa-check"></i><b>18.1</b> Contribution guideline</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Workshop: Spatial multi-omics data analysis with Giotto Suite</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="visium-part-ii" class="section level1 hasAnchor" number="8">
<h1><span class="header-section-number">8</span> Visium Part II<a href="visium-part-ii.html#visium-part-ii" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Joselyn Cristina Chávez Fuentes</p>
<p>August 6th 2024</p>
<div id="load-the-object" class="section level2 hasAnchor" number="8.1">
<h2><span class="header-section-number">8.1</span> Load the object<a href="visium-part-ii.html#load-the-object" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb58"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb58-1"><a href="visium-part-ii.html#cb58-1" tabindex="-1"></a><span class="fu">library</span>(Giotto)</span>
<span id="cb58-2"><a href="visium-part-ii.html#cb58-2" tabindex="-1"></a></span>
<span id="cb58-3"><a href="visium-part-ii.html#cb58-3" tabindex="-1"></a>visium_brain <span class="ot">&lt;-</span> <span class="fu">loadGiotto</span>(<span class="st">&quot;visium_brain_object&quot;</span>)</span></code></pre></div>
</div>
<div id="differential-expression" class="section level2 hasAnchor" number="8.2">
<h2><span class="header-section-number">8.2</span> Differential expression<a href="visium-part-ii.html#differential-expression" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="gini-markers" class="section level3 hasAnchor" number="8.2.1">
<h3><span class="header-section-number">8.2.1</span> Gini markers<a href="visium-part-ii.html#gini-markers" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li>Calculate the top marker genes per cluster using the gini method</li>
</ul>
<div class="sourceCode" id="cb59"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb59-1"><a href="visium-part-ii.html#cb59-1" tabindex="-1"></a>gini_markers <span class="ot">&lt;-</span> <span class="fu">findMarkers_one_vs_all</span>(<span class="at">gobject =</span> visium_brain,</span>
<span id="cb59-2"><a href="visium-part-ii.html#cb59-2" tabindex="-1"></a>                                       <span class="at">method =</span> <span class="st">&quot;gini&quot;</span>,</span>
<span id="cb59-3"><a href="visium-part-ii.html#cb59-3" tabindex="-1"></a>                                       <span class="at">expression_values =</span> <span class="st">&quot;normalized&quot;</span>,</span>
<span id="cb59-4"><a href="visium-part-ii.html#cb59-4" tabindex="-1"></a>                                       <span class="at">cluster_column =</span> <span class="st">&quot;leiden_clus&quot;</span>,</span>
<span id="cb59-5"><a href="visium-part-ii.html#cb59-5" tabindex="-1"></a>                                       <span class="at">min_feats =</span> <span class="dv">10</span>)</span>
<span id="cb59-6"><a href="visium-part-ii.html#cb59-6" tabindex="-1"></a></span>
<span id="cb59-7"><a href="visium-part-ii.html#cb59-7" tabindex="-1"></a>topgenes_gini <span class="ot">&lt;-</span> gini_markers[, <span class="fu">head</span>(.SD, <span class="dv">2</span>), by <span class="ot">=</span> <span class="st">&quot;cluster&quot;</span>]<span class="sc">$</span>feats</span></code></pre></div>
<ul>
<li>Visualize</li>
</ul>
<div class="sourceCode" id="cb60"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb60-1"><a href="visium-part-ii.html#cb60-1" tabindex="-1"></a><span class="fu">violinPlot</span>(visium_brain, </span>
<span id="cb60-2"><a href="visium-part-ii.html#cb60-2" tabindex="-1"></a>           <span class="at">feats =</span> <span class="fu">unique</span>(topgenes_gini), </span>
<span id="cb60-3"><a href="visium-part-ii.html#cb60-3" tabindex="-1"></a>           <span class="at">cluster_column =</span> <span class="st">&quot;leiden_clus&quot;</span>,</span>
<span id="cb60-4"><a href="visium-part-ii.html#cb60-4" tabindex="-1"></a>           <span class="at">strip_text =</span> <span class="dv">6</span>, </span>
<span id="cb60-5"><a href="visium-part-ii.html#cb60-5" tabindex="-1"></a>           <span class="at">strip_position =</span> <span class="st">&quot;right&quot;</span>,</span>
<span id="cb60-6"><a href="visium-part-ii.html#cb60-6" tabindex="-1"></a>           <span class="at">save_param =</span> <span class="fu">list</span>(<span class="at">base_width =</span> <span class="dv">5</span>, <span class="at">base_height =</span> <span class="dv">30</span>))</span></code></pre></div>
<p><img src="img/02_session1/0-violinPlot.png" width="50%" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb61"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb61-1"><a href="visium-part-ii.html#cb61-1" tabindex="-1"></a><span class="fu">plotMetaDataHeatmap</span>(visium_brain, </span>
<span id="cb61-2"><a href="visium-part-ii.html#cb61-2" tabindex="-1"></a>                    <span class="at">selected_feats =</span> <span class="fu">unique</span>(topgenes_gini),</span>
<span id="cb61-3"><a href="visium-part-ii.html#cb61-3" tabindex="-1"></a>                    <span class="at">metadata_cols =</span> <span class="st">&quot;leiden_clus&quot;</span>,</span>
<span id="cb61-4"><a href="visium-part-ii.html#cb61-4" tabindex="-1"></a>                    <span class="at">x_text_size =</span> <span class="dv">10</span>, <span class="at">y_text_size =</span> <span class="dv">10</span>)</span></code></pre></div>
<p><img src="img/02_session1/1-plotMetaDataHeatmap.png" width="80%" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb62"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb62-1"><a href="visium-part-ii.html#cb62-1" tabindex="-1"></a><span class="fu">dimFeatPlot2D</span>(visium_brain, </span>
<span id="cb62-2"><a href="visium-part-ii.html#cb62-2" tabindex="-1"></a>              <span class="at">expression_values =</span> <span class="st">&quot;scaled&quot;</span>,</span>
<span id="cb62-3"><a href="visium-part-ii.html#cb62-3" tabindex="-1"></a>              <span class="at">feats =</span> <span class="fu">sort</span>(<span class="fu">unique</span>(topgenes_gini)),</span>
<span id="cb62-4"><a href="visium-part-ii.html#cb62-4" tabindex="-1"></a>              <span class="at">cow_n_col =</span> <span class="dv">5</span>, </span>
<span id="cb62-5"><a href="visium-part-ii.html#cb62-5" tabindex="-1"></a>              <span class="at">point_size =</span> <span class="dv">1</span>,</span>
<span id="cb62-6"><a href="visium-part-ii.html#cb62-6" tabindex="-1"></a>              <span class="at">save_param =</span> <span class="fu">list</span>(<span class="at">base_width =</span> <span class="dv">15</span>, <span class="at">base_height =</span> <span class="dv">20</span>))</span></code></pre></div>
<p><img src="img/02_session1/2-dimFeatPlot2D.png" width="100%" style="display: block; margin: auto;" /></p>
</div>
<div id="scran-markers" class="section level3 hasAnchor" number="8.2.2">
<h3><span class="header-section-number">8.2.2</span> Scran markers<a href="visium-part-ii.html#scran-markers" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li>Calculate the top marker genes per cluster using the scran method</li>
</ul>
<div class="sourceCode" id="cb63"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb63-1"><a href="visium-part-ii.html#cb63-1" tabindex="-1"></a>scran_markers <span class="ot">&lt;-</span> <span class="fu">findMarkers_one_vs_all</span>(<span class="at">gobject =</span> visium_brain,</span>
<span id="cb63-2"><a href="visium-part-ii.html#cb63-2" tabindex="-1"></a>                                       <span class="at">method =</span> <span class="st">&quot;scran&quot;</span>,</span>
<span id="cb63-3"><a href="visium-part-ii.html#cb63-3" tabindex="-1"></a>                                       <span class="at">expression_values =</span> <span class="st">&quot;normalized&quot;</span>,</span>
<span id="cb63-4"><a href="visium-part-ii.html#cb63-4" tabindex="-1"></a>                                       <span class="at">cluster_column =</span> <span class="st">&quot;leiden_clus&quot;</span>,</span>
<span id="cb63-5"><a href="visium-part-ii.html#cb63-5" tabindex="-1"></a>                                       <span class="at">min_feats =</span> <span class="dv">10</span>)</span>
<span id="cb63-6"><a href="visium-part-ii.html#cb63-6" tabindex="-1"></a></span>
<span id="cb63-7"><a href="visium-part-ii.html#cb63-7" tabindex="-1"></a>topgenes_scran <span class="ot">&lt;-</span> scran_markers[, <span class="fu">head</span>(.SD, <span class="dv">2</span>), by <span class="ot">=</span> <span class="st">&quot;cluster&quot;</span>]<span class="sc">$</span>feats</span></code></pre></div>
<ul>
<li>Visualize</li>
</ul>
<div class="sourceCode" id="cb64"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb64-1"><a href="visium-part-ii.html#cb64-1" tabindex="-1"></a><span class="fu">violinPlot</span>(visium_brain, </span>
<span id="cb64-2"><a href="visium-part-ii.html#cb64-2" tabindex="-1"></a>           <span class="at">feats =</span> <span class="fu">unique</span>(topgenes_scran), </span>
<span id="cb64-3"><a href="visium-part-ii.html#cb64-3" tabindex="-1"></a>           <span class="at">cluster_column =</span> <span class="st">&quot;leiden_clus&quot;</span>,</span>
<span id="cb64-4"><a href="visium-part-ii.html#cb64-4" tabindex="-1"></a>           <span class="at">strip_text =</span> <span class="dv">6</span>, </span>
<span id="cb64-5"><a href="visium-part-ii.html#cb64-5" tabindex="-1"></a>           <span class="at">strip_position =</span> <span class="st">&quot;right&quot;</span>,</span>
<span id="cb64-6"><a href="visium-part-ii.html#cb64-6" tabindex="-1"></a>           <span class="at">save_param =</span> <span class="fu">list</span>(<span class="at">base_width =</span> <span class="dv">5</span>, <span class="at">base_height =</span> <span class="dv">30</span>))</span></code></pre></div>
<p><img src="img/02_session1/3-violinPlot.png" width="50%" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb65"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb65-1"><a href="visium-part-ii.html#cb65-1" tabindex="-1"></a><span class="fu">plotMetaDataHeatmap</span>(visium_brain, </span>
<span id="cb65-2"><a href="visium-part-ii.html#cb65-2" tabindex="-1"></a>                    <span class="at">selected_feats =</span> <span class="fu">unique</span>(topgenes_scran),</span>
<span id="cb65-3"><a href="visium-part-ii.html#cb65-3" tabindex="-1"></a>                    <span class="at">metadata_cols =</span> <span class="st">&quot;leiden_clus&quot;</span>,</span>
<span id="cb65-4"><a href="visium-part-ii.html#cb65-4" tabindex="-1"></a>                    <span class="at">x_text_size =</span> <span class="dv">10</span>, <span class="at">y_text_size =</span> <span class="dv">10</span>)</span></code></pre></div>
<p><img src="img/02_session1/4-plotMetaDataHeatmap.png" width="100%" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb66"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb66-1"><a href="visium-part-ii.html#cb66-1" tabindex="-1"></a><span class="fu">dimFeatPlot2D</span>(visium_brain, </span>
<span id="cb66-2"><a href="visium-part-ii.html#cb66-2" tabindex="-1"></a>              <span class="at">expression_values =</span> <span class="st">&quot;scaled&quot;</span>,</span>
<span id="cb66-3"><a href="visium-part-ii.html#cb66-3" tabindex="-1"></a>              <span class="at">feats =</span> <span class="fu">sort</span>(<span class="fu">unique</span>(topgenes_scran)),</span>
<span id="cb66-4"><a href="visium-part-ii.html#cb66-4" tabindex="-1"></a>              <span class="at">cow_n_col =</span> <span class="dv">5</span>, </span>
<span id="cb66-5"><a href="visium-part-ii.html#cb66-5" tabindex="-1"></a>              <span class="at">point_size =</span> <span class="dv">1</span>,</span>
<span id="cb66-6"><a href="visium-part-ii.html#cb66-6" tabindex="-1"></a>              <span class="at">save_param =</span> <span class="fu">list</span>(<span class="at">base_width =</span> <span class="dv">20</span>, <span class="at">base_height =</span> <span class="dv">20</span>))</span></code></pre></div>
<p><img src="img/02_session1/5-dimFeatPlot2D.png" width="100%" style="display: block; margin: auto;" /></p>
<p>The Gini method is preferred when identifying genes that exhibit significant expression differences across clusters, highlighting genes with distinct expression patters between different clusters.</p>
<p>The Scran method is preferred for robust differential expression analysis, especially when addressing technical variability or differences in sequencing depth across spatial locations.</p>
<p>In practice, it is often beneficial to apply both methods and compare results for a more complete understanding of differential gene expression across clusters.</p>
</div>
</div>
<div id="enrichment-deconvolution" class="section level2 hasAnchor" number="8.3">
<h2><span class="header-section-number">8.3</span> Enrichment &amp; Deconvolution<a href="visium-part-ii.html#enrichment-deconvolution" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Visium spatial transcriptomics does not provide single-cell resolution, making cell type annotation a harder problem. Giotto provides several ways to calculate enrichment of specific cell-type signature gene lists.</p>
<ul>
<li>Download the single-cell dataset</li>
</ul>
<div class="sourceCode" id="cb67"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb67-1"><a href="visium-part-ii.html#cb67-1" tabindex="-1"></a>GiottoData<span class="sc">::</span><span class="fu">getSpatialDataset</span>(<span class="at">dataset =</span> <span class="st">&quot;scRNA_mouse_brain&quot;</span>, <span class="at">directory =</span> <span class="st">&quot;data/&quot;</span>)</span></code></pre></div>
<ul>
<li>Create the single-cell object</li>
</ul>
<div class="sourceCode" id="cb68"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb68-1"><a href="visium-part-ii.html#cb68-1" tabindex="-1"></a>results_folder <span class="ot">&lt;-</span> <span class="st">&quot;results/&quot;</span></span>
<span id="cb68-2"><a href="visium-part-ii.html#cb68-2" tabindex="-1"></a></span>
<span id="cb68-3"><a href="visium-part-ii.html#cb68-3" tabindex="-1"></a>python_path <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb68-4"><a href="visium-part-ii.html#cb68-4" tabindex="-1"></a></span>
<span id="cb68-5"><a href="visium-part-ii.html#cb68-5" tabindex="-1"></a>instructions <span class="ot">&lt;-</span> <span class="fu">createGiottoInstructions</span>(</span>
<span id="cb68-6"><a href="visium-part-ii.html#cb68-6" tabindex="-1"></a>    <span class="at">save_dir =</span> results_folder,</span>
<span id="cb68-7"><a href="visium-part-ii.html#cb68-7" tabindex="-1"></a>    <span class="at">save_plot =</span> <span class="cn">TRUE</span>,</span>
<span id="cb68-8"><a href="visium-part-ii.html#cb68-8" tabindex="-1"></a>    <span class="at">show_plot =</span> <span class="cn">FALSE</span>,</span>
<span id="cb68-9"><a href="visium-part-ii.html#cb68-9" tabindex="-1"></a>    <span class="at">python_path =</span> python_path</span>
<span id="cb68-10"><a href="visium-part-ii.html#cb68-10" tabindex="-1"></a>)</span>
<span id="cb68-11"><a href="visium-part-ii.html#cb68-11" tabindex="-1"></a></span>
<span id="cb68-12"><a href="visium-part-ii.html#cb68-12" tabindex="-1"></a>sc_expression <span class="ot">&lt;-</span> <span class="st">&quot;data/brain_sc_expression_matrix.txt.gz&quot;</span></span>
<span id="cb68-13"><a href="visium-part-ii.html#cb68-13" tabindex="-1"></a>sc_metadata <span class="ot">&lt;-</span> <span class="st">&quot;data/brain_sc_metadata.csv&quot;</span></span>
<span id="cb68-14"><a href="visium-part-ii.html#cb68-14" tabindex="-1"></a></span>
<span id="cb68-15"><a href="visium-part-ii.html#cb68-15" tabindex="-1"></a>giotto_SC <span class="ot">&lt;-</span> <span class="fu">createGiottoObject</span>(<span class="at">expression =</span> sc_expression,</span>
<span id="cb68-16"><a href="visium-part-ii.html#cb68-16" tabindex="-1"></a>                                <span class="at">instructions =</span> instructions)</span>
<span id="cb68-17"><a href="visium-part-ii.html#cb68-17" tabindex="-1"></a></span>
<span id="cb68-18"><a href="visium-part-ii.html#cb68-18" tabindex="-1"></a>giotto_SC <span class="ot">&lt;-</span> <span class="fu">addCellMetadata</span>(giotto_SC, </span>
<span id="cb68-19"><a href="visium-part-ii.html#cb68-19" tabindex="-1"></a>                             <span class="at">new_metadata =</span> data.table<span class="sc">::</span><span class="fu">fread</span>(sc_metadata))</span>
<span id="cb68-20"><a href="visium-part-ii.html#cb68-20" tabindex="-1"></a></span>
<span id="cb68-21"><a href="visium-part-ii.html#cb68-21" tabindex="-1"></a>giotto_SC <span class="ot">&lt;-</span> <span class="fu">normalizeGiotto</span>(giotto_SC)</span></code></pre></div>
<div id="pagerank" class="section level3 hasAnchor" number="8.3.1">
<h3><span class="header-section-number">8.3.1</span> PAGE/Rank<a href="visium-part-ii.html#pagerank" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Parametric Analysis of Gene Set Enrichment (PAGE) and Rank enrichment both aim to determine whether a predefined set of genes show statistically significant differences in expression compared to other genes in the dataset.</p>
<ul>
<li>Calculate the cell type markers</li>
</ul>
<div class="sourceCode" id="cb69"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb69-1"><a href="visium-part-ii.html#cb69-1" tabindex="-1"></a>markers_scran <span class="ot">&lt;-</span> <span class="fu">findMarkers_one_vs_all</span>(<span class="at">gobject =</span> giotto_SC, </span>
<span id="cb69-2"><a href="visium-part-ii.html#cb69-2" tabindex="-1"></a>                                        <span class="at">method =</span> <span class="st">&quot;scran&quot;</span>,</span>
<span id="cb69-3"><a href="visium-part-ii.html#cb69-3" tabindex="-1"></a>                                        <span class="at">expression_values =</span> <span class="st">&quot;normalized&quot;</span>,</span>
<span id="cb69-4"><a href="visium-part-ii.html#cb69-4" tabindex="-1"></a>                                        <span class="at">cluster_column =</span> <span class="st">&quot;Class&quot;</span>, </span>
<span id="cb69-5"><a href="visium-part-ii.html#cb69-5" tabindex="-1"></a>                                        <span class="at">min_feats =</span> <span class="dv">3</span>)</span>
<span id="cb69-6"><a href="visium-part-ii.html#cb69-6" tabindex="-1"></a></span>
<span id="cb69-7"><a href="visium-part-ii.html#cb69-7" tabindex="-1"></a>top_markers <span class="ot">&lt;-</span> markers_scran[, <span class="fu">head</span>(.SD, <span class="dv">10</span>), by <span class="ot">=</span> <span class="st">&quot;cluster&quot;</span>]</span>
<span id="cb69-8"><a href="visium-part-ii.html#cb69-8" tabindex="-1"></a>celltypes <span class="ot">&lt;-</span> <span class="fu">levels</span>(<span class="fu">factor</span>(markers_scran<span class="sc">$</span>cluster)) </span></code></pre></div>
<ul>
<li>Create the signature matrix</li>
</ul>
<div class="sourceCode" id="cb70"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb70-1"><a href="visium-part-ii.html#cb70-1" tabindex="-1"></a>sign_list <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb70-2"><a href="visium-part-ii.html#cb70-2" tabindex="-1"></a></span>
<span id="cb70-3"><a href="visium-part-ii.html#cb70-3" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(celltypes)){</span>
<span id="cb70-4"><a href="visium-part-ii.html#cb70-4" tabindex="-1"></a>  sign_list[[i]] <span class="ot">=</span> top_markers[<span class="fu">which</span>(top_markers<span class="sc">$</span>cluster <span class="sc">==</span> celltypes[i]),]<span class="sc">$</span>feats</span>
<span id="cb70-5"><a href="visium-part-ii.html#cb70-5" tabindex="-1"></a>}</span>
<span id="cb70-6"><a href="visium-part-ii.html#cb70-6" tabindex="-1"></a></span>
<span id="cb70-7"><a href="visium-part-ii.html#cb70-7" tabindex="-1"></a>sign_matrix <span class="ot">&lt;-</span> <span class="fu">makeSignMatrixPAGE</span>(<span class="at">sign_names =</span> celltypes,</span>
<span id="cb70-8"><a href="visium-part-ii.html#cb70-8" tabindex="-1"></a>                                  <span class="at">sign_list =</span> sign_list)</span></code></pre></div>
<ul>
<li>Run the enrichment test with PAGE</li>
</ul>
<div class="sourceCode" id="cb71"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb71-1"><a href="visium-part-ii.html#cb71-1" tabindex="-1"></a>visium_brain <span class="ot">&lt;-</span> <span class="fu">runPAGEEnrich</span>(<span class="at">gobject =</span> visium_brain, </span>
<span id="cb71-2"><a href="visium-part-ii.html#cb71-2" tabindex="-1"></a>                              <span class="at">sign_matrix =</span> sign_matrix)</span></code></pre></div>
<ul>
<li>Visualize</li>
</ul>
<div class="sourceCode" id="cb72"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb72-1"><a href="visium-part-ii.html#cb72-1" tabindex="-1"></a>cell_types_PAGE <span class="ot">&lt;-</span> <span class="fu">colnames</span>(sign_matrix)</span>
<span id="cb72-2"><a href="visium-part-ii.html#cb72-2" tabindex="-1"></a></span>
<span id="cb72-3"><a href="visium-part-ii.html#cb72-3" tabindex="-1"></a><span class="fu">plotMetaDataCellsHeatmap</span>(<span class="at">gobject =</span> visium_brain,</span>
<span id="cb72-4"><a href="visium-part-ii.html#cb72-4" tabindex="-1"></a>                         <span class="at">metadata_cols =</span> <span class="st">&quot;leiden_clus&quot;</span>,</span>
<span id="cb72-5"><a href="visium-part-ii.html#cb72-5" tabindex="-1"></a>                         <span class="at">value_cols =</span> cell_types_PAGE,</span>
<span id="cb72-6"><a href="visium-part-ii.html#cb72-6" tabindex="-1"></a>                         <span class="at">spat_enr_names =</span> <span class="st">&quot;PAGE&quot;</span>,</span>
<span id="cb72-7"><a href="visium-part-ii.html#cb72-7" tabindex="-1"></a>                         <span class="at">x_text_size =</span> <span class="dv">8</span>,</span>
<span id="cb72-8"><a href="visium-part-ii.html#cb72-8" tabindex="-1"></a>                         <span class="at">y_text_size =</span> <span class="dv">8</span>)</span></code></pre></div>
<p><img src="img/02_session1/6-plotMetaDataCellsHeatmap.png" width="100%" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb73"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb73-1"><a href="visium-part-ii.html#cb73-1" tabindex="-1"></a><span class="fu">spatCellPlot2D</span>(<span class="at">gobject =</span> visium_brain,</span>
<span id="cb73-2"><a href="visium-part-ii.html#cb73-2" tabindex="-1"></a>               <span class="at">spat_enr_names =</span> <span class="st">&quot;PAGE&quot;</span>,</span>
<span id="cb73-3"><a href="visium-part-ii.html#cb73-3" tabindex="-1"></a>               <span class="at">cell_annotation_values =</span> cell_types_PAGE,</span>
<span id="cb73-4"><a href="visium-part-ii.html#cb73-4" tabindex="-1"></a>               <span class="at">cow_n_col =</span> <span class="dv">3</span>,</span>
<span id="cb73-5"><a href="visium-part-ii.html#cb73-5" tabindex="-1"></a>               <span class="at">coord_fix_ratio =</span> <span class="dv">1</span>, </span>
<span id="cb73-6"><a href="visium-part-ii.html#cb73-6" tabindex="-1"></a>               <span class="at">point_size =</span> <span class="dv">1</span>, </span>
<span id="cb73-7"><a href="visium-part-ii.html#cb73-7" tabindex="-1"></a>               <span class="at">show_legend =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p><img src="img/02_session1/7-spatCellPlot2D.png" width="100%" style="display: block; margin: auto;" /></p>
</div>
<div id="spatialdwls" class="section level3 hasAnchor" number="8.3.2">
<h3><span class="header-section-number">8.3.2</span> SpatialDWLS<a href="visium-part-ii.html#spatialdwls" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Spatial Diffusion Weighted Least Squares (DWLS) estimates the proportions of different cell types or spatial gradients of gene expression across spots in a tissue.</p>
<ul>
<li>Create the signature matrix</li>
</ul>
<div class="sourceCode" id="cb74"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb74-1"><a href="visium-part-ii.html#cb74-1" tabindex="-1"></a>sign_matrix <span class="ot">&lt;-</span> <span class="fu">makeSignMatrixDWLSfromMatrix</span>(</span>
<span id="cb74-2"><a href="visium-part-ii.html#cb74-2" tabindex="-1"></a>    <span class="at">matrix =</span> <span class="fu">getExpression</span>(giotto_SC,</span>
<span id="cb74-3"><a href="visium-part-ii.html#cb74-3" tabindex="-1"></a>                           <span class="at">values =</span> <span class="st">&quot;normalized&quot;</span>,</span>
<span id="cb74-4"><a href="visium-part-ii.html#cb74-4" tabindex="-1"></a>                           <span class="at">output =</span> <span class="st">&quot;matrix&quot;</span>),</span>
<span id="cb74-5"><a href="visium-part-ii.html#cb74-5" tabindex="-1"></a>    <span class="at">cell_type =</span> <span class="fu">pDataDT</span>(giotto_SC)<span class="sc">$</span>Class,</span>
<span id="cb74-6"><a href="visium-part-ii.html#cb74-6" tabindex="-1"></a>    <span class="at">sign_gene =</span> top_markers<span class="sc">$</span>feats)</span></code></pre></div>
<ul>
<li>Run the DWLS Deconvolution</li>
</ul>
<div class="sourceCode" id="cb75"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb75-1"><a href="visium-part-ii.html#cb75-1" tabindex="-1"></a>visium_brain <span class="ot">&lt;-</span> <span class="fu">runDWLSDeconv</span>(<span class="at">gobject =</span> visium_brain, </span>
<span id="cb75-2"><a href="visium-part-ii.html#cb75-2" tabindex="-1"></a>                              <span class="at">sign_matrix =</span> sign_matrix)</span></code></pre></div>
<ul>
<li>Visualize</li>
</ul>
<div class="sourceCode" id="cb76"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb76-1"><a href="visium-part-ii.html#cb76-1" tabindex="-1"></a><span class="co"># Plot DWLS deconvolution result with Pie plots</span></span>
<span id="cb76-2"><a href="visium-part-ii.html#cb76-2" tabindex="-1"></a><span class="fu">spatDeconvPlot</span>(visium_brain, </span>
<span id="cb76-3"><a href="visium-part-ii.html#cb76-3" tabindex="-1"></a>               <span class="at">show_image =</span> <span class="cn">FALSE</span>,</span>
<span id="cb76-4"><a href="visium-part-ii.html#cb76-4" tabindex="-1"></a>               <span class="at">radius =</span> <span class="dv">50</span>,</span>
<span id="cb76-5"><a href="visium-part-ii.html#cb76-5" tabindex="-1"></a>               <span class="at">save_param =</span> <span class="fu">list</span>(<span class="at">save_name =</span> <span class="st">&quot;8_spat_DWLS_pie_plot&quot;</span>))</span></code></pre></div>
<p><img src="img/02_session1/8_spat_DWLS_pie_plot.png" width="100%" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="spatial-expression-patterns" class="section level2 hasAnchor" number="8.4">
<h2><span class="header-section-number">8.4</span> Spatial expression patterns<a href="visium-part-ii.html#spatial-expression-patterns" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="spatial-variable-genes" class="section level3 hasAnchor" number="8.4.1">
<h3><span class="header-section-number">8.4.1</span> Spatial variable genes<a href="visium-part-ii.html#spatial-variable-genes" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li>Create a spatial network</li>
</ul>
<div class="sourceCode" id="cb77"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb77-1"><a href="visium-part-ii.html#cb77-1" tabindex="-1"></a>visium_brain <span class="ot">&lt;-</span> <span class="fu">createSpatialNetwork</span>(<span class="at">gobject =</span> visium_brain,</span>
<span id="cb77-2"><a href="visium-part-ii.html#cb77-2" tabindex="-1"></a>                                     <span class="at">method =</span> <span class="st">&quot;kNN&quot;</span>, </span>
<span id="cb77-3"><a href="visium-part-ii.html#cb77-3" tabindex="-1"></a>                                     <span class="at">k =</span> <span class="dv">6</span>,</span>
<span id="cb77-4"><a href="visium-part-ii.html#cb77-4" tabindex="-1"></a>                                     <span class="at">maximum_distance_knn =</span> <span class="dv">400</span>,</span>
<span id="cb77-5"><a href="visium-part-ii.html#cb77-5" tabindex="-1"></a>                                     <span class="at">name =</span> <span class="st">&quot;spatial_network&quot;</span>)</span>
<span id="cb77-6"><a href="visium-part-ii.html#cb77-6" tabindex="-1"></a></span>
<span id="cb77-7"><a href="visium-part-ii.html#cb77-7" tabindex="-1"></a><span class="fu">spatPlot2D</span>(<span class="at">gobject =</span> visium_brain,  </span>
<span id="cb77-8"><a href="visium-part-ii.html#cb77-8" tabindex="-1"></a>           <span class="at">show_network=</span> <span class="cn">TRUE</span>,</span>
<span id="cb77-9"><a href="visium-part-ii.html#cb77-9" tabindex="-1"></a>           <span class="at">network_color =</span> <span class="st">&quot;blue&quot;</span>, </span>
<span id="cb77-10"><a href="visium-part-ii.html#cb77-10" tabindex="-1"></a>           <span class="at">spatial_network_name =</span> <span class="st">&quot;spatial_network&quot;</span>)</span></code></pre></div>
<p><img src="img/02_session1/9-spatPlot2D.png" width="100%" style="display: block; margin: auto;" /></p>
<ul>
<li>Rank binarization</li>
</ul>
<div class="sourceCode" id="cb78"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb78-1"><a href="visium-part-ii.html#cb78-1" tabindex="-1"></a>ranktest <span class="ot">&lt;-</span> <span class="fu">binSpect</span>(visium_brain, </span>
<span id="cb78-2"><a href="visium-part-ii.html#cb78-2" tabindex="-1"></a>                     <span class="at">bin_method =</span> <span class="st">&quot;rank&quot;</span>,</span>
<span id="cb78-3"><a href="visium-part-ii.html#cb78-3" tabindex="-1"></a>                     <span class="at">calc_hub =</span> <span class="cn">TRUE</span>, </span>
<span id="cb78-4"><a href="visium-part-ii.html#cb78-4" tabindex="-1"></a>                     <span class="at">hub_min_int =</span> <span class="dv">5</span>,</span>
<span id="cb78-5"><a href="visium-part-ii.html#cb78-5" tabindex="-1"></a>                     <span class="at">spatial_network_name =</span> <span class="st">&quot;spatial_network&quot;</span>)</span></code></pre></div>
<ul>
<li>Visualize top results</li>
</ul>
<div class="sourceCode" id="cb79"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb79-1"><a href="visium-part-ii.html#cb79-1" tabindex="-1"></a><span class="fu">spatFeatPlot2D</span>(visium_brain, </span>
<span id="cb79-2"><a href="visium-part-ii.html#cb79-2" tabindex="-1"></a>               <span class="at">expression_values =</span> <span class="st">&quot;scaled&quot;</span>,</span>
<span id="cb79-3"><a href="visium-part-ii.html#cb79-3" tabindex="-1"></a>               <span class="at">feats =</span> ranktest<span class="sc">$</span>feats[<span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>], </span>
<span id="cb79-4"><a href="visium-part-ii.html#cb79-4" tabindex="-1"></a>               <span class="at">cow_n_col =</span> <span class="dv">2</span>, </span>
<span id="cb79-5"><a href="visium-part-ii.html#cb79-5" tabindex="-1"></a>               <span class="at">point_size =</span> <span class="dv">1</span>)</span></code></pre></div>
<p><img src="img/02_session1/10-spatFeatPlot2D.png" width="100%" style="display: block; margin: auto;" /></p>
</div>
<div id="spatial-co-expression-modules" class="section level3 hasAnchor" number="8.4.2">
<h3><span class="header-section-number">8.4.2</span> Spatial co-expression modules<a href="visium-part-ii.html#spatial-co-expression-modules" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li>Cluster the top 500 spatial genes into 20 clusters</li>
</ul>
<div class="sourceCode" id="cb80"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb80-1"><a href="visium-part-ii.html#cb80-1" tabindex="-1"></a>ext_spatial_genes <span class="ot">&lt;-</span> ranktest[<span class="dv">1</span><span class="sc">:</span><span class="dv">500</span>,]<span class="sc">$</span>feats</span></code></pre></div>
<ul>
<li>Use detectSpatialCorGenes function to calculate pairwise distances between genes.</li>
</ul>
<div class="sourceCode" id="cb81"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb81-1"><a href="visium-part-ii.html#cb81-1" tabindex="-1"></a>spat_cor_netw_DT <span class="ot">&lt;-</span> <span class="fu">detectSpatialCorFeats</span>(</span>
<span id="cb81-2"><a href="visium-part-ii.html#cb81-2" tabindex="-1"></a>    visium_brain,</span>
<span id="cb81-3"><a href="visium-part-ii.html#cb81-3" tabindex="-1"></a>    <span class="at">method =</span> <span class="st">&quot;network&quot;</span>,</span>
<span id="cb81-4"><a href="visium-part-ii.html#cb81-4" tabindex="-1"></a>    <span class="at">spatial_network_name =</span> <span class="st">&quot;spatial_network&quot;</span>,</span>
<span id="cb81-5"><a href="visium-part-ii.html#cb81-5" tabindex="-1"></a>    <span class="at">subset_feats =</span> ext_spatial_genes)</span></code></pre></div>
<ul>
<li>Identify most similar spatially correlated genes for one gene</li>
</ul>
<div class="sourceCode" id="cb82"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb82-1"><a href="visium-part-ii.html#cb82-1" tabindex="-1"></a>top10_genes <span class="ot">&lt;-</span> <span class="fu">showSpatialCorFeats</span>(spat_cor_netw_DT, </span>
<span id="cb82-2"><a href="visium-part-ii.html#cb82-2" tabindex="-1"></a>                                  <span class="at">feats =</span> <span class="st">&quot;Mbp&quot;</span>, </span>
<span id="cb82-3"><a href="visium-part-ii.html#cb82-3" tabindex="-1"></a>                                  <span class="at">show_top_feats =</span> <span class="dv">10</span>)</span></code></pre></div>
<ul>
<li>Visualize</li>
</ul>
<div class="sourceCode" id="cb83"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb83-1"><a href="visium-part-ii.html#cb83-1" tabindex="-1"></a><span class="fu">spatFeatPlot2D</span>(visium_brain, </span>
<span id="cb83-2"><a href="visium-part-ii.html#cb83-2" tabindex="-1"></a>               <span class="at">expression_values =</span> <span class="st">&quot;scaled&quot;</span>,</span>
<span id="cb83-3"><a href="visium-part-ii.html#cb83-3" tabindex="-1"></a>               <span class="at">feats =</span> top10_genes<span class="sc">$</span>variable[<span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>], </span>
<span id="cb83-4"><a href="visium-part-ii.html#cb83-4" tabindex="-1"></a>               <span class="at">point_size =</span> <span class="fl">1.5</span>)</span></code></pre></div>
<p><img src="img/02_session1/11-spatFeatPlot2D.png" width="100%" style="display: block; margin: auto;" /></p>
<ul>
<li>Cluster spatial genes</li>
</ul>
<div class="sourceCode" id="cb84"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb84-1"><a href="visium-part-ii.html#cb84-1" tabindex="-1"></a>spat_cor_netw_DT <span class="ot">&lt;-</span> <span class="fu">clusterSpatialCorFeats</span>(spat_cor_netw_DT, </span>
<span id="cb84-2"><a href="visium-part-ii.html#cb84-2" tabindex="-1"></a>                                           <span class="at">name =</span> <span class="st">&quot;spat_netw_clus&quot;</span>, </span>
<span id="cb84-3"><a href="visium-part-ii.html#cb84-3" tabindex="-1"></a>                                           <span class="at">k =</span> <span class="dv">20</span>)</span></code></pre></div>
<ul>
<li>Visualize clusters</li>
</ul>
<div class="sourceCode" id="cb85"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb85-1"><a href="visium-part-ii.html#cb85-1" tabindex="-1"></a><span class="fu">heatmSpatialCorFeats</span>(visium_brain,</span>
<span id="cb85-2"><a href="visium-part-ii.html#cb85-2" tabindex="-1"></a>                     <span class="at">spatCorObject =</span> spat_cor_netw_DT,</span>
<span id="cb85-3"><a href="visium-part-ii.html#cb85-3" tabindex="-1"></a>                     <span class="at">use_clus_name =</span> <span class="st">&quot;spat_netw_clus&quot;</span>,</span>
<span id="cb85-4"><a href="visium-part-ii.html#cb85-4" tabindex="-1"></a>                     <span class="at">heatmap_legend_param =</span> <span class="fu">list</span>(<span class="at">title =</span> <span class="cn">NULL</span>))</span></code></pre></div>
<p><img src="img/02_session1/12-heatmSpatialCorFeats.png" width="100%" style="display: block; margin: auto;" /></p>
<ul>
<li>Rank spatial correlated clusters and show genes for selected clusters</li>
</ul>
<div class="sourceCode" id="cb86"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb86-1"><a href="visium-part-ii.html#cb86-1" tabindex="-1"></a>netw_ranks <span class="ot">&lt;-</span> <span class="fu">rankSpatialCorGroups</span>(</span>
<span id="cb86-2"><a href="visium-part-ii.html#cb86-2" tabindex="-1"></a>  visium_brain,</span>
<span id="cb86-3"><a href="visium-part-ii.html#cb86-3" tabindex="-1"></a>  <span class="at">spatCorObject =</span> spat_cor_netw_DT, </span>
<span id="cb86-4"><a href="visium-part-ii.html#cb86-4" tabindex="-1"></a>  <span class="at">use_clus_name =</span> <span class="st">&quot;spat_netw_clus&quot;</span>)</span>
<span id="cb86-5"><a href="visium-part-ii.html#cb86-5" tabindex="-1"></a></span>
<span id="cb86-6"><a href="visium-part-ii.html#cb86-6" tabindex="-1"></a>top_netw_spat_cluster <span class="ot">&lt;-</span> <span class="fu">showSpatialCorFeats</span>(spat_cor_netw_DT, </span>
<span id="cb86-7"><a href="visium-part-ii.html#cb86-7" tabindex="-1"></a>                                            <span class="at">use_clus_name =</span> <span class="st">&quot;spat_netw_clus&quot;</span>,</span>
<span id="cb86-8"><a href="visium-part-ii.html#cb86-8" tabindex="-1"></a>                                            <span class="at">selected_clusters =</span> <span class="dv">6</span>, </span>
<span id="cb86-9"><a href="visium-part-ii.html#cb86-9" tabindex="-1"></a>                                            <span class="at">show_top_feats =</span> <span class="dv">1</span>)</span></code></pre></div>
<p><img src="img/02_session1/13-rankSpatialCorGroups.png" width="100%" style="display: block; margin: auto;" /></p>
<ul>
<li>Create metagene enrichment score for clusters</li>
</ul>
<div class="sourceCode" id="cb87"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb87-1"><a href="visium-part-ii.html#cb87-1" tabindex="-1"></a>cluster_genes_DT <span class="ot">&lt;-</span> <span class="fu">showSpatialCorFeats</span>(spat_cor_netw_DT, </span>
<span id="cb87-2"><a href="visium-part-ii.html#cb87-2" tabindex="-1"></a>                                        <span class="at">use_clus_name =</span> <span class="st">&quot;spat_netw_clus&quot;</span>, </span>
<span id="cb87-3"><a href="visium-part-ii.html#cb87-3" tabindex="-1"></a>                                        <span class="at">show_top_feats =</span> <span class="dv">1</span>)</span>
<span id="cb87-4"><a href="visium-part-ii.html#cb87-4" tabindex="-1"></a></span>
<span id="cb87-5"><a href="visium-part-ii.html#cb87-5" tabindex="-1"></a>cluster_genes <span class="ot">&lt;-</span> cluster_genes_DT<span class="sc">$</span>clus </span>
<span id="cb87-6"><a href="visium-part-ii.html#cb87-6" tabindex="-1"></a><span class="fu">names</span>(cluster_genes) <span class="ot">&lt;-</span> cluster_genes_DT<span class="sc">$</span>feat_ID</span>
<span id="cb87-7"><a href="visium-part-ii.html#cb87-7" tabindex="-1"></a></span>
<span id="cb87-8"><a href="visium-part-ii.html#cb87-8" tabindex="-1"></a>visium_brain <span class="ot">&lt;-</span> <span class="fu">createMetafeats</span>(visium_brain, </span>
<span id="cb87-9"><a href="visium-part-ii.html#cb87-9" tabindex="-1"></a>                                <span class="at">feat_clusters =</span> cluster_genes, </span>
<span id="cb87-10"><a href="visium-part-ii.html#cb87-10" tabindex="-1"></a>                                <span class="at">name =</span> <span class="st">&quot;cluster_metagene&quot;</span>)</span>
<span id="cb87-11"><a href="visium-part-ii.html#cb87-11" tabindex="-1"></a></span>
<span id="cb87-12"><a href="visium-part-ii.html#cb87-12" tabindex="-1"></a><span class="fu">spatCellPlot</span>(visium_brain,</span>
<span id="cb87-13"><a href="visium-part-ii.html#cb87-13" tabindex="-1"></a>             <span class="at">spat_enr_names =</span> <span class="st">&quot;cluster_metagene&quot;</span>,</span>
<span id="cb87-14"><a href="visium-part-ii.html#cb87-14" tabindex="-1"></a>             <span class="at">cell_annotation_values =</span> netw_ranks<span class="sc">$</span>clusters,</span>
<span id="cb87-15"><a href="visium-part-ii.html#cb87-15" tabindex="-1"></a>             <span class="at">point_size =</span> <span class="dv">1</span>, </span>
<span id="cb87-16"><a href="visium-part-ii.html#cb87-16" tabindex="-1"></a>             <span class="at">cow_n_col =</span> <span class="dv">5</span>)</span></code></pre></div>
<p><img src="img/02_session1/14-spatCellPlot2D.png" width="100%" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="spatially-informed-clusters" class="section level2 hasAnchor" number="8.5">
<h2><span class="header-section-number">8.5</span> Spatially informed clusters<a href="visium-part-ii.html#spatially-informed-clusters" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ul>
<li>Get the top 30 genes per spatial co-expression cluster</li>
</ul>
<div class="sourceCode" id="cb88"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb88-1"><a href="visium-part-ii.html#cb88-1" tabindex="-1"></a>coexpr_dt <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">data.table</span>(</span>
<span id="cb88-2"><a href="visium-part-ii.html#cb88-2" tabindex="-1"></a>  <span class="at">genes =</span> <span class="fu">names</span>(spat_cor_netw_DT<span class="sc">$</span>cor_clusters<span class="sc">$</span>spat_netw_clus),</span>
<span id="cb88-3"><a href="visium-part-ii.html#cb88-3" tabindex="-1"></a>  <span class="at">cluster =</span> spat_cor_netw_DT<span class="sc">$</span>cor_clusters<span class="sc">$</span>spat_netw_clus)</span>
<span id="cb88-4"><a href="visium-part-ii.html#cb88-4" tabindex="-1"></a></span>
<span id="cb88-5"><a href="visium-part-ii.html#cb88-5" tabindex="-1"></a>data.table<span class="sc">::</span><span class="fu">setorder</span>(coexpr_dt, cluster)</span>
<span id="cb88-6"><a href="visium-part-ii.html#cb88-6" tabindex="-1"></a></span>
<span id="cb88-7"><a href="visium-part-ii.html#cb88-7" tabindex="-1"></a>top30_coexpr_dt <span class="ot">&lt;-</span> coexpr_dt[, <span class="fu">head</span>(.SD, <span class="dv">30</span>) , by <span class="ot">=</span> cluster]</span>
<span id="cb88-8"><a href="visium-part-ii.html#cb88-8" tabindex="-1"></a></span>
<span id="cb88-9"><a href="visium-part-ii.html#cb88-9" tabindex="-1"></a>spatial_genes <span class="ot">&lt;-</span> top30_coexpr_dt<span class="sc">$</span>genes</span></code></pre></div>
<ul>
<li>Re-calculate the clustering</li>
</ul>
<div class="sourceCode" id="cb89"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb89-1"><a href="visium-part-ii.html#cb89-1" tabindex="-1"></a>visium_brain <span class="ot">&lt;-</span> <span class="fu">runPCA</span>(<span class="at">gobject =</span> visium_brain,</span>
<span id="cb89-2"><a href="visium-part-ii.html#cb89-2" tabindex="-1"></a>                       <span class="at">feats_to_use =</span> spatial_genes,</span>
<span id="cb89-3"><a href="visium-part-ii.html#cb89-3" tabindex="-1"></a>                       <span class="at">name =</span> <span class="st">&quot;custom_pca&quot;</span>)</span>
<span id="cb89-4"><a href="visium-part-ii.html#cb89-4" tabindex="-1"></a></span>
<span id="cb89-5"><a href="visium-part-ii.html#cb89-5" tabindex="-1"></a>visium_brain <span class="ot">&lt;-</span> <span class="fu">runUMAP</span>(visium_brain, </span>
<span id="cb89-6"><a href="visium-part-ii.html#cb89-6" tabindex="-1"></a>                        <span class="at">dim_reduction_name =</span> <span class="st">&quot;custom_pca&quot;</span>, </span>
<span id="cb89-7"><a href="visium-part-ii.html#cb89-7" tabindex="-1"></a>                        <span class="at">dimensions_to_use =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>,</span>
<span id="cb89-8"><a href="visium-part-ii.html#cb89-8" tabindex="-1"></a>                        <span class="at">name =</span> <span class="st">&quot;custom_umap&quot;</span>)</span>
<span id="cb89-9"><a href="visium-part-ii.html#cb89-9" tabindex="-1"></a></span>
<span id="cb89-10"><a href="visium-part-ii.html#cb89-10" tabindex="-1"></a>visium_brain <span class="ot">&lt;-</span> <span class="fu">createNearestNetwork</span>(<span class="at">gobject =</span> visium_brain,</span>
<span id="cb89-11"><a href="visium-part-ii.html#cb89-11" tabindex="-1"></a>                                     <span class="at">dim_reduction_name =</span> <span class="st">&quot;custom_pca&quot;</span>,</span>
<span id="cb89-12"><a href="visium-part-ii.html#cb89-12" tabindex="-1"></a>                                     <span class="at">dimensions_to_use =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>, </span>
<span id="cb89-13"><a href="visium-part-ii.html#cb89-13" tabindex="-1"></a>                                     <span class="at">k =</span> <span class="dv">5</span>,</span>
<span id="cb89-14"><a href="visium-part-ii.html#cb89-14" tabindex="-1"></a>                                     <span class="at">name =</span> <span class="st">&quot;custom_NN&quot;</span>)</span>
<span id="cb89-15"><a href="visium-part-ii.html#cb89-15" tabindex="-1"></a></span>
<span id="cb89-16"><a href="visium-part-ii.html#cb89-16" tabindex="-1"></a>visium_brain <span class="ot">&lt;-</span> <span class="fu">doLeidenCluster</span>(<span class="at">gobject =</span> visium_brain, </span>
<span id="cb89-17"><a href="visium-part-ii.html#cb89-17" tabindex="-1"></a>                                <span class="at">network_name =</span> <span class="st">&quot;custom_NN&quot;</span>,</span>
<span id="cb89-18"><a href="visium-part-ii.html#cb89-18" tabindex="-1"></a>                                <span class="at">resolution =</span> <span class="fl">0.15</span>, </span>
<span id="cb89-19"><a href="visium-part-ii.html#cb89-19" tabindex="-1"></a>                                <span class="at">n_iterations =</span> <span class="dv">1000</span>,</span>
<span id="cb89-20"><a href="visium-part-ii.html#cb89-20" tabindex="-1"></a>                                <span class="at">name =</span> <span class="st">&quot;custom_leiden&quot;</span>)</span></code></pre></div>
<ul>
<li>Visualize</li>
</ul>
<div class="sourceCode" id="cb90"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb90-1"><a href="visium-part-ii.html#cb90-1" tabindex="-1"></a><span class="fu">spatPlot2D</span>(visium_brain, </span>
<span id="cb90-2"><a href="visium-part-ii.html#cb90-2" tabindex="-1"></a>           <span class="at">cell_color =</span> <span class="st">&quot;custom_leiden&quot;</span>)</span></code></pre></div>
<p><img src="img/02_session1/15-spatPlot2D.png" width="100%" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb91"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb91-1"><a href="visium-part-ii.html#cb91-1" tabindex="-1"></a><span class="fu">plotUMAP</span>(<span class="at">gobject =</span> visium_brain, </span>
<span id="cb91-2"><a href="visium-part-ii.html#cb91-2" tabindex="-1"></a>         <span class="at">cell_color =</span> <span class="st">&quot;custom_leiden&quot;</span>, </span>
<span id="cb91-3"><a href="visium-part-ii.html#cb91-3" tabindex="-1"></a>         <span class="at">point_size =</span> <span class="fl">1.5</span>)</span></code></pre></div>
<p><img src="img/02_session1/16-UMAP.png" width="100%" style="display: block; margin: auto;" /></p>
</div>
<div id="spatial-domains-hmrf" class="section level2 hasAnchor" number="8.6">
<h2><span class="header-section-number">8.6</span> Spatial domains HMRF<a href="visium-part-ii.html#spatial-domains-hmrf" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Hidden Markov Random Field (HMRF) models capture spatial dependencies and segment tissue regions based on shared gene expression patterns.</p>
<ul>
<li>Do HMRF with different betas on top 30 genes per spatial co-expression module</li>
</ul>
<div class="sourceCode" id="cb92"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb92-1"><a href="visium-part-ii.html#cb92-1" tabindex="-1"></a>HMRF_spatial_genes <span class="ot">&lt;-</span> <span class="fu">doHMRF</span>(<span class="at">gobject =</span> visium_brain,</span>
<span id="cb92-2"><a href="visium-part-ii.html#cb92-2" tabindex="-1"></a>                            <span class="at">expression_values =</span> <span class="st">&quot;scaled&quot;</span>,</span>
<span id="cb92-3"><a href="visium-part-ii.html#cb92-3" tabindex="-1"></a>                            <span class="at">spatial_genes =</span> spatial_genes, </span>
<span id="cb92-4"><a href="visium-part-ii.html#cb92-4" tabindex="-1"></a>                            <span class="at">k =</span> <span class="dv">20</span>,</span>
<span id="cb92-5"><a href="visium-part-ii.html#cb92-5" tabindex="-1"></a>                            <span class="at">spatial_network_name =</span> <span class="st">&quot;spatial_network&quot;</span>,</span>
<span id="cb92-6"><a href="visium-part-ii.html#cb92-6" tabindex="-1"></a>                            <span class="at">betas =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">10</span>, <span class="dv">5</span>),</span>
<span id="cb92-7"><a href="visium-part-ii.html#cb92-7" tabindex="-1"></a>                            <span class="at">output_folder =</span> <span class="st">&quot;11_HMRF/&quot;</span>)</span>
<span id="cb92-8"><a href="visium-part-ii.html#cb92-8" tabindex="-1"></a></span>
<span id="cb92-9"><a href="visium-part-ii.html#cb92-9" tabindex="-1"></a>visium_brain <span class="ot">&lt;-</span> <span class="fu">addHMRF</span>(<span class="at">gobject =</span> visium_brain, </span>
<span id="cb92-10"><a href="visium-part-ii.html#cb92-10" tabindex="-1"></a>                        <span class="at">HMRFoutput =</span> HMRF_spatial_genes,</span>
<span id="cb92-11"><a href="visium-part-ii.html#cb92-11" tabindex="-1"></a>                        <span class="at">k =</span> <span class="dv">20</span>, </span>
<span id="cb92-12"><a href="visium-part-ii.html#cb92-12" tabindex="-1"></a>                        <span class="at">betas_to_add =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">10</span>, <span class="dv">20</span>, <span class="dv">30</span>, <span class="dv">40</span>),</span>
<span id="cb92-13"><a href="visium-part-ii.html#cb92-13" tabindex="-1"></a>                        <span class="at">hmrf_name =</span> <span class="st">&quot;HMRF&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb93"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb93-1"><a href="visium-part-ii.html#cb93-1" tabindex="-1"></a><span class="fu">spatPlot2D</span>(<span class="at">gobject =</span> visium_brain, </span>
<span id="cb93-2"><a href="visium-part-ii.html#cb93-2" tabindex="-1"></a>           <span class="at">cell_color =</span> <span class="st">&quot;HMRF_k20_b.40&quot;</span>)</span></code></pre></div>
<ul>
<li>Visualize</li>
</ul>
<p><img src="img/02_session1/17-spatPlot2D.png" width="100%" style="display: block; margin: auto;" /></p>
</div>
<div id="interactive-tools" class="section level2 hasAnchor" number="8.7">
<h2><span class="header-section-number">8.7</span> Interactive tools<a href="visium-part-ii.html#interactive-tools" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ul>
<li>Create a spatial plot</li>
</ul>
<div class="sourceCode" id="cb94"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb94-1"><a href="visium-part-ii.html#cb94-1" tabindex="-1"></a>brain_spatPlot <span class="ot">&lt;-</span> <span class="fu">spatPlot2D</span>(<span class="at">gobject =</span> visium_brain,</span>
<span id="cb94-2"><a href="visium-part-ii.html#cb94-2" tabindex="-1"></a>                          <span class="at">cell_color =</span> <span class="st">&quot;leiden_clus&quot;</span>,</span>
<span id="cb94-3"><a href="visium-part-ii.html#cb94-3" tabindex="-1"></a>                          <span class="at">show_image =</span> <span class="cn">FALSE</span>,</span>
<span id="cb94-4"><a href="visium-part-ii.html#cb94-4" tabindex="-1"></a>                          <span class="at">return_plot =</span> <span class="cn">TRUE</span>,</span>
<span id="cb94-5"><a href="visium-part-ii.html#cb94-5" tabindex="-1"></a>                          <span class="at">point_size =</span> <span class="dv">1</span>)</span>
<span id="cb94-6"><a href="visium-part-ii.html#cb94-6" tabindex="-1"></a></span>
<span id="cb94-7"><a href="visium-part-ii.html#cb94-7" tabindex="-1"></a>brain_spatPlot</span></code></pre></div>
<ul>
<li>Run the Shiny app</li>
</ul>
<div class="sourceCode" id="cb95"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb95-1"><a href="visium-part-ii.html#cb95-1" tabindex="-1"></a><span class="fu">plotInteractivePolygons</span>(brain_spatPlot)</span></code></pre></div>
<p><img src="img/02_session1/18-spatPlot2D.png" width="100%" style="display: block; margin: auto;" /></p>
<ul>
<li>Select the regions of interest and save the coordinates</li>
</ul>
<div class="sourceCode" id="cb96"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb96-1"><a href="visium-part-ii.html#cb96-1" tabindex="-1"></a>polygon_coordinates <span class="ot">&lt;-</span> <span class="fu">plotInteractivePolygons</span>(brain_spatPlot)</span></code></pre></div>
<p><img src="img/02_session1/19-spatPlot2D.png" width="60%" style="display: block; margin: auto;" /></p>
<ul>
<li>Transform the data.table or data.frame with coordinates into a Giotto polygon object</li>
</ul>
<div class="sourceCode" id="cb97"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb97-1"><a href="visium-part-ii.html#cb97-1" tabindex="-1"></a>giotto_polygons <span class="ot">&lt;-</span> <span class="fu">createGiottoPolygonsFromDfr</span>(polygon_coordinates, </span>
<span id="cb97-2"><a href="visium-part-ii.html#cb97-2" tabindex="-1"></a>                                               <span class="at">name =</span> <span class="st">&quot;selections&quot;</span>,</span>
<span id="cb97-3"><a href="visium-part-ii.html#cb97-3" tabindex="-1"></a>                                               <span class="at">calc_centroids =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<ul>
<li>Add the polygons (or spatial units) to the Giotto object</li>
</ul>
<div class="sourceCode" id="cb98"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb98-1"><a href="visium-part-ii.html#cb98-1" tabindex="-1"></a>visium_brain <span class="ot">&lt;-</span> <span class="fu">addGiottoPolygons</span>(<span class="at">gobject =</span> visium_brain,</span>
<span id="cb98-2"><a href="visium-part-ii.html#cb98-2" tabindex="-1"></a>                                  <span class="at">gpolygons =</span> <span class="fu">list</span>(giotto_polygons))</span></code></pre></div>
<ul>
<li>Add the corresponding polygon IDs to the cell metadata</li>
</ul>
<div class="sourceCode" id="cb99"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb99-1"><a href="visium-part-ii.html#cb99-1" tabindex="-1"></a>visium_brain <span class="ot">&lt;-</span> <span class="fu">addPolygonCells</span>(visium_brain,</span>
<span id="cb99-2"><a href="visium-part-ii.html#cb99-2" tabindex="-1"></a>                                <span class="at">polygon_name =</span> <span class="st">&quot;selections&quot;</span>)</span></code></pre></div>
<ul>
<li>Extract the coordinates and IDs from cells located within one or multiple regions of interest.</li>
</ul>
<div class="sourceCode" id="cb100"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb100-1"><a href="visium-part-ii.html#cb100-1" tabindex="-1"></a><span class="fu">getCellsFromPolygon</span>(visium_brain,</span>
<span id="cb100-2"><a href="visium-part-ii.html#cb100-2" tabindex="-1"></a>                    <span class="at">polygon_name =</span> <span class="st">&quot;selections&quot;</span>,</span>
<span id="cb100-3"><a href="visium-part-ii.html#cb100-3" tabindex="-1"></a>                    <span class="at">polygons =</span> <span class="st">&quot;polygon 1&quot;</span>)</span></code></pre></div>
<p>If no polygon name is provided, the function will retrieve cells located within all polygons</p>
<div class="sourceCode" id="cb101"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb101-1"><a href="visium-part-ii.html#cb101-1" tabindex="-1"></a><span class="fu">getCellsFromPolygon</span>(visium_brain,</span>
<span id="cb101-2"><a href="visium-part-ii.html#cb101-2" tabindex="-1"></a>                    <span class="at">polygon_name =</span> <span class="st">&quot;selections&quot;</span>)</span></code></pre></div>
<ul>
<li>Compare the expression levels of some genes of interest between the selected regions</li>
</ul>
<div class="sourceCode" id="cb102"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb102-1"><a href="visium-part-ii.html#cb102-1" tabindex="-1"></a><span class="fu">comparePolygonExpression</span>(visium_brain,</span>
<span id="cb102-2"><a href="visium-part-ii.html#cb102-2" tabindex="-1"></a>                         <span class="at">selected_feats =</span> <span class="fu">c</span>(<span class="st">&quot;Stmn1&quot;</span>, <span class="st">&quot;Psd&quot;</span>, <span class="st">&quot;Ly6h&quot;</span>))</span></code></pre></div>
<p><img src="img/02_session1/20-heatmap.png" width="60%" style="display: block; margin: auto;" /></p>
<ul>
<li>Calculate the top genes expressed within each region, then provide the result to compare polygons</li>
</ul>
<div class="sourceCode" id="cb103"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb103-1"><a href="visium-part-ii.html#cb103-1" tabindex="-1"></a>scran_results <span class="ot">&lt;-</span> <span class="fu">findMarkers_one_vs_all</span>(visium_brain,</span>
<span id="cb103-2"><a href="visium-part-ii.html#cb103-2" tabindex="-1"></a>                                        <span class="at">spat_unit =</span> <span class="st">&quot;cell&quot;</span>,</span>
<span id="cb103-3"><a href="visium-part-ii.html#cb103-3" tabindex="-1"></a>                                        <span class="at">feat_type =</span> <span class="st">&quot;rna&quot;</span>,</span>
<span id="cb103-4"><a href="visium-part-ii.html#cb103-4" tabindex="-1"></a>                                        <span class="at">method =</span> <span class="st">&quot;scran&quot;</span>,</span>
<span id="cb103-5"><a href="visium-part-ii.html#cb103-5" tabindex="-1"></a>                                        <span class="at">expression_values =</span> <span class="st">&quot;normalized&quot;</span>,</span>
<span id="cb103-6"><a href="visium-part-ii.html#cb103-6" tabindex="-1"></a>                                        <span class="at">cluster_column =</span> <span class="st">&quot;selections&quot;</span>,</span>
<span id="cb103-7"><a href="visium-part-ii.html#cb103-7" tabindex="-1"></a>                                        <span class="at">min_feats =</span> <span class="dv">2</span>)</span>
<span id="cb103-8"><a href="visium-part-ii.html#cb103-8" tabindex="-1"></a></span>
<span id="cb103-9"><a href="visium-part-ii.html#cb103-9" tabindex="-1"></a>top_genes <span class="ot">&lt;-</span> scran_results[, <span class="fu">head</span>(.SD, <span class="dv">2</span>), by <span class="ot">=</span> <span class="st">&quot;cluster&quot;</span>]<span class="sc">$</span>feats</span>
<span id="cb103-10"><a href="visium-part-ii.html#cb103-10" tabindex="-1"></a></span>
<span id="cb103-11"><a href="visium-part-ii.html#cb103-11" tabindex="-1"></a><span class="fu">comparePolygonExpression</span>(visium_brain,</span>
<span id="cb103-12"><a href="visium-part-ii.html#cb103-12" tabindex="-1"></a>                         <span class="at">selected_feats =</span> top_genes)</span></code></pre></div>
<p><img src="img/02_session1/21-heatmap.png" width="60%" style="display: block; margin: auto;" /></p>
<ul>
<li>Compare the abundance of cell types between the selected regions</li>
</ul>
<div class="sourceCode" id="cb104"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb104-1"><a href="visium-part-ii.html#cb104-1" tabindex="-1"></a><span class="fu">compareCellAbundance</span>(visium_brain)</span></code></pre></div>
<p><img src="img/02_session1/22-heatmap.png" width="60%" style="display: block; margin: auto;" /></p>
<ul>
<li>Use other columns within the cell metadata table to compare the cell type abundances</li>
</ul>
<div class="sourceCode" id="cb105"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb105-1"><a href="visium-part-ii.html#cb105-1" tabindex="-1"></a><span class="fu">compareCellAbundance</span>(visium_brain, </span>
<span id="cb105-2"><a href="visium-part-ii.html#cb105-2" tabindex="-1"></a>                     <span class="at">cell_type_column =</span> <span class="st">&quot;custom_leiden&quot;</span>)</span></code></pre></div>
<p><img src="img/02_session1/23-heatmap.png" width="60%" style="display: block; margin: auto;" /></p>
<ul>
<li>Use the spatPlot arguments to isolate and plot each region.</li>
</ul>
<div class="sourceCode" id="cb106"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb106-1"><a href="visium-part-ii.html#cb106-1" tabindex="-1"></a><span class="fu">spatPlot2D</span>(visium_brain,</span>
<span id="cb106-2"><a href="visium-part-ii.html#cb106-2" tabindex="-1"></a>           <span class="at">cell_color =</span> <span class="st">&quot;leiden_clus&quot;</span>,</span>
<span id="cb106-3"><a href="visium-part-ii.html#cb106-3" tabindex="-1"></a>           <span class="at">group_by =</span> <span class="st">&quot;selections&quot;</span>,</span>
<span id="cb106-4"><a href="visium-part-ii.html#cb106-4" tabindex="-1"></a>           <span class="at">cow_n_col =</span> <span class="dv">3</span>,</span>
<span id="cb106-5"><a href="visium-part-ii.html#cb106-5" tabindex="-1"></a>           <span class="at">point_size =</span> <span class="dv">2</span>,</span>
<span id="cb106-6"><a href="visium-part-ii.html#cb106-6" tabindex="-1"></a>           <span class="at">show_legend =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<p><img src="img/02_session1/24-spatPlot2D.png" width="80%" style="display: block; margin: auto;" /></p>
<ul>
<li>Color each cell by cluster, cell type or expression level.</li>
</ul>
<div class="sourceCode" id="cb107"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb107-1"><a href="visium-part-ii.html#cb107-1" tabindex="-1"></a><span class="fu">spatFeatPlot2D</span>(visium_brain,</span>
<span id="cb107-2"><a href="visium-part-ii.html#cb107-2" tabindex="-1"></a>               <span class="at">expression_values =</span> <span class="st">&quot;scaled&quot;</span>,</span>
<span id="cb107-3"><a href="visium-part-ii.html#cb107-3" tabindex="-1"></a>               <span class="at">group_by =</span> <span class="st">&quot;selections&quot;</span>,</span>
<span id="cb107-4"><a href="visium-part-ii.html#cb107-4" tabindex="-1"></a>               <span class="at">feats =</span> <span class="st">&quot;Psd&quot;</span>,</span>
<span id="cb107-5"><a href="visium-part-ii.html#cb107-5" tabindex="-1"></a>               <span class="at">point_size =</span> <span class="dv">2</span>)</span></code></pre></div>
<p><img src="img/02_session1/25-spatFeatPlot2D.png" width="80%" style="display: block; margin: auto;" /></p>
<ul>
<li>Plot again the polygons</li>
</ul>
<div class="sourceCode" id="cb108"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb108-1"><a href="visium-part-ii.html#cb108-1" tabindex="-1"></a><span class="fu">plotPolygons</span>(visium_brain,</span>
<span id="cb108-2"><a href="visium-part-ii.html#cb108-2" tabindex="-1"></a>             <span class="at">polygon_name =</span> <span class="st">&quot;selections&quot;</span>,</span>
<span id="cb108-3"><a href="visium-part-ii.html#cb108-3" tabindex="-1"></a>             <span class="at">x =</span> brain_spatPlot)</span></code></pre></div>
<p><img src="img/02_session1/26-spatPlot2D.png" width="60%" style="display: block; margin: auto;" /></p>
</div>
<div id="session-info-1" class="section level2 hasAnchor" number="8.8">
<h2><span class="header-section-number">8.8</span> Session info<a href="visium-part-ii.html#session-info-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb109"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb109-1"><a href="visium-part-ii.html#cb109-1" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code></pre></div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="visium-part-i.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="visium-hd.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
