<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>9 Visium HD | Workshop: Spatial multi-omics data analysis with Giotto Suite</title>
  <meta name="description" content="9 Visium HD | Workshop: Spatial multi-omics data analysis with Giotto Suite" />
  <meta name="generator" content="bookdown 0.39 and GitBook 2.6.7" />

  <meta property="og:title" content="9 Visium HD | Workshop: Spatial multi-omics data analysis with Giotto Suite" />
  <meta property="og:type" content="book" />
  
  
  <meta name="github-repo" content="drieslab/giotto_workshop_2024" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="9 Visium HD | Workshop: Spatial multi-omics data analysis with Giotto Suite" />
  
  
  

<meta name="author" content="Ruben Dries, Jiaji George Chen, Joselyn Cristina ChÃ¡vez-Fuentes, Junxiang Xu ,Edward Ruiz, Jeff Sheridan, Iqra Amin, Wen Wang" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="visium-part-ii.html"/>
<link rel="next" href="xenium-1.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Giotto Suite Workshop 2024</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#instructors"><i class="fa fa-check"></i><b>1.1</b> Instructors</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#topics-and-schedule"><i class="fa fa-check"></i><b>1.2</b> Topics and Schedule:</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#license"><i class="fa fa-check"></i><b>1.3</b> License</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="datasets-packages.html"><a href="datasets-packages.html"><i class="fa fa-check"></i><b>2</b> Datasets &amp; Packages</a>
<ul>
<li class="chapter" data-level="2.1" data-path="datasets-packages.html"><a href="datasets-packages.html#datasets-to-download"><i class="fa fa-check"></i><b>2.1</b> Datasets to download</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="datasets-packages.html"><a href="datasets-packages.html#zenodo-repository"><i class="fa fa-check"></i><b>2.1.1</b> Zenodo repository</a></li>
<li class="chapter" data-level="2.1.2" data-path="datasets-packages.html"><a href="datasets-packages.html#x-genomics-visium-mouse-brain-section-coronal-dataset"><i class="fa fa-check"></i><b>2.1.2</b> 10X Genomics Visium Mouse Brain Section (Coronal) dataset</a></li>
<li class="chapter" data-level="2.1.3" data-path="datasets-packages.html"><a href="datasets-packages.html#x-genomics-visium-hd-ffpe-human-colon-cancer"><i class="fa fa-check"></i><b>2.1.3</b> 10X Genomics Visium HD: FFPE Human Colon Cancer</a></li>
<li class="chapter" data-level="2.1.4" data-path="datasets-packages.html"><a href="datasets-packages.html#x-genomics-multi-modal-dataset"><i class="fa fa-check"></i><b>2.1.4</b> 10X Genomics multi-modal dataset</a></li>
<li class="chapter" data-level="2.1.5" data-path="datasets-packages.html"><a href="datasets-packages.html#x-genomics-multi-omics-visium-cytassist-human-tonsil-dataset"><i class="fa fa-check"></i><b>2.1.5</b> 10X Genomics multi-omics Visium CytAssist Human Tonsil dataset</a></li>
<li class="chapter" data-level="2.1.6" data-path="datasets-packages.html"><a href="datasets-packages.html#x-genomics-human-prostate-cancer-adenocarcinoma-with-invasive-carcinoma-ffpe"><i class="fa fa-check"></i><b>2.1.6</b> 10X Genomics Human Prostate Cancer Adenocarcinoma with Invasive Carcinoma (FFPE)</a></li>
<li class="chapter" data-level="2.1.7" data-path="datasets-packages.html"><a href="datasets-packages.html#x-genomics-normal-human-prostate-ffpe"><i class="fa fa-check"></i><b>2.1.7</b> 10X Genomics Normal Human Prostate (FFPE)</a></li>
<li class="chapter" data-level="2.1.8" data-path="datasets-packages.html"><a href="datasets-packages.html#xenium"><i class="fa fa-check"></i><b>2.1.8</b> Xenium</a></li>
<li class="chapter" data-level="2.1.9" data-path="datasets-packages.html"><a href="datasets-packages.html#merfish-cortex-dataset"><i class="fa fa-check"></i><b>2.1.9</b> MERFISH cortex dataset</a></li>
<li class="chapter" data-level="2.1.10" data-path="datasets-packages.html"><a href="datasets-packages.html#lunaphore-if-dataset"><i class="fa fa-check"></i><b>2.1.10</b> Lunaphore IF dataset</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="datasets-packages.html"><a href="datasets-packages.html#needed-packages"><i class="fa fa-check"></i><b>2.2</b> Needed packages</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="datasets-packages.html"><a href="datasets-packages.html#cran-dependencies"><i class="fa fa-check"></i><b>2.2.1</b> CRAN dependencies:</a></li>
<li class="chapter" data-level="2.2.2" data-path="datasets-packages.html"><a href="datasets-packages.html#terra-installation"><i class="fa fa-check"></i><b>2.2.2</b> <em>terra</em> installation</a></li>
<li class="chapter" data-level="2.2.3" data-path="datasets-packages.html"><a href="datasets-packages.html#matrix-installation"><i class="fa fa-check"></i><b>2.2.3</b> <em>Matrix</em> installation</a></li>
<li class="chapter" data-level="2.2.4" data-path="datasets-packages.html"><a href="datasets-packages.html#rtools-installation"><i class="fa fa-check"></i><b>2.2.4</b> <em>Rtools</em> installation</a></li>
<li class="chapter" data-level="2.2.5" data-path="datasets-packages.html"><a href="datasets-packages.html#giotto-installation"><i class="fa fa-check"></i><b>2.2.5</b> <em>Giotto</em> installation</a></li>
<li class="chapter" data-level="2.2.6" data-path="datasets-packages.html"><a href="datasets-packages.html#irlba-install"><i class="fa fa-check"></i><b>2.2.6</b> <em>irlba</em> install</a></li>
<li class="chapter" data-level="2.2.7" data-path="datasets-packages.html"><a href="datasets-packages.html#arrow-install"><i class="fa fa-check"></i><b>2.2.7</b> <em>arrow</em> install</a></li>
<li class="chapter" data-level="2.2.8" data-path="datasets-packages.html"><a href="datasets-packages.html#bioconductor-dependencies"><i class="fa fa-check"></i><b>2.2.8</b> Bioconductor dependencies:</a></li>
<li class="chapter" data-level="2.2.9" data-path="datasets-packages.html"><a href="datasets-packages.html#cran-packages"><i class="fa fa-check"></i><b>2.2.9</b> CRAN packages:</a></li>
<li class="chapter" data-level="2.2.10" data-path="datasets-packages.html"><a href="datasets-packages.html#packages-from-github"><i class="fa fa-check"></i><b>2.2.10</b> Packages from GitHub</a></li>
<li class="chapter" data-level="2.2.11" data-path="datasets-packages.html"><a href="datasets-packages.html#python-environments"><i class="fa fa-check"></i><b>2.2.11</b> Python environments</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html"><i class="fa fa-check"></i><b>3</b> Spatial omics technologies</a>
<ul>
<li class="chapter" data-level="3.1" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#presentation"><i class="fa fa-check"></i><b>3.1</b> Presentation</a></li>
<li class="chapter" data-level="3.2" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#short-summary"><i class="fa fa-check"></i><b>3.2</b> Short summary</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#why-do-we-need-spatial-omics-technologies"><i class="fa fa-check"></i><b>3.2.1</b> Why do we need spatial omics technologies?</a></li>
<li class="chapter" data-level="3.2.2" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#what-is-spatial-omics"><i class="fa fa-check"></i><b>3.2.2</b> What is spatial omics?</a></li>
<li class="chapter" data-level="3.2.3" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#what-are-the-main-spatial-omics-technologies"><i class="fa fa-check"></i><b>3.2.3</b> What are the main spatial omics technologies?</a></li>
<li class="chapter" data-level="3.2.4" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#other-spatial-omics-atac-seq-cuttag-lipidomics-etc"><i class="fa fa-check"></i><b>3.2.4</b> Other Spatial omics: ATAC-seq, CUT&amp;Tag, lipidomics, etc</a></li>
<li class="chapter" data-level="3.2.5" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#what-are-the-different-types-of-spatial-downstream-analyses"><i class="fa fa-check"></i><b>3.2.5</b> What are the different types of spatial downstream analyses?</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html"><i class="fa fa-check"></i><b>4</b> Introduction to the Giotto package</a>
<ul>
<li class="chapter" data-level="4.1" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html#presentation-1"><i class="fa fa-check"></i><b>4.1</b> Presentation</a></li>
<li class="chapter" data-level="4.2" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html#ecosystem"><i class="fa fa-check"></i><b>4.2</b> Ecosystem</a></li>
<li class="chapter" data-level="4.3" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html#installation-python-environment"><i class="fa fa-check"></i><b>4.3</b> Installation + python environment</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html#giotto-installation-1"><i class="fa fa-check"></i><b>4.3.1</b> Giotto installation</a></li>
<li class="chapter" data-level="4.3.2" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html#python-environment"><i class="fa fa-check"></i><b>4.3.2</b> Python environment</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html#giotto-instructions"><i class="fa fa-check"></i><b>4.4</b> Giotto instructions</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html"><i class="fa fa-check"></i><b>5</b> Data formatting and Pre-processing</a>
<ul>
<li class="chapter" data-level="5.1" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#data-formats"><i class="fa fa-check"></i><b>5.1</b> Data formats</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#general-formats"><i class="fa fa-check"></i><b>5.1.1</b> General formats</a></li>
<li class="chapter" data-level="5.1.2" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#matrix-formats"><i class="fa fa-check"></i><b>5.1.2</b> Matrix formats</a></li>
<li class="chapter" data-level="5.1.3" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#tabular-formats"><i class="fa fa-check"></i><b>5.1.3</b> Tabular formats</a></li>
<li class="chapter" data-level="5.1.4" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#spatial-formats"><i class="fa fa-check"></i><b>5.1.4</b> Spatial formats</a></li>
<li class="chapter" data-level="5.1.5" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#mask-files"><i class="fa fa-check"></i><b>5.1.5</b> Mask files</a></li>
<li class="chapter" data-level="5.1.6" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#images"><i class="fa fa-check"></i><b>5.1.6</b> images</a></li>
<li class="chapter" data-level="5.1.7" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#jsonlike-formats"><i class="fa fa-check"></i><b>5.1.7</b> jsonlike formats</a></li>
<li class="chapter" data-level="5.1.8" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#hierarchical-formats"><i class="fa fa-check"></i><b>5.1.8</b> Hierarchical formats</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#pre-processing"><i class="fa fa-check"></i><b>5.2</b> Pre-processing</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#expression-matrix"><i class="fa fa-check"></i><b>5.2.1</b> Expression matrix</a></li>
<li class="chapter" data-level="5.2.2" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#spatial-locations"><i class="fa fa-check"></i><b>5.2.2</b> Spatial locations</a></li>
<li class="chapter" data-level="5.2.3" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#giottopoints"><i class="fa fa-check"></i><b>5.2.3</b> giottoPoints</a></li>
<li class="chapter" data-level="5.2.4" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#giottopolygon"><i class="fa fa-check"></i><b>5.2.4</b> giottoPolygon</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#subobject-utility-functions"><i class="fa fa-check"></i><b>5.3</b> Subobject utility functions</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#ids"><i class="fa fa-check"></i><b>5.3.1</b> IDs</a></li>
<li class="chapter" data-level="5.3.2" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#bracket-subsetting-and-extraction"><i class="fa fa-check"></i><b>5.3.2</b> Bracket subsetting and extraction</a></li>
<li class="chapter" data-level="5.3.3" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#nesting-metadata-generics"><i class="fa fa-check"></i><b>5.3.3</b> Nesting metadata generics</a></li>
<li class="chapter" data-level="5.3.4" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#appending-to-a-giotto-object"><i class="fa fa-check"></i><b>5.3.4</b> Appending to a giotto object</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html"><i class="fa fa-check"></i><b>6</b> Creating a Giotto object</a>
<ul>
<li class="chapter" data-level="6.1" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#overview"><i class="fa fa-check"></i><b>6.1</b> Overview</a></li>
<li class="chapter" data-level="6.2" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#giottodata-modular-package"><i class="fa fa-check"></i><b>6.2</b> GiottoData modular package</a></li>
<li class="chapter" data-level="6.3" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#from-matrix-locations"><i class="fa fa-check"></i><b>6.3</b> From matrix + locations</a></li>
<li class="chapter" data-level="6.4" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#from-subcellular-raw-data-transcripts-or-images-polygons"><i class="fa fa-check"></i><b>6.4</b> From subcellular raw data (transcripts or images) + polygons</a></li>
<li class="chapter" data-level="6.5" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#from-piece-wise"><i class="fa fa-check"></i><b>6.5</b> From piece-wise</a></li>
<li class="chapter" data-level="6.6" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#using-convenience-functions-for-popular-technologies-vizgen-xenium-cosmx"><i class="fa fa-check"></i><b>6.6</b> Using convenience functions for popular technologies (Vizgen, Xenium, CosMx, â¦)</a></li>
<li class="chapter" data-level="6.7" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#plotting"><i class="fa fa-check"></i><b>6.7</b> Plotting</a>
<ul>
<li class="chapter" data-level="6.7.1" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#subobject-plotting"><i class="fa fa-check"></i><b>6.7.1</b> Subobject plotting</a></li>
<li class="chapter" data-level="6.7.2" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#additive-subobject-plotting"><i class="fa fa-check"></i><b>6.7.2</b> Additive subobject plotting</a></li>
<li class="chapter" data-level="6.7.3" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#giotto-object-plotting"><i class="fa fa-check"></i><b>6.7.3</b> Giotto object plotting</a></li>
</ul></li>
<li class="chapter" data-level="6.8" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#subsetting"><i class="fa fa-check"></i><b>6.8</b> Subsetting</a></li>
<li class="chapter" data-level="6.9" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#giottodata-mini-objects"><i class="fa fa-check"></i><b>6.9</b> <em>GiottoData</em> Mini objects</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="visium-part-i.html"><a href="visium-part-i.html"><i class="fa fa-check"></i><b>7</b> Visium Part I</a>
<ul>
<li class="chapter" data-level="7.1" data-path="visium-part-i.html"><a href="visium-part-i.html#the-visium-technology"><i class="fa fa-check"></i><b>7.1</b> The Visium technology</a></li>
<li class="chapter" data-level="7.2" data-path="visium-part-i.html"><a href="visium-part-i.html#introduction-to-the-spatial-dataset"><i class="fa fa-check"></i><b>7.2</b> Introduction to the spatial dataset</a></li>
<li class="chapter" data-level="7.3" data-path="visium-part-i.html"><a href="visium-part-i.html#download-dataset"><i class="fa fa-check"></i><b>7.3</b> Download dataset</a></li>
<li class="chapter" data-level="7.4" data-path="visium-part-i.html"><a href="visium-part-i.html#create-the-giotto-object"><i class="fa fa-check"></i><b>7.4</b> Create the Giotto object</a></li>
<li class="chapter" data-level="7.5" data-path="visium-part-i.html"><a href="visium-part-i.html#subset-on-spots-that-were-covered-by-tissue"><i class="fa fa-check"></i><b>7.5</b> Subset on spots that were covered by tissue</a></li>
<li class="chapter" data-level="7.6" data-path="visium-part-i.html"><a href="visium-part-i.html#quality-control"><i class="fa fa-check"></i><b>7.6</b> Quality control</a></li>
<li class="chapter" data-level="7.7" data-path="visium-part-i.html"><a href="visium-part-i.html#filtering"><i class="fa fa-check"></i><b>7.7</b> Filtering</a></li>
<li class="chapter" data-level="7.8" data-path="visium-part-i.html"><a href="visium-part-i.html#normalization"><i class="fa fa-check"></i><b>7.8</b> Normalization</a></li>
<li class="chapter" data-level="7.9" data-path="visium-part-i.html"><a href="visium-part-i.html#feature-selection"><i class="fa fa-check"></i><b>7.9</b> Feature selection</a>
<ul>
<li class="chapter" data-level="7.9.1" data-path="visium-part-i.html"><a href="visium-part-i.html#highly-variable-features"><i class="fa fa-check"></i><b>7.9.1</b> Highly Variable Features:</a></li>
</ul></li>
<li class="chapter" data-level="7.10" data-path="visium-part-i.html"><a href="visium-part-i.html#dimension-reduction"><i class="fa fa-check"></i><b>7.10</b> Dimension Reduction</a>
<ul>
<li class="chapter" data-level="7.10.1" data-path="visium-part-i.html"><a href="visium-part-i.html#pca"><i class="fa fa-check"></i><b>7.10.1</b> PCA</a></li>
<li class="chapter" data-level="7.10.2" data-path="visium-part-i.html"><a href="visium-part-i.html#umap"><i class="fa fa-check"></i><b>7.10.2</b> UMAP</a></li>
<li class="chapter" data-level="7.10.3" data-path="visium-part-i.html"><a href="visium-part-i.html#t-sne"><i class="fa fa-check"></i><b>7.10.3</b> t-SNE</a></li>
</ul></li>
<li class="chapter" data-level="7.11" data-path="visium-part-i.html"><a href="visium-part-i.html#clustering"><i class="fa fa-check"></i><b>7.11</b> Clustering</a>
<ul>
<li class="chapter" data-level="7.11.1" data-path="visium-part-i.html"><a href="visium-part-i.html#calculate-leiden-clustering"><i class="fa fa-check"></i><b>7.11.1</b> Calculate Leiden clustering</a></li>
<li class="chapter" data-level="7.11.2" data-path="visium-part-i.html"><a href="visium-part-i.html#calculate-louvain-clustering"><i class="fa fa-check"></i><b>7.11.2</b> Calculate Louvain clustering</a></li>
</ul></li>
<li class="chapter" data-level="7.12" data-path="visium-part-i.html"><a href="visium-part-i.html#save-the-object"><i class="fa fa-check"></i><b>7.12</b> Save the object</a></li>
<li class="chapter" data-level="7.13" data-path="visium-part-i.html"><a href="visium-part-i.html#session-info"><i class="fa fa-check"></i><b>7.13</b> Session info</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="visium-part-ii.html"><a href="visium-part-ii.html"><i class="fa fa-check"></i><b>8</b> Visium Part II</a>
<ul>
<li class="chapter" data-level="8.1" data-path="visium-part-ii.html"><a href="visium-part-ii.html#load-the-object"><i class="fa fa-check"></i><b>8.1</b> Load the object</a></li>
<li class="chapter" data-level="8.2" data-path="visium-part-ii.html"><a href="visium-part-ii.html#differential-expression"><i class="fa fa-check"></i><b>8.2</b> Differential expression</a>
<ul>
<li class="chapter" data-level="8.2.1" data-path="visium-part-ii.html"><a href="visium-part-ii.html#gini-markers"><i class="fa fa-check"></i><b>8.2.1</b> Gini markers</a></li>
<li class="chapter" data-level="8.2.2" data-path="visium-part-ii.html"><a href="visium-part-ii.html#scran-markers"><i class="fa fa-check"></i><b>8.2.2</b> Scran markers</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="visium-part-ii.html"><a href="visium-part-ii.html#enrichment-deconvolution"><i class="fa fa-check"></i><b>8.3</b> Enrichment &amp; Deconvolution</a>
<ul>
<li class="chapter" data-level="8.3.1" data-path="visium-part-ii.html"><a href="visium-part-ii.html#pagerank"><i class="fa fa-check"></i><b>8.3.1</b> PAGE/Rank</a></li>
<li class="chapter" data-level="8.3.2" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatialdwls"><i class="fa fa-check"></i><b>8.3.2</b> SpatialDWLS</a></li>
</ul></li>
<li class="chapter" data-level="8.4" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatial-expression-patterns"><i class="fa fa-check"></i><b>8.4</b> Spatial expression patterns</a>
<ul>
<li class="chapter" data-level="8.4.1" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatial-variable-genes"><i class="fa fa-check"></i><b>8.4.1</b> Spatial variable genes</a></li>
<li class="chapter" data-level="8.4.2" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatial-co-expression-modules"><i class="fa fa-check"></i><b>8.4.2</b> Spatial co-expression modules</a></li>
</ul></li>
<li class="chapter" data-level="8.5" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatially-informed-clusters"><i class="fa fa-check"></i><b>8.5</b> Spatially informed clusters</a></li>
<li class="chapter" data-level="8.6" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatial-domains-hmrf"><i class="fa fa-check"></i><b>8.6</b> Spatial domains HMRF</a></li>
<li class="chapter" data-level="8.7" data-path="visium-part-ii.html"><a href="visium-part-ii.html#interactive-tools"><i class="fa fa-check"></i><b>8.7</b> Interactive tools</a></li>
<li class="chapter" data-level="8.8" data-path="visium-part-ii.html"><a href="visium-part-ii.html#save-the-object-1"><i class="fa fa-check"></i><b>8.8</b> Save the object</a></li>
<li class="chapter" data-level="8.9" data-path="visium-part-ii.html"><a href="visium-part-ii.html#session-info-1"><i class="fa fa-check"></i><b>8.9</b> Session info</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="visium-hd.html"><a href="visium-hd.html"><i class="fa fa-check"></i><b>9</b> Visium HD</a>
<ul>
<li class="chapter" data-level="9.1" data-path="visium-hd.html"><a href="visium-hd.html#objective"><i class="fa fa-check"></i><b>9.1</b> Objective</a></li>
<li class="chapter" data-level="9.2" data-path="visium-hd.html"><a href="visium-hd.html#background"><i class="fa fa-check"></i><b>9.2</b> Background</a>
<ul>
<li class="chapter" data-level="9.2.1" data-path="visium-hd.html"><a href="visium-hd.html#visium-hd-technology"><i class="fa fa-check"></i><b>9.2.1</b> Visium HD Technology</a></li>
<li class="chapter" data-level="9.2.2" data-path="visium-hd.html"><a href="visium-hd.html#colorectal-cancer-sample"><i class="fa fa-check"></i><b>9.2.2</b> Colorectal Cancer Sample</a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="visium-hd.html"><a href="visium-hd.html#data-ingestion"><i class="fa fa-check"></i><b>9.3</b> Data Ingestion</a>
<ul>
<li class="chapter" data-level="9.3.1" data-path="visium-hd.html"><a href="visium-hd.html#visium-hd-output-data-format"><i class="fa fa-check"></i><b>9.3.1</b> Visium HD output data format</a></li>
<li class="chapter" data-level="9.3.2" data-path="visium-hd.html"><a href="visium-hd.html#mini-visium-hd-dataset"><i class="fa fa-check"></i><b>9.3.2</b> Mini Visium HD dataset</a></li>
<li class="chapter" data-level="9.3.3" data-path="visium-hd.html"><a href="visium-hd.html#giotto-visium-hd-convenience-function"><i class="fa fa-check"></i><b>9.3.3</b> Giotto Visium HD convenience function</a></li>
<li class="chapter" data-level="9.3.4" data-path="visium-hd.html"><a href="visium-hd.html#read-in-data-manually"><i class="fa fa-check"></i><b>9.3.4</b> Read in data manually</a></li>
</ul></li>
<li class="chapter" data-level="9.4" data-path="visium-hd.html"><a href="visium-hd.html#giotto-object-at-hexbin-400"><i class="fa fa-check"></i><b>9.4</b> Giotto object at Hexbin 400</a>
<ul>
<li class="chapter" data-level="9.4.1" data-path="visium-hd.html"><a href="visium-hd.html#create-giotto-points"><i class="fa fa-check"></i><b>9.4.1</b> create giotto points</a></li>
<li class="chapter" data-level="9.4.2" data-path="visium-hd.html"><a href="visium-hd.html#create-giotto-polygons"><i class="fa fa-check"></i><b>9.4.2</b> create giotto polygons</a></li>
<li class="chapter" data-level="9.4.3" data-path="visium-hd.html"><a href="visium-hd.html#combine-giotto-points-and-polygons-to-create-giotto-object"><i class="fa fa-check"></i><b>9.4.3</b> combine Giotto points and polygons to create Giotto object</a></li>
<li class="chapter" data-level="9.4.4" data-path="visium-hd.html"><a href="visium-hd.html#process-giotto-object"><i class="fa fa-check"></i><b>9.4.4</b> Process Giotto object</a></li>
<li class="chapter" data-level="9.4.5" data-path="visium-hd.html"><a href="visium-hd.html#dimension-reduction-clustering"><i class="fa fa-check"></i><b>9.4.5</b> Dimension reduction + clustering</a></li>
</ul></li>
<li class="chapter" data-level="9.5" data-path="visium-hd.html"><a href="visium-hd.html#hexbin-100"><i class="fa fa-check"></i><b>9.5</b> Hexbin 100</a>
<ul>
<li class="chapter" data-level="9.5.1" data-path="visium-hd.html"><a href="visium-hd.html#standard-subcellular-pipeline"><i class="fa fa-check"></i><b>9.5.1</b> Standard subcellular pipeline</a></li>
<li class="chapter" data-level="9.5.2" data-path="visium-hd.html"><a href="visium-hd.html#spatial-expression-patterns-1"><i class="fa fa-check"></i><b>9.5.2</b> Spatial expression patterns</a></li>
</ul></li>
<li class="chapter" data-level="9.6" data-path="visium-hd.html"><a href="visium-hd.html#hexbin-25"><i class="fa fa-check"></i><b>9.6</b> Hexbin 25</a>
<ul>
<li class="chapter" data-level="9.6.1" data-path="visium-hd.html"><a href="visium-hd.html#subcellular-workflow-and-projection-functions"><i class="fa fa-check"></i><b>9.6.1</b> Subcellular workflow and projection functions</a></li>
</ul></li>
<li class="chapter" data-level="9.7" data-path="visium-hd.html"><a href="visium-hd.html#database-backend---work-in-progress-but-coming-soon"><i class="fa fa-check"></i><b>9.7</b> Database backend - Work in progress, but coming soon!</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="xenium-1.html"><a href="xenium-1.html"><i class="fa fa-check"></i><b>10</b> Xenium</a>
<ul>
<li class="chapter" data-level="10.1" data-path="xenium-1.html"><a href="xenium-1.html#introduction-to-spatial-dataset"><i class="fa fa-check"></i><b>10.1</b> Introduction to spatial dataset</a>
<ul>
<li class="chapter" data-level="10.1.1" data-path="xenium-1.html"><a href="xenium-1.html#output-directory-structure"><i class="fa fa-check"></i><b>10.1.1</b> Output directory structure</a></li>
<li class="chapter" data-level="10.1.2" data-path="xenium-1.html"><a href="xenium-1.html#mini-xenium-dataset"><i class="fa fa-check"></i><b>10.1.2</b> Mini Xenium Dataset</a></li>
</ul></li>
<li class="chapter" data-level="10.2" data-path="xenium-1.html"><a href="xenium-1.html#data-prep"><i class="fa fa-check"></i><b>10.2</b> Data prep</a>
<ul>
<li class="chapter" data-level="10.2.1" data-path="xenium-1.html"><a href="xenium-1.html#image-conversion-may-change"><i class="fa fa-check"></i><b>10.2.1</b> Image conversion (may change)</a></li>
</ul></li>
<li class="chapter" data-level="10.3" data-path="xenium-1.html"><a href="xenium-1.html#convenience-function"><i class="fa fa-check"></i><b>10.3</b> Convenience function</a>
<ul>
<li class="chapter" data-level="10.3.1" data-path="xenium-1.html"><a href="xenium-1.html#specific-filepaths"><i class="fa fa-check"></i><b>10.3.1</b> Specific filepaths</a></li>
<li class="chapter" data-level="10.3.2" data-path="xenium-1.html"><a href="xenium-1.html#quality-value"><i class="fa fa-check"></i><b>10.3.2</b> Quality value</a></li>
<li class="chapter" data-level="10.3.3" data-path="xenium-1.html"><a href="xenium-1.html#transcript-type-splitting"><i class="fa fa-check"></i><b>10.3.3</b> Transcript type splitting</a></li>
<li class="chapter" data-level="10.3.4" data-path="xenium-1.html"><a href="xenium-1.html#centroids-calculation"><i class="fa fa-check"></i><b>10.3.4</b> Centroids calculation</a></li>
<li class="chapter" data-level="10.3.5" data-path="xenium-1.html"><a href="xenium-1.html#simple-visualization"><i class="fa fa-check"></i><b>10.3.5</b> Simple visualization</a></li>
</ul></li>
<li class="chapter" data-level="10.4" data-path="xenium-1.html"><a href="xenium-1.html#piecewise-loading"><i class="fa fa-check"></i><b>10.4</b> Piecewise loading</a>
<ul>
<li class="chapter" data-level="10.4.1" data-path="xenium-1.html"><a href="xenium-1.html#load-giottopoints-transcripts"><i class="fa fa-check"></i><b>10.4.1</b> load giottoPoints transcripts</a></li>
<li class="chapter" data-level="10.4.2" data-path="xenium-1.html"><a href="xenium-1.html#optional-loading-pre-aggregated-data"><i class="fa fa-check"></i><b>10.4.2</b> (optional) Loading pre-aggregated data</a></li>
</ul></li>
<li class="chapter" data-level="10.5" data-path="xenium-1.html"><a href="xenium-1.html#xenium-images"><i class="fa fa-check"></i><b>10.5</b> Xenium Images</a>
<ul>
<li class="chapter" data-level="10.5.1" data-path="xenium-1.html"><a href="xenium-1.html#image-metadata"><i class="fa fa-check"></i><b>10.5.1</b> Image metadata</a></li>
<li class="chapter" data-level="10.5.2" data-path="xenium-1.html"><a href="xenium-1.html#image-loading"><i class="fa fa-check"></i><b>10.5.2</b> Image loading</a></li>
</ul></li>
<li class="chapter" data-level="10.6" data-path="xenium-1.html"><a href="xenium-1.html#spatial-aggregation"><i class="fa fa-check"></i><b>10.6</b> Spatial aggregation</a></li>
<li class="chapter" data-level="10.7" data-path="xenium-1.html"><a href="xenium-1.html#aggregate-analyses-workflow"><i class="fa fa-check"></i><b>10.7</b> Aggregate analyses workflow</a>
<ul>
<li class="chapter" data-level="10.7.1" data-path="xenium-1.html"><a href="xenium-1.html#filtering-1"><i class="fa fa-check"></i><b>10.7.1</b> Filtering</a></li>
<li class="chapter" data-level="10.7.2" data-path="xenium-1.html"><a href="xenium-1.html#normalization-1"><i class="fa fa-check"></i><b>10.7.2</b> Normalization</a></li>
<li class="chapter" data-level="10.7.3" data-path="xenium-1.html"><a href="xenium-1.html#dimension-reduction-1"><i class="fa fa-check"></i><b>10.7.3</b> Dimension Reduction</a></li>
<li class="chapter" data-level="10.7.4" data-path="xenium-1.html"><a href="xenium-1.html#clustering-1"><i class="fa fa-check"></i><b>10.7.4</b> Clustering</a></li>
</ul></li>
<li class="chapter" data-level="10.8" data-path="xenium-1.html"><a href="xenium-1.html#niche-clustering"><i class="fa fa-check"></i><b>10.8</b> Niche clustering</a>
<ul>
<li class="chapter" data-level="10.8.1" data-path="xenium-1.html"><a href="xenium-1.html#spatial-network"><i class="fa fa-check"></i><b>10.8.1</b> Spatial network</a></li>
<li class="chapter" data-level="10.8.2" data-path="xenium-1.html"><a href="xenium-1.html#niche-calculation"><i class="fa fa-check"></i><b>10.8.2</b> Niche calculation</a></li>
<li class="chapter" data-level="10.8.3" data-path="xenium-1.html"><a href="xenium-1.html#kmeans-clustering-based-on-niche-signature"><i class="fa fa-check"></i><b>10.8.3</b> kmeans clustering based on niche signature</a></li>
</ul></li>
<li class="chapter" data-level="10.9" data-path="xenium-1.html"><a href="xenium-1.html#cell-proximity-enrichment"><i class="fa fa-check"></i><b>10.9</b> Cell proximity enrichment</a></li>
<li class="chapter" data-level="10.10" data-path="xenium-1.html"><a href="xenium-1.html#pseudovisium"><i class="fa fa-check"></i><b>10.10</b> Pseudovisium</a>
<ul>
<li class="chapter" data-level="10.10.1" data-path="xenium-1.html"><a href="xenium-1.html#pseudovisium-aggregation-and-workflow"><i class="fa fa-check"></i><b>10.10.1</b> Pseudovisium aggregation and workflow</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html"><i class="fa fa-check"></i><b>11</b> Spatial proteomics: Multiplexed Immunofluorescence</a>
<ul>
<li class="chapter" data-level="11.1" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#spatial-proteomics-technologies"><i class="fa fa-check"></i><b>11.1</b> Spatial Proteomics Technologies</a></li>
<li class="chapter" data-level="11.2" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#raw-data-type-coming-out-of-different-technologies"><i class="fa fa-check"></i><b>11.2</b> Raw data type coming out of different technologies</a>
<ul>
<li class="chapter" data-level="11.2.1" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#use-ome.tiff-as-an-example-output-data-to-begin-with"><i class="fa fa-check"></i><b>11.2.1</b> Use ome.tiff as an example output data to begin with</a></li>
<li class="chapter" data-level="11.2.2" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#use-single-channel-images-as-an-example-output-data-to-begin-with"><i class="fa fa-check"></i><b>11.2.2</b> Use single channel images as an example output data to begin with</a></li>
</ul></li>
<li class="chapter" data-level="11.3" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#cell-segmentation-file-to-get-single-cell-level-protein-expression"><i class="fa fa-check"></i><b>11.3</b> Cell Segmentation file to get single cell level protein expression</a>
<ul>
<li class="chapter" data-level="11.3.1" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#using-segmentation-output-file-from-deepcellmesmer-as-an-example."><i class="fa fa-check"></i><b>11.3.1</b> Using segmentation output file from DeepCell(mesmer) as an example.</a></li>
<li class="chapter" data-level="11.3.2" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#using-giotto-wrapper-of-cellpose-to-perform-segmentation"><i class="fa fa-check"></i><b>11.3.2</b> Using Giotto wrapper of Cellpose to perform segmentation</a></li>
</ul></li>
<li class="chapter" data-level="11.4" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#create-a-giotto-object-using-list-of-gitto-large-images-and-polygons"><i class="fa fa-check"></i><b>11.4</b> Create a Giotto Object using list of gitto large images and polygons</a>
<ul>
<li class="chapter" data-level="11.4.1" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#overlap-to-matrix"><i class="fa fa-check"></i><b>11.4.1</b> Overlap to matrix</a></li>
<li class="chapter" data-level="11.4.2" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#manipulate-expression-information"><i class="fa fa-check"></i><b>11.4.2</b> Manipulate Expression information</a></li>
<li class="chapter" data-level="11.4.3" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#rescale-polygons"><i class="fa fa-check"></i><b>11.4.3</b> Rescale polygons</a></li>
<li class="chapter" data-level="11.4.4" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#perform-clustering-and-differential-expression"><i class="fa fa-check"></i><b>11.4.4</b> Perform clustering and differential expression</a></li>
<li class="chapter" data-level="11.4.5" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#give-the-cluster-an-annotation-based-on-expression-values"><i class="fa fa-check"></i><b>11.4.5</b> Give the cluster an annotation based on expression values</a></li>
<li class="chapter" data-level="11.4.6" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#spatial-network-1"><i class="fa fa-check"></i><b>11.4.6</b> spatial network</a></li>
<li class="chapter" data-level="11.4.7" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#cell-neighborhood-cell-typecell-type-interactions"><i class="fa fa-check"></i><b>11.4.7</b> Cell Neighborhood: Cell-Type/Cell-Type Interactions</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="12" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html"><i class="fa fa-check"></i><b>12</b> Working with multiple samples</a>
<ul>
<li class="chapter" data-level="12.1" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#objective-1"><i class="fa fa-check"></i><b>12.1</b> Objective</a></li>
<li class="chapter" data-level="12.2" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#background-1"><i class="fa fa-check"></i><b>12.2</b> Background</a>
<ul>
<li class="chapter" data-level="12.2.1" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#dataset"><i class="fa fa-check"></i><b>12.2.1</b> Dataset</a></li>
<li class="chapter" data-level="12.2.2" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#visium-technology"><i class="fa fa-check"></i><b>12.2.2</b> Visium technology</a></li>
</ul></li>
<li class="chapter" data-level="12.3" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#create-individual-giotto-objects"><i class="fa fa-check"></i><b>12.3</b> Create individual giotto objects</a>
<ul>
<li class="chapter" data-level="12.3.1" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#download-the-data"><i class="fa fa-check"></i><b>12.3.1</b> Download the data</a></li>
</ul></li>
<li class="chapter" data-level="12.4" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#extracting-the-downloaded-files"><i class="fa fa-check"></i><b>12.4</b> Extracting the downloaded files</a>
<ul>
<li class="chapter" data-level="12.4.1" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#create-giotto-instructions"><i class="fa fa-check"></i><b>12.4.1</b> Create giotto instructions</a></li>
<li class="chapter" data-level="12.4.2" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#load-visium-data-into-giotto"><i class="fa fa-check"></i><b>12.4.2</b> Load visium data into Giotto</a></li>
<li class="chapter" data-level="12.4.3" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#healthy-prostate-tissue-coverage"><i class="fa fa-check"></i><b>12.4.3</b> Healthy prostate tissue coverage</a></li>
<li class="chapter" data-level="12.4.4" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#adenocarcinoma-prostate-tissue-coverage"><i class="fa fa-check"></i><b>12.4.4</b> Adenocarcinoma prostate tissue coverage</a></li>
<li class="chapter" data-level="12.4.5" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#showing-the-data-strucutre-for-the-inidividual-objects"><i class="fa fa-check"></i><b>12.4.5</b> Showing the data strucutre for the inidividual objects</a></li>
</ul></li>
<li class="chapter" data-level="12.5" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#join-giotto-objects"><i class="fa fa-check"></i><b>12.5</b> Join Giotto Objects</a></li>
<li class="chapter" data-level="12.6" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#visualizing-combined-datasets"><i class="fa fa-check"></i><b>12.6</b> Visualizing combined datasets</a>
<ul>
<li class="chapter" data-level="12.6.1" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#vizualizing-in-the-same-plot"><i class="fa fa-check"></i><b>12.6.1</b> Vizualizing in the same plot</a></li>
<li class="chapter" data-level="12.6.2" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#vizualizing-on-separate-plots"><i class="fa fa-check"></i><b>12.6.2</b> Vizualizing on separate plots</a></li>
</ul></li>
<li class="chapter" data-level="12.7" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#splitting-combined-dataset"><i class="fa fa-check"></i><b>12.7</b> Splitting combined dataset</a></li>
<li class="chapter" data-level="12.8" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#analyzing-joined-objects"><i class="fa fa-check"></i><b>12.8</b> Analyzing joined objects</a>
<ul>
<li class="chapter" data-level="12.8.1" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#normalization-and-adding-statistics"><i class="fa fa-check"></i><b>12.8.1</b> Normalization and adding statistics</a></li>
<li class="chapter" data-level="12.8.2" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#clustering-the-datasets"><i class="fa fa-check"></i><b>12.8.2</b> Clustering the datasets</a></li>
<li class="chapter" data-level="12.8.3" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#vizualizing-spatial-location-of-clusters"><i class="fa fa-check"></i><b>12.8.3</b> Vizualizing spatial location of clusters</a></li>
<li class="chapter" data-level="12.8.4" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#vizualizing-tissue-contribution-to-clusters"><i class="fa fa-check"></i><b>12.8.4</b> Vizualizing tissue contribution to clusters</a></li>
</ul></li>
<li class="chapter" data-level="12.9" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#perform-harmony-and-default-workflows"><i class="fa fa-check"></i><b>12.9</b> Perform Harmony and default workflows</a>
<ul>
<li class="chapter" data-level="12.9.1" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#clustering-harmonized-object"><i class="fa fa-check"></i><b>12.9.1</b> Clustering harmonized object</a></li>
<li class="chapter" data-level="12.9.2" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#vizualizing-the-tissue-contribution"><i class="fa fa-check"></i><b>12.9.2</b> Vizualizing the tissue contribution</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="13" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html"><i class="fa fa-check"></i><b>13</b> Spatial multi-modal analysis</a>
<ul>
<li class="chapter" data-level="13.1" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#overview-1"><i class="fa fa-check"></i><b>13.1</b> Overview</a></li>
<li class="chapter" data-level="13.2" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#spatial-manipulation"><i class="fa fa-check"></i><b>13.2</b> Spatial manipulation</a>
<ul>
<li class="chapter" data-level="13.2.1" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#spatial-transforms"><i class="fa fa-check"></i><b>13.2.1</b> Spatial transforms:</a></li>
<li class="chapter" data-level="13.2.2" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#spatial-utilities"><i class="fa fa-check"></i><b>13.2.2</b> Spatial utilities:</a></li>
<li class="chapter" data-level="13.2.3" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#spatial-classes"><i class="fa fa-check"></i><b>13.2.3</b> Spatial classes:</a></li>
</ul></li>
<li class="chapter" data-level="13.3" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#examples-of-the-simple-transforms-with-a-giottopolygon"><i class="fa fa-check"></i><b>13.3</b> Examples of the simple transforms with a giottoPolygon</a></li>
<li class="chapter" data-level="13.4" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#affine-transforms"><i class="fa fa-check"></i><b>13.4</b> Affine transforms</a></li>
<li class="chapter" data-level="13.5" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#image-transforms"><i class="fa fa-check"></i><b>13.5</b> Image transforms</a></li>
<li class="chapter" data-level="13.6" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#the-practical-usage-of-multi-modality-co-registration"><i class="fa fa-check"></i><b>13.6</b> The practical usage of multi-modality co-registration</a>
<ul>
<li class="chapter" data-level="13.6.1" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#example-dataset-xenium-breast-cancer-pre-release-pack"><i class="fa fa-check"></i><b>13.6.1</b> Example dataset: Xenium Breast Cancer pre-release pack</a></li>
<li class="chapter" data-level="13.6.2" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#target-coordinate-system"><i class="fa fa-check"></i><b>13.6.2</b> Target Coordinate system</a></li>
<li class="chapter" data-level="13.6.3" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#visium-to-register"><i class="fa fa-check"></i><b>13.6.3</b> Visium to register</a></li>
<li class="chapter" data-level="13.6.4" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#register-post-xenium-he-and-if-image"><i class="fa fa-check"></i><b>13.6.4</b> Register post-Xenium H&amp;E and IF image</a></li>
<li class="chapter" data-level="13.6.5" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#automatic-alignment-via-sift-feature-descriptor-matching-and-affine-transformation"><i class="fa fa-check"></i><b>13.6.5</b> Automatic alignment via SIFT feature descriptor matching and affine transformation</a></li>
<li class="chapter" data-level="13.6.6" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#final-notes"><i class="fa fa-check"></i><b>13.6.6</b> Final Notes</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="14" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html"><i class="fa fa-check"></i><b>14</b> Multi-omics integration</a>
<ul>
<li class="chapter" data-level="14.1" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#the-cytassist-technology"><i class="fa fa-check"></i><b>14.1</b> The CytAssist technology</a></li>
<li class="chapter" data-level="14.2" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#introduction-to-the-spatial-dataset-1"><i class="fa fa-check"></i><b>14.2</b> Introduction to the spatial dataset</a></li>
<li class="chapter" data-level="14.3" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#download-dataset-1"><i class="fa fa-check"></i><b>14.3</b> Download dataset</a></li>
<li class="chapter" data-level="14.4" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#create-the-giotto-object-1"><i class="fa fa-check"></i><b>14.4</b> Create the Giotto object</a></li>
<li class="chapter" data-level="14.5" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#subset-on-spots-that-were-covered-by-tissue-1"><i class="fa fa-check"></i><b>14.5</b> Subset on spots that were covered by tissue</a></li>
<li class="chapter" data-level="14.6" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#rna-processing"><i class="fa fa-check"></i><b>14.6</b> RNA processing</a></li>
<li class="chapter" data-level="14.7" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#protein-processing"><i class="fa fa-check"></i><b>14.7</b> Protein processing</a></li>
<li class="chapter" data-level="14.8" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#multi-omics-integration-1"><i class="fa fa-check"></i><b>14.8</b> Multi-omics integration</a></li>
<li class="chapter" data-level="14.9" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#session-info-2"><i class="fa fa-check"></i><b>14.9</b> Session info</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html"><i class="fa fa-check"></i><b>15</b> Interoperability with other frameworks</a>
<ul>
<li class="chapter" data-level="15.1" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#load-giotto-object"><i class="fa fa-check"></i><b>15.1</b> Load Giotto object</a></li>
<li class="chapter" data-level="15.2" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#seurat"><i class="fa fa-check"></i><b>15.2</b> Seurat</a>
<ul>
<li class="chapter" data-level="15.2.1" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#conversion-of-giotto-object-to-seurat-object"><i class="fa fa-check"></i><b>15.2.1</b> Conversion of Giotto Object to Seurat Object</a></li>
<li class="chapter" data-level="15.2.2" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#conversion-of-seurat-object-back-to-giotto-object"><i class="fa fa-check"></i><b>15.2.2</b> Conversion of Seurat object Back to Giotto Object</a></li>
</ul></li>
<li class="chapter" data-level="15.3" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#spatialexperiment"><i class="fa fa-check"></i><b>15.3</b> SpatialExperiment</a>
<ul>
<li class="chapter" data-level="15.3.1" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#convert-giotto-object-to-spatialexperiment-object"><i class="fa fa-check"></i><b>15.3.1</b> Convert Giotto Object to SpatialExperiment Object</a></li>
<li class="chapter" data-level="15.3.2" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#conversion-of-spatialexperiment-object-back-to-giotto"><i class="fa fa-check"></i><b>15.3.2</b> Conversion of SpatialExperiment object back to Giotto</a></li>
</ul></li>
<li class="chapter" data-level="15.4" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#anndata"><i class="fa fa-check"></i><b>15.4</b> AnnData</a>
<ul>
<li class="chapter" data-level="15.4.1" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#load-required-libraries"><i class="fa fa-check"></i><b>15.4.1</b> Load Required Libraries</a></li>
<li class="chapter" data-level="15.4.2" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#specify-path-for-results"><i class="fa fa-check"></i><b>15.4.2</b> Specify Path for Results</a></li>
<li class="chapter" data-level="15.4.3" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#giotto-to-anndata"><i class="fa fa-check"></i><b>15.4.3</b> Giotto To AnnData</a></li>
<li class="chapter" data-level="15.4.4" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#convert-anndata-to-giotto"><i class="fa fa-check"></i><b>15.4.4</b> Convert AnnData to Giotto</a></li>
</ul></li>
<li class="chapter" data-level="15.5" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#create-mini-vizgen-object"><i class="fa fa-check"></i><b>15.5</b> Create mini Vizgen object</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html"><i class="fa fa-check"></i><b>16</b> Interoperability with isolated tools</a>
<ul>
<li class="chapter" data-level="16.1" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#spatial-niche-trajectory-analysis-ontrac"><i class="fa fa-check"></i><b>16.1</b> Spatial niche trajectory analysis (ONTraC)</a>
<ul>
<li class="chapter" data-level="16.1.1" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#introduction-to-ontrac"><i class="fa fa-check"></i><b>16.1.1</b> Introduction to ONTraC</a></li>
<li class="chapter" data-level="16.1.2" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#introduction-to-merfish"><i class="fa fa-check"></i><b>16.1.2</b> Introduction to MERFISH</a></li>
<li class="chapter" data-level="16.1.3" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#settings"><i class="fa fa-check"></i><b>16.1.3</b> Settings</a></li>
<li class="chapter" data-level="16.1.4" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#dataset-1"><i class="fa fa-check"></i><b>16.1.4</b> Dataset</a></li>
<li class="chapter" data-level="16.1.5" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#create-the-giotto-object-2"><i class="fa fa-check"></i><b>16.1.5</b> Create the Giotto object</a></li>
<li class="chapter" data-level="16.1.6" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#perform-spatial-niche-trajectory-analysis-using-ontrac"><i class="fa fa-check"></i><b>16.1.6</b> Perform spatial niche trajectory analysis using ONTraC</a></li>
<li class="chapter" data-level="16.1.7" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#visualization"><i class="fa fa-check"></i><b>16.1.7</b> Visualization</a></li>
</ul></li>
<li class="chapter" data-level="16.2" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#session-info-3"><i class="fa fa-check"></i><b>16.2</b> Session info</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html"><i class="fa fa-check"></i><b>17</b> Interactivity with the R/Spatial ecosystem</a>
<ul>
<li class="chapter" data-level="17.1" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#kriging"><i class="fa fa-check"></i><b>17.1</b> Kriging</a>
<ul>
<li class="chapter" data-level="17.1.1" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#visium-technology-1"><i class="fa fa-check"></i><b>17.1.1</b> Visium technology</a></li>
<li class="chapter" data-level="17.1.2" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#dataset-2"><i class="fa fa-check"></i><b>17.1.2</b> Dataset</a></li>
<li class="chapter" data-level="17.1.3" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#generating-a-geojson-file-of-nuclei-location"><i class="fa fa-check"></i><b>17.1.3</b> Generating a geojson file of nuclei location</a></li>
</ul></li>
<li class="chapter" data-level="17.2" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#downloading-the-dataset"><i class="fa fa-check"></i><b>17.2</b> Downloading the dataset</a></li>
<li class="chapter" data-level="17.3" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#extracting-the-files"><i class="fa fa-check"></i><b>17.3</b> Extracting the files</a></li>
<li class="chapter" data-level="17.4" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#downloading-giotto-object-and-nuclei-segmentation"><i class="fa fa-check"></i><b>17.4</b> Downloading giotto object and nuclei segmentation</a></li>
<li class="chapter" data-level="17.5" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#importing-visium-data"><i class="fa fa-check"></i><b>17.5</b> Importing visium data</a>
<ul>
<li class="chapter" data-level="17.5.1" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#identifying-spatially-organized-features"><i class="fa fa-check"></i><b>17.5.1</b> Identifying spatially organized features</a></li>
</ul></li>
<li class="chapter" data-level="17.6" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#adding-cell-polygons-to-giotto-object"><i class="fa fa-check"></i><b>17.6</b> Adding cell polygons to Giotto object</a>
<ul>
<li class="chapter" data-level="17.6.1" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#read-in-the-poly-information"><i class="fa fa-check"></i><b>17.6.1</b> Read in the poly information</a></li>
<li class="chapter" data-level="17.6.2" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#vizualizing-polygons"><i class="fa fa-check"></i><b>17.6.2</b> Vizualizing polygons</a></li>
<li class="chapter" data-level="17.6.3" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#showing-giotto-object-prior-to-polygon-addition"><i class="fa fa-check"></i><b>17.6.3</b> Showing Giotto object prior to polygon addition</a></li>
<li class="chapter" data-level="17.6.4" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#adding-polygons-to-giotto-object"><i class="fa fa-check"></i><b>17.6.4</b> Adding polygons to giotto object</a></li>
<li class="chapter" data-level="17.6.5" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#check-polygon-information"><i class="fa fa-check"></i><b>17.6.5</b> Check polygon information</a></li>
</ul></li>
<li class="chapter" data-level="17.7" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#performing-kriging"><i class="fa fa-check"></i><b>17.7</b> Performing kriging</a>
<ul>
<li class="chapter" data-level="17.7.1" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#interpolating-features"><i class="fa fa-check"></i><b>17.7.1</b> Interpolating features</a></li>
<li class="chapter" data-level="17.7.2" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#expression-overlap"><i class="fa fa-check"></i><b>17.7.2</b> Expression overlap</a></li>
</ul></li>
<li class="chapter" data-level="17.8" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#reading-in-larger-dataset"><i class="fa fa-check"></i><b>17.8</b> Reading in larger dataset</a></li>
<li class="chapter" data-level="17.9" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#analyzing-interpolated-features"><i class="fa fa-check"></i><b>17.9</b> Analyzing interpolated features</a>
<ul>
<li class="chapter" data-level="17.9.1" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#filter-and-normalization"><i class="fa fa-check"></i><b>17.9.1</b> Filter and normalization</a></li>
<li class="chapter" data-level="17.9.2" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#visualizing-gene-expression-from-interpolated-expression"><i class="fa fa-check"></i><b>17.9.2</b> Visualizing gene expression from interpolated expression</a></li>
<li class="chapter" data-level="17.9.3" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#run-pca"><i class="fa fa-check"></i><b>17.9.3</b> Run PCA</a></li>
<li class="chapter" data-level="17.9.4" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#clustering-2"><i class="fa fa-check"></i><b>17.9.4</b> Clustering</a></li>
<li class="chapter" data-level="17.9.5" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#visualizing-clustering"><i class="fa fa-check"></i><b>17.9.5</b> Visualizing clustering</a></li>
<li class="chapter" data-level="17.9.6" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#cropping-objects"><i class="fa fa-check"></i><b>17.9.6</b> Cropping objects</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="18" data-path="contributing-to-giotto.html"><a href="contributing-to-giotto.html"><i class="fa fa-check"></i><b>18</b> Contributing to Giotto</a>
<ul>
<li class="chapter" data-level="18.1" data-path="contributing-to-giotto.html"><a href="contributing-to-giotto.html#contribution-guideline"><i class="fa fa-check"></i><b>18.1</b> Contribution guideline</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Workshop: Spatial multi-omics data analysis with Giotto Suite</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="visium-hd" class="section level1 hasAnchor" number="9">
<h1><span class="header-section-number">9</span> Visium HD<a href="visium-hd.html#visium-hd" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Ruben Dries &amp; Edward C. Ruiz</p>
<p>August 6th 2024</p>
<div id="objective" class="section level2 hasAnchor" number="9.1">
<h2><span class="header-section-number">9.1</span> Objective<a href="visium-hd.html#objective" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>This tutorial demonstrates how to process Visium HD data at the highest 2 micron bin resolution using <a href="https://drieslab.github.io/Giotto_website/">Giotto Suite</a>. Notably, a similar strategy can be used for other spatial sequencing methods that operate at the subcellular level, including:<br />
- Stereo-seq<br />
- <a href="https://drieslab.github.io/Giotto_website/articles/seqscope_mouse_liver.html">Seq-Scope</a><br />
- Open-ST</p>
<p>The resulting datasets from all these technologies can be very large since they provide both a high spatial resolution and genome-wide capture of <em>all</em> transcripts.</p>
</div>
<div id="background" class="section level2 hasAnchor" number="9.2">
<h2><span class="header-section-number">9.2</span> Background<a href="visium-hd.html#background" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="visium-hd-technology" class="section level3 hasAnchor" number="9.2.1">
<h3><span class="header-section-number">9.2.1</span> Visium HD Technology<a href="visium-hd.html#visium-hd-technology" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-236"></span>
<img src="img/02_session2/1.png" alt="Overview of Visium HD. Source: *10X Genomics*" width="80%" />
<p class="caption">
Figure 9.1: Overview of Visium HD. Source: <em>10X Genomics</em>
</p>
</div>
<p>Visium HD is a spatial transcriptomics technology recently developed by 10X Genomics. Details about this platform are discussed on the official <a href="https://www.10xgenomics.com/products/visium-hd-spatial-gene-expression">10X Genomics Visium HD website</a> and the preprint by <a href="https://doi.org/10.1101/2024.06.04.597233">Oliveira <em>et al.</em> 2024</a> on <em>bioRxiv</em>.</p>
<p>Visium HD has a 2 micron bin size resolution. The default SpaceRanger pipeline from 10X Genomics also returns aggregated data at the 8 and 16 micron bin size.</p>
</div>
<div id="colorectal-cancer-sample" class="section level3 hasAnchor" number="9.2.2">
<h3><span class="header-section-number">9.2.2</span> Colorectal Cancer Sample<a href="visium-hd.html#colorectal-cancer-sample" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-237"></span>
<img src="img/02_session2/2.png" alt="Colorectal Cancer Overview. Source: *10X Genomics*" width="80%" />
<p class="caption">
Figure 9.2: Colorectal Cancer Overview. Source: <em>10X Genomics</em>
</p>
</div>
<p>For this tutorial we will be using the publicly available Colorectal Cancer Visium HD dataset.</p>
<p>Details about this dataset and a link to download the raw data can be found at the <a href="https://www.10xgenomics.com/datasets/visium-hd-cytassist-gene-expression-libraries-of-human-crc">10X Genomics website</a>.</p>
</div>
</div>
<div id="data-ingestion" class="section level2 hasAnchor" number="9.3">
<h2><span class="header-section-number">9.3</span> Data Ingestion<a href="visium-hd.html#data-ingestion" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="visium-hd-output-data-format" class="section level3 hasAnchor" number="9.3.1">
<h3><span class="header-section-number">9.3.1</span> Visium HD output data format<a href="visium-hd.html#visium-hd-output-data-format" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-238"></span>
<img src="img/02_session2/3.png" alt="File structure of Visium HD data processed with spaceranger pipeline." width="80%" />
<p class="caption">
Figure 9.3: File structure of Visium HD data processed with spaceranger pipeline.
</p>
</div>
<p>Visium HD data processed with the spaceranger pipeline is organized in this format containing various files associated with the sample. The files highlighted in yellow are what we will be using to read in these datasets.</p>
</div>
<div id="mini-visium-hd-dataset" class="section level3 hasAnchor" number="9.3.2">
<h3><span class="header-section-number">9.3.2</span> Mini Visium HD dataset<a href="visium-hd.html#mini-visium-hd-dataset" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb182"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb182-1"><a href="visium-hd.html#cb182-1" tabindex="-1"></a><span class="fu">library</span>(Giotto)</span>
<span id="cb182-2"><a href="visium-hd.html#cb182-2" tabindex="-1"></a></span>
<span id="cb182-3"><a href="visium-hd.html#cb182-3" tabindex="-1"></a><span class="co"># set up paths</span></span>
<span id="cb182-4"><a href="visium-hd.html#cb182-4" tabindex="-1"></a>data_path <span class="ot">&lt;-</span> <span class="st">&quot;data/02_session2/&quot;</span></span>
<span id="cb182-5"><a href="visium-hd.html#cb182-5" tabindex="-1"></a>save_dir <span class="ot">&lt;-</span> <span class="st">&quot;results/02_session2/&quot;</span></span>
<span id="cb182-6"><a href="visium-hd.html#cb182-6" tabindex="-1"></a><span class="fu">dir.create</span>(save_dir, <span class="at">recursive =</span> <span class="cn">TRUE</span>)</span>
<span id="cb182-7"><a href="visium-hd.html#cb182-7" tabindex="-1"></a></span>
<span id="cb182-8"><a href="visium-hd.html#cb182-8" tabindex="-1"></a><span class="co"># download the mini dataset and untar</span></span>
<span id="cb182-9"><a href="visium-hd.html#cb182-9" tabindex="-1"></a><span class="fu">options</span>(<span class="st">&quot;timeout&quot;</span> <span class="ot">=</span> <span class="cn">Inf</span>)</span>
<span id="cb182-10"><a href="visium-hd.html#cb182-10" tabindex="-1"></a><span class="fu">download.file</span>(</span>
<span id="cb182-11"><a href="visium-hd.html#cb182-11" tabindex="-1"></a>    <span class="at">url =</span> <span class="st">&quot;https://zenodo.org/records/13212855/files/workshop_VisiumHD.zip?download=1&quot;</span>,</span>
<span id="cb182-12"><a href="visium-hd.html#cb182-12" tabindex="-1"></a>    <span class="at">destfile =</span> <span class="fu">file.path</span>(save_dir, <span class="st">&quot;workshop_visiumHD.zip&quot;</span>)</span>
<span id="cb182-13"><a href="visium-hd.html#cb182-13" tabindex="-1"></a>)</span>
<span id="cb182-14"><a href="visium-hd.html#cb182-14" tabindex="-1"></a><span class="fu">untar</span>(<span class="at">tarfile =</span> <span class="fu">file.path</span>(save_dir, <span class="st">&quot;workshop_visiumHD.zip&quot;</span>), </span>
<span id="cb182-15"><a href="visium-hd.html#cb182-15" tabindex="-1"></a>      <span class="at">exdir =</span> data_path)</span></code></pre></div>
</div>
<div id="giotto-visium-hd-convenience-function" class="section level3 hasAnchor" number="9.3.3">
<h3><span class="header-section-number">9.3.3</span> Giotto Visium HD convenience function<a href="visium-hd.html#giotto-visium-hd-convenience-function" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The easiest way to read in Visium HD data in Giotto is through our convenience function. This function will automatically read in the data at your desired resolution, align the images, and finally create a Giotto Object.</p>
<div class="sourceCode" id="cb183"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb183-1"><a href="visium-hd.html#cb183-1" tabindex="-1"></a><span class="co"># importVisiumHD()</span></span></code></pre></div>
</div>
<div id="read-in-data-manually" class="section level3 hasAnchor" number="9.3.4">
<h3><span class="header-section-number">9.3.4</span> Read in data manually<a href="visium-hd.html#read-in-data-manually" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>However, for this tutorial we will illustrate how to create your own Giotto object in a step-by-step manner, which can also be applied to other similar technologies as discussed in the <a href="#Objective">Objective</a> section.</p>
<div id="raw-expression-data" class="section level4 hasAnchor" number="9.3.4.1">
<h4><span class="header-section-number">9.3.4.1</span> Raw expression data<a href="visium-hd.html#raw-expression-data" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb184"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb184-1"><a href="visium-hd.html#cb184-1" tabindex="-1"></a>expression_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_path, <span class="st">&#39;workshop_visiumHD/Human_Colorectal_Cancer_workshop/square_002um/raw_feature_bc_matrix&#39;</span>)</span>
<span id="cb184-2"><a href="visium-hd.html#cb184-2" tabindex="-1"></a>expr_results <span class="ot">&lt;-</span> <span class="fu">get10Xmatrix</span>(<span class="at">path_to_data =</span> expression_path, </span>
<span id="cb184-3"><a href="visium-hd.html#cb184-3" tabindex="-1"></a>                             <span class="at">gene_column_index =</span> <span class="dv">1</span>)</span></code></pre></div>
</div>
<div id="tissue-positions-data" class="section level4 hasAnchor" number="9.3.4.2">
<h4><span class="header-section-number">9.3.4.2</span> Tissue positions data<a href="visium-hd.html#tissue-positions-data" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb185"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb185-1"><a href="visium-hd.html#cb185-1" tabindex="-1"></a>tissue_positions_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_path, <span class="st">&#39;workshop_visiumHD/Human_Colorectal_Cancer_workshop/square_002um/spatial/tissue_positions.parquet&#39;</span>)</span>
<span id="cb185-2"><a href="visium-hd.html#cb185-2" tabindex="-1"></a>tissue_positions <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">as.data.table</span>(arrow<span class="sc">::</span><span class="fu">read_parquet</span>(tissue_positions_path))</span></code></pre></div>
</div>
<div id="merge-expression-and-2-micron-position-data" class="section level4 hasAnchor" number="9.3.4.3">
<h4><span class="header-section-number">9.3.4.3</span> Merge expression and 2 micron position data<a href="visium-hd.html#merge-expression-and-2-micron-position-data" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb186"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb186-1"><a href="visium-hd.html#cb186-1" tabindex="-1"></a><span class="co"># convert expression matrix to minimal data.frame or data.table object</span></span>
<span id="cb186-2"><a href="visium-hd.html#cb186-2" tabindex="-1"></a>matrix_tile_dt <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">as.data.table</span>(Matrix<span class="sc">::</span><span class="fu">summary</span>(expr_results))</span>
<span id="cb186-3"><a href="visium-hd.html#cb186-3" tabindex="-1"></a>genes   <span class="ot">&lt;-</span> expr_results<span class="sc">@</span>Dimnames[[<span class="dv">1</span>]]</span>
<span id="cb186-4"><a href="visium-hd.html#cb186-4" tabindex="-1"></a>samples <span class="ot">&lt;-</span> expr_results<span class="sc">@</span>Dimnames[[<span class="dv">2</span>]]</span>
<span id="cb186-5"><a href="visium-hd.html#cb186-5" tabindex="-1"></a>matrix_tile_dt[, gene <span class="sc">:=</span> genes[i]]</span>
<span id="cb186-6"><a href="visium-hd.html#cb186-6" tabindex="-1"></a>matrix_tile_dt[, pixel <span class="sc">:=</span> samples[j]]</span></code></pre></div>
<p><img src="img/02_session2/4.png" width="80%" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb187"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb187-1"><a href="visium-hd.html#cb187-1" tabindex="-1"></a><span class="co"># merge data.table matrix and spatial coordinates to create input for Giotto Polygons</span></span>
<span id="cb187-2"><a href="visium-hd.html#cb187-2" tabindex="-1"></a>expr_pos_data <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">merge.data.table</span>(matrix_tile_dt, </span>
<span id="cb187-3"><a href="visium-hd.html#cb187-3" tabindex="-1"></a>                                             tissue_positions, </span>
<span id="cb187-4"><a href="visium-hd.html#cb187-4" tabindex="-1"></a>                                             <span class="at">by.x =</span> <span class="st">&#39;pixel&#39;</span>, </span>
<span id="cb187-5"><a href="visium-hd.html#cb187-5" tabindex="-1"></a>                                             <span class="at">by.y =</span> <span class="st">&#39;barcode&#39;</span>)</span>
<span id="cb187-6"><a href="visium-hd.html#cb187-6" tabindex="-1"></a></span>
<span id="cb187-7"><a href="visium-hd.html#cb187-7" tabindex="-1"></a>expr_pos_data <span class="ot">&lt;-</span> expr_pos_data[,.(pixel, pxl_row_in_fullres, pxl_col_in_fullres, gene, x)]</span>
<span id="cb187-8"><a href="visium-hd.html#cb187-8" tabindex="-1"></a><span class="fu">colnames</span>(expr_pos_data) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&#39;pixel&#39;</span>, <span class="st">&#39;x&#39;</span>, <span class="st">&#39;y&#39;</span>, <span class="st">&#39;gene&#39;</span>, <span class="st">&#39;count&#39;</span>)</span></code></pre></div>
<p><img src="img/02_session2/5.png" width="80%" style="display: block; margin: auto;" /></p>
</div>
</div>
</div>
<div id="giotto-object-at-hexbin-400" class="section level2 hasAnchor" number="9.4">
<h2><span class="header-section-number">9.4</span> Giotto object at Hexbin 400<a href="visium-hd.html#giotto-object-at-hexbin-400" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="create-giotto-points" class="section level3 hasAnchor" number="9.4.1">
<h3><span class="header-section-number">9.4.1</span> create giotto points<a href="visium-hd.html#create-giotto-points" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb188"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb188-1"><a href="visium-hd.html#cb188-1" tabindex="-1"></a>giotto_points <span class="ot">=</span> <span class="fu">createGiottoPoints</span>(<span class="at">x =</span> expr_pos_data[,.(x, y, gene, pixel, count)])</span></code></pre></div>
</div>
<div id="create-giotto-polygons" class="section level3 hasAnchor" number="9.4.2">
<h3><span class="header-section-number">9.4.2</span> create giotto polygons<a href="visium-hd.html#create-giotto-polygons" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="tiling-and-aggregation" class="section level4 hasAnchor" number="9.4.2.1">
<h4><span class="header-section-number">9.4.2.1</span> Tiling and aggregation<a href="visium-hd.html#tiling-and-aggregation" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>The Visium HD data is organized in a grid format. We can aggregate the data into larger bins to reduce the dimensionality of the data. Giotto Suite provides options to bin data not only with squares, but also through hexagons and triangles. Here we use a hexagon tesselation to aggregate the data into arbitrary cells.</p>
<div class="sourceCode" id="cb189"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb189-1"><a href="visium-hd.html#cb189-1" tabindex="-1"></a><span class="co"># create giotto polygons, here we create hexagons</span></span>
<span id="cb189-2"><a href="visium-hd.html#cb189-2" tabindex="-1"></a>hexbin400 <span class="ot">&lt;-</span> <span class="fu">tessellate</span>(<span class="at">extent =</span> <span class="fu">ext</span>(giotto_points), </span>
<span id="cb189-3"><a href="visium-hd.html#cb189-3" tabindex="-1"></a>                        <span class="at">shape =</span> <span class="st">&#39;hexagon&#39;</span>, </span>
<span id="cb189-4"><a href="visium-hd.html#cb189-4" tabindex="-1"></a>                        <span class="at">shape_size =</span> <span class="dv">400</span>, </span>
<span id="cb189-5"><a href="visium-hd.html#cb189-5" tabindex="-1"></a>                        <span class="at">name =</span> <span class="st">&#39;hex400&#39;</span>) </span>
<span id="cb189-6"><a href="visium-hd.html#cb189-6" tabindex="-1"></a><span class="fu">plot</span>(hexbin400)</span></code></pre></div>
<p><img src="img/02_session2/6.png" width="80%" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="combine-giotto-points-and-polygons-to-create-giotto-object" class="section level3 hasAnchor" number="9.4.3">
<h3><span class="header-section-number">9.4.3</span> combine Giotto points and polygons to create Giotto object<a href="visium-hd.html#combine-giotto-points-and-polygons-to-create-giotto-object" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb190"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb190-1"><a href="visium-hd.html#cb190-1" tabindex="-1"></a>instrs <span class="ot">=</span> <span class="fu">createGiottoInstructions</span>(</span>
<span id="cb190-2"><a href="visium-hd.html#cb190-2" tabindex="-1"></a>  <span class="at">save_dir =</span> results_folder,</span>
<span id="cb190-3"><a href="visium-hd.html#cb190-3" tabindex="-1"></a>  <span class="at">save_plot =</span> <span class="cn">TRUE</span>,</span>
<span id="cb190-4"><a href="visium-hd.html#cb190-4" tabindex="-1"></a>  <span class="at">show_plot =</span> <span class="cn">FALSE</span>,</span>
<span id="cb190-5"><a href="visium-hd.html#cb190-5" tabindex="-1"></a>  <span class="at">return_plot =</span> <span class="cn">FALSE</span></span>
<span id="cb190-6"><a href="visium-hd.html#cb190-6" tabindex="-1"></a>)</span>
<span id="cb190-7"><a href="visium-hd.html#cb190-7" tabindex="-1"></a></span>
<span id="cb190-8"><a href="visium-hd.html#cb190-8" tabindex="-1"></a>visiumHD <span class="ot">=</span> <span class="fu">createGiottoObjectSubcellular</span>(<span class="at">gpoints =</span> <span class="fu">list</span>(<span class="st">&#39;rna&#39;</span> <span class="ot">=</span> giotto_points),</span>
<span id="cb190-9"><a href="visium-hd.html#cb190-9" tabindex="-1"></a>                                         <span class="at">gpolygons =</span> <span class="fu">list</span>(<span class="st">&#39;hex400&#39;</span> <span class="ot">=</span> hexbin400),</span>
<span id="cb190-10"><a href="visium-hd.html#cb190-10" tabindex="-1"></a>                                         <span class="at">instructions =</span> instrs)</span>
<span id="cb190-11"><a href="visium-hd.html#cb190-11" tabindex="-1"></a></span>
<span id="cb190-12"><a href="visium-hd.html#cb190-12" tabindex="-1"></a>visiumHD <span class="ot">=</span> <span class="fu">addSpatialCentroidLocations</span>(<span class="at">gobject =</span> visiumHD,</span>
<span id="cb190-13"><a href="visium-hd.html#cb190-13" tabindex="-1"></a>                                       <span class="at">poly_info =</span> <span class="st">&#39;hex400&#39;</span>)</span></code></pre></div>
<p>Visualize the Giotto object. Make sure to set <code>expand_counts = TRUE</code> to expand the counts
column. Each spatial bin can have multiple transcripts/UMIs. This is different
compared to in situ technologies like seqFISH, MERFISH, Nanostring CosMx or Xenium.</p>
<div class="sourceCode" id="cb191"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb191-1"><a href="visium-hd.html#cb191-1" tabindex="-1"></a>feature_data <span class="ot">=</span> <span class="fu">fDataDT</span>(visiumHD)</span>
<span id="cb191-2"><a href="visium-hd.html#cb191-2" tabindex="-1"></a></span>
<span id="cb191-3"><a href="visium-hd.html#cb191-3" tabindex="-1"></a><span class="fu">spatInSituPlotPoints</span>(visiumHD,</span>
<span id="cb191-4"><a href="visium-hd.html#cb191-4" tabindex="-1"></a>                     <span class="at">show_image =</span> F,</span>
<span id="cb191-5"><a href="visium-hd.html#cb191-5" tabindex="-1"></a>                     <span class="at">feats =</span> <span class="fu">list</span>(<span class="st">&#39;rna&#39;</span> <span class="ot">=</span> feature_data<span class="sc">$</span>feat_ID[<span class="dv">10</span><span class="sc">:</span><span class="dv">20</span>]),</span>
<span id="cb191-6"><a href="visium-hd.html#cb191-6" tabindex="-1"></a>                     <span class="at">show_legend =</span> T,</span>
<span id="cb191-7"><a href="visium-hd.html#cb191-7" tabindex="-1"></a>                     <span class="at">spat_unit =</span> <span class="st">&#39;hex400&#39;</span>,</span>
<span id="cb191-8"><a href="visium-hd.html#cb191-8" tabindex="-1"></a>                     <span class="at">point_size =</span> <span class="fl">0.25</span>,</span>
<span id="cb191-9"><a href="visium-hd.html#cb191-9" tabindex="-1"></a>                     <span class="at">show_polygon =</span> <span class="cn">TRUE</span>,</span>
<span id="cb191-10"><a href="visium-hd.html#cb191-10" tabindex="-1"></a>                     <span class="at">use_overlap =</span> <span class="cn">FALSE</span>,</span>
<span id="cb191-11"><a href="visium-hd.html#cb191-11" tabindex="-1"></a>                     <span class="at">polygon_feat_type =</span> <span class="st">&#39;hex400&#39;</span>,</span>
<span id="cb191-12"><a href="visium-hd.html#cb191-12" tabindex="-1"></a>                     <span class="at">polygon_bg_color =</span> <span class="cn">NA</span>,</span>
<span id="cb191-13"><a href="visium-hd.html#cb191-13" tabindex="-1"></a>                     <span class="at">polygon_color =</span> <span class="st">&#39;white&#39;</span>,</span>
<span id="cb191-14"><a href="visium-hd.html#cb191-14" tabindex="-1"></a>                     <span class="at">polygon_line_size =</span> <span class="fl">0.1</span>,</span>
<span id="cb191-15"><a href="visium-hd.html#cb191-15" tabindex="-1"></a>                     <span class="at">expand_counts =</span> <span class="cn">TRUE</span>, </span>
<span id="cb191-16"><a href="visium-hd.html#cb191-16" tabindex="-1"></a>                     <span class="at">count_info_column =</span> <span class="st">&#39;count&#39;</span>,</span>
<span id="cb191-17"><a href="visium-hd.html#cb191-17" tabindex="-1"></a>                     <span class="at">jitter =</span> <span class="fu">c</span>(<span class="dv">25</span>,<span class="dv">25</span>))</span></code></pre></div>
<p><img src="img/02_session2/0-spatInSituPlotPoints.png" width="80%" style="display: block; margin: auto;" /></p>
<p>You can set <code>plot_method = scattermore</code> or <code>scattermost</code> to convert high-resolution
images to low(er) resolution rasterized images. Itâs usually faster and will save on disk space.</p>
<div class="sourceCode" id="cb192"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb192-1"><a href="visium-hd.html#cb192-1" tabindex="-1"></a><span class="fu">spatInSituPlotPoints</span>(visiumHD,</span>
<span id="cb192-2"><a href="visium-hd.html#cb192-2" tabindex="-1"></a>                     <span class="at">show_image =</span> F,</span>
<span id="cb192-3"><a href="visium-hd.html#cb192-3" tabindex="-1"></a>                     <span class="at">feats =</span> <span class="fu">list</span>(<span class="st">&#39;rna&#39;</span> <span class="ot">=</span> feature_data<span class="sc">$</span>feat_ID[<span class="dv">10</span><span class="sc">:</span><span class="dv">20</span>]),</span>
<span id="cb192-4"><a href="visium-hd.html#cb192-4" tabindex="-1"></a>                     <span class="at">show_legend =</span> T,</span>
<span id="cb192-5"><a href="visium-hd.html#cb192-5" tabindex="-1"></a>                     <span class="at">spat_unit =</span> <span class="st">&#39;hex400&#39;</span>,</span>
<span id="cb192-6"><a href="visium-hd.html#cb192-6" tabindex="-1"></a>                     <span class="at">point_size =</span> <span class="fl">0.25</span>,</span>
<span id="cb192-7"><a href="visium-hd.html#cb192-7" tabindex="-1"></a>                     <span class="at">show_polygon =</span> <span class="cn">TRUE</span>,</span>
<span id="cb192-8"><a href="visium-hd.html#cb192-8" tabindex="-1"></a>                     <span class="at">use_overlap =</span> <span class="cn">FALSE</span>,</span>
<span id="cb192-9"><a href="visium-hd.html#cb192-9" tabindex="-1"></a>                     <span class="at">polygon_feat_type =</span> <span class="st">&#39;hex400&#39;</span>,</span>
<span id="cb192-10"><a href="visium-hd.html#cb192-10" tabindex="-1"></a>                     <span class="at">polygon_bg_color =</span> <span class="cn">NA</span>,</span>
<span id="cb192-11"><a href="visium-hd.html#cb192-11" tabindex="-1"></a>                     <span class="at">polygon_color =</span> <span class="st">&#39;white&#39;</span>,</span>
<span id="cb192-12"><a href="visium-hd.html#cb192-12" tabindex="-1"></a>                     <span class="at">polygon_line_size =</span> <span class="fl">0.1</span>,</span>
<span id="cb192-13"><a href="visium-hd.html#cb192-13" tabindex="-1"></a>                     <span class="at">expand_counts =</span> <span class="cn">TRUE</span>, </span>
<span id="cb192-14"><a href="visium-hd.html#cb192-14" tabindex="-1"></a>                     <span class="at">count_info_column =</span> <span class="st">&#39;count&#39;</span>,</span>
<span id="cb192-15"><a href="visium-hd.html#cb192-15" tabindex="-1"></a>                     <span class="at">jitter =</span> <span class="fu">c</span>(<span class="dv">25</span>,<span class="dv">25</span>), </span>
<span id="cb192-16"><a href="visium-hd.html#cb192-16" tabindex="-1"></a>                     <span class="at">plot_method =</span> <span class="st">&#39;scattermore&#39;</span>)</span></code></pre></div>
<p><img src="img/02_session2/1-spatInSituPlotPoints.png" width="80%" style="display: block; margin: auto;" /></p>
</div>
<div id="process-giotto-object" class="section level3 hasAnchor" number="9.4.4">
<h3><span class="header-section-number">9.4.4</span> Process Giotto object<a href="visium-hd.html#process-giotto-object" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="calculate-overlap-between-points-and-polygons" class="section level4 hasAnchor" number="9.4.4.1">
<h4><span class="header-section-number">9.4.4.1</span> calculate overlap between points and polygons<a href="visium-hd.html#calculate-overlap-between-points-and-polygons" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb193"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb193-1"><a href="visium-hd.html#cb193-1" tabindex="-1"></a><span class="co"># calculate overlap between points and polygons</span></span>
<span id="cb193-2"><a href="visium-hd.html#cb193-2" tabindex="-1"></a>visiumHD <span class="ot">=</span> <span class="fu">calculateOverlap</span>(visiumHD,</span>
<span id="cb193-3"><a href="visium-hd.html#cb193-3" tabindex="-1"></a>                            <span class="at">spatial_info =</span> <span class="st">&#39;hex400&#39;</span>,</span>
<span id="cb193-4"><a href="visium-hd.html#cb193-4" tabindex="-1"></a>                            <span class="at">feat_info =</span> <span class="st">&#39;rna&#39;</span>)</span>
<span id="cb193-5"><a href="visium-hd.html#cb193-5" tabindex="-1"></a></span>
<span id="cb193-6"><a href="visium-hd.html#cb193-6" tabindex="-1"></a><span class="fu">showGiottoSpatialInfo</span>(visiumHD)</span></code></pre></div>
</div>
<div id="convert-overlap-results-to-a-bin-by-gene-matrix" class="section level4 hasAnchor" number="9.4.4.2">
<h4><span class="header-section-number">9.4.4.2</span> convert overlap results to a bin-by-gene matrix<a href="visium-hd.html#convert-overlap-results-to-a-bin-by-gene-matrix" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb194"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb194-1"><a href="visium-hd.html#cb194-1" tabindex="-1"></a><span class="co"># convert overlap results to bin by gene matrix</span></span>
<span id="cb194-2"><a href="visium-hd.html#cb194-2" tabindex="-1"></a>visiumHD <span class="ot">=</span> <span class="fu">overlapToMatrix</span>(visiumHD,</span>
<span id="cb194-3"><a href="visium-hd.html#cb194-3" tabindex="-1"></a>                           <span class="at">poly_info =</span> <span class="st">&#39;hex400&#39;</span>,</span>
<span id="cb194-4"><a href="visium-hd.html#cb194-4" tabindex="-1"></a>                           <span class="at">feat_info =</span> <span class="st">&#39;rna&#39;</span>,</span>
<span id="cb194-5"><a href="visium-hd.html#cb194-5" tabindex="-1"></a>                           <span class="at">name =</span> <span class="st">&#39;raw&#39;</span>)</span>
<span id="cb194-6"><a href="visium-hd.html#cb194-6" tabindex="-1"></a></span>
<span id="cb194-7"><a href="visium-hd.html#cb194-7" tabindex="-1"></a><span class="fu">activeSpatUnit</span>(visiumHD)</span></code></pre></div>
</div>
<div id="default-processing-steps" class="section level4 hasAnchor" number="9.4.4.3">
<h4><span class="header-section-number">9.4.4.3</span> default processing steps<a href="visium-hd.html#default-processing-steps" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb195"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb195-1"><a href="visium-hd.html#cb195-1" tabindex="-1"></a>visiumHD <span class="ot">&lt;-</span> <span class="fu">filterGiotto</span>(visiumHD,</span>
<span id="cb195-2"><a href="visium-hd.html#cb195-2" tabindex="-1"></a>                         <span class="at">expression_threshold =</span> <span class="dv">1</span>,</span>
<span id="cb195-3"><a href="visium-hd.html#cb195-3" tabindex="-1"></a>                         <span class="at">feat_det_in_min_cells =</span> <span class="dv">5</span>,</span>
<span id="cb195-4"><a href="visium-hd.html#cb195-4" tabindex="-1"></a>                         <span class="at">min_det_feats_per_cell =</span> <span class="dv">25</span>)</span>
<span id="cb195-5"><a href="visium-hd.html#cb195-5" tabindex="-1"></a></span>
<span id="cb195-6"><a href="visium-hd.html#cb195-6" tabindex="-1"></a><span class="co"># normalize</span></span>
<span id="cb195-7"><a href="visium-hd.html#cb195-7" tabindex="-1"></a>visiumHD <span class="ot">&lt;-</span> <span class="fu">normalizeGiotto</span>(visiumHD, </span>
<span id="cb195-8"><a href="visium-hd.html#cb195-8" tabindex="-1"></a>                            <span class="at">scalefactor =</span> <span class="dv">1000</span>, </span>
<span id="cb195-9"><a href="visium-hd.html#cb195-9" tabindex="-1"></a>                            <span class="at">verbose =</span> T)</span>
<span id="cb195-10"><a href="visium-hd.html#cb195-10" tabindex="-1"></a></span>
<span id="cb195-11"><a href="visium-hd.html#cb195-11" tabindex="-1"></a><span class="co"># add statistics</span></span>
<span id="cb195-12"><a href="visium-hd.html#cb195-12" tabindex="-1"></a>visiumHD <span class="ot">&lt;-</span> <span class="fu">addStatistics</span>(visiumHD)</span></code></pre></div>
</div>
<div id="visualize-number-of-features" class="section level4 hasAnchor" number="9.4.4.4">
<h4><span class="header-section-number">9.4.4.4</span> visualize number of features<a href="visium-hd.html#visualize-number-of-features" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>At the centroid level.</p>
<div class="sourceCode" id="cb196"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb196-1"><a href="visium-hd.html#cb196-1" tabindex="-1"></a><span class="co"># each dot here represents a 200x200 aggregation of spatial barcodes (bin size 200)</span></span>
<span id="cb196-2"><a href="visium-hd.html#cb196-2" tabindex="-1"></a><span class="fu">spatPlot2D</span>(<span class="at">gobject =</span> visiumHD,</span>
<span id="cb196-3"><a href="visium-hd.html#cb196-3" tabindex="-1"></a>           <span class="at">cell_color =</span> <span class="st">&quot;nr_feats&quot;</span>,</span>
<span id="cb196-4"><a href="visium-hd.html#cb196-4" tabindex="-1"></a>           <span class="at">color_as_factor =</span> F,</span>
<span id="cb196-5"><a href="visium-hd.html#cb196-5" tabindex="-1"></a>           <span class="at">point_size =</span> <span class="fl">2.5</span>)</span></code></pre></div>
<p><img src="img/02_session2/2-spatPlot2D.png" width="100%" style="display: block; margin: auto;" /></p>
<p>Using the spatial polygon (hexagon) tiles</p>
<div class="sourceCode" id="cb197"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb197-1"><a href="visium-hd.html#cb197-1" tabindex="-1"></a><span class="fu">spatInSituPlotPoints</span>(visiumHD,</span>
<span id="cb197-2"><a href="visium-hd.html#cb197-2" tabindex="-1"></a>                     <span class="at">show_image =</span> F,</span>
<span id="cb197-3"><a href="visium-hd.html#cb197-3" tabindex="-1"></a>                     <span class="at">feats =</span> <span class="cn">NULL</span>,</span>
<span id="cb197-4"><a href="visium-hd.html#cb197-4" tabindex="-1"></a>                     <span class="at">show_legend =</span> F,</span>
<span id="cb197-5"><a href="visium-hd.html#cb197-5" tabindex="-1"></a>                     <span class="at">spat_unit =</span> <span class="st">&#39;hex400&#39;</span>,</span>
<span id="cb197-6"><a href="visium-hd.html#cb197-6" tabindex="-1"></a>                     <span class="at">point_size =</span> <span class="fl">0.1</span>,</span>
<span id="cb197-7"><a href="visium-hd.html#cb197-7" tabindex="-1"></a>                     <span class="at">show_polygon =</span> <span class="cn">TRUE</span>,</span>
<span id="cb197-8"><a href="visium-hd.html#cb197-8" tabindex="-1"></a>                     <span class="at">use_overlap =</span> <span class="cn">TRUE</span>,</span>
<span id="cb197-9"><a href="visium-hd.html#cb197-9" tabindex="-1"></a>                     <span class="at">polygon_feat_type =</span> <span class="st">&#39;hex400&#39;</span>, </span>
<span id="cb197-10"><a href="visium-hd.html#cb197-10" tabindex="-1"></a>                     <span class="at">polygon_fill =</span> <span class="st">&#39;nr_feats&#39;</span>, </span>
<span id="cb197-11"><a href="visium-hd.html#cb197-11" tabindex="-1"></a>                     <span class="at">polygon_fill_as_factor =</span> F,</span>
<span id="cb197-12"><a href="visium-hd.html#cb197-12" tabindex="-1"></a>                     <span class="at">polygon_bg_color =</span> <span class="cn">NA</span>,</span>
<span id="cb197-13"><a href="visium-hd.html#cb197-13" tabindex="-1"></a>                     <span class="at">polygon_color =</span> <span class="st">&#39;white&#39;</span>,</span>
<span id="cb197-14"><a href="visium-hd.html#cb197-14" tabindex="-1"></a>                     <span class="at">polygon_line_size =</span> <span class="fl">0.1</span>)</span></code></pre></div>
<p><img src="img/02_session2/3-spatInSituPlotPoints.png" width="100%" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="dimension-reduction-clustering" class="section level3 hasAnchor" number="9.4.5">
<h3><span class="header-section-number">9.4.5</span> Dimension reduction + clustering<a href="visium-hd.html#dimension-reduction-clustering" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="highly-variable-features-pca" class="section level4 hasAnchor" number="9.4.5.1">
<h4><span class="header-section-number">9.4.5.1</span> Highly variable features + PCA<a href="visium-hd.html#highly-variable-features-pca" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb198"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb198-1"><a href="visium-hd.html#cb198-1" tabindex="-1"></a>visiumHD <span class="ot">&lt;-</span> <span class="fu">calculateHVF</span>(visiumHD, </span>
<span id="cb198-2"><a href="visium-hd.html#cb198-2" tabindex="-1"></a>                         <span class="at">zscore_threshold =</span> <span class="dv">1</span>)</span>
<span id="cb198-3"><a href="visium-hd.html#cb198-3" tabindex="-1"></a>visiumHD <span class="ot">&lt;-</span> <span class="fu">runPCA</span>(visiumHD, </span>
<span id="cb198-4"><a href="visium-hd.html#cb198-4" tabindex="-1"></a>                   <span class="at">expression_values =</span> <span class="st">&#39;normalized&#39;</span>, </span>
<span id="cb198-5"><a href="visium-hd.html#cb198-5" tabindex="-1"></a>                   <span class="at">feats_to_use =</span> <span class="st">&#39;hvf&#39;</span>)</span>
<span id="cb198-6"><a href="visium-hd.html#cb198-6" tabindex="-1"></a><span class="fu">screePlot</span>(visiumHD, <span class="at">ncp =</span> <span class="dv">30</span>)</span></code></pre></div>
<p><img src="img/02_session2/4-screePlot.png" width="100%" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb199"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb199-1"><a href="visium-hd.html#cb199-1" tabindex="-1"></a><span class="fu">plotPCA</span>(visiumHD)</span></code></pre></div>
<p><img src="img/02_session2/5-PCA.png" width="100%" style="display: block; margin: auto;" /></p>
</div>
<div id="umap-reduction-for-visualization" class="section level4 hasAnchor" number="9.4.5.2">
<h4><span class="header-section-number">9.4.5.2</span> UMAP reduction for visualization<a href="visium-hd.html#umap-reduction-for-visualization" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb200"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb200-1"><a href="visium-hd.html#cb200-1" tabindex="-1"></a>visiumHD <span class="ot">&lt;-</span> <span class="fu">runUMAP</span>(visiumHD, </span>
<span id="cb200-2"><a href="visium-hd.html#cb200-2" tabindex="-1"></a>                    <span class="at">dimensions_to_use =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">14</span>, </span>
<span id="cb200-3"><a href="visium-hd.html#cb200-3" tabindex="-1"></a>                    <span class="at">n_threads =</span> <span class="dv">10</span>)</span>
<span id="cb200-4"><a href="visium-hd.html#cb200-4" tabindex="-1"></a></span>
<span id="cb200-5"><a href="visium-hd.html#cb200-5" tabindex="-1"></a><span class="fu">plotUMAP</span>(<span class="at">gobject =</span> visiumHD,</span>
<span id="cb200-6"><a href="visium-hd.html#cb200-6" tabindex="-1"></a>         <span class="at">point_size =</span> <span class="dv">1</span>)</span></code></pre></div>
<p><img src="img/02_session2/6-UMAP.png" width="100%" style="display: block; margin: auto;" /></p>
</div>
<div id="create-network-based-on-expression-similarity-graph-partition-cluster" class="section level4 hasAnchor" number="9.4.5.3">
<h4><span class="header-section-number">9.4.5.3</span> Create network based on expression similarity + graph partition cluster<a href="visium-hd.html#create-network-based-on-expression-similarity-graph-partition-cluster" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb201"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb201-1"><a href="visium-hd.html#cb201-1" tabindex="-1"></a><span class="co"># sNN network (default)</span></span>
<span id="cb201-2"><a href="visium-hd.html#cb201-2" tabindex="-1"></a>visiumHD <span class="ot">&lt;-</span> <span class="fu">createNearestNetwork</span>(visiumHD, </span>
<span id="cb201-3"><a href="visium-hd.html#cb201-3" tabindex="-1"></a>                                 <span class="at">dimensions_to_use =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">14</span>, </span>
<span id="cb201-4"><a href="visium-hd.html#cb201-4" tabindex="-1"></a>                                 <span class="at">k =</span> <span class="dv">5</span>)</span>
<span id="cb201-5"><a href="visium-hd.html#cb201-5" tabindex="-1"></a></span>
<span id="cb201-6"><a href="visium-hd.html#cb201-6" tabindex="-1"></a></span>
<span id="cb201-7"><a href="visium-hd.html#cb201-7" tabindex="-1"></a><span class="do">## leiden clustering ####</span></span>
<span id="cb201-8"><a href="visium-hd.html#cb201-8" tabindex="-1"></a>visiumHD <span class="ot">&lt;-</span> <span class="fu">doLeidenClusterIgraph</span>(visiumHD, <span class="at">resolution =</span> <span class="fl">0.5</span>, <span class="at">n_iterations =</span> <span class="dv">1000</span>, <span class="at">spat_unit =</span> <span class="st">&#39;hex400&#39;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb202"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb202-1"><a href="visium-hd.html#cb202-1" tabindex="-1"></a><span class="fu">plotUMAP</span>(<span class="at">gobject =</span> visiumHD,</span>
<span id="cb202-2"><a href="visium-hd.html#cb202-2" tabindex="-1"></a>         <span class="at">cell_color =</span> <span class="st">&#39;leiden_clus&#39;</span>, </span>
<span id="cb202-3"><a href="visium-hd.html#cb202-3" tabindex="-1"></a>         <span class="at">point_size =</span> <span class="fl">1.5</span>,</span>
<span id="cb202-4"><a href="visium-hd.html#cb202-4" tabindex="-1"></a>         <span class="at">show_NN_network =</span> F, </span>
<span id="cb202-5"><a href="visium-hd.html#cb202-5" tabindex="-1"></a>         <span class="at">edge_alpha =</span> <span class="fl">0.05</span>)</span></code></pre></div>
<p><img src="img/02_session2/7-UMAP.png" width="100%" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb203"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb203-1"><a href="visium-hd.html#cb203-1" tabindex="-1"></a><span class="fu">spatInSituPlotPoints</span>(visiumHD,</span>
<span id="cb203-2"><a href="visium-hd.html#cb203-2" tabindex="-1"></a>                     <span class="at">show_image =</span> F,</span>
<span id="cb203-3"><a href="visium-hd.html#cb203-3" tabindex="-1"></a>                     <span class="at">feats =</span> <span class="cn">NULL</span>,</span>
<span id="cb203-4"><a href="visium-hd.html#cb203-4" tabindex="-1"></a>                     <span class="at">show_legend =</span> F,</span>
<span id="cb203-5"><a href="visium-hd.html#cb203-5" tabindex="-1"></a>                     <span class="at">spat_unit =</span> <span class="st">&#39;hex400&#39;</span>,</span>
<span id="cb203-6"><a href="visium-hd.html#cb203-6" tabindex="-1"></a>                     <span class="at">point_size =</span> <span class="fl">0.25</span>,</span>
<span id="cb203-7"><a href="visium-hd.html#cb203-7" tabindex="-1"></a>                     <span class="at">show_polygon =</span> <span class="cn">TRUE</span>,</span>
<span id="cb203-8"><a href="visium-hd.html#cb203-8" tabindex="-1"></a>                     <span class="at">use_overlap =</span> <span class="cn">FALSE</span>,</span>
<span id="cb203-9"><a href="visium-hd.html#cb203-9" tabindex="-1"></a>                     <span class="at">polygon_feat_type =</span> <span class="st">&#39;hex400&#39;</span>,</span>
<span id="cb203-10"><a href="visium-hd.html#cb203-10" tabindex="-1"></a>                     <span class="at">polygon_fill_as_factor =</span> <span class="cn">TRUE</span>,</span>
<span id="cb203-11"><a href="visium-hd.html#cb203-11" tabindex="-1"></a>                     <span class="at">polygon_fill =</span> <span class="st">&#39;leiden_clus&#39;</span>,</span>
<span id="cb203-12"><a href="visium-hd.html#cb203-12" tabindex="-1"></a>                     <span class="at">polygon_color =</span> <span class="st">&#39;black&#39;</span>,</span>
<span id="cb203-13"><a href="visium-hd.html#cb203-13" tabindex="-1"></a>                     <span class="at">polygon_line_size =</span> <span class="fl">0.3</span>)</span></code></pre></div>
<p><img src="img/02_session2/8-spatInSituPlotPoints.png" width="100%" style="display: block; margin: auto;" /></p>
</div>
</div>
</div>
<div id="hexbin-100" class="section level2 hasAnchor" number="9.5">
<h2><span class="header-section-number">9.5</span> Hexbin 100<a href="visium-hd.html#hexbin-100" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Goal is to create a higher resolution bin (hex100) and add to the Giotto object.</p>
<div id="standard-subcellular-pipeline" class="section level3 hasAnchor" number="9.5.1">
<h3><span class="header-section-number">9.5.1</span> Standard subcellular pipeline<a href="visium-hd.html#standard-subcellular-pipeline" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb204"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb204-1"><a href="visium-hd.html#cb204-1" tabindex="-1"></a>hexbin100 <span class="ot">&lt;-</span> <span class="fu">tessellate</span>(<span class="at">extent =</span> <span class="fu">ext</span>(visiumHD), </span>
<span id="cb204-2"><a href="visium-hd.html#cb204-2" tabindex="-1"></a>                        <span class="at">shape =</span> <span class="st">&#39;hexagon&#39;</span>, </span>
<span id="cb204-3"><a href="visium-hd.html#cb204-3" tabindex="-1"></a>                        <span class="at">shape_size =</span> <span class="dv">100</span>, </span>
<span id="cb204-4"><a href="visium-hd.html#cb204-4" tabindex="-1"></a>                        <span class="at">name =</span> <span class="st">&#39;hex100&#39;</span>) </span>
<span id="cb204-5"><a href="visium-hd.html#cb204-5" tabindex="-1"></a></span>
<span id="cb204-6"><a href="visium-hd.html#cb204-6" tabindex="-1"></a>visiumHD <span class="ot">=</span> <span class="fu">setPolygonInfo</span>(<span class="at">gobject =</span> visiumHD,</span>
<span id="cb204-7"><a href="visium-hd.html#cb204-7" tabindex="-1"></a>                          <span class="at">x =</span> hexbin100,</span>
<span id="cb204-8"><a href="visium-hd.html#cb204-8" tabindex="-1"></a>                          <span class="at">name =</span> <span class="st">&#39;hex100&#39;</span>,</span>
<span id="cb204-9"><a href="visium-hd.html#cb204-9" tabindex="-1"></a>                          <span class="at">initialize =</span> T)</span>
<span id="cb204-10"><a href="visium-hd.html#cb204-10" tabindex="-1"></a></span>
<span id="cb204-11"><a href="visium-hd.html#cb204-11" tabindex="-1"></a>visiumHD <span class="ot">=</span> <span class="fu">addSpatialCentroidLocations</span>(<span class="at">gobject =</span> visiumHD,</span>
<span id="cb204-12"><a href="visium-hd.html#cb204-12" tabindex="-1"></a>                                       <span class="at">poly_info =</span> <span class="st">&#39;hex100&#39;</span>)</span></code></pre></div>
<p>Set active spatial unit. This can also be set manually.</p>
<div class="sourceCode" id="cb205"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb205-1"><a href="visium-hd.html#cb205-1" tabindex="-1"></a><span class="fu">activeSpatUnit</span>(visiumHD) <span class="ot">&lt;-</span> <span class="st">&#39;hex100&#39;</span></span></code></pre></div>
<div class="sourceCode" id="cb206"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb206-1"><a href="visium-hd.html#cb206-1" tabindex="-1"></a><span class="fu">spatInSituPlotPoints</span>(visiumHD,</span>
<span id="cb206-2"><a href="visium-hd.html#cb206-2" tabindex="-1"></a>                     <span class="at">show_image =</span> F,</span>
<span id="cb206-3"><a href="visium-hd.html#cb206-3" tabindex="-1"></a>                     <span class="at">feats =</span> <span class="fu">list</span>(<span class="st">&#39;rna&#39;</span> <span class="ot">=</span> feature_data<span class="sc">$</span>feat_ID[<span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>]),</span>
<span id="cb206-4"><a href="visium-hd.html#cb206-4" tabindex="-1"></a>                     <span class="at">show_legend =</span> T,</span>
<span id="cb206-5"><a href="visium-hd.html#cb206-5" tabindex="-1"></a>                     <span class="at">spat_unit =</span> <span class="st">&#39;hex100&#39;</span>,</span>
<span id="cb206-6"><a href="visium-hd.html#cb206-6" tabindex="-1"></a>                     <span class="at">point_size =</span> <span class="fl">0.1</span>,</span>
<span id="cb206-7"><a href="visium-hd.html#cb206-7" tabindex="-1"></a>                     <span class="at">show_polygon =</span> <span class="cn">TRUE</span>,</span>
<span id="cb206-8"><a href="visium-hd.html#cb206-8" tabindex="-1"></a>                     <span class="at">use_overlap =</span> <span class="cn">FALSE</span>,</span>
<span id="cb206-9"><a href="visium-hd.html#cb206-9" tabindex="-1"></a>                     <span class="at">polygon_feat_type =</span> <span class="st">&#39;hex100&#39;</span>,</span>
<span id="cb206-10"><a href="visium-hd.html#cb206-10" tabindex="-1"></a>                     <span class="at">polygon_bg_color =</span> <span class="cn">NA</span>,</span>
<span id="cb206-11"><a href="visium-hd.html#cb206-11" tabindex="-1"></a>                     <span class="at">polygon_color =</span> <span class="st">&#39;white&#39;</span>,</span>
<span id="cb206-12"><a href="visium-hd.html#cb206-12" tabindex="-1"></a>                     <span class="at">polygon_line_size =</span> <span class="fl">0.2</span>,</span>
<span id="cb206-13"><a href="visium-hd.html#cb206-13" tabindex="-1"></a>                     <span class="at">expand_counts =</span> <span class="cn">TRUE</span>, </span>
<span id="cb206-14"><a href="visium-hd.html#cb206-14" tabindex="-1"></a>                     <span class="at">count_info_column =</span> <span class="st">&#39;count&#39;</span>,</span>
<span id="cb206-15"><a href="visium-hd.html#cb206-15" tabindex="-1"></a>                     <span class="at">jitter =</span> <span class="fu">c</span>(<span class="dv">25</span>,<span class="dv">25</span>))</span></code></pre></div>
<p><img src="img/02_session2/9-spatInSituPlotPoints.png" width="100%" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb207"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb207-1"><a href="visium-hd.html#cb207-1" tabindex="-1"></a>visiumHD <span class="ot">=</span> <span class="fu">calculateOverlap</span>(visiumHD,</span>
<span id="cb207-2"><a href="visium-hd.html#cb207-2" tabindex="-1"></a>                            <span class="at">spatial_info =</span> <span class="st">&#39;hex100&#39;</span>,</span>
<span id="cb207-3"><a href="visium-hd.html#cb207-3" tabindex="-1"></a>                            <span class="at">feat_info =</span> <span class="st">&#39;rna&#39;</span>)</span>
<span id="cb207-4"><a href="visium-hd.html#cb207-4" tabindex="-1"></a></span>
<span id="cb207-5"><a href="visium-hd.html#cb207-5" tabindex="-1"></a>visiumHD <span class="ot">=</span> <span class="fu">overlapToMatrix</span>(visiumHD,</span>
<span id="cb207-6"><a href="visium-hd.html#cb207-6" tabindex="-1"></a>                           <span class="at">poly_info =</span> <span class="st">&#39;hex100&#39;</span>,</span>
<span id="cb207-7"><a href="visium-hd.html#cb207-7" tabindex="-1"></a>                           <span class="at">feat_info =</span> <span class="st">&#39;rna&#39;</span>,</span>
<span id="cb207-8"><a href="visium-hd.html#cb207-8" tabindex="-1"></a>                           <span class="at">name =</span> <span class="st">&#39;raw&#39;</span>)</span>
<span id="cb207-9"><a href="visium-hd.html#cb207-9" tabindex="-1"></a></span>
<span id="cb207-10"><a href="visium-hd.html#cb207-10" tabindex="-1"></a>visiumHD <span class="ot">&lt;-</span> <span class="fu">filterGiotto</span>(visiumHD,</span>
<span id="cb207-11"><a href="visium-hd.html#cb207-11" tabindex="-1"></a>                         <span class="at">expression_threshold =</span> <span class="dv">1</span>,</span>
<span id="cb207-12"><a href="visium-hd.html#cb207-12" tabindex="-1"></a>                         <span class="at">feat_det_in_min_cells =</span> <span class="dv">10</span>,</span>
<span id="cb207-13"><a href="visium-hd.html#cb207-13" tabindex="-1"></a>                         <span class="at">min_det_feats_per_cell =</span> <span class="dv">10</span>)</span>
<span id="cb207-14"><a href="visium-hd.html#cb207-14" tabindex="-1"></a></span>
<span id="cb207-15"><a href="visium-hd.html#cb207-15" tabindex="-1"></a>visiumHD <span class="ot">&lt;-</span> <span class="fu">normalizeGiotto</span>(visiumHD, <span class="at">scalefactor =</span> <span class="dv">1000</span>, <span class="at">verbose =</span> T)</span>
<span id="cb207-16"><a href="visium-hd.html#cb207-16" tabindex="-1"></a></span>
<span id="cb207-17"><a href="visium-hd.html#cb207-17" tabindex="-1"></a>visiumHD <span class="ot">&lt;-</span> <span class="fu">addStatistics</span>(visiumHD)</span></code></pre></div>
<div class="sourceCode" id="cb208"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb208-1"><a href="visium-hd.html#cb208-1" tabindex="-1"></a><span class="fu">pDataDT</span>(visiumHD, <span class="at">spat_unit =</span> <span class="st">&#39;hex100&#39;</span>)</span>
<span id="cb208-2"><a href="visium-hd.html#cb208-2" tabindex="-1"></a><span class="fu">pDataDT</span>(visiumHD, <span class="at">spat_unit =</span> <span class="st">&#39;hex400&#39;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb209"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb209-1"><a href="visium-hd.html#cb209-1" tabindex="-1"></a><span class="do">## dimension reduction ####</span></span>
<span id="cb209-2"><a href="visium-hd.html#cb209-2" tabindex="-1"></a><span class="co"># --------------------------- #</span></span>
<span id="cb209-3"><a href="visium-hd.html#cb209-3" tabindex="-1"></a>visiumHD <span class="ot">&lt;-</span> <span class="fu">calculateHVF</span>(visiumHD, <span class="at">zscore_threshold =</span> <span class="dv">1</span>)</span>
<span id="cb209-4"><a href="visium-hd.html#cb209-4" tabindex="-1"></a></span>
<span id="cb209-5"><a href="visium-hd.html#cb209-5" tabindex="-1"></a>visiumHD <span class="ot">&lt;-</span> <span class="fu">runPCA</span>(visiumHD, <span class="at">expression_values =</span> <span class="st">&#39;normalized&#39;</span>, <span class="at">feats_to_use =</span> <span class="st">&#39;hvf&#39;</span>)</span>
<span id="cb209-6"><a href="visium-hd.html#cb209-6" tabindex="-1"></a></span>
<span id="cb209-7"><a href="visium-hd.html#cb209-7" tabindex="-1"></a><span class="fu">plotPCA</span>(visiumHD)</span></code></pre></div>
<p><img src="img/02_session2/10-PCA.png" width="100%" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb210"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb210-1"><a href="visium-hd.html#cb210-1" tabindex="-1"></a>visiumHD <span class="ot">&lt;-</span> <span class="fu">runUMAP</span>(visiumHD, <span class="at">dimensions_to_use =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">14</span>, <span class="at">n_threads =</span> <span class="dv">10</span>)</span>
<span id="cb210-2"><a href="visium-hd.html#cb210-2" tabindex="-1"></a></span>
<span id="cb210-3"><a href="visium-hd.html#cb210-3" tabindex="-1"></a><span class="co"># plot UMAP, coloring cells/points based on nr_feats</span></span>
<span id="cb210-4"><a href="visium-hd.html#cb210-4" tabindex="-1"></a><span class="fu">plotUMAP</span>(<span class="at">gobject =</span> visiumHD,</span>
<span id="cb210-5"><a href="visium-hd.html#cb210-5" tabindex="-1"></a>         <span class="at">point_size =</span> <span class="dv">2</span>)</span></code></pre></div>
<p><img src="img/02_session2/11-UMAP.png" width="100%" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb211"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb211-1"><a href="visium-hd.html#cb211-1" tabindex="-1"></a><span class="co"># sNN network (default)</span></span>
<span id="cb211-2"><a href="visium-hd.html#cb211-2" tabindex="-1"></a>visiumHD <span class="ot">&lt;-</span> <span class="fu">createNearestNetwork</span>(visiumHD, </span>
<span id="cb211-3"><a href="visium-hd.html#cb211-3" tabindex="-1"></a>                                 <span class="at">dimensions_to_use =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">14</span>, </span>
<span id="cb211-4"><a href="visium-hd.html#cb211-4" tabindex="-1"></a>                                 <span class="at">k =</span> <span class="dv">5</span>)</span>
<span id="cb211-5"><a href="visium-hd.html#cb211-5" tabindex="-1"></a></span>
<span id="cb211-6"><a href="visium-hd.html#cb211-6" tabindex="-1"></a><span class="do">## leiden clustering ####</span></span>
<span id="cb211-7"><a href="visium-hd.html#cb211-7" tabindex="-1"></a>visiumHD <span class="ot">&lt;-</span> <span class="fu">doLeidenClusterIgraph</span>(visiumHD, <span class="at">resolution =</span> <span class="fl">0.2</span>, <span class="at">n_iterations =</span> <span class="dv">1000</span>)</span>
<span id="cb211-8"><a href="visium-hd.html#cb211-8" tabindex="-1"></a></span>
<span id="cb211-9"><a href="visium-hd.html#cb211-9" tabindex="-1"></a><span class="fu">plotUMAP</span>(<span class="at">gobject =</span> visiumHD,</span>
<span id="cb211-10"><a href="visium-hd.html#cb211-10" tabindex="-1"></a>         <span class="at">cell_color =</span> <span class="st">&#39;leiden_clus&#39;</span>, </span>
<span id="cb211-11"><a href="visium-hd.html#cb211-11" tabindex="-1"></a>         <span class="at">point_size =</span> <span class="fl">1.5</span>,</span>
<span id="cb211-12"><a href="visium-hd.html#cb211-12" tabindex="-1"></a>         <span class="at">show_NN_network =</span> F, </span>
<span id="cb211-13"><a href="visium-hd.html#cb211-13" tabindex="-1"></a>         <span class="at">edge_alpha =</span> <span class="fl">0.05</span>)</span></code></pre></div>
<p><img src="img/02_session2/12-UMAP.png" width="100%" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb212"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb212-1"><a href="visium-hd.html#cb212-1" tabindex="-1"></a><span class="fu">spatInSituPlotPoints</span>(visiumHD,</span>
<span id="cb212-2"><a href="visium-hd.html#cb212-2" tabindex="-1"></a>                     <span class="at">show_image =</span> F,</span>
<span id="cb212-3"><a href="visium-hd.html#cb212-3" tabindex="-1"></a>                     <span class="at">feats =</span> <span class="cn">NULL</span>,</span>
<span id="cb212-4"><a href="visium-hd.html#cb212-4" tabindex="-1"></a>                     <span class="at">show_legend =</span> F,</span>
<span id="cb212-5"><a href="visium-hd.html#cb212-5" tabindex="-1"></a>                     <span class="at">spat_unit =</span> <span class="st">&#39;hex100&#39;</span>,</span>
<span id="cb212-6"><a href="visium-hd.html#cb212-6" tabindex="-1"></a>                     <span class="at">point_size =</span> <span class="fl">0.5</span>,</span>
<span id="cb212-7"><a href="visium-hd.html#cb212-7" tabindex="-1"></a>                     <span class="at">show_polygon =</span> <span class="cn">TRUE</span>,</span>
<span id="cb212-8"><a href="visium-hd.html#cb212-8" tabindex="-1"></a>                     <span class="at">use_overlap =</span> <span class="cn">FALSE</span>,</span>
<span id="cb212-9"><a href="visium-hd.html#cb212-9" tabindex="-1"></a>                     <span class="at">polygon_feat_type =</span> <span class="st">&#39;hex100&#39;</span>,</span>
<span id="cb212-10"><a href="visium-hd.html#cb212-10" tabindex="-1"></a>                     <span class="at">polygon_fill_as_factor =</span> <span class="cn">TRUE</span>,</span>
<span id="cb212-11"><a href="visium-hd.html#cb212-11" tabindex="-1"></a>                     <span class="at">polygon_fill =</span> <span class="st">&#39;leiden_clus&#39;</span>,</span>
<span id="cb212-12"><a href="visium-hd.html#cb212-12" tabindex="-1"></a>                     <span class="at">polygon_color =</span> <span class="st">&#39;black&#39;</span>,</span>
<span id="cb212-13"><a href="visium-hd.html#cb212-13" tabindex="-1"></a>                     <span class="at">polygon_line_size =</span> <span class="fl">0.3</span>)</span></code></pre></div>
<p><img src="img/02_session2/13-spatInSituPlotPoints.png" width="100%" style="display: block; margin: auto;" /></p>
</div>
<div id="spatial-expression-patterns-1" class="section level3 hasAnchor" number="9.5.2">
<h3><span class="header-section-number">9.5.2</span> Spatial expression patterns<a href="visium-hd.html#spatial-expression-patterns-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="identify-single-genes" class="section level4 hasAnchor" number="9.5.2.1">
<h4><span class="header-section-number">9.5.2.1</span> Identify single genes<a href="visium-hd.html#identify-single-genes" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb213"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb213-1"><a href="visium-hd.html#cb213-1" tabindex="-1"></a>featData <span class="ot">=</span> <span class="fu">fDataDT</span>(visiumHD)</span>
<span id="cb213-2"><a href="visium-hd.html#cb213-2" tabindex="-1"></a>hvf_genes <span class="ot">=</span> featData[hvf <span class="sc">==</span> <span class="st">&#39;yes&#39;</span>]<span class="sc">$</span>feat_ID</span>
<span id="cb213-3"><a href="visium-hd.html#cb213-3" tabindex="-1"></a></span>
<span id="cb213-4"><a href="visium-hd.html#cb213-4" tabindex="-1"></a>visiumHD <span class="ot">=</span> <span class="fu">createSpatialNetwork</span>(visiumHD,</span>
<span id="cb213-5"><a href="visium-hd.html#cb213-5" tabindex="-1"></a>                                <span class="at">name =</span> <span class="st">&#39;kNN_network&#39;</span>,</span>
<span id="cb213-6"><a href="visium-hd.html#cb213-6" tabindex="-1"></a>                                <span class="at">spat_unit =</span> <span class="st">&#39;hex100&#39;</span>, </span>
<span id="cb213-7"><a href="visium-hd.html#cb213-7" tabindex="-1"></a>                                <span class="at">method =</span> <span class="st">&#39;kNN&#39;</span>,</span>
<span id="cb213-8"><a href="visium-hd.html#cb213-8" tabindex="-1"></a>                                <span class="at">k =</span> <span class="dv">8</span>)</span>
<span id="cb213-9"><a href="visium-hd.html#cb213-9" tabindex="-1"></a></span>
<span id="cb213-10"><a href="visium-hd.html#cb213-10" tabindex="-1"></a>ranktest <span class="ot">=</span> <span class="fu">binSpect</span>(visiumHD,</span>
<span id="cb213-11"><a href="visium-hd.html#cb213-11" tabindex="-1"></a>                    <span class="at">spat_unit =</span> <span class="st">&#39;hex100&#39;</span>,</span>
<span id="cb213-12"><a href="visium-hd.html#cb213-12" tabindex="-1"></a>                    <span class="at">subset_feats =</span> hvf_genes,</span>
<span id="cb213-13"><a href="visium-hd.html#cb213-13" tabindex="-1"></a>                    <span class="at">bin_method =</span> <span class="st">&#39;rank&#39;</span>,</span>
<span id="cb213-14"><a href="visium-hd.html#cb213-14" tabindex="-1"></a>                    <span class="at">calc_hub =</span> <span class="cn">FALSE</span>,</span>
<span id="cb213-15"><a href="visium-hd.html#cb213-15" tabindex="-1"></a>                    <span class="at">do_fisher_test =</span> <span class="cn">TRUE</span>,</span>
<span id="cb213-16"><a href="visium-hd.html#cb213-16" tabindex="-1"></a>                    <span class="at">spatial_network_name =</span> <span class="st">&#39;kNN_network&#39;</span>)</span></code></pre></div>
<p>Visualize top 2 ranked spatial genes per expression bin:</p>
<div class="sourceCode" id="cb214"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb214-1"><a href="visium-hd.html#cb214-1" tabindex="-1"></a>set0 <span class="ot">=</span> ranktest[high_expr <span class="sc">&lt;</span> <span class="dv">50</span>][<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>]<span class="sc">$</span>feats</span>
<span id="cb214-2"><a href="visium-hd.html#cb214-2" tabindex="-1"></a>set1 <span class="ot">=</span> ranktest[high_expr <span class="sc">&gt;</span> <span class="dv">50</span> <span class="sc">&amp;</span> high_expr <span class="sc">&lt;</span> <span class="dv">100</span>][<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>]<span class="sc">$</span>feats</span>
<span id="cb214-3"><a href="visium-hd.html#cb214-3" tabindex="-1"></a>set2 <span class="ot">=</span> ranktest[high_expr <span class="sc">&gt;</span> <span class="dv">100</span> <span class="sc">&amp;</span> high_expr <span class="sc">&lt;</span> <span class="dv">200</span>][<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>]<span class="sc">$</span>feats</span>
<span id="cb214-4"><a href="visium-hd.html#cb214-4" tabindex="-1"></a>set3 <span class="ot">=</span> ranktest[high_expr <span class="sc">&gt;</span> <span class="dv">200</span> <span class="sc">&amp;</span> high_expr <span class="sc">&lt;</span> <span class="dv">400</span>][<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>]<span class="sc">$</span>feats</span>
<span id="cb214-5"><a href="visium-hd.html#cb214-5" tabindex="-1"></a>set4 <span class="ot">=</span> ranktest[high_expr <span class="sc">&gt;</span> <span class="dv">400</span> <span class="sc">&amp;</span> high_expr <span class="sc">&lt;</span> <span class="dv">1000</span>][<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>]<span class="sc">$</span>feats</span>
<span id="cb214-6"><a href="visium-hd.html#cb214-6" tabindex="-1"></a>set5 <span class="ot">=</span> ranktest[high_expr <span class="sc">&gt;</span> <span class="dv">1000</span>][<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>]<span class="sc">$</span>feats</span>
<span id="cb214-7"><a href="visium-hd.html#cb214-7" tabindex="-1"></a></span>
<span id="cb214-8"><a href="visium-hd.html#cb214-8" tabindex="-1"></a><span class="fu">spatFeatPlot2D</span>(visiumHD,</span>
<span id="cb214-9"><a href="visium-hd.html#cb214-9" tabindex="-1"></a>               <span class="at">expression_values =</span> <span class="st">&#39;scaled&#39;</span>,</span>
<span id="cb214-10"><a href="visium-hd.html#cb214-10" tabindex="-1"></a>               <span class="at">feats =</span> <span class="fu">c</span>(set0, set1, set2),</span>
<span id="cb214-11"><a href="visium-hd.html#cb214-11" tabindex="-1"></a>               <span class="at">gradient_style =</span> <span class="st">&quot;sequential&quot;</span>,</span>
<span id="cb214-12"><a href="visium-hd.html#cb214-12" tabindex="-1"></a>               <span class="at">cell_color_gradient =</span> <span class="fu">c</span>(<span class="st">&#39;blue&#39;</span>, <span class="st">&#39;white&#39;</span>, <span class="st">&#39;yellow&#39;</span>, <span class="st">&#39;orange&#39;</span>, <span class="st">&#39;red&#39;</span>, <span class="st">&#39;darkred&#39;</span>),</span>
<span id="cb214-13"><a href="visium-hd.html#cb214-13" tabindex="-1"></a>               <span class="at">cow_n_col =</span> <span class="dv">2</span>, <span class="at">point_size =</span> <span class="dv">1</span>)</span></code></pre></div>
<p><img src="img/02_session2/14-spatFeatPlot2D.png" width="100%" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb215"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb215-1"><a href="visium-hd.html#cb215-1" tabindex="-1"></a><span class="fu">spatFeatPlot2D</span>(visiumHD,</span>
<span id="cb215-2"><a href="visium-hd.html#cb215-2" tabindex="-1"></a>               <span class="at">expression_values =</span> <span class="st">&#39;scaled&#39;</span>,</span>
<span id="cb215-3"><a href="visium-hd.html#cb215-3" tabindex="-1"></a>               <span class="at">feats =</span> <span class="fu">c</span>(set3, set4, set5),</span>
<span id="cb215-4"><a href="visium-hd.html#cb215-4" tabindex="-1"></a>               <span class="at">gradient_style =</span> <span class="st">&quot;sequential&quot;</span>,</span>
<span id="cb215-5"><a href="visium-hd.html#cb215-5" tabindex="-1"></a>               <span class="at">cell_color_gradient =</span> <span class="fu">c</span>(<span class="st">&#39;blue&#39;</span>, <span class="st">&#39;white&#39;</span>, <span class="st">&#39;yellow&#39;</span>, <span class="st">&#39;orange&#39;</span>, <span class="st">&#39;red&#39;</span>, <span class="st">&#39;darkred&#39;</span>),</span>
<span id="cb215-6"><a href="visium-hd.html#cb215-6" tabindex="-1"></a>               <span class="at">cow_n_col =</span> <span class="dv">2</span>, <span class="at">point_size =</span> <span class="dv">1</span>)</span></code></pre></div>
<p><img src="img/02_session2/15-spatFeatPlot2D.png" width="100%" style="display: block; margin: auto;" /></p>
</div>
<div id="spatial-co-expression-modules-1" class="section level4 hasAnchor" number="9.5.2.2">
<h4><span class="header-section-number">9.5.2.2</span> Spatial co-expression modules<a href="visium-hd.html#spatial-co-expression-modules-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb216"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb216-1"><a href="visium-hd.html#cb216-1" tabindex="-1"></a>ext_spatial_genes <span class="ot">=</span> ranktest[adj.p.value <span class="sc">&lt;</span> <span class="fl">0.001</span>]<span class="sc">$</span>feats</span>
<span id="cb216-2"><a href="visium-hd.html#cb216-2" tabindex="-1"></a></span>
<span id="cb216-3"><a href="visium-hd.html#cb216-3" tabindex="-1"></a>spat_cor_netw_DT <span class="ot">=</span> <span class="fu">detectSpatialCorFeats</span>(visiumHD,</span>
<span id="cb216-4"><a href="visium-hd.html#cb216-4" tabindex="-1"></a>                                         <span class="at">method =</span> <span class="st">&#39;network&#39;</span>,</span>
<span id="cb216-5"><a href="visium-hd.html#cb216-5" tabindex="-1"></a>                                         <span class="at">spatial_network_name =</span> <span class="st">&#39;kNN_network&#39;</span>,</span>
<span id="cb216-6"><a href="visium-hd.html#cb216-6" tabindex="-1"></a>                                         <span class="at">subset_feats =</span> ext_spatial_genes)</span>
<span id="cb216-7"><a href="visium-hd.html#cb216-7" tabindex="-1"></a></span>
<span id="cb216-8"><a href="visium-hd.html#cb216-8" tabindex="-1"></a><span class="co"># cluster spatial genes</span></span>
<span id="cb216-9"><a href="visium-hd.html#cb216-9" tabindex="-1"></a>spat_cor_netw_DT <span class="ot">=</span> <span class="fu">clusterSpatialCorFeats</span>(spat_cor_netw_DT, </span>
<span id="cb216-10"><a href="visium-hd.html#cb216-10" tabindex="-1"></a>                                          <span class="at">name =</span> <span class="st">&#39;spat_netw_clus&#39;</span>, </span>
<span id="cb216-11"><a href="visium-hd.html#cb216-11" tabindex="-1"></a>                                          <span class="at">k =</span> <span class="dv">16</span>)</span>
<span id="cb216-12"><a href="visium-hd.html#cb216-12" tabindex="-1"></a></span>
<span id="cb216-13"><a href="visium-hd.html#cb216-13" tabindex="-1"></a><span class="co"># visualize clusters</span></span>
<span id="cb216-14"><a href="visium-hd.html#cb216-14" tabindex="-1"></a><span class="fu">heatmSpatialCorFeats</span>(visiumHD,</span>
<span id="cb216-15"><a href="visium-hd.html#cb216-15" tabindex="-1"></a>                     <span class="at">spatCorObject =</span> spat_cor_netw_DT,</span>
<span id="cb216-16"><a href="visium-hd.html#cb216-16" tabindex="-1"></a>                     <span class="at">use_clus_name =</span> <span class="st">&#39;spat_netw_clus&#39;</span>,</span>
<span id="cb216-17"><a href="visium-hd.html#cb216-17" tabindex="-1"></a>                     <span class="at">heatmap_legend_param =</span> <span class="fu">list</span>(<span class="at">title =</span> <span class="cn">NULL</span>))</span></code></pre></div>
<p><img src="img/02_session2/16-heatmSpatialCorFeats.png" width="100%" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb217"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb217-1"><a href="visium-hd.html#cb217-1" tabindex="-1"></a><span class="co"># create metagene enrichment score for clusters</span></span>
<span id="cb217-2"><a href="visium-hd.html#cb217-2" tabindex="-1"></a>cluster_genes_DT <span class="ot">=</span> <span class="fu">showSpatialCorFeats</span>(spat_cor_netw_DT, </span>
<span id="cb217-3"><a href="visium-hd.html#cb217-3" tabindex="-1"></a>                                       <span class="at">use_clus_name =</span> <span class="st">&#39;spat_netw_clus&#39;</span>, </span>
<span id="cb217-4"><a href="visium-hd.html#cb217-4" tabindex="-1"></a>                                       <span class="at">show_top_feats =</span> <span class="dv">1</span>)</span>
<span id="cb217-5"><a href="visium-hd.html#cb217-5" tabindex="-1"></a>cluster_genes <span class="ot">=</span> cluster_genes_DT<span class="sc">$</span>clus; <span class="fu">names</span>(cluster_genes) <span class="ot">=</span> cluster_genes_DT<span class="sc">$</span>feat_ID</span>
<span id="cb217-6"><a href="visium-hd.html#cb217-6" tabindex="-1"></a></span>
<span id="cb217-7"><a href="visium-hd.html#cb217-7" tabindex="-1"></a>visiumHD <span class="ot">=</span> <span class="fu">createMetafeats</span>(visiumHD,</span>
<span id="cb217-8"><a href="visium-hd.html#cb217-8" tabindex="-1"></a>                           <span class="at">expression_values =</span> <span class="st">&#39;normalized&#39;</span>,</span>
<span id="cb217-9"><a href="visium-hd.html#cb217-9" tabindex="-1"></a>                           <span class="at">feat_clusters =</span> cluster_genes,</span>
<span id="cb217-10"><a href="visium-hd.html#cb217-10" tabindex="-1"></a>                           <span class="at">name =</span> <span class="st">&#39;cluster_metagene&#39;</span>)</span>
<span id="cb217-11"><a href="visium-hd.html#cb217-11" tabindex="-1"></a></span>
<span id="cb217-12"><a href="visium-hd.html#cb217-12" tabindex="-1"></a><span class="fu">showGiottoSpatEnrichments</span>(visiumHD)</span></code></pre></div>
<div class="sourceCode" id="cb218"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb218-1"><a href="visium-hd.html#cb218-1" tabindex="-1"></a><span class="fu">spatCellPlot</span>(visiumHD,</span>
<span id="cb218-2"><a href="visium-hd.html#cb218-2" tabindex="-1"></a>             <span class="at">spat_enr_names =</span> <span class="st">&#39;cluster_metagene&#39;</span>,</span>
<span id="cb218-3"><a href="visium-hd.html#cb218-3" tabindex="-1"></a>             <span class="at">gradient_style =</span> <span class="st">&quot;sequential&quot;</span>,</span>
<span id="cb218-4"><a href="visium-hd.html#cb218-4" tabindex="-1"></a>             <span class="at">cell_color_gradient =</span> <span class="fu">c</span>(<span class="st">&#39;blue&#39;</span>, <span class="st">&#39;white&#39;</span>, <span class="st">&#39;yellow&#39;</span>, <span class="st">&#39;orange&#39;</span>, <span class="st">&#39;red&#39;</span>, <span class="st">&#39;darkred&#39;</span>),</span>
<span id="cb218-5"><a href="visium-hd.html#cb218-5" tabindex="-1"></a>             <span class="at">cell_annotation_values =</span> <span class="fu">as.character</span>(<span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>)),</span>
<span id="cb218-6"><a href="visium-hd.html#cb218-6" tabindex="-1"></a>             <span class="at">point_size =</span> <span class="dv">1</span>, <span class="at">cow_n_col =</span> <span class="dv">2</span>)</span></code></pre></div>
<p><img src="img/02_session2/17-spatCellPlot2D.png" width="100%" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb219"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb219-1"><a href="visium-hd.html#cb219-1" tabindex="-1"></a><span class="fu">spatCellPlot</span>(visiumHD,</span>
<span id="cb219-2"><a href="visium-hd.html#cb219-2" tabindex="-1"></a>             <span class="at">spat_enr_names =</span> <span class="st">&#39;cluster_metagene&#39;</span>,</span>
<span id="cb219-3"><a href="visium-hd.html#cb219-3" tabindex="-1"></a>             <span class="at">gradient_style =</span> <span class="st">&quot;sequential&quot;</span>,</span>
<span id="cb219-4"><a href="visium-hd.html#cb219-4" tabindex="-1"></a>             <span class="at">cell_color_gradient =</span> <span class="fu">c</span>(<span class="st">&#39;blue&#39;</span>, <span class="st">&#39;white&#39;</span>, <span class="st">&#39;yellow&#39;</span>, <span class="st">&#39;orange&#39;</span>, <span class="st">&#39;red&#39;</span>, <span class="st">&#39;darkred&#39;</span>),</span>
<span id="cb219-5"><a href="visium-hd.html#cb219-5" tabindex="-1"></a>             <span class="at">cell_annotation_values =</span> <span class="fu">as.character</span>(<span class="fu">c</span>(<span class="dv">5</span><span class="sc">:</span><span class="dv">8</span>)),</span>
<span id="cb219-6"><a href="visium-hd.html#cb219-6" tabindex="-1"></a>             <span class="at">point_size =</span> <span class="dv">1</span>, <span class="at">cow_n_col =</span> <span class="dv">2</span>)</span></code></pre></div>
<p><img src="img/02_session2/18-spatCellPlot2D.png" width="100%" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb220"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb220-1"><a href="visium-hd.html#cb220-1" tabindex="-1"></a><span class="fu">spatCellPlot</span>(visiumHD,</span>
<span id="cb220-2"><a href="visium-hd.html#cb220-2" tabindex="-1"></a>             <span class="at">spat_enr_names =</span> <span class="st">&#39;cluster_metagene&#39;</span>,</span>
<span id="cb220-3"><a href="visium-hd.html#cb220-3" tabindex="-1"></a>             <span class="at">gradient_style =</span> <span class="st">&quot;sequential&quot;</span>,</span>
<span id="cb220-4"><a href="visium-hd.html#cb220-4" tabindex="-1"></a>             <span class="at">cell_color_gradient =</span> <span class="fu">c</span>(<span class="st">&#39;blue&#39;</span>, <span class="st">&#39;white&#39;</span>, <span class="st">&#39;yellow&#39;</span>, <span class="st">&#39;orange&#39;</span>, <span class="st">&#39;red&#39;</span>, <span class="st">&#39;darkred&#39;</span>),</span>
<span id="cb220-5"><a href="visium-hd.html#cb220-5" tabindex="-1"></a>             <span class="at">cell_annotation_values =</span> <span class="fu">as.character</span>(<span class="fu">c</span>(<span class="dv">9</span><span class="sc">:</span><span class="dv">12</span>)),</span>
<span id="cb220-6"><a href="visium-hd.html#cb220-6" tabindex="-1"></a>             <span class="at">point_size =</span> <span class="dv">1</span>, <span class="at">cow_n_col =</span> <span class="dv">2</span>)</span></code></pre></div>
<p><img src="img/02_session2/19-spatCellPlot2D.png" width="100%" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb221"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb221-1"><a href="visium-hd.html#cb221-1" tabindex="-1"></a><span class="fu">spatCellPlot</span>(visiumHD,</span>
<span id="cb221-2"><a href="visium-hd.html#cb221-2" tabindex="-1"></a>             <span class="at">spat_enr_names =</span> <span class="st">&#39;cluster_metagene&#39;</span>,</span>
<span id="cb221-3"><a href="visium-hd.html#cb221-3" tabindex="-1"></a>             <span class="at">gradient_style =</span> <span class="st">&quot;sequential&quot;</span>,</span>
<span id="cb221-4"><a href="visium-hd.html#cb221-4" tabindex="-1"></a>             <span class="at">cell_color_gradient =</span> <span class="fu">c</span>(<span class="st">&#39;blue&#39;</span>, <span class="st">&#39;white&#39;</span>, <span class="st">&#39;yellow&#39;</span>, <span class="st">&#39;orange&#39;</span>, <span class="st">&#39;red&#39;</span>, <span class="st">&#39;darkred&#39;</span>),</span>
<span id="cb221-5"><a href="visium-hd.html#cb221-5" tabindex="-1"></a>             <span class="at">cell_annotation_values =</span> <span class="fu">as.character</span>(<span class="fu">c</span>(<span class="dv">13</span><span class="sc">:</span><span class="dv">16</span>)),</span>
<span id="cb221-6"><a href="visium-hd.html#cb221-6" tabindex="-1"></a>             <span class="at">point_size =</span> <span class="dv">1</span>, <span class="at">cow_n_col =</span> <span class="dv">2</span>)</span></code></pre></div>
<p><img src="img/02_session2/20-spatCellPlot2D.png" width="100%" style="display: block; margin: auto;" /></p>
</div>
<div id="plot-spatial-gene-groups" class="section level4 hasAnchor" number="9.5.2.3">
<h4><span class="header-section-number">9.5.2.3</span> Plot spatial gene groups<a href="visium-hd.html#plot-spatial-gene-groups" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb222"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb222-1"><a href="visium-hd.html#cb222-1" tabindex="-1"></a>balanced_genes <span class="ot">=</span> <span class="fu">getBalancedSpatCoexpressionFeats</span>(<span class="at">spatCorObject =</span> spat_cor_netw_DT, </span>
<span id="cb222-2"><a href="visium-hd.html#cb222-2" tabindex="-1"></a>                                                  <span class="at">maximum =</span> <span class="dv">5</span>)</span>
<span id="cb222-3"><a href="visium-hd.html#cb222-3" tabindex="-1"></a>selected_feats <span class="ot">=</span> <span class="fu">names</span>(balanced_genes)</span>
<span id="cb222-4"><a href="visium-hd.html#cb222-4" tabindex="-1"></a></span>
<span id="cb222-5"><a href="visium-hd.html#cb222-5" tabindex="-1"></a><span class="co"># give genes from same cluster same color</span></span>
<span id="cb222-6"><a href="visium-hd.html#cb222-6" tabindex="-1"></a>distinct_colors <span class="ot">=</span> <span class="fu">getDistinctColors</span>(<span class="at">n =</span> <span class="dv">20</span>)</span>
<span id="cb222-7"><a href="visium-hd.html#cb222-7" tabindex="-1"></a><span class="fu">names</span>(distinct_colors) <span class="ot">=</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">20</span></span>
<span id="cb222-8"><a href="visium-hd.html#cb222-8" tabindex="-1"></a></span>
<span id="cb222-9"><a href="visium-hd.html#cb222-9" tabindex="-1"></a>my_colors <span class="ot">=</span> distinct_colors[balanced_genes]</span>
<span id="cb222-10"><a href="visium-hd.html#cb222-10" tabindex="-1"></a><span class="fu">names</span>(my_colors) <span class="ot">=</span> <span class="fu">names</span>(balanced_genes)</span>
<span id="cb222-11"><a href="visium-hd.html#cb222-11" tabindex="-1"></a></span>
<span id="cb222-12"><a href="visium-hd.html#cb222-12" tabindex="-1"></a><span class="fu">spatInSituPlotPoints</span>(visiumHD,</span>
<span id="cb222-13"><a href="visium-hd.html#cb222-13" tabindex="-1"></a>                     <span class="at">show_image =</span> F,</span>
<span id="cb222-14"><a href="visium-hd.html#cb222-14" tabindex="-1"></a>                     <span class="at">feats =</span> <span class="fu">list</span>(<span class="st">&#39;rna&#39;</span> <span class="ot">=</span> selected_feats), </span>
<span id="cb222-15"><a href="visium-hd.html#cb222-15" tabindex="-1"></a>                     <span class="at">feats_color_code =</span> my_colors,</span>
<span id="cb222-16"><a href="visium-hd.html#cb222-16" tabindex="-1"></a>                     <span class="at">show_legend =</span> F,</span>
<span id="cb222-17"><a href="visium-hd.html#cb222-17" tabindex="-1"></a>                     <span class="at">spat_unit =</span> <span class="st">&#39;hex100&#39;</span>,</span>
<span id="cb222-18"><a href="visium-hd.html#cb222-18" tabindex="-1"></a>                     <span class="at">point_size =</span> <span class="fl">0.20</span>,</span>
<span id="cb222-19"><a href="visium-hd.html#cb222-19" tabindex="-1"></a>                     <span class="at">show_polygon =</span> <span class="cn">FALSE</span>,</span>
<span id="cb222-20"><a href="visium-hd.html#cb222-20" tabindex="-1"></a>                     <span class="at">use_overlap =</span> <span class="cn">FALSE</span>,</span>
<span id="cb222-21"><a href="visium-hd.html#cb222-21" tabindex="-1"></a>                     <span class="at">polygon_feat_type =</span> <span class="st">&#39;hex100&#39;</span>,</span>
<span id="cb222-22"><a href="visium-hd.html#cb222-22" tabindex="-1"></a>                     <span class="at">polygon_bg_color =</span> <span class="cn">NA</span>,</span>
<span id="cb222-23"><a href="visium-hd.html#cb222-23" tabindex="-1"></a>                     <span class="at">polygon_color =</span> <span class="st">&#39;white&#39;</span>,</span>
<span id="cb222-24"><a href="visium-hd.html#cb222-24" tabindex="-1"></a>                     <span class="at">polygon_line_size =</span> <span class="fl">0.01</span>,</span>
<span id="cb222-25"><a href="visium-hd.html#cb222-25" tabindex="-1"></a>                     <span class="at">expand_counts =</span> <span class="cn">TRUE</span>, </span>
<span id="cb222-26"><a href="visium-hd.html#cb222-26" tabindex="-1"></a>                     <span class="at">count_info_column =</span> <span class="st">&#39;count&#39;</span>,</span>
<span id="cb222-27"><a href="visium-hd.html#cb222-27" tabindex="-1"></a>                     <span class="at">jitter =</span> <span class="fu">c</span>(<span class="dv">25</span>,<span class="dv">25</span>))</span></code></pre></div>
<p><img src="img/02_session2/21-spatInSituPlotPoints.png" width="100%" style="display: block; margin: auto;" /></p>
</div>
</div>
</div>
<div id="hexbin-25" class="section level2 hasAnchor" number="9.6">
<h2><span class="header-section-number">9.6</span> Hexbin 25<a href="visium-hd.html#hexbin-25" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Goal is to create a higher resolution bin (hex25) and add to the Giotto object.
We will aim to identify individual cell types and local neighborhood niches.</p>
<div id="subcellular-workflow-and-projection-functions" class="section level3 hasAnchor" number="9.6.1">
<h3><span class="header-section-number">9.6.1</span> Subcellular workflow and projection functions<a href="visium-hd.html#subcellular-workflow-and-projection-functions" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li>filter and normalization workflow</li>
<li>PCA projection</li>
</ul>
<div class="sourceCode" id="cb223"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb223-1"><a href="visium-hd.html#cb223-1" tabindex="-1"></a>visiumHD_subset <span class="ot">=</span> <span class="fu">subsetGiottoLocs</span>(<span class="at">gobject =</span> visiumHD,</span>
<span id="cb223-2"><a href="visium-hd.html#cb223-2" tabindex="-1"></a>                                   <span class="at">x_min =</span> <span class="dv">16000</span>, </span>
<span id="cb223-3"><a href="visium-hd.html#cb223-3" tabindex="-1"></a>                                   <span class="at">x_max =</span> <span class="dv">20000</span>, </span>
<span id="cb223-4"><a href="visium-hd.html#cb223-4" tabindex="-1"></a>                                   <span class="at">y_min =</span> <span class="dv">44250</span>,</span>
<span id="cb223-5"><a href="visium-hd.html#cb223-5" tabindex="-1"></a>                                   <span class="at">y_max =</span> <span class="dv">45500</span>)</span></code></pre></div>
<div class="sourceCode" id="cb224"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb224-1"><a href="visium-hd.html#cb224-1" tabindex="-1"></a>hexbin25 <span class="ot">&lt;-</span> <span class="fu">tessellate</span>(<span class="at">extent =</span> <span class="fu">ext</span>(visiumHD_subset<span class="sc">@</span>feat_info<span class="sc">$</span>rna), </span>
<span id="cb224-2"><a href="visium-hd.html#cb224-2" tabindex="-1"></a>                       <span class="at">shape =</span> <span class="st">&#39;hexagon&#39;</span>, </span>
<span id="cb224-3"><a href="visium-hd.html#cb224-3" tabindex="-1"></a>                       <span class="at">shape_size =</span> <span class="dv">25</span>, </span>
<span id="cb224-4"><a href="visium-hd.html#cb224-4" tabindex="-1"></a>                       <span class="at">name =</span> <span class="st">&#39;hex25&#39;</span>) </span>
<span id="cb224-5"><a href="visium-hd.html#cb224-5" tabindex="-1"></a></span>
<span id="cb224-6"><a href="visium-hd.html#cb224-6" tabindex="-1"></a>visiumHD_subset <span class="ot">=</span> <span class="fu">setPolygonInfo</span>(<span class="at">gobject =</span> visiumHD_subset,</span>
<span id="cb224-7"><a href="visium-hd.html#cb224-7" tabindex="-1"></a>                                 <span class="at">x =</span> hexbin25,</span>
<span id="cb224-8"><a href="visium-hd.html#cb224-8" tabindex="-1"></a>                                 <span class="at">name =</span> <span class="st">&#39;hex25&#39;</span>,</span>
<span id="cb224-9"><a href="visium-hd.html#cb224-9" tabindex="-1"></a>                                 <span class="at">initialize =</span> T)</span>
<span id="cb224-10"><a href="visium-hd.html#cb224-10" tabindex="-1"></a></span>
<span id="cb224-11"><a href="visium-hd.html#cb224-11" tabindex="-1"></a>visiumHD_subset <span class="ot">=</span> <span class="fu">addSpatialCentroidLocations</span>(<span class="at">gobject =</span> visiumHD_subset,</span>
<span id="cb224-12"><a href="visium-hd.html#cb224-12" tabindex="-1"></a>                                              <span class="at">poly_info =</span> <span class="st">&#39;hex25&#39;</span>)</span>
<span id="cb224-13"><a href="visium-hd.html#cb224-13" tabindex="-1"></a></span>
<span id="cb224-14"><a href="visium-hd.html#cb224-14" tabindex="-1"></a><span class="fu">activeSpatUnit</span>(visiumHD_subset) <span class="ot">&lt;-</span> <span class="st">&#39;hex25&#39;</span></span></code></pre></div>
<div class="sourceCode" id="cb225"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb225-1"><a href="visium-hd.html#cb225-1" tabindex="-1"></a>visiumHD_subset <span class="ot">=</span> <span class="fu">calculateOverlap</span>(visiumHD_subset,</span>
<span id="cb225-2"><a href="visium-hd.html#cb225-2" tabindex="-1"></a>                                   <span class="at">spatial_info =</span> <span class="st">&#39;hex25&#39;</span>,</span>
<span id="cb225-3"><a href="visium-hd.html#cb225-3" tabindex="-1"></a>                                   <span class="at">feat_info =</span> <span class="st">&#39;rna&#39;</span>)</span>
<span id="cb225-4"><a href="visium-hd.html#cb225-4" tabindex="-1"></a></span>
<span id="cb225-5"><a href="visium-hd.html#cb225-5" tabindex="-1"></a><span class="co"># convert overlap results to bin by gene matrix</span></span>
<span id="cb225-6"><a href="visium-hd.html#cb225-6" tabindex="-1"></a>visiumHD_subset <span class="ot">=</span> <span class="fu">overlapToMatrix</span>(visiumHD_subset,</span>
<span id="cb225-7"><a href="visium-hd.html#cb225-7" tabindex="-1"></a>                                  <span class="at">poly_info =</span> <span class="st">&#39;hex25&#39;</span>,</span>
<span id="cb225-8"><a href="visium-hd.html#cb225-8" tabindex="-1"></a>                                  <span class="at">feat_info =</span> <span class="st">&#39;rna&#39;</span>,</span>
<span id="cb225-9"><a href="visium-hd.html#cb225-9" tabindex="-1"></a>                                  <span class="at">name =</span> <span class="st">&#39;raw&#39;</span>)</span>
<span id="cb225-10"><a href="visium-hd.html#cb225-10" tabindex="-1"></a></span>
<span id="cb225-11"><a href="visium-hd.html#cb225-11" tabindex="-1"></a>visiumHD_subset <span class="ot">&lt;-</span> <span class="fu">filterGiotto</span>(visiumHD_subset,</span>
<span id="cb225-12"><a href="visium-hd.html#cb225-12" tabindex="-1"></a>                                <span class="at">expression_threshold =</span> <span class="dv">1</span>,</span>
<span id="cb225-13"><a href="visium-hd.html#cb225-13" tabindex="-1"></a>                                <span class="at">feat_det_in_min_cells =</span> <span class="dv">25</span>,</span>
<span id="cb225-14"><a href="visium-hd.html#cb225-14" tabindex="-1"></a>                                <span class="at">min_det_feats_per_cell =</span> <span class="dv">25</span>)</span>
<span id="cb225-15"><a href="visium-hd.html#cb225-15" tabindex="-1"></a></span>
<span id="cb225-16"><a href="visium-hd.html#cb225-16" tabindex="-1"></a><span class="co"># normalize</span></span>
<span id="cb225-17"><a href="visium-hd.html#cb225-17" tabindex="-1"></a>visiumHD_subset <span class="ot">&lt;-</span> <span class="fu">normalizeGiotto</span>(visiumHD_subset, <span class="at">scalefactor =</span> <span class="dv">1000</span>, <span class="at">verbose =</span> T)</span>
<span id="cb225-18"><a href="visium-hd.html#cb225-18" tabindex="-1"></a></span>
<span id="cb225-19"><a href="visium-hd.html#cb225-19" tabindex="-1"></a><span class="co"># add statistics</span></span>
<span id="cb225-20"><a href="visium-hd.html#cb225-20" tabindex="-1"></a>visiumHD_subset <span class="ot">&lt;-</span> <span class="fu">addStatistics</span>(visiumHD_subset)</span></code></pre></div>
</div>
</div>
<div id="database-backend---work-in-progress-but-coming-soon" class="section level2 hasAnchor" number="9.7">
<h2><span class="header-section-number">9.7</span> Database backend - Work in progress, but coming soon!<a href="visium-hd.html#database-backend---work-in-progress-but-coming-soon" class="anchor-section" aria-label="Anchor link to header"></a></h2>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="visium-part-ii.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="xenium-1.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
