<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>9 Visium HD | Workshop: Spatial multi-omics data analysis with Giotto Suite</title>
  <meta name="description" content="9 Visium HD | Workshop: Spatial multi-omics data analysis with Giotto Suite" />
  <meta name="generator" content="bookdown 0.39 and GitBook 2.6.7" />

  <meta property="og:title" content="9 Visium HD | Workshop: Spatial multi-omics data analysis with Giotto Suite" />
  <meta property="og:type" content="book" />
  
  
  <meta name="github-repo" content="drieslab/giotto_workshop_2024" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="9 Visium HD | Workshop: Spatial multi-omics data analysis with Giotto Suite" />
  
  
  

<meta name="author" content="Ruben Dries, Jiaji George Chen, Joselyn Cristina Chávez-Fuentes, Junxiang Xu ,Edward Ruiz, Jeff Sheridan, Iqra Amin, Wen Wang" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="visium-part-ii.html"/>
<link rel="next" href="xenium-1.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Giotto Suite Workshop 2024</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#instructors"><i class="fa fa-check"></i><b>1.1</b> Instructors</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#topics-and-schedule"><i class="fa fa-check"></i><b>1.2</b> Topics and Schedule:</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#license"><i class="fa fa-check"></i><b>1.3</b> License</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="datasets-packages.html"><a href="datasets-packages.html"><i class="fa fa-check"></i><b>2</b> Datasets &amp; Packages</a>
<ul>
<li class="chapter" data-level="2.1" data-path="datasets-packages.html"><a href="datasets-packages.html#datasets-to-download"><i class="fa fa-check"></i><b>2.1</b> Datasets to download</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="datasets-packages.html"><a href="datasets-packages.html#zenodo-repository"><i class="fa fa-check"></i><b>2.1.1</b> Zenodo repository</a></li>
<li class="chapter" data-level="2.1.2" data-path="datasets-packages.html"><a href="datasets-packages.html#x-genomics-visium-mouse-brain-section-coronal-dataset"><i class="fa fa-check"></i><b>2.1.2</b> 10X Genomics Visium Mouse Brain Section (Coronal) dataset</a></li>
<li class="chapter" data-level="2.1.3" data-path="datasets-packages.html"><a href="datasets-packages.html#x-genomics-visium-hd-ffpe-human-colon-cancer"><i class="fa fa-check"></i><b>2.1.3</b> 10X Genomics Visium HD: FFPE Human Colon Cancer</a></li>
<li class="chapter" data-level="2.1.4" data-path="datasets-packages.html"><a href="datasets-packages.html#x-genomics-multi-modal-dataset"><i class="fa fa-check"></i><b>2.1.4</b> 10X Genomics multi-modal dataset</a></li>
<li class="chapter" data-level="2.1.5" data-path="datasets-packages.html"><a href="datasets-packages.html#x-genomics-multi-omics-visium-cytassist-human-tonsil-dataset"><i class="fa fa-check"></i><b>2.1.5</b> 10X Genomics multi-omics Visium CytAssist Human Tonsil dataset</a></li>
<li class="chapter" data-level="2.1.6" data-path="datasets-packages.html"><a href="datasets-packages.html#x-genomics-human-prostate-cancer-adenocarcinoma-with-invasive-carcinoma-ffpe"><i class="fa fa-check"></i><b>2.1.6</b> 10X Genomics Human Prostate Cancer Adenocarcinoma with Invasive Carcinoma (FFPE)</a></li>
<li class="chapter" data-level="2.1.7" data-path="datasets-packages.html"><a href="datasets-packages.html#x-genomics-normal-human-prostate-ffpe"><i class="fa fa-check"></i><b>2.1.7</b> 10X Genomics Normal Human Prostate (FFPE)</a></li>
<li class="chapter" data-level="2.1.8" data-path="datasets-packages.html"><a href="datasets-packages.html#xenium"><i class="fa fa-check"></i><b>2.1.8</b> Xenium</a></li>
<li class="chapter" data-level="2.1.9" data-path="datasets-packages.html"><a href="datasets-packages.html#merfish-cortex-dataset"><i class="fa fa-check"></i><b>2.1.9</b> MERFISH cortex dataset</a></li>
<li class="chapter" data-level="2.1.10" data-path="datasets-packages.html"><a href="datasets-packages.html#lunaphore-if-dataset"><i class="fa fa-check"></i><b>2.1.10</b> Lunaphore IF dataset</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="datasets-packages.html"><a href="datasets-packages.html#needed-packages"><i class="fa fa-check"></i><b>2.2</b> Needed packages</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="datasets-packages.html"><a href="datasets-packages.html#cran-dependencies"><i class="fa fa-check"></i><b>2.2.1</b> CRAN dependencies:</a></li>
<li class="chapter" data-level="2.2.2" data-path="datasets-packages.html"><a href="datasets-packages.html#terra-installation"><i class="fa fa-check"></i><b>2.2.2</b> <em>terra</em> installation</a></li>
<li class="chapter" data-level="2.2.3" data-path="datasets-packages.html"><a href="datasets-packages.html#matrix-installation"><i class="fa fa-check"></i><b>2.2.3</b> <em>Matrix</em> installation</a></li>
<li class="chapter" data-level="2.2.4" data-path="datasets-packages.html"><a href="datasets-packages.html#rtools-installation"><i class="fa fa-check"></i><b>2.2.4</b> <em>Rtools</em> installation</a></li>
<li class="chapter" data-level="2.2.5" data-path="datasets-packages.html"><a href="datasets-packages.html#giotto-installation"><i class="fa fa-check"></i><b>2.2.5</b> <em>Giotto</em> installation</a></li>
<li class="chapter" data-level="2.2.6" data-path="datasets-packages.html"><a href="datasets-packages.html#irlba-install"><i class="fa fa-check"></i><b>2.2.6</b> <em>irlba</em> install</a></li>
<li class="chapter" data-level="2.2.7" data-path="datasets-packages.html"><a href="datasets-packages.html#arrow-install"><i class="fa fa-check"></i><b>2.2.7</b> <em>arrow</em> install</a></li>
<li class="chapter" data-level="2.2.8" data-path="datasets-packages.html"><a href="datasets-packages.html#bioconductor-dependencies"><i class="fa fa-check"></i><b>2.2.8</b> Bioconductor dependencies:</a></li>
<li class="chapter" data-level="2.2.9" data-path="datasets-packages.html"><a href="datasets-packages.html#cran-packages"><i class="fa fa-check"></i><b>2.2.9</b> CRAN packages:</a></li>
<li class="chapter" data-level="2.2.10" data-path="datasets-packages.html"><a href="datasets-packages.html#packages-from-github"><i class="fa fa-check"></i><b>2.2.10</b> Packages from GitHub</a></li>
<li class="chapter" data-level="2.2.11" data-path="datasets-packages.html"><a href="datasets-packages.html#python-environments"><i class="fa fa-check"></i><b>2.2.11</b> Python environments</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html"><i class="fa fa-check"></i><b>3</b> Spatial omics technologies</a>
<ul>
<li class="chapter" data-level="3.1" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#presentation"><i class="fa fa-check"></i><b>3.1</b> Presentation</a></li>
<li class="chapter" data-level="3.2" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#short-summary"><i class="fa fa-check"></i><b>3.2</b> Short summary</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#why-do-we-need-spatial-omics-technologies"><i class="fa fa-check"></i><b>3.2.1</b> Why do we need spatial omics technologies?</a></li>
<li class="chapter" data-level="3.2.2" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#what-is-spatial-omics"><i class="fa fa-check"></i><b>3.2.2</b> What is spatial omics?</a></li>
<li class="chapter" data-level="3.2.3" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#what-are-the-main-spatial-omics-technologies"><i class="fa fa-check"></i><b>3.2.3</b> What are the main spatial omics technologies?</a></li>
<li class="chapter" data-level="3.2.4" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#other-spatial-omics-atac-seq-cuttag-lipidomics-etc"><i class="fa fa-check"></i><b>3.2.4</b> Other Spatial omics: ATAC-seq, CUT&amp;Tag, lipidomics, etc</a></li>
<li class="chapter" data-level="3.2.5" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#what-are-the-different-types-of-spatial-downstream-analyses"><i class="fa fa-check"></i><b>3.2.5</b> What are the different types of spatial downstream analyses?</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html"><i class="fa fa-check"></i><b>4</b> Introduction to the Giotto package</a>
<ul>
<li class="chapter" data-level="4.1" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html#presentation-1"><i class="fa fa-check"></i><b>4.1</b> Presentation</a></li>
<li class="chapter" data-level="4.2" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html#ecosystem"><i class="fa fa-check"></i><b>4.2</b> Ecosystem</a></li>
<li class="chapter" data-level="4.3" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html#installation-python-environment"><i class="fa fa-check"></i><b>4.3</b> Installation + python environment</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html#giotto-installation-1"><i class="fa fa-check"></i><b>4.3.1</b> Giotto installation</a></li>
<li class="chapter" data-level="4.3.2" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html#python-environment"><i class="fa fa-check"></i><b>4.3.2</b> Python environment</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html#giotto-instructions"><i class="fa fa-check"></i><b>4.4</b> Giotto instructions</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html"><i class="fa fa-check"></i><b>5</b> Data formatting and Pre-processing</a>
<ul>
<li class="chapter" data-level="5.1" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#data-formats"><i class="fa fa-check"></i><b>5.1</b> Data formats</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#general-formats"><i class="fa fa-check"></i><b>5.1.1</b> General formats</a></li>
<li class="chapter" data-level="5.1.2" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#matrix-formats"><i class="fa fa-check"></i><b>5.1.2</b> Matrix formats</a></li>
<li class="chapter" data-level="5.1.3" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#tabular-formats"><i class="fa fa-check"></i><b>5.1.3</b> Tabular formats</a></li>
<li class="chapter" data-level="5.1.4" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#spatial-formats"><i class="fa fa-check"></i><b>5.1.4</b> Spatial formats</a></li>
<li class="chapter" data-level="5.1.5" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#mask-files"><i class="fa fa-check"></i><b>5.1.5</b> Mask files</a></li>
<li class="chapter" data-level="5.1.6" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#images"><i class="fa fa-check"></i><b>5.1.6</b> images</a></li>
<li class="chapter" data-level="5.1.7" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#jsonlike-formats"><i class="fa fa-check"></i><b>5.1.7</b> jsonlike formats</a></li>
<li class="chapter" data-level="5.1.8" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#hierarchical-formats"><i class="fa fa-check"></i><b>5.1.8</b> Hierarchical formats</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#pre-processing"><i class="fa fa-check"></i><b>5.2</b> Pre-processing</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#expression-matrix"><i class="fa fa-check"></i><b>5.2.1</b> Expression matrix</a></li>
<li class="chapter" data-level="5.2.2" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#spatial-locations"><i class="fa fa-check"></i><b>5.2.2</b> Spatial locations</a></li>
<li class="chapter" data-level="5.2.3" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#giottopoints"><i class="fa fa-check"></i><b>5.2.3</b> giottoPoints</a></li>
<li class="chapter" data-level="5.2.4" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#giottopolygon"><i class="fa fa-check"></i><b>5.2.4</b> giottoPolygon</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#subobject-utility-functions"><i class="fa fa-check"></i><b>5.3</b> Subobject utility functions</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#ids"><i class="fa fa-check"></i><b>5.3.1</b> IDs</a></li>
<li class="chapter" data-level="5.3.2" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#bracket-subsetting-and-extraction"><i class="fa fa-check"></i><b>5.3.2</b> Bracket subsetting and extraction</a></li>
<li class="chapter" data-level="5.3.3" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#nesting-metadata-generics"><i class="fa fa-check"></i><b>5.3.3</b> Nesting metadata generics</a></li>
<li class="chapter" data-level="5.3.4" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#appending-to-a-giotto-object"><i class="fa fa-check"></i><b>5.3.4</b> Appending to a giotto object</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html"><i class="fa fa-check"></i><b>6</b> Creating a Giotto object</a>
<ul>
<li class="chapter" data-level="6.1" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#overview"><i class="fa fa-check"></i><b>6.1</b> Overview</a></li>
<li class="chapter" data-level="6.2" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#giottodata-modular-package"><i class="fa fa-check"></i><b>6.2</b> GiottoData modular package</a></li>
<li class="chapter" data-level="6.3" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#from-matrix-locations"><i class="fa fa-check"></i><b>6.3</b> From matrix + locations</a></li>
<li class="chapter" data-level="6.4" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#from-subcellular-raw-data-transcripts-or-images-polygons"><i class="fa fa-check"></i><b>6.4</b> From subcellular raw data (transcripts or images) + polygons</a></li>
<li class="chapter" data-level="6.5" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#from-piece-wise"><i class="fa fa-check"></i><b>6.5</b> From piece-wise</a></li>
<li class="chapter" data-level="6.6" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#using-convenience-functions-for-popular-technologies-vizgen-xenium-cosmx"><i class="fa fa-check"></i><b>6.6</b> Using convenience functions for popular technologies (Vizgen, Xenium, CosMx, …)</a></li>
<li class="chapter" data-level="6.7" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#plotting"><i class="fa fa-check"></i><b>6.7</b> Plotting</a>
<ul>
<li class="chapter" data-level="6.7.1" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#subobject-plotting"><i class="fa fa-check"></i><b>6.7.1</b> Subobject plotting</a></li>
<li class="chapter" data-level="6.7.2" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#additive-subobject-plotting"><i class="fa fa-check"></i><b>6.7.2</b> Additive subobject plotting</a></li>
<li class="chapter" data-level="6.7.3" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#giotto-object-plotting"><i class="fa fa-check"></i><b>6.7.3</b> Giotto object plotting</a></li>
</ul></li>
<li class="chapter" data-level="6.8" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#subsetting"><i class="fa fa-check"></i><b>6.8</b> Subsetting</a>
<ul>
<li class="chapter" data-level="6.8.1" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#id-subsetting"><i class="fa fa-check"></i><b>6.8.1</b> ID subsetting</a></li>
<li class="chapter" data-level="6.8.2" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#coordinate-based-subsetting"><i class="fa fa-check"></i><b>6.8.2</b> Coordinate-based subsetting</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="visium-part-i.html"><a href="visium-part-i.html"><i class="fa fa-check"></i><b>7</b> Visium Part I</a>
<ul>
<li class="chapter" data-level="7.1" data-path="visium-part-i.html"><a href="visium-part-i.html#the-visium-technology"><i class="fa fa-check"></i><b>7.1</b> The Visium technology</a></li>
<li class="chapter" data-level="7.2" data-path="visium-part-i.html"><a href="visium-part-i.html#introduction-to-the-spatial-dataset"><i class="fa fa-check"></i><b>7.2</b> Introduction to the spatial dataset</a></li>
<li class="chapter" data-level="7.3" data-path="visium-part-i.html"><a href="visium-part-i.html#download-dataset"><i class="fa fa-check"></i><b>7.3</b> Download dataset</a></li>
<li class="chapter" data-level="7.4" data-path="visium-part-i.html"><a href="visium-part-i.html#create-the-giotto-object"><i class="fa fa-check"></i><b>7.4</b> Create the Giotto object</a></li>
<li class="chapter" data-level="7.5" data-path="visium-part-i.html"><a href="visium-part-i.html#subset-on-spots-that-were-covered-by-tissue"><i class="fa fa-check"></i><b>7.5</b> Subset on spots that were covered by tissue</a></li>
<li class="chapter" data-level="7.6" data-path="visium-part-i.html"><a href="visium-part-i.html#quality-control"><i class="fa fa-check"></i><b>7.6</b> Quality control</a></li>
<li class="chapter" data-level="7.7" data-path="visium-part-i.html"><a href="visium-part-i.html#filtering"><i class="fa fa-check"></i><b>7.7</b> Filtering</a></li>
<li class="chapter" data-level="7.8" data-path="visium-part-i.html"><a href="visium-part-i.html#normalization"><i class="fa fa-check"></i><b>7.8</b> Normalization</a></li>
<li class="chapter" data-level="7.9" data-path="visium-part-i.html"><a href="visium-part-i.html#feature-selection"><i class="fa fa-check"></i><b>7.9</b> Feature selection</a>
<ul>
<li class="chapter" data-level="7.9.1" data-path="visium-part-i.html"><a href="visium-part-i.html#highly-variable-features"><i class="fa fa-check"></i><b>7.9.1</b> Highly Variable Features:</a></li>
</ul></li>
<li class="chapter" data-level="7.10" data-path="visium-part-i.html"><a href="visium-part-i.html#dimension-reduction"><i class="fa fa-check"></i><b>7.10</b> Dimension Reduction</a>
<ul>
<li class="chapter" data-level="7.10.1" data-path="visium-part-i.html"><a href="visium-part-i.html#pca"><i class="fa fa-check"></i><b>7.10.1</b> PCA</a></li>
<li class="chapter" data-level="7.10.2" data-path="visium-part-i.html"><a href="visium-part-i.html#umap"><i class="fa fa-check"></i><b>7.10.2</b> UMAP</a></li>
<li class="chapter" data-level="7.10.3" data-path="visium-part-i.html"><a href="visium-part-i.html#t-sne"><i class="fa fa-check"></i><b>7.10.3</b> t-SNE</a></li>
</ul></li>
<li class="chapter" data-level="7.11" data-path="visium-part-i.html"><a href="visium-part-i.html#clustering"><i class="fa fa-check"></i><b>7.11</b> Clustering</a>
<ul>
<li class="chapter" data-level="7.11.1" data-path="visium-part-i.html"><a href="visium-part-i.html#calculate-leiden-clustering"><i class="fa fa-check"></i><b>7.11.1</b> Calculate Leiden clustering</a></li>
<li class="chapter" data-level="7.11.2" data-path="visium-part-i.html"><a href="visium-part-i.html#calculate-louvain-clustering"><i class="fa fa-check"></i><b>7.11.2</b> Calculate Louvain clustering</a></li>
</ul></li>
<li class="chapter" data-level="7.12" data-path="visium-part-i.html"><a href="visium-part-i.html#save-the-object"><i class="fa fa-check"></i><b>7.12</b> Save the object</a></li>
<li class="chapter" data-level="7.13" data-path="visium-part-i.html"><a href="visium-part-i.html#session-info"><i class="fa fa-check"></i><b>7.13</b> Session info</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="visium-part-ii.html"><a href="visium-part-ii.html"><i class="fa fa-check"></i><b>8</b> Visium Part II</a>
<ul>
<li class="chapter" data-level="8.1" data-path="visium-part-ii.html"><a href="visium-part-ii.html#load-the-object"><i class="fa fa-check"></i><b>8.1</b> Load the object</a></li>
<li class="chapter" data-level="8.2" data-path="visium-part-ii.html"><a href="visium-part-ii.html#differential-expression"><i class="fa fa-check"></i><b>8.2</b> Differential expression</a>
<ul>
<li class="chapter" data-level="8.2.1" data-path="visium-part-ii.html"><a href="visium-part-ii.html#gini-markers"><i class="fa fa-check"></i><b>8.2.1</b> Gini markers</a></li>
<li class="chapter" data-level="8.2.2" data-path="visium-part-ii.html"><a href="visium-part-ii.html#scran-markers"><i class="fa fa-check"></i><b>8.2.2</b> Scran markers</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="visium-part-ii.html"><a href="visium-part-ii.html#enrichment-deconvolution"><i class="fa fa-check"></i><b>8.3</b> Enrichment &amp; Deconvolution</a>
<ul>
<li class="chapter" data-level="8.3.1" data-path="visium-part-ii.html"><a href="visium-part-ii.html#pagerank"><i class="fa fa-check"></i><b>8.3.1</b> PAGE/Rank</a></li>
<li class="chapter" data-level="8.3.2" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatialdwls"><i class="fa fa-check"></i><b>8.3.2</b> SpatialDWLS</a></li>
</ul></li>
<li class="chapter" data-level="8.4" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatial-expression-patterns"><i class="fa fa-check"></i><b>8.4</b> Spatial expression patterns</a>
<ul>
<li class="chapter" data-level="8.4.1" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatial-variable-genes"><i class="fa fa-check"></i><b>8.4.1</b> Spatial variable genes</a></li>
<li class="chapter" data-level="8.4.2" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatial-co-expression-modules"><i class="fa fa-check"></i><b>8.4.2</b> Spatial co-expression modules</a></li>
</ul></li>
<li class="chapter" data-level="8.5" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatially-informed-clusters"><i class="fa fa-check"></i><b>8.5</b> Spatially informed clusters</a></li>
<li class="chapter" data-level="8.6" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatial-domains-hmrf"><i class="fa fa-check"></i><b>8.6</b> Spatial domains HMRF</a></li>
<li class="chapter" data-level="8.7" data-path="visium-part-ii.html"><a href="visium-part-ii.html#interactive-tools"><i class="fa fa-check"></i><b>8.7</b> Interactive tools</a></li>
<li class="chapter" data-level="8.8" data-path="visium-part-ii.html"><a href="visium-part-ii.html#save-the-object-1"><i class="fa fa-check"></i><b>8.8</b> Save the object</a></li>
<li class="chapter" data-level="8.9" data-path="visium-part-ii.html"><a href="visium-part-ii.html#session-info-1"><i class="fa fa-check"></i><b>8.9</b> Session info</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="visium-hd.html"><a href="visium-hd.html"><i class="fa fa-check"></i><b>9</b> Visium HD</a>
<ul>
<li class="chapter" data-level="9.1" data-path="visium-hd.html"><a href="visium-hd.html#objective"><i class="fa fa-check"></i><b>9.1</b> Objective</a></li>
<li class="chapter" data-level="9.2" data-path="visium-hd.html"><a href="visium-hd.html#background"><i class="fa fa-check"></i><b>9.2</b> Background</a>
<ul>
<li class="chapter" data-level="9.2.1" data-path="visium-hd.html"><a href="visium-hd.html#visium-hd-technology"><i class="fa fa-check"></i><b>9.2.1</b> Visium HD Technology</a></li>
<li class="chapter" data-level="9.2.2" data-path="visium-hd.html"><a href="visium-hd.html#colorectal-cancer-sample"><i class="fa fa-check"></i><b>9.2.2</b> Colorectal Cancer Sample</a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="visium-hd.html"><a href="visium-hd.html#data-ingestion"><i class="fa fa-check"></i><b>9.3</b> Data Ingestion</a>
<ul>
<li class="chapter" data-level="9.3.1" data-path="visium-hd.html"><a href="visium-hd.html#visium-hd-output-data-format"><i class="fa fa-check"></i><b>9.3.1</b> Visium HD output data format</a></li>
<li class="chapter" data-level="9.3.2" data-path="visium-hd.html"><a href="visium-hd.html#mini-visium-hd-dataset"><i class="fa fa-check"></i><b>9.3.2</b> Mini Visium HD dataset</a></li>
<li class="chapter" data-level="9.3.3" data-path="visium-hd.html"><a href="visium-hd.html#giotto-visium-hd-convenience-function"><i class="fa fa-check"></i><b>9.3.3</b> Giotto Visium HD convenience function</a></li>
<li class="chapter" data-level="9.3.4" data-path="visium-hd.html"><a href="visium-hd.html#read-in-data-manually"><i class="fa fa-check"></i><b>9.3.4</b> Read in data manually</a></li>
</ul></li>
<li class="chapter" data-level="9.4" data-path="visium-hd.html"><a href="visium-hd.html#hexbin-400-giotto-object"><i class="fa fa-check"></i><b>9.4</b> Hexbin 400 Giotto object</a>
<ul>
<li class="chapter" data-level="9.4.1" data-path="visium-hd.html"><a href="visium-hd.html#create-giotto-points"><i class="fa fa-check"></i><b>9.4.1</b> create giotto points</a></li>
<li class="chapter" data-level="9.4.2" data-path="visium-hd.html"><a href="visium-hd.html#create-giotto-polygons"><i class="fa fa-check"></i><b>9.4.2</b> create giotto polygons</a></li>
<li class="chapter" data-level="9.4.3" data-path="visium-hd.html"><a href="visium-hd.html#combine-giotto-points-and-polygons-to-create-giotto-object"><i class="fa fa-check"></i><b>9.4.3</b> combine Giotto points and polygons to create Giotto object</a></li>
<li class="chapter" data-level="9.4.4" data-path="visium-hd.html"><a href="visium-hd.html#process-giotto-object"><i class="fa fa-check"></i><b>9.4.4</b> Process Giotto object</a></li>
</ul></li>
<li class="chapter" data-level="9.5" data-path="visium-hd.html"><a href="visium-hd.html#hexbin-100"><i class="fa fa-check"></i><b>9.5</b> Hexbin 100</a>
<ul>
<li class="chapter" data-level="9.5.1" data-path="visium-hd.html"><a href="visium-hd.html#standard-subcellular-pipeline"><i class="fa fa-check"></i><b>9.5.1</b> Standard subcellular pipeline</a></li>
<li class="chapter" data-level="9.5.2" data-path="visium-hd.html"><a href="visium-hd.html#spatial-expression-patterns-1"><i class="fa fa-check"></i><b>9.5.2</b> Spatial expression patterns</a></li>
</ul></li>
<li class="chapter" data-level="9.6" data-path="visium-hd.html"><a href="visium-hd.html#hexbin-25"><i class="fa fa-check"></i><b>9.6</b> Hexbin 25</a>
<ul>
<li class="chapter" data-level="9.6.1" data-path="visium-hd.html"><a href="visium-hd.html#subcellular-workflow"><i class="fa fa-check"></i><b>9.6.1</b> Subcellular workflow</a></li>
<li class="chapter" data-level="9.6.2" data-path="visium-hd.html"><a href="visium-hd.html#projections"><i class="fa fa-check"></i><b>9.6.2</b> Projections</a></li>
<li class="chapter" data-level="9.6.3" data-path="visium-hd.html"><a href="visium-hd.html#niche-clustering"><i class="fa fa-check"></i><b>9.6.3</b> Niche clustering</a></li>
</ul></li>
<li class="chapter" data-level="9.7" data-path="visium-hd.html"><a href="visium-hd.html#database-backend---work-in-progress-but-coming-soon"><i class="fa fa-check"></i><b>9.7</b> Database backend - Work in progress, but coming soon!</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="xenium-1.html"><a href="xenium-1.html"><i class="fa fa-check"></i><b>10</b> Xenium</a>
<ul>
<li class="chapter" data-level="10.1" data-path="xenium-1.html"><a href="xenium-1.html#introduction-to-spatial-dataset"><i class="fa fa-check"></i><b>10.1</b> Introduction to spatial dataset</a>
<ul>
<li class="chapter" data-level="10.1.1" data-path="xenium-1.html"><a href="xenium-1.html#output-directory-structure"><i class="fa fa-check"></i><b>10.1.1</b> Output directory structure</a></li>
<li class="chapter" data-level="10.1.2" data-path="xenium-1.html"><a href="xenium-1.html#mini-xenium-dataset"><i class="fa fa-check"></i><b>10.1.2</b> Mini Xenium Dataset</a></li>
</ul></li>
<li class="chapter" data-level="10.2" data-path="xenium-1.html"><a href="xenium-1.html#data-preparation"><i class="fa fa-check"></i><b>10.2</b> Data preparation</a>
<ul>
<li class="chapter" data-level="10.2.1" data-path="xenium-1.html"><a href="xenium-1.html#image-conversion-may-change"><i class="fa fa-check"></i><b>10.2.1</b> Image conversion (may change)</a></li>
</ul></li>
<li class="chapter" data-level="10.3" data-path="xenium-1.html"><a href="xenium-1.html#convenience-function"><i class="fa fa-check"></i><b>10.3</b> Convenience function</a>
<ul>
<li class="chapter" data-level="10.3.1" data-path="xenium-1.html"><a href="xenium-1.html#specific-filepaths"><i class="fa fa-check"></i><b>10.3.1</b> Specific filepaths</a></li>
<li class="chapter" data-level="10.3.2" data-path="xenium-1.html"><a href="xenium-1.html#quality-value"><i class="fa fa-check"></i><b>10.3.2</b> Quality value</a></li>
<li class="chapter" data-level="10.3.3" data-path="xenium-1.html"><a href="xenium-1.html#transcript-type-splitting"><i class="fa fa-check"></i><b>10.3.3</b> Transcript type splitting</a></li>
<li class="chapter" data-level="10.3.4" data-path="xenium-1.html"><a href="xenium-1.html#centroids-calculation"><i class="fa fa-check"></i><b>10.3.4</b> Centroids calculation</a></li>
<li class="chapter" data-level="10.3.5" data-path="xenium-1.html"><a href="xenium-1.html#simple-visualization"><i class="fa fa-check"></i><b>10.3.5</b> Simple visualization</a></li>
</ul></li>
<li class="chapter" data-level="10.4" data-path="xenium-1.html"><a href="xenium-1.html#piecewise-loading"><i class="fa fa-check"></i><b>10.4</b> Piecewise loading</a>
<ul>
<li class="chapter" data-level="10.4.1" data-path="xenium-1.html"><a href="xenium-1.html#load-giottopoints-transcripts"><i class="fa fa-check"></i><b>10.4.1</b> Load giottoPoints transcripts</a></li>
<li class="chapter" data-level="10.4.2" data-path="xenium-1.html"><a href="xenium-1.html#optional-loading-pre-aggregated-data"><i class="fa fa-check"></i><b>10.4.2</b> (optional) Loading pre-aggregated data</a></li>
</ul></li>
<li class="chapter" data-level="10.5" data-path="xenium-1.html"><a href="xenium-1.html#xenium-images"><i class="fa fa-check"></i><b>10.5</b> Xenium Images</a>
<ul>
<li class="chapter" data-level="10.5.1" data-path="xenium-1.html"><a href="xenium-1.html#image-metadata"><i class="fa fa-check"></i><b>10.5.1</b> Image metadata</a></li>
<li class="chapter" data-level="10.5.2" data-path="xenium-1.html"><a href="xenium-1.html#image-loading"><i class="fa fa-check"></i><b>10.5.2</b> Image loading</a></li>
</ul></li>
<li class="chapter" data-level="10.6" data-path="xenium-1.html"><a href="xenium-1.html#spatial-aggregation"><i class="fa fa-check"></i><b>10.6</b> Spatial aggregation</a></li>
<li class="chapter" data-level="10.7" data-path="xenium-1.html"><a href="xenium-1.html#aggregate-analyses-workflow"><i class="fa fa-check"></i><b>10.7</b> Aggregate analyses workflow</a>
<ul>
<li class="chapter" data-level="10.7.1" data-path="xenium-1.html"><a href="xenium-1.html#transcripts-per-cell"><i class="fa fa-check"></i><b>10.7.1</b> Transcripts per cell</a></li>
<li class="chapter" data-level="10.7.2" data-path="xenium-1.html"><a href="xenium-1.html#filtering-1"><i class="fa fa-check"></i><b>10.7.2</b> Filtering</a></li>
<li class="chapter" data-level="10.7.3" data-path="xenium-1.html"><a href="xenium-1.html#normalization-1"><i class="fa fa-check"></i><b>10.7.3</b> Normalization</a></li>
<li class="chapter" data-level="10.7.4" data-path="xenium-1.html"><a href="xenium-1.html#dimension-reduction-1"><i class="fa fa-check"></i><b>10.7.4</b> Dimension Reduction</a></li>
<li class="chapter" data-level="10.7.5" data-path="xenium-1.html"><a href="xenium-1.html#clustering-1"><i class="fa fa-check"></i><b>10.7.5</b> Clustering</a></li>
</ul></li>
<li class="chapter" data-level="10.8" data-path="xenium-1.html"><a href="xenium-1.html#niche-clustering-1"><i class="fa fa-check"></i><b>10.8</b> Niche clustering</a>
<ul>
<li class="chapter" data-level="10.8.1" data-path="xenium-1.html"><a href="xenium-1.html#spatial-network"><i class="fa fa-check"></i><b>10.8.1</b> Spatial network</a></li>
<li class="chapter" data-level="10.8.2" data-path="xenium-1.html"><a href="xenium-1.html#niche-calculation"><i class="fa fa-check"></i><b>10.8.2</b> Niche calculation</a></li>
<li class="chapter" data-level="10.8.3" data-path="xenium-1.html"><a href="xenium-1.html#k-means-clustering-based-on-niche-signature"><i class="fa fa-check"></i><b>10.8.3</b> k-means clustering based on niche signature</a></li>
</ul></li>
<li class="chapter" data-level="10.9" data-path="xenium-1.html"><a href="xenium-1.html#cell-proximity-enrichment"><i class="fa fa-check"></i><b>10.9</b> Cell proximity enrichment</a></li>
<li class="chapter" data-level="10.10" data-path="xenium-1.html"><a href="xenium-1.html#pseudovisium"><i class="fa fa-check"></i><b>10.10</b> Pseudovisium</a>
<ul>
<li class="chapter" data-level="10.10.1" data-path="xenium-1.html"><a href="xenium-1.html#pseudovisium-aggregation-and-workflow"><i class="fa fa-check"></i><b>10.10.1</b> Pseudovisium aggregation and workflow</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html"><i class="fa fa-check"></i><b>11</b> Spatial proteomics: Multiplexed Immunofluorescence</a>
<ul>
<li class="chapter" data-level="11.1" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#spatial-proteomics-technologies"><i class="fa fa-check"></i><b>11.1</b> Spatial Proteomics Technologies</a></li>
<li class="chapter" data-level="11.2" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#raw-data-type-coming-out-of-different-technologies"><i class="fa fa-check"></i><b>11.2</b> Raw data type coming out of different technologies</a>
<ul>
<li class="chapter" data-level="11.2.1" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#use-ome.tiff-as-an-example-output-data-to-begin-with"><i class="fa fa-check"></i><b>11.2.1</b> Use ome.tiff as an example output data to begin with</a></li>
<li class="chapter" data-level="11.2.2" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#use-single-channel-images-as-an-example-output-data-to-begin-with"><i class="fa fa-check"></i><b>11.2.2</b> Use single channel images as an example output data to begin with</a></li>
</ul></li>
<li class="chapter" data-level="11.3" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#cell-segmentation-file-to-get-single-cell-level-protein-expression"><i class="fa fa-check"></i><b>11.3</b> Cell Segmentation file to get single cell level protein expression</a>
<ul>
<li class="chapter" data-level="11.3.1" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#using-segmentation-output-file-from-deepcellmesmer-as-an-example."><i class="fa fa-check"></i><b>11.3.1</b> Using segmentation output file from DeepCell(mesmer) as an example.</a></li>
<li class="chapter" data-level="11.3.2" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#using-giotto-wrapper-of-cellpose-to-perform-segmentation"><i class="fa fa-check"></i><b>11.3.2</b> Using Giotto wrapper of Cellpose to perform segmentation</a></li>
</ul></li>
<li class="chapter" data-level="11.4" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#create-a-giotto-object-using-list-of-gitto-large-images-and-polygons"><i class="fa fa-check"></i><b>11.4</b> Create a Giotto Object using list of gitto large images and polygons</a>
<ul>
<li class="chapter" data-level="11.4.1" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#overlap-to-matrix"><i class="fa fa-check"></i><b>11.4.1</b> Overlap to matrix</a></li>
<li class="chapter" data-level="11.4.2" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#manipulate-expression-information"><i class="fa fa-check"></i><b>11.4.2</b> Manipulate Expression information</a></li>
<li class="chapter" data-level="11.4.3" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#rescale-polygons"><i class="fa fa-check"></i><b>11.4.3</b> Rescale polygons</a></li>
<li class="chapter" data-level="11.4.4" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#perform-clustering-and-differential-expression"><i class="fa fa-check"></i><b>11.4.4</b> Perform clustering and differential expression</a></li>
<li class="chapter" data-level="11.4.5" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#give-the-cluster-an-annotation-based-on-expression-values"><i class="fa fa-check"></i><b>11.4.5</b> Give the cluster an annotation based on expression values</a></li>
<li class="chapter" data-level="11.4.6" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#spatial-network-1"><i class="fa fa-check"></i><b>11.4.6</b> Spatial network</a></li>
<li class="chapter" data-level="11.4.7" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#cell-neighborhood-cell-typecell-type-interactions"><i class="fa fa-check"></i><b>11.4.7</b> Cell Neighborhood: Cell-Type/Cell-Type Interactions</a></li>
</ul></li>
<li class="chapter" data-level="11.5" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#session-info-2"><i class="fa fa-check"></i><b>11.5</b> Session info</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html"><i class="fa fa-check"></i><b>12</b> Working with multiple samples</a>
<ul>
<li class="chapter" data-level="12.1" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#objective-1"><i class="fa fa-check"></i><b>12.1</b> Objective</a></li>
<li class="chapter" data-level="12.2" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#background-1"><i class="fa fa-check"></i><b>12.2</b> Background</a>
<ul>
<li class="chapter" data-level="12.2.1" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#dataset"><i class="fa fa-check"></i><b>12.2.1</b> Dataset</a></li>
<li class="chapter" data-level="12.2.2" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#visium-technology"><i class="fa fa-check"></i><b>12.2.2</b> Visium technology</a></li>
</ul></li>
<li class="chapter" data-level="12.3" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#create-individual-giotto-objects"><i class="fa fa-check"></i><b>12.3</b> Create individual giotto objects</a>
<ul>
<li class="chapter" data-level="12.3.1" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#download-the-data"><i class="fa fa-check"></i><b>12.3.1</b> Download the data</a></li>
</ul></li>
<li class="chapter" data-level="12.4" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#extracting-the-downloaded-files"><i class="fa fa-check"></i><b>12.4</b> Extracting the downloaded files</a>
<ul>
<li class="chapter" data-level="12.4.1" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#create-giotto-instructions"><i class="fa fa-check"></i><b>12.4.1</b> Create giotto instructions</a></li>
<li class="chapter" data-level="12.4.2" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#load-visium-data-into-giotto"><i class="fa fa-check"></i><b>12.4.2</b> Load visium data into Giotto</a></li>
<li class="chapter" data-level="12.4.3" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#healthy-prostate-tissue-coverage"><i class="fa fa-check"></i><b>12.4.3</b> Healthy prostate tissue coverage</a></li>
<li class="chapter" data-level="12.4.4" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#adenocarcinoma-prostate-tissue-coverage"><i class="fa fa-check"></i><b>12.4.4</b> Adenocarcinoma prostate tissue coverage</a></li>
<li class="chapter" data-level="12.4.5" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#showing-the-data-strucutre-for-the-inidividual-objects"><i class="fa fa-check"></i><b>12.4.5</b> Showing the data strucutre for the inidividual objects</a></li>
</ul></li>
<li class="chapter" data-level="12.5" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#join-giotto-objects"><i class="fa fa-check"></i><b>12.5</b> Join Giotto Objects</a></li>
<li class="chapter" data-level="12.6" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#visualizing-combined-datasets"><i class="fa fa-check"></i><b>12.6</b> Visualizing combined datasets</a>
<ul>
<li class="chapter" data-level="12.6.1" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#vizualizing-in-the-same-plot"><i class="fa fa-check"></i><b>12.6.1</b> Vizualizing in the same plot</a></li>
<li class="chapter" data-level="12.6.2" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#visualizing-on-separate-plots"><i class="fa fa-check"></i><b>12.6.2</b> Visualizing on separate plots</a></li>
</ul></li>
<li class="chapter" data-level="12.7" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#splitting-combined-dataset"><i class="fa fa-check"></i><b>12.7</b> Splitting combined dataset</a></li>
<li class="chapter" data-level="12.8" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#analyzing-joined-objects"><i class="fa fa-check"></i><b>12.8</b> Analyzing joined objects</a>
<ul>
<li class="chapter" data-level="12.8.1" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#normalization-and-adding-statistics"><i class="fa fa-check"></i><b>12.8.1</b> Normalization and adding statistics</a></li>
<li class="chapter" data-level="12.8.2" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#clustering-the-datasets"><i class="fa fa-check"></i><b>12.8.2</b> Clustering the datasets</a></li>
<li class="chapter" data-level="12.8.3" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#vizualizing-spatial-location-of-clusters"><i class="fa fa-check"></i><b>12.8.3</b> Vizualizing spatial location of clusters</a></li>
<li class="chapter" data-level="12.8.4" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#vizualizing-tissue-contribution-to-clusters"><i class="fa fa-check"></i><b>12.8.4</b> Vizualizing tissue contribution to clusters</a></li>
</ul></li>
<li class="chapter" data-level="12.9" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#perform-harmony-and-default-workflows"><i class="fa fa-check"></i><b>12.9</b> Perform Harmony and default workflows</a>
<ul>
<li class="chapter" data-level="12.9.1" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#clustering-harmonized-object"><i class="fa fa-check"></i><b>12.9.1</b> Clustering harmonized object</a></li>
<li class="chapter" data-level="12.9.2" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#vizualizing-the-tissue-contribution"><i class="fa fa-check"></i><b>12.9.2</b> Vizualizing the tissue contribution</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="13" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html"><i class="fa fa-check"></i><b>13</b> Spatial multi-modal analysis</a>
<ul>
<li class="chapter" data-level="13.1" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#overview-1"><i class="fa fa-check"></i><b>13.1</b> Overview</a></li>
<li class="chapter" data-level="13.2" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#spatial-manipulation"><i class="fa fa-check"></i><b>13.2</b> Spatial manipulation</a>
<ul>
<li class="chapter" data-level="13.2.1" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#spatial-transforms"><i class="fa fa-check"></i><b>13.2.1</b> Spatial transforms:</a></li>
<li class="chapter" data-level="13.2.2" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#spatial-utilities"><i class="fa fa-check"></i><b>13.2.2</b> Spatial utilities:</a></li>
<li class="chapter" data-level="13.2.3" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#spatial-classes"><i class="fa fa-check"></i><b>13.2.3</b> Spatial classes:</a></li>
</ul></li>
<li class="chapter" data-level="13.3" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#examples-of-the-simple-transforms-with-a-giottopolygon"><i class="fa fa-check"></i><b>13.3</b> Examples of the simple transforms with a giottoPolygon</a></li>
<li class="chapter" data-level="13.4" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#affine-transforms"><i class="fa fa-check"></i><b>13.4</b> Affine transforms</a></li>
<li class="chapter" data-level="13.5" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#image-transforms"><i class="fa fa-check"></i><b>13.5</b> Image transforms</a></li>
<li class="chapter" data-level="13.6" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#the-practical-usage-of-multi-modality-co-registration"><i class="fa fa-check"></i><b>13.6</b> The practical usage of multi-modality co-registration</a>
<ul>
<li class="chapter" data-level="13.6.1" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#example-dataset-xenium-breast-cancer-pre-release-pack"><i class="fa fa-check"></i><b>13.6.1</b> Example dataset: Xenium Breast Cancer pre-release pack</a></li>
<li class="chapter" data-level="13.6.2" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#target-coordinate-system"><i class="fa fa-check"></i><b>13.6.2</b> Target Coordinate system</a></li>
<li class="chapter" data-level="13.6.3" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#visium-to-register"><i class="fa fa-check"></i><b>13.6.3</b> Visium to register</a></li>
<li class="chapter" data-level="13.6.4" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#register-post-xenium-he-and-if-image"><i class="fa fa-check"></i><b>13.6.4</b> Register post-Xenium H&amp;E and IF image</a></li>
<li class="chapter" data-level="13.6.5" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#automatic-alignment-via-sift-feature-descriptor-matching-and-affine-transformation"><i class="fa fa-check"></i><b>13.6.5</b> Automatic alignment via SIFT feature descriptor matching and affine transformation</a></li>
<li class="chapter" data-level="13.6.6" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#final-notes"><i class="fa fa-check"></i><b>13.6.6</b> Final Notes</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="14" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html"><i class="fa fa-check"></i><b>14</b> Multi-omics integration</a>
<ul>
<li class="chapter" data-level="14.1" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#the-cytassist-technology"><i class="fa fa-check"></i><b>14.1</b> The CytAssist technology</a></li>
<li class="chapter" data-level="14.2" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#introduction-to-the-spatial-dataset-1"><i class="fa fa-check"></i><b>14.2</b> Introduction to the spatial dataset</a></li>
<li class="chapter" data-level="14.3" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#download-dataset-1"><i class="fa fa-check"></i><b>14.3</b> Download dataset</a></li>
<li class="chapter" data-level="14.4" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#create-the-giotto-object-1"><i class="fa fa-check"></i><b>14.4</b> Create the Giotto object</a></li>
<li class="chapter" data-level="14.5" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#subset-on-spots-that-were-covered-by-tissue-1"><i class="fa fa-check"></i><b>14.5</b> Subset on spots that were covered by tissue</a></li>
<li class="chapter" data-level="14.6" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#rna-processing"><i class="fa fa-check"></i><b>14.6</b> RNA processing</a></li>
<li class="chapter" data-level="14.7" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#protein-processing"><i class="fa fa-check"></i><b>14.7</b> Protein processing</a></li>
<li class="chapter" data-level="14.8" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#multi-omics-integration-1"><i class="fa fa-check"></i><b>14.8</b> Multi-omics integration</a></li>
<li class="chapter" data-level="14.9" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#session-info-3"><i class="fa fa-check"></i><b>14.9</b> Session info</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html"><i class="fa fa-check"></i><b>15</b> Interoperability with other frameworks</a>
<ul>
<li class="chapter" data-level="15.1" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#load-giotto-object"><i class="fa fa-check"></i><b>15.1</b> Load Giotto object</a></li>
<li class="chapter" data-level="15.2" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#seurat"><i class="fa fa-check"></i><b>15.2</b> Seurat</a>
<ul>
<li class="chapter" data-level="15.2.1" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#conversion-of-giotto-object-to-seurat-object"><i class="fa fa-check"></i><b>15.2.1</b> Conversion of Giotto Object to Seurat Object</a></li>
<li class="chapter" data-level="15.2.2" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#conversion-of-seurat-object-back-to-giotto-object"><i class="fa fa-check"></i><b>15.2.2</b> Conversion of Seurat object Back to Giotto Object</a></li>
</ul></li>
<li class="chapter" data-level="15.3" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#spatialexperiment"><i class="fa fa-check"></i><b>15.3</b> SpatialExperiment</a>
<ul>
<li class="chapter" data-level="15.3.1" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#convert-giotto-object-to-spatialexperiment-object"><i class="fa fa-check"></i><b>15.3.1</b> Convert Giotto Object to SpatialExperiment Object</a></li>
<li class="chapter" data-level="15.3.2" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#conversion-of-spatialexperiment-object-back-to-giotto"><i class="fa fa-check"></i><b>15.3.2</b> Conversion of SpatialExperiment object back to Giotto</a></li>
</ul></li>
<li class="chapter" data-level="15.4" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#anndata"><i class="fa fa-check"></i><b>15.4</b> AnnData</a>
<ul>
<li class="chapter" data-level="15.4.1" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#load-required-libraries"><i class="fa fa-check"></i><b>15.4.1</b> Load Required Libraries</a></li>
<li class="chapter" data-level="15.4.2" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#specify-path-for-results"><i class="fa fa-check"></i><b>15.4.2</b> Specify Path for Results</a></li>
<li class="chapter" data-level="15.4.3" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#giotto-to-anndata"><i class="fa fa-check"></i><b>15.4.3</b> Giotto To AnnData</a></li>
<li class="chapter" data-level="15.4.4" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#convert-anndata-to-giotto"><i class="fa fa-check"></i><b>15.4.4</b> Convert AnnData to Giotto</a></li>
</ul></li>
<li class="chapter" data-level="15.5" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#create-mini-vizgen-object"><i class="fa fa-check"></i><b>15.5</b> Create mini Vizgen object</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html"><i class="fa fa-check"></i><b>16</b> Interoperability with isolated tools</a>
<ul>
<li class="chapter" data-level="16.1" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#spatial-niche-trajectory-analysis-ontrac"><i class="fa fa-check"></i><b>16.1</b> Spatial niche trajectory analysis (ONTraC)</a>
<ul>
<li class="chapter" data-level="16.1.1" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#introduction-to-ontrac"><i class="fa fa-check"></i><b>16.1.1</b> Introduction to ONTraC</a></li>
<li class="chapter" data-level="16.1.2" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#introduction-to-merfish"><i class="fa fa-check"></i><b>16.1.2</b> Introduction to MERFISH</a></li>
<li class="chapter" data-level="16.1.3" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#settings"><i class="fa fa-check"></i><b>16.1.3</b> Settings</a></li>
<li class="chapter" data-level="16.1.4" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#dataset-1"><i class="fa fa-check"></i><b>16.1.4</b> Dataset</a></li>
<li class="chapter" data-level="16.1.5" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#create-the-giotto-object-2"><i class="fa fa-check"></i><b>16.1.5</b> Create the Giotto object</a></li>
<li class="chapter" data-level="16.1.6" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#perform-spatial-niche-trajectory-analysis-using-ontrac"><i class="fa fa-check"></i><b>16.1.6</b> Perform spatial niche trajectory analysis using ONTraC</a></li>
<li class="chapter" data-level="16.1.7" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#visualization"><i class="fa fa-check"></i><b>16.1.7</b> Visualization</a></li>
</ul></li>
<li class="chapter" data-level="16.2" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#session-info-4"><i class="fa fa-check"></i><b>16.2</b> Session info</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html"><i class="fa fa-check"></i><b>17</b> Interactivity with the R/Spatial ecosystem</a>
<ul>
<li class="chapter" data-level="17.1" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#visium-technology-1"><i class="fa fa-check"></i><b>17.1</b> Visium technology</a></li>
<li class="chapter" data-level="17.2" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#gene-expression-interpolation-through-kriging"><i class="fa fa-check"></i><b>17.2</b> Gene expression interpolation through kriging</a>
<ul>
<li class="chapter" data-level="17.2.1" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#dataset-2"><i class="fa fa-check"></i><b>17.2.1</b> Dataset</a></li>
<li class="chapter" data-level="17.2.2" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#generating-a-geojson-file-of-nuclei-location"><i class="fa fa-check"></i><b>17.2.2</b> Generating a geojson file of nuclei location</a></li>
</ul></li>
<li class="chapter" data-level="17.3" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#downloading-the-dataset"><i class="fa fa-check"></i><b>17.3</b> Downloading the dataset</a></li>
<li class="chapter" data-level="17.4" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#extracting-the-files"><i class="fa fa-check"></i><b>17.4</b> Extracting the files</a></li>
<li class="chapter" data-level="17.5" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#downloading-giotto-object-and-nuclei-segmentation"><i class="fa fa-check"></i><b>17.5</b> Downloading giotto object and nuclei segmentation</a></li>
<li class="chapter" data-level="17.6" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#importing-visium-data"><i class="fa fa-check"></i><b>17.6</b> Importing visium data</a>
<ul>
<li class="chapter" data-level="17.6.1" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#identifying-spatially-organized-features"><i class="fa fa-check"></i><b>17.6.1</b> Identifying spatially organized features</a></li>
</ul></li>
<li class="chapter" data-level="17.7" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#performing-kriging"><i class="fa fa-check"></i><b>17.7</b> Performing kriging</a>
<ul>
<li class="chapter" data-level="17.7.1" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#interpolating-features"><i class="fa fa-check"></i><b>17.7.1</b> Interpolating features</a></li>
</ul></li>
<li class="chapter" data-level="17.8" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#adding-cell-polygons-to-giotto-object"><i class="fa fa-check"></i><b>17.8</b> Adding cell polygons to Giotto object</a>
<ul>
<li class="chapter" data-level="17.8.1" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#read-in-the-poly-information"><i class="fa fa-check"></i><b>17.8.1</b> Read in the poly information</a></li>
<li class="chapter" data-level="17.8.2" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#vizualizing-polygons"><i class="fa fa-check"></i><b>17.8.2</b> Vizualizing polygons</a></li>
<li class="chapter" data-level="17.8.3" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#showing-giotto-object-prior-to-polygon-addition"><i class="fa fa-check"></i><b>17.8.3</b> Showing Giotto object prior to polygon addition</a></li>
<li class="chapter" data-level="17.8.4" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#adding-polygons-to-giotto-object"><i class="fa fa-check"></i><b>17.8.4</b> Adding polygons to giotto object</a></li>
<li class="chapter" data-level="17.8.5" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#check-polygon-information"><i class="fa fa-check"></i><b>17.8.5</b> Check polygon information</a></li>
<li class="chapter" data-level="17.8.6" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#expression-overlap"><i class="fa fa-check"></i><b>17.8.6</b> Expression overlap</a></li>
</ul></li>
<li class="chapter" data-level="17.9" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#reading-in-larger-dataset"><i class="fa fa-check"></i><b>17.9</b> Reading in larger dataset</a></li>
<li class="chapter" data-level="17.10" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#analyzing-interpolated-features"><i class="fa fa-check"></i><b>17.10</b> Analyzing interpolated features</a>
<ul>
<li class="chapter" data-level="17.10.1" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#filter-and-normalization"><i class="fa fa-check"></i><b>17.10.1</b> Filter and normalization</a></li>
<li class="chapter" data-level="17.10.2" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#visualizing-gene-expression-from-interpolated-expression"><i class="fa fa-check"></i><b>17.10.2</b> Visualizing gene expression from interpolated expression</a></li>
<li class="chapter" data-level="17.10.3" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#run-pca"><i class="fa fa-check"></i><b>17.10.3</b> Run PCA</a></li>
<li class="chapter" data-level="17.10.4" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#clustering-2"><i class="fa fa-check"></i><b>17.10.4</b> Clustering</a></li>
<li class="chapter" data-level="17.10.5" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#visualizing-clustering"><i class="fa fa-check"></i><b>17.10.5</b> Visualizing clustering</a></li>
<li class="chapter" data-level="17.10.6" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#cropping-objects"><i class="fa fa-check"></i><b>17.10.6</b> Cropping objects</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="18" data-path="contributing-to-giotto.html"><a href="contributing-to-giotto.html"><i class="fa fa-check"></i><b>18</b> Contributing to Giotto</a>
<ul>
<li class="chapter" data-level="18.1" data-path="contributing-to-giotto.html"><a href="contributing-to-giotto.html#contribution-guideline"><i class="fa fa-check"></i><b>18.1</b> Contribution guideline</a></li>
<li class="chapter" data-level="18.2" data-path="contributing-to-giotto.html"><a href="contributing-to-giotto.html#coding-style"><i class="fa fa-check"></i><b>18.2</b> Coding Style</a></li>
<li class="chapter" data-level="18.3" data-path="contributing-to-giotto.html"><a href="contributing-to-giotto.html#stat-functions"><i class="fa fa-check"></i><b>18.3</b> Stat functions</a></li>
<li class="chapter" data-level="18.4" data-path="contributing-to-giotto.html"><a href="contributing-to-giotto.html#auxiliary-functions"><i class="fa fa-check"></i><b>18.4</b> Auxiliary functions</a></li>
<li class="chapter" data-level="18.5" data-path="contributing-to-giotto.html"><a href="contributing-to-giotto.html#package-imports"><i class="fa fa-check"></i><b>18.5</b> Package Imports</a></li>
<li class="chapter" data-level="18.6" data-path="contributing-to-giotto.html"><a href="contributing-to-giotto.html#python-code"><i class="fa fa-check"></i><b>18.6</b> Python code</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Workshop: Spatial multi-omics data analysis with Giotto Suite</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="visium-hd" class="section level1 hasAnchor" number="9">
<h1><span class="header-section-number">9</span> Visium HD<a href="visium-hd.html#visium-hd" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Ruben Dries &amp; Edward C. Ruiz</p>
<p>August 6th 2024</p>
<div id="objective" class="section level2 hasAnchor" number="9.1">
<h2><span class="header-section-number">9.1</span> Objective<a href="visium-hd.html#objective" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>This tutorial demonstrates how to process Visium HD data at the highest 2 micron bin resolution by using flexible tiling and aggregation steps that are available in <a href="https://drieslab.github.io/Giotto_website/">Giotto Suite</a>. Notably, a similar strategy can be used for other spatial sequencing methods that operate at the subcellular level, including:<br />
- Stereo-seq<br />
- <a href="https://drieslab.github.io/Giotto_website/articles/seqscope_mouse_liver.html">Seq-Scope</a><br />
- Open-ST</p>
<p>The resulting datasets from all these technologies can be very large since they provide both a high spatial resolution and genome-wide capture of <em>all</em> transcripts. We will also discuss how data projection strategies can be used to alleviate heavy computational tasks such as PCA, UMAP, or clustering.</p>
<p>This tutorial expects a general knowledge of common spatial analysis technologies that are available in Giotto Suite, such as those that have been discussed in the standard Visium tutorials (<a href="#Visium%20Part%20I">part I</a> and <a href="Visium%20Part%20II">part II</a>).</p>
</div>
<div id="background" class="section level2 hasAnchor" number="9.2">
<h2><span class="header-section-number">9.2</span> Background<a href="visium-hd.html#background" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="visium-hd-technology" class="section level3 hasAnchor" number="9.2.1">
<h3><span class="header-section-number">9.2.1</span> Visium HD Technology<a href="visium-hd.html#visium-hd-technology" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-258"></span>
<img src="img/02_session2/1.png" alt="Overview of Visium HD. Source: *10X Genomics*" width="80%" />
<p class="caption">
Figure 9.1: Overview of Visium HD. Source: <em>10X Genomics</em>
</p>
</div>
<p>Visium HD is a spatial transcriptomics technology recently developed by 10X Genomics. Details about this platform are discussed on the official <a href="https://www.10xgenomics.com/products/visium-hd-spatial-gene-expression">10X Genomics Visium HD website</a> and the preprint by <a href="https://doi.org/10.1101/2024.06.04.597233">Oliveira <em>et al.</em> 2024</a> on <em>bioRxiv</em>.</p>
<p>Visium HD has a 2 micron bin size resolution. The default SpaceRanger pipeline from 10X Genomics also returns aggregated data at the 8 and 16 micron bin size.</p>
</div>
<div id="colorectal-cancer-sample" class="section level3 hasAnchor" number="9.2.2">
<h3><span class="header-section-number">9.2.2</span> Colorectal Cancer Sample<a href="visium-hd.html#colorectal-cancer-sample" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-259"></span>
<img src="img/02_session2/2.png" alt="Colorectal Cancer Overview. Source: *10X Genomics*" width="80%" />
<p class="caption">
Figure 9.2: Colorectal Cancer Overview. Source: <em>10X Genomics</em>
</p>
</div>
<p>For this tutorial we will be using the publicly available Colorectal Cancer Visium HD dataset.</p>
<p>Details about this dataset and a link to download the raw data can be found at the <a href="https://www.10xgenomics.com/datasets/visium-hd-cytassist-gene-expression-libraries-of-human-crc">10X Genomics website</a>.</p>
</div>
</div>
<div id="data-ingestion" class="section level2 hasAnchor" number="9.3">
<h2><span class="header-section-number">9.3</span> Data Ingestion<a href="visium-hd.html#data-ingestion" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="visium-hd-output-data-format" class="section level3 hasAnchor" number="9.3.1">
<h3><span class="header-section-number">9.3.1</span> Visium HD output data format<a href="visium-hd.html#visium-hd-output-data-format" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-260"></span>
<img src="img/02_session2/3.png" alt="File structure of Visium HD data processed with spaceranger pipeline." width="80%" />
<p class="caption">
Figure 9.3: File structure of Visium HD data processed with spaceranger pipeline.
</p>
</div>
<p>Visium HD data processed with the spaceranger pipeline is organized in this format containing various files associated with the sample. The files highlighted in yellow are what we will be using to read in these datasets.</p>
<p><strong>Warning</strong>: the VisiumHD folder structure has very recently been updated and might be slightly different.</p>
</div>
<div id="mini-visium-hd-dataset" class="section level3 hasAnchor" number="9.3.2">
<h3><span class="header-section-number">9.3.2</span> Mini Visium HD dataset<a href="visium-hd.html#mini-visium-hd-dataset" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>For this workshop we will use a spatial subset and downsampled version of the original datasets. A VisiumHD folder similar to the original can be downloaded using the Zenodo link. Using this dataset will ensure that we will not run into major memory issues.</p>
<div class="sourceCode" id="cb201"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb201-1"><a href="visium-hd.html#cb201-1" tabindex="-1"></a><span class="fu">library</span>(Giotto)</span>
<span id="cb201-2"><a href="visium-hd.html#cb201-2" tabindex="-1"></a></span>
<span id="cb201-3"><a href="visium-hd.html#cb201-3" tabindex="-1"></a><span class="co"># set up paths</span></span>
<span id="cb201-4"><a href="visium-hd.html#cb201-4" tabindex="-1"></a>data_path <span class="ot">&lt;-</span> <span class="st">&quot;data/02_session2/&quot;</span></span>
<span id="cb201-5"><a href="visium-hd.html#cb201-5" tabindex="-1"></a>save_dir <span class="ot">&lt;-</span> <span class="st">&quot;results/02_session2/&quot;</span></span>
<span id="cb201-6"><a href="visium-hd.html#cb201-6" tabindex="-1"></a><span class="fu">dir.create</span>(save_dir, <span class="at">recursive =</span> <span class="cn">TRUE</span>)</span>
<span id="cb201-7"><a href="visium-hd.html#cb201-7" tabindex="-1"></a></span>
<span id="cb201-8"><a href="visium-hd.html#cb201-8" tabindex="-1"></a><span class="co"># download the mini dataset and untar</span></span>
<span id="cb201-9"><a href="visium-hd.html#cb201-9" tabindex="-1"></a><span class="fu">options</span>(<span class="st">&quot;timeout&quot;</span> <span class="ot">=</span> <span class="cn">Inf</span>)</span>
<span id="cb201-10"><a href="visium-hd.html#cb201-10" tabindex="-1"></a><span class="fu">download.file</span>(</span>
<span id="cb201-11"><a href="visium-hd.html#cb201-11" tabindex="-1"></a>  <span class="at">url =</span> <span class="st">&quot;https://zenodo.org/records/13226158/files/workshop_VisiumHD.zip?download=1&quot;</span>,</span>
<span id="cb201-12"><a href="visium-hd.html#cb201-12" tabindex="-1"></a>  <span class="at">destfile =</span> <span class="fu">file.path</span>(save_dir, <span class="st">&quot;workshop_visiumHD.zip&quot;</span>)</span>
<span id="cb201-13"><a href="visium-hd.html#cb201-13" tabindex="-1"></a>)</span>
<span id="cb201-14"><a href="visium-hd.html#cb201-14" tabindex="-1"></a><span class="fu">untar</span>(<span class="at">tarfile =</span> <span class="fu">file.path</span>(save_dir, <span class="st">&quot;workshop_visiumHD.zip&quot;</span>), </span>
<span id="cb201-15"><a href="visium-hd.html#cb201-15" tabindex="-1"></a>      <span class="at">exdir =</span> data_path)</span></code></pre></div>
</div>
<div id="giotto-visium-hd-convenience-function" class="section level3 hasAnchor" number="9.3.3">
<h3><span class="header-section-number">9.3.3</span> Giotto Visium HD convenience function<a href="visium-hd.html#giotto-visium-hd-convenience-function" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The easiest way to read in Visium HD data in Giotto is through our convenience function. This function will automatically read in the data at your desired resolution, align the images, and finally create a Giotto Object.</p>
<div class="sourceCode" id="cb202"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb202-1"><a href="visium-hd.html#cb202-1" tabindex="-1"></a><span class="co"># importVisiumHD()</span></span></code></pre></div>
</div>
<div id="read-in-data-manually" class="section level3 hasAnchor" number="9.3.4">
<h3><span class="header-section-number">9.3.4</span> Read in data manually<a href="visium-hd.html#read-in-data-manually" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>However, for this tutorial we will illustrate how to create your own Giotto object in a step-by-step manner, which can also be applied to other similar technologies as discussed in the <a href="#Objective">Objective</a> section.</p>
<div id="raw-expression-data" class="section level4 hasAnchor" number="9.3.4.1">
<h4><span class="header-section-number">9.3.4.1</span> Raw expression data<a href="visium-hd.html#raw-expression-data" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb203"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb203-1"><a href="visium-hd.html#cb203-1" tabindex="-1"></a>expression_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_path, <span class="st">&#39;/Human_Colorectal_Cancer_workshop/square_002um/raw_feature_bc_matrix&#39;</span>)</span>
<span id="cb203-2"><a href="visium-hd.html#cb203-2" tabindex="-1"></a>expr_results <span class="ot">&lt;-</span> <span class="fu">get10Xmatrix</span>(<span class="at">path_to_data =</span> expression_path, </span>
<span id="cb203-3"><a href="visium-hd.html#cb203-3" tabindex="-1"></a>                             <span class="at">gene_column_index =</span> <span class="dv">1</span>)</span></code></pre></div>
</div>
<div id="tissue-positions-data" class="section level4 hasAnchor" number="9.3.4.2">
<h4><span class="header-section-number">9.3.4.2</span> Tissue positions data<a href="visium-hd.html#tissue-positions-data" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb204"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb204-1"><a href="visium-hd.html#cb204-1" tabindex="-1"></a>tissue_positions_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_path, <span class="st">&#39;/Human_Colorectal_Cancer_workshop/square_002um/spatial/tissue_positions.parquet&#39;</span>)</span>
<span id="cb204-2"><a href="visium-hd.html#cb204-2" tabindex="-1"></a>tissue_positions <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">as.data.table</span>(arrow<span class="sc">::</span><span class="fu">read_parquet</span>(tissue_positions_path))</span></code></pre></div>
</div>
<div id="merge-expression-and-2-micron-position-data" class="section level4 hasAnchor" number="9.3.4.3">
<h4><span class="header-section-number">9.3.4.3</span> Merge expression and 2 micron position data<a href="visium-hd.html#merge-expression-and-2-micron-position-data" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb205"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb205-1"><a href="visium-hd.html#cb205-1" tabindex="-1"></a><span class="co"># convert expression matrix to minimal data.frame or data.table object</span></span>
<span id="cb205-2"><a href="visium-hd.html#cb205-2" tabindex="-1"></a>matrix_tile_dt <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">as.data.table</span>(Matrix<span class="sc">::</span><span class="fu">summary</span>(expr_results))</span>
<span id="cb205-3"><a href="visium-hd.html#cb205-3" tabindex="-1"></a>genes   <span class="ot">&lt;-</span> expr_results<span class="sc">@</span>Dimnames[[<span class="dv">1</span>]]</span>
<span id="cb205-4"><a href="visium-hd.html#cb205-4" tabindex="-1"></a>samples <span class="ot">&lt;-</span> expr_results<span class="sc">@</span>Dimnames[[<span class="dv">2</span>]]</span>
<span id="cb205-5"><a href="visium-hd.html#cb205-5" tabindex="-1"></a>matrix_tile_dt[, gene <span class="sc">:=</span> genes[i]]</span>
<span id="cb205-6"><a href="visium-hd.html#cb205-6" tabindex="-1"></a>matrix_tile_dt[, pixel <span class="sc">:=</span> samples[j]]</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-266"></span>
<img src="img/02_session2/4.png" alt="Genes expressed for each 2 µm pixel in the array dimensions." width="80%" />
<p class="caption">
Figure 9.4: Genes expressed for each 2 µm pixel in the array dimensions.
</p>
</div>
<div class="sourceCode" id="cb206"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb206-1"><a href="visium-hd.html#cb206-1" tabindex="-1"></a><span class="co"># merge data.table matrix and spatial coordinates to create input for Giotto Polygons</span></span>
<span id="cb206-2"><a href="visium-hd.html#cb206-2" tabindex="-1"></a>expr_pos_data <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">merge.data.table</span>(matrix_tile_dt, </span>
<span id="cb206-3"><a href="visium-hd.html#cb206-3" tabindex="-1"></a>                                             tissue_positions, </span>
<span id="cb206-4"><a href="visium-hd.html#cb206-4" tabindex="-1"></a>                                             <span class="at">by.x =</span> <span class="st">&#39;pixel&#39;</span>, </span>
<span id="cb206-5"><a href="visium-hd.html#cb206-5" tabindex="-1"></a>                                             <span class="at">by.y =</span> <span class="st">&#39;barcode&#39;</span>)</span>
<span id="cb206-6"><a href="visium-hd.html#cb206-6" tabindex="-1"></a></span>
<span id="cb206-7"><a href="visium-hd.html#cb206-7" tabindex="-1"></a>expr_pos_data <span class="ot">&lt;-</span> expr_pos_data[,.(pixel, pxl_row_in_fullres, pxl_col_in_fullres, gene, x)]</span>
<span id="cb206-8"><a href="visium-hd.html#cb206-8" tabindex="-1"></a><span class="fu">colnames</span>(expr_pos_data) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&#39;pixel&#39;</span>, <span class="st">&#39;x&#39;</span>, <span class="st">&#39;y&#39;</span>, <span class="st">&#39;gene&#39;</span>, <span class="st">&#39;count&#39;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-268"></span>
<img src="img/02_session2/5.png" alt="Genes expressed with count for each 2 µm pixel in the spatial dimensions." width="80%" />
<p class="caption">
Figure 9.5: Genes expressed with count for each 2 µm pixel in the spatial dimensions.
</p>
</div>
</div>
</div>
</div>
<div id="hexbin-400-giotto-object" class="section level2 hasAnchor" number="9.4">
<h2><span class="header-section-number">9.4</span> Hexbin 400 Giotto object<a href="visium-hd.html#hexbin-400-giotto-object" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="create-giotto-points" class="section level3 hasAnchor" number="9.4.1">
<h3><span class="header-section-number">9.4.1</span> create giotto points<a href="visium-hd.html#create-giotto-points" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The giottoPoints object represents the spatial expression information for each transcript: - gene id - count or UMI<br />
- spatial pixel location (x, y)</p>
<div class="sourceCode" id="cb207"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb207-1"><a href="visium-hd.html#cb207-1" tabindex="-1"></a>giotto_points <span class="ot">=</span> <span class="fu">createGiottoPoints</span>(<span class="at">x =</span> expr_pos_data[,.(x, y, gene, pixel, count)])</span></code></pre></div>
</div>
<div id="create-giotto-polygons" class="section level3 hasAnchor" number="9.4.2">
<h3><span class="header-section-number">9.4.2</span> create giotto polygons<a href="visium-hd.html#create-giotto-polygons" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="tiling-and-aggregation" class="section level4 hasAnchor" number="9.4.2.1">
<h4><span class="header-section-number">9.4.2.1</span> Tiling and aggregation<a href="visium-hd.html#tiling-and-aggregation" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>The Visium HD data is organized in a grid format. We can aggregate the data into larger bins to reduce the resolution of the data. Giotto Suite can work with any type of polygon information and already provides ready-to-use options for binning data with squares, triangles, and hexagons. Here we will use a hexagon tesselation to aggregate the data into arbitrary bins.</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-270"></span>
<img src="img/02_session2/6.png" alt="Hexagon properties" width="80%" />
<p class="caption">
Figure 9.6: Hexagon properties
</p>
</div>
<div class="sourceCode" id="cb208"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb208-1"><a href="visium-hd.html#cb208-1" tabindex="-1"></a><span class="co"># create giotto polygons, here we create hexagons</span></span>
<span id="cb208-2"><a href="visium-hd.html#cb208-2" tabindex="-1"></a>hexbin400 <span class="ot">&lt;-</span> <span class="fu">tessellate</span>(<span class="at">extent =</span> <span class="fu">ext</span>(giotto_points), </span>
<span id="cb208-3"><a href="visium-hd.html#cb208-3" tabindex="-1"></a>                        <span class="at">shape =</span> <span class="st">&#39;hexagon&#39;</span>, </span>
<span id="cb208-4"><a href="visium-hd.html#cb208-4" tabindex="-1"></a>                        <span class="at">shape_size =</span> <span class="dv">400</span>, </span>
<span id="cb208-5"><a href="visium-hd.html#cb208-5" tabindex="-1"></a>                        <span class="at">name =</span> <span class="st">&#39;hex400&#39;</span>) </span>
<span id="cb208-6"><a href="visium-hd.html#cb208-6" tabindex="-1"></a><span class="fu">plot</span>(hexbin400)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-272"></span>
<img src="img/02_session2/7.png" alt="Giotto polygon in a hexagon shape for overlapping visium HD expression data." width="80%" />
<p class="caption">
Figure 9.7: Giotto polygon in a hexagon shape for overlapping visium HD expression data.
</p>
</div>
</div>
</div>
<div id="combine-giotto-points-and-polygons-to-create-giotto-object" class="section level3 hasAnchor" number="9.4.3">
<h3><span class="header-section-number">9.4.3</span> combine Giotto points and polygons to create Giotto object<a href="visium-hd.html#combine-giotto-points-and-polygons-to-create-giotto-object" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb209"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb209-1"><a href="visium-hd.html#cb209-1" tabindex="-1"></a>instrs <span class="ot">=</span> <span class="fu">createGiottoInstructions</span>(</span>
<span id="cb209-2"><a href="visium-hd.html#cb209-2" tabindex="-1"></a>  <span class="at">save_dir =</span> save_dir,</span>
<span id="cb209-3"><a href="visium-hd.html#cb209-3" tabindex="-1"></a>  <span class="at">save_plot =</span> <span class="cn">TRUE</span>,</span>
<span id="cb209-4"><a href="visium-hd.html#cb209-4" tabindex="-1"></a>  <span class="at">show_plot =</span> <span class="cn">FALSE</span>,</span>
<span id="cb209-5"><a href="visium-hd.html#cb209-5" tabindex="-1"></a>  <span class="at">return_plot =</span> <span class="cn">FALSE</span></span>
<span id="cb209-6"><a href="visium-hd.html#cb209-6" tabindex="-1"></a>)</span>
<span id="cb209-7"><a href="visium-hd.html#cb209-7" tabindex="-1"></a></span>
<span id="cb209-8"><a href="visium-hd.html#cb209-8" tabindex="-1"></a><span class="co"># gpoints provides spatial gene expression information</span></span>
<span id="cb209-9"><a href="visium-hd.html#cb209-9" tabindex="-1"></a><span class="co"># gpolygons provides spatial unit information (here = hexagon tiles)</span></span>
<span id="cb209-10"><a href="visium-hd.html#cb209-10" tabindex="-1"></a>visiumHD <span class="ot">=</span> <span class="fu">createGiottoObjectSubcellular</span>(<span class="at">gpoints =</span> <span class="fu">list</span>(<span class="st">&#39;rna&#39;</span> <span class="ot">=</span> giotto_points),</span>
<span id="cb209-11"><a href="visium-hd.html#cb209-11" tabindex="-1"></a>                                         <span class="at">gpolygons =</span> <span class="fu">list</span>(<span class="st">&#39;hex400&#39;</span> <span class="ot">=</span> hexbin400),</span>
<span id="cb209-12"><a href="visium-hd.html#cb209-12" tabindex="-1"></a>                                         <span class="at">instructions =</span> instrs)</span>
<span id="cb209-13"><a href="visium-hd.html#cb209-13" tabindex="-1"></a></span>
<span id="cb209-14"><a href="visium-hd.html#cb209-14" tabindex="-1"></a><span class="co"># create spatial centroids for each spatial unit (hexagon)</span></span>
<span id="cb209-15"><a href="visium-hd.html#cb209-15" tabindex="-1"></a>visiumHD <span class="ot">=</span> <span class="fu">addSpatialCentroidLocations</span>(<span class="at">gobject =</span> visiumHD,</span>
<span id="cb209-16"><a href="visium-hd.html#cb209-16" tabindex="-1"></a>                                       <span class="at">poly_info =</span> <span class="st">&#39;hex400&#39;</span>)</span></code></pre></div>
<p>Visualize the Giotto object. Make sure to set <code>expand_counts = TRUE</code> to expand the counts column. Each spatial bin can have multiple transcripts/UMIs. This is different compared to in situ technologies like seqFISH, MERFISH, Nanostring CosMx or Xenium.</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-274"></span>
<img src="img/02_session2/8.png" alt="Schematic showing effect of expand counts and jitter." width="80%" />
<p class="caption">
Figure 9.8: Schematic showing effect of expand counts and jitter.
</p>
</div>
<p>Show the giotto points (transcripts) and polygons (hexagons) together using <code>spatInSituPlotPoints</code>:</p>
<div class="sourceCode" id="cb210"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb210-1"><a href="visium-hd.html#cb210-1" tabindex="-1"></a>feature_data <span class="ot">=</span> <span class="fu">fDataDT</span>(visiumHD)</span>
<span id="cb210-2"><a href="visium-hd.html#cb210-2" tabindex="-1"></a></span>
<span id="cb210-3"><a href="visium-hd.html#cb210-3" tabindex="-1"></a><span class="fu">spatInSituPlotPoints</span>(visiumHD,</span>
<span id="cb210-4"><a href="visium-hd.html#cb210-4" tabindex="-1"></a>                     <span class="at">show_image =</span> F,</span>
<span id="cb210-5"><a href="visium-hd.html#cb210-5" tabindex="-1"></a>                     <span class="at">feats =</span> <span class="fu">list</span>(<span class="st">&#39;rna&#39;</span> <span class="ot">=</span> feature_data<span class="sc">$</span>feat_ID[<span class="dv">10</span><span class="sc">:</span><span class="dv">20</span>]),</span>
<span id="cb210-6"><a href="visium-hd.html#cb210-6" tabindex="-1"></a>                     <span class="at">show_legend =</span> T,</span>
<span id="cb210-7"><a href="visium-hd.html#cb210-7" tabindex="-1"></a>                     <span class="at">spat_unit =</span> <span class="st">&#39;hex400&#39;</span>,</span>
<span id="cb210-8"><a href="visium-hd.html#cb210-8" tabindex="-1"></a>                     <span class="at">point_size =</span> <span class="fl">0.25</span>,</span>
<span id="cb210-9"><a href="visium-hd.html#cb210-9" tabindex="-1"></a>                     <span class="at">show_polygon =</span> <span class="cn">TRUE</span>,</span>
<span id="cb210-10"><a href="visium-hd.html#cb210-10" tabindex="-1"></a>                     <span class="at">use_overlap =</span> <span class="cn">FALSE</span>,</span>
<span id="cb210-11"><a href="visium-hd.html#cb210-11" tabindex="-1"></a>                     <span class="at">polygon_feat_type =</span> <span class="st">&#39;hex400&#39;</span>,</span>
<span id="cb210-12"><a href="visium-hd.html#cb210-12" tabindex="-1"></a>                     <span class="at">polygon_bg_color =</span> <span class="cn">NA</span>,</span>
<span id="cb210-13"><a href="visium-hd.html#cb210-13" tabindex="-1"></a>                     <span class="at">polygon_color =</span> <span class="st">&#39;white&#39;</span>,</span>
<span id="cb210-14"><a href="visium-hd.html#cb210-14" tabindex="-1"></a>                     <span class="at">polygon_line_size =</span> <span class="fl">0.1</span>,</span>
<span id="cb210-15"><a href="visium-hd.html#cb210-15" tabindex="-1"></a>                     <span class="at">expand_counts =</span> <span class="cn">TRUE</span>, </span>
<span id="cb210-16"><a href="visium-hd.html#cb210-16" tabindex="-1"></a>                     <span class="at">count_info_column =</span> <span class="st">&#39;count&#39;</span>,</span>
<span id="cb210-17"><a href="visium-hd.html#cb210-17" tabindex="-1"></a>                     <span class="at">jitter =</span> <span class="fu">c</span>(<span class="dv">25</span>,<span class="dv">25</span>))</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-276"></span>
<img src="img/02_session2/0-spatInSituPlotPoints.png" alt="Overlap of gene expression with the hex400 polygons. Each dot represents a single gene. Jitter used to better vizualize individual transcripts" width="80%" />
<p class="caption">
Figure 9.9: Overlap of gene expression with the hex400 polygons. Each dot represents a single gene. Jitter used to better vizualize individual transcripts
</p>
</div>
<p>You can set <code>plot_method = scattermore</code> or <code>scattermost</code> to convert high-resolution images to low(er) resolution rasterized images. It’s usually faster and will save on disk space.</p>
<div class="sourceCode" id="cb211"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb211-1"><a href="visium-hd.html#cb211-1" tabindex="-1"></a><span class="fu">spatInSituPlotPoints</span>(visiumHD,</span>
<span id="cb211-2"><a href="visium-hd.html#cb211-2" tabindex="-1"></a>                     <span class="at">show_image =</span> F,</span>
<span id="cb211-3"><a href="visium-hd.html#cb211-3" tabindex="-1"></a>                     <span class="at">feats =</span> <span class="fu">list</span>(<span class="st">&#39;rna&#39;</span> <span class="ot">=</span> feature_data<span class="sc">$</span>feat_ID[<span class="dv">10</span><span class="sc">:</span><span class="dv">20</span>]),</span>
<span id="cb211-4"><a href="visium-hd.html#cb211-4" tabindex="-1"></a>                     <span class="at">show_legend =</span> T,</span>
<span id="cb211-5"><a href="visium-hd.html#cb211-5" tabindex="-1"></a>                     <span class="at">spat_unit =</span> <span class="st">&#39;hex400&#39;</span>,</span>
<span id="cb211-6"><a href="visium-hd.html#cb211-6" tabindex="-1"></a>                     <span class="at">point_size =</span> <span class="fl">0.25</span>,</span>
<span id="cb211-7"><a href="visium-hd.html#cb211-7" tabindex="-1"></a>                     <span class="at">show_polygon =</span> <span class="cn">TRUE</span>,</span>
<span id="cb211-8"><a href="visium-hd.html#cb211-8" tabindex="-1"></a>                     <span class="at">use_overlap =</span> <span class="cn">FALSE</span>,</span>
<span id="cb211-9"><a href="visium-hd.html#cb211-9" tabindex="-1"></a>                     <span class="at">polygon_feat_type =</span> <span class="st">&#39;hex400&#39;</span>,</span>
<span id="cb211-10"><a href="visium-hd.html#cb211-10" tabindex="-1"></a>                     <span class="at">polygon_bg_color =</span> <span class="cn">NA</span>,</span>
<span id="cb211-11"><a href="visium-hd.html#cb211-11" tabindex="-1"></a>                     <span class="at">polygon_color =</span> <span class="st">&#39;white&#39;</span>,</span>
<span id="cb211-12"><a href="visium-hd.html#cb211-12" tabindex="-1"></a>                     <span class="at">polygon_line_size =</span> <span class="fl">0.1</span>,</span>
<span id="cb211-13"><a href="visium-hd.html#cb211-13" tabindex="-1"></a>                     <span class="at">expand_counts =</span> <span class="cn">TRUE</span>, </span>
<span id="cb211-14"><a href="visium-hd.html#cb211-14" tabindex="-1"></a>                     <span class="at">count_info_column =</span> <span class="st">&#39;count&#39;</span>,</span>
<span id="cb211-15"><a href="visium-hd.html#cb211-15" tabindex="-1"></a>                     <span class="at">jitter =</span> <span class="fu">c</span>(<span class="dv">25</span>,<span class="dv">25</span>), </span>
<span id="cb211-16"><a href="visium-hd.html#cb211-16" tabindex="-1"></a>                     <span class="at">plot_method =</span> <span class="st">&#39;scattermore&#39;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-278"></span>
<img src="img/02_session2/1-spatInSituPlotPoints.png" alt="Overlap of gene expression with the hex400 polygons. Genes/transcripts are rasterized. Jitter used to better vizualize individual transcripts" width="80%" />
<p class="caption">
Figure 9.10: Overlap of gene expression with the hex400 polygons. Genes/transcripts are rasterized. Jitter used to better vizualize individual transcripts
</p>
</div>
</div>
<div id="process-giotto-object" class="section level3 hasAnchor" number="9.4.4">
<h3><span class="header-section-number">9.4.4</span> Process Giotto object<a href="visium-hd.html#process-giotto-object" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="calculate-overlap-between-points-and-polygons" class="section level4 hasAnchor" number="9.4.4.1">
<h4><span class="header-section-number">9.4.4.1</span> calculate overlap between points and polygons<a href="visium-hd.html#calculate-overlap-between-points-and-polygons" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>At the moment the giotto points (transcripts) and polygons (hexagons) are two separate layers of information. Here we will determine which transcripts overlap with which hexagons so that we can aggregate the gene expression information and convert this into a gene expression matrix (genes-by-hexagons) that can be used in default spatial pipelines.</p>
<div class="sourceCode" id="cb212"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb212-1"><a href="visium-hd.html#cb212-1" tabindex="-1"></a><span class="co"># calculate overlap between points and polygons</span></span>
<span id="cb212-2"><a href="visium-hd.html#cb212-2" tabindex="-1"></a>visiumHD <span class="ot">=</span> <span class="fu">calculateOverlap</span>(visiumHD,</span>
<span id="cb212-3"><a href="visium-hd.html#cb212-3" tabindex="-1"></a>                            <span class="at">spatial_info =</span> <span class="st">&#39;hex400&#39;</span>,</span>
<span id="cb212-4"><a href="visium-hd.html#cb212-4" tabindex="-1"></a>                            <span class="at">feat_info =</span> <span class="st">&#39;rna&#39;</span>)</span>
<span id="cb212-5"><a href="visium-hd.html#cb212-5" tabindex="-1"></a></span>
<span id="cb212-6"><a href="visium-hd.html#cb212-6" tabindex="-1"></a><span class="fu">showGiottoSpatialInfo</span>(visiumHD)</span></code></pre></div>
</div>
<div id="convert-overlap-results-to-a-gene-by-hexagon-matrix" class="section level4 hasAnchor" number="9.4.4.2">
<h4><span class="header-section-number">9.4.4.2</span> convert overlap results to a gene-by-hexagon matrix<a href="visium-hd.html#convert-overlap-results-to-a-gene-by-hexagon-matrix" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb213"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb213-1"><a href="visium-hd.html#cb213-1" tabindex="-1"></a><span class="co"># convert overlap results to bin by gene matrix</span></span>
<span id="cb213-2"><a href="visium-hd.html#cb213-2" tabindex="-1"></a>visiumHD <span class="ot">=</span> <span class="fu">overlapToMatrix</span>(visiumHD,</span>
<span id="cb213-3"><a href="visium-hd.html#cb213-3" tabindex="-1"></a>                           <span class="at">poly_info =</span> <span class="st">&#39;hex400&#39;</span>,</span>
<span id="cb213-4"><a href="visium-hd.html#cb213-4" tabindex="-1"></a>                           <span class="at">feat_info =</span> <span class="st">&#39;rna&#39;</span>,</span>
<span id="cb213-5"><a href="visium-hd.html#cb213-5" tabindex="-1"></a>                           <span class="at">name =</span> <span class="st">&#39;raw&#39;</span>)</span>
<span id="cb213-6"><a href="visium-hd.html#cb213-6" tabindex="-1"></a></span>
<span id="cb213-7"><a href="visium-hd.html#cb213-7" tabindex="-1"></a><span class="co"># this action will automatically create an active spatial unit, ie. hexbin 400</span></span>
<span id="cb213-8"><a href="visium-hd.html#cb213-8" tabindex="-1"></a><span class="fu">activeSpatUnit</span>(visiumHD)</span></code></pre></div>
</div>
<div id="default-processing-steps" class="section level4 hasAnchor" number="9.4.4.3">
<h4><span class="header-section-number">9.4.4.3</span> default processing steps<a href="visium-hd.html#default-processing-steps" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>This part is similar to that described in the Visium tutorials (<a href="#Visium%20Part%20I">Part I</a> and <a href="#Visium%20Part%20II">Part II</a>).</p>
<div class="sourceCode" id="cb214"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb214-1"><a href="visium-hd.html#cb214-1" tabindex="-1"></a><span class="co"># filter on gene expression matrix</span></span>
<span id="cb214-2"><a href="visium-hd.html#cb214-2" tabindex="-1"></a>visiumHD <span class="ot">&lt;-</span> <span class="fu">filterGiotto</span>(visiumHD,</span>
<span id="cb214-3"><a href="visium-hd.html#cb214-3" tabindex="-1"></a>                         <span class="at">expression_threshold =</span> <span class="dv">1</span>,</span>
<span id="cb214-4"><a href="visium-hd.html#cb214-4" tabindex="-1"></a>                         <span class="at">feat_det_in_min_cells =</span> <span class="dv">5</span>,</span>
<span id="cb214-5"><a href="visium-hd.html#cb214-5" tabindex="-1"></a>                         <span class="at">min_det_feats_per_cell =</span> <span class="dv">25</span>)</span>
<span id="cb214-6"><a href="visium-hd.html#cb214-6" tabindex="-1"></a></span>
<span id="cb214-7"><a href="visium-hd.html#cb214-7" tabindex="-1"></a><span class="co"># normalize and scale gene expression data</span></span>
<span id="cb214-8"><a href="visium-hd.html#cb214-8" tabindex="-1"></a>visiumHD <span class="ot">&lt;-</span> <span class="fu">normalizeGiotto</span>(visiumHD, </span>
<span id="cb214-9"><a href="visium-hd.html#cb214-9" tabindex="-1"></a>                            <span class="at">scalefactor =</span> <span class="dv">1000</span>, </span>
<span id="cb214-10"><a href="visium-hd.html#cb214-10" tabindex="-1"></a>                            <span class="at">verbose =</span> T)</span>
<span id="cb214-11"><a href="visium-hd.html#cb214-11" tabindex="-1"></a></span>
<span id="cb214-12"><a href="visium-hd.html#cb214-12" tabindex="-1"></a><span class="co"># add cell and gene statistics</span></span>
<span id="cb214-13"><a href="visium-hd.html#cb214-13" tabindex="-1"></a>visiumHD <span class="ot">&lt;-</span> <span class="fu">addStatistics</span>(visiumHD)</span></code></pre></div>
<div id="visualize-number-of-features" class="section level5 hasAnchor" number="9.4.4.3.1">
<h5><span class="header-section-number">9.4.4.3.1</span> visualize number of features<a href="visium-hd.html#visualize-number-of-features" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<p>At the centroid level.</p>
<div class="sourceCode" id="cb215"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb215-1"><a href="visium-hd.html#cb215-1" tabindex="-1"></a><span class="co"># each dot here represents a 200x200 aggregation of spatial barcodes (bin size 200)</span></span>
<span id="cb215-2"><a href="visium-hd.html#cb215-2" tabindex="-1"></a><span class="fu">spatPlot2D</span>(<span class="at">gobject =</span> visiumHD,</span>
<span id="cb215-3"><a href="visium-hd.html#cb215-3" tabindex="-1"></a>           <span class="at">cell_color =</span> <span class="st">&quot;nr_feats&quot;</span>,</span>
<span id="cb215-4"><a href="visium-hd.html#cb215-4" tabindex="-1"></a>           <span class="at">color_as_factor =</span> F,</span>
<span id="cb215-5"><a href="visium-hd.html#cb215-5" tabindex="-1"></a>           <span class="at">point_size =</span> <span class="fl">2.5</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-283"></span>
<img src="img/02_session2/2-spatPlot2D.png" alt="Number of features detected in each of the centroids." width="100%" />
<p class="caption">
Figure 9.11: Number of features detected in each of the centroids.
</p>
</div>
<p>Using the spatial polygon (hexagon) tiles</p>
<div class="sourceCode" id="cb216"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb216-1"><a href="visium-hd.html#cb216-1" tabindex="-1"></a><span class="fu">spatInSituPlotPoints</span>(visiumHD,</span>
<span id="cb216-2"><a href="visium-hd.html#cb216-2" tabindex="-1"></a>                     <span class="at">show_image =</span> F,</span>
<span id="cb216-3"><a href="visium-hd.html#cb216-3" tabindex="-1"></a>                     <span class="at">feats =</span> <span class="cn">NULL</span>,</span>
<span id="cb216-4"><a href="visium-hd.html#cb216-4" tabindex="-1"></a>                     <span class="at">show_legend =</span> F,</span>
<span id="cb216-5"><a href="visium-hd.html#cb216-5" tabindex="-1"></a>                     <span class="at">spat_unit =</span> <span class="st">&#39;hex400&#39;</span>,</span>
<span id="cb216-6"><a href="visium-hd.html#cb216-6" tabindex="-1"></a>                     <span class="at">point_size =</span> <span class="fl">0.1</span>,</span>
<span id="cb216-7"><a href="visium-hd.html#cb216-7" tabindex="-1"></a>                     <span class="at">show_polygon =</span> <span class="cn">TRUE</span>,</span>
<span id="cb216-8"><a href="visium-hd.html#cb216-8" tabindex="-1"></a>                     <span class="at">use_overlap =</span> <span class="cn">TRUE</span>,</span>
<span id="cb216-9"><a href="visium-hd.html#cb216-9" tabindex="-1"></a>                     <span class="at">polygon_feat_type =</span> <span class="st">&#39;hex400&#39;</span>, </span>
<span id="cb216-10"><a href="visium-hd.html#cb216-10" tabindex="-1"></a>                     <span class="at">polygon_fill =</span> <span class="st">&#39;nr_feats&#39;</span>, </span>
<span id="cb216-11"><a href="visium-hd.html#cb216-11" tabindex="-1"></a>                     <span class="at">polygon_fill_as_factor =</span> F,</span>
<span id="cb216-12"><a href="visium-hd.html#cb216-12" tabindex="-1"></a>                     <span class="at">polygon_bg_color =</span> <span class="cn">NA</span>,</span>
<span id="cb216-13"><a href="visium-hd.html#cb216-13" tabindex="-1"></a>                     <span class="at">polygon_color =</span> <span class="st">&#39;white&#39;</span>,</span>
<span id="cb216-14"><a href="visium-hd.html#cb216-14" tabindex="-1"></a>                     <span class="at">polygon_line_size =</span> <span class="fl">0.1</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-285"></span>
<img src="img/02_session2/3-spatInSituPlotPoints.png" alt="Number of features detected in each of the hex400 polygons." width="100%" />
<p class="caption">
Figure 9.12: Number of features detected in each of the hex400 polygons.
</p>
</div>
</div>
</div>
<div id="dimension-reduction-clustering" class="section level4 hasAnchor" number="9.4.4.4">
<h4><span class="header-section-number">9.4.4.4</span> Dimension reduction + clustering<a href="visium-hd.html#dimension-reduction-clustering" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div id="highly-variable-features-pca" class="section level5 hasAnchor" number="9.4.4.4.1">
<h5><span class="header-section-number">9.4.4.4.1</span> Highly variable features + PCA<a href="visium-hd.html#highly-variable-features-pca" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<div class="sourceCode" id="cb217"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb217-1"><a href="visium-hd.html#cb217-1" tabindex="-1"></a>visiumHD <span class="ot">&lt;-</span> <span class="fu">calculateHVF</span>(visiumHD, </span>
<span id="cb217-2"><a href="visium-hd.html#cb217-2" tabindex="-1"></a>                         <span class="at">zscore_threshold =</span> <span class="dv">1</span>)</span>
<span id="cb217-3"><a href="visium-hd.html#cb217-3" tabindex="-1"></a>visiumHD <span class="ot">&lt;-</span> <span class="fu">runPCA</span>(visiumHD, </span>
<span id="cb217-4"><a href="visium-hd.html#cb217-4" tabindex="-1"></a>                   <span class="at">expression_values =</span> <span class="st">&#39;normalized&#39;</span>, </span>
<span id="cb217-5"><a href="visium-hd.html#cb217-5" tabindex="-1"></a>                   <span class="at">feats_to_use =</span> <span class="st">&#39;hvf&#39;</span>)</span>
<span id="cb217-6"><a href="visium-hd.html#cb217-6" tabindex="-1"></a><span class="fu">screePlot</span>(visiumHD, <span class="at">ncp =</span> <span class="dv">30</span>)</span></code></pre></div>
<p><img src="img/02_session2/4-screePlot.png" width="100%" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb218"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb218-1"><a href="visium-hd.html#cb218-1" tabindex="-1"></a><span class="fu">plotPCA</span>(visiumHD)</span></code></pre></div>
<p><img src="img/02_session2/5-PCA.png" width="100%" style="display: block; margin: auto;" /></p>
</div>
<div id="umap-reduction-for-visualization" class="section level5 hasAnchor" number="9.4.4.4.2">
<h5><span class="header-section-number">9.4.4.4.2</span> UMAP reduction for visualization<a href="visium-hd.html#umap-reduction-for-visualization" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<div class="sourceCode" id="cb219"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb219-1"><a href="visium-hd.html#cb219-1" tabindex="-1"></a>visiumHD <span class="ot">&lt;-</span> <span class="fu">runUMAP</span>(visiumHD, </span>
<span id="cb219-2"><a href="visium-hd.html#cb219-2" tabindex="-1"></a>                    <span class="at">dimensions_to_use =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">14</span>, </span>
<span id="cb219-3"><a href="visium-hd.html#cb219-3" tabindex="-1"></a>                    <span class="at">n_threads =</span> <span class="dv">10</span>)</span>
<span id="cb219-4"><a href="visium-hd.html#cb219-4" tabindex="-1"></a></span>
<span id="cb219-5"><a href="visium-hd.html#cb219-5" tabindex="-1"></a><span class="fu">plotUMAP</span>(<span class="at">gobject =</span> visiumHD,</span>
<span id="cb219-6"><a href="visium-hd.html#cb219-6" tabindex="-1"></a>         <span class="at">point_size =</span> <span class="dv">1</span>)</span></code></pre></div>
<p><img src="img/02_session2/6-UMAP.png" width="100%" style="display: block; margin: auto;" /></p>
</div>
<div id="create-network-based-on-expression-similarity-graph-partition-cluster" class="section level5 hasAnchor" number="9.4.4.4.3">
<h5><span class="header-section-number">9.4.4.4.3</span> Create network based on expression similarity + graph partition cluster<a href="visium-hd.html#create-network-based-on-expression-similarity-graph-partition-cluster" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<div class="sourceCode" id="cb220"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb220-1"><a href="visium-hd.html#cb220-1" tabindex="-1"></a><span class="co"># sNN network (default)</span></span>
<span id="cb220-2"><a href="visium-hd.html#cb220-2" tabindex="-1"></a>visiumHD <span class="ot">&lt;-</span> <span class="fu">createNearestNetwork</span>(visiumHD, </span>
<span id="cb220-3"><a href="visium-hd.html#cb220-3" tabindex="-1"></a>                                 <span class="at">dimensions_to_use =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">14</span>, </span>
<span id="cb220-4"><a href="visium-hd.html#cb220-4" tabindex="-1"></a>                                 <span class="at">k =</span> <span class="dv">5</span>)</span>
<span id="cb220-5"><a href="visium-hd.html#cb220-5" tabindex="-1"></a></span>
<span id="cb220-6"><a href="visium-hd.html#cb220-6" tabindex="-1"></a><span class="do">## leiden clustering ####</span></span>
<span id="cb220-7"><a href="visium-hd.html#cb220-7" tabindex="-1"></a>visiumHD <span class="ot">&lt;-</span> <span class="fu">doLeidenClusterIgraph</span>(visiumHD, <span class="at">resolution =</span> <span class="fl">0.5</span>, <span class="at">n_iterations =</span> <span class="dv">1000</span>, <span class="at">spat_unit =</span> <span class="st">&#39;hex400&#39;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb221"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb221-1"><a href="visium-hd.html#cb221-1" tabindex="-1"></a><span class="fu">plotUMAP</span>(<span class="at">gobject =</span> visiumHD,</span>
<span id="cb221-2"><a href="visium-hd.html#cb221-2" tabindex="-1"></a>         <span class="at">cell_color =</span> <span class="st">&#39;leiden_clus&#39;</span>, </span>
<span id="cb221-3"><a href="visium-hd.html#cb221-3" tabindex="-1"></a>         <span class="at">point_size =</span> <span class="fl">1.5</span>,</span>
<span id="cb221-4"><a href="visium-hd.html#cb221-4" tabindex="-1"></a>         <span class="at">show_NN_network =</span> F, </span>
<span id="cb221-5"><a href="visium-hd.html#cb221-5" tabindex="-1"></a>         <span class="at">edge_alpha =</span> <span class="fl">0.05</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-294"></span>
<img src="img/02_session2/7-UMAP.png" alt="Leiden clustering for the hex400 bins." width="100%" />
<p class="caption">
Figure 9.13: Leiden clustering for the hex400 bins.
</p>
</div>
<div class="sourceCode" id="cb222"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb222-1"><a href="visium-hd.html#cb222-1" tabindex="-1"></a><span class="fu">spatInSituPlotPoints</span>(visiumHD,</span>
<span id="cb222-2"><a href="visium-hd.html#cb222-2" tabindex="-1"></a>                     <span class="at">show_image =</span> F,</span>
<span id="cb222-3"><a href="visium-hd.html#cb222-3" tabindex="-1"></a>                     <span class="at">feats =</span> <span class="cn">NULL</span>,</span>
<span id="cb222-4"><a href="visium-hd.html#cb222-4" tabindex="-1"></a>                     <span class="at">show_legend =</span> F,</span>
<span id="cb222-5"><a href="visium-hd.html#cb222-5" tabindex="-1"></a>                     <span class="at">spat_unit =</span> <span class="st">&#39;hex400&#39;</span>,</span>
<span id="cb222-6"><a href="visium-hd.html#cb222-6" tabindex="-1"></a>                     <span class="at">point_size =</span> <span class="fl">0.25</span>,</span>
<span id="cb222-7"><a href="visium-hd.html#cb222-7" tabindex="-1"></a>                     <span class="at">show_polygon =</span> <span class="cn">TRUE</span>,</span>
<span id="cb222-8"><a href="visium-hd.html#cb222-8" tabindex="-1"></a>                     <span class="at">use_overlap =</span> <span class="cn">FALSE</span>,</span>
<span id="cb222-9"><a href="visium-hd.html#cb222-9" tabindex="-1"></a>                     <span class="at">polygon_feat_type =</span> <span class="st">&#39;hex400&#39;</span>,</span>
<span id="cb222-10"><a href="visium-hd.html#cb222-10" tabindex="-1"></a>                     <span class="at">polygon_fill_as_factor =</span> <span class="cn">TRUE</span>,</span>
<span id="cb222-11"><a href="visium-hd.html#cb222-11" tabindex="-1"></a>                     <span class="at">polygon_fill =</span> <span class="st">&#39;leiden_clus&#39;</span>,</span>
<span id="cb222-12"><a href="visium-hd.html#cb222-12" tabindex="-1"></a>                     <span class="at">polygon_color =</span> <span class="st">&#39;black&#39;</span>,</span>
<span id="cb222-13"><a href="visium-hd.html#cb222-13" tabindex="-1"></a>                     <span class="at">polygon_line_size =</span> <span class="fl">0.3</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-296"></span>
<img src="img/02_session2/8-spatInSituPlotPoints.png" alt="Spat plot for hex400 bin colored by leiden clusters." width="100%" />
<p class="caption">
Figure 9.14: Spat plot for hex400 bin colored by leiden clusters.
</p>
</div>
</div>
</div>
</div>
</div>
<div id="hexbin-100" class="section level2 hasAnchor" number="9.5">
<h2><span class="header-section-number">9.5</span> Hexbin 100<a href="visium-hd.html#hexbin-100" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Observation: Hexbin 400 results in very coarse information about the tissue.</p>
<p>Goal is to create a higher resolution bin (hex100), then add this to the Giotto object to compare difference in resolution.</p>
<div id="standard-subcellular-pipeline" class="section level3 hasAnchor" number="9.5.1">
<h3><span class="header-section-number">9.5.1</span> Standard subcellular pipeline<a href="visium-hd.html#standard-subcellular-pipeline" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li>Create new spatial unit layer, e.g. with tessellate function<br />
</li>
<li>Add spatial units to Giottoo object<br />
</li>
<li>Calculate centroids (optional)<br />
</li>
<li>Compute overlap between transcript and polygon (hexagon) locations.<br />
</li>
<li>Convert overlap data into a gene-by-polygon matrix</li>
</ol>
<div class="sourceCode" id="cb223"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb223-1"><a href="visium-hd.html#cb223-1" tabindex="-1"></a>hexbin100 <span class="ot">&lt;-</span> <span class="fu">tessellate</span>(<span class="at">extent =</span> <span class="fu">ext</span>(visiumHD), </span>
<span id="cb223-2"><a href="visium-hd.html#cb223-2" tabindex="-1"></a>                        <span class="at">shape =</span> <span class="st">&#39;hexagon&#39;</span>, </span>
<span id="cb223-3"><a href="visium-hd.html#cb223-3" tabindex="-1"></a>                        <span class="at">shape_size =</span> <span class="dv">100</span>, </span>
<span id="cb223-4"><a href="visium-hd.html#cb223-4" tabindex="-1"></a>                        <span class="at">name =</span> <span class="st">&#39;hex100&#39;</span>) </span>
<span id="cb223-5"><a href="visium-hd.html#cb223-5" tabindex="-1"></a></span>
<span id="cb223-6"><a href="visium-hd.html#cb223-6" tabindex="-1"></a>visiumHD <span class="ot">=</span> <span class="fu">setPolygonInfo</span>(<span class="at">gobject =</span> visiumHD,</span>
<span id="cb223-7"><a href="visium-hd.html#cb223-7" tabindex="-1"></a>                          <span class="at">x =</span> hexbin100,</span>
<span id="cb223-8"><a href="visium-hd.html#cb223-8" tabindex="-1"></a>                          <span class="at">name =</span> <span class="st">&#39;hex100&#39;</span>,</span>
<span id="cb223-9"><a href="visium-hd.html#cb223-9" tabindex="-1"></a>                          <span class="at">initialize =</span> T)</span>
<span id="cb223-10"><a href="visium-hd.html#cb223-10" tabindex="-1"></a></span>
<span id="cb223-11"><a href="visium-hd.html#cb223-11" tabindex="-1"></a>visiumHD <span class="ot">=</span> <span class="fu">addSpatialCentroidLocations</span>(<span class="at">gobject =</span> visiumHD,</span>
<span id="cb223-12"><a href="visium-hd.html#cb223-12" tabindex="-1"></a>                                       <span class="at">poly_info =</span> <span class="st">&#39;hex100&#39;</span>)</span></code></pre></div>
<p>Set active spatial unit. This can also be set manually in each function.</p>
<div class="sourceCode" id="cb224"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb224-1"><a href="visium-hd.html#cb224-1" tabindex="-1"></a><span class="fu">activeSpatUnit</span>(visiumHD) <span class="ot">&lt;-</span> <span class="st">&#39;hex100&#39;</span></span></code></pre></div>
<p>Let’s visualize the higher resolution hexagons.</p>
<div class="sourceCode" id="cb225"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb225-1"><a href="visium-hd.html#cb225-1" tabindex="-1"></a><span class="fu">spatInSituPlotPoints</span>(visiumHD,</span>
<span id="cb225-2"><a href="visium-hd.html#cb225-2" tabindex="-1"></a>                     <span class="at">show_image =</span> F,</span>
<span id="cb225-3"><a href="visium-hd.html#cb225-3" tabindex="-1"></a>                     <span class="at">feats =</span> <span class="fu">list</span>(<span class="st">&#39;rna&#39;</span> <span class="ot">=</span> feature_data<span class="sc">$</span>feat_ID[<span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>]),</span>
<span id="cb225-4"><a href="visium-hd.html#cb225-4" tabindex="-1"></a>                     <span class="at">show_legend =</span> T,</span>
<span id="cb225-5"><a href="visium-hd.html#cb225-5" tabindex="-1"></a>                     <span class="at">spat_unit =</span> <span class="st">&#39;hex100&#39;</span>,</span>
<span id="cb225-6"><a href="visium-hd.html#cb225-6" tabindex="-1"></a>                     <span class="at">point_size =</span> <span class="fl">0.1</span>,</span>
<span id="cb225-7"><a href="visium-hd.html#cb225-7" tabindex="-1"></a>                     <span class="at">show_polygon =</span> <span class="cn">TRUE</span>,</span>
<span id="cb225-8"><a href="visium-hd.html#cb225-8" tabindex="-1"></a>                     <span class="at">use_overlap =</span> <span class="cn">FALSE</span>,</span>
<span id="cb225-9"><a href="visium-hd.html#cb225-9" tabindex="-1"></a>                     <span class="at">polygon_feat_type =</span> <span class="st">&#39;hex100&#39;</span>,</span>
<span id="cb225-10"><a href="visium-hd.html#cb225-10" tabindex="-1"></a>                     <span class="at">polygon_bg_color =</span> <span class="cn">NA</span>,</span>
<span id="cb225-11"><a href="visium-hd.html#cb225-11" tabindex="-1"></a>                     <span class="at">polygon_color =</span> <span class="st">&#39;white&#39;</span>,</span>
<span id="cb225-12"><a href="visium-hd.html#cb225-12" tabindex="-1"></a>                     <span class="at">polygon_line_size =</span> <span class="fl">0.2</span>,</span>
<span id="cb225-13"><a href="visium-hd.html#cb225-13" tabindex="-1"></a>                     <span class="at">expand_counts =</span> <span class="cn">TRUE</span>, </span>
<span id="cb225-14"><a href="visium-hd.html#cb225-14" tabindex="-1"></a>                     <span class="at">count_info_column =</span> <span class="st">&#39;count&#39;</span>,</span>
<span id="cb225-15"><a href="visium-hd.html#cb225-15" tabindex="-1"></a>                     <span class="at">jitter =</span> <span class="fu">c</span>(<span class="dv">25</span>,<span class="dv">25</span>))</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-300"></span>
<img src="img/02_session2/9-spatInSituPlotPoints.png" alt="Polygon overlay of hex100 bins over 2 µm pixel. Jitter applied to vizualize individual features." width="100%" />
<p class="caption">
Figure 9.15: Polygon overlay of hex100 bins over 2 µm pixel. Jitter applied to vizualize individual features.
</p>
</div>
<div class="sourceCode" id="cb226"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb226-1"><a href="visium-hd.html#cb226-1" tabindex="-1"></a>visiumHD <span class="ot">=</span> <span class="fu">calculateOverlap</span>(visiumHD,</span>
<span id="cb226-2"><a href="visium-hd.html#cb226-2" tabindex="-1"></a>                            <span class="at">spatial_info =</span> <span class="st">&#39;hex100&#39;</span>,</span>
<span id="cb226-3"><a href="visium-hd.html#cb226-3" tabindex="-1"></a>                            <span class="at">feat_info =</span> <span class="st">&#39;rna&#39;</span>)</span>
<span id="cb226-4"><a href="visium-hd.html#cb226-4" tabindex="-1"></a></span>
<span id="cb226-5"><a href="visium-hd.html#cb226-5" tabindex="-1"></a>visiumHD <span class="ot">=</span> <span class="fu">overlapToMatrix</span>(visiumHD,</span>
<span id="cb226-6"><a href="visium-hd.html#cb226-6" tabindex="-1"></a>                           <span class="at">poly_info =</span> <span class="st">&#39;hex100&#39;</span>,</span>
<span id="cb226-7"><a href="visium-hd.html#cb226-7" tabindex="-1"></a>                           <span class="at">feat_info =</span> <span class="st">&#39;rna&#39;</span>,</span>
<span id="cb226-8"><a href="visium-hd.html#cb226-8" tabindex="-1"></a>                           <span class="at">name =</span> <span class="st">&#39;raw&#39;</span>)</span>
<span id="cb226-9"><a href="visium-hd.html#cb226-9" tabindex="-1"></a></span>
<span id="cb226-10"><a href="visium-hd.html#cb226-10" tabindex="-1"></a>visiumHD <span class="ot">&lt;-</span> <span class="fu">filterGiotto</span>(visiumHD,</span>
<span id="cb226-11"><a href="visium-hd.html#cb226-11" tabindex="-1"></a>                         <span class="at">expression_threshold =</span> <span class="dv">1</span>,</span>
<span id="cb226-12"><a href="visium-hd.html#cb226-12" tabindex="-1"></a>                         <span class="at">feat_det_in_min_cells =</span> <span class="dv">10</span>,</span>
<span id="cb226-13"><a href="visium-hd.html#cb226-13" tabindex="-1"></a>                         <span class="at">min_det_feats_per_cell =</span> <span class="dv">10</span>)</span>
<span id="cb226-14"><a href="visium-hd.html#cb226-14" tabindex="-1"></a></span>
<span id="cb226-15"><a href="visium-hd.html#cb226-15" tabindex="-1"></a>visiumHD <span class="ot">&lt;-</span> <span class="fu">normalizeGiotto</span>(visiumHD, <span class="at">scalefactor =</span> <span class="dv">1000</span>, <span class="at">verbose =</span> T)</span>
<span id="cb226-16"><a href="visium-hd.html#cb226-16" tabindex="-1"></a></span>
<span id="cb226-17"><a href="visium-hd.html#cb226-17" tabindex="-1"></a>visiumHD <span class="ot">&lt;-</span> <span class="fu">addStatistics</span>(visiumHD)</span></code></pre></div>
<p>Your Giotto object will have metadata for each spatial unit.</p>
<div class="sourceCode" id="cb227"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb227-1"><a href="visium-hd.html#cb227-1" tabindex="-1"></a><span class="fu">pDataDT</span>(visiumHD, <span class="at">spat_unit =</span> <span class="st">&#39;hex100&#39;</span>)</span>
<span id="cb227-2"><a href="visium-hd.html#cb227-2" tabindex="-1"></a><span class="fu">pDataDT</span>(visiumHD, <span class="at">spat_unit =</span> <span class="st">&#39;hex400&#39;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb228"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb228-1"><a href="visium-hd.html#cb228-1" tabindex="-1"></a><span class="do">## dimension reduction ####</span></span>
<span id="cb228-2"><a href="visium-hd.html#cb228-2" tabindex="-1"></a><span class="co"># --------------------------- #</span></span>
<span id="cb228-3"><a href="visium-hd.html#cb228-3" tabindex="-1"></a>visiumHD <span class="ot">&lt;-</span> <span class="fu">calculateHVF</span>(visiumHD, <span class="at">zscore_threshold =</span> <span class="dv">1</span>)</span>
<span id="cb228-4"><a href="visium-hd.html#cb228-4" tabindex="-1"></a></span>
<span id="cb228-5"><a href="visium-hd.html#cb228-5" tabindex="-1"></a>visiumHD <span class="ot">&lt;-</span> <span class="fu">runPCA</span>(visiumHD, <span class="at">expression_values =</span> <span class="st">&#39;normalized&#39;</span>, <span class="at">feats_to_use =</span> <span class="st">&#39;hvf&#39;</span>)</span>
<span id="cb228-6"><a href="visium-hd.html#cb228-6" tabindex="-1"></a></span>
<span id="cb228-7"><a href="visium-hd.html#cb228-7" tabindex="-1"></a><span class="fu">plotPCA</span>(visiumHD)</span></code></pre></div>
<p><img src="img/02_session2/10-PCA.png" width="100%" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb229"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb229-1"><a href="visium-hd.html#cb229-1" tabindex="-1"></a>visiumHD <span class="ot">&lt;-</span> <span class="fu">runUMAP</span>(visiumHD, <span class="at">dimensions_to_use =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">14</span>, <span class="at">n_threads =</span> <span class="dv">10</span>)</span>
<span id="cb229-2"><a href="visium-hd.html#cb229-2" tabindex="-1"></a></span>
<span id="cb229-3"><a href="visium-hd.html#cb229-3" tabindex="-1"></a><span class="co"># plot UMAP, coloring cells/points based on nr_feats</span></span>
<span id="cb229-4"><a href="visium-hd.html#cb229-4" tabindex="-1"></a><span class="fu">plotUMAP</span>(<span class="at">gobject =</span> visiumHD,</span>
<span id="cb229-5"><a href="visium-hd.html#cb229-5" tabindex="-1"></a>         <span class="at">point_size =</span> <span class="dv">2</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-306"></span>
<img src="img/02_session2/11-UMAP.png" alt="UMAP for the hex100 bin." width="100%" />
<p class="caption">
Figure 9.16: UMAP for the hex100 bin.
</p>
</div>
<div class="sourceCode" id="cb230"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb230-1"><a href="visium-hd.html#cb230-1" tabindex="-1"></a><span class="co"># sNN network (default)</span></span>
<span id="cb230-2"><a href="visium-hd.html#cb230-2" tabindex="-1"></a>visiumHD <span class="ot">&lt;-</span> <span class="fu">createNearestNetwork</span>(visiumHD, </span>
<span id="cb230-3"><a href="visium-hd.html#cb230-3" tabindex="-1"></a>                                 <span class="at">dimensions_to_use =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">14</span>, </span>
<span id="cb230-4"><a href="visium-hd.html#cb230-4" tabindex="-1"></a>                                 <span class="at">k =</span> <span class="dv">5</span>)</span>
<span id="cb230-5"><a href="visium-hd.html#cb230-5" tabindex="-1"></a></span>
<span id="cb230-6"><a href="visium-hd.html#cb230-6" tabindex="-1"></a><span class="do">## leiden clustering ####</span></span>
<span id="cb230-7"><a href="visium-hd.html#cb230-7" tabindex="-1"></a>visiumHD <span class="ot">&lt;-</span> <span class="fu">doLeidenClusterIgraph</span>(visiumHD, <span class="at">resolution =</span> <span class="fl">0.2</span>, <span class="at">n_iterations =</span> <span class="dv">1000</span>)</span>
<span id="cb230-8"><a href="visium-hd.html#cb230-8" tabindex="-1"></a></span>
<span id="cb230-9"><a href="visium-hd.html#cb230-9" tabindex="-1"></a><span class="fu">plotUMAP</span>(<span class="at">gobject =</span> visiumHD,</span>
<span id="cb230-10"><a href="visium-hd.html#cb230-10" tabindex="-1"></a>         <span class="at">cell_color =</span> <span class="st">&#39;leiden_clus&#39;</span>, </span>
<span id="cb230-11"><a href="visium-hd.html#cb230-11" tabindex="-1"></a>         <span class="at">point_size =</span> <span class="fl">1.5</span>,</span>
<span id="cb230-12"><a href="visium-hd.html#cb230-12" tabindex="-1"></a>         <span class="at">show_NN_network =</span> F, </span>
<span id="cb230-13"><a href="visium-hd.html#cb230-13" tabindex="-1"></a>         <span class="at">edge_alpha =</span> <span class="fl">0.05</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-308"></span>
<img src="img/02_session2/12-UMAP.png" alt="UMAP for the hex100 bin colored by ledien clusters." width="100%" />
<p class="caption">
Figure 9.17: UMAP for the hex100 bin colored by ledien clusters.
</p>
</div>
<div class="sourceCode" id="cb231"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb231-1"><a href="visium-hd.html#cb231-1" tabindex="-1"></a><span class="fu">spatInSituPlotPoints</span>(visiumHD,</span>
<span id="cb231-2"><a href="visium-hd.html#cb231-2" tabindex="-1"></a>                     <span class="at">show_image =</span> F,</span>
<span id="cb231-3"><a href="visium-hd.html#cb231-3" tabindex="-1"></a>                     <span class="at">feats =</span> <span class="cn">NULL</span>,</span>
<span id="cb231-4"><a href="visium-hd.html#cb231-4" tabindex="-1"></a>                     <span class="at">show_legend =</span> F,</span>
<span id="cb231-5"><a href="visium-hd.html#cb231-5" tabindex="-1"></a>                     <span class="at">spat_unit =</span> <span class="st">&#39;hex100&#39;</span>,</span>
<span id="cb231-6"><a href="visium-hd.html#cb231-6" tabindex="-1"></a>                     <span class="at">point_size =</span> <span class="fl">0.5</span>,</span>
<span id="cb231-7"><a href="visium-hd.html#cb231-7" tabindex="-1"></a>                     <span class="at">show_polygon =</span> <span class="cn">TRUE</span>,</span>
<span id="cb231-8"><a href="visium-hd.html#cb231-8" tabindex="-1"></a>                     <span class="at">use_overlap =</span> <span class="cn">FALSE</span>,</span>
<span id="cb231-9"><a href="visium-hd.html#cb231-9" tabindex="-1"></a>                     <span class="at">polygon_feat_type =</span> <span class="st">&#39;hex100&#39;</span>,</span>
<span id="cb231-10"><a href="visium-hd.html#cb231-10" tabindex="-1"></a>                     <span class="at">polygon_fill_as_factor =</span> <span class="cn">TRUE</span>,</span>
<span id="cb231-11"><a href="visium-hd.html#cb231-11" tabindex="-1"></a>                     <span class="at">polygon_fill =</span> <span class="st">&#39;leiden_clus&#39;</span>,</span>
<span id="cb231-12"><a href="visium-hd.html#cb231-12" tabindex="-1"></a>                     <span class="at">polygon_color =</span> <span class="st">&#39;black&#39;</span>,</span>
<span id="cb231-13"><a href="visium-hd.html#cb231-13" tabindex="-1"></a>                     <span class="at">polygon_line_size =</span> <span class="fl">0.3</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-310"></span>
<img src="img/02_session2/13-spatInSituPlotPoints.png" alt="Spat plot for the hex100 bin colored by leiden clusters." width="100%" />
<p class="caption">
Figure 9.18: Spat plot for the hex100 bin colored by leiden clusters.
</p>
</div>
<p>This resolution definitely shows more promise to identify interesting spatial patterns.</p>
</div>
<div id="spatial-expression-patterns-1" class="section level3 hasAnchor" number="9.5.2">
<h3><span class="header-section-number">9.5.2</span> Spatial expression patterns<a href="visium-hd.html#spatial-expression-patterns-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="identify-single-genes" class="section level4 hasAnchor" number="9.5.2.1">
<h4><span class="header-section-number">9.5.2.1</span> Identify single genes<a href="visium-hd.html#identify-single-genes" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Here we will use binSpect as a quick method to rank genes with high potential for spatial coherent expression patterns.</p>
<div class="sourceCode" id="cb232"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb232-1"><a href="visium-hd.html#cb232-1" tabindex="-1"></a>featData <span class="ot">=</span> <span class="fu">fDataDT</span>(visiumHD)</span>
<span id="cb232-2"><a href="visium-hd.html#cb232-2" tabindex="-1"></a>hvf_genes <span class="ot">=</span> featData[hvf <span class="sc">==</span> <span class="st">&#39;yes&#39;</span>]<span class="sc">$</span>feat_ID</span>
<span id="cb232-3"><a href="visium-hd.html#cb232-3" tabindex="-1"></a></span>
<span id="cb232-4"><a href="visium-hd.html#cb232-4" tabindex="-1"></a>visiumHD <span class="ot">=</span> <span class="fu">createSpatialNetwork</span>(visiumHD,</span>
<span id="cb232-5"><a href="visium-hd.html#cb232-5" tabindex="-1"></a>                                <span class="at">name =</span> <span class="st">&#39;kNN_network&#39;</span>,</span>
<span id="cb232-6"><a href="visium-hd.html#cb232-6" tabindex="-1"></a>                                <span class="at">spat_unit =</span> <span class="st">&#39;hex100&#39;</span>, </span>
<span id="cb232-7"><a href="visium-hd.html#cb232-7" tabindex="-1"></a>                                <span class="at">method =</span> <span class="st">&#39;kNN&#39;</span>,</span>
<span id="cb232-8"><a href="visium-hd.html#cb232-8" tabindex="-1"></a>                                <span class="at">k =</span> <span class="dv">8</span>)</span>
<span id="cb232-9"><a href="visium-hd.html#cb232-9" tabindex="-1"></a></span>
<span id="cb232-10"><a href="visium-hd.html#cb232-10" tabindex="-1"></a>ranktest <span class="ot">=</span> <span class="fu">binSpect</span>(visiumHD,</span>
<span id="cb232-11"><a href="visium-hd.html#cb232-11" tabindex="-1"></a>                    <span class="at">spat_unit =</span> <span class="st">&#39;hex100&#39;</span>,</span>
<span id="cb232-12"><a href="visium-hd.html#cb232-12" tabindex="-1"></a>                    <span class="at">subset_feats =</span> hvf_genes,</span>
<span id="cb232-13"><a href="visium-hd.html#cb232-13" tabindex="-1"></a>                    <span class="at">bin_method =</span> <span class="st">&#39;rank&#39;</span>,</span>
<span id="cb232-14"><a href="visium-hd.html#cb232-14" tabindex="-1"></a>                    <span class="at">calc_hub =</span> <span class="cn">FALSE</span>,</span>
<span id="cb232-15"><a href="visium-hd.html#cb232-15" tabindex="-1"></a>                    <span class="at">do_fisher_test =</span> <span class="cn">TRUE</span>,</span>
<span id="cb232-16"><a href="visium-hd.html#cb232-16" tabindex="-1"></a>                    <span class="at">spatial_network_name =</span> <span class="st">&#39;kNN_network&#39;</span>)</span></code></pre></div>
<p>Visualize top 2 ranked spatial genes per expression bin:</p>
<div class="sourceCode" id="cb233"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb233-1"><a href="visium-hd.html#cb233-1" tabindex="-1"></a>set0 <span class="ot">=</span> ranktest[high_expr <span class="sc">&lt;</span> <span class="dv">50</span>][<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>]<span class="sc">$</span>feats</span>
<span id="cb233-2"><a href="visium-hd.html#cb233-2" tabindex="-1"></a>set1 <span class="ot">=</span> ranktest[high_expr <span class="sc">&gt;</span> <span class="dv">50</span> <span class="sc">&amp;</span> high_expr <span class="sc">&lt;</span> <span class="dv">100</span>][<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>]<span class="sc">$</span>feats</span>
<span id="cb233-3"><a href="visium-hd.html#cb233-3" tabindex="-1"></a>set2 <span class="ot">=</span> ranktest[high_expr <span class="sc">&gt;</span> <span class="dv">100</span> <span class="sc">&amp;</span> high_expr <span class="sc">&lt;</span> <span class="dv">200</span>][<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>]<span class="sc">$</span>feats</span>
<span id="cb233-4"><a href="visium-hd.html#cb233-4" tabindex="-1"></a>set3 <span class="ot">=</span> ranktest[high_expr <span class="sc">&gt;</span> <span class="dv">200</span> <span class="sc">&amp;</span> high_expr <span class="sc">&lt;</span> <span class="dv">400</span>][<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>]<span class="sc">$</span>feats</span>
<span id="cb233-5"><a href="visium-hd.html#cb233-5" tabindex="-1"></a>set4 <span class="ot">=</span> ranktest[high_expr <span class="sc">&gt;</span> <span class="dv">400</span> <span class="sc">&amp;</span> high_expr <span class="sc">&lt;</span> <span class="dv">1000</span>][<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>]<span class="sc">$</span>feats</span>
<span id="cb233-6"><a href="visium-hd.html#cb233-6" tabindex="-1"></a>set5 <span class="ot">=</span> ranktest[high_expr <span class="sc">&gt;</span> <span class="dv">1000</span>][<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>]<span class="sc">$</span>feats</span>
<span id="cb233-7"><a href="visium-hd.html#cb233-7" tabindex="-1"></a></span>
<span id="cb233-8"><a href="visium-hd.html#cb233-8" tabindex="-1"></a><span class="fu">spatFeatPlot2D</span>(visiumHD,</span>
<span id="cb233-9"><a href="visium-hd.html#cb233-9" tabindex="-1"></a>               <span class="at">expression_values =</span> <span class="st">&#39;scaled&#39;</span>,</span>
<span id="cb233-10"><a href="visium-hd.html#cb233-10" tabindex="-1"></a>               <span class="at">feats =</span> <span class="fu">c</span>(set0, set1, set2),</span>
<span id="cb233-11"><a href="visium-hd.html#cb233-11" tabindex="-1"></a>               <span class="at">gradient_style =</span> <span class="st">&quot;sequential&quot;</span>,</span>
<span id="cb233-12"><a href="visium-hd.html#cb233-12" tabindex="-1"></a>               <span class="at">cell_color_gradient =</span> <span class="fu">c</span>(<span class="st">&#39;blue&#39;</span>, <span class="st">&#39;white&#39;</span>, <span class="st">&#39;yellow&#39;</span>, <span class="st">&#39;orange&#39;</span>, <span class="st">&#39;red&#39;</span>, <span class="st">&#39;darkred&#39;</span>),</span>
<span id="cb233-13"><a href="visium-hd.html#cb233-13" tabindex="-1"></a>               <span class="at">cow_n_col =</span> <span class="dv">2</span>, <span class="at">point_size =</span> <span class="dv">1</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-313"></span>
<img src="img/02_session2/14-spatFeatPlot2D.png" alt="Spat feature plot showing gene expression for the top 2 ranked spatial genes per expression bin (&lt;50, &gt;50 and &gt;100) across the hex100 bin." width="100%" />
<p class="caption">
Figure 9.19: Spat feature plot showing gene expression for the top 2 ranked spatial genes per expression bin (&lt;50, &gt;50 and &gt;100) across the hex100 bin.
</p>
</div>
<div class="sourceCode" id="cb234"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb234-1"><a href="visium-hd.html#cb234-1" tabindex="-1"></a><span class="fu">spatFeatPlot2D</span>(visiumHD,</span>
<span id="cb234-2"><a href="visium-hd.html#cb234-2" tabindex="-1"></a>               <span class="at">expression_values =</span> <span class="st">&#39;scaled&#39;</span>,</span>
<span id="cb234-3"><a href="visium-hd.html#cb234-3" tabindex="-1"></a>               <span class="at">feats =</span> <span class="fu">c</span>(set3, set4, set5),</span>
<span id="cb234-4"><a href="visium-hd.html#cb234-4" tabindex="-1"></a>               <span class="at">gradient_style =</span> <span class="st">&quot;sequential&quot;</span>,</span>
<span id="cb234-5"><a href="visium-hd.html#cb234-5" tabindex="-1"></a>               <span class="at">cell_color_gradient =</span> <span class="fu">c</span>(<span class="st">&#39;blue&#39;</span>, <span class="st">&#39;white&#39;</span>, <span class="st">&#39;yellow&#39;</span>, <span class="st">&#39;orange&#39;</span>, <span class="st">&#39;red&#39;</span>, <span class="st">&#39;darkred&#39;</span>),</span>
<span id="cb234-6"><a href="visium-hd.html#cb234-6" tabindex="-1"></a>               <span class="at">cow_n_col =</span> <span class="dv">2</span>, <span class="at">point_size =</span> <span class="dv">1</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-315"></span>
<img src="img/02_session2/15-spatFeatPlot2D.png" alt="Spat feature plot showing gene expression for the top 2 ranked spatial genes per expression bin (&gt;200, &gt;400 and &gt;1000) across the hex100 bin." width="100%" />
<p class="caption">
Figure 9.20: Spat feature plot showing gene expression for the top 2 ranked spatial genes per expression bin (&gt;200, &gt;400 and &gt;1000) across the hex100 bin.
</p>
</div>
</div>
<div id="spatial-co-expression-modules-1" class="section level4 hasAnchor" number="9.5.2.2">
<h4><span class="header-section-number">9.5.2.2</span> Spatial co-expression modules<a href="visium-hd.html#spatial-co-expression-modules-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Investigating individual genes is a good start, but here we would like to identify recurrent spatial expression patterns that are shared by spatial co-expression modules that might represent spatially organized biological processes.</p>
<div class="sourceCode" id="cb235"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb235-1"><a href="visium-hd.html#cb235-1" tabindex="-1"></a>ext_spatial_genes <span class="ot">=</span> ranktest[adj.p.value <span class="sc">&lt;</span> <span class="fl">0.001</span>]<span class="sc">$</span>feats</span>
<span id="cb235-2"><a href="visium-hd.html#cb235-2" tabindex="-1"></a></span>
<span id="cb235-3"><a href="visium-hd.html#cb235-3" tabindex="-1"></a>spat_cor_netw_DT <span class="ot">=</span> <span class="fu">detectSpatialCorFeats</span>(visiumHD,</span>
<span id="cb235-4"><a href="visium-hd.html#cb235-4" tabindex="-1"></a>                                         <span class="at">method =</span> <span class="st">&#39;network&#39;</span>,</span>
<span id="cb235-5"><a href="visium-hd.html#cb235-5" tabindex="-1"></a>                                         <span class="at">spatial_network_name =</span> <span class="st">&#39;kNN_network&#39;</span>,</span>
<span id="cb235-6"><a href="visium-hd.html#cb235-6" tabindex="-1"></a>                                         <span class="at">subset_feats =</span> ext_spatial_genes)</span>
<span id="cb235-7"><a href="visium-hd.html#cb235-7" tabindex="-1"></a></span>
<span id="cb235-8"><a href="visium-hd.html#cb235-8" tabindex="-1"></a><span class="co"># cluster spatial genes</span></span>
<span id="cb235-9"><a href="visium-hd.html#cb235-9" tabindex="-1"></a>spat_cor_netw_DT <span class="ot">=</span> <span class="fu">clusterSpatialCorFeats</span>(spat_cor_netw_DT, </span>
<span id="cb235-10"><a href="visium-hd.html#cb235-10" tabindex="-1"></a>                                          <span class="at">name =</span> <span class="st">&#39;spat_netw_clus&#39;</span>, </span>
<span id="cb235-11"><a href="visium-hd.html#cb235-11" tabindex="-1"></a>                                          <span class="at">k =</span> <span class="dv">16</span>)</span>
<span id="cb235-12"><a href="visium-hd.html#cb235-12" tabindex="-1"></a></span>
<span id="cb235-13"><a href="visium-hd.html#cb235-13" tabindex="-1"></a><span class="co"># visualize clusters</span></span>
<span id="cb235-14"><a href="visium-hd.html#cb235-14" tabindex="-1"></a><span class="fu">heatmSpatialCorFeats</span>(visiumHD,</span>
<span id="cb235-15"><a href="visium-hd.html#cb235-15" tabindex="-1"></a>                     <span class="at">spatCorObject =</span> spat_cor_netw_DT,</span>
<span id="cb235-16"><a href="visium-hd.html#cb235-16" tabindex="-1"></a>                     <span class="at">use_clus_name =</span> <span class="st">&#39;spat_netw_clus&#39;</span>,</span>
<span id="cb235-17"><a href="visium-hd.html#cb235-17" tabindex="-1"></a>                     <span class="at">heatmap_legend_param =</span> <span class="fu">list</span>(<span class="at">title =</span> <span class="cn">NULL</span>))</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-317"></span>
<img src="img/02_session2/16-heatmSpatialCorFeats.png" alt="Heatmap showing spatially correlated genes split into 16 clusters." width="100%" />
<p class="caption">
Figure 9.21: Heatmap showing spatially correlated genes split into 16 clusters.
</p>
</div>
<div class="sourceCode" id="cb236"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb236-1"><a href="visium-hd.html#cb236-1" tabindex="-1"></a><span class="co"># create metagene enrichment score for clusters</span></span>
<span id="cb236-2"><a href="visium-hd.html#cb236-2" tabindex="-1"></a>cluster_genes_DT <span class="ot">=</span> <span class="fu">showSpatialCorFeats</span>(spat_cor_netw_DT, </span>
<span id="cb236-3"><a href="visium-hd.html#cb236-3" tabindex="-1"></a>                                       <span class="at">use_clus_name =</span> <span class="st">&#39;spat_netw_clus&#39;</span>, </span>
<span id="cb236-4"><a href="visium-hd.html#cb236-4" tabindex="-1"></a>                                       <span class="at">show_top_feats =</span> <span class="dv">1</span>)</span>
<span id="cb236-5"><a href="visium-hd.html#cb236-5" tabindex="-1"></a>cluster_genes <span class="ot">=</span> cluster_genes_DT<span class="sc">$</span>clus; <span class="fu">names</span>(cluster_genes) <span class="ot">=</span> cluster_genes_DT<span class="sc">$</span>feat_ID</span>
<span id="cb236-6"><a href="visium-hd.html#cb236-6" tabindex="-1"></a></span>
<span id="cb236-7"><a href="visium-hd.html#cb236-7" tabindex="-1"></a>visiumHD <span class="ot">=</span> <span class="fu">createMetafeats</span>(visiumHD,</span>
<span id="cb236-8"><a href="visium-hd.html#cb236-8" tabindex="-1"></a>                           <span class="at">expression_values =</span> <span class="st">&#39;normalized&#39;</span>,</span>
<span id="cb236-9"><a href="visium-hd.html#cb236-9" tabindex="-1"></a>                           <span class="at">feat_clusters =</span> cluster_genes,</span>
<span id="cb236-10"><a href="visium-hd.html#cb236-10" tabindex="-1"></a>                           <span class="at">name =</span> <span class="st">&#39;cluster_metagene&#39;</span>)</span>
<span id="cb236-11"><a href="visium-hd.html#cb236-11" tabindex="-1"></a></span>
<span id="cb236-12"><a href="visium-hd.html#cb236-12" tabindex="-1"></a><span class="fu">showGiottoSpatEnrichments</span>(visiumHD)</span></code></pre></div>
<div class="sourceCode" id="cb237"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb237-1"><a href="visium-hd.html#cb237-1" tabindex="-1"></a><span class="fu">spatCellPlot</span>(visiumHD,</span>
<span id="cb237-2"><a href="visium-hd.html#cb237-2" tabindex="-1"></a>             <span class="at">spat_enr_names =</span> <span class="st">&#39;cluster_metagene&#39;</span>,</span>
<span id="cb237-3"><a href="visium-hd.html#cb237-3" tabindex="-1"></a>             <span class="at">gradient_style =</span> <span class="st">&quot;sequential&quot;</span>,</span>
<span id="cb237-4"><a href="visium-hd.html#cb237-4" tabindex="-1"></a>             <span class="at">cell_color_gradient =</span> <span class="fu">c</span>(<span class="st">&#39;blue&#39;</span>, <span class="st">&#39;white&#39;</span>, <span class="st">&#39;yellow&#39;</span>, <span class="st">&#39;orange&#39;</span>, <span class="st">&#39;red&#39;</span>, <span class="st">&#39;darkred&#39;</span>),</span>
<span id="cb237-5"><a href="visium-hd.html#cb237-5" tabindex="-1"></a>             <span class="at">cell_annotation_values =</span> <span class="fu">as.character</span>(<span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>)),</span>
<span id="cb237-6"><a href="visium-hd.html#cb237-6" tabindex="-1"></a>             <span class="at">point_size =</span> <span class="dv">1</span>, <span class="at">cow_n_col =</span> <span class="dv">2</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-320"></span>
<img src="img/02_session2/17-spatCellPlot2D.png" alt="Spat plot vizualizing metagenes (1-4) based on spatially correlated genes vizualized on the hex100 bin" width="100%" />
<p class="caption">
Figure 9.22: Spat plot vizualizing metagenes (1-4) based on spatially correlated genes vizualized on the hex100 bin
</p>
</div>
<div class="sourceCode" id="cb238"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb238-1"><a href="visium-hd.html#cb238-1" tabindex="-1"></a><span class="fu">spatCellPlot</span>(visiumHD,</span>
<span id="cb238-2"><a href="visium-hd.html#cb238-2" tabindex="-1"></a>             <span class="at">spat_enr_names =</span> <span class="st">&#39;cluster_metagene&#39;</span>,</span>
<span id="cb238-3"><a href="visium-hd.html#cb238-3" tabindex="-1"></a>             <span class="at">gradient_style =</span> <span class="st">&quot;sequential&quot;</span>,</span>
<span id="cb238-4"><a href="visium-hd.html#cb238-4" tabindex="-1"></a>             <span class="at">cell_color_gradient =</span> <span class="fu">c</span>(<span class="st">&#39;blue&#39;</span>, <span class="st">&#39;white&#39;</span>, <span class="st">&#39;yellow&#39;</span>, <span class="st">&#39;orange&#39;</span>, <span class="st">&#39;red&#39;</span>, <span class="st">&#39;darkred&#39;</span>),</span>
<span id="cb238-5"><a href="visium-hd.html#cb238-5" tabindex="-1"></a>             <span class="at">cell_annotation_values =</span> <span class="fu">as.character</span>(<span class="fu">c</span>(<span class="dv">5</span><span class="sc">:</span><span class="dv">8</span>)),</span>
<span id="cb238-6"><a href="visium-hd.html#cb238-6" tabindex="-1"></a>             <span class="at">point_size =</span> <span class="dv">1</span>, <span class="at">cow_n_col =</span> <span class="dv">2</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-322"></span>
<img src="img/02_session2/18-spatCellPlot2D.png" alt="Spat plot vizualizing metagenes (5-8) based on spatially correlated genes vizualized on the hex100 bin" width="100%" />
<p class="caption">
Figure 9.23: Spat plot vizualizing metagenes (5-8) based on spatially correlated genes vizualized on the hex100 bin
</p>
</div>
<div class="sourceCode" id="cb239"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb239-1"><a href="visium-hd.html#cb239-1" tabindex="-1"></a><span class="fu">spatCellPlot</span>(visiumHD,</span>
<span id="cb239-2"><a href="visium-hd.html#cb239-2" tabindex="-1"></a>             <span class="at">spat_enr_names =</span> <span class="st">&#39;cluster_metagene&#39;</span>,</span>
<span id="cb239-3"><a href="visium-hd.html#cb239-3" tabindex="-1"></a>             <span class="at">gradient_style =</span> <span class="st">&quot;sequential&quot;</span>,</span>
<span id="cb239-4"><a href="visium-hd.html#cb239-4" tabindex="-1"></a>             <span class="at">cell_color_gradient =</span> <span class="fu">c</span>(<span class="st">&#39;blue&#39;</span>, <span class="st">&#39;white&#39;</span>, <span class="st">&#39;yellow&#39;</span>, <span class="st">&#39;orange&#39;</span>, <span class="st">&#39;red&#39;</span>, <span class="st">&#39;darkred&#39;</span>),</span>
<span id="cb239-5"><a href="visium-hd.html#cb239-5" tabindex="-1"></a>             <span class="at">cell_annotation_values =</span> <span class="fu">as.character</span>(<span class="fu">c</span>(<span class="dv">9</span><span class="sc">:</span><span class="dv">12</span>)),</span>
<span id="cb239-6"><a href="visium-hd.html#cb239-6" tabindex="-1"></a>             <span class="at">point_size =</span> <span class="dv">1</span>, <span class="at">cow_n_col =</span> <span class="dv">2</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-324"></span>
<img src="img/02_session2/19-spatCellPlot2D.png" alt="Spat plot vizualizing metagenes (9-12) based on spatially correlated genes vizualized on the hex100 bin" width="100%" />
<p class="caption">
Figure 9.24: Spat plot vizualizing metagenes (9-12) based on spatially correlated genes vizualized on the hex100 bin
</p>
</div>
<div class="sourceCode" id="cb240"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb240-1"><a href="visium-hd.html#cb240-1" tabindex="-1"></a><span class="fu">spatCellPlot</span>(visiumHD,</span>
<span id="cb240-2"><a href="visium-hd.html#cb240-2" tabindex="-1"></a>             <span class="at">spat_enr_names =</span> <span class="st">&#39;cluster_metagene&#39;</span>,</span>
<span id="cb240-3"><a href="visium-hd.html#cb240-3" tabindex="-1"></a>             <span class="at">gradient_style =</span> <span class="st">&quot;sequential&quot;</span>,</span>
<span id="cb240-4"><a href="visium-hd.html#cb240-4" tabindex="-1"></a>             <span class="at">cell_color_gradient =</span> <span class="fu">c</span>(<span class="st">&#39;blue&#39;</span>, <span class="st">&#39;white&#39;</span>, <span class="st">&#39;yellow&#39;</span>, <span class="st">&#39;orange&#39;</span>, <span class="st">&#39;red&#39;</span>, <span class="st">&#39;darkred&#39;</span>),</span>
<span id="cb240-5"><a href="visium-hd.html#cb240-5" tabindex="-1"></a>             <span class="at">cell_annotation_values =</span> <span class="fu">as.character</span>(<span class="fu">c</span>(<span class="dv">13</span><span class="sc">:</span><span class="dv">16</span>)),</span>
<span id="cb240-6"><a href="visium-hd.html#cb240-6" tabindex="-1"></a>             <span class="at">point_size =</span> <span class="dv">1</span>, <span class="at">cow_n_col =</span> <span class="dv">2</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-326"></span>
<img src="img/02_session2/20-spatCellPlot2D.png" alt="Spat plot vizualizing metagenes (13-16) based on spatially correlated genes vizualized on the hex100 bin" width="100%" />
<p class="caption">
Figure 9.25: Spat plot vizualizing metagenes (13-16) based on spatially correlated genes vizualized on the hex100 bin
</p>
</div>
<p>A simple follow up analysis could be to perform gene set enrichment analysis on each spatial co-expression module.</p>
</div>
<div id="plot-spatial-gene-groups" class="section level4 hasAnchor" number="9.5.2.3">
<h4><span class="header-section-number">9.5.2.3</span> Plot spatial gene groups<a href="visium-hd.html#plot-spatial-gene-groups" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Hack! Vendors of spatial technologies typically like to show very interesting spatial gene expression patterns. Here we will follow a similar strategy by selecting a balanced set of genes for each spatial co-expression module and then to simply give them the same color in the <code>spatInSituPlotPoints</code> function.</p>
<div class="sourceCode" id="cb241"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb241-1"><a href="visium-hd.html#cb241-1" tabindex="-1"></a>balanced_genes <span class="ot">=</span> <span class="fu">getBalancedSpatCoexpressionFeats</span>(<span class="at">spatCorObject =</span> spat_cor_netw_DT, </span>
<span id="cb241-2"><a href="visium-hd.html#cb241-2" tabindex="-1"></a>                                                  <span class="at">maximum =</span> <span class="dv">5</span>)</span>
<span id="cb241-3"><a href="visium-hd.html#cb241-3" tabindex="-1"></a>selected_feats <span class="ot">=</span> <span class="fu">names</span>(balanced_genes)</span>
<span id="cb241-4"><a href="visium-hd.html#cb241-4" tabindex="-1"></a></span>
<span id="cb241-5"><a href="visium-hd.html#cb241-5" tabindex="-1"></a><span class="co"># give genes from same cluster same color</span></span>
<span id="cb241-6"><a href="visium-hd.html#cb241-6" tabindex="-1"></a>distinct_colors <span class="ot">=</span> <span class="fu">getDistinctColors</span>(<span class="at">n =</span> <span class="dv">20</span>)</span>
<span id="cb241-7"><a href="visium-hd.html#cb241-7" tabindex="-1"></a><span class="fu">names</span>(distinct_colors) <span class="ot">=</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">20</span></span>
<span id="cb241-8"><a href="visium-hd.html#cb241-8" tabindex="-1"></a></span>
<span id="cb241-9"><a href="visium-hd.html#cb241-9" tabindex="-1"></a>my_colors <span class="ot">=</span> distinct_colors[balanced_genes]</span>
<span id="cb241-10"><a href="visium-hd.html#cb241-10" tabindex="-1"></a><span class="fu">names</span>(my_colors) <span class="ot">=</span> <span class="fu">names</span>(balanced_genes)</span>
<span id="cb241-11"><a href="visium-hd.html#cb241-11" tabindex="-1"></a></span>
<span id="cb241-12"><a href="visium-hd.html#cb241-12" tabindex="-1"></a><span class="fu">spatInSituPlotPoints</span>(visiumHD,</span>
<span id="cb241-13"><a href="visium-hd.html#cb241-13" tabindex="-1"></a>                     <span class="at">show_image =</span> F,</span>
<span id="cb241-14"><a href="visium-hd.html#cb241-14" tabindex="-1"></a>                     <span class="at">feats =</span> <span class="fu">list</span>(<span class="st">&#39;rna&#39;</span> <span class="ot">=</span> selected_feats), </span>
<span id="cb241-15"><a href="visium-hd.html#cb241-15" tabindex="-1"></a>                     <span class="at">feats_color_code =</span> my_colors,</span>
<span id="cb241-16"><a href="visium-hd.html#cb241-16" tabindex="-1"></a>                     <span class="at">show_legend =</span> F,</span>
<span id="cb241-17"><a href="visium-hd.html#cb241-17" tabindex="-1"></a>                     <span class="at">spat_unit =</span> <span class="st">&#39;hex100&#39;</span>,</span>
<span id="cb241-18"><a href="visium-hd.html#cb241-18" tabindex="-1"></a>                     <span class="at">point_size =</span> <span class="fl">0.20</span>,</span>
<span id="cb241-19"><a href="visium-hd.html#cb241-19" tabindex="-1"></a>                     <span class="at">show_polygon =</span> <span class="cn">FALSE</span>,</span>
<span id="cb241-20"><a href="visium-hd.html#cb241-20" tabindex="-1"></a>                     <span class="at">use_overlap =</span> <span class="cn">FALSE</span>,</span>
<span id="cb241-21"><a href="visium-hd.html#cb241-21" tabindex="-1"></a>                     <span class="at">polygon_feat_type =</span> <span class="st">&#39;hex100&#39;</span>,</span>
<span id="cb241-22"><a href="visium-hd.html#cb241-22" tabindex="-1"></a>                     <span class="at">polygon_bg_color =</span> <span class="cn">NA</span>,</span>
<span id="cb241-23"><a href="visium-hd.html#cb241-23" tabindex="-1"></a>                     <span class="at">polygon_color =</span> <span class="st">&#39;white&#39;</span>,</span>
<span id="cb241-24"><a href="visium-hd.html#cb241-24" tabindex="-1"></a>                     <span class="at">polygon_line_size =</span> <span class="fl">0.01</span>,</span>
<span id="cb241-25"><a href="visium-hd.html#cb241-25" tabindex="-1"></a>                     <span class="at">expand_counts =</span> <span class="cn">TRUE</span>, </span>
<span id="cb241-26"><a href="visium-hd.html#cb241-26" tabindex="-1"></a>                     <span class="at">count_info_column =</span> <span class="st">&#39;count&#39;</span>,</span>
<span id="cb241-27"><a href="visium-hd.html#cb241-27" tabindex="-1"></a>                     <span class="at">jitter =</span> <span class="fu">c</span>(<span class="dv">25</span>,<span class="dv">25</span>))</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-328"></span>
<img src="img/02_session2/21-spatInSituPlotPoints.png" alt="Coloring individual features based on the spatially correlated gene clusters." width="100%" />
<p class="caption">
Figure 9.26: Coloring individual features based on the spatially correlated gene clusters.
</p>
</div>
</div>
</div>
</div>
<div id="hexbin-25" class="section level2 hasAnchor" number="9.6">
<h2><span class="header-section-number">9.6</span> Hexbin 25<a href="visium-hd.html#hexbin-25" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Goal is to create a higher resolution bin (hex25) and add to the Giotto object. We will aim to identify individual cell types and local neighborhood niches.</p>
<div id="subcellular-workflow" class="section level3 hasAnchor" number="9.6.1">
<h3><span class="header-section-number">9.6.1</span> Subcellular workflow<a href="visium-hd.html#subcellular-workflow" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li>filter and normalization workflow</li>
</ul>
<div class="sourceCode" id="cb242"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb242-1"><a href="visium-hd.html#cb242-1" tabindex="-1"></a>visiumHD_subset <span class="ot">=</span> <span class="fu">subsetGiottoLocs</span>(<span class="at">gobject =</span> visiumHD,</span>
<span id="cb242-2"><a href="visium-hd.html#cb242-2" tabindex="-1"></a>                                   <span class="at">x_min =</span> <span class="dv">16000</span>, </span>
<span id="cb242-3"><a href="visium-hd.html#cb242-3" tabindex="-1"></a>                                   <span class="at">x_max =</span> <span class="dv">20000</span>, </span>
<span id="cb242-4"><a href="visium-hd.html#cb242-4" tabindex="-1"></a>                                   <span class="at">y_min =</span> <span class="dv">44250</span>,</span>
<span id="cb242-5"><a href="visium-hd.html#cb242-5" tabindex="-1"></a>                                   <span class="at">y_max =</span> <span class="dv">45500</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-330"></span>
<img src="img/02_session2/9.png" alt="Coloring individual features based on the spatially correlated gene clusters + subset rectangle." width="100%" />
<p class="caption">
Figure 9.27: Coloring individual features based on the spatially correlated gene clusters + subset rectangle.
</p>
</div>
<p>Plot visiumHD subset with hexbin100 polygons:</p>
<div class="sourceCode" id="cb243"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb243-1"><a href="visium-hd.html#cb243-1" tabindex="-1"></a><span class="fu">spatInSituPlotPoints</span>(visiumHD_subset,</span>
<span id="cb243-2"><a href="visium-hd.html#cb243-2" tabindex="-1"></a>                     <span class="at">show_image =</span> F,</span>
<span id="cb243-3"><a href="visium-hd.html#cb243-3" tabindex="-1"></a>                     <span class="at">feats =</span> <span class="cn">NULL</span>,</span>
<span id="cb243-4"><a href="visium-hd.html#cb243-4" tabindex="-1"></a>                     <span class="at">show_legend =</span> F,</span>
<span id="cb243-5"><a href="visium-hd.html#cb243-5" tabindex="-1"></a>                     <span class="at">spat_unit =</span> <span class="st">&#39;hex100&#39;</span>,</span>
<span id="cb243-6"><a href="visium-hd.html#cb243-6" tabindex="-1"></a>                     <span class="at">point_size =</span> <span class="fl">0.5</span>,</span>
<span id="cb243-7"><a href="visium-hd.html#cb243-7" tabindex="-1"></a>                     <span class="at">show_polygon =</span> <span class="cn">TRUE</span>,</span>
<span id="cb243-8"><a href="visium-hd.html#cb243-8" tabindex="-1"></a>                     <span class="at">use_overlap =</span> <span class="cn">FALSE</span>,</span>
<span id="cb243-9"><a href="visium-hd.html#cb243-9" tabindex="-1"></a>                     <span class="at">polygon_feat_type =</span> <span class="st">&#39;hex100&#39;</span>,</span>
<span id="cb243-10"><a href="visium-hd.html#cb243-10" tabindex="-1"></a>                     <span class="at">polygon_fill_as_factor =</span> <span class="cn">TRUE</span>,</span>
<span id="cb243-11"><a href="visium-hd.html#cb243-11" tabindex="-1"></a>                     <span class="at">polygon_fill =</span> <span class="st">&#39;leiden_clus&#39;</span>,</span>
<span id="cb243-12"><a href="visium-hd.html#cb243-12" tabindex="-1"></a>                     <span class="at">polygon_color =</span> <span class="st">&#39;black&#39;</span>,</span>
<span id="cb243-13"><a href="visium-hd.html#cb243-13" tabindex="-1"></a>                     <span class="at">polygon_line_size =</span> <span class="fl">0.3</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-332"></span>
<img src="img/02_session2/22-spatInSituPlotPoints.png" alt="Hexbin100 colored by leiden clustering results" width="100%" />
<p class="caption">
Figure 9.28: Hexbin100 colored by leiden clustering results
</p>
</div>
<p>Plot visiumHD subset with selected gene features:</p>
<div class="sourceCode" id="cb244"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb244-1"><a href="visium-hd.html#cb244-1" tabindex="-1"></a><span class="fu">spatInSituPlotPoints</span>(visiumHD_subset,</span>
<span id="cb244-2"><a href="visium-hd.html#cb244-2" tabindex="-1"></a>                     <span class="at">show_image =</span> F,</span>
<span id="cb244-3"><a href="visium-hd.html#cb244-3" tabindex="-1"></a>                     <span class="at">feats =</span> <span class="fu">list</span>(<span class="st">&#39;rna&#39;</span> <span class="ot">=</span> selected_feats), </span>
<span id="cb244-4"><a href="visium-hd.html#cb244-4" tabindex="-1"></a>                     <span class="at">feats_color_code =</span> my_colors,</span>
<span id="cb244-5"><a href="visium-hd.html#cb244-5" tabindex="-1"></a>                     <span class="at">show_legend =</span> F,</span>
<span id="cb244-6"><a href="visium-hd.html#cb244-6" tabindex="-1"></a>                     <span class="at">spat_unit =</span> <span class="st">&#39;hex100&#39;</span>,</span>
<span id="cb244-7"><a href="visium-hd.html#cb244-7" tabindex="-1"></a>                     <span class="at">point_size =</span> <span class="fl">0.40</span>,</span>
<span id="cb244-8"><a href="visium-hd.html#cb244-8" tabindex="-1"></a>                     <span class="at">show_polygon =</span> <span class="cn">TRUE</span>,</span>
<span id="cb244-9"><a href="visium-hd.html#cb244-9" tabindex="-1"></a>                     <span class="at">use_overlap =</span> <span class="cn">FALSE</span>,</span>
<span id="cb244-10"><a href="visium-hd.html#cb244-10" tabindex="-1"></a>                     <span class="at">polygon_feat_type =</span> <span class="st">&#39;hex100&#39;</span>,</span>
<span id="cb244-11"><a href="visium-hd.html#cb244-11" tabindex="-1"></a>                     <span class="at">polygon_bg_color =</span> <span class="cn">NA</span>,</span>
<span id="cb244-12"><a href="visium-hd.html#cb244-12" tabindex="-1"></a>                     <span class="at">polygon_color =</span> <span class="st">&#39;white&#39;</span>,</span>
<span id="cb244-13"><a href="visium-hd.html#cb244-13" tabindex="-1"></a>                     <span class="at">polygon_line_size =</span> <span class="fl">0.05</span>,</span>
<span id="cb244-14"><a href="visium-hd.html#cb244-14" tabindex="-1"></a>                     <span class="at">jitter =</span> <span class="fu">c</span>(<span class="dv">25</span>,<span class="dv">25</span>))</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-334"></span>
<img src="img/02_session2/23-spatInSituPlotPoints.png" alt="Coloring individual features based on the spatially correlated gene clusters" width="100%" />
<p class="caption">
Figure 9.29: Coloring individual features based on the spatially correlated gene clusters
</p>
</div>
<p>Create smaller hexbin25 tessellations:</p>
<div class="sourceCode" id="cb245"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb245-1"><a href="visium-hd.html#cb245-1" tabindex="-1"></a>hexbin25 <span class="ot">&lt;-</span> <span class="fu">tessellate</span>(<span class="at">extent =</span> <span class="fu">ext</span>(visiumHD_subset<span class="sc">@</span>feat_info<span class="sc">$</span>rna), </span>
<span id="cb245-2"><a href="visium-hd.html#cb245-2" tabindex="-1"></a>                       <span class="at">shape =</span> <span class="st">&#39;hexagon&#39;</span>, </span>
<span id="cb245-3"><a href="visium-hd.html#cb245-3" tabindex="-1"></a>                       <span class="at">shape_size =</span> <span class="dv">25</span>, </span>
<span id="cb245-4"><a href="visium-hd.html#cb245-4" tabindex="-1"></a>                       <span class="at">name =</span> <span class="st">&#39;hex25&#39;</span>) </span>
<span id="cb245-5"><a href="visium-hd.html#cb245-5" tabindex="-1"></a></span>
<span id="cb245-6"><a href="visium-hd.html#cb245-6" tabindex="-1"></a>visiumHD_subset <span class="ot">=</span> <span class="fu">setPolygonInfo</span>(<span class="at">gobject =</span> visiumHD_subset,</span>
<span id="cb245-7"><a href="visium-hd.html#cb245-7" tabindex="-1"></a>                                 <span class="at">x =</span> hexbin25,</span>
<span id="cb245-8"><a href="visium-hd.html#cb245-8" tabindex="-1"></a>                                 <span class="at">name =</span> <span class="st">&#39;hex25&#39;</span>,</span>
<span id="cb245-9"><a href="visium-hd.html#cb245-9" tabindex="-1"></a>                                 <span class="at">initialize =</span> T)</span>
<span id="cb245-10"><a href="visium-hd.html#cb245-10" tabindex="-1"></a></span>
<span id="cb245-11"><a href="visium-hd.html#cb245-11" tabindex="-1"></a><span class="fu">showGiottoSpatialInfo</span>(visiumHD_subset)</span>
<span id="cb245-12"><a href="visium-hd.html#cb245-12" tabindex="-1"></a></span>
<span id="cb245-13"><a href="visium-hd.html#cb245-13" tabindex="-1"></a>visiumHD_subset <span class="ot">=</span> <span class="fu">addSpatialCentroidLocations</span>(<span class="at">gobject =</span> visiumHD_subset,</span>
<span id="cb245-14"><a href="visium-hd.html#cb245-14" tabindex="-1"></a>                                              <span class="at">poly_info =</span> <span class="st">&#39;hex25&#39;</span>)</span>
<span id="cb245-15"><a href="visium-hd.html#cb245-15" tabindex="-1"></a></span>
<span id="cb245-16"><a href="visium-hd.html#cb245-16" tabindex="-1"></a><span class="fu">activeSpatUnit</span>(visiumHD_subset) <span class="ot">&lt;-</span> <span class="st">&#39;hex25&#39;</span></span>
<span id="cb245-17"><a href="visium-hd.html#cb245-17" tabindex="-1"></a></span>
<span id="cb245-18"><a href="visium-hd.html#cb245-18" tabindex="-1"></a><span class="fu">spatInSituPlotPoints</span>(visiumHD_subset,</span>
<span id="cb245-19"><a href="visium-hd.html#cb245-19" tabindex="-1"></a>                     <span class="at">show_image =</span> F,</span>
<span id="cb245-20"><a href="visium-hd.html#cb245-20" tabindex="-1"></a>                     <span class="at">feats =</span> <span class="fu">list</span>(<span class="st">&#39;rna&#39;</span> <span class="ot">=</span> selected_feats), </span>
<span id="cb245-21"><a href="visium-hd.html#cb245-21" tabindex="-1"></a>                     <span class="at">feats_color_code =</span> my_colors,</span>
<span id="cb245-22"><a href="visium-hd.html#cb245-22" tabindex="-1"></a>                     <span class="at">show_legend =</span> F,</span>
<span id="cb245-23"><a href="visium-hd.html#cb245-23" tabindex="-1"></a>                     <span class="at">spat_unit =</span> <span class="st">&#39;hex25&#39;</span>,</span>
<span id="cb245-24"><a href="visium-hd.html#cb245-24" tabindex="-1"></a>                     <span class="at">point_size =</span> <span class="fl">0.40</span>,</span>
<span id="cb245-25"><a href="visium-hd.html#cb245-25" tabindex="-1"></a>                     <span class="at">show_polygon =</span> <span class="cn">TRUE</span>,</span>
<span id="cb245-26"><a href="visium-hd.html#cb245-26" tabindex="-1"></a>                     <span class="at">use_overlap =</span> <span class="cn">FALSE</span>,</span>
<span id="cb245-27"><a href="visium-hd.html#cb245-27" tabindex="-1"></a>                     <span class="at">polygon_feat_type =</span> <span class="st">&#39;hex25&#39;</span>,</span>
<span id="cb245-28"><a href="visium-hd.html#cb245-28" tabindex="-1"></a>                     <span class="at">polygon_bg_color =</span> <span class="cn">NA</span>,</span>
<span id="cb245-29"><a href="visium-hd.html#cb245-29" tabindex="-1"></a>                     <span class="at">polygon_color =</span> <span class="st">&#39;white&#39;</span>,</span>
<span id="cb245-30"><a href="visium-hd.html#cb245-30" tabindex="-1"></a>                     <span class="at">polygon_line_size =</span> <span class="fl">0.05</span>,</span>
<span id="cb245-31"><a href="visium-hd.html#cb245-31" tabindex="-1"></a>                     <span class="at">jitter =</span> <span class="fu">c</span>(<span class="dv">25</span>,<span class="dv">25</span>))</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-336"></span>
<img src="img/02_session2/24-spatInSituPlotPoints.png" alt="xxx" width="100%" />
<p class="caption">
Figure 9.30: xxx
</p>
</div>
<div class="sourceCode" id="cb246"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb246-1"><a href="visium-hd.html#cb246-1" tabindex="-1"></a>visiumHD_subset <span class="ot">=</span> <span class="fu">calculateOverlap</span>(visiumHD_subset,</span>
<span id="cb246-2"><a href="visium-hd.html#cb246-2" tabindex="-1"></a>                                   <span class="at">spatial_info =</span> <span class="st">&#39;hex25&#39;</span>,</span>
<span id="cb246-3"><a href="visium-hd.html#cb246-3" tabindex="-1"></a>                                   <span class="at">feat_info =</span> <span class="st">&#39;rna&#39;</span>)</span>
<span id="cb246-4"><a href="visium-hd.html#cb246-4" tabindex="-1"></a></span>
<span id="cb246-5"><a href="visium-hd.html#cb246-5" tabindex="-1"></a><span class="fu">showGiottoSpatialInfo</span>(visiumHD_subset)</span>
<span id="cb246-6"><a href="visium-hd.html#cb246-6" tabindex="-1"></a></span>
<span id="cb246-7"><a href="visium-hd.html#cb246-7" tabindex="-1"></a></span>
<span id="cb246-8"><a href="visium-hd.html#cb246-8" tabindex="-1"></a><span class="co"># convert overlap results to bin by gene matrix</span></span>
<span id="cb246-9"><a href="visium-hd.html#cb246-9" tabindex="-1"></a>visiumHD_subset <span class="ot">=</span> <span class="fu">overlapToMatrix</span>(visiumHD_subset,</span>
<span id="cb246-10"><a href="visium-hd.html#cb246-10" tabindex="-1"></a>                                  <span class="at">poly_info =</span> <span class="st">&#39;hex25&#39;</span>,</span>
<span id="cb246-11"><a href="visium-hd.html#cb246-11" tabindex="-1"></a>                                  <span class="at">feat_info =</span> <span class="st">&#39;rna&#39;</span>,</span>
<span id="cb246-12"><a href="visium-hd.html#cb246-12" tabindex="-1"></a>                                  <span class="at">name =</span> <span class="st">&#39;raw&#39;</span>)</span>
<span id="cb246-13"><a href="visium-hd.html#cb246-13" tabindex="-1"></a></span>
<span id="cb246-14"><a href="visium-hd.html#cb246-14" tabindex="-1"></a>visiumHD_subset <span class="ot">&lt;-</span> <span class="fu">filterGiotto</span>(visiumHD_subset,</span>
<span id="cb246-15"><a href="visium-hd.html#cb246-15" tabindex="-1"></a>                                <span class="at">expression_threshold =</span> <span class="dv">1</span>,</span>
<span id="cb246-16"><a href="visium-hd.html#cb246-16" tabindex="-1"></a>                                <span class="at">feat_det_in_min_cells =</span> <span class="dv">3</span>,</span>
<span id="cb246-17"><a href="visium-hd.html#cb246-17" tabindex="-1"></a>                                <span class="at">min_det_feats_per_cell =</span> <span class="dv">5</span>)</span>
<span id="cb246-18"><a href="visium-hd.html#cb246-18" tabindex="-1"></a></span>
<span id="cb246-19"><a href="visium-hd.html#cb246-19" tabindex="-1"></a><span class="fu">activeSpatUnit</span>(visiumHD_subset)</span>
<span id="cb246-20"><a href="visium-hd.html#cb246-20" tabindex="-1"></a></span>
<span id="cb246-21"><a href="visium-hd.html#cb246-21" tabindex="-1"></a></span>
<span id="cb246-22"><a href="visium-hd.html#cb246-22" tabindex="-1"></a><span class="co"># normalize</span></span>
<span id="cb246-23"><a href="visium-hd.html#cb246-23" tabindex="-1"></a>visiumHD_subset <span class="ot">&lt;-</span> <span class="fu">normalizeGiotto</span>(visiumHD_subset, <span class="at">scalefactor =</span> <span class="dv">1000</span>, <span class="at">verbose =</span> T)</span>
<span id="cb246-24"><a href="visium-hd.html#cb246-24" tabindex="-1"></a></span>
<span id="cb246-25"><a href="visium-hd.html#cb246-25" tabindex="-1"></a><span class="co"># add statistics</span></span>
<span id="cb246-26"><a href="visium-hd.html#cb246-26" tabindex="-1"></a>visiumHD_subset <span class="ot">&lt;-</span> <span class="fu">addStatistics</span>(visiumHD_subset)</span>
<span id="cb246-27"><a href="visium-hd.html#cb246-27" tabindex="-1"></a></span>
<span id="cb246-28"><a href="visium-hd.html#cb246-28" tabindex="-1"></a>feature_data <span class="ot">=</span> <span class="fu">fDataDT</span>(visiumHD_subset)</span>
<span id="cb246-29"><a href="visium-hd.html#cb246-29" tabindex="-1"></a></span>
<span id="cb246-30"><a href="visium-hd.html#cb246-30" tabindex="-1"></a>visiumHD_subset <span class="ot">&lt;-</span> <span class="fu">calculateHVF</span>(visiumHD_subset, <span class="at">zscore_threshold =</span> <span class="dv">1</span>)</span></code></pre></div>
</div>
<div id="projections" class="section level3 hasAnchor" number="9.6.2">
<h3><span class="header-section-number">9.6.2</span> Projections<a href="visium-hd.html#projections" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li>PCA projection from random subset.<br />
</li>
<li>UMAP projection from random subset.</li>
<li>cluster result projection from subsampled Giotto object + kNN voting</li>
</ul>
<div id="pca-with-projection" class="section level4 hasAnchor" number="9.6.2.1">
<h4><span class="header-section-number">9.6.2.1</span> PCA with projection<a href="visium-hd.html#pca-with-projection" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb247"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb247-1"><a href="visium-hd.html#cb247-1" tabindex="-1"></a>n_25_percent <span class="ot">&lt;-</span> <span class="fu">round</span>(<span class="fu">length</span>(<span class="fu">spatIDs</span>(visiumHD_subset, <span class="st">&#39;hex25&#39;</span>)) <span class="sc">*</span> <span class="fl">0.25</span>)</span>
<span id="cb247-2"><a href="visium-hd.html#cb247-2" tabindex="-1"></a></span>
<span id="cb247-3"><a href="visium-hd.html#cb247-3" tabindex="-1"></a><span class="co"># pca projection on subset</span></span>
<span id="cb247-4"><a href="visium-hd.html#cb247-4" tabindex="-1"></a>visiumHD_subset <span class="ot">&lt;-</span> <span class="fu">runPCAprojection</span>(</span>
<span id="cb247-5"><a href="visium-hd.html#cb247-5" tabindex="-1"></a>  <span class="at">gobject =</span> visiumHD_subset,</span>
<span id="cb247-6"><a href="visium-hd.html#cb247-6" tabindex="-1"></a>  <span class="at">spat_unit =</span> <span class="st">&quot;hex25&quot;</span>,</span>
<span id="cb247-7"><a href="visium-hd.html#cb247-7" tabindex="-1"></a>  <span class="at">feats_to_use =</span> <span class="st">&#39;hvf&#39;</span>,</span>
<span id="cb247-8"><a href="visium-hd.html#cb247-8" tabindex="-1"></a>  <span class="at">name =</span> <span class="st">&#39;pca.projection&#39;</span>,</span>
<span id="cb247-9"><a href="visium-hd.html#cb247-9" tabindex="-1"></a>  <span class="at">set_seed =</span> <span class="cn">TRUE</span>,</span>
<span id="cb247-10"><a href="visium-hd.html#cb247-10" tabindex="-1"></a>  <span class="at">seed_number =</span> <span class="dv">12345</span>,</span>
<span id="cb247-11"><a href="visium-hd.html#cb247-11" tabindex="-1"></a>  <span class="at">random_subset =</span> n_25_percent</span>
<span id="cb247-12"><a href="visium-hd.html#cb247-12" tabindex="-1"></a>)</span>
<span id="cb247-13"><a href="visium-hd.html#cb247-13" tabindex="-1"></a></span>
<span id="cb247-14"><a href="visium-hd.html#cb247-14" tabindex="-1"></a><span class="fu">showGiottoDimRed</span>(visiumHD_subset)</span>
<span id="cb247-15"><a href="visium-hd.html#cb247-15" tabindex="-1"></a><span class="fu">plotPCA</span>(visiumHD_subset, <span class="at">dim_reduction_name =</span> <span class="st">&#39;pca.projection&#39;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-339"></span>
<img src="img/02_session2/25-PCA.png" alt="xxx" width="100%" />
<p class="caption">
Figure 9.31: xxx
</p>
</div>
</div>
<div id="umap-with-projection" class="section level4 hasAnchor" number="9.6.2.2">
<h4><span class="header-section-number">9.6.2.2</span> UMAP with projection<a href="visium-hd.html#umap-with-projection" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb248"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb248-1"><a href="visium-hd.html#cb248-1" tabindex="-1"></a><span class="co"># umap projection on subset</span></span>
<span id="cb248-2"><a href="visium-hd.html#cb248-2" tabindex="-1"></a>visiumHD_subset <span class="ot">&lt;-</span> <span class="fu">runUMAPprojection</span>(</span>
<span id="cb248-3"><a href="visium-hd.html#cb248-3" tabindex="-1"></a>  <span class="at">gobject =</span> visiumHD_subset,</span>
<span id="cb248-4"><a href="visium-hd.html#cb248-4" tabindex="-1"></a>  <span class="at">spat_unit =</span> <span class="st">&quot;hex25&quot;</span>,</span>
<span id="cb248-5"><a href="visium-hd.html#cb248-5" tabindex="-1"></a>  <span class="at">dim_reduction_to_use =</span> <span class="st">&#39;pca&#39;</span>,</span>
<span id="cb248-6"><a href="visium-hd.html#cb248-6" tabindex="-1"></a>  <span class="at">dim_reduction_name =</span> <span class="st">&quot;pca.projection&quot;</span>,</span>
<span id="cb248-7"><a href="visium-hd.html#cb248-7" tabindex="-1"></a>  <span class="at">dimensions_to_use =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>,</span>
<span id="cb248-8"><a href="visium-hd.html#cb248-8" tabindex="-1"></a>  <span class="at">name =</span> <span class="st">&quot;umap.projection&quot;</span>,</span>
<span id="cb248-9"><a href="visium-hd.html#cb248-9" tabindex="-1"></a>  <span class="at">random_subset =</span> n_25_percent, </span>
<span id="cb248-10"><a href="visium-hd.html#cb248-10" tabindex="-1"></a>  <span class="at">n_neighbors =</span> <span class="dv">10</span>,</span>
<span id="cb248-11"><a href="visium-hd.html#cb248-11" tabindex="-1"></a>  <span class="at">min_dist =</span> <span class="fl">0.005</span>,</span>
<span id="cb248-12"><a href="visium-hd.html#cb248-12" tabindex="-1"></a>  <span class="at">n_threads =</span> <span class="dv">4</span></span>
<span id="cb248-13"><a href="visium-hd.html#cb248-13" tabindex="-1"></a>)</span>
<span id="cb248-14"><a href="visium-hd.html#cb248-14" tabindex="-1"></a></span>
<span id="cb248-15"><a href="visium-hd.html#cb248-15" tabindex="-1"></a><span class="fu">showGiottoDimRed</span>(visiumHD_subset)</span>
<span id="cb248-16"><a href="visium-hd.html#cb248-16" tabindex="-1"></a></span>
<span id="cb248-17"><a href="visium-hd.html#cb248-17" tabindex="-1"></a><span class="co"># plot UMAP, coloring cells/points based on nr_feats</span></span>
<span id="cb248-18"><a href="visium-hd.html#cb248-18" tabindex="-1"></a><span class="fu">plotUMAP</span>(<span class="at">gobject =</span> visiumHD_subset,</span>
<span id="cb248-19"><a href="visium-hd.html#cb248-19" tabindex="-1"></a>         <span class="at">point_size =</span> <span class="dv">1</span>, </span>
<span id="cb248-20"><a href="visium-hd.html#cb248-20" tabindex="-1"></a>         <span class="at">dim_reduction_name =</span> <span class="st">&#39;umap.projection&#39;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-341"></span>
<img src="img/02_session2/26-UMAP.png" alt="xxx" width="100%" />
<p class="caption">
Figure 9.32: xxx
</p>
</div>
</div>
<div id="clustering-with-projection" class="section level4 hasAnchor" number="9.6.2.3">
<h4><span class="header-section-number">9.6.2.3</span> clustering with projection<a href="visium-hd.html#clustering-with-projection" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ol style="list-style-type: decimal">
<li>subsample Giotto object<br />
</li>
<li>perform clustering (e.g. hierarchical clustering)<br />
</li>
<li>project cluster results to full Giotto object using a kNN voting approach and a shared dimension reduction space (e.g. PCA)</li>
</ol>
<div class="sourceCode" id="cb249"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb249-1"><a href="visium-hd.html#cb249-1" tabindex="-1"></a><span class="co"># subset to smaller giotto object</span></span>
<span id="cb249-2"><a href="visium-hd.html#cb249-2" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">1234</span>)</span>
<span id="cb249-3"><a href="visium-hd.html#cb249-3" tabindex="-1"></a>subset_IDs <span class="ot">=</span> <span class="fu">sample</span>(<span class="at">x =</span> <span class="fu">spatIDs</span>(visiumHD_subset, <span class="st">&#39;hex25&#39;</span>), <span class="at">size =</span> n_25_percent)</span>
<span id="cb249-4"><a href="visium-hd.html#cb249-4" tabindex="-1"></a></span>
<span id="cb249-5"><a href="visium-hd.html#cb249-5" tabindex="-1"></a>temp_gobject <span class="ot">=</span> <span class="fu">subsetGiotto</span>(</span>
<span id="cb249-6"><a href="visium-hd.html#cb249-6" tabindex="-1"></a>  <span class="at">gobject =</span> visiumHD_subset,</span>
<span id="cb249-7"><a href="visium-hd.html#cb249-7" tabindex="-1"></a>  <span class="at">spat_unit =</span> <span class="st">&#39;hex25&#39;</span>,</span>
<span id="cb249-8"><a href="visium-hd.html#cb249-8" tabindex="-1"></a>  <span class="at">cell_ids =</span> subset_IDs</span>
<span id="cb249-9"><a href="visium-hd.html#cb249-9" tabindex="-1"></a>)</span>
<span id="cb249-10"><a href="visium-hd.html#cb249-10" tabindex="-1"></a></span>
<span id="cb249-11"><a href="visium-hd.html#cb249-11" tabindex="-1"></a></span>
<span id="cb249-12"><a href="visium-hd.html#cb249-12" tabindex="-1"></a><span class="co"># hierarchical clustering</span></span>
<span id="cb249-13"><a href="visium-hd.html#cb249-13" tabindex="-1"></a>temp_gobject <span class="ot">=</span> <span class="fu">doHclust</span>(<span class="at">gobject =</span> temp_gobject, </span>
<span id="cb249-14"><a href="visium-hd.html#cb249-14" tabindex="-1"></a>                        <span class="at">spat_unit =</span> <span class="st">&#39;hex25&#39;</span>, </span>
<span id="cb249-15"><a href="visium-hd.html#cb249-15" tabindex="-1"></a>                        <span class="at">k =</span> <span class="dv">8</span>, <span class="at">name =</span> <span class="st">&#39;sub_hclust&#39;</span>, </span>
<span id="cb249-16"><a href="visium-hd.html#cb249-16" tabindex="-1"></a>                        <span class="at">dim_reduction_to_use =</span> <span class="st">&#39;pca&#39;</span>, </span>
<span id="cb249-17"><a href="visium-hd.html#cb249-17" tabindex="-1"></a>                        <span class="at">dim_reduction_name =</span> <span class="st">&#39;pca.projection&#39;</span>, </span>
<span id="cb249-18"><a href="visium-hd.html#cb249-18" tabindex="-1"></a>                        <span class="at">dimensions_to_use =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>)</span>
<span id="cb249-19"><a href="visium-hd.html#cb249-19" tabindex="-1"></a></span>
<span id="cb249-20"><a href="visium-hd.html#cb249-20" tabindex="-1"></a></span>
<span id="cb249-21"><a href="visium-hd.html#cb249-21" tabindex="-1"></a><span class="co"># show umap</span></span>
<span id="cb249-22"><a href="visium-hd.html#cb249-22" tabindex="-1"></a><span class="fu">dimPlot2D</span>(</span>
<span id="cb249-23"><a href="visium-hd.html#cb249-23" tabindex="-1"></a>  <span class="at">gobject =</span> temp_gobject,</span>
<span id="cb249-24"><a href="visium-hd.html#cb249-24" tabindex="-1"></a>  <span class="at">point_size =</span> <span class="fl">2.5</span>,</span>
<span id="cb249-25"><a href="visium-hd.html#cb249-25" tabindex="-1"></a>  <span class="at">spat_unit =</span> <span class="st">&#39;hex25&#39;</span>,</span>
<span id="cb249-26"><a href="visium-hd.html#cb249-26" tabindex="-1"></a>  <span class="at">dim_reduction_to_use =</span> <span class="st">&#39;umap&#39;</span>,</span>
<span id="cb249-27"><a href="visium-hd.html#cb249-27" tabindex="-1"></a>  <span class="at">dim_reduction_name =</span> <span class="st">&#39;umap.projection&#39;</span>,</span>
<span id="cb249-28"><a href="visium-hd.html#cb249-28" tabindex="-1"></a>  <span class="at">cell_color =</span> <span class="st">&#39;sub_hclust&#39;</span></span>
<span id="cb249-29"><a href="visium-hd.html#cb249-29" tabindex="-1"></a>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-343"></span>
<img src="img/02_session2/27-dimPlot2D.png" alt="xxx" width="100%" />
<p class="caption">
Figure 9.33: xxx
</p>
</div>
<div class="sourceCode" id="cb250"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb250-1"><a href="visium-hd.html#cb250-1" tabindex="-1"></a><span class="co"># project clusterings back to full dataset</span></span>
<span id="cb250-2"><a href="visium-hd.html#cb250-2" tabindex="-1"></a>visiumHD_subset <span class="ot">&lt;-</span> <span class="fu">doClusterProjection</span>(</span>
<span id="cb250-3"><a href="visium-hd.html#cb250-3" tabindex="-1"></a>  <span class="at">target_gobject =</span> visiumHD_subset,</span>
<span id="cb250-4"><a href="visium-hd.html#cb250-4" tabindex="-1"></a>  <span class="at">source_gobject =</span> temp_gobject,</span>
<span id="cb250-5"><a href="visium-hd.html#cb250-5" tabindex="-1"></a>  <span class="at">spat_unit =</span> <span class="st">&quot;hex25&quot;</span>,</span>
<span id="cb250-6"><a href="visium-hd.html#cb250-6" tabindex="-1"></a>  <span class="at">source_cluster_labels =</span> <span class="st">&quot;sub_hclust&quot;</span>,</span>
<span id="cb250-7"><a href="visium-hd.html#cb250-7" tabindex="-1"></a>  <span class="at">reduction_method =</span> <span class="st">&#39;pca&#39;</span>,</span>
<span id="cb250-8"><a href="visium-hd.html#cb250-8" tabindex="-1"></a>  <span class="at">reduction_name =</span> <span class="st">&#39;pca.projection&#39;</span>,</span>
<span id="cb250-9"><a href="visium-hd.html#cb250-9" tabindex="-1"></a>  <span class="at">prob =</span> <span class="cn">FALSE</span>,</span>
<span id="cb250-10"><a href="visium-hd.html#cb250-10" tabindex="-1"></a>  <span class="at">knn_k =</span> <span class="dv">5</span>,</span>
<span id="cb250-11"><a href="visium-hd.html#cb250-11" tabindex="-1"></a>  <span class="at">dimensions_to_use =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span></span>
<span id="cb250-12"><a href="visium-hd.html#cb250-12" tabindex="-1"></a>)</span>
<span id="cb250-13"><a href="visium-hd.html#cb250-13" tabindex="-1"></a></span>
<span id="cb250-14"><a href="visium-hd.html#cb250-14" tabindex="-1"></a><span class="fu">pDataDT</span>(visiumHD_subset)</span>
<span id="cb250-15"><a href="visium-hd.html#cb250-15" tabindex="-1"></a></span>
<span id="cb250-16"><a href="visium-hd.html#cb250-16" tabindex="-1"></a><span class="fu">dimPlot2D</span>(</span>
<span id="cb250-17"><a href="visium-hd.html#cb250-17" tabindex="-1"></a>  <span class="at">gobject =</span> visiumHD_subset,</span>
<span id="cb250-18"><a href="visium-hd.html#cb250-18" tabindex="-1"></a>  <span class="at">point_size =</span> <span class="fl">1.5</span>,</span>
<span id="cb250-19"><a href="visium-hd.html#cb250-19" tabindex="-1"></a>  <span class="at">spat_unit =</span> <span class="st">&#39;hex25&#39;</span>,</span>
<span id="cb250-20"><a href="visium-hd.html#cb250-20" tabindex="-1"></a>  <span class="at">dim_reduction_to_use =</span> <span class="st">&#39;umap&#39;</span>,</span>
<span id="cb250-21"><a href="visium-hd.html#cb250-21" tabindex="-1"></a>  <span class="at">dim_reduction_name =</span> <span class="st">&#39;umap.projection&#39;</span>,</span>
<span id="cb250-22"><a href="visium-hd.html#cb250-22" tabindex="-1"></a>  <span class="at">cell_color =</span> <span class="st">&#39;knn_labels&#39;</span></span>
<span id="cb250-23"><a href="visium-hd.html#cb250-23" tabindex="-1"></a>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-345"></span>
<img src="img/02_session2/28-dimPlot2D.png" alt="xxx" width="100%" />
<p class="caption">
Figure 9.34: xxx
</p>
</div>
<div class="sourceCode" id="cb251"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb251-1"><a href="visium-hd.html#cb251-1" tabindex="-1"></a><span class="fu">spatInSituPlotPoints</span>(visiumHD_subset,</span>
<span id="cb251-2"><a href="visium-hd.html#cb251-2" tabindex="-1"></a>                     <span class="at">show_image =</span> F,</span>
<span id="cb251-3"><a href="visium-hd.html#cb251-3" tabindex="-1"></a>                     <span class="at">feats =</span> <span class="cn">NULL</span>,</span>
<span id="cb251-4"><a href="visium-hd.html#cb251-4" tabindex="-1"></a>                     <span class="at">show_legend =</span> F,</span>
<span id="cb251-5"><a href="visium-hd.html#cb251-5" tabindex="-1"></a>                     <span class="at">spat_unit =</span> <span class="st">&#39;hex25&#39;</span>,</span>
<span id="cb251-6"><a href="visium-hd.html#cb251-6" tabindex="-1"></a>                     <span class="at">point_size =</span> <span class="fl">0.5</span>,</span>
<span id="cb251-7"><a href="visium-hd.html#cb251-7" tabindex="-1"></a>                     <span class="at">show_polygon =</span> <span class="cn">TRUE</span>,</span>
<span id="cb251-8"><a href="visium-hd.html#cb251-8" tabindex="-1"></a>                     <span class="at">use_overlap =</span> <span class="cn">FALSE</span>,</span>
<span id="cb251-9"><a href="visium-hd.html#cb251-9" tabindex="-1"></a>                     <span class="at">polygon_feat_type =</span> <span class="st">&#39;hex25&#39;</span>,</span>
<span id="cb251-10"><a href="visium-hd.html#cb251-10" tabindex="-1"></a>                     <span class="at">polygon_fill_as_factor =</span> <span class="cn">TRUE</span>,</span>
<span id="cb251-11"><a href="visium-hd.html#cb251-11" tabindex="-1"></a>                     <span class="at">polygon_fill =</span> <span class="st">&#39;knn_labels&#39;</span>,</span>
<span id="cb251-12"><a href="visium-hd.html#cb251-12" tabindex="-1"></a>                     <span class="at">polygon_color =</span> <span class="st">&#39;black&#39;</span>,</span>
<span id="cb251-13"><a href="visium-hd.html#cb251-13" tabindex="-1"></a>                     <span class="at">polygon_line_size =</span> <span class="fl">0.3</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-347"></span>
<img src="img/02_session2/29-spatInSituPlotPoints.png" alt="xxx" width="100%" />
<p class="caption">
Figure 9.35: xxx
</p>
</div>
</div>
</div>
<div id="niche-clustering" class="section level3 hasAnchor" number="9.6.3">
<h3><span class="header-section-number">9.6.3</span> Niche clustering<a href="visium-hd.html#niche-clustering" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Each cell will be clustered based on its neighboring cell type composition.</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-348"></span>
<img src="img/02_session2/10.png" alt="Schematic for niche clustering. Originally from CODEX." width="50%" />
<p class="caption">
Figure 9.36: Schematic for niche clustering. Originally from CODEX.
</p>
</div>
<p>Size of cellular niche is important and defines the tissue organization resolution.</p>
<div class="sourceCode" id="cb252"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb252-1"><a href="visium-hd.html#cb252-1" tabindex="-1"></a>visiumHD_subset <span class="ot">=</span> <span class="fu">createSpatialNetwork</span>(visiumHD_subset,</span>
<span id="cb252-2"><a href="visium-hd.html#cb252-2" tabindex="-1"></a>                                <span class="at">name =</span> <span class="st">&#39;kNN_network&#39;</span>,</span>
<span id="cb252-3"><a href="visium-hd.html#cb252-3" tabindex="-1"></a>                                <span class="at">spat_unit =</span> <span class="st">&#39;hex25&#39;</span>, </span>
<span id="cb252-4"><a href="visium-hd.html#cb252-4" tabindex="-1"></a>                                <span class="at">method =</span> <span class="st">&#39;kNN&#39;</span>,</span>
<span id="cb252-5"><a href="visium-hd.html#cb252-5" tabindex="-1"></a>                                <span class="at">k =</span> <span class="dv">6</span>)</span>
<span id="cb252-6"><a href="visium-hd.html#cb252-6" tabindex="-1"></a></span>
<span id="cb252-7"><a href="visium-hd.html#cb252-7" tabindex="-1"></a><span class="fu">pDataDT</span>(visiumHD_subset)</span>
<span id="cb252-8"><a href="visium-hd.html#cb252-8" tabindex="-1"></a>visiumHD_subset <span class="ot">=</span> <span class="fu">calculateSpatCellMetadataProportions</span>(<span class="at">gobject =</span> visiumHD_subset,</span>
<span id="cb252-9"><a href="visium-hd.html#cb252-9" tabindex="-1"></a>                                                       <span class="at">spat_unit =</span> <span class="st">&#39;hex25&#39;</span>,</span>
<span id="cb252-10"><a href="visium-hd.html#cb252-10" tabindex="-1"></a>                                                       <span class="at">feat_type =</span> <span class="st">&#39;rna&#39;</span>,</span>
<span id="cb252-11"><a href="visium-hd.html#cb252-11" tabindex="-1"></a>                                                       <span class="at">metadata_column =</span> <span class="st">&#39;knn_labels&#39;</span>,</span>
<span id="cb252-12"><a href="visium-hd.html#cb252-12" tabindex="-1"></a>                                                       <span class="at">spat_network =</span> <span class="st">&#39;kNN_network&#39;</span>)</span>
<span id="cb252-13"><a href="visium-hd.html#cb252-13" tabindex="-1"></a></span>
<span id="cb252-14"><a href="visium-hd.html#cb252-14" tabindex="-1"></a>prop_table <span class="ot">=</span> <span class="fu">getSpatialEnrichment</span>(visiumHD_subset, <span class="at">name =</span> <span class="st">&#39;proportion&#39;</span>, <span class="at">output =</span> <span class="st">&#39;data.table&#39;</span>)</span>
<span id="cb252-15"><a href="visium-hd.html#cb252-15" tabindex="-1"></a>prop_matrix <span class="ot">=</span> GiottoUtils<span class="sc">:::</span><span class="fu">dt_to_matrix</span>(prop_table)</span>
<span id="cb252-16"><a href="visium-hd.html#cb252-16" tabindex="-1"></a></span>
<span id="cb252-17"><a href="visium-hd.html#cb252-17" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">1234</span>)</span>
<span id="cb252-18"><a href="visium-hd.html#cb252-18" tabindex="-1"></a>prop_kmeans <span class="ot">=</span> <span class="fu">kmeans</span>(<span class="at">x =</span> prop_matrix, <span class="at">centers =</span> <span class="dv">10</span>, <span class="at">iter.max =</span> <span class="dv">1000</span>, <span class="at">nstart =</span> <span class="dv">100</span>)</span>
<span id="cb252-19"><a href="visium-hd.html#cb252-19" tabindex="-1"></a>prop_kmeansDT <span class="ot">=</span> data.table<span class="sc">::</span><span class="fu">data.table</span>(<span class="at">cell_ID =</span> <span class="fu">names</span>(prop_kmeans<span class="sc">$</span>cluster), <span class="at">niche =</span> prop_kmeans<span class="sc">$</span>cluster)</span>
<span id="cb252-20"><a href="visium-hd.html#cb252-20" tabindex="-1"></a></span>
<span id="cb252-21"><a href="visium-hd.html#cb252-21" tabindex="-1"></a>visiumHD_subset <span class="ot">=</span> <span class="fu">addCellMetadata</span>(visiumHD_subset, </span>
<span id="cb252-22"><a href="visium-hd.html#cb252-22" tabindex="-1"></a>                                  <span class="at">new_metadata =</span> prop_kmeansDT, </span>
<span id="cb252-23"><a href="visium-hd.html#cb252-23" tabindex="-1"></a>                                  <span class="at">by_column =</span> T, </span>
<span id="cb252-24"><a href="visium-hd.html#cb252-24" tabindex="-1"></a>                                  <span class="at">column_cell_ID =</span> <span class="st">&#39;cell_ID&#39;</span>)</span>
<span id="cb252-25"><a href="visium-hd.html#cb252-25" tabindex="-1"></a><span class="fu">pDataDT</span>(visiumHD_subset)</span>
<span id="cb252-26"><a href="visium-hd.html#cb252-26" tabindex="-1"></a></span>
<span id="cb252-27"><a href="visium-hd.html#cb252-27" tabindex="-1"></a><span class="fu">spatInSituPlotPoints</span>(visiumHD_subset,</span>
<span id="cb252-28"><a href="visium-hd.html#cb252-28" tabindex="-1"></a>                     <span class="at">show_image =</span> F,</span>
<span id="cb252-29"><a href="visium-hd.html#cb252-29" tabindex="-1"></a>                     <span class="at">feats =</span> <span class="cn">NULL</span>,</span>
<span id="cb252-30"><a href="visium-hd.html#cb252-30" tabindex="-1"></a>                     <span class="at">show_legend =</span> F,</span>
<span id="cb252-31"><a href="visium-hd.html#cb252-31" tabindex="-1"></a>                     <span class="at">spat_unit =</span> <span class="st">&#39;hex25&#39;</span>,</span>
<span id="cb252-32"><a href="visium-hd.html#cb252-32" tabindex="-1"></a>                     <span class="at">point_size =</span> <span class="fl">0.5</span>,</span>
<span id="cb252-33"><a href="visium-hd.html#cb252-33" tabindex="-1"></a>                     <span class="at">show_polygon =</span> <span class="cn">TRUE</span>,</span>
<span id="cb252-34"><a href="visium-hd.html#cb252-34" tabindex="-1"></a>                     <span class="at">use_overlap =</span> <span class="cn">FALSE</span>,</span>
<span id="cb252-35"><a href="visium-hd.html#cb252-35" tabindex="-1"></a>                     <span class="at">polygon_feat_type =</span> <span class="st">&#39;hex25&#39;</span>,</span>
<span id="cb252-36"><a href="visium-hd.html#cb252-36" tabindex="-1"></a>                     <span class="at">polygon_fill_as_factor =</span> <span class="cn">TRUE</span>,</span>
<span id="cb252-37"><a href="visium-hd.html#cb252-37" tabindex="-1"></a>                     <span class="at">polygon_fill =</span> <span class="st">&#39;niche&#39;</span>,</span>
<span id="cb252-38"><a href="visium-hd.html#cb252-38" tabindex="-1"></a>                     <span class="at">polygon_color =</span> <span class="st">&#39;black&#39;</span>,</span>
<span id="cb252-39"><a href="visium-hd.html#cb252-39" tabindex="-1"></a>                     <span class="at">polygon_line_size =</span> <span class="fl">0.3</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-350"></span>
<img src="img/02_session2/30-spatInSituPlotPoints.png" alt="xxx" width="100%" />
<p class="caption">
Figure 9.37: xxx
</p>
</div>
</div>
</div>
<div id="database-backend---work-in-progress-but-coming-soon" class="section level2 hasAnchor" number="9.7">
<h2><span class="header-section-number">9.7</span> Database backend - Work in progress, but coming soon!<a href="visium-hd.html#database-backend---work-in-progress-but-coming-soon" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Memory problems:<br />
- data ingestion<br />
- spatial operations<br />
- matrix operations<br />
- matrix and spatial geometry object sizes</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="visium-part-ii.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="xenium-1.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
