<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>8 Visium Part II | Workshop: Spatial multi-omics data analysis with Giotto Suite</title>
  <meta name="description" content="8 Visium Part II | Workshop: Spatial multi-omics data analysis with Giotto Suite" />
  <meta name="generator" content="bookdown 0.39 and GitBook 2.6.7" />

  <meta property="og:title" content="8 Visium Part II | Workshop: Spatial multi-omics data analysis with Giotto Suite" />
  <meta property="og:type" content="book" />
  
  
  <meta name="github-repo" content="drieslab/giotto_workshop_2024" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="8 Visium Part II | Workshop: Spatial multi-omics data analysis with Giotto Suite" />
  
  
  

<meta name="author" content="Ruben Dries, Jiaji George Chen, Joselyn Cristina Chávez-Fuentes, Junxiang Xu ,Edward Ruiz, Jeff Sheridan, Iqra Amin, Wen Wang" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="visium-part-i.html"/>
<link rel="next" href="visium-hd.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Giotto Suite Workshop 2024</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#instructors"><i class="fa fa-check"></i><b>1.1</b> Instructors</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#topics-and-schedule"><i class="fa fa-check"></i><b>1.2</b> Topics and Schedule:</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#license"><i class="fa fa-check"></i><b>1.3</b> License</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="datasets-packages.html"><a href="datasets-packages.html"><i class="fa fa-check"></i><b>2</b> Datasets &amp; Packages</a>
<ul>
<li class="chapter" data-level="2.1" data-path="datasets-packages.html"><a href="datasets-packages.html#datasets-to-download"><i class="fa fa-check"></i><b>2.1</b> Datasets to download</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="datasets-packages.html"><a href="datasets-packages.html#zenodo-repository"><i class="fa fa-check"></i><b>2.1.1</b> Zenodo repository</a></li>
<li class="chapter" data-level="2.1.2" data-path="datasets-packages.html"><a href="datasets-packages.html#x-genomics-visium-mouse-brain-section-coronal-dataset"><i class="fa fa-check"></i><b>2.1.2</b> 10X Genomics Visium Mouse Brain Section (Coronal) dataset</a></li>
<li class="chapter" data-level="2.1.3" data-path="datasets-packages.html"><a href="datasets-packages.html#x-genomics-visium-hd-ffpe-human-colon-cancer"><i class="fa fa-check"></i><b>2.1.3</b> 10X Genomics Visium HD: FFPE Human Colon Cancer</a></li>
<li class="chapter" data-level="2.1.4" data-path="datasets-packages.html"><a href="datasets-packages.html#x-genomics-multi-modal-dataset"><i class="fa fa-check"></i><b>2.1.4</b> 10X Genomics multi-modal dataset</a></li>
<li class="chapter" data-level="2.1.5" data-path="datasets-packages.html"><a href="datasets-packages.html#x-genomics-multi-omics-visium-cytassist-human-tonsil-dataset"><i class="fa fa-check"></i><b>2.1.5</b> 10X Genomics multi-omics Visium CytAssist Human Tonsil dataset</a></li>
<li class="chapter" data-level="2.1.6" data-path="datasets-packages.html"><a href="datasets-packages.html#x-genomics-human-prostate-cancer-adenocarcinoma-with-invasive-carcinoma-ffpe"><i class="fa fa-check"></i><b>2.1.6</b> 10X Genomics Human Prostate Cancer Adenocarcinoma with Invasive Carcinoma (FFPE)</a></li>
<li class="chapter" data-level="2.1.7" data-path="datasets-packages.html"><a href="datasets-packages.html#x-genomics-normal-human-prostate-ffpe"><i class="fa fa-check"></i><b>2.1.7</b> 10X Genomics Normal Human Prostate (FFPE)</a></li>
<li class="chapter" data-level="2.1.8" data-path="datasets-packages.html"><a href="datasets-packages.html#xenium"><i class="fa fa-check"></i><b>2.1.8</b> Xenium</a></li>
<li class="chapter" data-level="2.1.9" data-path="datasets-packages.html"><a href="datasets-packages.html#merfish-cortex-dataset"><i class="fa fa-check"></i><b>2.1.9</b> MERFISH cortex dataset</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="datasets-packages.html"><a href="datasets-packages.html#needed-packages"><i class="fa fa-check"></i><b>2.2</b> Needed packages</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="datasets-packages.html"><a href="datasets-packages.html#cran-dependencies"><i class="fa fa-check"></i><b>2.2.1</b> CRAN dependencies:</a></li>
<li class="chapter" data-level="2.2.2" data-path="datasets-packages.html"><a href="datasets-packages.html#terra-installation"><i class="fa fa-check"></i><b>2.2.2</b> <em>terra</em> installation</a></li>
<li class="chapter" data-level="2.2.3" data-path="datasets-packages.html"><a href="datasets-packages.html#matrix-installation"><i class="fa fa-check"></i><b>2.2.3</b> <em>Matrix</em> installation</a></li>
<li class="chapter" data-level="2.2.4" data-path="datasets-packages.html"><a href="datasets-packages.html#rtools-installation"><i class="fa fa-check"></i><b>2.2.4</b> <em>Rtools</em> installation</a></li>
<li class="chapter" data-level="2.2.5" data-path="datasets-packages.html"><a href="datasets-packages.html#giotto-installation"><i class="fa fa-check"></i><b>2.2.5</b> <em>Giotto</em> installation</a></li>
<li class="chapter" data-level="2.2.6" data-path="datasets-packages.html"><a href="datasets-packages.html#irlba-install"><i class="fa fa-check"></i><b>2.2.6</b> <em>irlba</em> install</a></li>
<li class="chapter" data-level="2.2.7" data-path="datasets-packages.html"><a href="datasets-packages.html#arrow-install"><i class="fa fa-check"></i><b>2.2.7</b> <em>arrow</em> install</a></li>
<li class="chapter" data-level="2.2.8" data-path="datasets-packages.html"><a href="datasets-packages.html#bioconductor-dependencies"><i class="fa fa-check"></i><b>2.2.8</b> Bioconductor dependencies:</a></li>
<li class="chapter" data-level="2.2.9" data-path="datasets-packages.html"><a href="datasets-packages.html#cran-packages"><i class="fa fa-check"></i><b>2.2.9</b> CRAN packages:</a></li>
<li class="chapter" data-level="2.2.10" data-path="datasets-packages.html"><a href="datasets-packages.html#packages-from-github"><i class="fa fa-check"></i><b>2.2.10</b> Packages from GitHub</a></li>
<li class="chapter" data-level="2.2.11" data-path="datasets-packages.html"><a href="datasets-packages.html#python-environments"><i class="fa fa-check"></i><b>2.2.11</b> Python environments</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html"><i class="fa fa-check"></i><b>3</b> Spatial omics technologies</a>
<ul>
<li class="chapter" data-level="3.1" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#presentation"><i class="fa fa-check"></i><b>3.1</b> Presentation</a></li>
<li class="chapter" data-level="3.2" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#short-summary"><i class="fa fa-check"></i><b>3.2</b> Short summary</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#why-do-we-need-spatial-omics-technologies"><i class="fa fa-check"></i><b>3.2.1</b> Why do we need spatial omics technologies?</a></li>
<li class="chapter" data-level="3.2.2" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#what-is-spatial-omics"><i class="fa fa-check"></i><b>3.2.2</b> What is spatial omics?</a></li>
<li class="chapter" data-level="3.2.3" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#what-are-the-main-spatial-omics-technologies"><i class="fa fa-check"></i><b>3.2.3</b> What are the main spatial omics technologies?</a></li>
<li class="chapter" data-level="3.2.4" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#other-spatial-omics-atac-seq-cuttag-lipidomics-etc"><i class="fa fa-check"></i><b>3.2.4</b> Other Spatial omics: ATAC-seq, CUT&amp;Tag, lipidomics, etc</a></li>
<li class="chapter" data-level="3.2.5" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#what-are-the-different-types-of-spatial-downstream-analyses"><i class="fa fa-check"></i><b>3.2.5</b> What are the different types of spatial downstream analyses?</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html"><i class="fa fa-check"></i><b>4</b> Introduction to the Giotto package</a>
<ul>
<li class="chapter" data-level="4.1" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html#presentation-1"><i class="fa fa-check"></i><b>4.1</b> Presentation</a></li>
<li class="chapter" data-level="4.2" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html#ecosystem"><i class="fa fa-check"></i><b>4.2</b> Ecosystem</a></li>
<li class="chapter" data-level="4.3" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html#installation-python-environment"><i class="fa fa-check"></i><b>4.3</b> Installation + python environment</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html#giotto-installation-1"><i class="fa fa-check"></i><b>4.3.1</b> Giotto installation</a></li>
<li class="chapter" data-level="4.3.2" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html#python-environment"><i class="fa fa-check"></i><b>4.3.2</b> Python environment</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html#giotto-instructions"><i class="fa fa-check"></i><b>4.4</b> Giotto instructions</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html"><i class="fa fa-check"></i><b>5</b> Data formatting and Pre-processing</a>
<ul>
<li class="chapter" data-level="5.1" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#data-formats"><i class="fa fa-check"></i><b>5.1</b> Data formats</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#general-formats"><i class="fa fa-check"></i><b>5.1.1</b> General formats</a></li>
<li class="chapter" data-level="5.1.2" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#matrix-formats"><i class="fa fa-check"></i><b>5.1.2</b> Matrix formats</a></li>
<li class="chapter" data-level="5.1.3" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#tabular-formats"><i class="fa fa-check"></i><b>5.1.3</b> Tabular formats</a></li>
<li class="chapter" data-level="5.1.4" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#spatial-formats"><i class="fa fa-check"></i><b>5.1.4</b> Spatial formats</a></li>
<li class="chapter" data-level="5.1.5" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#mask-files"><i class="fa fa-check"></i><b>5.1.5</b> Mask files</a></li>
<li class="chapter" data-level="5.1.6" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#images"><i class="fa fa-check"></i><b>5.1.6</b> images</a></li>
<li class="chapter" data-level="5.1.7" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#jsonlike-formats"><i class="fa fa-check"></i><b>5.1.7</b> jsonlike formats</a></li>
<li class="chapter" data-level="5.1.8" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#hierarchical-formats"><i class="fa fa-check"></i><b>5.1.8</b> Hierarchical formats</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#pre-processing"><i class="fa fa-check"></i><b>5.2</b> Pre-processing</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#expression-matrix"><i class="fa fa-check"></i><b>5.2.1</b> Expression matrix</a></li>
<li class="chapter" data-level="5.2.2" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#spatial-locations"><i class="fa fa-check"></i><b>5.2.2</b> Spatial locations</a></li>
<li class="chapter" data-level="5.2.3" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#giottopoints"><i class="fa fa-check"></i><b>5.2.3</b> giottoPoints</a></li>
<li class="chapter" data-level="5.2.4" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#giottopolygon"><i class="fa fa-check"></i><b>5.2.4</b> giottoPolygon</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#subobject-utility-functions"><i class="fa fa-check"></i><b>5.3</b> Subobject utility functions</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#ids"><i class="fa fa-check"></i><b>5.3.1</b> IDs</a></li>
<li class="chapter" data-level="5.3.2" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#bracket-subsetting-and-extraction"><i class="fa fa-check"></i><b>5.3.2</b> Bracket subsetting and extraction</a></li>
<li class="chapter" data-level="5.3.3" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#nesting-metadata-generics"><i class="fa fa-check"></i><b>5.3.3</b> Nesting metadata generics</a></li>
<li class="chapter" data-level="5.3.4" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#appending-to-a-giotto-object"><i class="fa fa-check"></i><b>5.3.4</b> Appending to a giotto object</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html"><i class="fa fa-check"></i><b>6</b> Creating a Giotto object</a>
<ul>
<li class="chapter" data-level="6.0.1" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#used-packages"><i class="fa fa-check"></i><b>6.0.1</b> Used packages</a></li>
<li class="chapter" data-level="6.1" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#overview"><i class="fa fa-check"></i><b>6.1</b> Overview</a></li>
<li class="chapter" data-level="6.2" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#from-matrix-locations"><i class="fa fa-check"></i><b>6.2</b> From matrix + locations</a></li>
<li class="chapter" data-level="6.3" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#from-subcellular-raw-data-transcripts-or-images-polygons"><i class="fa fa-check"></i><b>6.3</b> From subcellular raw data (transcripts or images) + polygons</a></li>
<li class="chapter" data-level="6.4" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#from-piece-wise"><i class="fa fa-check"></i><b>6.4</b> From piece-wise</a></li>
<li class="chapter" data-level="6.5" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#using-convenience-functions-for-popular-technologies-vizgen-xenium-cosmx"><i class="fa fa-check"></i><b>6.5</b> Using convenience functions for popular technologies (Vizgen, Xenium, CosMx, …)</a></li>
<li class="chapter" data-level="6.6" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#spatial-plots"><i class="fa fa-check"></i><b>6.6</b> Spatial plots</a></li>
<li class="chapter" data-level="6.7" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#subsetting"><i class="fa fa-check"></i><b>6.7</b> Subsetting</a></li>
<li class="chapter" data-level="6.8" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#mini-objects-giottodata"><i class="fa fa-check"></i><b>6.8</b> Mini objects &amp; GiottoData</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="visium-part-i.html"><a href="visium-part-i.html"><i class="fa fa-check"></i><b>7</b> Visium Part I</a>
<ul>
<li class="chapter" data-level="7.1" data-path="visium-part-i.html"><a href="visium-part-i.html#the-visium-technology"><i class="fa fa-check"></i><b>7.1</b> The Visium technology</a></li>
<li class="chapter" data-level="7.2" data-path="visium-part-i.html"><a href="visium-part-i.html#introduction-to-the-spatial-dataset"><i class="fa fa-check"></i><b>7.2</b> Introduction to the spatial dataset</a></li>
<li class="chapter" data-level="7.3" data-path="visium-part-i.html"><a href="visium-part-i.html#download-dataset"><i class="fa fa-check"></i><b>7.3</b> Download dataset</a></li>
<li class="chapter" data-level="7.4" data-path="visium-part-i.html"><a href="visium-part-i.html#create-the-giotto-object"><i class="fa fa-check"></i><b>7.4</b> Create the Giotto object</a></li>
<li class="chapter" data-level="7.5" data-path="visium-part-i.html"><a href="visium-part-i.html#subset-on-spots-that-were-covered-by-tissue"><i class="fa fa-check"></i><b>7.5</b> Subset on spots that were covered by tissue</a></li>
<li class="chapter" data-level="7.6" data-path="visium-part-i.html"><a href="visium-part-i.html#quality-control"><i class="fa fa-check"></i><b>7.6</b> Quality control</a></li>
<li class="chapter" data-level="7.7" data-path="visium-part-i.html"><a href="visium-part-i.html#filtering"><i class="fa fa-check"></i><b>7.7</b> Filtering</a></li>
<li class="chapter" data-level="7.8" data-path="visium-part-i.html"><a href="visium-part-i.html#normalization"><i class="fa fa-check"></i><b>7.8</b> Normalization</a></li>
<li class="chapter" data-level="7.9" data-path="visium-part-i.html"><a href="visium-part-i.html#feature-selection"><i class="fa fa-check"></i><b>7.9</b> Feature selection</a>
<ul>
<li class="chapter" data-level="7.9.1" data-path="visium-part-i.html"><a href="visium-part-i.html#highly-variable-features"><i class="fa fa-check"></i><b>7.9.1</b> Highly Variable Features:</a></li>
</ul></li>
<li class="chapter" data-level="7.10" data-path="visium-part-i.html"><a href="visium-part-i.html#dimension-reduction"><i class="fa fa-check"></i><b>7.10</b> Dimension Reduction</a>
<ul>
<li class="chapter" data-level="7.10.1" data-path="visium-part-i.html"><a href="visium-part-i.html#pca"><i class="fa fa-check"></i><b>7.10.1</b> PCA</a></li>
<li class="chapter" data-level="7.10.2" data-path="visium-part-i.html"><a href="visium-part-i.html#umap"><i class="fa fa-check"></i><b>7.10.2</b> UMAP</a></li>
<li class="chapter" data-level="7.10.3" data-path="visium-part-i.html"><a href="visium-part-i.html#t-sne"><i class="fa fa-check"></i><b>7.10.3</b> t-SNE</a></li>
</ul></li>
<li class="chapter" data-level="7.11" data-path="visium-part-i.html"><a href="visium-part-i.html#clustering"><i class="fa fa-check"></i><b>7.11</b> Clustering</a>
<ul>
<li class="chapter" data-level="7.11.1" data-path="visium-part-i.html"><a href="visium-part-i.html#calculate-leiden-clustering"><i class="fa fa-check"></i><b>7.11.1</b> Calculate Leiden clustering</a></li>
<li class="chapter" data-level="7.11.2" data-path="visium-part-i.html"><a href="visium-part-i.html#calculate-louvain-clustering"><i class="fa fa-check"></i><b>7.11.2</b> Calculate Louvain clustering</a></li>
</ul></li>
<li class="chapter" data-level="7.12" data-path="visium-part-i.html"><a href="visium-part-i.html#save-the-object"><i class="fa fa-check"></i><b>7.12</b> Save the object</a></li>
<li class="chapter" data-level="7.13" data-path="visium-part-i.html"><a href="visium-part-i.html#session-info"><i class="fa fa-check"></i><b>7.13</b> Session info</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="visium-part-ii.html"><a href="visium-part-ii.html"><i class="fa fa-check"></i><b>8</b> Visium Part II</a>
<ul>
<li class="chapter" data-level="8.1" data-path="visium-part-ii.html"><a href="visium-part-ii.html#load-the-object"><i class="fa fa-check"></i><b>8.1</b> Load the object</a></li>
<li class="chapter" data-level="8.2" data-path="visium-part-ii.html"><a href="visium-part-ii.html#differential-expression"><i class="fa fa-check"></i><b>8.2</b> Differential expression</a>
<ul>
<li class="chapter" data-level="8.2.1" data-path="visium-part-ii.html"><a href="visium-part-ii.html#gini-markers"><i class="fa fa-check"></i><b>8.2.1</b> Gini markers</a></li>
<li class="chapter" data-level="8.2.2" data-path="visium-part-ii.html"><a href="visium-part-ii.html#scran-markers"><i class="fa fa-check"></i><b>8.2.2</b> Scran markers</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="visium-part-ii.html"><a href="visium-part-ii.html#enrichment-deconvolution"><i class="fa fa-check"></i><b>8.3</b> Enrichment &amp; Deconvolution</a>
<ul>
<li class="chapter" data-level="8.3.1" data-path="visium-part-ii.html"><a href="visium-part-ii.html#pagerank"><i class="fa fa-check"></i><b>8.3.1</b> PAGE/Rank</a></li>
<li class="chapter" data-level="8.3.2" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatialdwls"><i class="fa fa-check"></i><b>8.3.2</b> SpatialDWLS</a></li>
</ul></li>
<li class="chapter" data-level="8.4" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatial-expression-patterns"><i class="fa fa-check"></i><b>8.4</b> Spatial expression patterns</a>
<ul>
<li class="chapter" data-level="8.4.1" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatial-variable-genes"><i class="fa fa-check"></i><b>8.4.1</b> Spatial variable genes</a></li>
<li class="chapter" data-level="8.4.2" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatial-co-expression-modules"><i class="fa fa-check"></i><b>8.4.2</b> Spatial co-expression modules</a></li>
</ul></li>
<li class="chapter" data-level="8.5" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatially-informed-clusters"><i class="fa fa-check"></i><b>8.5</b> Spatially informed clusters</a></li>
<li class="chapter" data-level="8.6" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatial-domains-hmrf"><i class="fa fa-check"></i><b>8.6</b> Spatial domains HMRF</a></li>
<li class="chapter" data-level="8.7" data-path="visium-part-ii.html"><a href="visium-part-ii.html#interactive-tools"><i class="fa fa-check"></i><b>8.7</b> Interactive tools</a></li>
<li class="chapter" data-level="8.8" data-path="visium-part-ii.html"><a href="visium-part-ii.html#session-info-1"><i class="fa fa-check"></i><b>8.8</b> Session info</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="visium-hd.html"><a href="visium-hd.html"><i class="fa fa-check"></i><b>9</b> Visium HD</a>
<ul>
<li class="chapter" data-level="9.1" data-path="visium-hd.html"><a href="visium-hd.html#objective"><i class="fa fa-check"></i><b>9.1</b> Objective</a></li>
<li class="chapter" data-level="9.2" data-path="visium-hd.html"><a href="visium-hd.html#background"><i class="fa fa-check"></i><b>9.2</b> Background</a>
<ul>
<li class="chapter" data-level="9.2.1" data-path="visium-hd.html"><a href="visium-hd.html#visium-hd-technology"><i class="fa fa-check"></i><b>9.2.1</b> Visium HD Technology</a></li>
<li class="chapter" data-level="9.2.2" data-path="visium-hd.html"><a href="visium-hd.html#colorectal-cancer-sample"><i class="fa fa-check"></i><b>9.2.2</b> Colorectal Cancer Sample</a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="visium-hd.html"><a href="visium-hd.html#data-ingestion"><i class="fa fa-check"></i><b>9.3</b> Data Ingestion</a>
<ul>
<li class="chapter" data-level="9.3.1" data-path="visium-hd.html"><a href="visium-hd.html#visium-hd-output-data-format"><i class="fa fa-check"></i><b>9.3.1</b> Visium HD output data format</a></li>
<li class="chapter" data-level="9.3.2" data-path="visium-hd.html"><a href="visium-hd.html#mini-visium-hd-dataset"><i class="fa fa-check"></i><b>9.3.2</b> Mini Visium HD dataset</a></li>
<li class="chapter" data-level="9.3.3" data-path="visium-hd.html"><a href="visium-hd.html#giotto-visium-hd-convenience-function"><i class="fa fa-check"></i><b>9.3.3</b> Giotto Visium HD convenience function</a></li>
<li class="chapter" data-level="9.3.4" data-path="visium-hd.html"><a href="visium-hd.html#read-in-data-manually"><i class="fa fa-check"></i><b>9.3.4</b> Read in data manually</a></li>
</ul></li>
<li class="chapter" data-level="9.4" data-path="visium-hd.html"><a href="visium-hd.html#create-giotto-object"><i class="fa fa-check"></i><b>9.4</b> Create Giotto object</a>
<ul>
<li class="chapter" data-level="9.4.1" data-path="visium-hd.html"><a href="visium-hd.html#create-giotto-points"><i class="fa fa-check"></i><b>9.4.1</b> create giotto points</a></li>
<li class="chapter" data-level="9.4.2" data-path="visium-hd.html"><a href="visium-hd.html#create-giotto-polygons"><i class="fa fa-check"></i><b>9.4.2</b> create giotto polygons</a></li>
<li class="chapter" data-level="9.4.3" data-path="visium-hd.html"><a href="visium-hd.html#combine-giotto-points-and-polygons-to-create-giotto-object"><i class="fa fa-check"></i><b>9.4.3</b> combine Giotto points and polygons to create Giotto object</a></li>
</ul></li>
<li class="chapter" data-level="9.5" data-path="visium-hd.html"><a href="visium-hd.html#process-giotto-object"><i class="fa fa-check"></i><b>9.5</b> Process Giotto object</a>
<ul>
<li class="chapter" data-level="9.5.1" data-path="visium-hd.html"><a href="visium-hd.html#calculate-overlap-between-points-and-polygons"><i class="fa fa-check"></i><b>9.5.1</b> calculate overlap between points and polygons</a></li>
<li class="chapter" data-level="9.5.2" data-path="visium-hd.html"><a href="visium-hd.html#convert-overlap-results-to-a-bin-by-gene-matrix"><i class="fa fa-check"></i><b>9.5.2</b> convert overlap results to a bin-by-gene matrix</a></li>
<li class="chapter" data-level="9.5.3" data-path="visium-hd.html"><a href="visium-hd.html#default-processing-steps"><i class="fa fa-check"></i><b>9.5.3</b> default processing steps</a></li>
<li class="chapter" data-level="9.5.4" data-path="visium-hd.html"><a href="visium-hd.html#visualize-number-of-features"><i class="fa fa-check"></i><b>9.5.4</b> visualize number of features</a></li>
</ul></li>
<li class="chapter" data-level="9.6" data-path="visium-hd.html"><a href="visium-hd.html#dimension-reduction-clustering"><i class="fa fa-check"></i><b>9.6</b> Dimension reduction + clustering</a>
<ul>
<li class="chapter" data-level="9.6.1" data-path="visium-hd.html"><a href="visium-hd.html#highly-variable-features-pca"><i class="fa fa-check"></i><b>9.6.1</b> Highly variable features + PCA</a></li>
<li class="chapter" data-level="9.6.2" data-path="visium-hd.html"><a href="visium-hd.html#umap-reduction-for-visualization"><i class="fa fa-check"></i><b>9.6.2</b> UMAP reduction for visualization</a></li>
<li class="chapter" data-level="9.6.3" data-path="visium-hd.html"><a href="visium-hd.html#create-network-based-on-expression-similarity-graph-partition-cluster"><i class="fa fa-check"></i><b>9.6.3</b> Create network based on expression similarity + graph partition cluster</a></li>
</ul></li>
<li class="chapter" data-level="9.7" data-path="visium-hd.html"><a href="visium-hd.html#hexbin-100"><i class="fa fa-check"></i><b>9.7</b> Hexbin 100</a>
<ul>
<li class="chapter" data-level="9.7.1" data-path="visium-hd.html"><a href="visium-hd.html#standard-subcellular-pipeline"><i class="fa fa-check"></i><b>9.7.1</b> Standard subcellular pipeline</a></li>
<li class="chapter" data-level="9.7.2" data-path="visium-hd.html"><a href="visium-hd.html#spatial-expression-patterns-1"><i class="fa fa-check"></i><b>9.7.2</b> Spatial expression patterns</a></li>
</ul></li>
<li class="chapter" data-level="9.8" data-path="visium-hd.html"><a href="visium-hd.html#hexbin-25"><i class="fa fa-check"></i><b>9.8</b> Hexbin 25</a>
<ul>
<li class="chapter" data-level="9.8.1" data-path="visium-hd.html"><a href="visium-hd.html#subcellular-workflow-and-projection-functions"><i class="fa fa-check"></i><b>9.8.1</b> Subcellular workflow and projection functions</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="xenium-1.html"><a href="xenium-1.html"><i class="fa fa-check"></i><b>10</b> Xenium</a>
<ul>
<li class="chapter" data-level="10.1" data-path="xenium-1.html"><a href="xenium-1.html#introduction-to-spatial-dataset"><i class="fa fa-check"></i><b>10.1</b> Introduction to spatial dataset</a>
<ul>
<li class="chapter" data-level="10.1.1" data-path="xenium-1.html"><a href="xenium-1.html#output-directory-structure"><i class="fa fa-check"></i><b>10.1.1</b> Output directory structure</a></li>
<li class="chapter" data-level="10.1.2" data-path="xenium-1.html"><a href="xenium-1.html#mini-xenium-dataset"><i class="fa fa-check"></i><b>10.1.2</b> Mini Xenium Dataset</a></li>
</ul></li>
<li class="chapter" data-level="10.2" data-path="xenium-1.html"><a href="xenium-1.html#data-prep"><i class="fa fa-check"></i><b>10.2</b> Data prep</a>
<ul>
<li class="chapter" data-level="10.2.1" data-path="xenium-1.html"><a href="xenium-1.html#image-conversion-may-change"><i class="fa fa-check"></i><b>10.2.1</b> Image conversion (may change)</a></li>
</ul></li>
<li class="chapter" data-level="10.3" data-path="xenium-1.html"><a href="xenium-1.html#convenience-function"><i class="fa fa-check"></i><b>10.3</b> Convenience function</a>
<ul>
<li class="chapter" data-level="10.3.1" data-path="xenium-1.html"><a href="xenium-1.html#specific-filepaths"><i class="fa fa-check"></i><b>10.3.1</b> Specific filepaths</a></li>
<li class="chapter" data-level="10.3.2" data-path="xenium-1.html"><a href="xenium-1.html#quality-value"><i class="fa fa-check"></i><b>10.3.2</b> Quality value</a></li>
<li class="chapter" data-level="10.3.3" data-path="xenium-1.html"><a href="xenium-1.html#transcript-type-splitting"><i class="fa fa-check"></i><b>10.3.3</b> Transcript type splitting</a></li>
<li class="chapter" data-level="10.3.4" data-path="xenium-1.html"><a href="xenium-1.html#centroids-calculation"><i class="fa fa-check"></i><b>10.3.4</b> Centroids calculation</a></li>
<li class="chapter" data-level="10.3.5" data-path="xenium-1.html"><a href="xenium-1.html#simple-visualization"><i class="fa fa-check"></i><b>10.3.5</b> Simple visualization</a></li>
</ul></li>
<li class="chapter" data-level="10.4" data-path="xenium-1.html"><a href="xenium-1.html#piecewise-loading"><i class="fa fa-check"></i><b>10.4</b> Piecewise loading</a>
<ul>
<li class="chapter" data-level="10.4.1" data-path="xenium-1.html"><a href="xenium-1.html#load-giottopoints-transcripts"><i class="fa fa-check"></i><b>10.4.1</b> load giottoPoints transcripts</a></li>
<li class="chapter" data-level="10.4.2" data-path="xenium-1.html"><a href="xenium-1.html#optional-loading-pre-aggregated-data"><i class="fa fa-check"></i><b>10.4.2</b> (optional) Loading pre-aggregated data</a></li>
</ul></li>
<li class="chapter" data-level="10.5" data-path="xenium-1.html"><a href="xenium-1.html#xenium-images"><i class="fa fa-check"></i><b>10.5</b> Xenium Images</a>
<ul>
<li class="chapter" data-level="10.5.1" data-path="xenium-1.html"><a href="xenium-1.html#image-metadata"><i class="fa fa-check"></i><b>10.5.1</b> Image metadata</a></li>
<li class="chapter" data-level="10.5.2" data-path="xenium-1.html"><a href="xenium-1.html#image-loading"><i class="fa fa-check"></i><b>10.5.2</b> Image loading</a></li>
</ul></li>
<li class="chapter" data-level="10.6" data-path="xenium-1.html"><a href="xenium-1.html#spatial-aggregation"><i class="fa fa-check"></i><b>10.6</b> Spatial aggregation</a></li>
<li class="chapter" data-level="10.7" data-path="xenium-1.html"><a href="xenium-1.html#aggregate-analyses-workflow"><i class="fa fa-check"></i><b>10.7</b> Aggregate analyses workflow</a>
<ul>
<li class="chapter" data-level="10.7.1" data-path="xenium-1.html"><a href="xenium-1.html#filtering-1"><i class="fa fa-check"></i><b>10.7.1</b> Filtering</a></li>
<li class="chapter" data-level="10.7.2" data-path="xenium-1.html"><a href="xenium-1.html#normalization-1"><i class="fa fa-check"></i><b>10.7.2</b> Normalization</a></li>
<li class="chapter" data-level="10.7.3" data-path="xenium-1.html"><a href="xenium-1.html#dimension-reduction-1"><i class="fa fa-check"></i><b>10.7.3</b> Dimension Reduction</a></li>
<li class="chapter" data-level="10.7.4" data-path="xenium-1.html"><a href="xenium-1.html#clustering-1"><i class="fa fa-check"></i><b>10.7.4</b> Clustering</a></li>
</ul></li>
<li class="chapter" data-level="10.8" data-path="xenium-1.html"><a href="xenium-1.html#niche-clustering"><i class="fa fa-check"></i><b>10.8</b> Niche clustering</a>
<ul>
<li class="chapter" data-level="10.8.1" data-path="xenium-1.html"><a href="xenium-1.html#spatial-network"><i class="fa fa-check"></i><b>10.8.1</b> Spatial network</a></li>
<li class="chapter" data-level="10.8.2" data-path="xenium-1.html"><a href="xenium-1.html#niche-calculation"><i class="fa fa-check"></i><b>10.8.2</b> Niche calculation</a></li>
<li class="chapter" data-level="10.8.3" data-path="xenium-1.html"><a href="xenium-1.html#kmeans-clustering-based-on-niche-signature"><i class="fa fa-check"></i><b>10.8.3</b> kmeans clustering based on niche signature</a></li>
</ul></li>
<li class="chapter" data-level="10.9" data-path="xenium-1.html"><a href="xenium-1.html#cell-proximity-enrichment"><i class="fa fa-check"></i><b>10.9</b> Cell proximity enrichment</a></li>
<li class="chapter" data-level="10.10" data-path="xenium-1.html"><a href="xenium-1.html#pseudovisium"><i class="fa fa-check"></i><b>10.10</b> Pseudovisium</a>
<ul>
<li class="chapter" data-level="10.10.1" data-path="xenium-1.html"><a href="xenium-1.html#pseudovisium-aggregation-and-workflow"><i class="fa fa-check"></i><b>10.10.1</b> Pseudovisium aggregation and workflow</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html"><i class="fa fa-check"></i><b>11</b> Spatial proteomics: Multiplexed Immunofluorescence</a>
<ul>
<li class="chapter" data-level="11.1" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#spatial-proteomics-technologies"><i class="fa fa-check"></i><b>11.1</b> Spatial Proteomics Technologies</a></li>
<li class="chapter" data-level="11.2" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#raw-data-type-coming-out-of-different-technologies"><i class="fa fa-check"></i><b>11.2</b> Raw data type coming out of different technologies</a>
<ul>
<li class="chapter" data-level="11.2.1" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#use-ome.tiff-as-an-example-output-data-to-begin-with"><i class="fa fa-check"></i><b>11.2.1</b> Use ome.tiff as an example output data to begin with</a></li>
<li class="chapter" data-level="11.2.2" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#use-single-channel-images-as-an-example-output-data-to-begin-with"><i class="fa fa-check"></i><b>11.2.2</b> Use single channel images as an example output data to begin with</a></li>
</ul></li>
<li class="chapter" data-level="11.3" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#cell-segmentation-file-to-get-single-cell-level-protein-expression"><i class="fa fa-check"></i><b>11.3</b> Cell Segmentation file to get single cell level protein expression</a>
<ul>
<li class="chapter" data-level="11.3.1" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#using-segmentation-output-file-from-deepcellmesmer-as-an-example."><i class="fa fa-check"></i><b>11.3.1</b> Using segmentation output file from DeepCell(mesmer) as an example.</a></li>
<li class="chapter" data-level="11.3.2" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#using-giotto-wrapper-of-cellpose-to-perform-segmentation"><i class="fa fa-check"></i><b>11.3.2</b> Using Giotto wrapper of Cellpose to perform segmentation</a></li>
</ul></li>
<li class="chapter" data-level="11.4" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#create-a-giotto-object-using-list-of-gitto-large-images-and-polygons"><i class="fa fa-check"></i><b>11.4</b> Create a Giotto Object using list of gitto large images and polygons</a>
<ul>
<li class="chapter" data-level="11.4.1" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#overlap-to-matrix"><i class="fa fa-check"></i><b>11.4.1</b> Overlap to matrix</a></li>
<li class="chapter" data-level="11.4.2" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#manipulate-expression-information"><i class="fa fa-check"></i><b>11.4.2</b> Manipulate Expression information</a></li>
<li class="chapter" data-level="11.4.3" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#rescale-polygons"><i class="fa fa-check"></i><b>11.4.3</b> Rescale polygons</a></li>
<li class="chapter" data-level="11.4.4" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#perform-clustering-and-differential-expression"><i class="fa fa-check"></i><b>11.4.4</b> Perform clustering and differential expression</a></li>
<li class="chapter" data-level="11.4.5" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#give-the-cluster-an-annotation-based-on-expression-values"><i class="fa fa-check"></i><b>11.4.5</b> Give the cluster an annotation based on expression values</a></li>
<li class="chapter" data-level="11.4.6" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#spatial-network-1"><i class="fa fa-check"></i><b>11.4.6</b> spatial network</a></li>
<li class="chapter" data-level="11.4.7" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#cell-neighborhood-cell-typecell-type-interactions"><i class="fa fa-check"></i><b>11.4.7</b> Cell Neighborhood: Cell-Type/Cell-Type Interactions</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="12" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html"><i class="fa fa-check"></i><b>12</b> Working with multiple samples</a>
<ul>
<li class="chapter" data-level="12.1" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#objective-1"><i class="fa fa-check"></i><b>12.1</b> Objective</a></li>
<li class="chapter" data-level="12.2" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#background-1"><i class="fa fa-check"></i><b>12.2</b> Background</a>
<ul>
<li class="chapter" data-level="12.2.1" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#dataset"><i class="fa fa-check"></i><b>12.2.1</b> Dataset</a></li>
<li class="chapter" data-level="12.2.2" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#visium-technology"><i class="fa fa-check"></i><b>12.2.2</b> Visium technology</a></li>
</ul></li>
<li class="chapter" data-level="12.3" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#create-individual-giotto-objects"><i class="fa fa-check"></i><b>12.3</b> Create individual giotto objects</a>
<ul>
<li class="chapter" data-level="12.3.1" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#download-the-data"><i class="fa fa-check"></i><b>12.3.1</b> Download the data</a></li>
</ul></li>
<li class="chapter" data-level="12.4" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#extracting-the-downloaded-files"><i class="fa fa-check"></i><b>12.4</b> Extracting the downloaded files</a>
<ul>
<li class="chapter" data-level="12.4.1" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#create-giotto-instructions"><i class="fa fa-check"></i><b>12.4.1</b> Create giotto instructions</a></li>
<li class="chapter" data-level="12.4.2" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#load-visium-data-into-giotto"><i class="fa fa-check"></i><b>12.4.2</b> Load visium data into Giotto</a></li>
<li class="chapter" data-level="12.4.3" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#healthy-prostate-tissue-coverage"><i class="fa fa-check"></i><b>12.4.3</b> Healthy prostate tissue coverage</a></li>
<li class="chapter" data-level="12.4.4" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#adenocarcinoma-prostate-tissue-coverage"><i class="fa fa-check"></i><b>12.4.4</b> Adenocarcinoma prostate tissue coverage</a></li>
<li class="chapter" data-level="12.4.5" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#showing-the-data-strucutre-for-the-inidividual-objects"><i class="fa fa-check"></i><b>12.4.5</b> Showing the data strucutre for the inidividual objects</a></li>
</ul></li>
<li class="chapter" data-level="12.5" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#join-giotto-objects"><i class="fa fa-check"></i><b>12.5</b> Join Giotto Objects</a></li>
<li class="chapter" data-level="12.6" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#visualizing-combined-datasets"><i class="fa fa-check"></i><b>12.6</b> Visualizing combined datasets</a>
<ul>
<li class="chapter" data-level="12.6.1" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#vizualizing-in-the-same-plot"><i class="fa fa-check"></i><b>12.6.1</b> Vizualizing in the same plot</a></li>
<li class="chapter" data-level="12.6.2" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#vizualizing-on-separate-plots"><i class="fa fa-check"></i><b>12.6.2</b> Vizualizing on separate plots</a></li>
</ul></li>
<li class="chapter" data-level="12.7" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#splitting-combined-dataset"><i class="fa fa-check"></i><b>12.7</b> Splitting combined dataset</a></li>
<li class="chapter" data-level="12.8" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#analyzing-joined-objects"><i class="fa fa-check"></i><b>12.8</b> Analyzing joined objects</a>
<ul>
<li class="chapter" data-level="12.8.1" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#normalization-and-adding-statistics"><i class="fa fa-check"></i><b>12.8.1</b> Normalization and adding statistics</a></li>
<li class="chapter" data-level="12.8.2" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#clustering-the-datasets"><i class="fa fa-check"></i><b>12.8.2</b> Clustering the datasets</a></li>
<li class="chapter" data-level="12.8.3" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#vizualizing-spatial-location-of-clusters"><i class="fa fa-check"></i><b>12.8.3</b> Vizualizing spatial location of clusters</a></li>
<li class="chapter" data-level="12.8.4" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#vizualizing-tissue-contribution-to-clusters"><i class="fa fa-check"></i><b>12.8.4</b> Vizualizing tissue contribution to clusters</a></li>
</ul></li>
<li class="chapter" data-level="12.9" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#perform-harmony-and-default-workflows"><i class="fa fa-check"></i><b>12.9</b> Perform Harmony and default workflows</a>
<ul>
<li class="chapter" data-level="12.9.1" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#clustering-harmonized-object"><i class="fa fa-check"></i><b>12.9.1</b> Clustering harmonized object</a></li>
<li class="chapter" data-level="12.9.2" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#vizualizing-the-tissue-contribution"><i class="fa fa-check"></i><b>12.9.2</b> Vizualizing the tissue contribution</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="13" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html"><i class="fa fa-check"></i><b>13</b> Spatial multi-modal analysis</a>
<ul>
<li class="chapter" data-level="13.1" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#overview-1"><i class="fa fa-check"></i><b>13.1</b> Overview</a></li>
<li class="chapter" data-level="13.2" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#spatial-manipulation"><i class="fa fa-check"></i><b>13.2</b> Spatial manipulation</a>
<ul>
<li class="chapter" data-level="13.2.1" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#spatial-transforms"><i class="fa fa-check"></i><b>13.2.1</b> Spatial transforms:</a></li>
<li class="chapter" data-level="13.2.2" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#spatial-utilities"><i class="fa fa-check"></i><b>13.2.2</b> Spatial utilities:</a></li>
<li class="chapter" data-level="13.2.3" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#spatial-classes"><i class="fa fa-check"></i><b>13.2.3</b> Spatial classes:</a></li>
</ul></li>
<li class="chapter" data-level="13.3" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#examples-of-the-simple-transforms-with-a-giottopolygon"><i class="fa fa-check"></i><b>13.3</b> Examples of the simple transforms with a giottoPolygon</a></li>
<li class="chapter" data-level="13.4" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#affine-transforms"><i class="fa fa-check"></i><b>13.4</b> Affine transforms</a></li>
<li class="chapter" data-level="13.5" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#image-transforms"><i class="fa fa-check"></i><b>13.5</b> Image transforms</a></li>
<li class="chapter" data-level="13.6" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#the-practical-usage-of-multi-modality-co-registration"><i class="fa fa-check"></i><b>13.6</b> The practical usage of multi-modality co-registration</a>
<ul>
<li class="chapter" data-level="13.6.1" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#example-dataset-xenium-breast-cancer-pre-release-pack"><i class="fa fa-check"></i><b>13.6.1</b> Example dataset: Xenium Breast Cancer pre-release pack</a></li>
<li class="chapter" data-level="13.6.2" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#target-coordinate-system"><i class="fa fa-check"></i><b>13.6.2</b> Target Coordinate system</a></li>
<li class="chapter" data-level="13.6.3" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#visium-to-register"><i class="fa fa-check"></i><b>13.6.3</b> Visium to register</a></li>
<li class="chapter" data-level="13.6.4" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#register-post-xenium-he-and-if-image"><i class="fa fa-check"></i><b>13.6.4</b> Register post-Xenium H&amp;E and IF image</a></li>
<li class="chapter" data-level="13.6.5" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#automatic-alignment-via-sift-feature-descriptor-matching-and-affine-transformation"><i class="fa fa-check"></i><b>13.6.5</b> Automatic alignment via SIFT feature descriptor matching and affine transformation</a></li>
<li class="chapter" data-level="13.6.6" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#final-notes"><i class="fa fa-check"></i><b>13.6.6</b> Final Notes</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="14" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html"><i class="fa fa-check"></i><b>14</b> Multi-omics integration</a>
<ul>
<li class="chapter" data-level="14.1" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#the-cytassist-technology"><i class="fa fa-check"></i><b>14.1</b> The CytAssist technology</a></li>
<li class="chapter" data-level="14.2" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#introduction-to-the-spatial-dataset-1"><i class="fa fa-check"></i><b>14.2</b> Introduction to the spatial dataset</a></li>
<li class="chapter" data-level="14.3" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#download-dataset-1"><i class="fa fa-check"></i><b>14.3</b> Download dataset</a></li>
<li class="chapter" data-level="14.4" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#create-the-giotto-object-1"><i class="fa fa-check"></i><b>14.4</b> Create the Giotto object</a></li>
<li class="chapter" data-level="14.5" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#subset-on-spots-that-were-covered-by-tissue-1"><i class="fa fa-check"></i><b>14.5</b> Subset on spots that were covered by tissue</a></li>
<li class="chapter" data-level="14.6" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#rna-processing"><i class="fa fa-check"></i><b>14.6</b> RNA processing</a></li>
<li class="chapter" data-level="14.7" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#protein-processing"><i class="fa fa-check"></i><b>14.7</b> Protein processing</a></li>
<li class="chapter" data-level="14.8" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#multi-omics-integration-1"><i class="fa fa-check"></i><b>14.8</b> Multi-omics integration</a></li>
<li class="chapter" data-level="14.9" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#session-info-2"><i class="fa fa-check"></i><b>14.9</b> Session info</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html"><i class="fa fa-check"></i><b>15</b> Interoperability with other frameworks</a>
<ul>
<li class="chapter" data-level="15.1" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#load-giotto-object"><i class="fa fa-check"></i><b>15.1</b> Load Giotto object</a></li>
<li class="chapter" data-level="15.2" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#seurat"><i class="fa fa-check"></i><b>15.2</b> Seurat</a>
<ul>
<li class="chapter" data-level="15.2.1" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#conversion-of-giotto-object-to-seurat-object"><i class="fa fa-check"></i><b>15.2.1</b> Conversion of Giotto Object to Seurat Object</a></li>
<li class="chapter" data-level="15.2.2" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#conversion-of-seurat-object-back-to-giotto-object"><i class="fa fa-check"></i><b>15.2.2</b> Conversion of Seurat object Back to Giotto Object</a></li>
</ul></li>
<li class="chapter" data-level="15.3" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#spatialexperiment"><i class="fa fa-check"></i><b>15.3</b> SpatialExperiment</a>
<ul>
<li class="chapter" data-level="15.3.1" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#convert-giotto-object-to-spatialexperiment-object"><i class="fa fa-check"></i><b>15.3.1</b> Convert Giotto Object to SpatialExperiment Object</a></li>
<li class="chapter" data-level="15.3.2" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#conversion-of-spatialexperiment-object-back-to-giotto"><i class="fa fa-check"></i><b>15.3.2</b> Conversion of SpatialExperiment object back to Giotto</a></li>
</ul></li>
<li class="chapter" data-level="15.4" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#anndata"><i class="fa fa-check"></i><b>15.4</b> AnnData</a>
<ul>
<li class="chapter" data-level="15.4.1" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#load-required-libraries"><i class="fa fa-check"></i><b>15.4.1</b> Load Required Libraries</a></li>
<li class="chapter" data-level="15.4.2" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#specify-path-for-results"><i class="fa fa-check"></i><b>15.4.2</b> Specify Path for Results</a></li>
<li class="chapter" data-level="15.4.3" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#giotto-to-anndata"><i class="fa fa-check"></i><b>15.4.3</b> Giotto To AnnData</a></li>
<li class="chapter" data-level="15.4.4" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#convert-anndata-to-giotto"><i class="fa fa-check"></i><b>15.4.4</b> Convert AnnData to Giotto</a></li>
</ul></li>
<li class="chapter" data-level="15.5" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#create-mini-vizgen-object"><i class="fa fa-check"></i><b>15.5</b> Create mini Vizgen object</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html"><i class="fa fa-check"></i><b>16</b> Interoperability with isolated tools</a>
<ul>
<li class="chapter" data-level="16.1" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#spatial-niche-trajectory-analysis-ontrac"><i class="fa fa-check"></i><b>16.1</b> Spatial niche trajectory analysis (ONTraC)</a>
<ul>
<li class="chapter" data-level="16.1.1" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#introduction-to-ontrac"><i class="fa fa-check"></i><b>16.1.1</b> Introduction to ONTraC</a></li>
<li class="chapter" data-level="16.1.2" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#introduction-to-merfish"><i class="fa fa-check"></i><b>16.1.2</b> Introduction to MERFISH</a></li>
<li class="chapter" data-level="16.1.3" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#settings"><i class="fa fa-check"></i><b>16.1.3</b> settings</a></li>
<li class="chapter" data-level="16.1.4" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#dataset-1"><i class="fa fa-check"></i><b>16.1.4</b> Dataset</a></li>
<li class="chapter" data-level="16.1.5" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#create-the-giotto-object-2"><i class="fa fa-check"></i><b>16.1.5</b> Create the Giotto object</a></li>
<li class="chapter" data-level="16.1.6" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#perform-spatial-niche-trajectory-analysis-using-ontrac"><i class="fa fa-check"></i><b>16.1.6</b> Perform spatial niche trajectory analysis using ONTraC</a></li>
<li class="chapter" data-level="16.1.7" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#visualization"><i class="fa fa-check"></i><b>16.1.7</b> visualization</a></li>
</ul></li>
<li class="chapter" data-level="16.2" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#session-info-3"><i class="fa fa-check"></i><b>16.2</b> Session info</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html"><i class="fa fa-check"></i><b>17</b> Interactivity with the R/Spatial ecosystem</a>
<ul>
<li class="chapter" data-level="17.1" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#kriging"><i class="fa fa-check"></i><b>17.1</b> Kriging</a>
<ul>
<li class="chapter" data-level="17.1.1" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#visium-technology-1"><i class="fa fa-check"></i><b>17.1.1</b> Visium technology</a></li>
<li class="chapter" data-level="17.1.2" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#dataset-2"><i class="fa fa-check"></i><b>17.1.2</b> Dataset</a></li>
<li class="chapter" data-level="17.1.3" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#generating-a-geojson-file-of-nuclei-location"><i class="fa fa-check"></i><b>17.1.3</b> Generating a geojson file of nuclei location</a></li>
</ul></li>
<li class="chapter" data-level="17.2" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#downloading-the-dataset"><i class="fa fa-check"></i><b>17.2</b> Downloading the dataset</a></li>
<li class="chapter" data-level="17.3" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#importing-visium-data"><i class="fa fa-check"></i><b>17.3</b> Importing visium data</a>
<ul>
<li class="chapter" data-level="17.3.1" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#identifying-spatially-organized-features"><i class="fa fa-check"></i><b>17.3.1</b> Identifying spatially organized features</a></li>
</ul></li>
<li class="chapter" data-level="17.4" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#adding-cell-polygons-to-giotto-object"><i class="fa fa-check"></i><b>17.4</b> Adding cell polygons to Giotto object</a>
<ul>
<li class="chapter" data-level="17.4.1" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#read-in-the-poly-information"><i class="fa fa-check"></i><b>17.4.1</b> Read in the poly information</a></li>
<li class="chapter" data-level="17.4.2" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#vizualizing-polygons"><i class="fa fa-check"></i><b>17.4.2</b> Vizualizing polygons</a></li>
<li class="chapter" data-level="17.4.3" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#showing-giotto-object-prior-to-polygon-addition"><i class="fa fa-check"></i><b>17.4.3</b> Showing Giotto object prior to polygon addition</a></li>
<li class="chapter" data-level="17.4.4" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#adding-polygons-to-giotto-object"><i class="fa fa-check"></i><b>17.4.4</b> Adding polygons to giotto object</a></li>
<li class="chapter" data-level="17.4.5" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#check-polygon-information"><i class="fa fa-check"></i><b>17.4.5</b> Check polygon information</a></li>
</ul></li>
<li class="chapter" data-level="17.5" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#performing-kriging"><i class="fa fa-check"></i><b>17.5</b> Performing kriging</a>
<ul>
<li class="chapter" data-level="17.5.1" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#interpolating-features"><i class="fa fa-check"></i><b>17.5.1</b> Interpolating features</a></li>
<li class="chapter" data-level="17.5.2" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#expression-overlap"><i class="fa fa-check"></i><b>17.5.2</b> Expression overlap</a></li>
</ul></li>
<li class="chapter" data-level="17.6" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#reading-in-larger-dataset"><i class="fa fa-check"></i><b>17.6</b> Reading in larger dataset</a></li>
<li class="chapter" data-level="17.7" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#analyzing-interpolated-features"><i class="fa fa-check"></i><b>17.7</b> Analyzing interpolated features</a>
<ul>
<li class="chapter" data-level="17.7.1" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#filter-and-normalization"><i class="fa fa-check"></i><b>17.7.1</b> Filter and normalization</a></li>
<li class="chapter" data-level="17.7.2" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#visualizing-gene-expression-from-interpolated-expression"><i class="fa fa-check"></i><b>17.7.2</b> Visualizing gene expression from interpolated expression</a></li>
<li class="chapter" data-level="17.7.3" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#run-pca"><i class="fa fa-check"></i><b>17.7.3</b> Run PCA</a></li>
<li class="chapter" data-level="17.7.4" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#clustering-2"><i class="fa fa-check"></i><b>17.7.4</b> Clustering</a></li>
<li class="chapter" data-level="17.7.5" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#visualizing-clustering"><i class="fa fa-check"></i><b>17.7.5</b> Visualizing clustering</a></li>
<li class="chapter" data-level="17.7.6" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#cropping-objects"><i class="fa fa-check"></i><b>17.7.6</b> Cropping objects</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="18" data-path="contributing-to-giotto.html"><a href="contributing-to-giotto.html"><i class="fa fa-check"></i><b>18</b> Contributing to Giotto</a>
<ul>
<li class="chapter" data-level="18.1" data-path="contributing-to-giotto.html"><a href="contributing-to-giotto.html#contribution-guideline"><i class="fa fa-check"></i><b>18.1</b> Contribution guideline</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Workshop: Spatial multi-omics data analysis with Giotto Suite</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="visium-part-ii" class="section level1 hasAnchor" number="8">
<h1><span class="header-section-number">8</span> Visium Part II<a href="visium-part-ii.html#visium-part-ii" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Joselyn Cristina Chávez Fuentes</p>
<p>August 6th 2024</p>
<div id="load-the-object" class="section level2 hasAnchor" number="8.1">
<h2><span class="header-section-number">8.1</span> Load the object<a href="visium-part-ii.html#load-the-object" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb107"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb107-1"><a href="visium-part-ii.html#cb107-1" tabindex="-1"></a><span class="fu">library</span>(Giotto)</span>
<span id="cb107-2"><a href="visium-part-ii.html#cb107-2" tabindex="-1"></a></span>
<span id="cb107-3"><a href="visium-part-ii.html#cb107-3" tabindex="-1"></a>visium_brain <span class="ot">&lt;-</span> <span class="fu">loadGiotto</span>(<span class="st">&quot;visium_brain_object&quot;</span>)</span></code></pre></div>
</div>
<div id="differential-expression" class="section level2 hasAnchor" number="8.2">
<h2><span class="header-section-number">8.2</span> Differential expression<a href="visium-part-ii.html#differential-expression" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="gini-markers" class="section level3 hasAnchor" number="8.2.1">
<h3><span class="header-section-number">8.2.1</span> Gini markers<a href="visium-part-ii.html#gini-markers" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The Gini method identifies genes that are very selectively expressed in a specific cluster, however not always expressed in all cells of that cluster. In other words, highly specific but not necessarily sensitive at the single-cell level.</p>
<ul>
<li>Calculate the top marker genes per cluster using the gini method.</li>
</ul>
<div class="sourceCode" id="cb108"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb108-1"><a href="visium-part-ii.html#cb108-1" tabindex="-1"></a>gini_markers <span class="ot">&lt;-</span> <span class="fu">findMarkers_one_vs_all</span>(<span class="at">gobject =</span> visium_brain,</span>
<span id="cb108-2"><a href="visium-part-ii.html#cb108-2" tabindex="-1"></a>                                       <span class="at">method =</span> <span class="st">&quot;gini&quot;</span>,</span>
<span id="cb108-3"><a href="visium-part-ii.html#cb108-3" tabindex="-1"></a>                                       <span class="at">expression_values =</span> <span class="st">&quot;normalized&quot;</span>,</span>
<span id="cb108-4"><a href="visium-part-ii.html#cb108-4" tabindex="-1"></a>                                       <span class="at">cluster_column =</span> <span class="st">&quot;leiden_clus&quot;</span>,</span>
<span id="cb108-5"><a href="visium-part-ii.html#cb108-5" tabindex="-1"></a>                                       <span class="at">min_feats =</span> <span class="dv">10</span>)</span>
<span id="cb108-6"><a href="visium-part-ii.html#cb108-6" tabindex="-1"></a></span>
<span id="cb108-7"><a href="visium-part-ii.html#cb108-7" tabindex="-1"></a>topgenes_gini <span class="ot">&lt;-</span> gini_markers[, <span class="fu">head</span>(.SD, <span class="dv">2</span>), by <span class="ot">=</span> <span class="st">&quot;cluster&quot;</span>]<span class="sc">$</span>feats</span></code></pre></div>
<ul>
<li>Visualize</li>
</ul>
<p>Plot the normalized expression distribution of the top expressed genes.</p>
<div class="sourceCode" id="cb109"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb109-1"><a href="visium-part-ii.html#cb109-1" tabindex="-1"></a><span class="fu">violinPlot</span>(visium_brain, </span>
<span id="cb109-2"><a href="visium-part-ii.html#cb109-2" tabindex="-1"></a>           <span class="at">feats =</span> <span class="fu">unique</span>(topgenes_gini), </span>
<span id="cb109-3"><a href="visium-part-ii.html#cb109-3" tabindex="-1"></a>           <span class="at">cluster_column =</span> <span class="st">&quot;leiden_clus&quot;</span>,</span>
<span id="cb109-4"><a href="visium-part-ii.html#cb109-4" tabindex="-1"></a>           <span class="at">strip_text =</span> <span class="dv">6</span>, </span>
<span id="cb109-5"><a href="visium-part-ii.html#cb109-5" tabindex="-1"></a>           <span class="at">strip_position =</span> <span class="st">&quot;right&quot;</span>,</span>
<span id="cb109-6"><a href="visium-part-ii.html#cb109-6" tabindex="-1"></a>           <span class="at">save_param =</span> <span class="fu">list</span>(<span class="at">base_width =</span> <span class="dv">5</span>, <span class="at">base_height =</span> <span class="dv">30</span>))</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-129"></span>
<img src="img/02_session1/0-violinPlot.png" alt="Violin plot showing the top gini genes normalized expression." width="50%" />
<p class="caption">
Figure 8.1: Violin plot showing the top gini genes normalized expression.
</p>
</div>
<p>Use the cluster IDs to create a heatmap with the normalized expression of the top expressed genes per cluster.</p>
<div class="sourceCode" id="cb110"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb110-1"><a href="visium-part-ii.html#cb110-1" tabindex="-1"></a><span class="fu">plotMetaDataHeatmap</span>(visium_brain, </span>
<span id="cb110-2"><a href="visium-part-ii.html#cb110-2" tabindex="-1"></a>                    <span class="at">selected_feats =</span> <span class="fu">unique</span>(topgenes_gini),</span>
<span id="cb110-3"><a href="visium-part-ii.html#cb110-3" tabindex="-1"></a>                    <span class="at">metadata_cols =</span> <span class="st">&quot;leiden_clus&quot;</span>,</span>
<span id="cb110-4"><a href="visium-part-ii.html#cb110-4" tabindex="-1"></a>                    <span class="at">x_text_size =</span> <span class="dv">10</span>, <span class="at">y_text_size =</span> <span class="dv">10</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-131"></span>
<img src="img/02_session1/1-plotMetaDataHeatmap.png" alt="Heatmap showing the top gini genes normalized expression per Leiden cluster." width="80%" />
<p class="caption">
Figure 8.2: Heatmap showing the top gini genes normalized expression per Leiden cluster.
</p>
</div>
<p>Visualize the scaled expression spatial distribution of the top expressed genes across the sample.</p>
<div class="sourceCode" id="cb111"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb111-1"><a href="visium-part-ii.html#cb111-1" tabindex="-1"></a><span class="fu">dimFeatPlot2D</span>(visium_brain, </span>
<span id="cb111-2"><a href="visium-part-ii.html#cb111-2" tabindex="-1"></a>              <span class="at">expression_values =</span> <span class="st">&quot;scaled&quot;</span>,</span>
<span id="cb111-3"><a href="visium-part-ii.html#cb111-3" tabindex="-1"></a>              <span class="at">feats =</span> <span class="fu">sort</span>(<span class="fu">unique</span>(topgenes_gini)),</span>
<span id="cb111-4"><a href="visium-part-ii.html#cb111-4" tabindex="-1"></a>              <span class="at">cow_n_col =</span> <span class="dv">5</span>, </span>
<span id="cb111-5"><a href="visium-part-ii.html#cb111-5" tabindex="-1"></a>              <span class="at">point_size =</span> <span class="dv">1</span>,</span>
<span id="cb111-6"><a href="visium-part-ii.html#cb111-6" tabindex="-1"></a>              <span class="at">save_param =</span> <span class="fu">list</span>(<span class="at">base_width =</span> <span class="dv">15</span>, <span class="at">base_height =</span> <span class="dv">20</span>))</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-133"></span>
<img src="img/02_session1/2-dimFeatPlot2D.png" alt="Spatial distribution of the top gini genes scaled expression." width="100%" />
<p class="caption">
Figure 8.3: Spatial distribution of the top gini genes scaled expression.
</p>
</div>
</div>
<div id="scran-markers" class="section level3 hasAnchor" number="8.2.2">
<h3><span class="header-section-number">8.2.2</span> Scran markers<a href="visium-part-ii.html#scran-markers" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The Scran method is preferred for robust differential expression analysis, especially when addressing technical variability or differences in sequencing depth across spatial locations. [redo]</p>
<ul>
<li>Calculate the top marker genes per cluster using the scran method</li>
</ul>
<div class="sourceCode" id="cb112"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb112-1"><a href="visium-part-ii.html#cb112-1" tabindex="-1"></a>scran_markers <span class="ot">&lt;-</span> <span class="fu">findMarkers_one_vs_all</span>(<span class="at">gobject =</span> visium_brain,</span>
<span id="cb112-2"><a href="visium-part-ii.html#cb112-2" tabindex="-1"></a>                                       <span class="at">method =</span> <span class="st">&quot;scran&quot;</span>,</span>
<span id="cb112-3"><a href="visium-part-ii.html#cb112-3" tabindex="-1"></a>                                       <span class="at">expression_values =</span> <span class="st">&quot;normalized&quot;</span>,</span>
<span id="cb112-4"><a href="visium-part-ii.html#cb112-4" tabindex="-1"></a>                                       <span class="at">cluster_column =</span> <span class="st">&quot;leiden_clus&quot;</span>,</span>
<span id="cb112-5"><a href="visium-part-ii.html#cb112-5" tabindex="-1"></a>                                       <span class="at">min_feats =</span> <span class="dv">10</span>)</span>
<span id="cb112-6"><a href="visium-part-ii.html#cb112-6" tabindex="-1"></a></span>
<span id="cb112-7"><a href="visium-part-ii.html#cb112-7" tabindex="-1"></a>topgenes_scran <span class="ot">&lt;-</span> scran_markers[, <span class="fu">head</span>(.SD, <span class="dv">2</span>), by <span class="ot">=</span> <span class="st">&quot;cluster&quot;</span>]<span class="sc">$</span>feats</span></code></pre></div>
<ul>
<li>Visualize</li>
</ul>
<p>Plot the normalized expression distribution of the top expressed genes.</p>
<div class="sourceCode" id="cb113"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb113-1"><a href="visium-part-ii.html#cb113-1" tabindex="-1"></a><span class="fu">violinPlot</span>(visium_brain, </span>
<span id="cb113-2"><a href="visium-part-ii.html#cb113-2" tabindex="-1"></a>           <span class="at">feats =</span> <span class="fu">unique</span>(topgenes_scran), </span>
<span id="cb113-3"><a href="visium-part-ii.html#cb113-3" tabindex="-1"></a>           <span class="at">cluster_column =</span> <span class="st">&quot;leiden_clus&quot;</span>,</span>
<span id="cb113-4"><a href="visium-part-ii.html#cb113-4" tabindex="-1"></a>           <span class="at">strip_text =</span> <span class="dv">6</span>, </span>
<span id="cb113-5"><a href="visium-part-ii.html#cb113-5" tabindex="-1"></a>           <span class="at">strip_position =</span> <span class="st">&quot;right&quot;</span>,</span>
<span id="cb113-6"><a href="visium-part-ii.html#cb113-6" tabindex="-1"></a>           <span class="at">save_param =</span> <span class="fu">list</span>(<span class="at">base_width =</span> <span class="dv">5</span>, <span class="at">base_height =</span> <span class="dv">30</span>))</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-136"></span>
<img src="img/02_session1/3-violinPlot.png" alt="Violin plot of the top scran genes normalized expression." width="50%" />
<p class="caption">
Figure 8.4: Violin plot of the top scran genes normalized expression.
</p>
</div>
<p>Use the cluster IDs to create a heatmap with the normalized expression of the top expressed genes per cluster.</p>
<div class="sourceCode" id="cb114"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb114-1"><a href="visium-part-ii.html#cb114-1" tabindex="-1"></a><span class="fu">plotMetaDataHeatmap</span>(visium_brain, </span>
<span id="cb114-2"><a href="visium-part-ii.html#cb114-2" tabindex="-1"></a>                    <span class="at">selected_feats =</span> <span class="fu">unique</span>(topgenes_scran),</span>
<span id="cb114-3"><a href="visium-part-ii.html#cb114-3" tabindex="-1"></a>                    <span class="at">metadata_cols =</span> <span class="st">&quot;leiden_clus&quot;</span>,</span>
<span id="cb114-4"><a href="visium-part-ii.html#cb114-4" tabindex="-1"></a>                    <span class="at">x_text_size =</span> <span class="dv">10</span>, <span class="at">y_text_size =</span> <span class="dv">10</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-138"></span>
<img src="img/02_session1/4-plotMetaDataHeatmap.png" alt="Heatmap showing the top scran genes normalized expression per Leiden cluster." width="100%" />
<p class="caption">
Figure 8.5: Heatmap showing the top scran genes normalized expression per Leiden cluster.
</p>
</div>
<p>Visualize the scaled expression spatial distribution of the top expressed genes across the sample.</p>
<div class="sourceCode" id="cb115"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb115-1"><a href="visium-part-ii.html#cb115-1" tabindex="-1"></a><span class="fu">dimFeatPlot2D</span>(visium_brain, </span>
<span id="cb115-2"><a href="visium-part-ii.html#cb115-2" tabindex="-1"></a>              <span class="at">expression_values =</span> <span class="st">&quot;scaled&quot;</span>,</span>
<span id="cb115-3"><a href="visium-part-ii.html#cb115-3" tabindex="-1"></a>              <span class="at">feats =</span> <span class="fu">sort</span>(<span class="fu">unique</span>(topgenes_scran)),</span>
<span id="cb115-4"><a href="visium-part-ii.html#cb115-4" tabindex="-1"></a>              <span class="at">cow_n_col =</span> <span class="dv">5</span>, </span>
<span id="cb115-5"><a href="visium-part-ii.html#cb115-5" tabindex="-1"></a>              <span class="at">point_size =</span> <span class="dv">1</span>,</span>
<span id="cb115-6"><a href="visium-part-ii.html#cb115-6" tabindex="-1"></a>              <span class="at">save_param =</span> <span class="fu">list</span>(<span class="at">base_width =</span> <span class="dv">20</span>, <span class="at">base_height =</span> <span class="dv">20</span>))</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-140"></span>
<img src="img/02_session1/5-dimFeatPlot2D.png" alt="Spatial distribution of the top scran genes scaled expression." width="100%" />
<p class="caption">
Figure 8.6: Spatial distribution of the top scran genes scaled expression.
</p>
</div>
<p>In practice, it is often beneficial to apply both Gini and Scran methods and compare results for a more complete understanding of differential gene expression across clusters.</p>
</div>
</div>
<div id="enrichment-deconvolution" class="section level2 hasAnchor" number="8.3">
<h2><span class="header-section-number">8.3</span> Enrichment &amp; Deconvolution<a href="visium-part-ii.html#enrichment-deconvolution" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Visium spatial transcriptomics does not provide single-cell resolution, making cell type annotation a harder problem. Giotto provides several ways to calculate enrichment of specific cell-type signature gene lists.</p>
<ul>
<li>Download the single-cell dataset</li>
</ul>
<div class="sourceCode" id="cb116"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb116-1"><a href="visium-part-ii.html#cb116-1" tabindex="-1"></a>GiottoData<span class="sc">::</span><span class="fu">getSpatialDataset</span>(<span class="at">dataset =</span> <span class="st">&quot;scRNA_mouse_brain&quot;</span>, </span>
<span id="cb116-2"><a href="visium-part-ii.html#cb116-2" tabindex="-1"></a>                              <span class="at">directory =</span> <span class="st">&quot;data/02_session1/&quot;</span>)</span></code></pre></div>
<ul>
<li>Create the single-cell object and run the normalization step</li>
</ul>
<div class="sourceCode" id="cb117"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb117-1"><a href="visium-part-ii.html#cb117-1" tabindex="-1"></a>results_folder <span class="ot">&lt;-</span> <span class="st">&quot;results/02_session1/&quot;</span></span>
<span id="cb117-2"><a href="visium-part-ii.html#cb117-2" tabindex="-1"></a></span>
<span id="cb117-3"><a href="visium-part-ii.html#cb117-3" tabindex="-1"></a>python_path <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb117-4"><a href="visium-part-ii.html#cb117-4" tabindex="-1"></a></span>
<span id="cb117-5"><a href="visium-part-ii.html#cb117-5" tabindex="-1"></a>instructions <span class="ot">&lt;-</span> <span class="fu">createGiottoInstructions</span>(</span>
<span id="cb117-6"><a href="visium-part-ii.html#cb117-6" tabindex="-1"></a>    <span class="at">save_dir =</span> results_folder,</span>
<span id="cb117-7"><a href="visium-part-ii.html#cb117-7" tabindex="-1"></a>    <span class="at">save_plot =</span> <span class="cn">TRUE</span>,</span>
<span id="cb117-8"><a href="visium-part-ii.html#cb117-8" tabindex="-1"></a>    <span class="at">show_plot =</span> <span class="cn">FALSE</span>,</span>
<span id="cb117-9"><a href="visium-part-ii.html#cb117-9" tabindex="-1"></a>    <span class="at">python_path =</span> python_path</span>
<span id="cb117-10"><a href="visium-part-ii.html#cb117-10" tabindex="-1"></a>)</span>
<span id="cb117-11"><a href="visium-part-ii.html#cb117-11" tabindex="-1"></a></span>
<span id="cb117-12"><a href="visium-part-ii.html#cb117-12" tabindex="-1"></a>sc_expression <span class="ot">&lt;-</span> <span class="st">&quot;data/02_session1/brain_sc_expression_matrix.txt.gz&quot;</span></span>
<span id="cb117-13"><a href="visium-part-ii.html#cb117-13" tabindex="-1"></a>sc_metadata <span class="ot">&lt;-</span> <span class="st">&quot;data/02_session1/brain_sc_metadata.csv&quot;</span></span>
<span id="cb117-14"><a href="visium-part-ii.html#cb117-14" tabindex="-1"></a></span>
<span id="cb117-15"><a href="visium-part-ii.html#cb117-15" tabindex="-1"></a>giotto_SC <span class="ot">&lt;-</span> <span class="fu">createGiottoObject</span>(<span class="at">expression =</span> sc_expression,</span>
<span id="cb117-16"><a href="visium-part-ii.html#cb117-16" tabindex="-1"></a>                                <span class="at">instructions =</span> instructions)</span>
<span id="cb117-17"><a href="visium-part-ii.html#cb117-17" tabindex="-1"></a></span>
<span id="cb117-18"><a href="visium-part-ii.html#cb117-18" tabindex="-1"></a>giotto_SC <span class="ot">&lt;-</span> <span class="fu">addCellMetadata</span>(giotto_SC, </span>
<span id="cb117-19"><a href="visium-part-ii.html#cb117-19" tabindex="-1"></a>                             <span class="at">new_metadata =</span> data.table<span class="sc">::</span><span class="fu">fread</span>(sc_metadata))</span>
<span id="cb117-20"><a href="visium-part-ii.html#cb117-20" tabindex="-1"></a></span>
<span id="cb117-21"><a href="visium-part-ii.html#cb117-21" tabindex="-1"></a>giotto_SC <span class="ot">&lt;-</span> <span class="fu">normalizeGiotto</span>(giotto_SC)</span></code></pre></div>
<div id="pagerank" class="section level3 hasAnchor" number="8.3.1">
<h3><span class="header-section-number">8.3.1</span> PAGE/Rank<a href="visium-part-ii.html#pagerank" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Parametric Analysis of Gene Set Enrichment (PAGE) and Rank enrichment both aim to determine whether a predefined set of genes show statistically significant differences in expression compared to other genes in the dataset.</p>
<ul>
<li>Calculate the cell type markers</li>
</ul>
<div class="sourceCode" id="cb118"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb118-1"><a href="visium-part-ii.html#cb118-1" tabindex="-1"></a>markers_scran <span class="ot">&lt;-</span> <span class="fu">findMarkers_one_vs_all</span>(<span class="at">gobject =</span> giotto_SC, </span>
<span id="cb118-2"><a href="visium-part-ii.html#cb118-2" tabindex="-1"></a>                                        <span class="at">method =</span> <span class="st">&quot;scran&quot;</span>,</span>
<span id="cb118-3"><a href="visium-part-ii.html#cb118-3" tabindex="-1"></a>                                        <span class="at">expression_values =</span> <span class="st">&quot;normalized&quot;</span>,</span>
<span id="cb118-4"><a href="visium-part-ii.html#cb118-4" tabindex="-1"></a>                                        <span class="at">cluster_column =</span> <span class="st">&quot;Class&quot;</span>, </span>
<span id="cb118-5"><a href="visium-part-ii.html#cb118-5" tabindex="-1"></a>                                        <span class="at">min_feats =</span> <span class="dv">3</span>)</span>
<span id="cb118-6"><a href="visium-part-ii.html#cb118-6" tabindex="-1"></a></span>
<span id="cb118-7"><a href="visium-part-ii.html#cb118-7" tabindex="-1"></a>top_markers <span class="ot">&lt;-</span> markers_scran[, <span class="fu">head</span>(.SD, <span class="dv">10</span>), by <span class="ot">=</span> <span class="st">&quot;cluster&quot;</span>]</span>
<span id="cb118-8"><a href="visium-part-ii.html#cb118-8" tabindex="-1"></a>celltypes <span class="ot">&lt;-</span> <span class="fu">levels</span>(<span class="fu">factor</span>(markers_scran<span class="sc">$</span>cluster)) </span></code></pre></div>
<ul>
<li>Create the signature matrix</li>
</ul>
<div class="sourceCode" id="cb119"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb119-1"><a href="visium-part-ii.html#cb119-1" tabindex="-1"></a>sign_list <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb119-2"><a href="visium-part-ii.html#cb119-2" tabindex="-1"></a></span>
<span id="cb119-3"><a href="visium-part-ii.html#cb119-3" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(celltypes)){</span>
<span id="cb119-4"><a href="visium-part-ii.html#cb119-4" tabindex="-1"></a>  sign_list[[i]] <span class="ot">=</span> top_markers[<span class="fu">which</span>(top_markers<span class="sc">$</span>cluster <span class="sc">==</span> celltypes[i]),]<span class="sc">$</span>feats</span>
<span id="cb119-5"><a href="visium-part-ii.html#cb119-5" tabindex="-1"></a>}</span>
<span id="cb119-6"><a href="visium-part-ii.html#cb119-6" tabindex="-1"></a></span>
<span id="cb119-7"><a href="visium-part-ii.html#cb119-7" tabindex="-1"></a>sign_matrix <span class="ot">&lt;-</span> <span class="fu">makeSignMatrixPAGE</span>(<span class="at">sign_names =</span> celltypes,</span>
<span id="cb119-8"><a href="visium-part-ii.html#cb119-8" tabindex="-1"></a>                                  <span class="at">sign_list =</span> sign_list)</span></code></pre></div>
<ul>
<li>Run the enrichment test with PAGE</li>
</ul>
<div class="sourceCode" id="cb120"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb120-1"><a href="visium-part-ii.html#cb120-1" tabindex="-1"></a>visium_brain <span class="ot">&lt;-</span> <span class="fu">runPAGEEnrich</span>(<span class="at">gobject =</span> visium_brain, </span>
<span id="cb120-2"><a href="visium-part-ii.html#cb120-2" tabindex="-1"></a>                              <span class="at">sign_matrix =</span> sign_matrix)</span></code></pre></div>
<ul>
<li>Visualize</li>
</ul>
<p>Create a heatmap showing the enrichment of cell types (from the single-cell data annotation) in the spatial dataset clusters.</p>
<div class="sourceCode" id="cb121"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb121-1"><a href="visium-part-ii.html#cb121-1" tabindex="-1"></a>cell_types_PAGE <span class="ot">&lt;-</span> <span class="fu">colnames</span>(sign_matrix)</span>
<span id="cb121-2"><a href="visium-part-ii.html#cb121-2" tabindex="-1"></a></span>
<span id="cb121-3"><a href="visium-part-ii.html#cb121-3" tabindex="-1"></a><span class="fu">plotMetaDataCellsHeatmap</span>(<span class="at">gobject =</span> visium_brain,</span>
<span id="cb121-4"><a href="visium-part-ii.html#cb121-4" tabindex="-1"></a>                         <span class="at">metadata_cols =</span> <span class="st">&quot;leiden_clus&quot;</span>,</span>
<span id="cb121-5"><a href="visium-part-ii.html#cb121-5" tabindex="-1"></a>                         <span class="at">value_cols =</span> cell_types_PAGE,</span>
<span id="cb121-6"><a href="visium-part-ii.html#cb121-6" tabindex="-1"></a>                         <span class="at">spat_enr_names =</span> <span class="st">&quot;PAGE&quot;</span>,</span>
<span id="cb121-7"><a href="visium-part-ii.html#cb121-7" tabindex="-1"></a>                         <span class="at">x_text_size =</span> <span class="dv">8</span>,</span>
<span id="cb121-8"><a href="visium-part-ii.html#cb121-8" tabindex="-1"></a>                         <span class="at">y_text_size =</span> <span class="dv">8</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-147"></span>
<img src="img/02_session1/6-plotMetaDataCellsHeatmap.png" alt="Cell types enrichment per Leiden cluster, identified using the PAGE method." width="100%" />
<p class="caption">
Figure 8.7: Cell types enrichment per Leiden cluster, identified using the PAGE method.
</p>
</div>
<p>Plot the spatial distribution of the cell types.</p>
<div class="sourceCode" id="cb122"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb122-1"><a href="visium-part-ii.html#cb122-1" tabindex="-1"></a><span class="fu">spatCellPlot2D</span>(<span class="at">gobject =</span> visium_brain,</span>
<span id="cb122-2"><a href="visium-part-ii.html#cb122-2" tabindex="-1"></a>               <span class="at">spat_enr_names =</span> <span class="st">&quot;PAGE&quot;</span>,</span>
<span id="cb122-3"><a href="visium-part-ii.html#cb122-3" tabindex="-1"></a>               <span class="at">cell_annotation_values =</span> cell_types_PAGE,</span>
<span id="cb122-4"><a href="visium-part-ii.html#cb122-4" tabindex="-1"></a>               <span class="at">cow_n_col =</span> <span class="dv">3</span>,</span>
<span id="cb122-5"><a href="visium-part-ii.html#cb122-5" tabindex="-1"></a>               <span class="at">coord_fix_ratio =</span> <span class="dv">1</span>, </span>
<span id="cb122-6"><a href="visium-part-ii.html#cb122-6" tabindex="-1"></a>               <span class="at">point_size =</span> <span class="dv">1</span>, </span>
<span id="cb122-7"><a href="visium-part-ii.html#cb122-7" tabindex="-1"></a>               <span class="at">show_legend =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-149"></span>
<img src="img/02_session1/7-spatCellPlot2D.png" alt="Spatial distribution of cell types identified using the PAGE method." width="100%" />
<p class="caption">
Figure 8.8: Spatial distribution of cell types identified using the PAGE method.
</p>
</div>
</div>
<div id="spatialdwls" class="section level3 hasAnchor" number="8.3.2">
<h3><span class="header-section-number">8.3.2</span> SpatialDWLS<a href="visium-part-ii.html#spatialdwls" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Spatial Dampened Weighted Least Squares (DWLS) estimates the proportions of different cell types across spots in a tissue.</p>
<ul>
<li>Create the signature matrix</li>
</ul>
<div class="sourceCode" id="cb123"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb123-1"><a href="visium-part-ii.html#cb123-1" tabindex="-1"></a>sign_matrix <span class="ot">&lt;-</span> <span class="fu">makeSignMatrixDWLSfromMatrix</span>(</span>
<span id="cb123-2"><a href="visium-part-ii.html#cb123-2" tabindex="-1"></a>    <span class="at">matrix =</span> <span class="fu">getExpression</span>(giotto_SC,</span>
<span id="cb123-3"><a href="visium-part-ii.html#cb123-3" tabindex="-1"></a>                           <span class="at">values =</span> <span class="st">&quot;normalized&quot;</span>,</span>
<span id="cb123-4"><a href="visium-part-ii.html#cb123-4" tabindex="-1"></a>                           <span class="at">output =</span> <span class="st">&quot;matrix&quot;</span>),</span>
<span id="cb123-5"><a href="visium-part-ii.html#cb123-5" tabindex="-1"></a>    <span class="at">cell_type =</span> <span class="fu">pDataDT</span>(giotto_SC)<span class="sc">$</span>Class,</span>
<span id="cb123-6"><a href="visium-part-ii.html#cb123-6" tabindex="-1"></a>    <span class="at">sign_gene =</span> top_markers<span class="sc">$</span>feats)</span></code></pre></div>
<ul>
<li>Run the DWLS Deconvolution</li>
</ul>
<p>This step may take a couple of minutes to run.</p>
<div class="sourceCode" id="cb124"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb124-1"><a href="visium-part-ii.html#cb124-1" tabindex="-1"></a>visium_brain <span class="ot">&lt;-</span> <span class="fu">runDWLSDeconv</span>(<span class="at">gobject =</span> visium_brain, </span>
<span id="cb124-2"><a href="visium-part-ii.html#cb124-2" tabindex="-1"></a>                              <span class="at">sign_matrix =</span> sign_matrix)</span></code></pre></div>
<ul>
<li>Visualize</li>
</ul>
<p>Plot the DWLS deconvolution result creating with pie plots showing the proportion of each cell type per spot.</p>
<div class="sourceCode" id="cb125"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb125-1"><a href="visium-part-ii.html#cb125-1" tabindex="-1"></a><span class="fu">spatDeconvPlot</span>(visium_brain, </span>
<span id="cb125-2"><a href="visium-part-ii.html#cb125-2" tabindex="-1"></a>               <span class="at">show_image =</span> <span class="cn">FALSE</span>,</span>
<span id="cb125-3"><a href="visium-part-ii.html#cb125-3" tabindex="-1"></a>               <span class="at">radius =</span> <span class="dv">50</span>,</span>
<span id="cb125-4"><a href="visium-part-ii.html#cb125-4" tabindex="-1"></a>               <span class="at">save_param =</span> <span class="fu">list</span>(<span class="at">save_name =</span> <span class="st">&quot;8_spat_DWLS_pie_plot&quot;</span>))</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-153"></span>
<img src="img/02_session1/8_spat_DWLS_pie_plot.png" alt="Spatial deconvolution plot showing the proportion of cell types per spot, identified using the DWLS method." width="100%" />
<p class="caption">
Figure 8.9: Spatial deconvolution plot showing the proportion of cell types per spot, identified using the DWLS method.
</p>
</div>
</div>
</div>
<div id="spatial-expression-patterns" class="section level2 hasAnchor" number="8.4">
<h2><span class="header-section-number">8.4</span> Spatial expression patterns<a href="visium-part-ii.html#spatial-expression-patterns" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="spatial-variable-genes" class="section level3 hasAnchor" number="8.4.1">
<h3><span class="header-section-number">8.4.1</span> Spatial variable genes<a href="visium-part-ii.html#spatial-variable-genes" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li>Create a spatial network</li>
</ul>
<div class="sourceCode" id="cb126"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb126-1"><a href="visium-part-ii.html#cb126-1" tabindex="-1"></a>visium_brain <span class="ot">&lt;-</span> <span class="fu">createSpatialNetwork</span>(<span class="at">gobject =</span> visium_brain,</span>
<span id="cb126-2"><a href="visium-part-ii.html#cb126-2" tabindex="-1"></a>                                     <span class="at">method =</span> <span class="st">&quot;kNN&quot;</span>, </span>
<span id="cb126-3"><a href="visium-part-ii.html#cb126-3" tabindex="-1"></a>                                     <span class="at">k =</span> <span class="dv">6</span>,</span>
<span id="cb126-4"><a href="visium-part-ii.html#cb126-4" tabindex="-1"></a>                                     <span class="at">maximum_distance_knn =</span> <span class="dv">400</span>,</span>
<span id="cb126-5"><a href="visium-part-ii.html#cb126-5" tabindex="-1"></a>                                     <span class="at">name =</span> <span class="st">&quot;spatial_network&quot;</span>)</span>
<span id="cb126-6"><a href="visium-part-ii.html#cb126-6" tabindex="-1"></a></span>
<span id="cb126-7"><a href="visium-part-ii.html#cb126-7" tabindex="-1"></a><span class="fu">spatPlot2D</span>(<span class="at">gobject =</span> visium_brain,  </span>
<span id="cb126-8"><a href="visium-part-ii.html#cb126-8" tabindex="-1"></a>           <span class="at">show_network=</span> <span class="cn">TRUE</span>,</span>
<span id="cb126-9"><a href="visium-part-ii.html#cb126-9" tabindex="-1"></a>           <span class="at">network_color =</span> <span class="st">&quot;blue&quot;</span>, </span>
<span id="cb126-10"><a href="visium-part-ii.html#cb126-10" tabindex="-1"></a>           <span class="at">spatial_network_name =</span> <span class="st">&quot;spatial_network&quot;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-155"></span>
<img src="img/02_session1/9-spatPlot2D.png" alt="Spatial network across spots in the Visium mouse sample." width="100%" />
<p class="caption">
Figure 8.10: Spatial network across spots in the Visium mouse sample.
</p>
</div>
<ul>
<li>Rank binarization</li>
</ul>
<p>Rank the genes on the spatial dataset depending on whether they exhibit a spatial pattern location or not.</p>
<p>This step may take a few minutes to run.</p>
<div class="sourceCode" id="cb127"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb127-1"><a href="visium-part-ii.html#cb127-1" tabindex="-1"></a>ranktest <span class="ot">&lt;-</span> <span class="fu">binSpect</span>(visium_brain, </span>
<span id="cb127-2"><a href="visium-part-ii.html#cb127-2" tabindex="-1"></a>                     <span class="at">bin_method =</span> <span class="st">&quot;rank&quot;</span>,</span>
<span id="cb127-3"><a href="visium-part-ii.html#cb127-3" tabindex="-1"></a>                     <span class="at">calc_hub =</span> <span class="cn">TRUE</span>, </span>
<span id="cb127-4"><a href="visium-part-ii.html#cb127-4" tabindex="-1"></a>                     <span class="at">hub_min_int =</span> <span class="dv">5</span>,</span>
<span id="cb127-5"><a href="visium-part-ii.html#cb127-5" tabindex="-1"></a>                     <span class="at">spatial_network_name =</span> <span class="st">&quot;spatial_network&quot;</span>)</span></code></pre></div>
<ul>
<li>Visualize top results</li>
</ul>
<p>Plot the scaled expression of genes with the highest probability of being spatial genes.</p>
<div class="sourceCode" id="cb128"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb128-1"><a href="visium-part-ii.html#cb128-1" tabindex="-1"></a><span class="fu">spatFeatPlot2D</span>(visium_brain, </span>
<span id="cb128-2"><a href="visium-part-ii.html#cb128-2" tabindex="-1"></a>               <span class="at">expression_values =</span> <span class="st">&quot;scaled&quot;</span>,</span>
<span id="cb128-3"><a href="visium-part-ii.html#cb128-3" tabindex="-1"></a>               <span class="at">feats =</span> ranktest<span class="sc">$</span>feats[<span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>], </span>
<span id="cb128-4"><a href="visium-part-ii.html#cb128-4" tabindex="-1"></a>               <span class="at">cow_n_col =</span> <span class="dv">2</span>, </span>
<span id="cb128-5"><a href="visium-part-ii.html#cb128-5" tabindex="-1"></a>               <span class="at">point_size =</span> <span class="dv">1</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-158"></span>
<img src="img/02_session1/10-spatFeatPlot2D.png" alt="Spatial distribution of the top spatial genes scaled expression." width="100%" />
<p class="caption">
Figure 8.11: Spatial distribution of the top spatial genes scaled expression.
</p>
</div>
</div>
<div id="spatial-co-expression-modules" class="section level3 hasAnchor" number="8.4.2">
<h3><span class="header-section-number">8.4.2</span> Spatial co-expression modules<a href="visium-part-ii.html#spatial-co-expression-modules" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li>Cluster the top 500 spatial genes into 20 clusters</li>
</ul>
<div class="sourceCode" id="cb129"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb129-1"><a href="visium-part-ii.html#cb129-1" tabindex="-1"></a>ext_spatial_genes <span class="ot">&lt;-</span> ranktest[<span class="dv">1</span><span class="sc">:</span><span class="dv">500</span>,]<span class="sc">$</span>feats</span></code></pre></div>
<ul>
<li>Use detectSpatialCorGenes function to calculate pairwise distances between genes.</li>
</ul>
<div class="sourceCode" id="cb130"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb130-1"><a href="visium-part-ii.html#cb130-1" tabindex="-1"></a>spat_cor_netw_DT <span class="ot">&lt;-</span> <span class="fu">detectSpatialCorFeats</span>(</span>
<span id="cb130-2"><a href="visium-part-ii.html#cb130-2" tabindex="-1"></a>    visium_brain,</span>
<span id="cb130-3"><a href="visium-part-ii.html#cb130-3" tabindex="-1"></a>    <span class="at">method =</span> <span class="st">&quot;network&quot;</span>,</span>
<span id="cb130-4"><a href="visium-part-ii.html#cb130-4" tabindex="-1"></a>    <span class="at">spatial_network_name =</span> <span class="st">&quot;spatial_network&quot;</span>,</span>
<span id="cb130-5"><a href="visium-part-ii.html#cb130-5" tabindex="-1"></a>    <span class="at">subset_feats =</span> ext_spatial_genes)</span></code></pre></div>
<ul>
<li>Identify most similar spatially correlated genes for one gene</li>
</ul>
<div class="sourceCode" id="cb131"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb131-1"><a href="visium-part-ii.html#cb131-1" tabindex="-1"></a>top10_genes <span class="ot">&lt;-</span> <span class="fu">showSpatialCorFeats</span>(spat_cor_netw_DT, </span>
<span id="cb131-2"><a href="visium-part-ii.html#cb131-2" tabindex="-1"></a>                                  <span class="at">feats =</span> <span class="st">&quot;Mbp&quot;</span>, </span>
<span id="cb131-3"><a href="visium-part-ii.html#cb131-3" tabindex="-1"></a>                                  <span class="at">show_top_feats =</span> <span class="dv">10</span>)</span></code></pre></div>
<ul>
<li>Visualize</li>
</ul>
<p>Plot the scaled expression of the 3 genes with most similar spatial patterns to Mbp.</p>
<div class="sourceCode" id="cb132"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb132-1"><a href="visium-part-ii.html#cb132-1" tabindex="-1"></a><span class="fu">spatFeatPlot2D</span>(visium_brain, </span>
<span id="cb132-2"><a href="visium-part-ii.html#cb132-2" tabindex="-1"></a>               <span class="at">expression_values =</span> <span class="st">&quot;scaled&quot;</span>,</span>
<span id="cb132-3"><a href="visium-part-ii.html#cb132-3" tabindex="-1"></a>               <span class="at">feats =</span> top10_genes<span class="sc">$</span>variable[<span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>], </span>
<span id="cb132-4"><a href="visium-part-ii.html#cb132-4" tabindex="-1"></a>               <span class="at">point_size =</span> <span class="fl">1.5</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-163"></span>
<img src="img/02_session1/11-spatFeatPlot2D.png" alt="Spatial distribution of the scaled expression of 3 genes with similar spatial pattern to Mbp." width="100%" />
<p class="caption">
Figure 8.12: Spatial distribution of the scaled expression of 3 genes with similar spatial pattern to Mbp.
</p>
</div>
<ul>
<li>Cluster spatial genes</li>
</ul>
<div class="sourceCode" id="cb133"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb133-1"><a href="visium-part-ii.html#cb133-1" tabindex="-1"></a>spat_cor_netw_DT <span class="ot">&lt;-</span> <span class="fu">clusterSpatialCorFeats</span>(spat_cor_netw_DT, </span>
<span id="cb133-2"><a href="visium-part-ii.html#cb133-2" tabindex="-1"></a>                                           <span class="at">name =</span> <span class="st">&quot;spat_netw_clus&quot;</span>, </span>
<span id="cb133-3"><a href="visium-part-ii.html#cb133-3" tabindex="-1"></a>                                           <span class="at">k =</span> <span class="dv">20</span>)</span></code></pre></div>
<ul>
<li>Visualize clusters</li>
</ul>
<p>Plot the correlation of the top 500 spatial genes with their assigned cluster.</p>
<div class="sourceCode" id="cb134"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb134-1"><a href="visium-part-ii.html#cb134-1" tabindex="-1"></a><span class="fu">heatmSpatialCorFeats</span>(visium_brain,</span>
<span id="cb134-2"><a href="visium-part-ii.html#cb134-2" tabindex="-1"></a>                     <span class="at">spatCorObject =</span> spat_cor_netw_DT,</span>
<span id="cb134-3"><a href="visium-part-ii.html#cb134-3" tabindex="-1"></a>                     <span class="at">use_clus_name =</span> <span class="st">&quot;spat_netw_clus&quot;</span>,</span>
<span id="cb134-4"><a href="visium-part-ii.html#cb134-4" tabindex="-1"></a>                     <span class="at">heatmap_legend_param =</span> <span class="fu">list</span>(<span class="at">title =</span> <span class="cn">NULL</span>))</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-166"></span>
<img src="img/02_session1/12-heatmSpatialCorFeats.png" alt="Correlations heatmap between spatial genes and correlated clusters." width="100%" />
<p class="caption">
Figure 8.13: Correlations heatmap between spatial genes and correlated clusters.
</p>
</div>
<ul>
<li>Rank spatial correlated clusters and show genes for selected clusters</li>
</ul>
<div class="sourceCode" id="cb135"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb135-1"><a href="visium-part-ii.html#cb135-1" tabindex="-1"></a>netw_ranks <span class="ot">&lt;-</span> <span class="fu">rankSpatialCorGroups</span>(</span>
<span id="cb135-2"><a href="visium-part-ii.html#cb135-2" tabindex="-1"></a>  visium_brain,</span>
<span id="cb135-3"><a href="visium-part-ii.html#cb135-3" tabindex="-1"></a>  <span class="at">spatCorObject =</span> spat_cor_netw_DT, </span>
<span id="cb135-4"><a href="visium-part-ii.html#cb135-4" tabindex="-1"></a>  <span class="at">use_clus_name =</span> <span class="st">&quot;spat_netw_clus&quot;</span>)</span></code></pre></div>
<p>Plot the correlation and number of spatial genes in each cluster.</p>
<div class="sourceCode" id="cb136"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb136-1"><a href="visium-part-ii.html#cb136-1" tabindex="-1"></a>top_netw_spat_cluster <span class="ot">&lt;-</span> <span class="fu">showSpatialCorFeats</span>(spat_cor_netw_DT, </span>
<span id="cb136-2"><a href="visium-part-ii.html#cb136-2" tabindex="-1"></a>                                            <span class="at">use_clus_name =</span> <span class="st">&quot;spat_netw_clus&quot;</span>,</span>
<span id="cb136-3"><a href="visium-part-ii.html#cb136-3" tabindex="-1"></a>                                            <span class="at">selected_clusters =</span> <span class="dv">6</span>, </span>
<span id="cb136-4"><a href="visium-part-ii.html#cb136-4" tabindex="-1"></a>                                            <span class="at">show_top_feats =</span> <span class="dv">1</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-169"></span>
<img src="img/02_session1/13-rankSpatialCorGroups.png" alt="Ranking of spatial correlated groups. Size indicates the number spatial genes per group." width="100%" />
<p class="caption">
Figure 8.14: Ranking of spatial correlated groups. Size indicates the number spatial genes per group.
</p>
</div>
<ul>
<li>Create the metagene enrichment score per co-expression cluster</li>
</ul>
<div class="sourceCode" id="cb137"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb137-1"><a href="visium-part-ii.html#cb137-1" tabindex="-1"></a>cluster_genes_DT <span class="ot">&lt;-</span> <span class="fu">showSpatialCorFeats</span>(spat_cor_netw_DT, </span>
<span id="cb137-2"><a href="visium-part-ii.html#cb137-2" tabindex="-1"></a>                                        <span class="at">use_clus_name =</span> <span class="st">&quot;spat_netw_clus&quot;</span>, </span>
<span id="cb137-3"><a href="visium-part-ii.html#cb137-3" tabindex="-1"></a>                                        <span class="at">show_top_feats =</span> <span class="dv">1</span>)</span>
<span id="cb137-4"><a href="visium-part-ii.html#cb137-4" tabindex="-1"></a></span>
<span id="cb137-5"><a href="visium-part-ii.html#cb137-5" tabindex="-1"></a>cluster_genes <span class="ot">&lt;-</span> cluster_genes_DT<span class="sc">$</span>clus </span>
<span id="cb137-6"><a href="visium-part-ii.html#cb137-6" tabindex="-1"></a><span class="fu">names</span>(cluster_genes) <span class="ot">&lt;-</span> cluster_genes_DT<span class="sc">$</span>feat_ID</span>
<span id="cb137-7"><a href="visium-part-ii.html#cb137-7" tabindex="-1"></a></span>
<span id="cb137-8"><a href="visium-part-ii.html#cb137-8" tabindex="-1"></a>visium_brain <span class="ot">&lt;-</span> <span class="fu">createMetafeats</span>(visium_brain, </span>
<span id="cb137-9"><a href="visium-part-ii.html#cb137-9" tabindex="-1"></a>                                <span class="at">feat_clusters =</span> cluster_genes, </span>
<span id="cb137-10"><a href="visium-part-ii.html#cb137-10" tabindex="-1"></a>                                <span class="at">name =</span> <span class="st">&quot;cluster_metagene&quot;</span>)</span></code></pre></div>
<p>Plot the spatial distribution of the metagene enrichment scores of each spatial co-expression cluster.</p>
<div class="sourceCode" id="cb138"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb138-1"><a href="visium-part-ii.html#cb138-1" tabindex="-1"></a><span class="fu">spatCellPlot</span>(visium_brain,</span>
<span id="cb138-2"><a href="visium-part-ii.html#cb138-2" tabindex="-1"></a>             <span class="at">spat_enr_names =</span> <span class="st">&quot;cluster_metagene&quot;</span>,</span>
<span id="cb138-3"><a href="visium-part-ii.html#cb138-3" tabindex="-1"></a>             <span class="at">cell_annotation_values =</span> netw_ranks<span class="sc">$</span>clusters,</span>
<span id="cb138-4"><a href="visium-part-ii.html#cb138-4" tabindex="-1"></a>             <span class="at">point_size =</span> <span class="dv">1</span>, </span>
<span id="cb138-5"><a href="visium-part-ii.html#cb138-5" tabindex="-1"></a>             <span class="at">cow_n_col =</span> <span class="dv">5</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-172"></span>
<img src="img/02_session1/14-spatCellPlot2D.png" alt="Spatial distribution of metagene enrichment scores per co-expression cluster." width="100%" />
<p class="caption">
Figure 8.15: Spatial distribution of metagene enrichment scores per co-expression cluster.
</p>
</div>
</div>
</div>
<div id="spatially-informed-clusters" class="section level2 hasAnchor" number="8.5">
<h2><span class="header-section-number">8.5</span> Spatially informed clusters<a href="visium-part-ii.html#spatially-informed-clusters" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ul>
<li>Get the top 30 genes per spatial co-expression cluster</li>
</ul>
<div class="sourceCode" id="cb139"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb139-1"><a href="visium-part-ii.html#cb139-1" tabindex="-1"></a>coexpr_dt <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">data.table</span>(</span>
<span id="cb139-2"><a href="visium-part-ii.html#cb139-2" tabindex="-1"></a>  <span class="at">genes =</span> <span class="fu">names</span>(spat_cor_netw_DT<span class="sc">$</span>cor_clusters<span class="sc">$</span>spat_netw_clus),</span>
<span id="cb139-3"><a href="visium-part-ii.html#cb139-3" tabindex="-1"></a>  <span class="at">cluster =</span> spat_cor_netw_DT<span class="sc">$</span>cor_clusters<span class="sc">$</span>spat_netw_clus)</span>
<span id="cb139-4"><a href="visium-part-ii.html#cb139-4" tabindex="-1"></a></span>
<span id="cb139-5"><a href="visium-part-ii.html#cb139-5" tabindex="-1"></a>data.table<span class="sc">::</span><span class="fu">setorder</span>(coexpr_dt, cluster)</span>
<span id="cb139-6"><a href="visium-part-ii.html#cb139-6" tabindex="-1"></a></span>
<span id="cb139-7"><a href="visium-part-ii.html#cb139-7" tabindex="-1"></a>top30_coexpr_dt <span class="ot">&lt;-</span> coexpr_dt[, <span class="fu">head</span>(.SD, <span class="dv">30</span>) , by <span class="ot">=</span> cluster]</span>
<span id="cb139-8"><a href="visium-part-ii.html#cb139-8" tabindex="-1"></a></span>
<span id="cb139-9"><a href="visium-part-ii.html#cb139-9" tabindex="-1"></a>spatial_genes <span class="ot">&lt;-</span> top30_coexpr_dt<span class="sc">$</span>genes</span></code></pre></div>
<ul>
<li>Re-calculate the clustering</li>
</ul>
<p>Use the spatial genes to calculate again the principal components, umap, network and clustering</p>
<div class="sourceCode" id="cb140"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb140-1"><a href="visium-part-ii.html#cb140-1" tabindex="-1"></a>visium_brain <span class="ot">&lt;-</span> <span class="fu">runPCA</span>(<span class="at">gobject =</span> visium_brain,</span>
<span id="cb140-2"><a href="visium-part-ii.html#cb140-2" tabindex="-1"></a>                       <span class="at">feats_to_use =</span> spatial_genes,</span>
<span id="cb140-3"><a href="visium-part-ii.html#cb140-3" tabindex="-1"></a>                       <span class="at">name =</span> <span class="st">&quot;custom_pca&quot;</span>)</span>
<span id="cb140-4"><a href="visium-part-ii.html#cb140-4" tabindex="-1"></a></span>
<span id="cb140-5"><a href="visium-part-ii.html#cb140-5" tabindex="-1"></a>visium_brain <span class="ot">&lt;-</span> <span class="fu">runUMAP</span>(visium_brain, </span>
<span id="cb140-6"><a href="visium-part-ii.html#cb140-6" tabindex="-1"></a>                        <span class="at">dim_reduction_name =</span> <span class="st">&quot;custom_pca&quot;</span>, </span>
<span id="cb140-7"><a href="visium-part-ii.html#cb140-7" tabindex="-1"></a>                        <span class="at">dimensions_to_use =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>,</span>
<span id="cb140-8"><a href="visium-part-ii.html#cb140-8" tabindex="-1"></a>                        <span class="at">name =</span> <span class="st">&quot;custom_umap&quot;</span>)</span>
<span id="cb140-9"><a href="visium-part-ii.html#cb140-9" tabindex="-1"></a></span>
<span id="cb140-10"><a href="visium-part-ii.html#cb140-10" tabindex="-1"></a>visium_brain <span class="ot">&lt;-</span> <span class="fu">createNearestNetwork</span>(<span class="at">gobject =</span> visium_brain,</span>
<span id="cb140-11"><a href="visium-part-ii.html#cb140-11" tabindex="-1"></a>                                     <span class="at">dim_reduction_name =</span> <span class="st">&quot;custom_pca&quot;</span>,</span>
<span id="cb140-12"><a href="visium-part-ii.html#cb140-12" tabindex="-1"></a>                                     <span class="at">dimensions_to_use =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>, </span>
<span id="cb140-13"><a href="visium-part-ii.html#cb140-13" tabindex="-1"></a>                                     <span class="at">k =</span> <span class="dv">5</span>,</span>
<span id="cb140-14"><a href="visium-part-ii.html#cb140-14" tabindex="-1"></a>                                     <span class="at">name =</span> <span class="st">&quot;custom_NN&quot;</span>)</span>
<span id="cb140-15"><a href="visium-part-ii.html#cb140-15" tabindex="-1"></a></span>
<span id="cb140-16"><a href="visium-part-ii.html#cb140-16" tabindex="-1"></a>visium_brain <span class="ot">&lt;-</span> <span class="fu">doLeidenCluster</span>(<span class="at">gobject =</span> visium_brain, </span>
<span id="cb140-17"><a href="visium-part-ii.html#cb140-17" tabindex="-1"></a>                                <span class="at">network_name =</span> <span class="st">&quot;custom_NN&quot;</span>,</span>
<span id="cb140-18"><a href="visium-part-ii.html#cb140-18" tabindex="-1"></a>                                <span class="at">resolution =</span> <span class="fl">0.15</span>, </span>
<span id="cb140-19"><a href="visium-part-ii.html#cb140-19" tabindex="-1"></a>                                <span class="at">n_iterations =</span> <span class="dv">1000</span>,</span>
<span id="cb140-20"><a href="visium-part-ii.html#cb140-20" tabindex="-1"></a>                                <span class="at">name =</span> <span class="st">&quot;custom_leiden&quot;</span>)</span></code></pre></div>
<ul>
<li>Visualize</li>
</ul>
<p>Plot the spatial distribution of the Leiden clusters calculated based on the spatial genes.</p>
<div class="sourceCode" id="cb141"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb141-1"><a href="visium-part-ii.html#cb141-1" tabindex="-1"></a><span class="fu">spatPlot2D</span>(visium_brain, </span>
<span id="cb141-2"><a href="visium-part-ii.html#cb141-2" tabindex="-1"></a>           <span class="at">cell_color =</span> <span class="st">&quot;custom_leiden&quot;</span>,</span>
<span id="cb141-3"><a href="visium-part-ii.html#cb141-3" tabindex="-1"></a>           <span class="at">point_size =</span> <span class="dv">3</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-176"></span>
<img src="img/02_session1/15-spatPlot2D.png" alt="Spatial distribution of Leiden clusters calculated using spatial genes." width="100%" />
<p class="caption">
Figure 8.16: Spatial distribution of Leiden clusters calculated using spatial genes.
</p>
</div>
<p>Plot the UMAP and color the spots using the Leiden clusters calculated based on the spatial genes.</p>
<div class="sourceCode" id="cb142"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb142-1"><a href="visium-part-ii.html#cb142-1" tabindex="-1"></a><span class="fu">plotUMAP</span>(<span class="at">gobject =</span> visium_brain, </span>
<span id="cb142-2"><a href="visium-part-ii.html#cb142-2" tabindex="-1"></a>         <span class="at">cell_color =</span> <span class="st">&quot;custom_leiden&quot;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-178"></span>
<img src="img/02_session1/16-UMAP.png" alt="UMAP plot, colors indicate the Leiden clusters calculated using spatial genes." width="100%" />
<p class="caption">
Figure 8.17: UMAP plot, colors indicate the Leiden clusters calculated using spatial genes.
</p>
</div>
</div>
<div id="spatial-domains-hmrf" class="section level2 hasAnchor" number="8.6">
<h2><span class="header-section-number">8.6</span> Spatial domains HMRF<a href="visium-part-ii.html#spatial-domains-hmrf" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Hidden Markov Random Field (HMRF) models capture spatial dependencies and segment tissue regions based on shared and gene expression patterns.</p>
<ul>
<li>Do HMRF with different betas on top 30 genes per spatial co-expression module</li>
</ul>
<p>This step may take several minutes to run.</p>
<div class="sourceCode" id="cb143"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb143-1"><a href="visium-part-ii.html#cb143-1" tabindex="-1"></a>HMRF_spatial_genes <span class="ot">&lt;-</span> <span class="fu">doHMRF</span>(<span class="at">gobject =</span> visium_brain,</span>
<span id="cb143-2"><a href="visium-part-ii.html#cb143-2" tabindex="-1"></a>                            <span class="at">expression_values =</span> <span class="st">&quot;scaled&quot;</span>,</span>
<span id="cb143-3"><a href="visium-part-ii.html#cb143-3" tabindex="-1"></a>                            <span class="at">spatial_genes =</span> spatial_genes, </span>
<span id="cb143-4"><a href="visium-part-ii.html#cb143-4" tabindex="-1"></a>                            <span class="at">k =</span> <span class="dv">20</span>,</span>
<span id="cb143-5"><a href="visium-part-ii.html#cb143-5" tabindex="-1"></a>                            <span class="at">spatial_network_name =</span> <span class="st">&quot;spatial_network&quot;</span>,</span>
<span id="cb143-6"><a href="visium-part-ii.html#cb143-6" tabindex="-1"></a>                            <span class="at">betas =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">10</span>, <span class="dv">5</span>),</span>
<span id="cb143-7"><a href="visium-part-ii.html#cb143-7" tabindex="-1"></a>                            <span class="at">output_folder =</span> <span class="st">&quot;11_HMRF/&quot;</span>)</span></code></pre></div>
<p>Add the HMRF results to the giotto object</p>
<div class="sourceCode" id="cb144"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb144-1"><a href="visium-part-ii.html#cb144-1" tabindex="-1"></a>visium_brain <span class="ot">&lt;-</span> <span class="fu">addHMRF</span>(<span class="at">gobject =</span> visium_brain, </span>
<span id="cb144-2"><a href="visium-part-ii.html#cb144-2" tabindex="-1"></a>                        <span class="at">HMRFoutput =</span> HMRF_spatial_genes,</span>
<span id="cb144-3"><a href="visium-part-ii.html#cb144-3" tabindex="-1"></a>                        <span class="at">k =</span> <span class="dv">20</span>, </span>
<span id="cb144-4"><a href="visium-part-ii.html#cb144-4" tabindex="-1"></a>                        <span class="at">betas_to_add =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">10</span>, <span class="dv">20</span>, <span class="dv">30</span>, <span class="dv">40</span>),</span>
<span id="cb144-5"><a href="visium-part-ii.html#cb144-5" tabindex="-1"></a>                        <span class="at">hmrf_name =</span> <span class="st">&quot;HMRF&quot;</span>)</span></code></pre></div>
<ul>
<li>Visualize</li>
</ul>
<p>Plot the spatial distribution of the HMRF domains.</p>
<div class="sourceCode" id="cb145"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb145-1"><a href="visium-part-ii.html#cb145-1" tabindex="-1"></a><span class="fu">spatPlot2D</span>(<span class="at">gobject =</span> visium_brain, </span>
<span id="cb145-2"><a href="visium-part-ii.html#cb145-2" tabindex="-1"></a>           <span class="at">cell_color =</span> <span class="st">&quot;HMRF_k20_b.40&quot;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-182"></span>
<img src="img/02_session1/17-spatPlot2D.png" alt="Spatial distribution of HMRF domains." width="100%" />
<p class="caption">
Figure 8.18: Spatial distribution of HMRF domains.
</p>
</div>
</div>
<div id="interactive-tools" class="section level2 hasAnchor" number="8.7">
<h2><span class="header-section-number">8.7</span> Interactive tools<a href="visium-part-ii.html#interactive-tools" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We have integrated a shiny app in Giotto to interactively select regions of a spatial plot.</p>
<ul>
<li>Create a spatial plot</li>
</ul>
<div class="sourceCode" id="cb146"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb146-1"><a href="visium-part-ii.html#cb146-1" tabindex="-1"></a>brain_spatPlot <span class="ot">&lt;-</span> <span class="fu">spatPlot2D</span>(<span class="at">gobject =</span> visium_brain,</span>
<span id="cb146-2"><a href="visium-part-ii.html#cb146-2" tabindex="-1"></a>                          <span class="at">cell_color =</span> <span class="st">&quot;leiden_clus&quot;</span>,</span>
<span id="cb146-3"><a href="visium-part-ii.html#cb146-3" tabindex="-1"></a>                          <span class="at">show_image =</span> <span class="cn">FALSE</span>,</span>
<span id="cb146-4"><a href="visium-part-ii.html#cb146-4" tabindex="-1"></a>                          <span class="at">return_plot =</span> <span class="cn">TRUE</span>,</span>
<span id="cb146-5"><a href="visium-part-ii.html#cb146-5" tabindex="-1"></a>                          <span class="at">point_size =</span> <span class="dv">1</span>)</span>
<span id="cb146-6"><a href="visium-part-ii.html#cb146-6" tabindex="-1"></a></span>
<span id="cb146-7"><a href="visium-part-ii.html#cb146-7" tabindex="-1"></a>brain_spatPlot</span></code></pre></div>
<ul>
<li>Run the Shiny app</li>
</ul>
<div class="sourceCode" id="cb147"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb147-1"><a href="visium-part-ii.html#cb147-1" tabindex="-1"></a><span class="fu">plotInteractivePolygons</span>(brain_spatPlot)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-185"></span>
<img src="img/02_session1/18-spatPlot2D.png" alt="Shiny app using the visium brain sample." width="100%" />
<p class="caption">
Figure 8.19: Shiny app using the visium brain sample.
</p>
</div>
<ul>
<li>Select the regions of interest and save the coordinates</li>
</ul>
<div class="sourceCode" id="cb148"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb148-1"><a href="visium-part-ii.html#cb148-1" tabindex="-1"></a>polygon_coordinates <span class="ot">&lt;-</span> <span class="fu">plotInteractivePolygons</span>(brain_spatPlot)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-187"></span>
<img src="img/02_session1/19-spatPlot2D.png" alt="Polygons selected using the interactive Shiny app." width="60%" />
<p class="caption">
Figure 8.20: Polygons selected using the interactive Shiny app.
</p>
</div>
<ul>
<li>Transform the data.table or data.frame with coordinates into a Giotto polygon object</li>
</ul>
<div class="sourceCode" id="cb149"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb149-1"><a href="visium-part-ii.html#cb149-1" tabindex="-1"></a>giotto_polygons <span class="ot">&lt;-</span> <span class="fu">createGiottoPolygonsFromDfr</span>(polygon_coordinates, </span>
<span id="cb149-2"><a href="visium-part-ii.html#cb149-2" tabindex="-1"></a>                                               <span class="at">name =</span> <span class="st">&quot;selections&quot;</span>,</span>
<span id="cb149-3"><a href="visium-part-ii.html#cb149-3" tabindex="-1"></a>                                               <span class="at">calc_centroids =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<ul>
<li>Add the polygons to the Giotto object</li>
</ul>
<div class="sourceCode" id="cb150"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb150-1"><a href="visium-part-ii.html#cb150-1" tabindex="-1"></a>visium_brain <span class="ot">&lt;-</span> <span class="fu">addGiottoPolygons</span>(<span class="at">gobject =</span> visium_brain,</span>
<span id="cb150-2"><a href="visium-part-ii.html#cb150-2" tabindex="-1"></a>                                  <span class="at">gpolygons =</span> <span class="fu">list</span>(giotto_polygons))</span></code></pre></div>
<ul>
<li>Add the corresponding polygon IDs to the cell metadata</li>
</ul>
<div class="sourceCode" id="cb151"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb151-1"><a href="visium-part-ii.html#cb151-1" tabindex="-1"></a>visium_brain <span class="ot">&lt;-</span> <span class="fu">addPolygonCells</span>(visium_brain,</span>
<span id="cb151-2"><a href="visium-part-ii.html#cb151-2" tabindex="-1"></a>                                <span class="at">polygon_name =</span> <span class="st">&quot;selections&quot;</span>)</span></code></pre></div>
<ul>
<li>Extract the coordinates and IDs from cells located within one or multiple regions of interest.</li>
</ul>
<div class="sourceCode" id="cb152"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb152-1"><a href="visium-part-ii.html#cb152-1" tabindex="-1"></a><span class="fu">getCellsFromPolygon</span>(visium_brain,</span>
<span id="cb152-2"><a href="visium-part-ii.html#cb152-2" tabindex="-1"></a>                    <span class="at">polygon_name =</span> <span class="st">&quot;selections&quot;</span>,</span>
<span id="cb152-3"><a href="visium-part-ii.html#cb152-3" tabindex="-1"></a>                    <span class="at">polygons =</span> <span class="st">&quot;polygon 1&quot;</span>)</span></code></pre></div>
<p>If no polygon name is provided, the function will retrieve cells located within all polygons</p>
<div class="sourceCode" id="cb153"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb153-1"><a href="visium-part-ii.html#cb153-1" tabindex="-1"></a><span class="fu">getCellsFromPolygon</span>(visium_brain,</span>
<span id="cb153-2"><a href="visium-part-ii.html#cb153-2" tabindex="-1"></a>                    <span class="at">polygon_name =</span> <span class="st">&quot;selections&quot;</span>)</span></code></pre></div>
<ul>
<li>Compare the expression levels of some genes of interest between the selected regions</li>
</ul>
<div class="sourceCode" id="cb154"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb154-1"><a href="visium-part-ii.html#cb154-1" tabindex="-1"></a><span class="fu">comparePolygonExpression</span>(visium_brain,</span>
<span id="cb154-2"><a href="visium-part-ii.html#cb154-2" tabindex="-1"></a>                         <span class="at">selected_feats =</span> <span class="fu">c</span>(<span class="st">&quot;Stmn1&quot;</span>, <span class="st">&quot;Psd&quot;</span>, <span class="st">&quot;Ly6h&quot;</span>))</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-194"></span>
<img src="img/02_session1/20-heatmap.png" alt="Heatmap showing the z-scores of three genes per selected polygon." width="60%" />
<p class="caption">
Figure 8.21: Heatmap showing the z-scores of three genes per selected polygon.
</p>
</div>
<ul>
<li>Calculate the top genes expressed within each region, then provide the result to compare polygons</li>
</ul>
<div class="sourceCode" id="cb155"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb155-1"><a href="visium-part-ii.html#cb155-1" tabindex="-1"></a>scran_results <span class="ot">&lt;-</span> <span class="fu">findMarkers_one_vs_all</span>(</span>
<span id="cb155-2"><a href="visium-part-ii.html#cb155-2" tabindex="-1"></a>  visium_brain,</span>
<span id="cb155-3"><a href="visium-part-ii.html#cb155-3" tabindex="-1"></a>  <span class="at">spat_unit =</span> <span class="st">&quot;cell&quot;</span>,</span>
<span id="cb155-4"><a href="visium-part-ii.html#cb155-4" tabindex="-1"></a>  <span class="at">feat_type =</span> <span class="st">&quot;rna&quot;</span>,</span>
<span id="cb155-5"><a href="visium-part-ii.html#cb155-5" tabindex="-1"></a>  <span class="at">method =</span> <span class="st">&quot;scran&quot;</span>,</span>
<span id="cb155-6"><a href="visium-part-ii.html#cb155-6" tabindex="-1"></a>  <span class="at">expression_values =</span> <span class="st">&quot;normalized&quot;</span>,</span>
<span id="cb155-7"><a href="visium-part-ii.html#cb155-7" tabindex="-1"></a>  <span class="at">cluster_column =</span> <span class="st">&quot;selections&quot;</span>,</span>
<span id="cb155-8"><a href="visium-part-ii.html#cb155-8" tabindex="-1"></a>  <span class="at">min_feats =</span> <span class="dv">2</span>)</span>
<span id="cb155-9"><a href="visium-part-ii.html#cb155-9" tabindex="-1"></a></span>
<span id="cb155-10"><a href="visium-part-ii.html#cb155-10" tabindex="-1"></a>top_genes <span class="ot">&lt;-</span> scran_results[, <span class="fu">head</span>(.SD, <span class="dv">2</span>), by <span class="ot">=</span> <span class="st">&quot;cluster&quot;</span>]<span class="sc">$</span>feats</span>
<span id="cb155-11"><a href="visium-part-ii.html#cb155-11" tabindex="-1"></a></span>
<span id="cb155-12"><a href="visium-part-ii.html#cb155-12" tabindex="-1"></a><span class="fu">comparePolygonExpression</span>(visium_brain,</span>
<span id="cb155-13"><a href="visium-part-ii.html#cb155-13" tabindex="-1"></a>                         <span class="at">selected_feats =</span> top_genes)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-196"></span>
<img src="img/02_session1/21-heatmap.png" alt="Heatmap showing the z-scores of top scran genes per selected polygon." width="60%" />
<p class="caption">
Figure 8.22: Heatmap showing the z-scores of top scran genes per selected polygon.
</p>
</div>
<ul>
<li>Compare the abundance of cell types between the selected regions</li>
</ul>
<div class="sourceCode" id="cb156"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb156-1"><a href="visium-part-ii.html#cb156-1" tabindex="-1"></a><span class="fu">compareCellAbundance</span>(visium_brain)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-198"></span>
<img src="img/02_session1/22-heatmap.png" alt="Heatmap showing the cell abundance per selected polygon." width="60%" />
<p class="caption">
Figure 8.23: Heatmap showing the cell abundance per selected polygon.
</p>
</div>
<ul>
<li>Use other columns within the cell metadata table to compare the cell type abundances</li>
</ul>
<div class="sourceCode" id="cb157"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb157-1"><a href="visium-part-ii.html#cb157-1" tabindex="-1"></a><span class="fu">compareCellAbundance</span>(visium_brain, </span>
<span id="cb157-2"><a href="visium-part-ii.html#cb157-2" tabindex="-1"></a>                     <span class="at">cell_type_column =</span> <span class="st">&quot;custom_leiden&quot;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-200"></span>
<img src="img/02_session1/23-heatmap.png" alt="Heatmap showing the Leiden clusters abundance per selected polygon." width="60%" />
<p class="caption">
Figure 8.24: Heatmap showing the Leiden clusters abundance per selected polygon.
</p>
</div>
<ul>
<li>Use the spatPlot arguments to isolate and plot each region.</li>
</ul>
<div class="sourceCode" id="cb158"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb158-1"><a href="visium-part-ii.html#cb158-1" tabindex="-1"></a><span class="fu">spatPlot2D</span>(visium_brain,</span>
<span id="cb158-2"><a href="visium-part-ii.html#cb158-2" tabindex="-1"></a>           <span class="at">cell_color =</span> <span class="st">&quot;leiden_clus&quot;</span>,</span>
<span id="cb158-3"><a href="visium-part-ii.html#cb158-3" tabindex="-1"></a>           <span class="at">group_by =</span> <span class="st">&quot;selections&quot;</span>,</span>
<span id="cb158-4"><a href="visium-part-ii.html#cb158-4" tabindex="-1"></a>           <span class="at">cow_n_col =</span> <span class="dv">3</span>,</span>
<span id="cb158-5"><a href="visium-part-ii.html#cb158-5" tabindex="-1"></a>           <span class="at">point_size =</span> <span class="dv">2</span>,</span>
<span id="cb158-6"><a href="visium-part-ii.html#cb158-6" tabindex="-1"></a>           <span class="at">show_legend =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-202"></span>
<img src="img/02_session1/24-spatPlot2D.png" alt="Spatial distribution of Leiden clusters across the selected polygons." width="80%" />
<p class="caption">
Figure 8.25: Spatial distribution of Leiden clusters across the selected polygons.
</p>
</div>
<ul>
<li>Color each cell by cluster, cell type or expression level.</li>
</ul>
<div class="sourceCode" id="cb159"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb159-1"><a href="visium-part-ii.html#cb159-1" tabindex="-1"></a><span class="fu">spatFeatPlot2D</span>(visium_brain,</span>
<span id="cb159-2"><a href="visium-part-ii.html#cb159-2" tabindex="-1"></a>               <span class="at">expression_values =</span> <span class="st">&quot;scaled&quot;</span>,</span>
<span id="cb159-3"><a href="visium-part-ii.html#cb159-3" tabindex="-1"></a>               <span class="at">group_by =</span> <span class="st">&quot;selections&quot;</span>,</span>
<span id="cb159-4"><a href="visium-part-ii.html#cb159-4" tabindex="-1"></a>               <span class="at">feats =</span> <span class="st">&quot;Psd&quot;</span>,</span>
<span id="cb159-5"><a href="visium-part-ii.html#cb159-5" tabindex="-1"></a>               <span class="at">point_size =</span> <span class="dv">2</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-204"></span>
<img src="img/02_session1/25-spatFeatPlot2D.png" alt="Spatial distribution of Psd scaled expression across the selected polygons." width="80%" />
<p class="caption">
Figure 8.26: Spatial distribution of Psd scaled expression across the selected polygons.
</p>
</div>
<ul>
<li>Plot again the polygons</li>
</ul>
<div class="sourceCode" id="cb160"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb160-1"><a href="visium-part-ii.html#cb160-1" tabindex="-1"></a><span class="fu">plotPolygons</span>(visium_brain,</span>
<span id="cb160-2"><a href="visium-part-ii.html#cb160-2" tabindex="-1"></a>             <span class="at">polygon_name =</span> <span class="st">&quot;selections&quot;</span>,</span>
<span id="cb160-3"><a href="visium-part-ii.html#cb160-3" tabindex="-1"></a>             <span class="at">x =</span> brain_spatPlot)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-206"></span>
<img src="img/02_session1/26-spatPlot2D.png" alt="Spatial location of selected polygons." width="60%" />
<p class="caption">
Figure 8.27: Spatial location of selected polygons.
</p>
</div>
</div>
<div id="session-info-1" class="section level2 hasAnchor" number="8.8">
<h2><span class="header-section-number">8.8</span> Session info<a href="visium-part-ii.html#session-info-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb161"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb161-1"><a href="visium-part-ii.html#cb161-1" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code></pre></div>
<div class="sourceCode" id="cb162"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb162-1"><a href="visium-part-ii.html#cb162-1" tabindex="-1"></a>R version <span class="dv">4</span>.<span class="fl">4.1</span> (<span class="dv">2024-06-14</span>)</span>
<span id="cb162-2"><a href="visium-part-ii.html#cb162-2" tabindex="-1"></a>Platform<span class="sc">:</span> aarch64<span class="sc">-</span>apple<span class="sc">-</span>darwin20</span>
<span id="cb162-3"><a href="visium-part-ii.html#cb162-3" tabindex="-1"></a>Running under<span class="sc">:</span> macOS Sonoma <span class="fl">14.5</span></span>
<span id="cb162-4"><a href="visium-part-ii.html#cb162-4" tabindex="-1"></a></span>
<span id="cb162-5"><a href="visium-part-ii.html#cb162-5" tabindex="-1"></a>Matrix products<span class="sc">:</span> default</span>
<span id="cb162-6"><a href="visium-part-ii.html#cb162-6" tabindex="-1"></a>BLAS<span class="sc">:</span>   <span class="er">/</span>System<span class="sc">/</span>Library<span class="sc">/</span>Frameworks<span class="sc">/</span>Accelerate.framework<span class="sc">/</span>Versions<span class="sc">/</span>A<span class="sc">/</span>Frameworks<span class="sc">/</span>vecLib.framework<span class="sc">/</span>Versions<span class="sc">/</span>A<span class="sc">/</span>libBLAS.dylib </span>
<span id="cb162-7"><a href="visium-part-ii.html#cb162-7" tabindex="-1"></a>LAPACK<span class="sc">:</span> <span class="er">/</span>Library<span class="sc">/</span>Frameworks<span class="sc">/</span>R.framework<span class="sc">/</span>Versions<span class="sc">/</span><span class="fl">4.4</span><span class="sc">-</span>arm64<span class="sc">/</span>Resources<span class="sc">/</span>lib<span class="sc">/</span>libRlapack.dylib;  LAPACK version <span class="dv">3</span>.<span class="fl">12.0</span></span>
<span id="cb162-8"><a href="visium-part-ii.html#cb162-8" tabindex="-1"></a></span>
<span id="cb162-9"><a href="visium-part-ii.html#cb162-9" tabindex="-1"></a>locale<span class="sc">:</span></span>
<span id="cb162-10"><a href="visium-part-ii.html#cb162-10" tabindex="-1"></a>[<span class="dv">1</span>] en_US.UTF<span class="dv">-8</span><span class="sc">/</span>en_US.UTF<span class="dv">-8</span><span class="sc">/</span>en_US.UTF<span class="dv">-8</span><span class="sc">/</span>C<span class="sc">/</span>en_US.UTF<span class="dv">-8</span><span class="sc">/</span>en_US.UTF<span class="dv">-8</span></span>
<span id="cb162-11"><a href="visium-part-ii.html#cb162-11" tabindex="-1"></a></span>
<span id="cb162-12"><a href="visium-part-ii.html#cb162-12" tabindex="-1"></a>time zone<span class="sc">:</span> America<span class="sc">/</span>New_York</span>
<span id="cb162-13"><a href="visium-part-ii.html#cb162-13" tabindex="-1"></a>tzcode source<span class="sc">:</span> internal</span>
<span id="cb162-14"><a href="visium-part-ii.html#cb162-14" tabindex="-1"></a></span>
<span id="cb162-15"><a href="visium-part-ii.html#cb162-15" tabindex="-1"></a>attached base packages<span class="sc">:</span></span>
<span id="cb162-16"><a href="visium-part-ii.html#cb162-16" tabindex="-1"></a>[<span class="dv">1</span>] stats     graphics  grDevices utils     datasets  methods   base     </span>
<span id="cb162-17"><a href="visium-part-ii.html#cb162-17" tabindex="-1"></a></span>
<span id="cb162-18"><a href="visium-part-ii.html#cb162-18" tabindex="-1"></a>other attached packages<span class="sc">:</span></span>
<span id="cb162-19"><a href="visium-part-ii.html#cb162-19" tabindex="-1"></a>[<span class="dv">1</span>] shiny_1.<span class="dv">8</span>.<span class="fl">1.1</span>     Giotto_4.<span class="fl">1.0</span>      GiottoClass_0.<span class="fl">3.3</span></span>
<span id="cb162-20"><a href="visium-part-ii.html#cb162-20" tabindex="-1"></a></span>
<span id="cb162-21"><a href="visium-part-ii.html#cb162-21" tabindex="-1"></a>loaded via a <span class="fu">namespace</span> (and not attached)<span class="sc">:</span></span>
<span id="cb162-22"><a href="visium-part-ii.html#cb162-22" tabindex="-1"></a>  [<span class="dv">1</span>] later_1.<span class="fl">3.2</span>                 tibble_3.<span class="fl">2.1</span>               </span>
<span id="cb162-23"><a href="visium-part-ii.html#cb162-23" tabindex="-1"></a>  [<span class="dv">3</span>] R.oo_1.<span class="fl">26.0</span>                 polyclip_1<span class="fl">.10</span><span class="dv">-7</span>            </span>
<span id="cb162-24"><a href="visium-part-ii.html#cb162-24" tabindex="-1"></a>  [<span class="dv">5</span>] lifecycle_1.<span class="fl">0.4</span>             edgeR_4.<span class="fl">2.1</span>                </span>
<span id="cb162-25"><a href="visium-part-ii.html#cb162-25" tabindex="-1"></a>  [<span class="dv">7</span>] doParallel_1.<span class="fl">0.17</span>           lattice_0<span class="fl">.22</span><span class="dv">-6</span>             </span>
<span id="cb162-26"><a href="visium-part-ii.html#cb162-26" tabindex="-1"></a>  [<span class="dv">9</span>] MASS_7<span class="fl">.3</span><span class="dv">-61</span>                 backports_1.<span class="fl">5.0</span>            </span>
<span id="cb162-27"><a href="visium-part-ii.html#cb162-27" tabindex="-1"></a> [<span class="dv">11</span>] magrittr_2.<span class="fl">0.3</span>              sass_0.<span class="fl">4.9</span>                 </span>
<span id="cb162-28"><a href="visium-part-ii.html#cb162-28" tabindex="-1"></a> [<span class="dv">13</span>] limma_3.<span class="fl">60.4</span>                plotly_4.<span class="fl">10.4</span>              </span>
<span id="cb162-29"><a href="visium-part-ii.html#cb162-29" tabindex="-1"></a> [<span class="dv">15</span>] rmarkdown_2<span class="fl">.27</span>              jquerylib_0.<span class="fl">1.4</span>            </span>
<span id="cb162-30"><a href="visium-part-ii.html#cb162-30" tabindex="-1"></a> [<span class="dv">17</span>] yaml_2.<span class="fl">3.9</span>                  metapod_1.<span class="fl">12.0</span>             </span>
<span id="cb162-31"><a href="visium-part-ii.html#cb162-31" tabindex="-1"></a> [<span class="dv">19</span>] httpuv_1.<span class="fl">6.15</span>               sp_2<span class="fl">.1</span><span class="dv">-4</span>                   </span>
<span id="cb162-32"><a href="visium-part-ii.html#cb162-32" tabindex="-1"></a> [<span class="dv">21</span>] reticulate_1.<span class="fl">38.0</span>           cowplot_1.<span class="fl">1.3</span>              </span>
<span id="cb162-33"><a href="visium-part-ii.html#cb162-33" tabindex="-1"></a> [<span class="dv">23</span>] RColorBrewer_1<span class="fl">.1</span><span class="dv">-3</span>          abind_1<span class="fl">.4</span><span class="dv">-5</span>                </span>
<span id="cb162-34"><a href="visium-part-ii.html#cb162-34" tabindex="-1"></a> [<span class="dv">25</span>] zlibbioc_1.<span class="fl">50.0</span>             quadprog_1<span class="fl">.5</span><span class="dv">-8</span>             </span>
<span id="cb162-35"><a href="visium-part-ii.html#cb162-35" tabindex="-1"></a> [<span class="dv">27</span>] GenomicRanges_1.<span class="fl">56.1</span>        purrr_1.<span class="fl">0.2</span>                </span>
<span id="cb162-36"><a href="visium-part-ii.html#cb162-36" tabindex="-1"></a> [<span class="dv">29</span>] R.utils_2.<span class="fl">12.3</span>              BiocGenerics_0.<span class="fl">50.0</span>        </span>
<span id="cb162-37"><a href="visium-part-ii.html#cb162-37" tabindex="-1"></a> [<span class="dv">31</span>] tweenr_2.<span class="fl">0.3</span>                circlize_0.<span class="fl">4.16</span>            </span>
<span id="cb162-38"><a href="visium-part-ii.html#cb162-38" tabindex="-1"></a> [<span class="dv">33</span>] GenomeInfoDbData_1.<span class="fl">2.12</span>     IRanges_2.<span class="fl">38.1</span>             </span>
<span id="cb162-39"><a href="visium-part-ii.html#cb162-39" tabindex="-1"></a> [<span class="dv">35</span>] S4Vectors_0.<span class="fl">42.1</span>            ggrepel_0.<span class="fl">9.5</span>              </span>
<span id="cb162-40"><a href="visium-part-ii.html#cb162-40" tabindex="-1"></a> [<span class="dv">37</span>] irlba_2.<span class="dv">3</span>.<span class="fl">5.1</span>               terra_1<span class="fl">.7</span><span class="dv">-78</span>               </span>
<span id="cb162-41"><a href="visium-part-ii.html#cb162-41" tabindex="-1"></a> [<span class="dv">39</span>] dqrng_0.<span class="fl">4.1</span>                 DelayedMatrixStats_1.<span class="fl">26.0</span>  </span>
<span id="cb162-42"><a href="visium-part-ii.html#cb162-42" tabindex="-1"></a> [<span class="dv">41</span>] colorRamp2_0.<span class="fl">1.0</span>            codetools_0<span class="fl">.2</span><span class="dv">-20</span>           </span>
<span id="cb162-43"><a href="visium-part-ii.html#cb162-43" tabindex="-1"></a> [<span class="dv">43</span>] DelayedArray_0.<span class="fl">30.1</span>         scuttle_1.<span class="fl">14.0</span>             </span>
<span id="cb162-44"><a href="visium-part-ii.html#cb162-44" tabindex="-1"></a> [<span class="dv">45</span>] ggforce_0.<span class="fl">4.2</span>               tidyselect_1.<span class="fl">2.1</span>           </span>
<span id="cb162-45"><a href="visium-part-ii.html#cb162-45" tabindex="-1"></a> [<span class="dv">47</span>] shape_1.<span class="dv">4</span>.<span class="fl">6.1</span>               UCSC.utils_1.<span class="fl">0.0</span>           </span>
<span id="cb162-46"><a href="visium-part-ii.html#cb162-46" tabindex="-1"></a> [<span class="dv">49</span>] farver_2.<span class="fl">1.2</span>                ScaledMatrix_1.<span class="fl">12.0</span>        </span>
<span id="cb162-47"><a href="visium-part-ii.html#cb162-47" tabindex="-1"></a> [<span class="dv">51</span>] matrixStats_1.<span class="fl">3.0</span>           stats4_4.<span class="fl">4.1</span>               </span>
<span id="cb162-48"><a href="visium-part-ii.html#cb162-48" tabindex="-1"></a> [<span class="dv">53</span>] GiottoData_0.<span class="dv">2</span>.<span class="fl">12.0</span>         jsonlite_1.<span class="fl">8.8</span>             </span>
<span id="cb162-49"><a href="visium-part-ii.html#cb162-49" tabindex="-1"></a> [<span class="dv">55</span>] GetoptLong_1.<span class="fl">0.5</span>            BiocNeighbors_1.<span class="fl">22.0</span>       </span>
<span id="cb162-50"><a href="visium-part-ii.html#cb162-50" tabindex="-1"></a> [<span class="dv">57</span>] progressr_0.<span class="fl">14.0</span>            iterators_1.<span class="fl">0.14</span>           </span>
<span id="cb162-51"><a href="visium-part-ii.html#cb162-51" tabindex="-1"></a> [<span class="dv">59</span>] systemfonts_1.<span class="fl">1.0</span>           foreach_1.<span class="fl">5.2</span>              </span>
<span id="cb162-52"><a href="visium-part-ii.html#cb162-52" tabindex="-1"></a> [<span class="dv">61</span>] dbscan_1<span class="fl">.2</span><span class="dv">-0</span>                tools_4.<span class="fl">4.1</span>                </span>
<span id="cb162-53"><a href="visium-part-ii.html#cb162-53" tabindex="-1"></a> [<span class="dv">63</span>] ragg_1.<span class="fl">3.2</span>                  Rcpp_1.<span class="fl">0.13</span>                </span>
<span id="cb162-54"><a href="visium-part-ii.html#cb162-54" tabindex="-1"></a> [<span class="dv">65</span>] glue_1.<span class="fl">7.0</span>                  SparseArray_1.<span class="fl">4.8</span>          </span>
<span id="cb162-55"><a href="visium-part-ii.html#cb162-55" tabindex="-1"></a> [<span class="dv">67</span>] xfun_0<span class="fl">.46</span>                   MatrixGenerics_1.<span class="fl">16.0</span>      </span>
<span id="cb162-56"><a href="visium-part-ii.html#cb162-56" tabindex="-1"></a> [<span class="dv">69</span>] GenomeInfoDb_1.<span class="fl">40.1</span>         dplyr_1.<span class="fl">1.4</span>                </span>
<span id="cb162-57"><a href="visium-part-ii.html#cb162-57" tabindex="-1"></a> [<span class="dv">71</span>] withr_3.<span class="fl">0.0</span>                 fastmap_1.<span class="fl">2.0</span>              </span>
<span id="cb162-58"><a href="visium-part-ii.html#cb162-58" tabindex="-1"></a> [<span class="dv">73</span>] bluster_1.<span class="fl">14.0</span>              fansi_1.<span class="fl">0.6</span>                </span>
<span id="cb162-59"><a href="visium-part-ii.html#cb162-59" tabindex="-1"></a> [<span class="dv">75</span>] digest_0.<span class="fl">6.36</span>               rsvd_1.<span class="fl">0.5</span>                 </span>
<span id="cb162-60"><a href="visium-part-ii.html#cb162-60" tabindex="-1"></a> [<span class="dv">77</span>] R6_2.<span class="fl">5.1</span>                    mime_0<span class="fl">.12</span>                  </span>
<span id="cb162-61"><a href="visium-part-ii.html#cb162-61" tabindex="-1"></a> [<span class="dv">79</span>] textshaping_0.<span class="fl">4.0</span>           colorspace_2<span class="fl">.1</span><span class="dv">-0</span>           </span>
<span id="cb162-62"><a href="visium-part-ii.html#cb162-62" tabindex="-1"></a> [<span class="dv">81</span>] scattermore_1<span class="fl">.2</span>             Cairo_1<span class="fl">.6</span><span class="dv">-2</span>                </span>
<span id="cb162-63"><a href="visium-part-ii.html#cb162-63" tabindex="-1"></a> [<span class="dv">83</span>] gtools_3.<span class="fl">9.5</span>                R.methodsS3_1.<span class="fl">8.2</span>          </span>
<span id="cb162-64"><a href="visium-part-ii.html#cb162-64" tabindex="-1"></a> [<span class="dv">85</span>] utf8_1.<span class="fl">2.4</span>                  tidyr_1.<span class="fl">3.1</span>                </span>
<span id="cb162-65"><a href="visium-part-ii.html#cb162-65" tabindex="-1"></a> [<span class="dv">87</span>] generics_0.<span class="fl">1.3</span>              data.table_1.<span class="fl">15.4</span>          </span>
<span id="cb162-66"><a href="visium-part-ii.html#cb162-66" tabindex="-1"></a> [<span class="dv">89</span>] FNN_1.<span class="fl">1.4</span>                   httr_1.<span class="fl">4.7</span>                 </span>
<span id="cb162-67"><a href="visium-part-ii.html#cb162-67" tabindex="-1"></a> [<span class="dv">91</span>] htmlwidgets_1.<span class="fl">6.4</span>           S4Arrays_1.<span class="fl">4.1</span>             </span>
<span id="cb162-68"><a href="visium-part-ii.html#cb162-68" tabindex="-1"></a> [<span class="dv">93</span>] scatterpie_0.<span class="fl">2.3</span>            uwot_0.<span class="fl">2.2</span>                 </span>
<span id="cb162-69"><a href="visium-part-ii.html#cb162-69" tabindex="-1"></a> [<span class="dv">95</span>] pkgconfig_2.<span class="fl">0.3</span>             gtable_0.<span class="fl">3.5</span>               </span>
<span id="cb162-70"><a href="visium-part-ii.html#cb162-70" tabindex="-1"></a> [<span class="dv">97</span>] ComplexHeatmap_2.<span class="fl">20.0</span>       GiottoVisuals_0.<span class="fl">2.4</span>        </span>
<span id="cb162-71"><a href="visium-part-ii.html#cb162-71" tabindex="-1"></a> [<span class="dv">99</span>] SingleCellExperiment_1.<span class="fl">26.0</span> XVector_0.<span class="fl">44.0</span>             </span>
<span id="cb162-72"><a href="visium-part-ii.html#cb162-72" tabindex="-1"></a>[<span class="dv">101</span>] htmltools_0.<span class="dv">5</span>.<span class="fl">8.1</span>           bookdown_0<span class="fl">.40</span>              </span>
<span id="cb162-73"><a href="visium-part-ii.html#cb162-73" tabindex="-1"></a>[<span class="dv">103</span>] clue_0<span class="fl">.3</span><span class="dv">-65</span>                 scales_1.<span class="fl">3.0</span>               </span>
<span id="cb162-74"><a href="visium-part-ii.html#cb162-74" tabindex="-1"></a>[<span class="dv">105</span>] Biobase_2.<span class="fl">64.0</span>              GiottoUtils_0.<span class="fl">1.10</span>         </span>
<span id="cb162-75"><a href="visium-part-ii.html#cb162-75" tabindex="-1"></a>[<span class="dv">107</span>] png_0<span class="fl">.1</span><span class="dv">-8</span>                   SpatialExperiment_1.<span class="fl">14.0</span>   </span>
<span id="cb162-76"><a href="visium-part-ii.html#cb162-76" tabindex="-1"></a>[<span class="dv">109</span>] scran_1.<span class="fl">32.0</span>                ggfun_0.<span class="fl">1.5</span>                </span>
<span id="cb162-77"><a href="visium-part-ii.html#cb162-77" tabindex="-1"></a>[<span class="dv">111</span>] knitr_1<span class="fl">.48</span>                  rstudioapi_0.<span class="fl">16.0</span>          </span>
<span id="cb162-78"><a href="visium-part-ii.html#cb162-78" tabindex="-1"></a>[<span class="dv">113</span>] reshape2_1.<span class="fl">4.4</span>              rjson_0.<span class="fl">2.21</span>               </span>
<span id="cb162-79"><a href="visium-part-ii.html#cb162-79" tabindex="-1"></a>[<span class="dv">115</span>] checkmate_2.<span class="fl">3.1</span>             cachem_1.<span class="fl">1.0</span>               </span>
<span id="cb162-80"><a href="visium-part-ii.html#cb162-80" tabindex="-1"></a>[<span class="dv">117</span>] GlobalOptions_0.<span class="fl">1.2</span>         stringr_1.<span class="fl">5.1</span>              </span>
<span id="cb162-81"><a href="visium-part-ii.html#cb162-81" tabindex="-1"></a>[<span class="dv">119</span>] parallel_4.<span class="fl">4.1</span>              miniUI_0.<span class="dv">1</span>.<span class="fl">1.1</span>             </span>
<span id="cb162-82"><a href="visium-part-ii.html#cb162-82" tabindex="-1"></a>[<span class="dv">121</span>] RcppZiggurat_0.<span class="fl">1.6</span>          pillar_1.<span class="fl">9.0</span>               </span>
<span id="cb162-83"><a href="visium-part-ii.html#cb162-83" tabindex="-1"></a>[<span class="dv">123</span>] grid_4.<span class="fl">4.1</span>                  vctrs_0.<span class="fl">6.5</span>                </span>
<span id="cb162-84"><a href="visium-part-ii.html#cb162-84" tabindex="-1"></a>[<span class="dv">125</span>] promises_1.<span class="fl">3.0</span>              BiocSingular_1.<span class="fl">20.0</span>        </span>
<span id="cb162-85"><a href="visium-part-ii.html#cb162-85" tabindex="-1"></a>[<span class="dv">127</span>] beachmat_2.<span class="fl">20.0</span>             xtable_1<span class="fl">.8</span><span class="dv">-4</span>               </span>
<span id="cb162-86"><a href="visium-part-ii.html#cb162-86" tabindex="-1"></a>[<span class="dv">129</span>] cluster_2.<span class="fl">1.6</span>               evaluate_0.<span class="fl">24.0</span>            </span>
<span id="cb162-87"><a href="visium-part-ii.html#cb162-87" tabindex="-1"></a>[<span class="dv">131</span>] magick_2.<span class="fl">8.4</span>                cli_3.<span class="fl">6.3</span>                  </span>
<span id="cb162-88"><a href="visium-part-ii.html#cb162-88" tabindex="-1"></a>[<span class="dv">133</span>] locfit_1<span class="fl">.5-9.10</span>             compiler_4.<span class="fl">4.1</span>             </span>
<span id="cb162-89"><a href="visium-part-ii.html#cb162-89" tabindex="-1"></a>[<span class="dv">135</span>] rlang_1.<span class="fl">1.4</span>                 crayon_1.<span class="fl">5.3</span>               </span>
<span id="cb162-90"><a href="visium-part-ii.html#cb162-90" tabindex="-1"></a>[<span class="dv">137</span>] labeling_0.<span class="fl">4.3</span>              plyr_1.<span class="fl">8.9</span>                 </span>
<span id="cb162-91"><a href="visium-part-ii.html#cb162-91" tabindex="-1"></a>[<span class="dv">139</span>] stringi_1.<span class="fl">8.4</span>               viridisLite_0.<span class="fl">4.2</span>          </span>
<span id="cb162-92"><a href="visium-part-ii.html#cb162-92" tabindex="-1"></a>[<span class="dv">141</span>] deldir_2<span class="fl">.0</span><span class="dv">-4</span>                BiocParallel_1.<span class="fl">38.0</span>        </span>
<span id="cb162-93"><a href="visium-part-ii.html#cb162-93" tabindex="-1"></a>[<span class="dv">143</span>] munsell_0.<span class="fl">5.1</span>               lazyeval_0.<span class="fl">2.2</span>             </span>
<span id="cb162-94"><a href="visium-part-ii.html#cb162-94" tabindex="-1"></a>[<span class="dv">145</span>] Matrix_1<span class="fl">.7</span><span class="dv">-0</span>                sparseMatrixStats_1.<span class="fl">16.0</span>   </span>
<span id="cb162-95"><a href="visium-part-ii.html#cb162-95" tabindex="-1"></a>[<span class="dv">147</span>] ggplot2_3.<span class="fl">5.1</span>               statmod_1.<span class="fl">5.0</span>              </span>
<span id="cb162-96"><a href="visium-part-ii.html#cb162-96" tabindex="-1"></a>[<span class="dv">149</span>] SummarizedExperiment_1.<span class="fl">34.0</span> Rfast_2.<span class="fl">1.0</span>                </span>
<span id="cb162-97"><a href="visium-part-ii.html#cb162-97" tabindex="-1"></a>[<span class="dv">151</span>] memoise_2.<span class="fl">0.1</span>               igraph_2.<span class="fl">0.3</span>               </span>
<span id="cb162-98"><a href="visium-part-ii.html#cb162-98" tabindex="-1"></a>[<span class="dv">153</span>] bslib_0.<span class="fl">7.0</span>                 RcppParallel_5.<span class="fl">1.8</span>  </span></code></pre></div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="visium-part-i.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="visium-hd.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
