<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>10 Xenium | Workshop: Spatial multi-omics data analysis with Giotto Suite</title>
  <meta name="description" content="10 Xenium | Workshop: Spatial multi-omics data analysis with Giotto Suite" />
  <meta name="generator" content="bookdown 0.39 and GitBook 2.6.7" />

  <meta property="og:title" content="10 Xenium | Workshop: Spatial multi-omics data analysis with Giotto Suite" />
  <meta property="og:type" content="book" />
  
  
  <meta name="github-repo" content="drieslab/giotto_workshop_2024" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="10 Xenium | Workshop: Spatial multi-omics data analysis with Giotto Suite" />
  
  
  

<meta name="author" content="Ruben Dries, Jiaji George Chen, Joselyn Cristina ChÃ¡vez-Fuentes, Junxiang Xu ,Edward Ruiz, Jeff Sheridan, Iqra Amin, Wen Wang" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="visium-hd.html"/>
<link rel="next" href="spatial-proteomics-multiplexed-immunofluorescence.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Giotto Suite Workshop 2024</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#instructors"><i class="fa fa-check"></i><b>1.1</b> Instructors</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#topics-and-schedule"><i class="fa fa-check"></i><b>1.2</b> Topics and Schedule:</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#license"><i class="fa fa-check"></i><b>1.3</b> License</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="datasets-packages.html"><a href="datasets-packages.html"><i class="fa fa-check"></i><b>2</b> Datasets &amp; Packages</a>
<ul>
<li class="chapter" data-level="2.1" data-path="datasets-packages.html"><a href="datasets-packages.html#datasets-to-download"><i class="fa fa-check"></i><b>2.1</b> Datasets to download</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="datasets-packages.html"><a href="datasets-packages.html#zenodo-repository"><i class="fa fa-check"></i><b>2.1.1</b> Zenodo repository</a></li>
<li class="chapter" data-level="2.1.2" data-path="datasets-packages.html"><a href="datasets-packages.html#x-genomics-visium-mouse-brain-section-coronal-dataset"><i class="fa fa-check"></i><b>2.1.2</b> 10X Genomics Visium Mouse Brain Section (Coronal) dataset</a></li>
<li class="chapter" data-level="2.1.3" data-path="datasets-packages.html"><a href="datasets-packages.html#x-genomics-visium-hd-ffpe-human-colon-cancer"><i class="fa fa-check"></i><b>2.1.3</b> 10X Genomics Visium HD: FFPE Human Colon Cancer</a></li>
<li class="chapter" data-level="2.1.4" data-path="datasets-packages.html"><a href="datasets-packages.html#x-genomics-multi-modal-dataset"><i class="fa fa-check"></i><b>2.1.4</b> 10X Genomics multi-modal dataset</a></li>
<li class="chapter" data-level="2.1.5" data-path="datasets-packages.html"><a href="datasets-packages.html#x-genomics-multi-omics-visium-cytassist-human-tonsil-dataset"><i class="fa fa-check"></i><b>2.1.5</b> 10X Genomics multi-omics Visium CytAssist Human Tonsil dataset</a></li>
<li class="chapter" data-level="2.1.6" data-path="datasets-packages.html"><a href="datasets-packages.html#x-genomics-human-prostate-cancer-adenocarcinoma-with-invasive-carcinoma-ffpe"><i class="fa fa-check"></i><b>2.1.6</b> 10X Genomics Human Prostate Cancer Adenocarcinoma with Invasive Carcinoma (FFPE)</a></li>
<li class="chapter" data-level="2.1.7" data-path="datasets-packages.html"><a href="datasets-packages.html#x-genomics-normal-human-prostate-ffpe"><i class="fa fa-check"></i><b>2.1.7</b> 10X Genomics Normal Human Prostate (FFPE)</a></li>
<li class="chapter" data-level="2.1.8" data-path="datasets-packages.html"><a href="datasets-packages.html#xenium"><i class="fa fa-check"></i><b>2.1.8</b> Xenium</a></li>
<li class="chapter" data-level="2.1.9" data-path="datasets-packages.html"><a href="datasets-packages.html#merfish-cortex-dataset"><i class="fa fa-check"></i><b>2.1.9</b> MERFISH cortex dataset</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="datasets-packages.html"><a href="datasets-packages.html#needed-packages"><i class="fa fa-check"></i><b>2.2</b> Needed packages</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="datasets-packages.html"><a href="datasets-packages.html#cran-dependencies"><i class="fa fa-check"></i><b>2.2.1</b> CRAN dependencies:</a></li>
<li class="chapter" data-level="2.2.2" data-path="datasets-packages.html"><a href="datasets-packages.html#terra-installation"><i class="fa fa-check"></i><b>2.2.2</b> <em>terra</em> installation</a></li>
<li class="chapter" data-level="2.2.3" data-path="datasets-packages.html"><a href="datasets-packages.html#matrix-installation"><i class="fa fa-check"></i><b>2.2.3</b> <em>Matrix</em> installation</a></li>
<li class="chapter" data-level="2.2.4" data-path="datasets-packages.html"><a href="datasets-packages.html#rtools-installation"><i class="fa fa-check"></i><b>2.2.4</b> <em>Rtools</em> installation</a></li>
<li class="chapter" data-level="2.2.5" data-path="datasets-packages.html"><a href="datasets-packages.html#giotto-installation"><i class="fa fa-check"></i><b>2.2.5</b> <em>Giotto</em> installation</a></li>
<li class="chapter" data-level="2.2.6" data-path="datasets-packages.html"><a href="datasets-packages.html#irlba-install"><i class="fa fa-check"></i><b>2.2.6</b> <em>irlba</em> install</a></li>
<li class="chapter" data-level="2.2.7" data-path="datasets-packages.html"><a href="datasets-packages.html#arrow-install"><i class="fa fa-check"></i><b>2.2.7</b> <em>arrow</em> install</a></li>
<li class="chapter" data-level="2.2.8" data-path="datasets-packages.html"><a href="datasets-packages.html#bioconductor-dependencies"><i class="fa fa-check"></i><b>2.2.8</b> Bioconductor dependencies:</a></li>
<li class="chapter" data-level="2.2.9" data-path="datasets-packages.html"><a href="datasets-packages.html#cran-packages"><i class="fa fa-check"></i><b>2.2.9</b> CRAN packages:</a></li>
<li class="chapter" data-level="2.2.10" data-path="datasets-packages.html"><a href="datasets-packages.html#packages-from-github"><i class="fa fa-check"></i><b>2.2.10</b> Packages from GitHub</a></li>
<li class="chapter" data-level="2.2.11" data-path="datasets-packages.html"><a href="datasets-packages.html#python-environments"><i class="fa fa-check"></i><b>2.2.11</b> Python environments</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html"><i class="fa fa-check"></i><b>3</b> Spatial omics technologies</a>
<ul>
<li class="chapter" data-level="3.1" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#presentation"><i class="fa fa-check"></i><b>3.1</b> Presentation</a></li>
<li class="chapter" data-level="3.2" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#short-summary"><i class="fa fa-check"></i><b>3.2</b> Short summary</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#why-do-we-need-spatial-omics-technologies"><i class="fa fa-check"></i><b>3.2.1</b> Why do we need spatial omics technologies?</a></li>
<li class="chapter" data-level="3.2.2" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#what-is-spatial-omics"><i class="fa fa-check"></i><b>3.2.2</b> What is spatial omics?</a></li>
<li class="chapter" data-level="3.2.3" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#what-are-the-main-spatial-omics-technologies"><i class="fa fa-check"></i><b>3.2.3</b> What are the main spatial omics technologies?</a></li>
<li class="chapter" data-level="3.2.4" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#other-spatial-omics-atac-seq-cuttag-lipidomics-etc"><i class="fa fa-check"></i><b>3.2.4</b> Other Spatial omics: ATAC-seq, CUT&amp;Tag, lipidomics, etc</a></li>
<li class="chapter" data-level="3.2.5" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#what-are-the-different-types-of-spatial-downstream-analyses"><i class="fa fa-check"></i><b>3.2.5</b> What are the different types of spatial downstream analyses?</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html"><i class="fa fa-check"></i><b>4</b> Introduction to the Giotto package</a>
<ul>
<li class="chapter" data-level="4.1" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html#presentation-1"><i class="fa fa-check"></i><b>4.1</b> Presentation</a></li>
<li class="chapter" data-level="4.2" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html#ecosystem"><i class="fa fa-check"></i><b>4.2</b> Ecosystem</a></li>
<li class="chapter" data-level="4.3" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html#installation-python-environment"><i class="fa fa-check"></i><b>4.3</b> Installation + python environment</a></li>
<li class="chapter" data-level="4.4" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html#giotto-instructions"><i class="fa fa-check"></i><b>4.4</b> Giotto instructions</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html"><i class="fa fa-check"></i><b>5</b> Data formatting and Pre-processing</a>
<ul>
<li class="chapter" data-level="5.1" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#data-formats"><i class="fa fa-check"></i><b>5.1</b> Data formats</a></li>
<li class="chapter" data-level="5.2" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#pre-processing"><i class="fa fa-check"></i><b>5.2</b> Pre-processing</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html"><i class="fa fa-check"></i><b>6</b> Creating a Giotto object</a>
<ul>
<li class="chapter" data-level="6.0.1" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#used-packages"><i class="fa fa-check"></i><b>6.0.1</b> Used packages</a></li>
<li class="chapter" data-level="6.1" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#overview"><i class="fa fa-check"></i><b>6.1</b> Overview</a></li>
<li class="chapter" data-level="6.2" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#from-matrix-locations"><i class="fa fa-check"></i><b>6.2</b> From matrix + locations</a></li>
<li class="chapter" data-level="6.3" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#from-subcellular-raw-data-transcripts-or-images-polygons"><i class="fa fa-check"></i><b>6.3</b> From subcellular raw data (transcripts or images) + polygons</a></li>
<li class="chapter" data-level="6.4" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#from-piece-wise"><i class="fa fa-check"></i><b>6.4</b> From piece-wise</a></li>
<li class="chapter" data-level="6.5" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#using-convenience-functions-for-popular-technologies-vizgen-xenium-cosmx"><i class="fa fa-check"></i><b>6.5</b> Using convenience functions for popular technologies (Vizgen, Xenium, CosMx, â¦)</a></li>
<li class="chapter" data-level="6.6" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#spatial-plots"><i class="fa fa-check"></i><b>6.6</b> Spatial plots</a></li>
<li class="chapter" data-level="6.7" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#subsetting"><i class="fa fa-check"></i><b>6.7</b> Subsetting</a></li>
<li class="chapter" data-level="6.8" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#mini-objects-giottodata"><i class="fa fa-check"></i><b>6.8</b> Mini objects &amp; GiottoData</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="visium-part-i.html"><a href="visium-part-i.html"><i class="fa fa-check"></i><b>7</b> Visium Part I</a>
<ul>
<li class="chapter" data-level="7.1" data-path="visium-part-i.html"><a href="visium-part-i.html#the-visium-technology"><i class="fa fa-check"></i><b>7.1</b> The Visium technology</a></li>
<li class="chapter" data-level="7.2" data-path="visium-part-i.html"><a href="visium-part-i.html#introduction-to-the-spatial-dataset"><i class="fa fa-check"></i><b>7.2</b> Introduction to the spatial dataset</a></li>
<li class="chapter" data-level="7.3" data-path="visium-part-i.html"><a href="visium-part-i.html#download-dataset"><i class="fa fa-check"></i><b>7.3</b> Download dataset</a></li>
<li class="chapter" data-level="7.4" data-path="visium-part-i.html"><a href="visium-part-i.html#create-the-giotto-object"><i class="fa fa-check"></i><b>7.4</b> Create the Giotto object</a></li>
<li class="chapter" data-level="7.5" data-path="visium-part-i.html"><a href="visium-part-i.html#subset-on-spots-that-were-covered-by-tissue"><i class="fa fa-check"></i><b>7.5</b> Subset on spots that were covered by tissue</a></li>
<li class="chapter" data-level="7.6" data-path="visium-part-i.html"><a href="visium-part-i.html#quality-control"><i class="fa fa-check"></i><b>7.6</b> Quality control</a></li>
<li class="chapter" data-level="7.7" data-path="visium-part-i.html"><a href="visium-part-i.html#filtering"><i class="fa fa-check"></i><b>7.7</b> Filtering</a></li>
<li class="chapter" data-level="7.8" data-path="visium-part-i.html"><a href="visium-part-i.html#normalization"><i class="fa fa-check"></i><b>7.8</b> Normalization</a></li>
<li class="chapter" data-level="7.9" data-path="visium-part-i.html"><a href="visium-part-i.html#feature-selection"><i class="fa fa-check"></i><b>7.9</b> Feature selection</a>
<ul>
<li class="chapter" data-level="7.9.1" data-path="visium-part-i.html"><a href="visium-part-i.html#highly-variable-features"><i class="fa fa-check"></i><b>7.9.1</b> Highly Variable Features:</a></li>
</ul></li>
<li class="chapter" data-level="7.10" data-path="visium-part-i.html"><a href="visium-part-i.html#dimension-reduction"><i class="fa fa-check"></i><b>7.10</b> Dimension Reduction</a>
<ul>
<li class="chapter" data-level="7.10.1" data-path="visium-part-i.html"><a href="visium-part-i.html#pca"><i class="fa fa-check"></i><b>7.10.1</b> PCA</a></li>
<li class="chapter" data-level="7.10.2" data-path="visium-part-i.html"><a href="visium-part-i.html#umap"><i class="fa fa-check"></i><b>7.10.2</b> UMAP</a></li>
<li class="chapter" data-level="7.10.3" data-path="visium-part-i.html"><a href="visium-part-i.html#t-sne"><i class="fa fa-check"></i><b>7.10.3</b> t-SNE</a></li>
</ul></li>
<li class="chapter" data-level="7.11" data-path="visium-part-i.html"><a href="visium-part-i.html#clustering"><i class="fa fa-check"></i><b>7.11</b> Clustering</a>
<ul>
<li class="chapter" data-level="7.11.1" data-path="visium-part-i.html"><a href="visium-part-i.html#calculate-leiden-clustering"><i class="fa fa-check"></i><b>7.11.1</b> Calculate Leiden clustering</a></li>
<li class="chapter" data-level="7.11.2" data-path="visium-part-i.html"><a href="visium-part-i.html#calculate-louvain-clustering"><i class="fa fa-check"></i><b>7.11.2</b> Calculate Louvain clustering</a></li>
</ul></li>
<li class="chapter" data-level="7.12" data-path="visium-part-i.html"><a href="visium-part-i.html#save-the-object"><i class="fa fa-check"></i><b>7.12</b> Save the object</a></li>
<li class="chapter" data-level="7.13" data-path="visium-part-i.html"><a href="visium-part-i.html#session-info"><i class="fa fa-check"></i><b>7.13</b> Session info</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="visium-part-ii.html"><a href="visium-part-ii.html"><i class="fa fa-check"></i><b>8</b> Visium Part II</a>
<ul>
<li class="chapter" data-level="8.1" data-path="visium-part-ii.html"><a href="visium-part-ii.html#load-the-object"><i class="fa fa-check"></i><b>8.1</b> Load the object</a></li>
<li class="chapter" data-level="8.2" data-path="visium-part-ii.html"><a href="visium-part-ii.html#differential-expression"><i class="fa fa-check"></i><b>8.2</b> Differential expression</a>
<ul>
<li class="chapter" data-level="8.2.1" data-path="visium-part-ii.html"><a href="visium-part-ii.html#gini-markers"><i class="fa fa-check"></i><b>8.2.1</b> Gini markers</a></li>
<li class="chapter" data-level="8.2.2" data-path="visium-part-ii.html"><a href="visium-part-ii.html#scran-markers"><i class="fa fa-check"></i><b>8.2.2</b> Scran markers</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="visium-part-ii.html"><a href="visium-part-ii.html#enrichment-deconvolution"><i class="fa fa-check"></i><b>8.3</b> Enrichment &amp; Deconvolution</a>
<ul>
<li class="chapter" data-level="8.3.1" data-path="visium-part-ii.html"><a href="visium-part-ii.html#pagerank"><i class="fa fa-check"></i><b>8.3.1</b> PAGE/Rank</a></li>
<li class="chapter" data-level="8.3.2" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatialdwls"><i class="fa fa-check"></i><b>8.3.2</b> SpatialDWLS</a></li>
</ul></li>
<li class="chapter" data-level="8.4" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatial-expression-patterns"><i class="fa fa-check"></i><b>8.4</b> Spatial expression patterns</a>
<ul>
<li class="chapter" data-level="8.4.1" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatial-variable-genes"><i class="fa fa-check"></i><b>8.4.1</b> Spatial variable genes</a></li>
<li class="chapter" data-level="8.4.2" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatial-co-expression-modules"><i class="fa fa-check"></i><b>8.4.2</b> Spatial co-expression modules</a></li>
</ul></li>
<li class="chapter" data-level="8.5" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatially-informed-clusters"><i class="fa fa-check"></i><b>8.5</b> Spatially informed clusters</a></li>
<li class="chapter" data-level="8.6" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatial-domains-hmrf"><i class="fa fa-check"></i><b>8.6</b> Spatial domains HMRF</a></li>
<li class="chapter" data-level="8.7" data-path="visium-part-ii.html"><a href="visium-part-ii.html#interactive-tools"><i class="fa fa-check"></i><b>8.7</b> Interactive tools</a></li>
<li class="chapter" data-level="8.8" data-path="visium-part-ii.html"><a href="visium-part-ii.html#session-info-1"><i class="fa fa-check"></i><b>8.8</b> Session info</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="visium-hd.html"><a href="visium-hd.html"><i class="fa fa-check"></i><b>9</b> Visium HD</a>
<ul>
<li class="chapter" data-level="9.1" data-path="visium-hd.html"><a href="visium-hd.html#objective"><i class="fa fa-check"></i><b>9.1</b> Objective</a></li>
<li class="chapter" data-level="9.2" data-path="visium-hd.html"><a href="visium-hd.html#background"><i class="fa fa-check"></i><b>9.2</b> Background</a>
<ul>
<li class="chapter" data-level="9.2.1" data-path="visium-hd.html"><a href="visium-hd.html#visium-hd-technology"><i class="fa fa-check"></i><b>9.2.1</b> Visium HD Technology</a></li>
<li class="chapter" data-level="9.2.2" data-path="visium-hd.html"><a href="visium-hd.html#colorectal-cancer-sample"><i class="fa fa-check"></i><b>9.2.2</b> Colorectal Cancer Sample</a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="visium-hd.html"><a href="visium-hd.html#data-ingestion"><i class="fa fa-check"></i><b>9.3</b> Data Ingestion</a>
<ul>
<li class="chapter" data-level="9.3.1" data-path="visium-hd.html"><a href="visium-hd.html#visium-hd-output-data-format"><i class="fa fa-check"></i><b>9.3.1</b> Visium HD output data format</a></li>
<li class="chapter" data-level="9.3.2" data-path="visium-hd.html"><a href="visium-hd.html#read-in-raw-data"><i class="fa fa-check"></i><b>9.3.2</b> Read in raw data</a></li>
<li class="chapter" data-level="9.3.3" data-path="visium-hd.html"><a href="visium-hd.html#giotto-visium-hd-convenience-function"><i class="fa fa-check"></i><b>9.3.3</b> Giotto Visium HD convenience function</a></li>
</ul></li>
<li class="chapter" data-level="9.4" data-path="visium-hd.html"><a href="visium-hd.html#tiling-and-aggregation"><i class="fa fa-check"></i><b>9.4</b> Tiling and aggregation</a></li>
<li class="chapter" data-level="9.5" data-path="visium-hd.html"><a href="visium-hd.html#scalability-and-projection-functions"><i class="fa fa-check"></i><b>9.5</b> Scalability and projection functions</a></li>
<li class="chapter" data-level="9.6" data-path="visium-hd.html"><a href="visium-hd.html#spatial-expression-patterns-1"><i class="fa fa-check"></i><b>9.6</b> Spatial expression patterns</a></li>
<li class="chapter" data-level="9.7" data-path="visium-hd.html"><a href="visium-hd.html#spatial-co-expression-modules-1"><i class="fa fa-check"></i><b>9.7</b> Spatial co-expression modules</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="xenium-1.html"><a href="xenium-1.html"><i class="fa fa-check"></i><b>10</b> Xenium</a>
<ul>
<li class="chapter" data-level="10.1" data-path="xenium-1.html"><a href="xenium-1.html#introduction-to-spatial-dataset"><i class="fa fa-check"></i><b>10.1</b> Introduction to spatial dataset</a></li>
<li class="chapter" data-level="10.2" data-path="xenium-1.html"><a href="xenium-1.html#output-directory-structure"><i class="fa fa-check"></i><b>10.2</b> Output directory structure</a></li>
<li class="chapter" data-level="10.3" data-path="xenium-1.html"><a href="xenium-1.html#mini-xenium-dataset"><i class="fa fa-check"></i><b>10.3</b> Mini Xenium Dataset</a></li>
<li class="chapter" data-level="10.4" data-path="xenium-1.html"><a href="xenium-1.html#data-prep"><i class="fa fa-check"></i><b>10.4</b> Data prep</a>
<ul>
<li class="chapter" data-level="10.4.1" data-path="xenium-1.html"><a href="xenium-1.html#image-conversion-may-change"><i class="fa fa-check"></i><b>10.4.1</b> Image conversion (may change)</a></li>
</ul></li>
<li class="chapter" data-level="10.5" data-path="xenium-1.html"><a href="xenium-1.html#convenience-function"><i class="fa fa-check"></i><b>10.5</b> Convenience function</a>
<ul>
<li class="chapter" data-level="10.5.1" data-path="xenium-1.html"><a href="xenium-1.html#specific-filepaths"><i class="fa fa-check"></i><b>10.5.1</b> Specific filepaths</a></li>
<li class="chapter" data-level="10.5.2" data-path="xenium-1.html"><a href="xenium-1.html#quality-value"><i class="fa fa-check"></i><b>10.5.2</b> Quality value</a></li>
<li class="chapter" data-level="10.5.3" data-path="xenium-1.html"><a href="xenium-1.html#transcript-type-splitting"><i class="fa fa-check"></i><b>10.5.3</b> Transcript type splitting</a></li>
<li class="chapter" data-level="10.5.4" data-path="xenium-1.html"><a href="xenium-1.html#centroids-calculation"><i class="fa fa-check"></i><b>10.5.4</b> Centroids calculation</a></li>
<li class="chapter" data-level="10.5.5" data-path="xenium-1.html"><a href="xenium-1.html#simple-visualization"><i class="fa fa-check"></i><b>10.5.5</b> Simple visualization</a></li>
</ul></li>
<li class="chapter" data-level="10.6" data-path="xenium-1.html"><a href="xenium-1.html#piecewise-loading"><i class="fa fa-check"></i><b>10.6</b> Piecewise loading</a>
<ul>
<li class="chapter" data-level="10.6.1" data-path="xenium-1.html"><a href="xenium-1.html#load-giottopoints-transcripts"><i class="fa fa-check"></i><b>10.6.1</b> load giottoPoints transcripts</a></li>
<li class="chapter" data-level="10.6.2" data-path="xenium-1.html"><a href="xenium-1.html#optional-loading-pre-aggregated-data"><i class="fa fa-check"></i><b>10.6.2</b> (optional) Loading pre-aggregated data</a></li>
</ul></li>
<li class="chapter" data-level="10.7" data-path="xenium-1.html"><a href="xenium-1.html#xenium-images"><i class="fa fa-check"></i><b>10.7</b> Xenium Images</a>
<ul>
<li class="chapter" data-level="10.7.1" data-path="xenium-1.html"><a href="xenium-1.html#image-metadata"><i class="fa fa-check"></i><b>10.7.1</b> Image metadata</a></li>
<li class="chapter" data-level="10.7.2" data-path="xenium-1.html"><a href="xenium-1.html#image-loading"><i class="fa fa-check"></i><b>10.7.2</b> Image loading</a></li>
</ul></li>
<li class="chapter" data-level="10.8" data-path="xenium-1.html"><a href="xenium-1.html#read-in-raw-data-1"><i class="fa fa-check"></i><b>10.8</b> Read in raw data</a></li>
<li class="chapter" data-level="10.9" data-path="xenium-1.html"><a href="xenium-1.html#spatial-aggregation"><i class="fa fa-check"></i><b>10.9</b> Spatial aggregation</a></li>
<li class="chapter" data-level="10.10" data-path="xenium-1.html"><a href="xenium-1.html#aggregate-analyses-workflow"><i class="fa fa-check"></i><b>10.10</b> Aggregate analyses workflow</a></li>
<li class="chapter" data-level="10.11" data-path="xenium-1.html"><a href="xenium-1.html#niche-clustering"><i class="fa fa-check"></i><b>10.11</b> Niche clustering</a></li>
<li class="chapter" data-level="10.12" data-path="xenium-1.html"><a href="xenium-1.html#pseudovisium"><i class="fa fa-check"></i><b>10.12</b> Pseudovisium</a></li>
<li class="chapter" data-level="10.13" data-path="xenium-1.html"><a href="xenium-1.html#transcript-enrichment-gsea"><i class="fa fa-check"></i><b>10.13</b> Transcript enrichment GSEA</a></li>
<li class="chapter" data-level="10.14" data-path="xenium-1.html"><a href="xenium-1.html#spatial-location-analysis"><i class="fa fa-check"></i><b>10.14</b> Spatial location analysis</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html"><i class="fa fa-check"></i><b>11</b> Spatial proteomics: Multiplexed Immunofluorescence</a>
<ul>
<li class="chapter" data-level="11.1" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#spatial-proteomics-technologies"><i class="fa fa-check"></i><b>11.1</b> Spatial Proteomics Technologies</a></li>
<li class="chapter" data-level="11.2" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#raw-data-type-coming-out-of-different-technologies"><i class="fa fa-check"></i><b>11.2</b> Raw data type coming out of different technologies</a>
<ul>
<li class="chapter" data-level="11.2.1" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#use-ome.tiff-as-an-example-output-data-to-begin-with"><i class="fa fa-check"></i><b>11.2.1</b> Use ome.tiff as an example output data to begin with</a></li>
<li class="chapter" data-level="11.2.2" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#use-single-channel-images-as-an-example-output-data-to-begin-with"><i class="fa fa-check"></i><b>11.2.2</b> Use single channel images as an example output data to begin with</a></li>
</ul></li>
<li class="chapter" data-level="11.3" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#cell-segmentation-file-to-get-single-cell-level-protein-expression"><i class="fa fa-check"></i><b>11.3</b> Cell Segmentation file to get single cell level protein expression</a>
<ul>
<li class="chapter" data-level="11.3.1" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#using-segmentation-output-file-from-deepcellmesmer-as-an-example."><i class="fa fa-check"></i><b>11.3.1</b> Using segmentation output file from DeepCell(mesmer) as an example.</a></li>
<li class="chapter" data-level="11.3.2" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#using-giotto-wrapper-of-cellpose-to-perform-segmentation"><i class="fa fa-check"></i><b>11.3.2</b> Using Giotto wrapper of Cellpose to perform segmentation</a></li>
</ul></li>
<li class="chapter" data-level="11.4" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#create-a-giotto-object-using-list-of-gitto-large-images-and-polygons"><i class="fa fa-check"></i><b>11.4</b> Create a Giotto Object using list of gitto large images and polygons</a>
<ul>
<li class="chapter" data-level="11.4.1" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#overlap-to-matrix"><i class="fa fa-check"></i><b>11.4.1</b> Overlap to matrix</a></li>
<li class="chapter" data-level="11.4.2" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#manipulate-expression-information"><i class="fa fa-check"></i><b>11.4.2</b> Manipulate Expression information</a></li>
<li class="chapter" data-level="11.4.3" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#rescale-polygons"><i class="fa fa-check"></i><b>11.4.3</b> Rescale polygons</a></li>
<li class="chapter" data-level="11.4.4" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#perform-clustering-and-differential-expression"><i class="fa fa-check"></i><b>11.4.4</b> Perform clustering and differential expression</a></li>
<li class="chapter" data-level="11.4.5" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#give-the-cluster-an-annotation-based-on-expression-values"><i class="fa fa-check"></i><b>11.4.5</b> Give the cluster an annotation based on expression values</a></li>
<li class="chapter" data-level="11.4.6" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#spatial-network"><i class="fa fa-check"></i><b>11.4.6</b> spatial network</a></li>
<li class="chapter" data-level="11.4.7" data-path="spatial-proteomics-multiplexed-immunofluorescence.html"><a href="spatial-proteomics-multiplexed-immunofluorescence.html#cell-neighborhood-cell-typecell-type-interactions"><i class="fa fa-check"></i><b>11.4.7</b> Cell Neighborhood: Cell-Type/Cell-Type Interactions</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="12" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html"><i class="fa fa-check"></i><b>12</b> Working with multiple samples</a>
<ul>
<li class="chapter" data-level="12.1" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#objective-1"><i class="fa fa-check"></i><b>12.1</b> Objective</a></li>
<li class="chapter" data-level="12.2" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#background-1"><i class="fa fa-check"></i><b>12.2</b> Background</a>
<ul>
<li class="chapter" data-level="12.2.1" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#dataset"><i class="fa fa-check"></i><b>12.2.1</b> Dataset</a></li>
<li class="chapter" data-level="12.2.2" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#visium-technology"><i class="fa fa-check"></i><b>12.2.2</b> Visium technology</a></li>
</ul></li>
<li class="chapter" data-level="12.3" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#create-individual-giotto-objects"><i class="fa fa-check"></i><b>12.3</b> Create individual giotto objects</a>
<ul>
<li class="chapter" data-level="12.3.1" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#download-the-data"><i class="fa fa-check"></i><b>12.3.1</b> Download the data</a></li>
<li class="chapter" data-level="12.3.2" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#create-giotto-instructions"><i class="fa fa-check"></i><b>12.3.2</b> Create giotto instructions</a></li>
<li class="chapter" data-level="12.3.3" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#load-visium-data-into-giotto"><i class="fa fa-check"></i><b>12.3.3</b> Load visium data into Giotto</a></li>
<li class="chapter" data-level="12.3.4" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#healthy-prostate-tissue-coverage"><i class="fa fa-check"></i><b>12.3.4</b> Healthy prostate tissue coverage</a></li>
<li class="chapter" data-level="12.3.5" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#adenocarcinoma-prostate-tissue-coverage"><i class="fa fa-check"></i><b>12.3.5</b> Adenocarcinoma prostate tissue coverage</a></li>
<li class="chapter" data-level="12.3.6" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#showing-the-data-strucutre-for-the-inidividual-objects"><i class="fa fa-check"></i><b>12.3.6</b> Showing the data strucutre for the inidividual objects</a></li>
</ul></li>
<li class="chapter" data-level="12.4" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#join-giotto-objects"><i class="fa fa-check"></i><b>12.4</b> Join Giotto Objects</a></li>
<li class="chapter" data-level="12.5" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#visualizing-combined-datasets"><i class="fa fa-check"></i><b>12.5</b> Visualizing combined datasets</a>
<ul>
<li class="chapter" data-level="12.5.1" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#vizualizing-in-the-same-plot"><i class="fa fa-check"></i><b>12.5.1</b> Vizualizing in the same plot</a></li>
<li class="chapter" data-level="12.5.2" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#vizualizing-on-separate-plots"><i class="fa fa-check"></i><b>12.5.2</b> Vizualizing on separate plots</a></li>
</ul></li>
<li class="chapter" data-level="12.6" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#splitting-combined-dataset"><i class="fa fa-check"></i><b>12.6</b> Splitting combined dataset</a></li>
<li class="chapter" data-level="12.7" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#analyzing-joined-objects"><i class="fa fa-check"></i><b>12.7</b> Analyzing joined objects</a>
<ul>
<li class="chapter" data-level="12.7.1" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#normalization-and-adding-statistics"><i class="fa fa-check"></i><b>12.7.1</b> Normalization and adding statistics</a></li>
<li class="chapter" data-level="12.7.2" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#clustering-the-datasets"><i class="fa fa-check"></i><b>12.7.2</b> Clustering the datasets</a></li>
<li class="chapter" data-level="12.7.3" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#vizualizing-spatial-location-of-clusters"><i class="fa fa-check"></i><b>12.7.3</b> Vizualizing spatial location of clusters</a></li>
<li class="chapter" data-level="12.7.4" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#vizualizing-tissue-contribution-to-clusters"><i class="fa fa-check"></i><b>12.7.4</b> Vizualizing tissue contribution to clusters</a></li>
</ul></li>
<li class="chapter" data-level="12.8" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#perform-harmony-and-default-workflows"><i class="fa fa-check"></i><b>12.8</b> Perform Harmony and default workflows</a>
<ul>
<li class="chapter" data-level="12.8.1" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#clustering-harmonized-object"><i class="fa fa-check"></i><b>12.8.1</b> Clustering harmonized object</a></li>
<li class="chapter" data-level="12.8.2" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#vizualizing-the-tissue-contribution"><i class="fa fa-check"></i><b>12.8.2</b> Vizualizing the tissue contribution</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="13" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html"><i class="fa fa-check"></i><b>13</b> Spatial multi-modal analysis</a>
<ul>
<li class="chapter" data-level="13.1" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#overview-1"><i class="fa fa-check"></i><b>13.1</b> Overview</a></li>
<li class="chapter" data-level="13.2" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#spatial-manipulation"><i class="fa fa-check"></i><b>13.2</b> Spatial manipulation</a>
<ul>
<li class="chapter" data-level="13.2.1" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#spatial-transforms"><i class="fa fa-check"></i><b>13.2.1</b> Spatial transforms:</a></li>
<li class="chapter" data-level="13.2.2" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#spatial-utilities"><i class="fa fa-check"></i><b>13.2.2</b> Spatial utilities:</a></li>
<li class="chapter" data-level="13.2.3" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#spatial-classes"><i class="fa fa-check"></i><b>13.2.3</b> Spatial classes:</a></li>
</ul></li>
<li class="chapter" data-level="13.3" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#examples-of-the-simple-transforms-with-a-giottopolygon"><i class="fa fa-check"></i><b>13.3</b> Examples of the simple transforms with a giottoPolygon</a></li>
<li class="chapter" data-level="13.4" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#affine-transforms"><i class="fa fa-check"></i><b>13.4</b> Affine transforms</a></li>
<li class="chapter" data-level="13.5" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#image-transforms"><i class="fa fa-check"></i><b>13.5</b> Image transforms</a></li>
<li class="chapter" data-level="13.6" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#co-registration-of-datasets"><i class="fa fa-check"></i><b>13.6</b> Co-registration of datasets</a></li>
<li class="chapter" data-level="13.7" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#examples-in-giotto-suite-manuscript"><i class="fa fa-check"></i><b>13.7</b> Examples in giotto suite manuscript</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html"><i class="fa fa-check"></i><b>14</b> Multi-omics integration</a>
<ul>
<li class="chapter" data-level="14.1" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#the-cytassist-technology"><i class="fa fa-check"></i><b>14.1</b> The CytAssist technology</a></li>
<li class="chapter" data-level="14.2" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#introduction-to-the-spatial-dataset-1"><i class="fa fa-check"></i><b>14.2</b> Introduction to the spatial dataset</a></li>
<li class="chapter" data-level="14.3" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#download-dataset-1"><i class="fa fa-check"></i><b>14.3</b> Download dataset</a></li>
<li class="chapter" data-level="14.4" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#create-the-giotto-object-1"><i class="fa fa-check"></i><b>14.4</b> Create the Giotto object</a></li>
<li class="chapter" data-level="14.5" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#subset-on-spots-that-were-covered-by-tissue-1"><i class="fa fa-check"></i><b>14.5</b> Subset on spots that were covered by tissue</a></li>
<li class="chapter" data-level="14.6" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#rna-processing"><i class="fa fa-check"></i><b>14.6</b> RNA processing</a></li>
<li class="chapter" data-level="14.7" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#protein-processing"><i class="fa fa-check"></i><b>14.7</b> Protein processing</a></li>
<li class="chapter" data-level="14.8" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#multi-omics-integration-1"><i class="fa fa-check"></i><b>14.8</b> Multi-omics integration</a></li>
<li class="chapter" data-level="14.9" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#session-info-2"><i class="fa fa-check"></i><b>14.9</b> Session info</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html"><i class="fa fa-check"></i><b>15</b> Interoperability with other frameworks</a>
<ul>
<li class="chapter" data-level="15.1" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#load-giotto-object"><i class="fa fa-check"></i><b>15.1</b> Load Giotto object</a></li>
<li class="chapter" data-level="15.2" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#seurat"><i class="fa fa-check"></i><b>15.2</b> Seurat</a>
<ul>
<li class="chapter" data-level="15.2.1" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#conversion-of-giotto-object-to-seurat-object"><i class="fa fa-check"></i><b>15.2.1</b> Conversion of Giotto Object to Seurat Object</a></li>
<li class="chapter" data-level="15.2.2" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#conversion-of-seurat-object-back-to-giotto-object"><i class="fa fa-check"></i><b>15.2.2</b> Conversion of Seurat object Back to Giotto Object</a></li>
</ul></li>
<li class="chapter" data-level="15.3" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#spatialexperiment"><i class="fa fa-check"></i><b>15.3</b> SpatialExperiment</a>
<ul>
<li class="chapter" data-level="15.3.1" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#convert-giotto-object-to-spatialexperiment-object"><i class="fa fa-check"></i><b>15.3.1</b> Convert Giotto Object to SpatialExperiment Object</a></li>
<li class="chapter" data-level="15.3.2" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#conversion-of-spatialexperiment-object-back-to-giotto"><i class="fa fa-check"></i><b>15.3.2</b> Conversion of SpatialExperiment object back to Giotto</a></li>
</ul></li>
<li class="chapter" data-level="15.4" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#anndata"><i class="fa fa-check"></i><b>15.4</b> AnnData</a>
<ul>
<li class="chapter" data-level="15.4.1" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#load-required-libraries"><i class="fa fa-check"></i><b>15.4.1</b> Load Required Libraries</a></li>
<li class="chapter" data-level="15.4.2" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#specify-path-for-results"><i class="fa fa-check"></i><b>15.4.2</b> Specify Path for Results</a></li>
<li class="chapter" data-level="15.4.3" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#giotto-to-anndata"><i class="fa fa-check"></i><b>15.4.3</b> Giotto To AnnData</a></li>
<li class="chapter" data-level="15.4.4" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#convert-anndata-to-giotto"><i class="fa fa-check"></i><b>15.4.4</b> Convert AnnData to Giotto</a></li>
</ul></li>
<li class="chapter" data-level="15.5" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#create-mini-vizgen-object"><i class="fa fa-check"></i><b>15.5</b> Create mini Vizgen object</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html"><i class="fa fa-check"></i><b>16</b> Interoperability with isolated tools</a>
<ul>
<li class="chapter" data-level="16.1" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#spatial-niche-trajectory-analysis"><i class="fa fa-check"></i><b>16.1</b> Spatial niche trajectory analysis</a>
<ul>
<li class="chapter" data-level="16.1.1" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#dataset-download"><i class="fa fa-check"></i><b>16.1.1</b> Dataset download</a></li>
<li class="chapter" data-level="16.1.2" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#create-the-giotto-object-2"><i class="fa fa-check"></i><b>16.1.2</b> Create the Giotto object</a></li>
<li class="chapter" data-level="16.1.3" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#visualization"><i class="fa fa-check"></i><b>16.1.3</b> visualization</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="17" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html"><i class="fa fa-check"></i><b>17</b> Interactivity with the R/Spatial ecosystem</a>
<ul>
<li class="chapter" data-level="17.1" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#kriging"><i class="fa fa-check"></i><b>17.1</b> Kriging</a>
<ul>
<li class="chapter" data-level="17.1.1" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#visium-technology-1"><i class="fa fa-check"></i><b>17.1.1</b> Visium technology</a></li>
<li class="chapter" data-level="17.1.2" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#dataset-1"><i class="fa fa-check"></i><b>17.1.2</b> Dataset</a></li>
<li class="chapter" data-level="17.1.3" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#generating-a-geojson-file-of-nuclei-location"><i class="fa fa-check"></i><b>17.1.3</b> Generating a geojson file of nuclei location</a></li>
</ul></li>
<li class="chapter" data-level="17.2" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#downloading-the-dataset"><i class="fa fa-check"></i><b>17.2</b> Downloading the dataset</a></li>
<li class="chapter" data-level="17.3" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#importing-visium-data"><i class="fa fa-check"></i><b>17.3</b> Importing visium data</a>
<ul>
<li class="chapter" data-level="17.3.1" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#identifying-spatially-organized-features"><i class="fa fa-check"></i><b>17.3.1</b> Identifying spatially organized features</a></li>
</ul></li>
<li class="chapter" data-level="17.4" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#adding-cell-polygons-to-giotto-object"><i class="fa fa-check"></i><b>17.4</b> Adding cell polygons to Giotto object</a>
<ul>
<li class="chapter" data-level="17.4.1" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#read-in-the-poly-information"><i class="fa fa-check"></i><b>17.4.1</b> Read in the poly information</a></li>
<li class="chapter" data-level="17.4.2" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#vizualizing-polygons"><i class="fa fa-check"></i><b>17.4.2</b> Vizualizing polygons</a></li>
<li class="chapter" data-level="17.4.3" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#showing-giotto-object-prior-to-polygon-addition"><i class="fa fa-check"></i><b>17.4.3</b> Showing Giotto object prior to polygon addition</a></li>
<li class="chapter" data-level="17.4.4" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#adding-polygons-to-giotto-object"><i class="fa fa-check"></i><b>17.4.4</b> Adding polygons to giotto object</a></li>
<li class="chapter" data-level="17.4.5" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#check-polygon-information"><i class="fa fa-check"></i><b>17.4.5</b> Check polygon information</a></li>
</ul></li>
<li class="chapter" data-level="17.5" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#performing-kriging"><i class="fa fa-check"></i><b>17.5</b> Performing kriging</a>
<ul>
<li class="chapter" data-level="17.5.1" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#interpolating-features"><i class="fa fa-check"></i><b>17.5.1</b> Interpolating features</a></li>
<li class="chapter" data-level="17.5.2" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#expression-overlap"><i class="fa fa-check"></i><b>17.5.2</b> Expression overlap</a></li>
</ul></li>
<li class="chapter" data-level="17.6" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#reading-in-larger-dataset"><i class="fa fa-check"></i><b>17.6</b> Reading in larger dataset</a></li>
<li class="chapter" data-level="17.7" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#analyzing-interpolated-features"><i class="fa fa-check"></i><b>17.7</b> Analyzing interpolated features</a>
<ul>
<li class="chapter" data-level="17.7.1" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#filter-and-normalization"><i class="fa fa-check"></i><b>17.7.1</b> Filter and normalization</a></li>
<li class="chapter" data-level="17.7.2" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#visualizing-gene-expression-from-interpolated-expression"><i class="fa fa-check"></i><b>17.7.2</b> Visualizing gene expression from interpolated expression</a></li>
<li class="chapter" data-level="17.7.3" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#run-pca"><i class="fa fa-check"></i><b>17.7.3</b> Run PCA</a></li>
<li class="chapter" data-level="17.7.4" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#clustering-1"><i class="fa fa-check"></i><b>17.7.4</b> Clustering</a></li>
<li class="chapter" data-level="17.7.5" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#visualizing-clustering"><i class="fa fa-check"></i><b>17.7.5</b> Visualizing clustering</a></li>
<li class="chapter" data-level="17.7.6" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#cropping-objects"><i class="fa fa-check"></i><b>17.7.6</b> Cropping objects</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="18" data-path="contributing-to-giotto.html"><a href="contributing-to-giotto.html"><i class="fa fa-check"></i><b>18</b> Contributing to Giotto</a>
<ul>
<li class="chapter" data-level="18.1" data-path="contributing-to-giotto.html"><a href="contributing-to-giotto.html#contribution-guideline"><i class="fa fa-check"></i><b>18.1</b> Contribution guideline</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Workshop: Spatial multi-omics data analysis with Giotto Suite</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="xenium-1" class="section level1 hasAnchor" number="10">
<h1><span class="header-section-number">10</span> Xenium<a href="xenium-1.html#xenium-1" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Jiaji George Chen</p>
<p>August 6th 2024</p>
<div id="introduction-to-spatial-dataset" class="section level2 hasAnchor" number="10.1">
<h2><span class="header-section-number">10.1</span> Introduction to spatial dataset<a href="xenium-1.html#introduction-to-spatial-dataset" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="img/02_session3/Xenium-General-Workflow-1024x709.png" width="80%" style="display: block; margin: auto;" /></p>
<p>This is the 10X Xenium FFPE Human Lung Cancer dataset. Xenium captures
individual transcript detections with a spatial resolution of 100s of
nanometers, providing an extremely highly resolved subcellular spatial
dataset. This particular dataset also showcases their recent
multimodal cell segmentation outputs.</p>
<p>The Xenium Human Multi-Tissue and Cancer Panel (377) genes was used. The
exported data is from their Xenium Onboard Analysis v2.0.0 pipeline.</p>
<p>The full data for this example can be found here:
<a href="https://www.10xgenomics.com/datasets/preview-data-ffpe-human-lung-cancer-with-xenium-multimodal-cell-segmentation-1-standard">here</a></p>
<p>The relevant items are:</p>
<ul>
<li>Xenium Output Bundle (full)</li>
<li>Supplemental: Post-Xenium H&amp;E image (OME-TIFF)</li>
<li>Supplemental: H&amp;E Image Alignment File (CSV)</li>
</ul>
<p><strong>Additional package requirements</strong></p>
<p>When working with this data and trying to open the parquet files, you will need <em>arrow</em> built with ZTSD support. See the datasets &amp; packages section for specific install instructions.</p>
</div>
<div id="output-directory-structure" class="section level2 hasAnchor" number="10.2">
<h2><span class="header-section-number">10.2</span> Output directory structure<a href="xenium-1.html#output-directory-structure" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<pre><code>    âââ analysis.tar.gz
    âââ analysis.zarr.zip
    âââ analysis_summary.html
    âââ aux_outputs.tar.gz
    âââ transcripts.csv.gz
    âââ transcripts.parquet
    âââ transcripts.zarr.zip
    âââ cell_boundaries.csv.gz
    âââ cell_boundaries.parquet
    âââ nucleus_boundaries.csv.gz
    âââ nucleus_boundaries.parquet
    âââ cell_feature_matrix.tar.gz
    âââ cell_feature_matrix
    â   âââ barcodes.tsv.gz
    â   âââ features.tsv.gz
    â   âââ matrix.mtx.gz
    âââ cell_feature_matrix.h5
    âââ cell_feature_matrix.zarr.zip
    âââ cells.csv.gz
    âââ cells.parquet
    âââ cells.zarr.zip
    âââ experiment.xenium
    âââ gene_panel.json
    âââ metrics_summary.csv
    âââ morphology.ome.tif
    âââ morphology_focus
    â   âââ morphology_focus_0000.ome.tif
    â   âââ morphology_focus_0001.ome.tif
    â   âââ morphology_focus_0002.ome.tif
    â   âââ morphology_focus_0003.ome.tif
    âââ Xenium_V1_humanLung_Cancer_FFPE_he_image.ome.tif
    âââ Xenium_V1_humanLung_Cancer_FFPE_he_imagealignment.csv</code></pre>
<p>The above directory structuring and naming is characteristic of Xenium v2.0 pipeline outputs. The only items that may not be exactly the same across all outputs are the morphology focus directory and the naming of the aligned image items.</p>
<p>For the morphology focus images, you may have fewer images if the experiment did not include the multimodal cell segmentation. As for the aligned images, this is usually done after the Xenium experiment concludes and is added on using Xenium Explorer. Naming and location of the aligned image (<code>he_image.ome.tif</code>) and associated alignment info <code>he_imagealignment.csv</code> are entirely up to the user.</p>
</div>
<div id="mini-xenium-dataset" class="section level2 hasAnchor" number="10.3">
<h2><span class="header-section-number">10.3</span> Mini Xenium Dataset<a href="xenium-1.html#mini-xenium-dataset" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb120"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb120-1"><a href="xenium-1.html#cb120-1" tabindex="-1"></a><span class="fu">library</span>(Giotto)</span>
<span id="cb120-2"><a href="xenium-1.html#cb120-2" tabindex="-1"></a><span class="co"># set up paths</span></span>
<span id="cb120-3"><a href="xenium-1.html#cb120-3" tabindex="-1"></a>data_path <span class="ot">&lt;-</span> <span class="st">&quot;data/02_session3/&quot;</span></span>
<span id="cb120-4"><a href="xenium-1.html#cb120-4" tabindex="-1"></a>save_dir <span class="ot">&lt;-</span> <span class="st">&quot;results/02_session3/&quot;</span></span>
<span id="cb120-5"><a href="xenium-1.html#cb120-5" tabindex="-1"></a><span class="fu">dir.create</span>(save_dir, <span class="at">recursive =</span> <span class="cn">TRUE</span>)</span>
<span id="cb120-6"><a href="xenium-1.html#cb120-6" tabindex="-1"></a></span>
<span id="cb120-7"><a href="xenium-1.html#cb120-7" tabindex="-1"></a><span class="co"># download the mini dataset and untar</span></span>
<span id="cb120-8"><a href="xenium-1.html#cb120-8" tabindex="-1"></a><span class="fu">options</span>(<span class="st">&quot;timeout&quot;</span> <span class="ot">=</span> <span class="cn">Inf</span>)</span>
<span id="cb120-9"><a href="xenium-1.html#cb120-9" tabindex="-1"></a><span class="fu">download.file</span>(</span>
<span id="cb120-10"><a href="xenium-1.html#cb120-10" tabindex="-1"></a>    <span class="at">url =</span> <span class="st">&quot;https://zenodo.org/records/13207308/files/workshop_xenium.zip?download=1&quot;</span>,</span>
<span id="cb120-11"><a href="xenium-1.html#cb120-11" tabindex="-1"></a>    <span class="at">destfile =</span> <span class="fu">file.path</span>(save_dir, <span class="st">&quot;workshop_xenium.zip&quot;</span>)</span>
<span id="cb120-12"><a href="xenium-1.html#cb120-12" tabindex="-1"></a>)</span>
<span id="cb120-13"><a href="xenium-1.html#cb120-13" tabindex="-1"></a><span class="fu">untar</span>(<span class="at">tarfile =</span> <span class="fu">file.path</span>(save_dir, <span class="st">&quot;workshop_xenium.zip&quot;</span>), </span>
<span id="cb120-14"><a href="xenium-1.html#cb120-14" tabindex="-1"></a>      <span class="at">exdir =</span> data_path)</span></code></pre></div>
<p>In order to speed up the steps of the workshop and make it locally runnable, we provide a subset of the full dataset.</p>
<pre><code>- Full: -16.039, 12342.984, -3511.515, -294.455 (xmin, xmax, ymin, ymax)
- Mini: 6000,    7000,      -2200,     -1400    (xmin, xmax, ymin, ymax)</code></pre>
<div class="figure"><span style="display:block;" id="fig:unnamed-chunk-172"></span>
<img src="img/02_session3/mini_roi.png" alt="Shown is the H&amp;E aligned to the Xenium dataset with micron scaling. The blue bounds mark out the area provided as a mini dataset"  />
<p class="caption">
Figure 10.1: Shown is the H&amp;E aligned to the Xenium dataset with micron scaling. The blue bounds mark out the area provided as a mini dataset
</p>
</div>
</div>
<div id="data-prep" class="section level2 hasAnchor" number="10.4">
<h2><span class="header-section-number">10.4</span> Data prep<a href="xenium-1.html#data-prep" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="image-conversion-may-change" class="section level3 hasAnchor" number="10.4.1">
<h3><span class="header-section-number">10.4.1</span> Image conversion (may change)<a href="xenium-1.html#image-conversion-may-change" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>First is actually dealing with the image formats. Xenium generates <code>ome.tif</code> images which Giotto is currently not fully compatible with. So we convert them to normal <code>tif</code> images using <code>ometif_to_tif()</code> which works through the python <em>tifffile</em> package.</p>
<p>The image files can then be loaded in downstream steps.</p>
<p><em>These commented out steps are not needed for today since the mini dataset provides .tif images that have already been spatially aligned and converted. However, the code needed to do this is provided below.</em></p>
<div class="sourceCode" id="cb122"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb122-1"><a href="xenium-1.html#cb122-1" tabindex="-1"></a><span class="co"># image_paths &lt;- list.files(</span></span>
<span id="cb122-2"><a href="xenium-1.html#cb122-2" tabindex="-1"></a><span class="co">#   data_path, pattern = &quot;morphology_focus|he_image.ome&quot;, </span></span>
<span id="cb122-3"><a href="xenium-1.html#cb122-3" tabindex="-1"></a><span class="co">#   recursive = TRUE, full.names = TRUE</span></span>
<span id="cb122-4"><a href="xenium-1.html#cb122-4" tabindex="-1"></a><span class="co"># )</span></span></code></pre></div>
<p><code>ometif_to_tif()</code> <code>output_dir</code> can be specified, but by default, it writes to a new subdirectory called <code>tif_exports</code> underneath the source imageâs directory.
<em>Keep in mind that where the exported tifs get exported to should be where downstream image reading functions should point to. The code run today is with the filepaths that the mini dataset has.</em></p>
<div class="sourceCode" id="cb123"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb123-1"><a href="xenium-1.html#cb123-1" tabindex="-1"></a><span class="co"># lapply(image_paths, function(img) {</span></span>
<span id="cb123-2"><a href="xenium-1.html#cb123-2" tabindex="-1"></a><span class="co">#     GiottoClass::ometif_to_tif(img, overwrite = TRUE)</span></span>
<span id="cb123-3"><a href="xenium-1.html#cb123-3" tabindex="-1"></a><span class="co"># })</span></span></code></pre></div>
<p>We are also working on a method of directly accessing the <code>ome.tifs</code> for better compatibility in the future.</p>
</div>
</div>
<div id="convenience-function" class="section level2 hasAnchor" number="10.5">
<h2><span class="header-section-number">10.5</span> Convenience function<a href="xenium-1.html#convenience-function" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Giotto has flexible methods for working with the Xenium outputs. The <code>createGiottoXeniumObject()</code> will generate a <code>giotto</code> object in a single step when provided the output directory.</p>
<p>The default behavior is to load:</p>
<ul>
<li>transcripts information</li>
<li>cell and nucleus boundaries</li>
<li>feature metadata (gene_panel.json)</li>
</ul>
<p>For the full dataset (HPC): <em>time: /1-2min | memory: 24GBC</em></p>
<div class="sourceCode" id="cb124"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb124-1"><a href="xenium-1.html#cb124-1" tabindex="-1"></a>?createGiottoXeniumObject</span>
<span id="cb124-2"><a href="xenium-1.html#cb124-2" tabindex="-1"></a>g <span class="ot">&lt;-</span> <span class="fu">createGiottoXeniumObject</span>(<span class="at">xenium_dir =</span> data_path)</span>
<span id="cb124-3"><a href="xenium-1.html#cb124-3" tabindex="-1"></a><span class="co"># set instructions</span></span>
<span id="cb124-4"><a href="xenium-1.html#cb124-4" tabindex="-1"></a><span class="fu">instructions</span>(g, <span class="st">&quot;save_dir&quot;</span>) <span class="ot">&lt;-</span> save_dir</span>
<span id="cb124-5"><a href="xenium-1.html#cb124-5" tabindex="-1"></a><span class="fu">instructions</span>(g, <span class="st">&quot;save_plot&quot;</span>) <span class="ot">&lt;-</span> <span class="cn">TRUE</span></span></code></pre></div>
<p>There are a lot of other params for additional or alternative items you can load. The next subsections will explain a couple of them.</p>
<div id="specific-filepaths" class="section level3 hasAnchor" number="10.5.1">
<h3><span class="header-section-number">10.5.1</span> Specific filepaths<a href="xenium-1.html#specific-filepaths" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<pre><code>    expression_path = ,
    cell_metadata_path = ,
    transcript_path = ,
    bounds_path = ,
    gene_panel_json_path = , </code></pre>
<p>The convenience function auto-detects filepaths based on the Xenium directory path and the preferred file formats</p>
<ul>
<li><code>.parquet</code> for tabular (vs <code>.csv</code>)</li>
<li><code>.h5</code> for matrix over other formats when available (vs <code>.mtx</code>)</li>
<li><code>.zarr</code> is currently not supported.</li>
</ul>
<p>When you need to use a different file format or something is not in the expected output structure, you can supply a specific filepath to the convenience function using these params.</p>
</div>
<div id="quality-value" class="section level3 hasAnchor" number="10.5.2">
<h3><span class="header-section-number">10.5.2</span> Quality value<a href="xenium-1.html#quality-value" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<pre><code>    qv_threshold = 20 # default</code></pre>
<p>The <em>Quality Value</em> is a Phred-based 0-40 value that 10X provides for every detection in their transcripts output. Higher values mean higher confidence in the decoded transcript identity. By default 10X uses a cutoff of QV = 20 for transcripts to use downstream.</p>
<p>_*setting a value other than 20 will make the loaded dataset different from the 10X-provided expression matrix and cell metadata._</p>
<details>
<summary>
QV Calculation
</summary>
<ol style="list-style-type: decimal">
<li>Raw Q-score based on how likely it is that an observed code is to be the codeword that it gets mapped to vs less likely codeword.</li>
<li>Adjustment of raw Q-score by binning the transcripts by Q-value then adjusting the exact Q per bin based on proportion of <strong>Negative Control Codewords</strong> detected within.</li>
</ol>
<p><a href="https://www.10xgenomics.com/support/software/xenium-onboard-analysis/latest/algorithms-overview/xoa-algorithms#qvs">further info</a></p>
</details>
</div>
<div id="transcript-type-splitting" class="section level3 hasAnchor" number="10.5.3">
<h3><span class="header-section-number">10.5.3</span> Transcript type splitting<a href="xenium-1.html#transcript-type-splitting" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<pre><code>    feat_type =  c(
        &quot;rna&quot;,
        &quot;NegControlProbe&quot;,
        &quot;UnassignedCodeword&quot;,
        &quot;NegControlCodeword&quot;
    ),
    split_keyword = list(
        c(&quot;NegControlProbe&quot;),
        c(&quot;UnassignedCodeword&quot;),
        c(&quot;NegControlCodeword)&quot;
    )</code></pre>
<p>There are 4 types of transcript detections that 10X reports with their v2.0 pipeline:</p>
<ul>
<li><strong>Gene expression</strong> - This is the rna gene detections</li>
<li><strong>Negative Control Codeword</strong> - (QC) Codewords that do not map to genes, but are in the codebook. Used to determine specificity of decoding algorithm.</li>
<li><strong>Negative Control Probe</strong> - (QC) Probes in panel but target non-biological sequences. Used to determine specificity of assay.</li>
<li><strong>Unassigned Codeword</strong> - (QC) Codewords that should not be used in the current panel</li>
</ul>
<p>With V3 on their Xenium prime outputs, there is additionally:</p>
<ul>
<li><strong>Genomic Control Codeword</strong> (QC) Probes for intergenic genomic DNA instead of transcripts</li>
</ul>
<hr>
<p>The main thing to watch out for is that the other probe types should be separated out from the the <strong>Gene expression</strong> or <strong>rna</strong> feature type.</p>
<p>How to deal with these different types of detections is easily adjustable. With the <code>feat_type</code> param you declare which categories/<code>feat_types</code> you want to split transcript detections into. Then with <code>split_keyword</code>, you provide a list of character vectors containing <code>grep()</code> terms to search for.</p>
<p>Note that there are 4 <code>feat_types</code> declared in this set of defaults, but 3 items passed to <code>split_keyword</code>. Any transcripts not matched by items in <code>split_keyword</code>, get categorized as the first provided <code>feat_type</code> (ârnaâ).</p>
</div>
<div id="centroids-calculation" class="section level3 hasAnchor" number="10.5.4">
<h3><span class="header-section-number">10.5.4</span> Centroids calculation<a href="xenium-1.html#centroids-calculation" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Several Giotto operations require that a set of centroids are calculated for polygon spatial units.</p>
<div class="sourceCode" id="cb128"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb128-1"><a href="xenium-1.html#cb128-1" tabindex="-1"></a>g <span class="ot">&lt;-</span> <span class="fu">addSpatialCentroidLocations</span>(g, <span class="at">poly_info =</span> <span class="st">&quot;cell&quot;</span>)</span>
<span id="cb128-2"><a href="xenium-1.html#cb128-2" tabindex="-1"></a>g <span class="ot">&lt;-</span> <span class="fu">addSpatialCentroidLocations</span>(g, <span class="at">poly_info =</span> <span class="st">&quot;nucleus&quot;</span>)</span></code></pre></div>
</div>
<div id="simple-visualization" class="section level3 hasAnchor" number="10.5.5">
<h3><span class="header-section-number">10.5.5</span> Simple visualization<a href="xenium-1.html#simple-visualization" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb129"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb129-1"><a href="xenium-1.html#cb129-1" tabindex="-1"></a><span class="fu">spatInSituPlotPoints</span>(g,</span>
<span id="cb129-2"><a href="xenium-1.html#cb129-2" tabindex="-1"></a>    <span class="at">polygon_feat_type =</span> <span class="st">&quot;cell&quot;</span>,</span>
<span id="cb129-3"><a href="xenium-1.html#cb129-3" tabindex="-1"></a>    <span class="at">feats =</span> <span class="fu">list</span>(<span class="at">rna =</span> <span class="fu">head</span>(<span class="fu">featIDs</span>(g))), </span>
<span id="cb129-4"><a href="xenium-1.html#cb129-4" tabindex="-1"></a>    <span class="at">use_overlap =</span> <span class="cn">FALSE</span>, </span>
<span id="cb129-5"><a href="xenium-1.html#cb129-5" tabindex="-1"></a>    <span class="at">polygon_color =</span> <span class="st">&quot;cyan&quot;</span>, </span>
<span id="cb129-6"><a href="xenium-1.html#cb129-6" tabindex="-1"></a>    <span class="at">polygon_line_size =</span> <span class="fl">0.1</span></span>
<span id="cb129-7"><a href="xenium-1.html#cb129-7" tabindex="-1"></a>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-178"></span>
<img src="img/02_session3/situ_plot_head.png" alt="Simple subcellular plotting to check data" width="80%" />
<p class="caption">
Figure 10.2: Simple subcellular plotting to check data
</p>
</div>
</div>
</div>
<div id="piecewise-loading" class="section level2 hasAnchor" number="10.6">
<h2><span class="header-section-number">10.6</span> Piecewise loading<a href="xenium-1.html#piecewise-loading" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Giotto also provides the <code>importXenium()</code> import utility that allows independent creation of compatible Giotto subobjects for more flexibility.</p>
<div class="sourceCode" id="cb130"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb130-1"><a href="xenium-1.html#cb130-1" tabindex="-1"></a>x <span class="ot">&lt;-</span> <span class="fu">importXenium</span>(data_path)</span>
<span id="cb130-2"><a href="xenium-1.html#cb130-2" tabindex="-1"></a><span class="fu">force</span>(x)</span></code></pre></div>
<pre><code>Giotto &lt;XeniumReader&gt;
dir       : data/02_session3/ 
qv_cutoff : 20 
filetype  : transcripts -- parquet
            boundaries  -- parquet
            expression  -- h5
            cell_meta   -- parquet
funs      : load_transcripts()
            load_polys()
            load_cellmeta()
            load_featmeta()
            load_expression()
            load_image()
            load_aligned_image()
            create_gobject()</code></pre>
<div id="load-giottopoints-transcripts" class="section level3 hasAnchor" number="10.6.1">
<h3><span class="header-section-number">10.6.1</span> load giottoPoints transcripts<a href="xenium-1.html#load-giottopoints-transcripts" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb132"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb132-1"><a href="xenium-1.html#cb132-1" tabindex="-1"></a>x<span class="sc">$</span>qv <span class="ot">&lt;-</span> <span class="dv">20</span> <span class="co"># default</span></span>
<span id="cb132-2"><a href="xenium-1.html#cb132-2" tabindex="-1"></a>tx <span class="ot">&lt;-</span> x<span class="sc">$</span><span class="fu">load_transcripts</span>()</span>
<span id="cb132-3"><a href="xenium-1.html#cb132-3" tabindex="-1"></a><span class="fu">plot</span>(tx[[<span class="dv">1</span>]]<span class="sc">$</span>rna, <span class="at">dens =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-181"></span>
<img src="img/02_session3/tx_plot.png" alt="plot of Gene expression ('rna') density" width="80%" />
<p class="caption">
Figure 10.3: plot of Gene expression (ârnaâ) density
</p>
</div>
<div class="sourceCode" id="cb133"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb133-1"><a href="xenium-1.html#cb133-1" tabindex="-1"></a><span class="fu">force</span>(tx[[<span class="dv">1</span>]]<span class="sc">$</span>rna)</span>
<span id="cb133-2"><a href="xenium-1.html#cb133-2" tabindex="-1"></a><span class="fu">rm</span>(tx) <span class="co"># save space</span></span></code></pre></div>
<pre><code>An object of class giottoPoints
feat_type : &quot;rna&quot;
Feature Information:
 class       : SpatVector 
 geometry    : points 
 dimensions  : 479097, 10  (geometries, attributes)
 extent      : 6000.001, 7000, -2200, -1400.012  (xmin, xmax, ymin, ymax)
 coord. ref. :  
 names       : feat_ID   transcript_id    cell_id overlaps_nucleus z_location    qv fov_name
 type        :   &lt;chr&gt;           &lt;chr&gt;      &lt;chr&gt;            &lt;int&gt;      &lt;num&gt; &lt;num&gt;    &lt;chr&gt;
 values      :   FBLN1 281487861612869 mcnjadoe-1                0      19.32    40      B11
                PDGFRB 281487861612872 mcnjbidl-1                1      18.75    40      B11
                PDGFRB 281487861612873 mcnjbidl-1                1      18.74    40      B11
 nucleus_distance codeword_index feat_ID_uniq
            &lt;num&gt;          &lt;int&gt;        &lt;int&gt;
                0            334            1
                0            289            2
                0            289            3</code></pre>
</div>
<div id="optional-loading-pre-aggregated-data" class="section level3 hasAnchor" number="10.6.2">
<h3><span class="header-section-number">10.6.2</span> (optional) Loading pre-aggregated data<a href="xenium-1.html#optional-loading-pre-aggregated-data" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Giotto can spatially aggregate the transcripts information based on a provided set of boundaries information, however 10X also provides a pre-aggregated set of cell by feature information and metadata. These values may be slightly different from those calculated by Giottoâs pipeline, and are not loaded by default.</p>
<p>Some care needs to be taken when loading this information:</p>
<ul>
<li>The <code>feat_type</code> of the loaded expression information should be matched to the used <code>feat_type</code> params passed to the convenience function.</li>
<li>The <code>qv_threshold</code> used must be 20 since the 10X outputs are based on that cutoff.</li>
</ul>
<div class="sourceCode" id="cb135"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb135-1"><a href="xenium-1.html#cb135-1" tabindex="-1"></a>x<span class="sc">$</span>filetype<span class="sc">$</span>expression <span class="ot">&lt;-</span> <span class="st">&quot;mtx&quot;</span> <span class="co"># change to mtx instead of .h5 which is not in the mini dataset</span></span>
<span id="cb135-2"><a href="xenium-1.html#cb135-2" tabindex="-1"></a>ex <span class="ot">&lt;-</span> x<span class="sc">$</span><span class="fu">load_expression</span>()</span>
<span id="cb135-3"><a href="xenium-1.html#cb135-3" tabindex="-1"></a><span class="fu">featType</span>(ex)</span></code></pre></div>
<pre><code>[1] &quot;rna&quot;                       &quot;Negative Control Probe&quot;    &quot;Negative Control Codeword&quot;
[4] &quot;Unassigned Codeword&quot;      </code></pre>
<p>The feature types here do not match what we established for the transcripts, so we can just change them.</p>
<div class="sourceCode" id="cb137"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb137-1"><a href="xenium-1.html#cb137-1" tabindex="-1"></a><span class="fu">force</span>(g)</span></code></pre></div>
<pre><code>An object of class giotto 
&gt;Active spat_unit:  cell 
&gt;Active feat_type:  rna 
[SUBCELLULAR INFO]
polygons      : cell nucleus 
features      : rna NegControlProbe UnassignedCodeword NegControlCodeword 
[AGGREGATE INFO]
spatial locations ----------------
  [cell] raw
  [nucleus] raw</code></pre>
<div class="sourceCode" id="cb139"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb139-1"><a href="xenium-1.html#cb139-1" tabindex="-1"></a><span class="fu">featType</span>(ex[[<span class="dv">2</span>]]) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;NegControlProbe&quot;</span>)</span>
<span id="cb139-2"><a href="xenium-1.html#cb139-2" tabindex="-1"></a><span class="fu">featType</span>(ex[[<span class="dv">3</span>]]) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;NegControlCodeword&quot;</span>)</span>
<span id="cb139-3"><a href="xenium-1.html#cb139-3" tabindex="-1"></a><span class="fu">featType</span>(ex[[<span class="dv">4</span>]]) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;UnassignedCodeword&quot;</span>)</span></code></pre></div>
<p>Then we can just append them to the Giotto object.<br />
Here we set up a second object since we will be using Giottoâs own aggregation method downstream.</p>
<div class="sourceCode" id="cb140"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb140-1"><a href="xenium-1.html#cb140-1" tabindex="-1"></a>g2 <span class="ot">&lt;-</span> g</span>
<span id="cb140-2"><a href="xenium-1.html#cb140-2" tabindex="-1"></a><span class="co"># append the expression info</span></span>
<span id="cb140-3"><a href="xenium-1.html#cb140-3" tabindex="-1"></a>g2 <span class="ot">&lt;-</span> <span class="fu">setGiotto</span>(g2, ex)</span>
<span id="cb140-4"><a href="xenium-1.html#cb140-4" tabindex="-1"></a></span>
<span id="cb140-5"><a href="xenium-1.html#cb140-5" tabindex="-1"></a><span class="co"># load cell metadata</span></span>
<span id="cb140-6"><a href="xenium-1.html#cb140-6" tabindex="-1"></a>cx <span class="ot">&lt;-</span> x<span class="sc">$</span><span class="fu">load_cellmeta</span>()</span>
<span id="cb140-7"><a href="xenium-1.html#cb140-7" tabindex="-1"></a>g2 <span class="ot">&lt;-</span> <span class="fu">setGiotto</span>(g2, cx)</span>
<span id="cb140-8"><a href="xenium-1.html#cb140-8" tabindex="-1"></a><span class="fu">force</span>(g2)</span></code></pre></div>
<pre><code>An object of class giotto 
&gt;Active spat_unit:  cell 
&gt;Active feat_type:  rna 
[SUBCELLULAR INFO]
polygons      : cell nucleus 
features      : rna NegControlProbe UnassignedCodeword NegControlCodeword 
[AGGREGATE INFO]
expression -----------------------
  [cell][rna] raw
  [cell][NegControlProbe] raw
  [cell][NegControlCodeword] raw
  [cell][UnassignedCodeword] raw
spatial locations ----------------
  [cell] raw
  [nucleus] raw</code></pre>
<div class="sourceCode" id="cb142"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb142-1"><a href="xenium-1.html#cb142-1" tabindex="-1"></a><span class="fu">spatInSituPlotPoints</span>(g2,</span>
<span id="cb142-2"><a href="xenium-1.html#cb142-2" tabindex="-1"></a>        <span class="co"># polygon shading params</span></span>
<span id="cb142-3"><a href="xenium-1.html#cb142-3" tabindex="-1"></a>    <span class="at">polygon_fill =</span> <span class="st">&quot;cell_area&quot;</span>,</span>
<span id="cb142-4"><a href="xenium-1.html#cb142-4" tabindex="-1"></a>    <span class="at">polygon_fill_as_factor =</span> <span class="cn">FALSE</span>,</span>
<span id="cb142-5"><a href="xenium-1.html#cb142-5" tabindex="-1"></a>    <span class="at">polygon_fill_gradient_style =</span> <span class="st">&quot;sequential&quot;</span>,</span>
<span id="cb142-6"><a href="xenium-1.html#cb142-6" tabindex="-1"></a>    <span class="co"># polygon line params</span></span>
<span id="cb142-7"><a href="xenium-1.html#cb142-7" tabindex="-1"></a>    <span class="at">polygon_color =</span> <span class="st">&quot;grey&quot;</span>,</span>
<span id="cb142-8"><a href="xenium-1.html#cb142-8" tabindex="-1"></a>    <span class="at">polygon_line_size =</span> <span class="fl">0.1</span></span>
<span id="cb142-9"><a href="xenium-1.html#cb142-9" tabindex="-1"></a>)</span>
<span id="cb142-10"><a href="xenium-1.html#cb142-10" tabindex="-1"></a></span>
<span id="cb142-11"><a href="xenium-1.html#cb142-11" tabindex="-1"></a><span class="fu">spatInSituPlotPoints</span>(g2,</span>
<span id="cb142-12"><a href="xenium-1.html#cb142-12" tabindex="-1"></a>    <span class="co"># polygon shading params</span></span>
<span id="cb142-13"><a href="xenium-1.html#cb142-13" tabindex="-1"></a>    <span class="at">polygon_fill =</span> <span class="st">&quot;transcript_counts&quot;</span>,</span>
<span id="cb142-14"><a href="xenium-1.html#cb142-14" tabindex="-1"></a>    <span class="at">polygon_fill_as_factor =</span> <span class="cn">FALSE</span>,</span>
<span id="cb142-15"><a href="xenium-1.html#cb142-15" tabindex="-1"></a>    <span class="at">polygon_fill_gradient_style =</span> <span class="st">&quot;sequential&quot;</span>,</span>
<span id="cb142-16"><a href="xenium-1.html#cb142-16" tabindex="-1"></a>    <span class="co"># polygon line params</span></span>
<span id="cb142-17"><a href="xenium-1.html#cb142-17" tabindex="-1"></a>    <span class="at">polygon_color =</span> <span class="st">&quot;grey&quot;</span>,</span>
<span id="cb142-18"><a href="xenium-1.html#cb142-18" tabindex="-1"></a>    <span class="at">polygon_line_size =</span> <span class="fl">0.1</span></span>
<span id="cb142-19"><a href="xenium-1.html#cb142-19" tabindex="-1"></a>)</span></code></pre></div>
<div class="figure"><span style="display:block;" id="fig:unnamed-chunk-188"></span>
<img src="img/02_session3/cmeta_plots.png" alt="Example plot using 10X metadata. Left is 'cell_area', right is 'transcript_counts'"  />
<p class="caption">
Figure 10.4: Example plot using 10X metadata. Left is âcell_areaâ, right is âtranscript_countsâ
</p>
</div>
<div class="sourceCode" id="cb143"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb143-1"><a href="xenium-1.html#cb143-1" tabindex="-1"></a><span class="fu">rm</span>(g2) <span class="co"># save space</span></span></code></pre></div>
</div>
</div>
<div id="xenium-images" class="section level2 hasAnchor" number="10.7">
<h2><span class="header-section-number">10.7</span> Xenium Images<a href="xenium-1.html#xenium-images" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Xenium outputs have several image outputs. For this dataset:</p>
<ul>
<li><p><code>morphology.ome.tif</code> is a z-stacked image of the DAPI staining, with z levels separated as pages within the <code>ome.tif</code>. In this dataset, only pages 6 and 7 are really in focus.</p></li>
<li><p><code>morphology_focus</code> is a folder containing single-channel image(s), but with the original z information collapsed into a single in-focus layer. For all datasets, image 0000 will be DAPI staining, but if you have additional stains, such as the multimodal segmentation, they will also be here. These are the recommended immunofluorescence staining images to import.</p></li>
<li><p><code>Xenium_V1_humanLung_Cancer_FFPE_he_image.ome.tif</code> is an added on (in this case H&amp;E) image with manual affine registration.</p>
<pre><code>  # 3D stack - DAPI

          # 2D fusion images
  # - DAPI
  # - stainings for multimodal segmentation

  # Maximum intensity projection (MIP) of the morphology image.
  # (Xenium Outputs v1.0 - 1.9. only)</code></pre></li>
<li><p>image mip</p></li>
</ul>
<div id="image-metadata" class="section level3 hasAnchor" number="10.7.1">
<h3><span class="header-section-number">10.7.1</span> Image metadata<a href="xenium-1.html#image-metadata" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb145"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb145-1"><a href="xenium-1.html#cb145-1" tabindex="-1"></a>img_xml_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_path, <span class="st">&quot;morphology_focus&quot;</span>, <span class="st">&quot;morphology_focus_0000.xml&quot;</span>)</span>
<span id="cb145-2"><a href="xenium-1.html#cb145-2" tabindex="-1"></a>omemeta <span class="ot">&lt;-</span> xml2<span class="sc">::</span><span class="fu">read_xml</span>(img_xml_path)</span>
<span id="cb145-3"><a href="xenium-1.html#cb145-3" tabindex="-1"></a>res <span class="ot">&lt;-</span> xml2<span class="sc">::</span><span class="fu">xml_find_all</span>(omemeta, <span class="st">&quot;//d1:Channel&quot;</span>, <span class="at">ns =</span> xml2<span class="sc">::</span><span class="fu">xml_ns</span>(omemeta))</span>
<span id="cb145-4"><a href="xenium-1.html#cb145-4" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">Reduce</span>(rbind, xml2<span class="sc">::</span><span class="fu">xml_attrs</span>(res))</span>
<span id="cb145-5"><a href="xenium-1.html#cb145-5" tabindex="-1"></a><span class="fu">rownames</span>(res) <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb145-6"><a href="xenium-1.html#cb145-6" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(res)</span>
<span id="cb145-7"><a href="xenium-1.html#cb145-7" tabindex="-1"></a><span class="fu">force</span>(res)</span></code></pre></div>
<pre><code>         ID                   Name SamplesPerPixel
1 Channel:0                   DAPI               1
2 Channel:1                    18S               1
3 Channel:2 ATP1A1/CD45/E-Cadherin               1
4 Channel:3      alphaSMA/Vimentin               1</code></pre>
</div>
<div id="image-loading" class="section level3 hasAnchor" number="10.7.2">
<h3><span class="header-section-number">10.7.2</span> Image loading<a href="xenium-1.html#image-loading" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<pre><code>    load_images = list(
        img1 = &quot;[img_path1.tif]&quot;,
        img2 = &quot;[img_path2.tif]&quot;,
        img3 = &quot;...&quot;
    ),
    load_aligned_images = list(
        aligned_img = c(
            &quot;[path to image.tif]&quot;,
            &quot;imagealignment.csv&quot;
        )
    ),

)</code></pre>
<p>morphology_focus images only require a scaling by the pixel to micron scaling factor. This can be found from the <code>experiment.xenium</code> to be 0.2125 for this dataset and is auto-detected by the convenience functions.</p>
<p>aligned images are usually additional images added on with manual affine registration.</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-191"></span>
<img src="img/02_session3/img_orig_ext.png" alt="Spatial extent/bounds of transcripts (red), immunofluorescence morphology focus images (blue), H&amp;E aligned image (gold). Lower right shows the affine matrix for aligning the H&amp;E" width="80%" />
<p class="caption">
Figure 10.5: Spatial extent/bounds of transcripts (red), immunofluorescence morphology focus images (blue), H&amp;E aligned image (gold). Lower right shows the affine matrix for aligning the H&amp;E
</p>
</div>
</div>
</div>
<div id="read-in-raw-data-1" class="section level2 hasAnchor" number="10.8">
<h2><span class="header-section-number">10.8</span> Read in raw data<a href="xenium-1.html#read-in-raw-data-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ul>
<li>Transcript coordinates</li>
<li>Polygon coordinates</li>
<li>Visualizations</li>
</ul>
</div>
<div id="spatial-aggregation" class="section level2 hasAnchor" number="10.9">
<h2><span class="header-section-number">10.9</span> Spatial aggregation<a href="xenium-1.html#spatial-aggregation" class="anchor-section" aria-label="Anchor link to header"></a></h2>
</div>
<div id="aggregate-analyses-workflow" class="section level2 hasAnchor" number="10.10">
<h2><span class="header-section-number">10.10</span> Aggregate analyses workflow<a href="xenium-1.html#aggregate-analyses-workflow" class="anchor-section" aria-label="Anchor link to header"></a></h2>
</div>
<div id="niche-clustering" class="section level2 hasAnchor" number="10.11">
<h2><span class="header-section-number">10.11</span> Niche clustering<a href="xenium-1.html#niche-clustering" class="anchor-section" aria-label="Anchor link to header"></a></h2>
</div>
<div id="pseudovisium" class="section level2 hasAnchor" number="10.12">
<h2><span class="header-section-number">10.12</span> Pseudovisium<a href="xenium-1.html#pseudovisium" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ul>
<li>Feature/molecule specific analysis</li>
</ul>
</div>
<div id="transcript-enrichment-gsea" class="section level2 hasAnchor" number="10.13">
<h2><span class="header-section-number">10.13</span> Transcript enrichment GSEA<a href="xenium-1.html#transcript-enrichment-gsea" class="anchor-section" aria-label="Anchor link to header"></a></h2>
</div>
<div id="spatial-location-analysis" class="section level2 hasAnchor" number="10.14">
<h2><span class="header-section-number">10.14</span> Spatial location analysis<a href="xenium-1.html#spatial-location-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h2>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="visium-hd.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="spatial-proteomics-multiplexed-immunofluorescence.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
