<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>7 Visium Part II | Workshop: Spatial multi-omics data analysis with Giotto Suite</title>
  <meta name="description" content="7 Visium Part II | Workshop: Spatial multi-omics data analysis with Giotto Suite" />
  <meta name="generator" content="bookdown 0.39 and GitBook 2.6.7" />

  <meta property="og:title" content="7 Visium Part II | Workshop: Spatial multi-omics data analysis with Giotto Suite" />
  <meta property="og:type" content="book" />
  
  
  <meta name="github-repo" content="drieslab/giotto_workshop_2024" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="7 Visium Part II | Workshop: Spatial multi-omics data analysis with Giotto Suite" />
  
  
  

<meta name="author" content="Ruben Dries, Jiaji George Chen, Joselyn Cristina Chávez-Fuentes, Junxiang Xu , Edward Ruiz, Jeff Sheridan, Wen Wang" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="visium-part-i.html"/>
<link rel="next" href="visium-hd.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Giotto Workshop 2024</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#instructors"><i class="fa fa-check"></i><b>1.1</b> Instructors</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#topics-and-schedule"><i class="fa fa-check"></i><b>1.2</b> Topics and Schedule:</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#license"><i class="fa fa-check"></i><b>1.3</b> License</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html"><i class="fa fa-check"></i><b>2</b> Spatial omics technologies</a>
<ul>
<li class="chapter" data-level="2.1" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#slides"><i class="fa fa-check"></i><b>2.1</b> Slides</a></li>
<li class="chapter" data-level="2.2" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#spatial-sequencing"><i class="fa fa-check"></i><b>2.2</b> Spatial sequencing</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html"><i class="fa fa-check"></i><b>3</b> Introduction to the Giotto package</a>
<ul>
<li class="chapter" data-level="3.1" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html#slides-1"><i class="fa fa-check"></i><b>3.1</b> Slides</a></li>
<li class="chapter" data-level="3.2" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html#ecosystem"><i class="fa fa-check"></i><b>3.2</b> Ecosystem</a></li>
<li class="chapter" data-level="3.3" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html#installation-python-environment"><i class="fa fa-check"></i><b>3.3</b> Installation + python environment</a></li>
<li class="chapter" data-level="3.4" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html#giotto-instructions"><i class="fa fa-check"></i><b>3.4</b> Giotto instructions</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html"><i class="fa fa-check"></i><b>4</b> Data formatting and Pre-processing</a>
<ul>
<li class="chapter" data-level="4.1" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#data-formats"><i class="fa fa-check"></i><b>4.1</b> Data formats</a></li>
<li class="chapter" data-level="4.2" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#pre-processing"><i class="fa fa-check"></i><b>4.2</b> Pre-processing</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html"><i class="fa fa-check"></i><b>5</b> Creating a Giotto object</a>
<ul>
<li class="chapter" data-level="5.1" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#from-matrix-locations"><i class="fa fa-check"></i><b>5.1</b> From matrix + locations</a></li>
<li class="chapter" data-level="5.2" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#from-subcellular-raw-data-transcripts-or-images-polygons"><i class="fa fa-check"></i><b>5.2</b> From subcellular raw data (transcripts or images) + polygons</a></li>
<li class="chapter" data-level="5.3" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#using-convenience-functions-for-popular-technologies-vizgen-xenium-cosmx"><i class="fa fa-check"></i><b>5.3</b> Using convenience functions for popular technologies (Vizgen, Xenium, CosMx, …)</a></li>
<li class="chapter" data-level="5.4" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#spatial-plots"><i class="fa fa-check"></i><b>5.4</b> Spatial plots</a></li>
<li class="chapter" data-level="5.5" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#subsetting"><i class="fa fa-check"></i><b>5.5</b> Subsetting</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="visium-part-i.html"><a href="visium-part-i.html"><i class="fa fa-check"></i><b>6</b> Visium Part I</a>
<ul>
<li class="chapter" data-level="6.1" data-path="visium-part-i.html"><a href="visium-part-i.html#introduction-to-spatial-dataset"><i class="fa fa-check"></i><b>6.1</b> Introduction to spatial dataset</a></li>
<li class="chapter" data-level="6.2" data-path="visium-part-i.html"><a href="visium-part-i.html#create-the-giotto-object"><i class="fa fa-check"></i><b>6.2</b> Create the Giotto object</a></li>
<li class="chapter" data-level="6.3" data-path="visium-part-i.html"><a href="visium-part-i.html#subset-on-spots-that-were-covered-by-tissue"><i class="fa fa-check"></i><b>6.3</b> Subset on spots that were covered by tissue</a></li>
<li class="chapter" data-level="6.4" data-path="visium-part-i.html"><a href="visium-part-i.html#quality-control"><i class="fa fa-check"></i><b>6.4</b> Quality control</a></li>
<li class="chapter" data-level="6.5" data-path="visium-part-i.html"><a href="visium-part-i.html#filtering"><i class="fa fa-check"></i><b>6.5</b> Filtering</a></li>
<li class="chapter" data-level="6.6" data-path="visium-part-i.html"><a href="visium-part-i.html#normalization"><i class="fa fa-check"></i><b>6.6</b> Normalization</a></li>
<li class="chapter" data-level="6.7" data-path="visium-part-i.html"><a href="visium-part-i.html#feature-selection"><i class="fa fa-check"></i><b>6.7</b> Feature selection</a>
<ul>
<li class="chapter" data-level="6.7.1" data-path="visium-part-i.html"><a href="visium-part-i.html#highly-variable-features"><i class="fa fa-check"></i><b>6.7.1</b> Highly Variable Features:</a></li>
<li class="chapter" data-level="6.7.2" data-path="visium-part-i.html"><a href="visium-part-i.html#spatial-variable-genes"><i class="fa fa-check"></i><b>6.7.2</b> Spatial variable genes</a></li>
</ul></li>
<li class="chapter" data-level="6.8" data-path="visium-part-i.html"><a href="visium-part-i.html#dimension-reduction"><i class="fa fa-check"></i><b>6.8</b> Dimension Reduction</a></li>
<li class="chapter" data-level="6.9" data-path="visium-part-i.html"><a href="visium-part-i.html#clustering"><i class="fa fa-check"></i><b>6.9</b> Clustering</a>
<ul>
<li class="chapter" data-level="6.9.1" data-path="visium-part-i.html"><a href="visium-part-i.html#non-spatial"><i class="fa fa-check"></i><b>6.9.1</b> Non-spatial</a></li>
<li class="chapter" data-level="6.9.2" data-path="visium-part-i.html"><a href="visium-part-i.html#spatial"><i class="fa fa-check"></i><b>6.9.2</b> Spatial</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="visium-part-ii.html"><a href="visium-part-ii.html"><i class="fa fa-check"></i><b>7</b> Visium Part II</a>
<ul>
<li class="chapter" data-level="7.1" data-path="visium-part-ii.html"><a href="visium-part-ii.html#differential-expression"><i class="fa fa-check"></i><b>7.1</b> Differential expression</a></li>
<li class="chapter" data-level="7.2" data-path="visium-part-ii.html"><a href="visium-part-ii.html#enrichment-deconvolution"><i class="fa fa-check"></i><b>7.2</b> Enrichment &amp; Deconvolution</a></li>
<li class="chapter" data-level="7.3" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatial-expression-patterns"><i class="fa fa-check"></i><b>7.3</b> Spatial expression patterns</a></li>
<li class="chapter" data-level="7.4" data-path="visium-part-ii.html"><a href="visium-part-ii.html#interactive-tools"><i class="fa fa-check"></i><b>7.4</b> Interactive tools</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="visium-hd.html"><a href="visium-hd.html"><i class="fa fa-check"></i><b>8</b> Visium HD</a>
<ul>
<li class="chapter" data-level="8.1" data-path="visium-hd.html"><a href="visium-hd.html#tiling-and-aggregation"><i class="fa fa-check"></i><b>8.1</b> Tiling and aggregation</a></li>
<li class="chapter" data-level="8.2" data-path="visium-hd.html"><a href="visium-hd.html#scalability-and-projection-functions"><i class="fa fa-check"></i><b>8.2</b> Scalability and projection functions</a></li>
<li class="chapter" data-level="8.3" data-path="visium-hd.html"><a href="visium-hd.html#spatial-expression-patterns-1"><i class="fa fa-check"></i><b>8.3</b> Spatial expression patterns</a></li>
<li class="chapter" data-level="8.4" data-path="visium-hd.html"><a href="visium-hd.html#spatial-co-expression-modules"><i class="fa fa-check"></i><b>8.4</b> Spatial co-expression modules</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="xenium.html"><a href="xenium.html"><i class="fa fa-check"></i><b>9</b> Xenium</a>
<ul>
<li class="chapter" data-level="9.1" data-path="xenium.html"><a href="xenium.html#read-in-raw-data"><i class="fa fa-check"></i><b>9.1</b> Read in raw data</a></li>
<li class="chapter" data-level="9.2" data-path="xenium.html"><a href="xenium.html#overlap-txs-polygons"><i class="fa fa-check"></i><b>9.2</b> Overlap txs &amp; polygons</a></li>
<li class="chapter" data-level="9.3" data-path="xenium.html"><a href="xenium.html#transcript-enrichment-gsea"><i class="fa fa-check"></i><b>9.3</b> Transcript enrichment GSEA</a></li>
<li class="chapter" data-level="9.4" data-path="xenium.html"><a href="xenium.html#spatial-location-analysis"><i class="fa fa-check"></i><b>9.4</b> Spatial location analysis</a></li>
<li class="chapter" data-level="9.5" data-path="xenium.html"><a href="xenium.html#visualizations"><i class="fa fa-check"></i><b>9.5</b> Visualizations</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="spatial-proteomics-multiplex-if.html"><a href="spatial-proteomics-multiplex-if.html"><i class="fa fa-check"></i><b>10</b> Spatial proteomics: multiplex IF</a>
<ul>
<li class="chapter" data-level="10.1" data-path="spatial-proteomics-multiplex-if.html"><a href="spatial-proteomics-multiplex-if.html#read-in-raw-data-1"><i class="fa fa-check"></i><b>10.1</b> Read in raw data</a></li>
<li class="chapter" data-level="10.2" data-path="spatial-proteomics-multiplex-if.html"><a href="spatial-proteomics-multiplex-if.html#overlap-intensity-workflows"><i class="fa fa-check"></i><b>10.2</b> Overlap intensity &amp; workflows</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html"><i class="fa fa-check"></i><b>11</b> Working with multiple samples</a>
<ul>
<li class="chapter" data-level="11.1" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#create-individual-giotto-objects"><i class="fa fa-check"></i><b>11.1</b> Create individual giotto objects</a></li>
<li class="chapter" data-level="11.2" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#join-giotto-objects"><i class="fa fa-check"></i><b>11.2</b> Join Giotto Objects</a></li>
<li class="chapter" data-level="11.3" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#perform-harmony-and-default-workflows"><i class="fa fa-check"></i><b>11.3</b> Perform Harmony and default workflows</a></li>
<li class="chapter" data-level="11.4" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#visualizations-1"><i class="fa fa-check"></i><b>11.4</b> Visualizations</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html"><i class="fa fa-check"></i><b>12</b> Spatial multi-modal analysis</a>
<ul>
<li class="chapter" data-level="12.1" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#co-registration-of-datasets"><i class="fa fa-check"></i><b>12.1</b> Co-registration of datasets</a></li>
<li class="chapter" data-level="12.2" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#examples-in-giotto-suite-manuscript"><i class="fa fa-check"></i><b>12.2</b> Examples in giotto suite manuscript</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html"><i class="fa fa-check"></i><b>13</b> Multi-omics integration</a>
<ul>
<li class="chapter" data-level="13.1" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#example-in-giotto-suite-manuscript"><i class="fa fa-check"></i><b>13.1</b> Example in giotto suite manuscript</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html"><i class="fa fa-check"></i><b>14</b> Interoperability with other frameworks</a>
<ul>
<li class="chapter" data-level="14.1" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#anndataspatialdata"><i class="fa fa-check"></i><b>14.1</b> AnnData/SpatialData</a></li>
<li class="chapter" data-level="14.2" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#spatialexperiment"><i class="fa fa-check"></i><b>14.2</b> SpatialExperiment</a></li>
<li class="chapter" data-level="14.3" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#seurat"><i class="fa fa-check"></i><b>14.3</b> Seurat</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html"><i class="fa fa-check"></i><b>15</b> Interoperability with isolated tools</a>
<ul>
<li class="chapter" data-level="15.1" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#spatial-niche-trajectory-analysis"><i class="fa fa-check"></i><b>15.1</b> Spatial niche trajectory analysis</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html"><i class="fa fa-check"></i><b>16</b> Interactivity with the R/Spatial ecosystem</a>
<ul>
<li class="chapter" data-level="16.1" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#kriging"><i class="fa fa-check"></i><b>16.1</b> Kriging</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="contributing-to-giotto.html"><a href="contributing-to-giotto.html"><i class="fa fa-check"></i><b>17</b> Contributing to Giotto</a>
<ul>
<li class="chapter" data-level="17.1" data-path="contributing-to-giotto.html"><a href="contributing-to-giotto.html#contribution-guideline"><i class="fa fa-check"></i><b>17.1</b> Contribution guideline</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Workshop: Spatial multi-omics data analysis with Giotto Suite</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="visium-part-ii" class="section level1 hasAnchor" number="7">
<h1><span class="header-section-number">7</span> Visium Part II<a href="visium-part-ii.html#visium-part-ii" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Joselyn Cristina Chávez Fuentes</p>
<p>August 6th 2024</p>
<div id="differential-expression" class="section level2 hasAnchor" number="7.1">
<h2><span class="header-section-number">7.1</span> Differential expression<a href="visium-part-ii.html#differential-expression" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ul>
<li>text</li>
</ul>
</div>
<div id="enrichment-deconvolution" class="section level2 hasAnchor" number="7.2">
<h2><span class="header-section-number">7.2</span> Enrichment &amp; Deconvolution<a href="visium-part-ii.html#enrichment-deconvolution" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ul>
<li>PAGE/Rank</li>
<li>SpatialDWLS</li>
<li>Visualizations</li>
</ul>
</div>
<div id="spatial-expression-patterns" class="section level2 hasAnchor" number="7.3">
<h2><span class="header-section-number">7.3</span> Spatial expression patterns<a href="visium-part-ii.html#spatial-expression-patterns" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ul>
<li>Spatial variable genes</li>
</ul>
<p>Rank binarization</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="visium-part-ii.html#cb27-1" tabindex="-1"></a>ranktest <span class="ot">=</span> <span class="fu">binSpect</span>(visium_brain, </span>
<span id="cb27-2"><a href="visium-part-ii.html#cb27-2" tabindex="-1"></a>                    <span class="at">bin_method =</span> <span class="st">&#39;rank&#39;</span>,</span>
<span id="cb27-3"><a href="visium-part-ii.html#cb27-3" tabindex="-1"></a>                    <span class="at">calc_hub =</span> <span class="cn">TRUE</span>, </span>
<span id="cb27-4"><a href="visium-part-ii.html#cb27-4" tabindex="-1"></a>                    <span class="at">hub_min_int =</span> <span class="dv">5</span>,</span>
<span id="cb27-5"><a href="visium-part-ii.html#cb27-5" tabindex="-1"></a>                    <span class="at">spatial_network_name =</span> <span class="st">&#39;spatial_network&#39;</span>)</span></code></pre></div>
<p>Visualize</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="visium-part-ii.html#cb28-1" tabindex="-1"></a><span class="fu">spatFeatPlot2D</span>(visium_brain, </span>
<span id="cb28-2"><a href="visium-part-ii.html#cb28-2" tabindex="-1"></a>               <span class="at">expression_values =</span> <span class="st">&#39;scaled&#39;</span>,</span>
<span id="cb28-3"><a href="visium-part-ii.html#cb28-3" tabindex="-1"></a>               <span class="at">feats =</span> ranktest<span class="sc">$</span>feats[<span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>], </span>
<span id="cb28-4"><a href="visium-part-ii.html#cb28-4" tabindex="-1"></a>               <span class="at">cow_n_col =</span> <span class="dv">2</span>, </span>
<span id="cb28-5"><a href="visium-part-ii.html#cb28-5" tabindex="-1"></a>               <span class="at">point_size =</span> <span class="fl">1.5</span>)</span></code></pre></div>
<ul>
<li>Spatial co-expression modules</li>
</ul>
<p>Cluster the top 500 spatial genes into 20 clusters</p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="visium-part-ii.html#cb29-1" tabindex="-1"></a>ext_spatial_genes <span class="ot">=</span> ranktest[<span class="dv">1</span><span class="sc">:</span><span class="dv">1500</span>,]<span class="sc">$</span>feats</span></code></pre></div>
<p>Use existing detectSpatialCorGenes function to calculate pairwise distances between genes (but set network_smoothing=0 to use default clustering)</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="visium-part-ii.html#cb30-1" tabindex="-1"></a>spat_cor_netw_DT <span class="ot">=</span> <span class="fu">detectSpatialCorFeats</span>(visium_brain,</span>
<span id="cb30-2"><a href="visium-part-ii.html#cb30-2" tabindex="-1"></a>                                         <span class="at">method =</span> <span class="st">&#39;network&#39;</span>,</span>
<span id="cb30-3"><a href="visium-part-ii.html#cb30-3" tabindex="-1"></a>                                         <span class="at">spatial_network_name =</span> <span class="st">&#39;spatial_network&#39;</span>,</span>
<span id="cb30-4"><a href="visium-part-ii.html#cb30-4" tabindex="-1"></a>                                         <span class="at">subset_feats =</span> ext_spatial_genes)</span></code></pre></div>
<p>Identify most similar spatially correlated genes for one gene</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="visium-part-ii.html#cb31-1" tabindex="-1"></a>top10_genes <span class="ot">=</span> <span class="fu">showSpatialCorFeats</span>(spat_cor_netw_DT, </span>
<span id="cb31-2"><a href="visium-part-ii.html#cb31-2" tabindex="-1"></a>                                  <span class="at">feats =</span> <span class="st">&#39;Ptprn&#39;</span>, </span>
<span id="cb31-3"><a href="visium-part-ii.html#cb31-3" tabindex="-1"></a>                                  <span class="at">show_top_feats =</span> <span class="dv">10</span>)</span></code></pre></div>
<p>Visualize</p>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="visium-part-ii.html#cb32-1" tabindex="-1"></a><span class="fu">spatFeatPlot2D</span>(visium_brain, </span>
<span id="cb32-2"><a href="visium-part-ii.html#cb32-2" tabindex="-1"></a>               <span class="at">expression_values =</span> <span class="st">&#39;scaled&#39;</span>,</span>
<span id="cb32-3"><a href="visium-part-ii.html#cb32-3" tabindex="-1"></a>               <span class="at">feats =</span> top10_genes<span class="sc">$</span>variable[<span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>], </span>
<span id="cb32-4"><a href="visium-part-ii.html#cb32-4" tabindex="-1"></a>               <span class="at">point_size =</span> <span class="dv">3</span>)</span></code></pre></div>
<p>Cluster spatial genes</p>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="visium-part-ii.html#cb33-1" tabindex="-1"></a>spat_cor_netw_DT <span class="ot">=</span> <span class="fu">clusterSpatialCorFeats</span>(spat_cor_netw_DT, </span>
<span id="cb33-2"><a href="visium-part-ii.html#cb33-2" tabindex="-1"></a>                                          <span class="at">name =</span> <span class="st">&#39;spat_netw_clus&#39;</span>, </span>
<span id="cb33-3"><a href="visium-part-ii.html#cb33-3" tabindex="-1"></a>                                          <span class="at">k =</span> <span class="dv">20</span>)</span></code></pre></div>
<p>Visualize clusters</p>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="visium-part-ii.html#cb34-1" tabindex="-1"></a><span class="fu">heatmSpatialCorFeats</span>(visium_brain,</span>
<span id="cb34-2"><a href="visium-part-ii.html#cb34-2" tabindex="-1"></a>                     <span class="at">spatCorObject =</span> spat_cor_netw_DT,</span>
<span id="cb34-3"><a href="visium-part-ii.html#cb34-3" tabindex="-1"></a>                     <span class="at">use_clus_name =</span> <span class="st">&#39;spat_netw_clus&#39;</span>,</span>
<span id="cb34-4"><a href="visium-part-ii.html#cb34-4" tabindex="-1"></a>                     <span class="at">heatmap_legend_param =</span> <span class="fu">list</span>(<span class="at">title =</span> <span class="cn">NULL</span>),</span>
<span id="cb34-5"><a href="visium-part-ii.html#cb34-5" tabindex="-1"></a>                     <span class="at">save_param =</span> <span class="fu">list</span>(<span class="at">base_height =</span> <span class="dv">6</span>, <span class="at">base_width =</span> <span class="dv">8</span>, <span class="at">units =</span> <span class="st">&#39;cm&#39;</span>))</span></code></pre></div>
<p>Rank spatial correlated clusters and show genes for selected clusters</p>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="visium-part-ii.html#cb35-1" tabindex="-1"></a>netw_ranks <span class="ot">=</span> <span class="fu">rankSpatialCorGroups</span>(visium_brain,</span>
<span id="cb35-2"><a href="visium-part-ii.html#cb35-2" tabindex="-1"></a>                                  <span class="at">spatCorObject =</span> spat_cor_netw_DT, </span>
<span id="cb35-3"><a href="visium-part-ii.html#cb35-3" tabindex="-1"></a>                                  <span class="at">use_clus_name =</span> <span class="st">&#39;spat_netw_clus&#39;</span>,</span>
<span id="cb35-4"><a href="visium-part-ii.html#cb35-4" tabindex="-1"></a>                                  <span class="at">save_param =</span> <span class="fu">list</span>(<span class="at">base_height =</span> <span class="dv">3</span>, <span class="at">base_width =</span> <span class="dv">5</span>))</span>
<span id="cb35-5"><a href="visium-part-ii.html#cb35-5" tabindex="-1"></a></span>
<span id="cb35-6"><a href="visium-part-ii.html#cb35-6" tabindex="-1"></a>top_netw_spat_cluster <span class="ot">=</span> <span class="fu">showSpatialCorFeats</span>(spat_cor_netw_DT, </span>
<span id="cb35-7"><a href="visium-part-ii.html#cb35-7" tabindex="-1"></a>                                            <span class="at">use_clus_name =</span> <span class="st">&#39;spat_netw_clus&#39;</span>,</span>
<span id="cb35-8"><a href="visium-part-ii.html#cb35-8" tabindex="-1"></a>                                            <span class="at">selected_clusters =</span> <span class="dv">6</span>, </span>
<span id="cb35-9"><a href="visium-part-ii.html#cb35-9" tabindex="-1"></a>                                            <span class="at">show_top_feats =</span> <span class="dv">1</span>)</span></code></pre></div>
<p>Create metagene enrichment score for clusters</p>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="visium-part-ii.html#cb36-1" tabindex="-1"></a>cluster_genes_DT <span class="ot">=</span> <span class="fu">showSpatialCorFeats</span>(spat_cor_netw_DT, </span>
<span id="cb36-2"><a href="visium-part-ii.html#cb36-2" tabindex="-1"></a>                                       <span class="at">use_clus_name =</span> <span class="st">&#39;spat_netw_clus&#39;</span>, </span>
<span id="cb36-3"><a href="visium-part-ii.html#cb36-3" tabindex="-1"></a>                                       <span class="at">show_top_feats =</span> <span class="dv">1</span>)</span>
<span id="cb36-4"><a href="visium-part-ii.html#cb36-4" tabindex="-1"></a></span>
<span id="cb36-5"><a href="visium-part-ii.html#cb36-5" tabindex="-1"></a>cluster_genes <span class="ot">=</span> cluster_genes_DT<span class="sc">$</span>clus </span>
<span id="cb36-6"><a href="visium-part-ii.html#cb36-6" tabindex="-1"></a><span class="fu">names</span>(cluster_genes) <span class="ot">=</span> cluster_genes_DT<span class="sc">$</span>feat_ID</span>
<span id="cb36-7"><a href="visium-part-ii.html#cb36-7" tabindex="-1"></a></span>
<span id="cb36-8"><a href="visium-part-ii.html#cb36-8" tabindex="-1"></a>visium_brain <span class="ot">&lt;-</span> <span class="fu">createMetafeats</span>(visium_brain, </span>
<span id="cb36-9"><a href="visium-part-ii.html#cb36-9" tabindex="-1"></a>                                <span class="at">feat_clusters =</span> cluster_genes, </span>
<span id="cb36-10"><a href="visium-part-ii.html#cb36-10" tabindex="-1"></a>                                <span class="at">name =</span> <span class="st">&#39;cluster_metagene&#39;</span>)</span>
<span id="cb36-11"><a href="visium-part-ii.html#cb36-11" tabindex="-1"></a></span>
<span id="cb36-12"><a href="visium-part-ii.html#cb36-12" tabindex="-1"></a><span class="fu">spatCellPlot</span>(visium_brain,</span>
<span id="cb36-13"><a href="visium-part-ii.html#cb36-13" tabindex="-1"></a>             <span class="at">spat_enr_names =</span> <span class="st">&#39;cluster_metagene&#39;</span>,</span>
<span id="cb36-14"><a href="visium-part-ii.html#cb36-14" tabindex="-1"></a>             <span class="at">cell_annotation_values =</span> netw_ranks<span class="sc">$</span>clusters,</span>
<span id="cb36-15"><a href="visium-part-ii.html#cb36-15" tabindex="-1"></a>             <span class="at">point_size =</span> <span class="dv">1</span>, </span>
<span id="cb36-16"><a href="visium-part-ii.html#cb36-16" tabindex="-1"></a>             <span class="at">cow_n_col =</span> <span class="dv">5</span>, </span>
<span id="cb36-17"><a href="visium-part-ii.html#cb36-17" tabindex="-1"></a>             <span class="at">save_param =</span> <span class="fu">list</span>(<span class="at">base_width =</span> <span class="dv">15</span>))</span></code></pre></div>
<ul>
<li>Spatially informed clusters</li>
</ul>
<p>Get the top 30 genes per spatial co-expression cluster</p>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="visium-part-ii.html#cb37-1" tabindex="-1"></a><span class="fu">table</span>(spat_cor_netw_DT<span class="sc">$</span>cor_clusters<span class="sc">$</span>spat_netw_clus)</span>
<span id="cb37-2"><a href="visium-part-ii.html#cb37-2" tabindex="-1"></a></span>
<span id="cb37-3"><a href="visium-part-ii.html#cb37-3" tabindex="-1"></a>coexpr_dt <span class="ot">=</span> data.table<span class="sc">::</span><span class="fu">data.table</span>(<span class="at">genes =</span> <span class="fu">names</span>(spat_cor_netw_DT<span class="sc">$</span>cor_clusters<span class="sc">$</span>spat_netw_clus),</span>
<span id="cb37-4"><a href="visium-part-ii.html#cb37-4" tabindex="-1"></a>                                   <span class="at">cluster =</span> spat_cor_netw_DT<span class="sc">$</span>cor_clusters<span class="sc">$</span>spat_netw_clus)</span>
<span id="cb37-5"><a href="visium-part-ii.html#cb37-5" tabindex="-1"></a></span>
<span id="cb37-6"><a href="visium-part-ii.html#cb37-6" tabindex="-1"></a>data.table<span class="sc">::</span><span class="fu">setorder</span>(coexpr_dt, cluster)</span>
<span id="cb37-7"><a href="visium-part-ii.html#cb37-7" tabindex="-1"></a>top30_coexpr_dt <span class="ot">=</span> coexpr_dt[, <span class="fu">head</span>(.SD, <span class="dv">30</span>) , by <span class="ot">=</span> cluster]</span>
<span id="cb37-8"><a href="visium-part-ii.html#cb37-8" tabindex="-1"></a>my_spatial_genes <span class="ot">=</span> top30_coexpr_dt<span class="sc">$</span>genes</span></code></pre></div>
<p>Re-calculate the clustering</p>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb38-1"><a href="visium-part-ii.html#cb38-1" tabindex="-1"></a>visium_brain <span class="ot">&lt;-</span> <span class="fu">runPCA</span>(<span class="at">gobject =</span> visium_brain,</span>
<span id="cb38-2"><a href="visium-part-ii.html#cb38-2" tabindex="-1"></a>                       <span class="at">feats_to_use =</span> my_spatial_genes,</span>
<span id="cb38-3"><a href="visium-part-ii.html#cb38-3" tabindex="-1"></a>                       <span class="at">name =</span> <span class="st">&#39;custom_pca&#39;</span>)</span>
<span id="cb38-4"><a href="visium-part-ii.html#cb38-4" tabindex="-1"></a></span>
<span id="cb38-5"><a href="visium-part-ii.html#cb38-5" tabindex="-1"></a>visium_brain <span class="ot">&lt;-</span> <span class="fu">runUMAP</span>(visium_brain, </span>
<span id="cb38-6"><a href="visium-part-ii.html#cb38-6" tabindex="-1"></a>                        <span class="at">dim_reduction_name =</span> <span class="st">&#39;custom_pca&#39;</span>, </span>
<span id="cb38-7"><a href="visium-part-ii.html#cb38-7" tabindex="-1"></a>                        <span class="at">dimensions_to_use =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>,</span>
<span id="cb38-8"><a href="visium-part-ii.html#cb38-8" tabindex="-1"></a>                        <span class="at">name =</span> <span class="st">&#39;custom_umap&#39;</span>)</span>
<span id="cb38-9"><a href="visium-part-ii.html#cb38-9" tabindex="-1"></a></span>
<span id="cb38-10"><a href="visium-part-ii.html#cb38-10" tabindex="-1"></a>visium_brain <span class="ot">&lt;-</span> <span class="fu">createNearestNetwork</span>(<span class="at">gobject =</span> visium_brain,</span>
<span id="cb38-11"><a href="visium-part-ii.html#cb38-11" tabindex="-1"></a>                                     <span class="at">dim_reduction_name =</span> <span class="st">&#39;custom_pca&#39;</span>,</span>
<span id="cb38-12"><a href="visium-part-ii.html#cb38-12" tabindex="-1"></a>                                     <span class="at">dimensions_to_use =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>, </span>
<span id="cb38-13"><a href="visium-part-ii.html#cb38-13" tabindex="-1"></a>                                     <span class="at">k =</span> <span class="dv">5</span>,</span>
<span id="cb38-14"><a href="visium-part-ii.html#cb38-14" tabindex="-1"></a>                                     <span class="at">name =</span> <span class="st">&#39;custom_NN&#39;</span>)</span>
<span id="cb38-15"><a href="visium-part-ii.html#cb38-15" tabindex="-1"></a></span>
<span id="cb38-16"><a href="visium-part-ii.html#cb38-16" tabindex="-1"></a>visium_brain <span class="ot">&lt;-</span> <span class="fu">doLeidenCluster</span>(<span class="at">gobject =</span> visium_brain, </span>
<span id="cb38-17"><a href="visium-part-ii.html#cb38-17" tabindex="-1"></a>                                <span class="at">network_name =</span> <span class="st">&#39;custom_NN&#39;</span>,</span>
<span id="cb38-18"><a href="visium-part-ii.html#cb38-18" tabindex="-1"></a>                                <span class="at">resolution =</span> <span class="fl">0.15</span>, </span>
<span id="cb38-19"><a href="visium-part-ii.html#cb38-19" tabindex="-1"></a>                                <span class="at">n_iterations =</span> <span class="dv">1000</span>,</span>
<span id="cb38-20"><a href="visium-part-ii.html#cb38-20" tabindex="-1"></a>                                <span class="at">name =</span> <span class="st">&#39;custom_leiden&#39;</span>)</span>
<span id="cb38-21"><a href="visium-part-ii.html#cb38-21" tabindex="-1"></a></span>
<span id="cb38-22"><a href="visium-part-ii.html#cb38-22" tabindex="-1"></a>cell_meta <span class="ot">=</span> <span class="fu">pDataDT</span>(visium_brain)</span>
<span id="cb38-23"><a href="visium-part-ii.html#cb38-23" tabindex="-1"></a>cell_clusters <span class="ot">=</span> <span class="fu">unique</span>(cell_meta<span class="sc">$</span>custom_leiden)</span>
<span id="cb38-24"><a href="visium-part-ii.html#cb38-24" tabindex="-1"></a></span>
<span id="cb38-25"><a href="visium-part-ii.html#cb38-25" tabindex="-1"></a>selected_colors <span class="ot">=</span> <span class="fu">getDistinctColors</span>(<span class="fu">length</span>(cell_clusters))</span>
<span id="cb38-26"><a href="visium-part-ii.html#cb38-26" tabindex="-1"></a><span class="fu">names</span>(selected_colors) <span class="ot">=</span> cell_clusters</span></code></pre></div>
<p>Visualize</p>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="visium-part-ii.html#cb39-1" tabindex="-1"></a><span class="fu">spatPlot2D</span>(visium_brain, </span>
<span id="cb39-2"><a href="visium-part-ii.html#cb39-2" tabindex="-1"></a>           <span class="at">cell_color =</span> <span class="st">&#39;custom_leiden&#39;</span>, </span>
<span id="cb39-3"><a href="visium-part-ii.html#cb39-3" tabindex="-1"></a>           <span class="at">cell_color_code =</span> selected_colors, </span>
<span id="cb39-4"><a href="visium-part-ii.html#cb39-4" tabindex="-1"></a>           <span class="at">coord_fix_ratio =</span> <span class="dv">1</span>)</span></code></pre></div>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="visium-part-ii.html#cb40-1" tabindex="-1"></a><span class="fu">plotUMAP</span>(<span class="at">gobject =</span> visium_brain, </span>
<span id="cb40-2"><a href="visium-part-ii.html#cb40-2" tabindex="-1"></a>         <span class="at">cell_color =</span> <span class="st">&#39;custom_leiden&#39;</span>, </span>
<span id="cb40-3"><a href="visium-part-ii.html#cb40-3" tabindex="-1"></a>         <span class="at">cell_color_code =</span> selected_colors, </span>
<span id="cb40-4"><a href="visium-part-ii.html#cb40-4" tabindex="-1"></a>         <span class="at">point_size =</span> <span class="fl">1.5</span>)</span></code></pre></div>
<ul>
<li>Spatial domains HMRF</li>
</ul>
<p>Do HMRF with different betas on top 30 genes per spatial co-expression module</p>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="visium-part-ii.html#cb41-1" tabindex="-1"></a>hmrf_folder <span class="ot">=</span> <span class="fu">paste0</span>(results_folder,<span class="st">&#39;/&#39;</span>,<span class="st">&#39;11_HMRF/&#39;</span>)</span>
<span id="cb41-2"><a href="visium-part-ii.html#cb41-2" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">file.exists</span>(hmrf_folder)) <span class="fu">dir.create</span>(hmrf_folder, <span class="at">recursive =</span> T)</span>
<span id="cb41-3"><a href="visium-part-ii.html#cb41-3" tabindex="-1"></a></span>
<span id="cb41-4"><a href="visium-part-ii.html#cb41-4" tabindex="-1"></a>HMRF_spatial_genes <span class="ot">=</span> <span class="fu">doHMRF</span>(<span class="at">gobject =</span> visium_brain,</span>
<span id="cb41-5"><a href="visium-part-ii.html#cb41-5" tabindex="-1"></a>                            <span class="at">expression_values =</span> <span class="st">&#39;scaled&#39;</span>,</span>
<span id="cb41-6"><a href="visium-part-ii.html#cb41-6" tabindex="-1"></a>                            <span class="at">spatial_genes =</span> my_spatial_genes, </span>
<span id="cb41-7"><a href="visium-part-ii.html#cb41-7" tabindex="-1"></a>                            <span class="at">k =</span> <span class="dv">20</span>,</span>
<span id="cb41-8"><a href="visium-part-ii.html#cb41-8" tabindex="-1"></a>                            <span class="at">spatial_network_name=</span><span class="st">&quot;spatial_network&quot;</span>,</span>
<span id="cb41-9"><a href="visium-part-ii.html#cb41-9" tabindex="-1"></a>                            <span class="at">betas =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">10</span>, <span class="dv">5</span>),</span>
<span id="cb41-10"><a href="visium-part-ii.html#cb41-10" tabindex="-1"></a>                            <span class="at">output_folder =</span> <span class="fu">paste0</span>(hmrf_folder, <span class="st">&#39;/&#39;</span>, <span class="st">&#39;Spatial_genes/SG_topgenes_k20_scaled&#39;</span>))</span>
<span id="cb41-11"><a href="visium-part-ii.html#cb41-11" tabindex="-1"></a></span>
<span id="cb41-12"><a href="visium-part-ii.html#cb41-12" tabindex="-1"></a>visium_brain <span class="ot">&lt;-</span> <span class="fu">addHMRF</span>(<span class="at">gobject =</span> visium_brain, </span>
<span id="cb41-13"><a href="visium-part-ii.html#cb41-13" tabindex="-1"></a>                        <span class="at">HMRFoutput =</span> HMRF_spatial_genes,</span>
<span id="cb41-14"><a href="visium-part-ii.html#cb41-14" tabindex="-1"></a>                        <span class="at">k =</span> <span class="dv">20</span>, </span>
<span id="cb41-15"><a href="visium-part-ii.html#cb41-15" tabindex="-1"></a>                        <span class="at">betas_to_add =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">10</span>, <span class="dv">20</span>, <span class="dv">30</span>, <span class="dv">40</span>),</span>
<span id="cb41-16"><a href="visium-part-ii.html#cb41-16" tabindex="-1"></a>                        <span class="at">hmrf_name =</span> <span class="st">&#39;HMRF&#39;</span>)</span>
<span id="cb41-17"><a href="visium-part-ii.html#cb41-17" tabindex="-1"></a></span>
<span id="cb41-18"><a href="visium-part-ii.html#cb41-18" tabindex="-1"></a><span class="fu">spatPlot2D</span>(<span class="at">gobject =</span> visium_brain, </span>
<span id="cb41-19"><a href="visium-part-ii.html#cb41-19" tabindex="-1"></a>           <span class="at">cell_color =</span> <span class="st">&#39;HMRF_k20_b.40&#39;</span>)</span></code></pre></div>
</div>
<div id="interactive-tools" class="section level2 hasAnchor" number="7.4">
<h2><span class="header-section-number">7.4</span> Interactive tools<a href="visium-part-ii.html#interactive-tools" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>text</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="visium-part-i.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="visium-hd.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
