<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>7 Visium Part II | Workshop: Spatial multi-omics data analysis with Giotto Suite</title>
  <meta name="description" content="7 Visium Part II | Workshop: Spatial multi-omics data analysis with Giotto Suite" />
  <meta name="generator" content="bookdown 0.40 and GitBook 2.6.7" />

  <meta property="og:title" content="7 Visium Part II | Workshop: Spatial multi-omics data analysis with Giotto Suite" />
  <meta property="og:type" content="book" />
  
  
  <meta name="github-repo" content="drieslab/giotto_workshop_2024" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="7 Visium Part II | Workshop: Spatial multi-omics data analysis with Giotto Suite" />
  
  
  

<meta name="author" content="Ruben Dries, Jiaji George Chen, Joselyn Cristina Chávez-Fuentes, Junxiang Xu ,Edward Ruiz, Jeff Sheridan, Iqra Amin, Wen Wang" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="visium-part-i.html"/>
<link rel="next" href="visium-hd.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Giotto Workshop 2024</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#instructors"><i class="fa fa-check"></i><b>1.1</b> Instructors</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#topics-and-schedule"><i class="fa fa-check"></i><b>1.2</b> Topics and Schedule:</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#license"><i class="fa fa-check"></i><b>1.3</b> License</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html"><i class="fa fa-check"></i><b>2</b> Spatial omics technologies</a>
<ul>
<li class="chapter" data-level="2.1" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#slides"><i class="fa fa-check"></i><b>2.1</b> Slides</a></li>
<li class="chapter" data-level="2.2" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#spatial-sequencing"><i class="fa fa-check"></i><b>2.2</b> Spatial sequencing</a></li>
<li class="chapter" data-level="2.3" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#spatial-in-situ"><i class="fa fa-check"></i><b>2.3</b> Spatial in situ</a></li>
<li class="chapter" data-level="2.4" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#spatial-proteomics"><i class="fa fa-check"></i><b>2.4</b> Spatial proteomics</a></li>
<li class="chapter" data-level="2.5" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#other-spatial-omics-atac-seq-cutrun-lipidomics-etc"><i class="fa fa-check"></i><b>2.5</b> Other Spatial omics: ATAC-seq, CUT&amp;RUN, lipidomics, etc</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html"><i class="fa fa-check"></i><b>3</b> Introduction to the Giotto package</a>
<ul>
<li class="chapter" data-level="3.1" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html#slides-1"><i class="fa fa-check"></i><b>3.1</b> Slides</a></li>
<li class="chapter" data-level="3.2" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html#ecosystem"><i class="fa fa-check"></i><b>3.2</b> Ecosystem</a></li>
<li class="chapter" data-level="3.3" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html#installation-python-environment"><i class="fa fa-check"></i><b>3.3</b> Installation + python environment</a></li>
<li class="chapter" data-level="3.4" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html#giotto-instructions"><i class="fa fa-check"></i><b>3.4</b> Giotto instructions</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html"><i class="fa fa-check"></i><b>4</b> Data formatting and Pre-processing</a>
<ul>
<li class="chapter" data-level="4.1" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#data-formats"><i class="fa fa-check"></i><b>4.1</b> Data formats</a></li>
<li class="chapter" data-level="4.2" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#pre-processing"><i class="fa fa-check"></i><b>4.2</b> Pre-processing</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html"><i class="fa fa-check"></i><b>5</b> Creating a Giotto object</a>
<ul>
<li class="chapter" data-level="5.1" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#overview"><i class="fa fa-check"></i><b>5.1</b> Overview</a></li>
<li class="chapter" data-level="5.2" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#from-matrix-locations"><i class="fa fa-check"></i><b>5.2</b> From matrix + locations</a></li>
<li class="chapter" data-level="5.3" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#from-subcellular-raw-data-transcripts-or-images-polygons"><i class="fa fa-check"></i><b>5.3</b> From subcellular raw data (transcripts or images) + polygons</a></li>
<li class="chapter" data-level="5.4" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#from-piece-wise"><i class="fa fa-check"></i><b>5.4</b> From piece-wise</a></li>
<li class="chapter" data-level="5.5" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#using-convenience-functions-for-popular-technologies-vizgen-xenium-cosmx"><i class="fa fa-check"></i><b>5.5</b> Using convenience functions for popular technologies (Vizgen, Xenium, CosMx, …)</a></li>
<li class="chapter" data-level="5.6" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#spatial-plots"><i class="fa fa-check"></i><b>5.6</b> Spatial plots</a></li>
<li class="chapter" data-level="5.7" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#subsetting"><i class="fa fa-check"></i><b>5.7</b> Subsetting</a></li>
<li class="chapter" data-level="5.8" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#mini-objects-giottodata"><i class="fa fa-check"></i><b>5.8</b> Mini objects &amp; GiottoData</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="visium-part-i.html"><a href="visium-part-i.html"><i class="fa fa-check"></i><b>6</b> Visium Part I</a>
<ul>
<li class="chapter" data-level="6.1" data-path="visium-part-i.html"><a href="visium-part-i.html#introduction-to-the-spatial-dataset"><i class="fa fa-check"></i><b>6.1</b> Introduction to the spatial dataset</a></li>
<li class="chapter" data-level="6.2" data-path="visium-part-i.html"><a href="visium-part-i.html#download-dataset"><i class="fa fa-check"></i><b>6.2</b> Download dataset</a></li>
<li class="chapter" data-level="6.3" data-path="visium-part-i.html"><a href="visium-part-i.html#create-the-giotto-object"><i class="fa fa-check"></i><b>6.3</b> Create the Giotto object</a></li>
<li class="chapter" data-level="6.4" data-path="visium-part-i.html"><a href="visium-part-i.html#subset-on-spots-that-were-covered-by-tissue"><i class="fa fa-check"></i><b>6.4</b> Subset on spots that were covered by tissue</a></li>
<li class="chapter" data-level="6.5" data-path="visium-part-i.html"><a href="visium-part-i.html#quality-control"><i class="fa fa-check"></i><b>6.5</b> Quality control</a></li>
<li class="chapter" data-level="6.6" data-path="visium-part-i.html"><a href="visium-part-i.html#filtering"><i class="fa fa-check"></i><b>6.6</b> Filtering</a></li>
<li class="chapter" data-level="6.7" data-path="visium-part-i.html"><a href="visium-part-i.html#normalization"><i class="fa fa-check"></i><b>6.7</b> Normalization</a></li>
<li class="chapter" data-level="6.8" data-path="visium-part-i.html"><a href="visium-part-i.html#feature-selection"><i class="fa fa-check"></i><b>6.8</b> Feature selection</a>
<ul>
<li class="chapter" data-level="6.8.1" data-path="visium-part-i.html"><a href="visium-part-i.html#highly-variable-features"><i class="fa fa-check"></i><b>6.8.1</b> Highly Variable Features:</a></li>
</ul></li>
<li class="chapter" data-level="6.9" data-path="visium-part-i.html"><a href="visium-part-i.html#dimension-reduction"><i class="fa fa-check"></i><b>6.9</b> Dimension Reduction</a>
<ul>
<li class="chapter" data-level="6.9.1" data-path="visium-part-i.html"><a href="visium-part-i.html#pca"><i class="fa fa-check"></i><b>6.9.1</b> PCA</a></li>
<li class="chapter" data-level="6.9.2" data-path="visium-part-i.html"><a href="visium-part-i.html#umap"><i class="fa fa-check"></i><b>6.9.2</b> UMAP</a></li>
<li class="chapter" data-level="6.9.3" data-path="visium-part-i.html"><a href="visium-part-i.html#t-sne"><i class="fa fa-check"></i><b>6.9.3</b> t-SNE</a></li>
</ul></li>
<li class="chapter" data-level="6.10" data-path="visium-part-i.html"><a href="visium-part-i.html#clustering"><i class="fa fa-check"></i><b>6.10</b> Clustering</a>
<ul>
<li class="chapter" data-level="6.10.1" data-path="visium-part-i.html"><a href="visium-part-i.html#calculate-leiden-clustering"><i class="fa fa-check"></i><b>6.10.1</b> Calculate Leiden clustering</a></li>
<li class="chapter" data-level="6.10.2" data-path="visium-part-i.html"><a href="visium-part-i.html#calculate-louvain-clustering"><i class="fa fa-check"></i><b>6.10.2</b> Calculate Louvain clustering</a></li>
</ul></li>
<li class="chapter" data-level="6.11" data-path="visium-part-i.html"><a href="visium-part-i.html#save-the-object"><i class="fa fa-check"></i><b>6.11</b> Save the object</a></li>
<li class="chapter" data-level="6.12" data-path="visium-part-i.html"><a href="visium-part-i.html#session-info"><i class="fa fa-check"></i><b>6.12</b> Session info</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="visium-part-ii.html"><a href="visium-part-ii.html"><i class="fa fa-check"></i><b>7</b> Visium Part II</a>
<ul>
<li class="chapter" data-level="7.1" data-path="visium-part-ii.html"><a href="visium-part-ii.html#load-the-object"><i class="fa fa-check"></i><b>7.1</b> Load the object</a></li>
<li class="chapter" data-level="7.2" data-path="visium-part-ii.html"><a href="visium-part-ii.html#differential-expression"><i class="fa fa-check"></i><b>7.2</b> Differential expression</a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="visium-part-ii.html"><a href="visium-part-ii.html#gini-markers"><i class="fa fa-check"></i><b>7.2.1</b> Gini markers</a></li>
<li class="chapter" data-level="7.2.2" data-path="visium-part-ii.html"><a href="visium-part-ii.html#scran-markers"><i class="fa fa-check"></i><b>7.2.2</b> Scran markers</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="visium-part-ii.html"><a href="visium-part-ii.html#enrichment-deconvolution"><i class="fa fa-check"></i><b>7.3</b> Enrichment &amp; Deconvolution</a>
<ul>
<li class="chapter" data-level="7.3.1" data-path="visium-part-ii.html"><a href="visium-part-ii.html#pagerank"><i class="fa fa-check"></i><b>7.3.1</b> PAGE/Rank</a></li>
<li class="chapter" data-level="7.3.2" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatialdwls"><i class="fa fa-check"></i><b>7.3.2</b> SpatialDWLS</a></li>
</ul></li>
<li class="chapter" data-level="7.4" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatial-expression-patterns"><i class="fa fa-check"></i><b>7.4</b> Spatial expression patterns</a>
<ul>
<li class="chapter" data-level="7.4.1" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatial-variable-genes"><i class="fa fa-check"></i><b>7.4.1</b> Spatial variable genes</a></li>
<li class="chapter" data-level="7.4.2" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatial-co-expression-modules"><i class="fa fa-check"></i><b>7.4.2</b> Spatial co-expression modules</a></li>
</ul></li>
<li class="chapter" data-level="7.5" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatially-informed-clusters"><i class="fa fa-check"></i><b>7.5</b> Spatially informed clusters</a></li>
<li class="chapter" data-level="7.6" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatial-domains-hmrf"><i class="fa fa-check"></i><b>7.6</b> Spatial domains HMRF</a></li>
<li class="chapter" data-level="7.7" data-path="visium-part-ii.html"><a href="visium-part-ii.html#interactive-tools"><i class="fa fa-check"></i><b>7.7</b> Interactive tools</a></li>
<li class="chapter" data-level="7.8" data-path="visium-part-ii.html"><a href="visium-part-ii.html#session-info-1"><i class="fa fa-check"></i><b>7.8</b> Session info</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="visium-hd.html"><a href="visium-hd.html"><i class="fa fa-check"></i><b>8</b> Visium HD</a>
<ul>
<li class="chapter" data-level="8.1" data-path="visium-hd.html"><a href="visium-hd.html#objective"><i class="fa fa-check"></i><b>8.1</b> Objective</a></li>
<li class="chapter" data-level="8.2" data-path="visium-hd.html"><a href="visium-hd.html#background"><i class="fa fa-check"></i><b>8.2</b> Background</a>
<ul>
<li class="chapter" data-level="8.2.1" data-path="visium-hd.html"><a href="visium-hd.html#visium-hd-technology"><i class="fa fa-check"></i><b>8.2.1</b> Visium HD Technology</a></li>
<li class="chapter" data-level="8.2.2" data-path="visium-hd.html"><a href="visium-hd.html#colorectal-cancer-sample"><i class="fa fa-check"></i><b>8.2.2</b> Colorectal Cancer Sample</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="visium-hd.html"><a href="visium-hd.html#data-ingestion"><i class="fa fa-check"></i><b>8.3</b> Data Ingestion</a>
<ul>
<li class="chapter" data-level="8.3.1" data-path="visium-hd.html"><a href="visium-hd.html#visium-hd-output-data-format"><i class="fa fa-check"></i><b>8.3.1</b> Visium HD output data format</a></li>
<li class="chapter" data-level="8.3.2" data-path="visium-hd.html"><a href="visium-hd.html#read-in-raw-data"><i class="fa fa-check"></i><b>8.3.2</b> Read in raw data</a></li>
<li class="chapter" data-level="8.3.3" data-path="visium-hd.html"><a href="visium-hd.html#giotto-visium-hd-convenience-function"><i class="fa fa-check"></i><b>8.3.3</b> Giotto Visium HD convenience function</a></li>
</ul></li>
<li class="chapter" data-level="8.4" data-path="visium-hd.html"><a href="visium-hd.html#tiling-and-aggregation"><i class="fa fa-check"></i><b>8.4</b> Tiling and aggregation</a></li>
<li class="chapter" data-level="8.5" data-path="visium-hd.html"><a href="visium-hd.html#scalability-and-projection-functions"><i class="fa fa-check"></i><b>8.5</b> Scalability and projection functions</a></li>
<li class="chapter" data-level="8.6" data-path="visium-hd.html"><a href="visium-hd.html#spatial-expression-patterns-1"><i class="fa fa-check"></i><b>8.6</b> Spatial expression patterns</a></li>
<li class="chapter" data-level="8.7" data-path="visium-hd.html"><a href="visium-hd.html#spatial-co-expression-modules-1"><i class="fa fa-check"></i><b>8.7</b> Spatial co-expression modules</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="xenium.html"><a href="xenium.html"><i class="fa fa-check"></i><b>9</b> Xenium</a>
<ul>
<li class="chapter" data-level="9.1" data-path="xenium.html"><a href="xenium.html#introduction-to-spatial-dataset"><i class="fa fa-check"></i><b>9.1</b> Introduction to spatial dataset</a></li>
<li class="chapter" data-level="9.2" data-path="xenium.html"><a href="xenium.html#additional-package-requirements"><i class="fa fa-check"></i><b>9.2</b> Additional package requirements</a></li>
<li class="chapter" data-level="9.3" data-path="xenium.html"><a href="xenium.html#introduce-mini-object"><i class="fa fa-check"></i><b>9.3</b> Introduce mini object</a></li>
<li class="chapter" data-level="9.4" data-path="xenium.html"><a href="xenium.html#read-in-raw-data-1"><i class="fa fa-check"></i><b>9.4</b> Read in raw data</a></li>
<li class="chapter" data-level="9.5" data-path="xenium.html"><a href="xenium.html#overlap-txs-polygons"><i class="fa fa-check"></i><b>9.5</b> Overlap txs &amp; polygons</a></li>
<li class="chapter" data-level="9.6" data-path="xenium.html"><a href="xenium.html#transcript-enrichment-gsea"><i class="fa fa-check"></i><b>9.6</b> Transcript enrichment GSEA</a></li>
<li class="chapter" data-level="9.7" data-path="xenium.html"><a href="xenium.html#spatial-location-analysis"><i class="fa fa-check"></i><b>9.7</b> Spatial location analysis</a></li>
<li class="chapter" data-level="9.8" data-path="xenium.html"><a href="xenium.html#visualizations"><i class="fa fa-check"></i><b>9.8</b> Visualizations</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="exploration.html"><a href="exploration.html"><i class="fa fa-check"></i><b>10</b> —– exploration</a></li>
<li class="chapter" data-level="11" data-path="image.html"><a href="image.html"><i class="fa fa-check"></i><b>11</b> IMAGE</a></li>
<li class="chapter" data-level="12" data-path="ometif_2_tifimage_path-output_file-out_tif-chunk_rows-chunk_rows-overwrite-true.html"><a href="ometif_2_tifimage_path-output_file-out_tif-chunk_rows-chunk_rows-overwrite-true.html"><i class="fa fa-check"></i><b>12</b> ometif_2_tif(image_path, output_file = out_tif, chunk_rows = chunk_rows, overwrite = TRUE)</a></li>
<li class="chapter" data-level="13" data-path="example-usage.html"><a href="example-usage.html"><i class="fa fa-check"></i><b>13</b> # Example usage</a></li>
<li class="chapter" data-level="14" data-path="input_file-pathtoinputfile.ome.html"><a href="input_file-pathtoinputfile.ome.html"><i class="fa fa-check"></i><b>14</b> input_file = “path/to/input/file.ome.tif”</a></li>
<li class="chapter" data-level="15" data-path="output_file-pathtooutputfile.html"><a href="output_file-pathtooutputfile.html"><i class="fa fa-check"></i><b>15</b> output_file = “path/to/output/file.tif”</a></li>
<li class="chapter" data-level="16" data-path="chunk_size-1000-1000-adjust-the-chunk-size-as-needed.html"><a href="chunk_size-1000-1000-adjust-the-chunk-size-as-needed.html"><i class="fa fa-check"></i><b>16</b> chunk_size = (1000, 1000) # Adjust the chunk size as needed</a></li>
<li class="chapter" data-level="17" data-path="num_processes-4-adjust-the-number-of-processes-based-on-available-cpu-cores.html"><a href="num_processes-4-adjust-the-number-of-processes-based-on-available-cpu-cores.html"><i class="fa fa-check"></i><b>17</b> num_processes = 4 # Adjust the number of processes based on available CPU cores</a></li>
<li class="chapter" data-level="18" data-path="section.html"><a href="section.html"><i class="fa fa-check"></i><b>18</b> </a></li>
<li class="chapter" data-level="19" data-path="ometif_2_tifinput_file-output_file-chunk_size-num_processes.html"><a href="ometif_2_tifinput_file-output_file-chunk_size-num_processes.html"><i class="fa fa-check"></i><b>19</b> ometif_2_tif(input_file, output_file, chunk_size, num_processes)</a></li>
<li class="chapter" data-level="20" data-path="transcripts.html"><a href="transcripts.html"><i class="fa fa-check"></i><b>20</b> TRANSCRIPTS</a></li>
<li class="chapter" data-level="21" data-path="where-are-the-issue-tx-at.html"><a href="where-are-the-issue-tx-at.html"><i class="fa fa-check"></i><b>21</b> where are the issue TX at?</a></li>
<li class="chapter" data-level="22" data-path="polys.html"><a href="polys.html"><i class="fa fa-check"></i><b>22</b> POLYS</a></li>
<li class="chapter" data-level="23" data-path="giotto.html"><a href="giotto.html"><i class="fa fa-check"></i><b>23</b> giotto</a></li>
<li class="chapter" data-level="24" data-path="no-hvf-calc-since-there-are-not-that-many-different-gene-species.html"><a href="no-hvf-calc-since-there-are-not-that-many-different-gene-species.html"><i class="fa fa-check"></i><b>24</b> no HVF calc since there are not that many different gene species</a></li>
<li class="chapter" data-level="25" data-path="set-seed-for-reproducibility.html"><a href="set-seed-for-reproducibility.html"><i class="fa fa-check"></i><b>25</b> set seed for reproducibility</a></li>
<li class="chapter" data-level="26" data-path="dimplot2d.html"><a href="dimplot2d.html"><i class="fa fa-check"></i><b>26</b> dimPlot2D(</a></li>
<li class="chapter" data-level="27" data-path="g_mini.html"><a href="g_mini.html"><i class="fa fa-check"></i><b>27</b> g_mini,</a></li>
<li class="chapter" data-level="28" data-path="spat_unit-cell.html"><a href="spat_unit-cell.html"><i class="fa fa-check"></i><b>28</b> spat_unit = “cell”,</a></li>
<li class="chapter" data-level="29" data-path="feat_type-rna.html"><a href="feat_type-rna.html"><i class="fa fa-check"></i><b>29</b> feat_type = “rna”,</a></li>
<li class="chapter" data-level="30" data-path="cell_color-leiden_clus.html"><a href="cell_color-leiden_clus.html"><i class="fa fa-check"></i><b>30</b> cell_color = “leiden_clus”,</a></li>
<li class="chapter" data-level="31" data-path="dim_reduction_name-umap.html"><a href="dim_reduction_name-umap.html"><i class="fa fa-check"></i><b>31</b> dim_reduction_name = “umap.projection”,</a></li>
<li class="chapter" data-level="32" data-path="dim_reduction_to_use-umap.html"><a href="dim_reduction_to_use-umap.html"><i class="fa fa-check"></i><b>32</b> dim_reduction_to_use = “umap”,</a></li>
<li class="chapter" data-level="33" data-path="save_param-list.html"><a href="save_param-list.html"><i class="fa fa-check"></i><b>33</b> save_param = list(</a></li>
<li class="chapter" data-level="34" data-path="save_name-mini_umap_1.html"><a href="save_name-mini_umap_1.html"><i class="fa fa-check"></i><b>34</b> save_name = “mini_umap_1.2”</a></li>
<li class="chapter" data-level="35" data-path="section-1.html"><a href="section-1.html"><i class="fa fa-check"></i><b>35</b> )</a></li>
<li class="chapter" data-level="36" data-path="section-2.html"><a href="section-2.html"><i class="fa fa-check"></i><b>36</b> )</a></li>
<li class="chapter" data-level="37" data-path="seems-like-a-good-res.html"><a href="seems-like-a-good-res.html"><i class="fa fa-check"></i><b>37</b> 1.4 seems like a good res</a></li>
<li class="chapter" data-level="38" data-path="project-values-back-via-knn-classifier-fnnknn.html"><a href="project-values-back-via-knn-classifier-fnnknn.html"><i class="fa fa-check"></i><b>38</b> project values back via kNN classifier (FNN::knn)</a></li>
<li class="chapter" data-level="39" data-path="spatplot2dg-cell_color-major_clus.html"><a href="spatplot2dg-cell_color-major_clus.html"><i class="fa fa-check"></i><b>39</b> spatPlot2D(g, cell_color = “major_clus”,</a></li>
<li class="chapter" data-level="40" data-path="point_size-0.html"><a href="point_size-0.html"><i class="fa fa-check"></i><b>40</b> point_size = 0.1,</a></li>
<li class="chapter" data-level="41" data-path="point_shape-no_border.html"><a href="point_shape-no_border.html"><i class="fa fa-check"></i><b>41</b> point_shape = “no_border”,</a></li>
<li class="chapter" data-level="42" data-path="cell_color_code-clus_colorc1-1017-29c1359-14-17-11.html"><a href="cell_color_code-clus_colorc1-1017-29c1359-14-17-11.html"><i class="fa fa-check"></i><b>42</b> cell_color_code = clus_color[c(1, 10:17, 2:9)][c(1:3,5:9, 14, 17, 11)],</a></li>
<li class="chapter" data-level="43" data-path="background-black.html"><a href="background-black.html"><i class="fa fa-check"></i><b>43</b> background = “black”,</a></li>
<li class="chapter" data-level="44" data-path="save_param-list-1.html"><a href="save_param-list-1.html"><i class="fa fa-check"></i><b>44</b> save_param = list(</a></li>
<li class="chapter" data-level="45" data-path="base_width-15.html"><a href="base_width-15.html"><i class="fa fa-check"></i><b>45</b> base_width = 15,</a></li>
<li class="chapter" data-level="46" data-path="base_height-6.html"><a href="base_height-6.html"><i class="fa fa-check"></i><b>46</b> base_height = 6,</a></li>
<li class="chapter" data-level="47" data-path="save_name-spat_major.html"><a href="save_name-spat_major.html"><i class="fa fa-check"></i><b>47</b> save_name = “spat_major”</a></li>
<li class="chapter" data-level="48" data-path="section-3.html"><a href="section-3.html"><i class="fa fa-check"></i><b>48</b> ))</a></li>
<li class="chapter" data-level="49" data-path="niches.html"><a href="niches.html"><i class="fa fa-check"></i><b>49</b> niches</a></li>
<li class="chapter" data-level="50" data-path="visualize-niche-level-enrichment-for-leiden-cluster-3-in-spat_unit-cell.html"><a href="visualize-niche-level-enrichment-for-leiden-cluster-3-in-spat_unit-cell.html"><i class="fa fa-check"></i><b>50</b> visualize niche-level enrichment for leiden cluster 3 in spat_unit “cell”</a></li>
<li class="chapter" data-level="51" data-path="these-enrichments-are-essentially-a-measure-of-how-many-cells-of-each.html"><a href="these-enrichments-are-essentially-a-measure-of-how-many-cells-of-each.html"><i class="fa fa-check"></i><b>51</b> These enrichments are essentially a measure of how many cells of each</a></li>
<li class="chapter" data-level="52" data-path="leiden-cluster-exist-in-the-local-region.html"><a href="leiden-cluster-exist-in-the-local-region.html"><i class="fa fa-check"></i><b>52</b> leiden cluster exist in the local region</a></li>
<li class="chapter" data-level="53" data-path="section-4.html"><a href="section-4.html"><i class="fa fa-check"></i><b>53</b> </a></li>
<li class="chapter" data-level="54" data-path="using-kmeans-we-can-classify-each-cell-by-its-niche-leiden-cluster-proportions.html"><a href="using-kmeans-we-can-classify-each-cell-by-its-niche-leiden-cluster-proportions.html"><i class="fa fa-check"></i><b>54</b> Using kmeans, we can classify each cell by its niche leiden cluster proportions</a></li>
<li class="chapter" data-level="55" data-path="add-kmeans-clustering-of-niches-to-cell-metadata.html"><a href="add-kmeans-clustering-of-niches-to-cell-metadata.html"><i class="fa fa-check"></i><b>55</b> add kmeans clustering of niches to cell metadata</a></li>
<li class="chapter" data-level="56" data-path="spatially-visualize-the-niches.html"><a href="spatially-visualize-the-niches.html"><i class="fa fa-check"></i><b>56</b> Spatially visualize the niches</a></li>
<li class="chapter" data-level="57" data-path="gene-enrichment.html"><a href="gene-enrichment.html"><i class="fa fa-check"></i><b>57</b> Gene enrichment</a></li>
<li class="chapter" data-level="58" data-path="violinplot.html"><a href="violinplot.html"><i class="fa fa-check"></i><b>58</b> violinplot</a></li>
<li class="chapter" data-level="59" data-path="spatial-proteomics-multiplex-if.html"><a href="spatial-proteomics-multiplex-if.html"><i class="fa fa-check"></i><b>59</b> Spatial proteomics: multiplex IF</a>
<ul>
<li class="chapter" data-level="59.1" data-path="spatial-proteomics-multiplex-if.html"><a href="spatial-proteomics-multiplex-if.html#read-in-raw-data-2"><i class="fa fa-check"></i><b>59.1</b> Read in raw data</a></li>
<li class="chapter" data-level="59.2" data-path="spatial-proteomics-multiplex-if.html"><a href="spatial-proteomics-multiplex-if.html#overlap-intensity-workflows"><i class="fa fa-check"></i><b>59.2</b> Overlap intensity &amp; workflows</a></li>
</ul></li>
<li class="chapter" data-level="60" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html"><i class="fa fa-check"></i><b>60</b> Working with multiple samples</a>
<ul>
<li class="chapter" data-level="60.1" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#create-individual-giotto-objects"><i class="fa fa-check"></i><b>60.1</b> Create individual giotto objects</a></li>
<li class="chapter" data-level="60.2" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#join-giotto-objects"><i class="fa fa-check"></i><b>60.2</b> Join Giotto Objects</a></li>
<li class="chapter" data-level="60.3" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#perform-harmony-and-default-workflows"><i class="fa fa-check"></i><b>60.3</b> Perform Harmony and default workflows</a></li>
<li class="chapter" data-level="60.4" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#visualizations-1"><i class="fa fa-check"></i><b>60.4</b> Visualizations</a></li>
</ul></li>
<li class="chapter" data-level="61" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html"><i class="fa fa-check"></i><b>61</b> Spatial multi-modal analysis</a>
<ul>
<li class="chapter" data-level="61.1" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#spatial-transforms"><i class="fa fa-check"></i><b>61.1</b> Spatial transforms</a>
<ul>
<li class="chapter" data-level="61.1.1" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#spatial-transforms-1"><i class="fa fa-check"></i><b>61.1.1</b> Spatial transforms:</a></li>
<li class="chapter" data-level="61.1.2" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#spatial-utilities"><i class="fa fa-check"></i><b>61.1.2</b> Spatial utilities:</a></li>
<li class="chapter" data-level="61.1.3" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#spatial-classes"><i class="fa fa-check"></i><b>61.1.3</b> Spatial classes:</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="62" data-path="image-transforms.html"><a href="image-transforms.html"><i class="fa fa-check"></i><b>62</b> Image transforms</a>
<ul>
<li class="chapter" data-level="62.1" data-path="image-transforms.html"><a href="image-transforms.html#co-registration-of-datasets"><i class="fa fa-check"></i><b>62.1</b> Co-registration of datasets</a></li>
<li class="chapter" data-level="62.2" data-path="image-transforms.html"><a href="image-transforms.html#examples-in-giotto-suite-manuscript"><i class="fa fa-check"></i><b>62.2</b> Examples in giotto suite manuscript</a></li>
</ul></li>
<li class="chapter" data-level="63" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html"><i class="fa fa-check"></i><b>63</b> Multi-omics integration</a>
<ul>
<li class="chapter" data-level="63.1" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#introduction-to-the-spatial-dataset-1"><i class="fa fa-check"></i><b>63.1</b> Introduction to the spatial dataset</a></li>
<li class="chapter" data-level="63.2" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#download-dataset-1"><i class="fa fa-check"></i><b>63.2</b> Download dataset</a></li>
<li class="chapter" data-level="63.3" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#create-the-giotto-object-1"><i class="fa fa-check"></i><b>63.3</b> Create the Giotto object</a></li>
<li class="chapter" data-level="63.4" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#subset-on-spots-that-were-covered-by-tissue-1"><i class="fa fa-check"></i><b>63.4</b> Subset on spots that were covered by tissue</a></li>
<li class="chapter" data-level="63.5" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#rna-processing"><i class="fa fa-check"></i><b>63.5</b> RNA processing</a></li>
<li class="chapter" data-level="63.6" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#protein-processing"><i class="fa fa-check"></i><b>63.6</b> Protein processing</a></li>
<li class="chapter" data-level="63.7" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#multi-omics-integration-1"><i class="fa fa-check"></i><b>63.7</b> Multi-omics integration</a></li>
<li class="chapter" data-level="63.8" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#session-info-2"><i class="fa fa-check"></i><b>63.8</b> Session info</a></li>
</ul></li>
<li class="chapter" data-level="64" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html"><i class="fa fa-check"></i><b>64</b> Interoperability with other frameworks</a>
<ul>
<li class="chapter" data-level="64.1" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#load-giotto-object"><i class="fa fa-check"></i><b>64.1</b> Load Giotto object</a></li>
<li class="chapter" data-level="64.2" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#spatialexperiment"><i class="fa fa-check"></i><b>64.2</b> SpatialExperiment</a>
<ul>
<li class="chapter" data-level="64.2.1" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#convert-giotto-object-to-spatialexperiment-object"><i class="fa fa-check"></i><b>64.2.1</b> Convert Giotto Object to SpatialExperiment Object</a></li>
</ul></li>
<li class="chapter" data-level="64.3" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#identify-top-spatially-variable-genes-with-nnsvg"><i class="fa fa-check"></i><b>64.3</b> Identify top spatially variable genes with nnSVG</a>
<ul>
<li class="chapter" data-level="64.3.1" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#conversion-of-spatialexperiment-object-back-to-giotto"><i class="fa fa-check"></i><b>64.3.1</b> Conversion of SpatialExperiment object back to Giotto</a></li>
<li class="chapter" data-level="64.3.2" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#plotting-top-genes-from-nnsvg-in-giotto"><i class="fa fa-check"></i><b>64.3.2</b> Plotting top genes from nnSVG in Giotto</a></li>
</ul></li>
<li class="chapter" data-level="64.4" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#seurat"><i class="fa fa-check"></i><b>64.4</b> Seurat</a>
<ul>
<li class="chapter" data-level="64.4.1" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#giotto-to-seurat"><i class="fa fa-check"></i><b>64.4.1</b> Giotto to Seurat</a></li>
</ul></li>
<li class="chapter" data-level="64.5" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#anndataspatialdata"><i class="fa fa-check"></i><b>64.5</b> AnnData/SpatialData</a></li>
</ul></li>
<li class="chapter" data-level="65" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html"><i class="fa fa-check"></i><b>65</b> Interoperability with isolated tools</a>
<ul>
<li class="chapter" data-level="65.1" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#spatial-niche-trajectory-analysis"><i class="fa fa-check"></i><b>65.1</b> Spatial niche trajectory analysis</a>
<ul>
<li class="chapter" data-level="65.1.1" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#prepare"><i class="fa fa-check"></i><b>65.1.1</b> Prepare</a></li>
<li class="chapter" data-level="65.1.2" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#running"><i class="fa fa-check"></i><b>65.1.2</b> Running</a></li>
<li class="chapter" data-level="65.1.3" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#visualization"><i class="fa fa-check"></i><b>65.1.3</b> Visualization</a></li>
<li class="chapter" data-level="65.1.4" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#section-5"><i class="fa fa-check"></i><b>65.1.4</b> </a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="66" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html"><i class="fa fa-check"></i><b>66</b> Interactivity with the R/Spatial ecosystem</a>
<ul>
<li class="chapter" data-level="66.1" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#kriging"><i class="fa fa-check"></i><b>66.1</b> Kriging</a></li>
</ul></li>
<li class="chapter" data-level="67" data-path="contributing-to-giotto.html"><a href="contributing-to-giotto.html"><i class="fa fa-check"></i><b>67</b> Contributing to Giotto</a>
<ul>
<li class="chapter" data-level="67.1" data-path="contributing-to-giotto.html"><a href="contributing-to-giotto.html#contribution-guideline"><i class="fa fa-check"></i><b>67.1</b> Contribution guideline</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Workshop: Spatial multi-omics data analysis with Giotto Suite</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="visium-part-ii" class="section level1 hasAnchor" number="7">
<h1><span class="header-section-number">7</span> Visium Part II<a href="visium-part-ii.html#visium-part-ii" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Joselyn Cristina Chávez Fuentes</p>
<p>August 6th 2024</p>
<div id="load-the-object" class="section level2 hasAnchor" number="7.1">
<h2><span class="header-section-number">7.1</span> Load the object<a href="visium-part-ii.html#load-the-object" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb46-1"><a href="visium-part-ii.html#cb46-1" tabindex="-1"></a><span class="fu">library</span>(Giotto)</span>
<span id="cb46-2"><a href="visium-part-ii.html#cb46-2" tabindex="-1"></a></span>
<span id="cb46-3"><a href="visium-part-ii.html#cb46-3" tabindex="-1"></a>visium_brain <span class="ot">&lt;-</span> <span class="fu">loadGiotto</span>(<span class="st">&quot;visium_brain_object&quot;</span>)</span></code></pre></div>
</div>
<div id="differential-expression" class="section level2 hasAnchor" number="7.2">
<h2><span class="header-section-number">7.2</span> Differential expression<a href="visium-part-ii.html#differential-expression" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="gini-markers" class="section level3 hasAnchor" number="7.2.1">
<h3><span class="header-section-number">7.2.1</span> Gini markers<a href="visium-part-ii.html#gini-markers" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li>Calculate the top marker genes per cluster using the gini method</li>
</ul>
<div class="sourceCode" id="cb47"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb47-1"><a href="visium-part-ii.html#cb47-1" tabindex="-1"></a>gini_markers <span class="ot">&lt;-</span> <span class="fu">findMarkers_one_vs_all</span>(<span class="at">gobject =</span> visium_brain,</span>
<span id="cb47-2"><a href="visium-part-ii.html#cb47-2" tabindex="-1"></a>                                       <span class="at">method =</span> <span class="st">&quot;gini&quot;</span>,</span>
<span id="cb47-3"><a href="visium-part-ii.html#cb47-3" tabindex="-1"></a>                                       <span class="at">expression_values =</span> <span class="st">&quot;normalized&quot;</span>,</span>
<span id="cb47-4"><a href="visium-part-ii.html#cb47-4" tabindex="-1"></a>                                       <span class="at">cluster_column =</span> <span class="st">&quot;leiden_clus&quot;</span>,</span>
<span id="cb47-5"><a href="visium-part-ii.html#cb47-5" tabindex="-1"></a>                                       <span class="at">min_feats =</span> <span class="dv">10</span>)</span>
<span id="cb47-6"><a href="visium-part-ii.html#cb47-6" tabindex="-1"></a></span>
<span id="cb47-7"><a href="visium-part-ii.html#cb47-7" tabindex="-1"></a>topgenes_gini <span class="ot">&lt;-</span> gini_markers[, <span class="fu">head</span>(.SD, <span class="dv">2</span>), by <span class="ot">=</span> <span class="st">&quot;cluster&quot;</span>]<span class="sc">$</span>feats</span></code></pre></div>
<ul>
<li>Visualize</li>
</ul>
<div class="sourceCode" id="cb48"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb48-1"><a href="visium-part-ii.html#cb48-1" tabindex="-1"></a><span class="fu">violinPlot</span>(visium_brain, </span>
<span id="cb48-2"><a href="visium-part-ii.html#cb48-2" tabindex="-1"></a>           <span class="at">feats =</span> <span class="fu">unique</span>(topgenes_gini), </span>
<span id="cb48-3"><a href="visium-part-ii.html#cb48-3" tabindex="-1"></a>           <span class="at">cluster_column =</span> <span class="st">&quot;leiden_clus&quot;</span>,</span>
<span id="cb48-4"><a href="visium-part-ii.html#cb48-4" tabindex="-1"></a>           <span class="at">strip_text =</span> <span class="dv">6</span>, </span>
<span id="cb48-5"><a href="visium-part-ii.html#cb48-5" tabindex="-1"></a>           <span class="at">strip_position =</span> <span class="st">&quot;right&quot;</span>,</span>
<span id="cb48-6"><a href="visium-part-ii.html#cb48-6" tabindex="-1"></a>           <span class="at">save_param =</span> <span class="fu">list</span>(<span class="at">base_width =</span> <span class="dv">5</span>, <span class="at">base_height =</span> <span class="dv">30</span>))</span></code></pre></div>
<p><img src="img/02_session1/0-violinPlot.png" width="50%" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb49"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb49-1"><a href="visium-part-ii.html#cb49-1" tabindex="-1"></a><span class="fu">plotMetaDataHeatmap</span>(visium_brain, </span>
<span id="cb49-2"><a href="visium-part-ii.html#cb49-2" tabindex="-1"></a>                    <span class="at">selected_feats =</span> <span class="fu">unique</span>(topgenes_gini),</span>
<span id="cb49-3"><a href="visium-part-ii.html#cb49-3" tabindex="-1"></a>                    <span class="at">metadata_cols =</span> <span class="st">&quot;leiden_clus&quot;</span>,</span>
<span id="cb49-4"><a href="visium-part-ii.html#cb49-4" tabindex="-1"></a>                    <span class="at">x_text_size =</span> <span class="dv">10</span>, <span class="at">y_text_size =</span> <span class="dv">10</span>)</span></code></pre></div>
<p><img src="img/02_session1/1-plotMetaDataHeatmap.png" width="80%" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb50"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb50-1"><a href="visium-part-ii.html#cb50-1" tabindex="-1"></a><span class="fu">dimFeatPlot2D</span>(visium_brain, </span>
<span id="cb50-2"><a href="visium-part-ii.html#cb50-2" tabindex="-1"></a>              <span class="at">expression_values =</span> <span class="st">&quot;scaled&quot;</span>,</span>
<span id="cb50-3"><a href="visium-part-ii.html#cb50-3" tabindex="-1"></a>              <span class="at">feats =</span> <span class="fu">sort</span>(<span class="fu">unique</span>(topgenes_gini)),</span>
<span id="cb50-4"><a href="visium-part-ii.html#cb50-4" tabindex="-1"></a>              <span class="at">cow_n_col =</span> <span class="dv">5</span>, </span>
<span id="cb50-5"><a href="visium-part-ii.html#cb50-5" tabindex="-1"></a>              <span class="at">point_size =</span> <span class="dv">1</span>,</span>
<span id="cb50-6"><a href="visium-part-ii.html#cb50-6" tabindex="-1"></a>              <span class="at">save_param =</span> <span class="fu">list</span>(<span class="at">base_width =</span> <span class="dv">15</span>, <span class="at">base_height =</span> <span class="dv">20</span>))</span></code></pre></div>
<p><img src="img/02_session1/2-dimFeatPlot2D.png" width="100%" style="display: block; margin: auto;" /></p>
</div>
<div id="scran-markers" class="section level3 hasAnchor" number="7.2.2">
<h3><span class="header-section-number">7.2.2</span> Scran markers<a href="visium-part-ii.html#scran-markers" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li>Calculate the top marker genes per cluster using the scran method</li>
</ul>
<div class="sourceCode" id="cb51"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb51-1"><a href="visium-part-ii.html#cb51-1" tabindex="-1"></a>scran_markers <span class="ot">&lt;-</span> <span class="fu">findMarkers_one_vs_all</span>(<span class="at">gobject =</span> visium_brain,</span>
<span id="cb51-2"><a href="visium-part-ii.html#cb51-2" tabindex="-1"></a>                                       <span class="at">method =</span> <span class="st">&quot;scran&quot;</span>,</span>
<span id="cb51-3"><a href="visium-part-ii.html#cb51-3" tabindex="-1"></a>                                       <span class="at">expression_values =</span> <span class="st">&quot;normalized&quot;</span>,</span>
<span id="cb51-4"><a href="visium-part-ii.html#cb51-4" tabindex="-1"></a>                                       <span class="at">cluster_column =</span> <span class="st">&quot;leiden_clus&quot;</span>,</span>
<span id="cb51-5"><a href="visium-part-ii.html#cb51-5" tabindex="-1"></a>                                       <span class="at">min_feats =</span> <span class="dv">10</span>)</span>
<span id="cb51-6"><a href="visium-part-ii.html#cb51-6" tabindex="-1"></a></span>
<span id="cb51-7"><a href="visium-part-ii.html#cb51-7" tabindex="-1"></a>topgenes_scran <span class="ot">&lt;-</span> scran_markers[, <span class="fu">head</span>(.SD, <span class="dv">2</span>), by <span class="ot">=</span> <span class="st">&quot;cluster&quot;</span>]<span class="sc">$</span>feats</span></code></pre></div>
<ul>
<li>Visualize</li>
</ul>
<div class="sourceCode" id="cb52"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb52-1"><a href="visium-part-ii.html#cb52-1" tabindex="-1"></a><span class="fu">violinPlot</span>(visium_brain, </span>
<span id="cb52-2"><a href="visium-part-ii.html#cb52-2" tabindex="-1"></a>           <span class="at">feats =</span> <span class="fu">unique</span>(topgenes_scran), </span>
<span id="cb52-3"><a href="visium-part-ii.html#cb52-3" tabindex="-1"></a>           <span class="at">cluster_column =</span> <span class="st">&quot;leiden_clus&quot;</span>,</span>
<span id="cb52-4"><a href="visium-part-ii.html#cb52-4" tabindex="-1"></a>           <span class="at">strip_text =</span> <span class="dv">6</span>, </span>
<span id="cb52-5"><a href="visium-part-ii.html#cb52-5" tabindex="-1"></a>           <span class="at">strip_position =</span> <span class="st">&quot;right&quot;</span>,</span>
<span id="cb52-6"><a href="visium-part-ii.html#cb52-6" tabindex="-1"></a>           <span class="at">save_param =</span> <span class="fu">list</span>(<span class="at">base_width =</span> <span class="dv">5</span>, <span class="at">base_height =</span> <span class="dv">30</span>))</span></code></pre></div>
<p><img src="img/02_session1/3-violinPlot.png" width="50%" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb53"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb53-1"><a href="visium-part-ii.html#cb53-1" tabindex="-1"></a><span class="fu">plotMetaDataHeatmap</span>(visium_brain, </span>
<span id="cb53-2"><a href="visium-part-ii.html#cb53-2" tabindex="-1"></a>                    <span class="at">selected_feats =</span> <span class="fu">unique</span>(topgenes_scran),</span>
<span id="cb53-3"><a href="visium-part-ii.html#cb53-3" tabindex="-1"></a>                    <span class="at">metadata_cols =</span> <span class="st">&quot;leiden_clus&quot;</span>,</span>
<span id="cb53-4"><a href="visium-part-ii.html#cb53-4" tabindex="-1"></a>                    <span class="at">x_text_size =</span> <span class="dv">10</span>, <span class="at">y_text_size =</span> <span class="dv">10</span>)</span></code></pre></div>
<p><img src="img/02_session1/4-plotMetaDataHeatmap.png" width="100%" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb54"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb54-1"><a href="visium-part-ii.html#cb54-1" tabindex="-1"></a><span class="fu">dimFeatPlot2D</span>(visium_brain, </span>
<span id="cb54-2"><a href="visium-part-ii.html#cb54-2" tabindex="-1"></a>              <span class="at">expression_values =</span> <span class="st">&quot;scaled&quot;</span>,</span>
<span id="cb54-3"><a href="visium-part-ii.html#cb54-3" tabindex="-1"></a>              <span class="at">feats =</span> <span class="fu">sort</span>(<span class="fu">unique</span>(topgenes_scran)),</span>
<span id="cb54-4"><a href="visium-part-ii.html#cb54-4" tabindex="-1"></a>              <span class="at">cow_n_col =</span> <span class="dv">5</span>, </span>
<span id="cb54-5"><a href="visium-part-ii.html#cb54-5" tabindex="-1"></a>              <span class="at">point_size =</span> <span class="dv">1</span>,</span>
<span id="cb54-6"><a href="visium-part-ii.html#cb54-6" tabindex="-1"></a>              <span class="at">save_param =</span> <span class="fu">list</span>(<span class="at">base_width =</span> <span class="dv">20</span>, <span class="at">base_height =</span> <span class="dv">20</span>))</span></code></pre></div>
<p><img src="img/02_session1/5-dimFeatPlot2D.png" width="100%" style="display: block; margin: auto;" /></p>
<p>The Gini method is preferred when identifying genes that exhibit significant expression differences across clusters, highlighting genes with distinct expression patters between different clusters.</p>
<p>The Scran method is preferred for robust differential expression analysis, especially when addressing technical variability or differences in sequencing depth across spatial locations.</p>
<p>In practice, it is often beneficial to apply both methods and compare results for a more complete understanding of differential gene expression across clusters.</p>
</div>
</div>
<div id="enrichment-deconvolution" class="section level2 hasAnchor" number="7.3">
<h2><span class="header-section-number">7.3</span> Enrichment &amp; Deconvolution<a href="visium-part-ii.html#enrichment-deconvolution" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Visium spatial transcriptomics does not provide single-cell resolution, making cell type annotation a harder problem. Giotto provides several ways to calculate enrichment of specific cell-type signature gene lists.</p>
<ul>
<li>Download the single-cell dataset</li>
</ul>
<div class="sourceCode" id="cb55"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb55-1"><a href="visium-part-ii.html#cb55-1" tabindex="-1"></a>GiottoData<span class="sc">::</span><span class="fu">getSpatialDataset</span>(<span class="at">dataset =</span> <span class="st">&quot;scRNA_mouse_brain&quot;</span>, <span class="at">directory =</span> <span class="st">&quot;data/&quot;</span>)</span></code></pre></div>
<ul>
<li>Create the single-cell object</li>
</ul>
<div class="sourceCode" id="cb56"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb56-1"><a href="visium-part-ii.html#cb56-1" tabindex="-1"></a>results_folder <span class="ot">&lt;-</span> <span class="st">&quot;results/&quot;</span></span>
<span id="cb56-2"><a href="visium-part-ii.html#cb56-2" tabindex="-1"></a></span>
<span id="cb56-3"><a href="visium-part-ii.html#cb56-3" tabindex="-1"></a>python_path <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb56-4"><a href="visium-part-ii.html#cb56-4" tabindex="-1"></a></span>
<span id="cb56-5"><a href="visium-part-ii.html#cb56-5" tabindex="-1"></a>instructions <span class="ot">&lt;-</span> <span class="fu">createGiottoInstructions</span>(</span>
<span id="cb56-6"><a href="visium-part-ii.html#cb56-6" tabindex="-1"></a>    <span class="at">save_dir =</span> results_folder,</span>
<span id="cb56-7"><a href="visium-part-ii.html#cb56-7" tabindex="-1"></a>    <span class="at">save_plot =</span> <span class="cn">TRUE</span>,</span>
<span id="cb56-8"><a href="visium-part-ii.html#cb56-8" tabindex="-1"></a>    <span class="at">show_plot =</span> <span class="cn">FALSE</span>,</span>
<span id="cb56-9"><a href="visium-part-ii.html#cb56-9" tabindex="-1"></a>    <span class="at">python_path =</span> python_path</span>
<span id="cb56-10"><a href="visium-part-ii.html#cb56-10" tabindex="-1"></a>)</span>
<span id="cb56-11"><a href="visium-part-ii.html#cb56-11" tabindex="-1"></a></span>
<span id="cb56-12"><a href="visium-part-ii.html#cb56-12" tabindex="-1"></a>sc_expression <span class="ot">&lt;-</span> <span class="st">&quot;data/brain_sc_expression_matrix.txt.gz&quot;</span></span>
<span id="cb56-13"><a href="visium-part-ii.html#cb56-13" tabindex="-1"></a>sc_metadata <span class="ot">&lt;-</span> <span class="st">&quot;data/brain_sc_metadata.csv&quot;</span></span>
<span id="cb56-14"><a href="visium-part-ii.html#cb56-14" tabindex="-1"></a></span>
<span id="cb56-15"><a href="visium-part-ii.html#cb56-15" tabindex="-1"></a>giotto_SC <span class="ot">&lt;-</span> <span class="fu">createGiottoObject</span>(<span class="at">expression =</span> sc_expression,</span>
<span id="cb56-16"><a href="visium-part-ii.html#cb56-16" tabindex="-1"></a>                                <span class="at">instructions =</span> instructions)</span>
<span id="cb56-17"><a href="visium-part-ii.html#cb56-17" tabindex="-1"></a></span>
<span id="cb56-18"><a href="visium-part-ii.html#cb56-18" tabindex="-1"></a>giotto_SC <span class="ot">&lt;-</span> <span class="fu">addCellMetadata</span>(giotto_SC, </span>
<span id="cb56-19"><a href="visium-part-ii.html#cb56-19" tabindex="-1"></a>                             <span class="at">new_metadata =</span> data.table<span class="sc">::</span><span class="fu">fread</span>(sc_metadata))</span>
<span id="cb56-20"><a href="visium-part-ii.html#cb56-20" tabindex="-1"></a></span>
<span id="cb56-21"><a href="visium-part-ii.html#cb56-21" tabindex="-1"></a>giotto_SC <span class="ot">&lt;-</span> <span class="fu">normalizeGiotto</span>(giotto_SC)</span></code></pre></div>
<div id="pagerank" class="section level3 hasAnchor" number="7.3.1">
<h3><span class="header-section-number">7.3.1</span> PAGE/Rank<a href="visium-part-ii.html#pagerank" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Parametric Analysis of Gene Set Enrichment (PAGE) and Rank enrichment both aim to determine whether a predefined set of genes show statistically significant differences in expression compared to other genes in the dataset.</p>
<ul>
<li>Calculate the cell type markers</li>
</ul>
<div class="sourceCode" id="cb57"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb57-1"><a href="visium-part-ii.html#cb57-1" tabindex="-1"></a>markers_scran <span class="ot">&lt;-</span> <span class="fu">findMarkers_one_vs_all</span>(<span class="at">gobject =</span> giotto_SC, </span>
<span id="cb57-2"><a href="visium-part-ii.html#cb57-2" tabindex="-1"></a>                                        <span class="at">method =</span> <span class="st">&quot;scran&quot;</span>,</span>
<span id="cb57-3"><a href="visium-part-ii.html#cb57-3" tabindex="-1"></a>                                        <span class="at">expression_values =</span> <span class="st">&quot;normalized&quot;</span>,</span>
<span id="cb57-4"><a href="visium-part-ii.html#cb57-4" tabindex="-1"></a>                                        <span class="at">cluster_column =</span> <span class="st">&quot;Class&quot;</span>, </span>
<span id="cb57-5"><a href="visium-part-ii.html#cb57-5" tabindex="-1"></a>                                        <span class="at">min_feats =</span> <span class="dv">3</span>)</span>
<span id="cb57-6"><a href="visium-part-ii.html#cb57-6" tabindex="-1"></a></span>
<span id="cb57-7"><a href="visium-part-ii.html#cb57-7" tabindex="-1"></a>top_markers <span class="ot">&lt;-</span> markers_scran[, <span class="fu">head</span>(.SD, <span class="dv">10</span>), by <span class="ot">=</span> <span class="st">&quot;cluster&quot;</span>]</span>
<span id="cb57-8"><a href="visium-part-ii.html#cb57-8" tabindex="-1"></a>celltypes <span class="ot">&lt;-</span> <span class="fu">levels</span>(<span class="fu">factor</span>(markers_scran<span class="sc">$</span>cluster)) </span></code></pre></div>
<ul>
<li>Create the signature matrix</li>
</ul>
<div class="sourceCode" id="cb58"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb58-1"><a href="visium-part-ii.html#cb58-1" tabindex="-1"></a>sign_list <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb58-2"><a href="visium-part-ii.html#cb58-2" tabindex="-1"></a></span>
<span id="cb58-3"><a href="visium-part-ii.html#cb58-3" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(celltypes)){</span>
<span id="cb58-4"><a href="visium-part-ii.html#cb58-4" tabindex="-1"></a>  sign_list[[i]] <span class="ot">=</span> top_markers[<span class="fu">which</span>(top_markers<span class="sc">$</span>cluster <span class="sc">==</span> celltypes[i]),]<span class="sc">$</span>feats</span>
<span id="cb58-5"><a href="visium-part-ii.html#cb58-5" tabindex="-1"></a>}</span>
<span id="cb58-6"><a href="visium-part-ii.html#cb58-6" tabindex="-1"></a></span>
<span id="cb58-7"><a href="visium-part-ii.html#cb58-7" tabindex="-1"></a>sign_matrix <span class="ot">&lt;-</span> <span class="fu">makeSignMatrixPAGE</span>(<span class="at">sign_names =</span> celltypes,</span>
<span id="cb58-8"><a href="visium-part-ii.html#cb58-8" tabindex="-1"></a>                                  <span class="at">sign_list =</span> sign_list)</span></code></pre></div>
<ul>
<li>Run the enrichment test with PAGE</li>
</ul>
<div class="sourceCode" id="cb59"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb59-1"><a href="visium-part-ii.html#cb59-1" tabindex="-1"></a>visium_brain <span class="ot">&lt;-</span> <span class="fu">runPAGEEnrich</span>(<span class="at">gobject =</span> visium_brain, </span>
<span id="cb59-2"><a href="visium-part-ii.html#cb59-2" tabindex="-1"></a>                              <span class="at">sign_matrix =</span> sign_matrix)</span></code></pre></div>
<ul>
<li>Visualize</li>
</ul>
<div class="sourceCode" id="cb60"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb60-1"><a href="visium-part-ii.html#cb60-1" tabindex="-1"></a>cell_types_PAGE <span class="ot">&lt;-</span> <span class="fu">colnames</span>(sign_matrix)</span>
<span id="cb60-2"><a href="visium-part-ii.html#cb60-2" tabindex="-1"></a></span>
<span id="cb60-3"><a href="visium-part-ii.html#cb60-3" tabindex="-1"></a><span class="fu">plotMetaDataCellsHeatmap</span>(<span class="at">gobject =</span> visium_brain,</span>
<span id="cb60-4"><a href="visium-part-ii.html#cb60-4" tabindex="-1"></a>                         <span class="at">metadata_cols =</span> <span class="st">&quot;leiden_clus&quot;</span>,</span>
<span id="cb60-5"><a href="visium-part-ii.html#cb60-5" tabindex="-1"></a>                         <span class="at">value_cols =</span> cell_types_PAGE,</span>
<span id="cb60-6"><a href="visium-part-ii.html#cb60-6" tabindex="-1"></a>                         <span class="at">spat_enr_names =</span> <span class="st">&quot;PAGE&quot;</span>,</span>
<span id="cb60-7"><a href="visium-part-ii.html#cb60-7" tabindex="-1"></a>                         <span class="at">x_text_size =</span> <span class="dv">8</span>,</span>
<span id="cb60-8"><a href="visium-part-ii.html#cb60-8" tabindex="-1"></a>                         <span class="at">y_text_size =</span> <span class="dv">8</span>)</span></code></pre></div>
<p><img src="img/02_session1/6-plotMetaDataCellsHeatmap.png" width="100%" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb61"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb61-1"><a href="visium-part-ii.html#cb61-1" tabindex="-1"></a><span class="fu">spatCellPlot2D</span>(<span class="at">gobject =</span> visium_brain,</span>
<span id="cb61-2"><a href="visium-part-ii.html#cb61-2" tabindex="-1"></a>               <span class="at">spat_enr_names =</span> <span class="st">&quot;PAGE&quot;</span>,</span>
<span id="cb61-3"><a href="visium-part-ii.html#cb61-3" tabindex="-1"></a>               <span class="at">cell_annotation_values =</span> cell_types_PAGE,</span>
<span id="cb61-4"><a href="visium-part-ii.html#cb61-4" tabindex="-1"></a>               <span class="at">cow_n_col =</span> <span class="dv">3</span>,</span>
<span id="cb61-5"><a href="visium-part-ii.html#cb61-5" tabindex="-1"></a>               <span class="at">coord_fix_ratio =</span> <span class="dv">1</span>, </span>
<span id="cb61-6"><a href="visium-part-ii.html#cb61-6" tabindex="-1"></a>               <span class="at">point_size =</span> <span class="dv">1</span>, </span>
<span id="cb61-7"><a href="visium-part-ii.html#cb61-7" tabindex="-1"></a>               <span class="at">show_legend =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p><img src="img/02_session1/7-spatCellPlot2D.png" width="100%" style="display: block; margin: auto;" /></p>
</div>
<div id="spatialdwls" class="section level3 hasAnchor" number="7.3.2">
<h3><span class="header-section-number">7.3.2</span> SpatialDWLS<a href="visium-part-ii.html#spatialdwls" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Spatial Diffusion Weighted Least Squares (DWLS) estimates the proportions of different cell types or spatial gradients of gene expression across spots in a tissue.</p>
<ul>
<li>Create the signature matrix</li>
</ul>
<div class="sourceCode" id="cb62"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb62-1"><a href="visium-part-ii.html#cb62-1" tabindex="-1"></a>sign_matrix <span class="ot">&lt;-</span> <span class="fu">makeSignMatrixDWLSfromMatrix</span>(</span>
<span id="cb62-2"><a href="visium-part-ii.html#cb62-2" tabindex="-1"></a>    <span class="at">matrix =</span> <span class="fu">getExpression</span>(giotto_SC,</span>
<span id="cb62-3"><a href="visium-part-ii.html#cb62-3" tabindex="-1"></a>                           <span class="at">values =</span> <span class="st">&quot;normalized&quot;</span>,</span>
<span id="cb62-4"><a href="visium-part-ii.html#cb62-4" tabindex="-1"></a>                           <span class="at">output =</span> <span class="st">&quot;matrix&quot;</span>),</span>
<span id="cb62-5"><a href="visium-part-ii.html#cb62-5" tabindex="-1"></a>    <span class="at">cell_type =</span> <span class="fu">pDataDT</span>(giotto_SC)<span class="sc">$</span>Class,</span>
<span id="cb62-6"><a href="visium-part-ii.html#cb62-6" tabindex="-1"></a>    <span class="at">sign_gene =</span> top_markers<span class="sc">$</span>feats)</span></code></pre></div>
<ul>
<li>Run the DWLS Deconvolution</li>
</ul>
<div class="sourceCode" id="cb63"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb63-1"><a href="visium-part-ii.html#cb63-1" tabindex="-1"></a>visium_brain <span class="ot">&lt;-</span> <span class="fu">runDWLSDeconv</span>(<span class="at">gobject =</span> visium_brain, </span>
<span id="cb63-2"><a href="visium-part-ii.html#cb63-2" tabindex="-1"></a>                              <span class="at">sign_matrix =</span> sign_matrix)</span></code></pre></div>
<ul>
<li>Visualize</li>
</ul>
<div class="sourceCode" id="cb64"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb64-1"><a href="visium-part-ii.html#cb64-1" tabindex="-1"></a><span class="co"># Plot DWLS deconvolution result with Pie plots</span></span>
<span id="cb64-2"><a href="visium-part-ii.html#cb64-2" tabindex="-1"></a><span class="fu">spatDeconvPlot</span>(visium_brain, </span>
<span id="cb64-3"><a href="visium-part-ii.html#cb64-3" tabindex="-1"></a>               <span class="at">show_image =</span> <span class="cn">FALSE</span>,</span>
<span id="cb64-4"><a href="visium-part-ii.html#cb64-4" tabindex="-1"></a>               <span class="at">radius =</span> <span class="dv">50</span>,</span>
<span id="cb64-5"><a href="visium-part-ii.html#cb64-5" tabindex="-1"></a>               <span class="at">save_param =</span> <span class="fu">list</span>(<span class="at">save_name =</span> <span class="st">&quot;8_spat_DWLS_pie_plot&quot;</span>))</span></code></pre></div>
<p><img src="img/02_session1/8_spat_DWLS_pie_plot.png" width="100%" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="spatial-expression-patterns" class="section level2 hasAnchor" number="7.4">
<h2><span class="header-section-number">7.4</span> Spatial expression patterns<a href="visium-part-ii.html#spatial-expression-patterns" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="spatial-variable-genes" class="section level3 hasAnchor" number="7.4.1">
<h3><span class="header-section-number">7.4.1</span> Spatial variable genes<a href="visium-part-ii.html#spatial-variable-genes" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li>Create a spatial network</li>
</ul>
<div class="sourceCode" id="cb65"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb65-1"><a href="visium-part-ii.html#cb65-1" tabindex="-1"></a>visium_brain <span class="ot">&lt;-</span> <span class="fu">createSpatialNetwork</span>(<span class="at">gobject =</span> visium_brain,</span>
<span id="cb65-2"><a href="visium-part-ii.html#cb65-2" tabindex="-1"></a>                                     <span class="at">method =</span> <span class="st">&quot;kNN&quot;</span>, </span>
<span id="cb65-3"><a href="visium-part-ii.html#cb65-3" tabindex="-1"></a>                                     <span class="at">k =</span> <span class="dv">6</span>,</span>
<span id="cb65-4"><a href="visium-part-ii.html#cb65-4" tabindex="-1"></a>                                     <span class="at">maximum_distance_knn =</span> <span class="dv">400</span>,</span>
<span id="cb65-5"><a href="visium-part-ii.html#cb65-5" tabindex="-1"></a>                                     <span class="at">name =</span> <span class="st">&quot;spatial_network&quot;</span>)</span>
<span id="cb65-6"><a href="visium-part-ii.html#cb65-6" tabindex="-1"></a></span>
<span id="cb65-7"><a href="visium-part-ii.html#cb65-7" tabindex="-1"></a><span class="fu">spatPlot2D</span>(<span class="at">gobject =</span> visium_brain,  </span>
<span id="cb65-8"><a href="visium-part-ii.html#cb65-8" tabindex="-1"></a>           <span class="at">show_network=</span> <span class="cn">TRUE</span>,</span>
<span id="cb65-9"><a href="visium-part-ii.html#cb65-9" tabindex="-1"></a>           <span class="at">network_color =</span> <span class="st">&quot;blue&quot;</span>, </span>
<span id="cb65-10"><a href="visium-part-ii.html#cb65-10" tabindex="-1"></a>           <span class="at">spatial_network_name =</span> <span class="st">&quot;spatial_network&quot;</span>)</span></code></pre></div>
<p><img src="img/02_session1/9-spatPlot2D.png" width="100%" style="display: block; margin: auto;" /></p>
<ul>
<li>Rank binarization</li>
</ul>
<div class="sourceCode" id="cb66"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb66-1"><a href="visium-part-ii.html#cb66-1" tabindex="-1"></a>ranktest <span class="ot">&lt;-</span> <span class="fu">binSpect</span>(visium_brain, </span>
<span id="cb66-2"><a href="visium-part-ii.html#cb66-2" tabindex="-1"></a>                     <span class="at">bin_method =</span> <span class="st">&quot;rank&quot;</span>,</span>
<span id="cb66-3"><a href="visium-part-ii.html#cb66-3" tabindex="-1"></a>                     <span class="at">calc_hub =</span> <span class="cn">TRUE</span>, </span>
<span id="cb66-4"><a href="visium-part-ii.html#cb66-4" tabindex="-1"></a>                     <span class="at">hub_min_int =</span> <span class="dv">5</span>,</span>
<span id="cb66-5"><a href="visium-part-ii.html#cb66-5" tabindex="-1"></a>                     <span class="at">spatial_network_name =</span> <span class="st">&quot;spatial_network&quot;</span>)</span></code></pre></div>
<ul>
<li>Visualize top results</li>
</ul>
<div class="sourceCode" id="cb67"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb67-1"><a href="visium-part-ii.html#cb67-1" tabindex="-1"></a><span class="fu">spatFeatPlot2D</span>(visium_brain, </span>
<span id="cb67-2"><a href="visium-part-ii.html#cb67-2" tabindex="-1"></a>               <span class="at">expression_values =</span> <span class="st">&quot;scaled&quot;</span>,</span>
<span id="cb67-3"><a href="visium-part-ii.html#cb67-3" tabindex="-1"></a>               <span class="at">feats =</span> ranktest<span class="sc">$</span>feats[<span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>], </span>
<span id="cb67-4"><a href="visium-part-ii.html#cb67-4" tabindex="-1"></a>               <span class="at">cow_n_col =</span> <span class="dv">2</span>, </span>
<span id="cb67-5"><a href="visium-part-ii.html#cb67-5" tabindex="-1"></a>               <span class="at">point_size =</span> <span class="dv">1</span>)</span></code></pre></div>
<p><img src="img/02_session1/10-spatFeatPlot2D.png" width="100%" style="display: block; margin: auto;" /></p>
</div>
<div id="spatial-co-expression-modules" class="section level3 hasAnchor" number="7.4.2">
<h3><span class="header-section-number">7.4.2</span> Spatial co-expression modules<a href="visium-part-ii.html#spatial-co-expression-modules" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li>Cluster the top 500 spatial genes into 20 clusters</li>
</ul>
<div class="sourceCode" id="cb68"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb68-1"><a href="visium-part-ii.html#cb68-1" tabindex="-1"></a>ext_spatial_genes <span class="ot">&lt;-</span> ranktest[<span class="dv">1</span><span class="sc">:</span><span class="dv">500</span>,]<span class="sc">$</span>feats</span></code></pre></div>
<ul>
<li>Use detectSpatialCorGenes function to calculate pairwise distances between genes.</li>
</ul>
<div class="sourceCode" id="cb69"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb69-1"><a href="visium-part-ii.html#cb69-1" tabindex="-1"></a>spat_cor_netw_DT <span class="ot">&lt;-</span> <span class="fu">detectSpatialCorFeats</span>(</span>
<span id="cb69-2"><a href="visium-part-ii.html#cb69-2" tabindex="-1"></a>    visium_brain,</span>
<span id="cb69-3"><a href="visium-part-ii.html#cb69-3" tabindex="-1"></a>    <span class="at">method =</span> <span class="st">&quot;network&quot;</span>,</span>
<span id="cb69-4"><a href="visium-part-ii.html#cb69-4" tabindex="-1"></a>    <span class="at">spatial_network_name =</span> <span class="st">&quot;spatial_network&quot;</span>,</span>
<span id="cb69-5"><a href="visium-part-ii.html#cb69-5" tabindex="-1"></a>    <span class="at">subset_feats =</span> ext_spatial_genes)</span></code></pre></div>
<ul>
<li>Identify most similar spatially correlated genes for one gene</li>
</ul>
<div class="sourceCode" id="cb70"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb70-1"><a href="visium-part-ii.html#cb70-1" tabindex="-1"></a>top10_genes <span class="ot">&lt;-</span> <span class="fu">showSpatialCorFeats</span>(spat_cor_netw_DT, </span>
<span id="cb70-2"><a href="visium-part-ii.html#cb70-2" tabindex="-1"></a>                                  <span class="at">feats =</span> <span class="st">&quot;Mbp&quot;</span>, </span>
<span id="cb70-3"><a href="visium-part-ii.html#cb70-3" tabindex="-1"></a>                                  <span class="at">show_top_feats =</span> <span class="dv">10</span>)</span></code></pre></div>
<ul>
<li>Visualize</li>
</ul>
<div class="sourceCode" id="cb71"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb71-1"><a href="visium-part-ii.html#cb71-1" tabindex="-1"></a><span class="fu">spatFeatPlot2D</span>(visium_brain, </span>
<span id="cb71-2"><a href="visium-part-ii.html#cb71-2" tabindex="-1"></a>               <span class="at">expression_values =</span> <span class="st">&quot;scaled&quot;</span>,</span>
<span id="cb71-3"><a href="visium-part-ii.html#cb71-3" tabindex="-1"></a>               <span class="at">feats =</span> top10_genes<span class="sc">$</span>variable[<span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>], </span>
<span id="cb71-4"><a href="visium-part-ii.html#cb71-4" tabindex="-1"></a>               <span class="at">point_size =</span> <span class="fl">1.5</span>)</span></code></pre></div>
<p><img src="img/02_session1/11-spatFeatPlot2D.png" width="100%" style="display: block; margin: auto;" /></p>
<ul>
<li>Cluster spatial genes</li>
</ul>
<div class="sourceCode" id="cb72"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb72-1"><a href="visium-part-ii.html#cb72-1" tabindex="-1"></a>spat_cor_netw_DT <span class="ot">&lt;-</span> <span class="fu">clusterSpatialCorFeats</span>(spat_cor_netw_DT, </span>
<span id="cb72-2"><a href="visium-part-ii.html#cb72-2" tabindex="-1"></a>                                           <span class="at">name =</span> <span class="st">&quot;spat_netw_clus&quot;</span>, </span>
<span id="cb72-3"><a href="visium-part-ii.html#cb72-3" tabindex="-1"></a>                                           <span class="at">k =</span> <span class="dv">20</span>)</span></code></pre></div>
<ul>
<li>Visualize clusters</li>
</ul>
<div class="sourceCode" id="cb73"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb73-1"><a href="visium-part-ii.html#cb73-1" tabindex="-1"></a><span class="fu">heatmSpatialCorFeats</span>(visium_brain,</span>
<span id="cb73-2"><a href="visium-part-ii.html#cb73-2" tabindex="-1"></a>                     <span class="at">spatCorObject =</span> spat_cor_netw_DT,</span>
<span id="cb73-3"><a href="visium-part-ii.html#cb73-3" tabindex="-1"></a>                     <span class="at">use_clus_name =</span> <span class="st">&quot;spat_netw_clus&quot;</span>,</span>
<span id="cb73-4"><a href="visium-part-ii.html#cb73-4" tabindex="-1"></a>                     <span class="at">heatmap_legend_param =</span> <span class="fu">list</span>(<span class="at">title =</span> <span class="cn">NULL</span>))</span></code></pre></div>
<p><img src="img/02_session1/12-heatmSpatialCorFeats.png" width="100%" style="display: block; margin: auto;" /></p>
<ul>
<li>Rank spatial correlated clusters and show genes for selected clusters</li>
</ul>
<div class="sourceCode" id="cb74"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb74-1"><a href="visium-part-ii.html#cb74-1" tabindex="-1"></a>netw_ranks <span class="ot">&lt;-</span> <span class="fu">rankSpatialCorGroups</span>(</span>
<span id="cb74-2"><a href="visium-part-ii.html#cb74-2" tabindex="-1"></a>  visium_brain,</span>
<span id="cb74-3"><a href="visium-part-ii.html#cb74-3" tabindex="-1"></a>  <span class="at">spatCorObject =</span> spat_cor_netw_DT, </span>
<span id="cb74-4"><a href="visium-part-ii.html#cb74-4" tabindex="-1"></a>  <span class="at">use_clus_name =</span> <span class="st">&quot;spat_netw_clus&quot;</span>)</span>
<span id="cb74-5"><a href="visium-part-ii.html#cb74-5" tabindex="-1"></a></span>
<span id="cb74-6"><a href="visium-part-ii.html#cb74-6" tabindex="-1"></a>top_netw_spat_cluster <span class="ot">&lt;-</span> <span class="fu">showSpatialCorFeats</span>(spat_cor_netw_DT, </span>
<span id="cb74-7"><a href="visium-part-ii.html#cb74-7" tabindex="-1"></a>                                            <span class="at">use_clus_name =</span> <span class="st">&quot;spat_netw_clus&quot;</span>,</span>
<span id="cb74-8"><a href="visium-part-ii.html#cb74-8" tabindex="-1"></a>                                            <span class="at">selected_clusters =</span> <span class="dv">6</span>, </span>
<span id="cb74-9"><a href="visium-part-ii.html#cb74-9" tabindex="-1"></a>                                            <span class="at">show_top_feats =</span> <span class="dv">1</span>)</span></code></pre></div>
<p><img src="img/02_session1/13-rankSpatialCorGroups.png" width="100%" style="display: block; margin: auto;" /></p>
<ul>
<li>Create metagene enrichment score for clusters</li>
</ul>
<div class="sourceCode" id="cb75"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb75-1"><a href="visium-part-ii.html#cb75-1" tabindex="-1"></a>cluster_genes_DT <span class="ot">&lt;-</span> <span class="fu">showSpatialCorFeats</span>(spat_cor_netw_DT, </span>
<span id="cb75-2"><a href="visium-part-ii.html#cb75-2" tabindex="-1"></a>                                        <span class="at">use_clus_name =</span> <span class="st">&quot;spat_netw_clus&quot;</span>, </span>
<span id="cb75-3"><a href="visium-part-ii.html#cb75-3" tabindex="-1"></a>                                        <span class="at">show_top_feats =</span> <span class="dv">1</span>)</span>
<span id="cb75-4"><a href="visium-part-ii.html#cb75-4" tabindex="-1"></a></span>
<span id="cb75-5"><a href="visium-part-ii.html#cb75-5" tabindex="-1"></a>cluster_genes <span class="ot">&lt;-</span> cluster_genes_DT<span class="sc">$</span>clus </span>
<span id="cb75-6"><a href="visium-part-ii.html#cb75-6" tabindex="-1"></a><span class="fu">names</span>(cluster_genes) <span class="ot">&lt;-</span> cluster_genes_DT<span class="sc">$</span>feat_ID</span>
<span id="cb75-7"><a href="visium-part-ii.html#cb75-7" tabindex="-1"></a></span>
<span id="cb75-8"><a href="visium-part-ii.html#cb75-8" tabindex="-1"></a>visium_brain <span class="ot">&lt;-</span> <span class="fu">createMetafeats</span>(visium_brain, </span>
<span id="cb75-9"><a href="visium-part-ii.html#cb75-9" tabindex="-1"></a>                                <span class="at">feat_clusters =</span> cluster_genes, </span>
<span id="cb75-10"><a href="visium-part-ii.html#cb75-10" tabindex="-1"></a>                                <span class="at">name =</span> <span class="st">&quot;cluster_metagene&quot;</span>)</span>
<span id="cb75-11"><a href="visium-part-ii.html#cb75-11" tabindex="-1"></a></span>
<span id="cb75-12"><a href="visium-part-ii.html#cb75-12" tabindex="-1"></a><span class="fu">spatCellPlot</span>(visium_brain,</span>
<span id="cb75-13"><a href="visium-part-ii.html#cb75-13" tabindex="-1"></a>             <span class="at">spat_enr_names =</span> <span class="st">&quot;cluster_metagene&quot;</span>,</span>
<span id="cb75-14"><a href="visium-part-ii.html#cb75-14" tabindex="-1"></a>             <span class="at">cell_annotation_values =</span> netw_ranks<span class="sc">$</span>clusters,</span>
<span id="cb75-15"><a href="visium-part-ii.html#cb75-15" tabindex="-1"></a>             <span class="at">point_size =</span> <span class="dv">1</span>, </span>
<span id="cb75-16"><a href="visium-part-ii.html#cb75-16" tabindex="-1"></a>             <span class="at">cow_n_col =</span> <span class="dv">5</span>)</span></code></pre></div>
<p><img src="img/02_session1/14-spatCellPlot2D.png" width="100%" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="spatially-informed-clusters" class="section level2 hasAnchor" number="7.5">
<h2><span class="header-section-number">7.5</span> Spatially informed clusters<a href="visium-part-ii.html#spatially-informed-clusters" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ul>
<li>Get the top 30 genes per spatial co-expression cluster</li>
</ul>
<div class="sourceCode" id="cb76"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb76-1"><a href="visium-part-ii.html#cb76-1" tabindex="-1"></a>coexpr_dt <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">data.table</span>(</span>
<span id="cb76-2"><a href="visium-part-ii.html#cb76-2" tabindex="-1"></a>  <span class="at">genes =</span> <span class="fu">names</span>(spat_cor_netw_DT<span class="sc">$</span>cor_clusters<span class="sc">$</span>spat_netw_clus),</span>
<span id="cb76-3"><a href="visium-part-ii.html#cb76-3" tabindex="-1"></a>  <span class="at">cluster =</span> spat_cor_netw_DT<span class="sc">$</span>cor_clusters<span class="sc">$</span>spat_netw_clus)</span>
<span id="cb76-4"><a href="visium-part-ii.html#cb76-4" tabindex="-1"></a></span>
<span id="cb76-5"><a href="visium-part-ii.html#cb76-5" tabindex="-1"></a>data.table<span class="sc">::</span><span class="fu">setorder</span>(coexpr_dt, cluster)</span>
<span id="cb76-6"><a href="visium-part-ii.html#cb76-6" tabindex="-1"></a></span>
<span id="cb76-7"><a href="visium-part-ii.html#cb76-7" tabindex="-1"></a>top30_coexpr_dt <span class="ot">&lt;-</span> coexpr_dt[, <span class="fu">head</span>(.SD, <span class="dv">30</span>) , by <span class="ot">=</span> cluster]</span>
<span id="cb76-8"><a href="visium-part-ii.html#cb76-8" tabindex="-1"></a></span>
<span id="cb76-9"><a href="visium-part-ii.html#cb76-9" tabindex="-1"></a>spatial_genes <span class="ot">&lt;-</span> top30_coexpr_dt<span class="sc">$</span>genes</span></code></pre></div>
<ul>
<li>Re-calculate the clustering</li>
</ul>
<div class="sourceCode" id="cb77"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb77-1"><a href="visium-part-ii.html#cb77-1" tabindex="-1"></a>visium_brain <span class="ot">&lt;-</span> <span class="fu">runPCA</span>(<span class="at">gobject =</span> visium_brain,</span>
<span id="cb77-2"><a href="visium-part-ii.html#cb77-2" tabindex="-1"></a>                       <span class="at">feats_to_use =</span> spatial_genes,</span>
<span id="cb77-3"><a href="visium-part-ii.html#cb77-3" tabindex="-1"></a>                       <span class="at">name =</span> <span class="st">&quot;custom_pca&quot;</span>)</span>
<span id="cb77-4"><a href="visium-part-ii.html#cb77-4" tabindex="-1"></a></span>
<span id="cb77-5"><a href="visium-part-ii.html#cb77-5" tabindex="-1"></a>visium_brain <span class="ot">&lt;-</span> <span class="fu">runUMAP</span>(visium_brain, </span>
<span id="cb77-6"><a href="visium-part-ii.html#cb77-6" tabindex="-1"></a>                        <span class="at">dim_reduction_name =</span> <span class="st">&quot;custom_pca&quot;</span>, </span>
<span id="cb77-7"><a href="visium-part-ii.html#cb77-7" tabindex="-1"></a>                        <span class="at">dimensions_to_use =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>,</span>
<span id="cb77-8"><a href="visium-part-ii.html#cb77-8" tabindex="-1"></a>                        <span class="at">name =</span> <span class="st">&quot;custom_umap&quot;</span>)</span>
<span id="cb77-9"><a href="visium-part-ii.html#cb77-9" tabindex="-1"></a></span>
<span id="cb77-10"><a href="visium-part-ii.html#cb77-10" tabindex="-1"></a>visium_brain <span class="ot">&lt;-</span> <span class="fu">createNearestNetwork</span>(<span class="at">gobject =</span> visium_brain,</span>
<span id="cb77-11"><a href="visium-part-ii.html#cb77-11" tabindex="-1"></a>                                     <span class="at">dim_reduction_name =</span> <span class="st">&quot;custom_pca&quot;</span>,</span>
<span id="cb77-12"><a href="visium-part-ii.html#cb77-12" tabindex="-1"></a>                                     <span class="at">dimensions_to_use =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>, </span>
<span id="cb77-13"><a href="visium-part-ii.html#cb77-13" tabindex="-1"></a>                                     <span class="at">k =</span> <span class="dv">5</span>,</span>
<span id="cb77-14"><a href="visium-part-ii.html#cb77-14" tabindex="-1"></a>                                     <span class="at">name =</span> <span class="st">&quot;custom_NN&quot;</span>)</span>
<span id="cb77-15"><a href="visium-part-ii.html#cb77-15" tabindex="-1"></a></span>
<span id="cb77-16"><a href="visium-part-ii.html#cb77-16" tabindex="-1"></a>visium_brain <span class="ot">&lt;-</span> <span class="fu">doLeidenCluster</span>(<span class="at">gobject =</span> visium_brain, </span>
<span id="cb77-17"><a href="visium-part-ii.html#cb77-17" tabindex="-1"></a>                                <span class="at">network_name =</span> <span class="st">&quot;custom_NN&quot;</span>,</span>
<span id="cb77-18"><a href="visium-part-ii.html#cb77-18" tabindex="-1"></a>                                <span class="at">resolution =</span> <span class="fl">0.15</span>, </span>
<span id="cb77-19"><a href="visium-part-ii.html#cb77-19" tabindex="-1"></a>                                <span class="at">n_iterations =</span> <span class="dv">1000</span>,</span>
<span id="cb77-20"><a href="visium-part-ii.html#cb77-20" tabindex="-1"></a>                                <span class="at">name =</span> <span class="st">&quot;custom_leiden&quot;</span>)</span></code></pre></div>
<ul>
<li>Visualize</li>
</ul>
<div class="sourceCode" id="cb78"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb78-1"><a href="visium-part-ii.html#cb78-1" tabindex="-1"></a><span class="fu">spatPlot2D</span>(visium_brain, </span>
<span id="cb78-2"><a href="visium-part-ii.html#cb78-2" tabindex="-1"></a>           <span class="at">cell_color =</span> <span class="st">&quot;custom_leiden&quot;</span>)</span></code></pre></div>
<p><img src="img/02_session1/15-spatPlot2D.png" width="100%" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb79"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb79-1"><a href="visium-part-ii.html#cb79-1" tabindex="-1"></a><span class="fu">plotUMAP</span>(<span class="at">gobject =</span> visium_brain, </span>
<span id="cb79-2"><a href="visium-part-ii.html#cb79-2" tabindex="-1"></a>         <span class="at">cell_color =</span> <span class="st">&quot;custom_leiden&quot;</span>, </span>
<span id="cb79-3"><a href="visium-part-ii.html#cb79-3" tabindex="-1"></a>         <span class="at">point_size =</span> <span class="fl">1.5</span>)</span></code></pre></div>
<p><img src="img/02_session1/16-UMAP.png" width="100%" style="display: block; margin: auto;" /></p>
</div>
<div id="spatial-domains-hmrf" class="section level2 hasAnchor" number="7.6">
<h2><span class="header-section-number">7.6</span> Spatial domains HMRF<a href="visium-part-ii.html#spatial-domains-hmrf" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Hidden Markov Random Field (HMRF) models capture spatial dependencies and segment tissue regions based on shared gene expression patterns.</p>
<ul>
<li>Do HMRF with different betas on top 30 genes per spatial co-expression module</li>
</ul>
<div class="sourceCode" id="cb80"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb80-1"><a href="visium-part-ii.html#cb80-1" tabindex="-1"></a>HMRF_spatial_genes <span class="ot">&lt;-</span> <span class="fu">doHMRF</span>(<span class="at">gobject =</span> visium_brain,</span>
<span id="cb80-2"><a href="visium-part-ii.html#cb80-2" tabindex="-1"></a>                            <span class="at">expression_values =</span> <span class="st">&quot;scaled&quot;</span>,</span>
<span id="cb80-3"><a href="visium-part-ii.html#cb80-3" tabindex="-1"></a>                            <span class="at">spatial_genes =</span> spatial_genes, </span>
<span id="cb80-4"><a href="visium-part-ii.html#cb80-4" tabindex="-1"></a>                            <span class="at">k =</span> <span class="dv">20</span>,</span>
<span id="cb80-5"><a href="visium-part-ii.html#cb80-5" tabindex="-1"></a>                            <span class="at">spatial_network_name =</span> <span class="st">&quot;spatial_network&quot;</span>,</span>
<span id="cb80-6"><a href="visium-part-ii.html#cb80-6" tabindex="-1"></a>                            <span class="at">betas =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">10</span>, <span class="dv">5</span>),</span>
<span id="cb80-7"><a href="visium-part-ii.html#cb80-7" tabindex="-1"></a>                            <span class="at">output_folder =</span> <span class="st">&quot;11_HMRF/&quot;</span>)</span>
<span id="cb80-8"><a href="visium-part-ii.html#cb80-8" tabindex="-1"></a></span>
<span id="cb80-9"><a href="visium-part-ii.html#cb80-9" tabindex="-1"></a>visium_brain <span class="ot">&lt;-</span> <span class="fu">addHMRF</span>(<span class="at">gobject =</span> visium_brain, </span>
<span id="cb80-10"><a href="visium-part-ii.html#cb80-10" tabindex="-1"></a>                        <span class="at">HMRFoutput =</span> HMRF_spatial_genes,</span>
<span id="cb80-11"><a href="visium-part-ii.html#cb80-11" tabindex="-1"></a>                        <span class="at">k =</span> <span class="dv">20</span>, </span>
<span id="cb80-12"><a href="visium-part-ii.html#cb80-12" tabindex="-1"></a>                        <span class="at">betas_to_add =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">10</span>, <span class="dv">20</span>, <span class="dv">30</span>, <span class="dv">40</span>),</span>
<span id="cb80-13"><a href="visium-part-ii.html#cb80-13" tabindex="-1"></a>                        <span class="at">hmrf_name =</span> <span class="st">&quot;HMRF&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb81"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb81-1"><a href="visium-part-ii.html#cb81-1" tabindex="-1"></a><span class="fu">spatPlot2D</span>(<span class="at">gobject =</span> visium_brain, </span>
<span id="cb81-2"><a href="visium-part-ii.html#cb81-2" tabindex="-1"></a>           <span class="at">cell_color =</span> <span class="st">&quot;HMRF_k20_b.40&quot;</span>)</span></code></pre></div>
<ul>
<li>Visualize</li>
</ul>
<p><img src="img/02_session1/17-spatPlot2D.png" width="100%" style="display: block; margin: auto;" /></p>
</div>
<div id="interactive-tools" class="section level2 hasAnchor" number="7.7">
<h2><span class="header-section-number">7.7</span> Interactive tools<a href="visium-part-ii.html#interactive-tools" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ul>
<li>Create a spatial plot</li>
</ul>
<div class="sourceCode" id="cb82"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb82-1"><a href="visium-part-ii.html#cb82-1" tabindex="-1"></a>brain_spatPlot <span class="ot">&lt;-</span> <span class="fu">spatPlot2D</span>(<span class="at">gobject =</span> visium_brain,</span>
<span id="cb82-2"><a href="visium-part-ii.html#cb82-2" tabindex="-1"></a>                          <span class="at">cell_color =</span> <span class="st">&quot;leiden_clus&quot;</span>,</span>
<span id="cb82-3"><a href="visium-part-ii.html#cb82-3" tabindex="-1"></a>                          <span class="at">show_image =</span> <span class="cn">FALSE</span>,</span>
<span id="cb82-4"><a href="visium-part-ii.html#cb82-4" tabindex="-1"></a>                          <span class="at">return_plot =</span> <span class="cn">TRUE</span>,</span>
<span id="cb82-5"><a href="visium-part-ii.html#cb82-5" tabindex="-1"></a>                          <span class="at">point_size =</span> <span class="dv">1</span>)</span>
<span id="cb82-6"><a href="visium-part-ii.html#cb82-6" tabindex="-1"></a></span>
<span id="cb82-7"><a href="visium-part-ii.html#cb82-7" tabindex="-1"></a>brain_spatPlot</span></code></pre></div>
<ul>
<li>Run the Shiny app</li>
</ul>
<div class="sourceCode" id="cb83"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb83-1"><a href="visium-part-ii.html#cb83-1" tabindex="-1"></a><span class="fu">plotInteractivePolygons</span>(brain_spatPlot)</span></code></pre></div>
<p><img src="img/02_session1/18-spatPlot2D.png" width="100%" style="display: block; margin: auto;" /></p>
<ul>
<li>Select the regions of interest and save the coordinates</li>
</ul>
<div class="sourceCode" id="cb84"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb84-1"><a href="visium-part-ii.html#cb84-1" tabindex="-1"></a>polygon_coordinates <span class="ot">&lt;-</span> <span class="fu">plotInteractivePolygons</span>(brain_spatPlot)</span></code></pre></div>
<p><img src="img/02_session1/19-spatPlot2D.png" width="60%" style="display: block; margin: auto;" /></p>
<ul>
<li>Transform the data.table or data.frame with coordinates into a Giotto polygon object</li>
</ul>
<div class="sourceCode" id="cb85"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb85-1"><a href="visium-part-ii.html#cb85-1" tabindex="-1"></a>giotto_polygons <span class="ot">&lt;-</span> <span class="fu">createGiottoPolygonsFromDfr</span>(polygon_coordinates, </span>
<span id="cb85-2"><a href="visium-part-ii.html#cb85-2" tabindex="-1"></a>                                               <span class="at">name =</span> <span class="st">&quot;selections&quot;</span>,</span>
<span id="cb85-3"><a href="visium-part-ii.html#cb85-3" tabindex="-1"></a>                                               <span class="at">calc_centroids =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<ul>
<li>Add the polygons (or spatial units) to the Giotto object</li>
</ul>
<div class="sourceCode" id="cb86"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb86-1"><a href="visium-part-ii.html#cb86-1" tabindex="-1"></a>visium_brain <span class="ot">&lt;-</span> <span class="fu">addGiottoPolygons</span>(<span class="at">gobject =</span> visium_brain,</span>
<span id="cb86-2"><a href="visium-part-ii.html#cb86-2" tabindex="-1"></a>                                  <span class="at">gpolygons =</span> <span class="fu">list</span>(giotto_polygons))</span></code></pre></div>
<ul>
<li>Add the corresponding polygon IDs to the cell metadata</li>
</ul>
<div class="sourceCode" id="cb87"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb87-1"><a href="visium-part-ii.html#cb87-1" tabindex="-1"></a>visium_brain <span class="ot">&lt;-</span> <span class="fu">addPolygonCells</span>(visium_brain,</span>
<span id="cb87-2"><a href="visium-part-ii.html#cb87-2" tabindex="-1"></a>                                <span class="at">polygon_name =</span> <span class="st">&quot;selections&quot;</span>)</span></code></pre></div>
<ul>
<li>Extract the coordinates and IDs from cells located within one or multiple regions of interest.</li>
</ul>
<div class="sourceCode" id="cb88"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb88-1"><a href="visium-part-ii.html#cb88-1" tabindex="-1"></a><span class="fu">getCellsFromPolygon</span>(visium_brain,</span>
<span id="cb88-2"><a href="visium-part-ii.html#cb88-2" tabindex="-1"></a>                    <span class="at">polygon_name =</span> <span class="st">&quot;selections&quot;</span>,</span>
<span id="cb88-3"><a href="visium-part-ii.html#cb88-3" tabindex="-1"></a>                    <span class="at">polygons =</span> <span class="st">&quot;polygon 1&quot;</span>)</span></code></pre></div>
<p>If no polygon name is provided, the function will retrieve cells located within all polygons</p>
<div class="sourceCode" id="cb89"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb89-1"><a href="visium-part-ii.html#cb89-1" tabindex="-1"></a><span class="fu">getCellsFromPolygon</span>(visium_brain,</span>
<span id="cb89-2"><a href="visium-part-ii.html#cb89-2" tabindex="-1"></a>                    <span class="at">polygon_name =</span> <span class="st">&quot;selections&quot;</span>)</span></code></pre></div>
<ul>
<li>Compare the expression levels of some genes of interest between the selected regions</li>
</ul>
<div class="sourceCode" id="cb90"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb90-1"><a href="visium-part-ii.html#cb90-1" tabindex="-1"></a><span class="fu">comparePolygonExpression</span>(visium_brain,</span>
<span id="cb90-2"><a href="visium-part-ii.html#cb90-2" tabindex="-1"></a>                         <span class="at">selected_feats =</span> <span class="fu">c</span>(<span class="st">&quot;Stmn1&quot;</span>, <span class="st">&quot;Psd&quot;</span>, <span class="st">&quot;Ly6h&quot;</span>))</span></code></pre></div>
<p><img src="img/02_session1/20-heatmap.png" width="60%" style="display: block; margin: auto;" /></p>
<ul>
<li>Calculate the top genes expressed within each region, then provide the result to compare polygons</li>
</ul>
<div class="sourceCode" id="cb91"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb91-1"><a href="visium-part-ii.html#cb91-1" tabindex="-1"></a>scran_results <span class="ot">&lt;-</span> <span class="fu">findMarkers_one_vs_all</span>(visium_brain,</span>
<span id="cb91-2"><a href="visium-part-ii.html#cb91-2" tabindex="-1"></a>                                        <span class="at">spat_unit =</span> <span class="st">&quot;cell&quot;</span>,</span>
<span id="cb91-3"><a href="visium-part-ii.html#cb91-3" tabindex="-1"></a>                                        <span class="at">feat_type =</span> <span class="st">&quot;rna&quot;</span>,</span>
<span id="cb91-4"><a href="visium-part-ii.html#cb91-4" tabindex="-1"></a>                                        <span class="at">method =</span> <span class="st">&quot;scran&quot;</span>,</span>
<span id="cb91-5"><a href="visium-part-ii.html#cb91-5" tabindex="-1"></a>                                        <span class="at">expression_values =</span> <span class="st">&quot;normalized&quot;</span>,</span>
<span id="cb91-6"><a href="visium-part-ii.html#cb91-6" tabindex="-1"></a>                                        <span class="at">cluster_column =</span> <span class="st">&quot;selections&quot;</span>,</span>
<span id="cb91-7"><a href="visium-part-ii.html#cb91-7" tabindex="-1"></a>                                        <span class="at">min_feats =</span> <span class="dv">2</span>)</span>
<span id="cb91-8"><a href="visium-part-ii.html#cb91-8" tabindex="-1"></a></span>
<span id="cb91-9"><a href="visium-part-ii.html#cb91-9" tabindex="-1"></a>top_genes <span class="ot">&lt;-</span> scran_results[, <span class="fu">head</span>(.SD, <span class="dv">2</span>), by <span class="ot">=</span> <span class="st">&quot;cluster&quot;</span>]<span class="sc">$</span>feats</span>
<span id="cb91-10"><a href="visium-part-ii.html#cb91-10" tabindex="-1"></a></span>
<span id="cb91-11"><a href="visium-part-ii.html#cb91-11" tabindex="-1"></a><span class="fu">comparePolygonExpression</span>(visium_brain,</span>
<span id="cb91-12"><a href="visium-part-ii.html#cb91-12" tabindex="-1"></a>                         <span class="at">selected_feats =</span> top_genes)</span></code></pre></div>
<p><img src="img/02_session1/21-heatmap.png" width="60%" style="display: block; margin: auto;" /></p>
<ul>
<li>Compare the abundance of cell types between the selected regions</li>
</ul>
<div class="sourceCode" id="cb92"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb92-1"><a href="visium-part-ii.html#cb92-1" tabindex="-1"></a><span class="fu">compareCellAbundance</span>(visium_brain)</span></code></pre></div>
<p><img src="img/02_session1/22-heatmap.png" width="60%" style="display: block; margin: auto;" /></p>
<ul>
<li>Use other columns within the cell metadata table to compare the cell type abundances</li>
</ul>
<div class="sourceCode" id="cb93"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb93-1"><a href="visium-part-ii.html#cb93-1" tabindex="-1"></a><span class="fu">compareCellAbundance</span>(visium_brain, </span>
<span id="cb93-2"><a href="visium-part-ii.html#cb93-2" tabindex="-1"></a>                     <span class="at">cell_type_column =</span> <span class="st">&quot;custom_leiden&quot;</span>)</span></code></pre></div>
<p><img src="img/02_session1/23-heatmap.png" width="60%" style="display: block; margin: auto;" /></p>
<ul>
<li>Use the spatPlot arguments to isolate and plot each region.</li>
</ul>
<div class="sourceCode" id="cb94"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb94-1"><a href="visium-part-ii.html#cb94-1" tabindex="-1"></a><span class="fu">spatPlot2D</span>(visium_brain,</span>
<span id="cb94-2"><a href="visium-part-ii.html#cb94-2" tabindex="-1"></a>           <span class="at">cell_color =</span> <span class="st">&quot;leiden_clus&quot;</span>,</span>
<span id="cb94-3"><a href="visium-part-ii.html#cb94-3" tabindex="-1"></a>           <span class="at">group_by =</span> <span class="st">&quot;selections&quot;</span>,</span>
<span id="cb94-4"><a href="visium-part-ii.html#cb94-4" tabindex="-1"></a>           <span class="at">cow_n_col =</span> <span class="dv">3</span>,</span>
<span id="cb94-5"><a href="visium-part-ii.html#cb94-5" tabindex="-1"></a>           <span class="at">point_size =</span> <span class="dv">2</span>,</span>
<span id="cb94-6"><a href="visium-part-ii.html#cb94-6" tabindex="-1"></a>           <span class="at">show_legend =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<p><img src="img/02_session1/24-spatPlot2D.png" width="80%" style="display: block; margin: auto;" /></p>
<ul>
<li>Color each cell by cluster, cell type or expression level.</li>
</ul>
<div class="sourceCode" id="cb95"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb95-1"><a href="visium-part-ii.html#cb95-1" tabindex="-1"></a><span class="fu">spatFeatPlot2D</span>(visium_brain,</span>
<span id="cb95-2"><a href="visium-part-ii.html#cb95-2" tabindex="-1"></a>               <span class="at">expression_values =</span> <span class="st">&quot;scaled&quot;</span>,</span>
<span id="cb95-3"><a href="visium-part-ii.html#cb95-3" tabindex="-1"></a>               <span class="at">group_by =</span> <span class="st">&quot;selections&quot;</span>,</span>
<span id="cb95-4"><a href="visium-part-ii.html#cb95-4" tabindex="-1"></a>               <span class="at">feats =</span> <span class="st">&quot;Psd&quot;</span>,</span>
<span id="cb95-5"><a href="visium-part-ii.html#cb95-5" tabindex="-1"></a>               <span class="at">point_size =</span> <span class="dv">2</span>)</span></code></pre></div>
<p><img src="img/02_session1/25-spatFeatPlot2D.png" width="80%" style="display: block; margin: auto;" /></p>
<ul>
<li>Plot again the polygons</li>
</ul>
<div class="sourceCode" id="cb96"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb96-1"><a href="visium-part-ii.html#cb96-1" tabindex="-1"></a><span class="fu">plotPolygons</span>(visium_brain,</span>
<span id="cb96-2"><a href="visium-part-ii.html#cb96-2" tabindex="-1"></a>             <span class="at">polygon_name =</span> <span class="st">&quot;selections&quot;</span>,</span>
<span id="cb96-3"><a href="visium-part-ii.html#cb96-3" tabindex="-1"></a>             <span class="at">x =</span> brain_spatPlot)</span></code></pre></div>
<p><img src="img/02_session1/26-spatPlot2D.png" width="60%" style="display: block; margin: auto;" /></p>
</div>
<div id="session-info-1" class="section level2 hasAnchor" number="7.8">
<h2><span class="header-section-number">7.8</span> Session info<a href="visium-part-ii.html#session-info-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb97"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb97-1"><a href="visium-part-ii.html#cb97-1" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code></pre></div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="visium-part-i.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="visium-hd.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
