<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>6 Visium Part I | Workshop: Spatial multi-omics data analysis with Giotto Suite</title>
  <meta name="description" content="6 Visium Part I | Workshop: Spatial multi-omics data analysis with Giotto Suite" />
  <meta name="generator" content="bookdown 0.40 and GitBook 2.6.7" />

  <meta property="og:title" content="6 Visium Part I | Workshop: Spatial multi-omics data analysis with Giotto Suite" />
  <meta property="og:type" content="book" />
  
  
  <meta name="github-repo" content="drieslab/giotto_workshop_2024" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="6 Visium Part I | Workshop: Spatial multi-omics data analysis with Giotto Suite" />
  
  
  

<meta name="author" content="Ruben Dries, Jiaji George Chen, Joselyn Cristina Chávez-Fuentes, Junxiang Xu ,Edward Ruiz, Jeff Sheridan, Iqra Amin, Wen Wang" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="creating-a-giotto-object.html"/>
<link rel="next" href="visium-part-ii.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Giotto Workshop 2024</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#instructors"><i class="fa fa-check"></i><b>1.1</b> Instructors</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#topics-and-schedule"><i class="fa fa-check"></i><b>1.2</b> Topics and Schedule:</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#license"><i class="fa fa-check"></i><b>1.3</b> License</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html"><i class="fa fa-check"></i><b>2</b> Spatial omics technologies</a>
<ul>
<li class="chapter" data-level="2.1" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#presentation"><i class="fa fa-check"></i><b>2.1</b> Presentation</a></li>
<li class="chapter" data-level="2.2" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#short-summary"><i class="fa fa-check"></i><b>2.2</b> Short summary</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#why-do-we-need-spatial-omics-technologies"><i class="fa fa-check"></i><b>2.2.1</b> Why do we need spatial omics technologies?</a></li>
<li class="chapter" data-level="2.2.2" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#what-is-spatial-omics"><i class="fa fa-check"></i><b>2.2.2</b> What is spatial omics?</a></li>
<li class="chapter" data-level="2.2.3" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#what-are-the-main-spatial-omics-technologies"><i class="fa fa-check"></i><b>2.2.3</b> What are the main spatial omics technologies?</a></li>
<li class="chapter" data-level="2.2.4" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#other-spatial-omics-atac-seq-cutrun-lipidomics-etc"><i class="fa fa-check"></i><b>2.2.4</b> Other Spatial omics: ATAC-seq, CUT&amp;RUN, lipidomics, etc</a></li>
<li class="chapter" data-level="2.2.5" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#what-are-the-different-types-of-spatial-downstream-analyses"><i class="fa fa-check"></i><b>2.2.5</b> What are the different types of spatial downstream analyses?</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html"><i class="fa fa-check"></i><b>3</b> Introduction to the Giotto package</a>
<ul>
<li class="chapter" data-level="3.1" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html#slides"><i class="fa fa-check"></i><b>3.1</b> Slides</a></li>
<li class="chapter" data-level="3.2" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html#ecosystem"><i class="fa fa-check"></i><b>3.2</b> Ecosystem</a></li>
<li class="chapter" data-level="3.3" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html#installation-python-environment"><i class="fa fa-check"></i><b>3.3</b> Installation + python environment</a></li>
<li class="chapter" data-level="3.4" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html#giotto-instructions"><i class="fa fa-check"></i><b>3.4</b> Giotto instructions</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html"><i class="fa fa-check"></i><b>4</b> Data formatting and Pre-processing</a>
<ul>
<li class="chapter" data-level="4.1" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#data-formats"><i class="fa fa-check"></i><b>4.1</b> Data formats</a></li>
<li class="chapter" data-level="4.2" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#pre-processing"><i class="fa fa-check"></i><b>4.2</b> Pre-processing</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html"><i class="fa fa-check"></i><b>5</b> Creating a Giotto object</a>
<ul>
<li class="chapter" data-level="5.1" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#overview"><i class="fa fa-check"></i><b>5.1</b> Overview</a></li>
<li class="chapter" data-level="5.2" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#from-matrix-locations"><i class="fa fa-check"></i><b>5.2</b> From matrix + locations</a></li>
<li class="chapter" data-level="5.3" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#from-subcellular-raw-data-transcripts-or-images-polygons"><i class="fa fa-check"></i><b>5.3</b> From subcellular raw data (transcripts or images) + polygons</a></li>
<li class="chapter" data-level="5.4" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#from-piece-wise"><i class="fa fa-check"></i><b>5.4</b> From piece-wise</a></li>
<li class="chapter" data-level="5.5" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#using-convenience-functions-for-popular-technologies-vizgen-xenium-cosmx"><i class="fa fa-check"></i><b>5.5</b> Using convenience functions for popular technologies (Vizgen, Xenium, CosMx, …)</a></li>
<li class="chapter" data-level="5.6" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#spatial-plots"><i class="fa fa-check"></i><b>5.6</b> Spatial plots</a></li>
<li class="chapter" data-level="5.7" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#subsetting"><i class="fa fa-check"></i><b>5.7</b> Subsetting</a></li>
<li class="chapter" data-level="5.8" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#mini-objects-giottodata"><i class="fa fa-check"></i><b>5.8</b> Mini objects &amp; GiottoData</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="visium-part-i.html"><a href="visium-part-i.html"><i class="fa fa-check"></i><b>6</b> Visium Part I</a>
<ul>
<li class="chapter" data-level="6.1" data-path="visium-part-i.html"><a href="visium-part-i.html#introduction-to-the-spatial-dataset"><i class="fa fa-check"></i><b>6.1</b> Introduction to the spatial dataset</a></li>
<li class="chapter" data-level="6.2" data-path="visium-part-i.html"><a href="visium-part-i.html#download-dataset"><i class="fa fa-check"></i><b>6.2</b> Download dataset</a></li>
<li class="chapter" data-level="6.3" data-path="visium-part-i.html"><a href="visium-part-i.html#create-the-giotto-object"><i class="fa fa-check"></i><b>6.3</b> Create the Giotto object</a></li>
<li class="chapter" data-level="6.4" data-path="visium-part-i.html"><a href="visium-part-i.html#subset-on-spots-that-were-covered-by-tissue"><i class="fa fa-check"></i><b>6.4</b> Subset on spots that were covered by tissue</a></li>
<li class="chapter" data-level="6.5" data-path="visium-part-i.html"><a href="visium-part-i.html#quality-control"><i class="fa fa-check"></i><b>6.5</b> Quality control</a></li>
<li class="chapter" data-level="6.6" data-path="visium-part-i.html"><a href="visium-part-i.html#filtering"><i class="fa fa-check"></i><b>6.6</b> Filtering</a></li>
<li class="chapter" data-level="6.7" data-path="visium-part-i.html"><a href="visium-part-i.html#normalization"><i class="fa fa-check"></i><b>6.7</b> Normalization</a></li>
<li class="chapter" data-level="6.8" data-path="visium-part-i.html"><a href="visium-part-i.html#feature-selection"><i class="fa fa-check"></i><b>6.8</b> Feature selection</a>
<ul>
<li class="chapter" data-level="6.8.1" data-path="visium-part-i.html"><a href="visium-part-i.html#highly-variable-features"><i class="fa fa-check"></i><b>6.8.1</b> Highly Variable Features:</a></li>
</ul></li>
<li class="chapter" data-level="6.9" data-path="visium-part-i.html"><a href="visium-part-i.html#dimension-reduction"><i class="fa fa-check"></i><b>6.9</b> Dimension Reduction</a>
<ul>
<li class="chapter" data-level="6.9.1" data-path="visium-part-i.html"><a href="visium-part-i.html#pca"><i class="fa fa-check"></i><b>6.9.1</b> PCA</a></li>
<li class="chapter" data-level="6.9.2" data-path="visium-part-i.html"><a href="visium-part-i.html#umap"><i class="fa fa-check"></i><b>6.9.2</b> UMAP</a></li>
<li class="chapter" data-level="6.9.3" data-path="visium-part-i.html"><a href="visium-part-i.html#t-sne"><i class="fa fa-check"></i><b>6.9.3</b> t-SNE</a></li>
</ul></li>
<li class="chapter" data-level="6.10" data-path="visium-part-i.html"><a href="visium-part-i.html#clustering"><i class="fa fa-check"></i><b>6.10</b> Clustering</a>
<ul>
<li class="chapter" data-level="6.10.1" data-path="visium-part-i.html"><a href="visium-part-i.html#calculate-leiden-clustering"><i class="fa fa-check"></i><b>6.10.1</b> Calculate Leiden clustering</a></li>
<li class="chapter" data-level="6.10.2" data-path="visium-part-i.html"><a href="visium-part-i.html#calculate-louvain-clustering"><i class="fa fa-check"></i><b>6.10.2</b> Calculate Louvain clustering</a></li>
</ul></li>
<li class="chapter" data-level="6.11" data-path="visium-part-i.html"><a href="visium-part-i.html#save-the-object"><i class="fa fa-check"></i><b>6.11</b> Save the object</a></li>
<li class="chapter" data-level="6.12" data-path="visium-part-i.html"><a href="visium-part-i.html#session-info"><i class="fa fa-check"></i><b>6.12</b> Session info</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="visium-part-ii.html"><a href="visium-part-ii.html"><i class="fa fa-check"></i><b>7</b> Visium Part II</a>
<ul>
<li class="chapter" data-level="7.1" data-path="visium-part-ii.html"><a href="visium-part-ii.html#load-the-object"><i class="fa fa-check"></i><b>7.1</b> Load the object</a></li>
<li class="chapter" data-level="7.2" data-path="visium-part-ii.html"><a href="visium-part-ii.html#differential-expression"><i class="fa fa-check"></i><b>7.2</b> Differential expression</a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="visium-part-ii.html"><a href="visium-part-ii.html#gini-markers"><i class="fa fa-check"></i><b>7.2.1</b> Gini markers</a></li>
<li class="chapter" data-level="7.2.2" data-path="visium-part-ii.html"><a href="visium-part-ii.html#scran-markers"><i class="fa fa-check"></i><b>7.2.2</b> Scran markers</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="visium-part-ii.html"><a href="visium-part-ii.html#enrichment-deconvolution"><i class="fa fa-check"></i><b>7.3</b> Enrichment &amp; Deconvolution</a>
<ul>
<li class="chapter" data-level="7.3.1" data-path="visium-part-ii.html"><a href="visium-part-ii.html#pagerank"><i class="fa fa-check"></i><b>7.3.1</b> PAGE/Rank</a></li>
<li class="chapter" data-level="7.3.2" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatialdwls"><i class="fa fa-check"></i><b>7.3.2</b> SpatialDWLS</a></li>
</ul></li>
<li class="chapter" data-level="7.4" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatial-expression-patterns"><i class="fa fa-check"></i><b>7.4</b> Spatial expression patterns</a>
<ul>
<li class="chapter" data-level="7.4.1" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatial-variable-genes"><i class="fa fa-check"></i><b>7.4.1</b> Spatial variable genes</a></li>
<li class="chapter" data-level="7.4.2" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatial-co-expression-modules"><i class="fa fa-check"></i><b>7.4.2</b> Spatial co-expression modules</a></li>
</ul></li>
<li class="chapter" data-level="7.5" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatially-informed-clusters"><i class="fa fa-check"></i><b>7.5</b> Spatially informed clusters</a></li>
<li class="chapter" data-level="7.6" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatial-domains-hmrf"><i class="fa fa-check"></i><b>7.6</b> Spatial domains HMRF</a></li>
<li class="chapter" data-level="7.7" data-path="visium-part-ii.html"><a href="visium-part-ii.html#interactive-tools"><i class="fa fa-check"></i><b>7.7</b> Interactive tools</a></li>
<li class="chapter" data-level="7.8" data-path="visium-part-ii.html"><a href="visium-part-ii.html#session-info-1"><i class="fa fa-check"></i><b>7.8</b> Session info</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="visium-hd.html"><a href="visium-hd.html"><i class="fa fa-check"></i><b>8</b> Visium HD</a>
<ul>
<li class="chapter" data-level="8.1" data-path="visium-hd.html"><a href="visium-hd.html#objective"><i class="fa fa-check"></i><b>8.1</b> Objective</a></li>
<li class="chapter" data-level="8.2" data-path="visium-hd.html"><a href="visium-hd.html#background"><i class="fa fa-check"></i><b>8.2</b> Background</a>
<ul>
<li class="chapter" data-level="8.2.1" data-path="visium-hd.html"><a href="visium-hd.html#visium-hd-technology"><i class="fa fa-check"></i><b>8.2.1</b> Visium HD Technology</a></li>
<li class="chapter" data-level="8.2.2" data-path="visium-hd.html"><a href="visium-hd.html#colorectal-cancer-sample"><i class="fa fa-check"></i><b>8.2.2</b> Colorectal Cancer Sample</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="visium-hd.html"><a href="visium-hd.html#data-ingestion"><i class="fa fa-check"></i><b>8.3</b> Data Ingestion</a>
<ul>
<li class="chapter" data-level="8.3.1" data-path="visium-hd.html"><a href="visium-hd.html#visium-hd-output-data-format"><i class="fa fa-check"></i><b>8.3.1</b> Visium HD output data format</a></li>
<li class="chapter" data-level="8.3.2" data-path="visium-hd.html"><a href="visium-hd.html#read-in-raw-data"><i class="fa fa-check"></i><b>8.3.2</b> Read in raw data</a></li>
<li class="chapter" data-level="8.3.3" data-path="visium-hd.html"><a href="visium-hd.html#giotto-visium-hd-convenience-function"><i class="fa fa-check"></i><b>8.3.3</b> Giotto Visium HD convenience function</a></li>
</ul></li>
<li class="chapter" data-level="8.4" data-path="visium-hd.html"><a href="visium-hd.html#tiling-and-aggregation"><i class="fa fa-check"></i><b>8.4</b> Tiling and aggregation</a></li>
<li class="chapter" data-level="8.5" data-path="visium-hd.html"><a href="visium-hd.html#scalability-and-projection-functions"><i class="fa fa-check"></i><b>8.5</b> Scalability and projection functions</a></li>
<li class="chapter" data-level="8.6" data-path="visium-hd.html"><a href="visium-hd.html#spatial-expression-patterns-1"><i class="fa fa-check"></i><b>8.6</b> Spatial expression patterns</a></li>
<li class="chapter" data-level="8.7" data-path="visium-hd.html"><a href="visium-hd.html#spatial-co-expression-modules-1"><i class="fa fa-check"></i><b>8.7</b> Spatial co-expression modules</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="xenium.html"><a href="xenium.html"><i class="fa fa-check"></i><b>9</b> Xenium</a>
<ul>
<li class="chapter" data-level="9.1" data-path="xenium.html"><a href="xenium.html#introduction-to-spatial-dataset"><i class="fa fa-check"></i><b>9.1</b> Introduction to spatial dataset</a></li>
<li class="chapter" data-level="9.2" data-path="xenium.html"><a href="xenium.html#additional-package-requirements"><i class="fa fa-check"></i><b>9.2</b> Additional package requirements</a></li>
<li class="chapter" data-level="9.3" data-path="xenium.html"><a href="xenium.html#introduce-mini-object"><i class="fa fa-check"></i><b>9.3</b> Introduce mini object</a></li>
<li class="chapter" data-level="9.4" data-path="xenium.html"><a href="xenium.html#read-in-raw-data-1"><i class="fa fa-check"></i><b>9.4</b> Read in raw data</a></li>
<li class="chapter" data-level="9.5" data-path="xenium.html"><a href="xenium.html#overlap-txs-polygons"><i class="fa fa-check"></i><b>9.5</b> Overlap txs &amp; polygons</a></li>
<li class="chapter" data-level="9.6" data-path="xenium.html"><a href="xenium.html#transcript-enrichment-gsea"><i class="fa fa-check"></i><b>9.6</b> Transcript enrichment GSEA</a></li>
<li class="chapter" data-level="9.7" data-path="xenium.html"><a href="xenium.html#spatial-location-analysis"><i class="fa fa-check"></i><b>9.7</b> Spatial location analysis</a></li>
<li class="chapter" data-level="9.8" data-path="xenium.html"><a href="xenium.html#visualizations"><i class="fa fa-check"></i><b>9.8</b> Visualizations</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="exploration.html"><a href="exploration.html"><i class="fa fa-check"></i><b>10</b> —– exploration</a></li>
<li class="chapter" data-level="11" data-path="image.html"><a href="image.html"><i class="fa fa-check"></i><b>11</b> IMAGE</a></li>
<li class="chapter" data-level="12" data-path="ometif_2_tifimage_path-output_file-out_tif-chunk_rows-chunk_rows-overwrite-true.html"><a href="ometif_2_tifimage_path-output_file-out_tif-chunk_rows-chunk_rows-overwrite-true.html"><i class="fa fa-check"></i><b>12</b> ometif_2_tif(image_path, output_file = out_tif, chunk_rows = chunk_rows, overwrite = TRUE)</a></li>
<li class="chapter" data-level="13" data-path="example-usage.html"><a href="example-usage.html"><i class="fa fa-check"></i><b>13</b> # Example usage</a></li>
<li class="chapter" data-level="14" data-path="input_file-pathtoinputfile.ome.html"><a href="input_file-pathtoinputfile.ome.html"><i class="fa fa-check"></i><b>14</b> input_file = “path/to/input/file.ome.tif”</a></li>
<li class="chapter" data-level="15" data-path="output_file-pathtooutputfile.html"><a href="output_file-pathtooutputfile.html"><i class="fa fa-check"></i><b>15</b> output_file = “path/to/output/file.tif”</a></li>
<li class="chapter" data-level="16" data-path="chunk_size-1000-1000-adjust-the-chunk-size-as-needed.html"><a href="chunk_size-1000-1000-adjust-the-chunk-size-as-needed.html"><i class="fa fa-check"></i><b>16</b> chunk_size = (1000, 1000) # Adjust the chunk size as needed</a></li>
<li class="chapter" data-level="17" data-path="num_processes-4-adjust-the-number-of-processes-based-on-available-cpu-cores.html"><a href="num_processes-4-adjust-the-number-of-processes-based-on-available-cpu-cores.html"><i class="fa fa-check"></i><b>17</b> num_processes = 4 # Adjust the number of processes based on available CPU cores</a></li>
<li class="chapter" data-level="18" data-path="section.html"><a href="section.html"><i class="fa fa-check"></i><b>18</b> </a></li>
<li class="chapter" data-level="19" data-path="ometif_2_tifinput_file-output_file-chunk_size-num_processes.html"><a href="ometif_2_tifinput_file-output_file-chunk_size-num_processes.html"><i class="fa fa-check"></i><b>19</b> ometif_2_tif(input_file, output_file, chunk_size, num_processes)</a></li>
<li class="chapter" data-level="20" data-path="transcripts.html"><a href="transcripts.html"><i class="fa fa-check"></i><b>20</b> TRANSCRIPTS</a></li>
<li class="chapter" data-level="21" data-path="where-are-the-issue-tx-at.html"><a href="where-are-the-issue-tx-at.html"><i class="fa fa-check"></i><b>21</b> where are the issue TX at?</a></li>
<li class="chapter" data-level="22" data-path="polys.html"><a href="polys.html"><i class="fa fa-check"></i><b>22</b> POLYS</a></li>
<li class="chapter" data-level="23" data-path="giotto.html"><a href="giotto.html"><i class="fa fa-check"></i><b>23</b> giotto</a></li>
<li class="chapter" data-level="24" data-path="no-hvf-calc-since-there-are-not-that-many-different-gene-species.html"><a href="no-hvf-calc-since-there-are-not-that-many-different-gene-species.html"><i class="fa fa-check"></i><b>24</b> no HVF calc since there are not that many different gene species</a></li>
<li class="chapter" data-level="25" data-path="set-seed-for-reproducibility.html"><a href="set-seed-for-reproducibility.html"><i class="fa fa-check"></i><b>25</b> set seed for reproducibility</a></li>
<li class="chapter" data-level="26" data-path="dimplot2d.html"><a href="dimplot2d.html"><i class="fa fa-check"></i><b>26</b> dimPlot2D(</a></li>
<li class="chapter" data-level="27" data-path="g_mini.html"><a href="g_mini.html"><i class="fa fa-check"></i><b>27</b> g_mini,</a></li>
<li class="chapter" data-level="28" data-path="spat_unit-cell.html"><a href="spat_unit-cell.html"><i class="fa fa-check"></i><b>28</b> spat_unit = “cell”,</a></li>
<li class="chapter" data-level="29" data-path="feat_type-rna.html"><a href="feat_type-rna.html"><i class="fa fa-check"></i><b>29</b> feat_type = “rna”,</a></li>
<li class="chapter" data-level="30" data-path="cell_color-leiden_clus.html"><a href="cell_color-leiden_clus.html"><i class="fa fa-check"></i><b>30</b> cell_color = “leiden_clus”,</a></li>
<li class="chapter" data-level="31" data-path="dim_reduction_name-umap.html"><a href="dim_reduction_name-umap.html"><i class="fa fa-check"></i><b>31</b> dim_reduction_name = “umap.projection”,</a></li>
<li class="chapter" data-level="32" data-path="dim_reduction_to_use-umap.html"><a href="dim_reduction_to_use-umap.html"><i class="fa fa-check"></i><b>32</b> dim_reduction_to_use = “umap”,</a></li>
<li class="chapter" data-level="33" data-path="save_param-list.html"><a href="save_param-list.html"><i class="fa fa-check"></i><b>33</b> save_param = list(</a></li>
<li class="chapter" data-level="34" data-path="save_name-mini_umap_1.html"><a href="save_name-mini_umap_1.html"><i class="fa fa-check"></i><b>34</b> save_name = “mini_umap_1.2”</a></li>
<li class="chapter" data-level="35" data-path="section-1.html"><a href="section-1.html"><i class="fa fa-check"></i><b>35</b> )</a></li>
<li class="chapter" data-level="36" data-path="section-2.html"><a href="section-2.html"><i class="fa fa-check"></i><b>36</b> )</a></li>
<li class="chapter" data-level="37" data-path="seems-like-a-good-res.html"><a href="seems-like-a-good-res.html"><i class="fa fa-check"></i><b>37</b> 1.4 seems like a good res</a></li>
<li class="chapter" data-level="38" data-path="project-values-back-via-knn-classifier-fnnknn.html"><a href="project-values-back-via-knn-classifier-fnnknn.html"><i class="fa fa-check"></i><b>38</b> project values back via kNN classifier (FNN::knn)</a></li>
<li class="chapter" data-level="39" data-path="spatplot2dg-cell_color-major_clus.html"><a href="spatplot2dg-cell_color-major_clus.html"><i class="fa fa-check"></i><b>39</b> spatPlot2D(g, cell_color = “major_clus”,</a></li>
<li class="chapter" data-level="40" data-path="point_size-0.html"><a href="point_size-0.html"><i class="fa fa-check"></i><b>40</b> point_size = 0.1,</a></li>
<li class="chapter" data-level="41" data-path="point_shape-no_border.html"><a href="point_shape-no_border.html"><i class="fa fa-check"></i><b>41</b> point_shape = “no_border”,</a></li>
<li class="chapter" data-level="42" data-path="cell_color_code-clus_colorc1-1017-29c1359-14-17-11.html"><a href="cell_color_code-clus_colorc1-1017-29c1359-14-17-11.html"><i class="fa fa-check"></i><b>42</b> cell_color_code = clus_color[c(1, 10:17, 2:9)][c(1:3,5:9, 14, 17, 11)],</a></li>
<li class="chapter" data-level="43" data-path="background-black.html"><a href="background-black.html"><i class="fa fa-check"></i><b>43</b> background = “black”,</a></li>
<li class="chapter" data-level="44" data-path="save_param-list-1.html"><a href="save_param-list-1.html"><i class="fa fa-check"></i><b>44</b> save_param = list(</a></li>
<li class="chapter" data-level="45" data-path="base_width-15.html"><a href="base_width-15.html"><i class="fa fa-check"></i><b>45</b> base_width = 15,</a></li>
<li class="chapter" data-level="46" data-path="base_height-6.html"><a href="base_height-6.html"><i class="fa fa-check"></i><b>46</b> base_height = 6,</a></li>
<li class="chapter" data-level="47" data-path="save_name-spat_major.html"><a href="save_name-spat_major.html"><i class="fa fa-check"></i><b>47</b> save_name = “spat_major”</a></li>
<li class="chapter" data-level="48" data-path="section-3.html"><a href="section-3.html"><i class="fa fa-check"></i><b>48</b> ))</a></li>
<li class="chapter" data-level="49" data-path="niches.html"><a href="niches.html"><i class="fa fa-check"></i><b>49</b> niches</a></li>
<li class="chapter" data-level="50" data-path="visualize-niche-level-enrichment-for-leiden-cluster-3-in-spat_unit-cell.html"><a href="visualize-niche-level-enrichment-for-leiden-cluster-3-in-spat_unit-cell.html"><i class="fa fa-check"></i><b>50</b> visualize niche-level enrichment for leiden cluster 3 in spat_unit “cell”</a></li>
<li class="chapter" data-level="51" data-path="these-enrichments-are-essentially-a-measure-of-how-many-cells-of-each.html"><a href="these-enrichments-are-essentially-a-measure-of-how-many-cells-of-each.html"><i class="fa fa-check"></i><b>51</b> These enrichments are essentially a measure of how many cells of each</a></li>
<li class="chapter" data-level="52" data-path="leiden-cluster-exist-in-the-local-region.html"><a href="leiden-cluster-exist-in-the-local-region.html"><i class="fa fa-check"></i><b>52</b> leiden cluster exist in the local region</a></li>
<li class="chapter" data-level="53" data-path="section-4.html"><a href="section-4.html"><i class="fa fa-check"></i><b>53</b> </a></li>
<li class="chapter" data-level="54" data-path="using-kmeans-we-can-classify-each-cell-by-its-niche-leiden-cluster-proportions.html"><a href="using-kmeans-we-can-classify-each-cell-by-its-niche-leiden-cluster-proportions.html"><i class="fa fa-check"></i><b>54</b> Using kmeans, we can classify each cell by its niche leiden cluster proportions</a></li>
<li class="chapter" data-level="55" data-path="add-kmeans-clustering-of-niches-to-cell-metadata.html"><a href="add-kmeans-clustering-of-niches-to-cell-metadata.html"><i class="fa fa-check"></i><b>55</b> add kmeans clustering of niches to cell metadata</a></li>
<li class="chapter" data-level="56" data-path="spatially-visualize-the-niches.html"><a href="spatially-visualize-the-niches.html"><i class="fa fa-check"></i><b>56</b> Spatially visualize the niches</a></li>
<li class="chapter" data-level="57" data-path="gene-enrichment.html"><a href="gene-enrichment.html"><i class="fa fa-check"></i><b>57</b> Gene enrichment</a></li>
<li class="chapter" data-level="58" data-path="violinplot.html"><a href="violinplot.html"><i class="fa fa-check"></i><b>58</b> violinplot</a></li>
<li class="chapter" data-level="59" data-path="spatial-proteomics-multiplex-if.html"><a href="spatial-proteomics-multiplex-if.html"><i class="fa fa-check"></i><b>59</b> Spatial proteomics: multiplex IF</a>
<ul>
<li class="chapter" data-level="59.1" data-path="spatial-proteomics-multiplex-if.html"><a href="spatial-proteomics-multiplex-if.html#read-in-raw-data-2"><i class="fa fa-check"></i><b>59.1</b> Read in raw data</a></li>
<li class="chapter" data-level="59.2" data-path="spatial-proteomics-multiplex-if.html"><a href="spatial-proteomics-multiplex-if.html#overlap-intensity-workflows"><i class="fa fa-check"></i><b>59.2</b> Overlap intensity &amp; workflows</a></li>
</ul></li>
<li class="chapter" data-level="60" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html"><i class="fa fa-check"></i><b>60</b> Working with multiple samples</a>
<ul>
<li class="chapter" data-level="60.1" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#create-individual-giotto-objects"><i class="fa fa-check"></i><b>60.1</b> Create individual giotto objects</a></li>
<li class="chapter" data-level="60.2" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#join-giotto-objects"><i class="fa fa-check"></i><b>60.2</b> Join Giotto Objects</a></li>
<li class="chapter" data-level="60.3" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#perform-harmony-and-default-workflows"><i class="fa fa-check"></i><b>60.3</b> Perform Harmony and default workflows</a></li>
<li class="chapter" data-level="60.4" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#visualizations-1"><i class="fa fa-check"></i><b>60.4</b> Visualizations</a></li>
</ul></li>
<li class="chapter" data-level="61" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html"><i class="fa fa-check"></i><b>61</b> Spatial multi-modal analysis</a>
<ul>
<li class="chapter" data-level="61.1" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#spatial-transforms"><i class="fa fa-check"></i><b>61.1</b> Spatial transforms</a>
<ul>
<li class="chapter" data-level="61.1.1" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#spatial-transforms-1"><i class="fa fa-check"></i><b>61.1.1</b> Spatial transforms:</a></li>
<li class="chapter" data-level="61.1.2" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#spatial-utilities"><i class="fa fa-check"></i><b>61.1.2</b> Spatial utilities:</a></li>
<li class="chapter" data-level="61.1.3" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#spatial-classes"><i class="fa fa-check"></i><b>61.1.3</b> Spatial classes:</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="62" data-path="image-transforms.html"><a href="image-transforms.html"><i class="fa fa-check"></i><b>62</b> Image transforms</a>
<ul>
<li class="chapter" data-level="62.1" data-path="image-transforms.html"><a href="image-transforms.html#co-registration-of-datasets"><i class="fa fa-check"></i><b>62.1</b> Co-registration of datasets</a></li>
<li class="chapter" data-level="62.2" data-path="image-transforms.html"><a href="image-transforms.html#examples-in-giotto-suite-manuscript"><i class="fa fa-check"></i><b>62.2</b> Examples in giotto suite manuscript</a></li>
</ul></li>
<li class="chapter" data-level="63" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html"><i class="fa fa-check"></i><b>63</b> Multi-omics integration</a>
<ul>
<li class="chapter" data-level="63.1" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#introduction-to-the-spatial-dataset-1"><i class="fa fa-check"></i><b>63.1</b> Introduction to the spatial dataset</a></li>
<li class="chapter" data-level="63.2" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#download-dataset-1"><i class="fa fa-check"></i><b>63.2</b> Download dataset</a></li>
<li class="chapter" data-level="63.3" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#create-the-giotto-object-1"><i class="fa fa-check"></i><b>63.3</b> Create the Giotto object</a></li>
<li class="chapter" data-level="63.4" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#subset-on-spots-that-were-covered-by-tissue-1"><i class="fa fa-check"></i><b>63.4</b> Subset on spots that were covered by tissue</a></li>
<li class="chapter" data-level="63.5" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#rna-processing"><i class="fa fa-check"></i><b>63.5</b> RNA processing</a></li>
<li class="chapter" data-level="63.6" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#protein-processing"><i class="fa fa-check"></i><b>63.6</b> Protein processing</a></li>
<li class="chapter" data-level="63.7" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#multi-omics-integration-1"><i class="fa fa-check"></i><b>63.7</b> Multi-omics integration</a></li>
<li class="chapter" data-level="63.8" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#session-info-2"><i class="fa fa-check"></i><b>63.8</b> Session info</a></li>
</ul></li>
<li class="chapter" data-level="64" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html"><i class="fa fa-check"></i><b>64</b> Interoperability with other frameworks</a>
<ul>
<li class="chapter" data-level="64.1" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#load-giotto-object"><i class="fa fa-check"></i><b>64.1</b> Load Giotto object</a></li>
<li class="chapter" data-level="64.2" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#spatialexperiment"><i class="fa fa-check"></i><b>64.2</b> SpatialExperiment</a>
<ul>
<li class="chapter" data-level="64.2.1" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#convert-giotto-object-to-spatialexperiment-object"><i class="fa fa-check"></i><b>64.2.1</b> Convert Giotto Object to SpatialExperiment Object</a></li>
</ul></li>
<li class="chapter" data-level="64.3" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#identify-top-spatially-variable-genes-with-nnsvg"><i class="fa fa-check"></i><b>64.3</b> Identify top spatially variable genes with nnSVG</a>
<ul>
<li class="chapter" data-level="64.3.1" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#conversion-of-spatialexperiment-object-back-to-giotto"><i class="fa fa-check"></i><b>64.3.1</b> Conversion of SpatialExperiment object back to Giotto</a></li>
<li class="chapter" data-level="64.3.2" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#plotting-top-genes-from-nnsvg-in-giotto"><i class="fa fa-check"></i><b>64.3.2</b> Plotting top genes from nnSVG in Giotto</a></li>
</ul></li>
<li class="chapter" data-level="64.4" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#seurat"><i class="fa fa-check"></i><b>64.4</b> Seurat</a>
<ul>
<li class="chapter" data-level="64.4.1" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#giotto-to-seurat"><i class="fa fa-check"></i><b>64.4.1</b> Giotto to Seurat</a></li>
</ul></li>
<li class="chapter" data-level="64.5" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#anndataspatialdata"><i class="fa fa-check"></i><b>64.5</b> AnnData/SpatialData</a></li>
</ul></li>
<li class="chapter" data-level="65" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html"><i class="fa fa-check"></i><b>65</b> Interoperability with isolated tools</a>
<ul>
<li class="chapter" data-level="65.1" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#spatial-niche-trajectory-analysis"><i class="fa fa-check"></i><b>65.1</b> Spatial niche trajectory analysis</a>
<ul>
<li class="chapter" data-level="65.1.1" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#prepare"><i class="fa fa-check"></i><b>65.1.1</b> Prepare</a></li>
<li class="chapter" data-level="65.1.2" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#running"><i class="fa fa-check"></i><b>65.1.2</b> Running</a></li>
<li class="chapter" data-level="65.1.3" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#visualization"><i class="fa fa-check"></i><b>65.1.3</b> Visualization</a></li>
<li class="chapter" data-level="65.1.4" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#section-5"><i class="fa fa-check"></i><b>65.1.4</b> </a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="66" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html"><i class="fa fa-check"></i><b>66</b> Interactivity with the R/Spatial ecosystem</a>
<ul>
<li class="chapter" data-level="66.1" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#kriging"><i class="fa fa-check"></i><b>66.1</b> Kriging</a></li>
</ul></li>
<li class="chapter" data-level="67" data-path="contributing-to-giotto.html"><a href="contributing-to-giotto.html"><i class="fa fa-check"></i><b>67</b> Contributing to Giotto</a>
<ul>
<li class="chapter" data-level="67.1" data-path="contributing-to-giotto.html"><a href="contributing-to-giotto.html#contribution-guideline"><i class="fa fa-check"></i><b>67.1</b> Contribution guideline</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Workshop: Spatial multi-omics data analysis with Giotto Suite</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="visium-part-i" class="section level1 hasAnchor" number="6">
<h1><span class="header-section-number">6</span> Visium Part I<a href="visium-part-i.html#visium-part-i" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Joselyn Cristina Chávez Fuentes</p>
<p>August 5th 2024</p>
<div id="introduction-to-the-spatial-dataset" class="section level2 hasAnchor" number="6.1">
<h2><span class="header-section-number">6.1</span> Introduction to the spatial dataset<a href="visium-part-i.html#introduction-to-the-spatial-dataset" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The Visium brain data to run this tutorial can be found <a href="https://support.10xgenomics.com/spatial-gene-expression/datasets/1.1.0/V1_Adult_Mouse_Brain">here</a></p>
</div>
<div id="download-dataset" class="section level2 hasAnchor" number="6.2">
<h2><span class="header-section-number">6.2</span> Download dataset<a href="visium-part-i.html#download-dataset" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>You need to download the expression matrix and spatial information by running these commands:</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="visium-part-i.html#cb9-1" tabindex="-1"></a><span class="fu">dir.create</span>(<span class="st">&quot;data&quot;</span>)</span>
<span id="cb9-2"><a href="visium-part-i.html#cb9-2" tabindex="-1"></a></span>
<span id="cb9-3"><a href="visium-part-i.html#cb9-3" tabindex="-1"></a><span class="fu">download.file</span>(<span class="at">url =</span> <span class="st">&quot;https://cf.10xgenomics.com/samples/spatial-exp/1.1.0/V1_Adult_Mouse_Brain/V1_Adult_Mouse_Brain_raw_feature_bc_matrix.tar.gz&quot;</span>,</span>
<span id="cb9-4"><a href="visium-part-i.html#cb9-4" tabindex="-1"></a>              <span class="at">destfile =</span> <span class="st">&quot;data/V1_Adult_Mouse_Brain_raw_feature_bc_matrix.tar.gz&quot;</span>)</span>
<span id="cb9-5"><a href="visium-part-i.html#cb9-5" tabindex="-1"></a></span>
<span id="cb9-6"><a href="visium-part-i.html#cb9-6" tabindex="-1"></a><span class="fu">download.file</span>(<span class="at">url =</span> <span class="st">&quot;https://cf.10xgenomics.com/samples/spatial-exp/1.1.0/V1_Adult_Mouse_Brain/V1_Adult_Mouse_Brain_spatial.tar.gz&quot;</span>,</span>
<span id="cb9-7"><a href="visium-part-i.html#cb9-7" tabindex="-1"></a>              <span class="at">destfile =</span> <span class="st">&quot;data/V1_Adult_Mouse_Brain_spatial.tar.gz&quot;</span>)</span></code></pre></div>
<p>After downloading, unzip the gz files. You should get the “raw_feature_bc_matrix” and “spatial” folders inside “data/”.</p>
</div>
<div id="create-the-giotto-object" class="section level2 hasAnchor" number="6.3">
<h2><span class="header-section-number">6.3</span> Create the Giotto object<a href="visium-part-i.html#create-the-giotto-object" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="visium-part-i.html#cb10-1" tabindex="-1"></a><span class="fu">library</span>(Giotto)</span>
<span id="cb10-2"><a href="visium-part-i.html#cb10-2" tabindex="-1"></a></span>
<span id="cb10-3"><a href="visium-part-i.html#cb10-3" tabindex="-1"></a><span class="do">## Set instructions</span></span>
<span id="cb10-4"><a href="visium-part-i.html#cb10-4" tabindex="-1"></a>results_folder <span class="ot">&lt;-</span> <span class="st">&quot;results/&quot;</span></span>
<span id="cb10-5"><a href="visium-part-i.html#cb10-5" tabindex="-1"></a></span>
<span id="cb10-6"><a href="visium-part-i.html#cb10-6" tabindex="-1"></a>python_path <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb10-7"><a href="visium-part-i.html#cb10-7" tabindex="-1"></a></span>
<span id="cb10-8"><a href="visium-part-i.html#cb10-8" tabindex="-1"></a>instructions <span class="ot">&lt;-</span> <span class="fu">createGiottoInstructions</span>(</span>
<span id="cb10-9"><a href="visium-part-i.html#cb10-9" tabindex="-1"></a>    <span class="at">save_dir =</span> results_folder,</span>
<span id="cb10-10"><a href="visium-part-i.html#cb10-10" tabindex="-1"></a>    <span class="at">save_plot =</span> <span class="cn">TRUE</span>,</span>
<span id="cb10-11"><a href="visium-part-i.html#cb10-11" tabindex="-1"></a>    <span class="at">show_plot =</span> <span class="cn">FALSE</span>,</span>
<span id="cb10-12"><a href="visium-part-i.html#cb10-12" tabindex="-1"></a>    <span class="at">return_plot =</span> <span class="cn">FALSE</span>,</span>
<span id="cb10-13"><a href="visium-part-i.html#cb10-13" tabindex="-1"></a>    <span class="at">python_path =</span> python_path</span>
<span id="cb10-14"><a href="visium-part-i.html#cb10-14" tabindex="-1"></a>)</span>
<span id="cb10-15"><a href="visium-part-i.html#cb10-15" tabindex="-1"></a></span>
<span id="cb10-16"><a href="visium-part-i.html#cb10-16" tabindex="-1"></a><span class="do">## Provide the path to the visium folder</span></span>
<span id="cb10-17"><a href="visium-part-i.html#cb10-17" tabindex="-1"></a>data_path <span class="ot">&lt;-</span> <span class="st">&quot;data&quot;</span></span>
<span id="cb10-18"><a href="visium-part-i.html#cb10-18" tabindex="-1"></a></span>
<span id="cb10-19"><a href="visium-part-i.html#cb10-19" tabindex="-1"></a><span class="do">## Create object directly from the visium folder</span></span>
<span id="cb10-20"><a href="visium-part-i.html#cb10-20" tabindex="-1"></a>visium_brain <span class="ot">&lt;-</span> <span class="fu">createGiottoVisiumObject</span>(</span>
<span id="cb10-21"><a href="visium-part-i.html#cb10-21" tabindex="-1"></a>    <span class="at">visium_dir =</span> data_path,</span>
<span id="cb10-22"><a href="visium-part-i.html#cb10-22" tabindex="-1"></a>    <span class="at">expr_data =</span> <span class="st">&quot;raw&quot;</span>,</span>
<span id="cb10-23"><a href="visium-part-i.html#cb10-23" tabindex="-1"></a>    <span class="at">png_name =</span> <span class="st">&quot;tissue_lowres_image.png&quot;</span>,</span>
<span id="cb10-24"><a href="visium-part-i.html#cb10-24" tabindex="-1"></a>    <span class="at">gene_column_index =</span> <span class="dv">2</span>,</span>
<span id="cb10-25"><a href="visium-part-i.html#cb10-25" tabindex="-1"></a>    <span class="at">instructions =</span> instructions</span>
<span id="cb10-26"><a href="visium-part-i.html#cb10-26" tabindex="-1"></a>)</span></code></pre></div>
</div>
<div id="subset-on-spots-that-were-covered-by-tissue" class="section level2 hasAnchor" number="6.4">
<h2><span class="header-section-number">6.4</span> Subset on spots that were covered by tissue<a href="visium-part-i.html#subset-on-spots-that-were-covered-by-tissue" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="visium-part-i.html#cb11-1" tabindex="-1"></a><span class="fu">spatPlot2D</span>(</span>
<span id="cb11-2"><a href="visium-part-i.html#cb11-2" tabindex="-1"></a>    <span class="at">gobject =</span> visium_brain,</span>
<span id="cb11-3"><a href="visium-part-i.html#cb11-3" tabindex="-1"></a>    <span class="at">cell_color =</span> <span class="st">&quot;in_tissue&quot;</span>,</span>
<span id="cb11-4"><a href="visium-part-i.html#cb11-4" tabindex="-1"></a>    <span class="at">point_size =</span> <span class="dv">2</span>,</span>
<span id="cb11-5"><a href="visium-part-i.html#cb11-5" tabindex="-1"></a>    <span class="at">cell_color_code =</span> <span class="fu">c</span>(<span class="st">&quot;0&quot;</span> <span class="ot">=</span> <span class="st">&quot;lightgrey&quot;</span>, <span class="st">&quot;1&quot;</span> <span class="ot">=</span> <span class="st">&quot;blue&quot;</span>),</span>
<span id="cb11-6"><a href="visium-part-i.html#cb11-6" tabindex="-1"></a>    <span class="at">show_image =</span> <span class="cn">TRUE</span>,</span>
<span id="cb11-7"><a href="visium-part-i.html#cb11-7" tabindex="-1"></a>    <span class="at">image_name =</span> <span class="st">&quot;image&quot;</span></span>
<span id="cb11-8"><a href="visium-part-i.html#cb11-8" tabindex="-1"></a>)</span></code></pre></div>
<p><img src="img/01_session5/0-spatPlot2D.png" width="80%" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="visium-part-i.html#cb12-1" tabindex="-1"></a>metadata <span class="ot">&lt;-</span> <span class="fu">getCellMetadata</span>(<span class="at">gobject =</span> visium_brain,</span>
<span id="cb12-2"><a href="visium-part-i.html#cb12-2" tabindex="-1"></a>                            <span class="at">output =</span> <span class="st">&quot;data.table&quot;</span>)</span>
<span id="cb12-3"><a href="visium-part-i.html#cb12-3" tabindex="-1"></a></span>
<span id="cb12-4"><a href="visium-part-i.html#cb12-4" tabindex="-1"></a>in_tissue_barcodes <span class="ot">&lt;-</span> metadata[in_tissue <span class="sc">==</span> <span class="dv">1</span>]<span class="sc">$</span>cell_ID</span>
<span id="cb12-5"><a href="visium-part-i.html#cb12-5" tabindex="-1"></a></span>
<span id="cb12-6"><a href="visium-part-i.html#cb12-6" tabindex="-1"></a>visium_brain <span class="ot">&lt;-</span> <span class="fu">subsetGiotto</span>(<span class="at">gobject =</span> visium_brain,</span>
<span id="cb12-7"><a href="visium-part-i.html#cb12-7" tabindex="-1"></a>                             <span class="at">cell_ids =</span> in_tissue_barcodes)</span></code></pre></div>
</div>
<div id="quality-control" class="section level2 hasAnchor" number="6.5">
<h2><span class="header-section-number">6.5</span> Quality control<a href="visium-part-i.html#quality-control" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ul>
<li>Statistics</li>
</ul>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="visium-part-i.html#cb13-1" tabindex="-1"></a>visium_brain_statistics <span class="ot">&lt;-</span> <span class="fu">addStatistics</span>(<span class="at">gobject =</span> visium_brain,</span>
<span id="cb13-2"><a href="visium-part-i.html#cb13-2" tabindex="-1"></a>                                         <span class="at">expression_values =</span> <span class="st">&quot;raw&quot;</span>)</span>
<span id="cb13-3"><a href="visium-part-i.html#cb13-3" tabindex="-1"></a></span>
<span id="cb13-4"><a href="visium-part-i.html#cb13-4" tabindex="-1"></a><span class="do">## visualize</span></span>
<span id="cb13-5"><a href="visium-part-i.html#cb13-5" tabindex="-1"></a><span class="fu">spatPlot2D</span>(<span class="at">gobject =</span> visium_brain_statistics, </span>
<span id="cb13-6"><a href="visium-part-i.html#cb13-6" tabindex="-1"></a>           <span class="at">cell_color =</span> <span class="st">&quot;nr_feats&quot;</span>, </span>
<span id="cb13-7"><a href="visium-part-i.html#cb13-7" tabindex="-1"></a>           <span class="at">color_as_factor =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<p><img src="img/01_session5/1-spatPlot2D.png" width="80%" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="visium-part-i.html#cb14-1" tabindex="-1"></a><span class="fu">filterDistributions</span>(<span class="at">gobject =</span> visium_brain_statistics, </span>
<span id="cb14-2"><a href="visium-part-i.html#cb14-2" tabindex="-1"></a>                    <span class="at">detection =</span> <span class="st">&quot;cells&quot;</span>)</span></code></pre></div>
<p><img src="img/01_session5/2-filterDistributions.png" width="50%" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="visium-part-i.html#cb15-1" tabindex="-1"></a><span class="fu">filterDistributions</span>(<span class="at">gobject =</span> visium_brain_statistics, </span>
<span id="cb15-2"><a href="visium-part-i.html#cb15-2" tabindex="-1"></a>                    <span class="at">detection =</span> <span class="st">&quot;feats&quot;</span>)</span></code></pre></div>
<p><img src="img/01_session5/3-filterDistributions.png" width="50%" style="display: block; margin: auto;" /></p>
<p>filterCombinations() may be used to test how different filtering parameters will affect the number of cells and features in the filtered data:</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="visium-part-i.html#cb16-1" tabindex="-1"></a><span class="fu">filterCombinations</span>(<span class="at">gobject =</span> visium_brain_statistics,</span>
<span id="cb16-2"><a href="visium-part-i.html#cb16-2" tabindex="-1"></a>                   <span class="at">expression_thresholds =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>, <span class="dv">3</span>),</span>
<span id="cb16-3"><a href="visium-part-i.html#cb16-3" tabindex="-1"></a>                   <span class="at">feat_det_in_min_cells =</span> <span class="fu">c</span>(<span class="dv">50</span>, <span class="dv">100</span>, <span class="dv">200</span>),</span>
<span id="cb16-4"><a href="visium-part-i.html#cb16-4" tabindex="-1"></a>                   <span class="at">min_det_feats_per_cell =</span> <span class="fu">c</span>(<span class="dv">500</span>, <span class="dv">1000</span>, <span class="dv">1500</span>))</span></code></pre></div>
<p><img src="img/01_session5/4-filterCombinations.png" width="50%" style="display: block; margin: auto;" /></p>
</div>
<div id="filtering" class="section level2 hasAnchor" number="6.6">
<h2><span class="header-section-number">6.6</span> Filtering<a href="visium-part-i.html#filtering" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="visium-part-i.html#cb17-1" tabindex="-1"></a>visium_brain <span class="ot">&lt;-</span> <span class="fu">filterGiotto</span>(</span>
<span id="cb17-2"><a href="visium-part-i.html#cb17-2" tabindex="-1"></a>    <span class="at">gobject =</span> visium_brain,</span>
<span id="cb17-3"><a href="visium-part-i.html#cb17-3" tabindex="-1"></a>    <span class="at">expression_threshold =</span> <span class="dv">1</span>,</span>
<span id="cb17-4"><a href="visium-part-i.html#cb17-4" tabindex="-1"></a>    <span class="at">feat_det_in_min_cells =</span> <span class="dv">50</span>,</span>
<span id="cb17-5"><a href="visium-part-i.html#cb17-5" tabindex="-1"></a>    <span class="at">min_det_feats_per_cell =</span> <span class="dv">1000</span>,</span>
<span id="cb17-6"><a href="visium-part-i.html#cb17-6" tabindex="-1"></a>    <span class="at">expression_values =</span> <span class="st">&quot;raw&quot;</span>,</span>
<span id="cb17-7"><a href="visium-part-i.html#cb17-7" tabindex="-1"></a>    <span class="at">verbose =</span> <span class="cn">TRUE</span></span>
<span id="cb17-8"><a href="visium-part-i.html#cb17-8" tabindex="-1"></a>)</span></code></pre></div>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="visium-part-i.html#cb18-1" tabindex="-1"></a>Feature type<span class="sc">:</span>  rna </span>
<span id="cb18-2"><a href="visium-part-i.html#cb18-2" tabindex="-1"></a>Number of cells removed<span class="sc">:</span>  <span class="dv">4</span>  out of  <span class="dv">2702</span> </span>
<span id="cb18-3"><a href="visium-part-i.html#cb18-3" tabindex="-1"></a>Number of feats removed<span class="sc">:</span>  <span class="dv">7311</span>  out of  <span class="dv">22125</span> </span></code></pre></div>
</div>
<div id="normalization" class="section level2 hasAnchor" number="6.7">
<h2><span class="header-section-number">6.7</span> Normalization<a href="visium-part-i.html#normalization" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="visium-part-i.html#cb19-1" tabindex="-1"></a>visium_brain <span class="ot">&lt;-</span> <span class="fu">normalizeGiotto</span>(</span>
<span id="cb19-2"><a href="visium-part-i.html#cb19-2" tabindex="-1"></a>    <span class="at">gobject =</span> visium_brain,</span>
<span id="cb19-3"><a href="visium-part-i.html#cb19-3" tabindex="-1"></a>    <span class="at">scalefactor =</span> <span class="dv">6000</span>,</span>
<span id="cb19-4"><a href="visium-part-i.html#cb19-4" tabindex="-1"></a>    <span class="at">verbose =</span> <span class="cn">TRUE</span></span>
<span id="cb19-5"><a href="visium-part-i.html#cb19-5" tabindex="-1"></a>)</span></code></pre></div>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="visium-part-i.html#cb20-1" tabindex="-1"></a>visium_brain <span class="ot">&lt;-</span> <span class="fu">addStatistics</span>(<span class="at">gobject =</span> visium_brain)</span>
<span id="cb20-2"><a href="visium-part-i.html#cb20-2" tabindex="-1"></a></span>
<span id="cb20-3"><a href="visium-part-i.html#cb20-3" tabindex="-1"></a><span class="do">## visualize</span></span>
<span id="cb20-4"><a href="visium-part-i.html#cb20-4" tabindex="-1"></a><span class="fu">spatPlot2D</span>(<span class="at">gobject =</span> visium_brain, </span>
<span id="cb20-5"><a href="visium-part-i.html#cb20-5" tabindex="-1"></a>           <span class="at">cell_color =</span> <span class="st">&quot;nr_feats&quot;</span>, </span>
<span id="cb20-6"><a href="visium-part-i.html#cb20-6" tabindex="-1"></a>           <span class="at">color_as_factor =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<p><img src="img/01_session5/5-spatPlot2D.png" width="80%" style="display: block; margin: auto;" /></p>
</div>
<div id="feature-selection" class="section level2 hasAnchor" number="6.8">
<h2><span class="header-section-number">6.8</span> Feature selection<a href="visium-part-i.html#feature-selection" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="highly-variable-features" class="section level3 hasAnchor" number="6.8.1">
<h3><span class="header-section-number">6.8.1</span> Highly Variable Features:<a href="visium-part-i.html#highly-variable-features" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Calculating Highly Variable Features (HVF) is necessary to identify genes (or features) that display significant variability across spatial locations. There are a few methods to choose from:</p>
<ul>
<li>loess regression</li>
</ul>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="visium-part-i.html#cb21-1" tabindex="-1"></a>visium_brain <span class="ot">&lt;-</span> <span class="fu">calculateHVF</span>(<span class="at">gobject =</span> visium_brain, </span>
<span id="cb21-2"><a href="visium-part-i.html#cb21-2" tabindex="-1"></a>                             <span class="at">method =</span> <span class="st">&quot;cov_loess&quot;</span>,</span>
<span id="cb21-3"><a href="visium-part-i.html#cb21-3" tabindex="-1"></a>                             <span class="at">save_plot =</span> <span class="cn">TRUE</span>,</span>
<span id="cb21-4"><a href="visium-part-i.html#cb21-4" tabindex="-1"></a>                             <span class="at">default_save_name =</span> <span class="st">&quot;HVFplot_loess&quot;</span>)</span></code></pre></div>
<p><img src="img/01_session5/6-HVFplot_loess.png" width="80%" style="display: block; margin: auto;" /></p>
<ul>
<li>pearson residuals</li>
</ul>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="visium-part-i.html#cb22-1" tabindex="-1"></a>visium_brain <span class="ot">&lt;-</span> <span class="fu">calculateHVF</span>(<span class="at">gobject =</span> visium_brain, </span>
<span id="cb22-2"><a href="visium-part-i.html#cb22-2" tabindex="-1"></a>                             <span class="at">method =</span> <span class="st">&quot;var_p_resid&quot;</span>,</span>
<span id="cb22-3"><a href="visium-part-i.html#cb22-3" tabindex="-1"></a>                             <span class="at">save_plot =</span> <span class="cn">TRUE</span>,</span>
<span id="cb22-4"><a href="visium-part-i.html#cb22-4" tabindex="-1"></a>                             <span class="at">default_save_name =</span> <span class="st">&quot;HVFplot_pearson&quot;</span>)</span></code></pre></div>
<p><img src="img/01_session5/7-HVFplot_pearson.png" width="80%" style="display: block; margin: auto;" /></p>
<ul>
<li>binned</li>
</ul>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="visium-part-i.html#cb23-1" tabindex="-1"></a>visium_brain <span class="ot">&lt;-</span> <span class="fu">calculateHVF</span>(<span class="at">gobject =</span> visium_brain, </span>
<span id="cb23-2"><a href="visium-part-i.html#cb23-2" tabindex="-1"></a>                             <span class="at">method =</span> <span class="st">&quot;cov_groups&quot;</span>,</span>
<span id="cb23-3"><a href="visium-part-i.html#cb23-3" tabindex="-1"></a>                             <span class="at">save_plot =</span> <span class="cn">TRUE</span>,</span>
<span id="cb23-4"><a href="visium-part-i.html#cb23-4" tabindex="-1"></a>                             <span class="at">default_save_name =</span> <span class="st">&quot;HVFplot_binned&quot;</span>)</span></code></pre></div>
<p><img src="img/01_session5/8-HVFplot_binned.png" width="80%" style="display: block; margin: auto;" /></p>
<p>Depending on the underlying distribution of the data:</p>
<ul>
<li><p>loess regression is used when the relationship between mean expression and variance is non-linear or can be described by a non-parametric model</p></li>
<li><p>pearson residuals are used when the relationship between mean expression and variance is linear or can be described by a parametric model</p></li>
<li><p>binned (covariance groups) are used when variability in gene expression differs across expression levels or spatial regions, without assuming a specific relationship between mean expression and variance</p></li>
</ul>
</div>
</div>
<div id="dimension-reduction" class="section level2 hasAnchor" number="6.9">
<h2><span class="header-section-number">6.9</span> Dimension Reduction<a href="visium-part-i.html#dimension-reduction" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="pca" class="section level3 hasAnchor" number="6.9.1">
<h3><span class="header-section-number">6.9.1</span> PCA<a href="visium-part-i.html#pca" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Principal Components Analysis (PCA) is applied to reduce the dimensionality of gene expression data by transforming it into principal components, which are linear combinations of genes ranked by the variance they explain, with the first components capturing the most variance.</p>
<ul>
<li>Default</li>
</ul>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="visium-part-i.html#cb24-1" tabindex="-1"></a>visium_brain <span class="ot">&lt;-</span> <span class="fu">runPCA</span>(<span class="at">gobject =</span> visium_brain)</span></code></pre></div>
<ul>
<li>Using specific features</li>
</ul>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="visium-part-i.html#cb25-1" tabindex="-1"></a>my_features <span class="ot">&lt;-</span> <span class="fu">head</span>(<span class="fu">getFeatureMetadata</span>(visium_brain, </span>
<span id="cb25-2"><a href="visium-part-i.html#cb25-2" tabindex="-1"></a>                                       <span class="at">output =</span> <span class="st">&quot;data.table&quot;</span>)<span class="sc">$</span>feat_ID, </span>
<span id="cb25-3"><a href="visium-part-i.html#cb25-3" tabindex="-1"></a>                    <span class="dv">1000</span>)</span>
<span id="cb25-4"><a href="visium-part-i.html#cb25-4" tabindex="-1"></a></span>
<span id="cb25-5"><a href="visium-part-i.html#cb25-5" tabindex="-1"></a>visium_brain <span class="ot">&lt;-</span> <span class="fu">runPCA</span>(<span class="at">gobject =</span> visium_brain,</span>
<span id="cb25-6"><a href="visium-part-i.html#cb25-6" tabindex="-1"></a>                       <span class="at">feats_to_use =</span> my_features,</span>
<span id="cb25-7"><a href="visium-part-i.html#cb25-7" tabindex="-1"></a>                       <span class="at">name =</span> <span class="st">&quot;custom_pca&quot;</span>)</span></code></pre></div>
<ul>
<li>Visualization</li>
</ul>
<p>Screeplot</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="visium-part-i.html#cb26-1" tabindex="-1"></a><span class="fu">screePlot</span>(<span class="at">gobject =</span> visium_brain, </span>
<span id="cb26-2"><a href="visium-part-i.html#cb26-2" tabindex="-1"></a>          <span class="at">ncp =</span> <span class="dv">30</span>)</span></code></pre></div>
<p><img src="img/01_session5/9-screePlot.png" width="80%" style="display: block; margin: auto;" /></p>
<p>PCA</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="visium-part-i.html#cb27-1" tabindex="-1"></a><span class="fu">plotPCA</span>(<span class="at">gobject =</span> visium_brain)</span></code></pre></div>
<p><img src="img/01_session5/10-PCA.png" width="80%" style="display: block; margin: auto;" /></p>
<p>Custom PCA</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="visium-part-i.html#cb28-1" tabindex="-1"></a><span class="fu">plotPCA</span>(<span class="at">gobject =</span> visium_brain,</span>
<span id="cb28-2"><a href="visium-part-i.html#cb28-2" tabindex="-1"></a>        <span class="at">dim_reduction_name =</span> <span class="st">&quot;custom_pca&quot;</span>)</span></code></pre></div>
<p><img src="img/01_session5/11-PCA.png" width="80%" style="display: block; margin: auto;" /></p>
<p>Unlike PCA, Unifold Manifold Approximation and Projection (UMAP) and t-Stochastic Neighbor Embedding (t-SNE) do not assume linearity. After running PCA, UMAP or t-SNE follows to further reduce PCs.</p>
</div>
<div id="umap" class="section level3 hasAnchor" number="6.9.2">
<h3><span class="header-section-number">6.9.2</span> UMAP<a href="visium-part-i.html#umap" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="visium-part-i.html#cb29-1" tabindex="-1"></a>visium_brain <span class="ot">&lt;-</span> <span class="fu">runUMAP</span>(visium_brain, </span>
<span id="cb29-2"><a href="visium-part-i.html#cb29-2" tabindex="-1"></a>                        <span class="at">dimensions_to_use =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>)</span></code></pre></div>
<ul>
<li>Visualization</li>
</ul>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="visium-part-i.html#cb30-1" tabindex="-1"></a><span class="fu">plotUMAP</span>(<span class="at">gobject =</span> visium_brain)</span></code></pre></div>
<p><img src="img/01_session5/12-UMAP.png" width="80%" style="display: block; margin: auto;" /></p>
</div>
<div id="t-sne" class="section level3 hasAnchor" number="6.9.3">
<h3><span class="header-section-number">6.9.3</span> t-SNE<a href="visium-part-i.html#t-sne" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="visium-part-i.html#cb31-1" tabindex="-1"></a>visium_brain <span class="ot">&lt;-</span> <span class="fu">runtSNE</span>(<span class="at">gobject =</span> visium_brain, </span>
<span id="cb31-2"><a href="visium-part-i.html#cb31-2" tabindex="-1"></a>                        <span class="at">dimensions_to_use =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>)</span></code></pre></div>
<ul>
<li>Visualization</li>
</ul>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="visium-part-i.html#cb32-1" tabindex="-1"></a><span class="fu">plotTSNE</span>(<span class="at">gobject =</span> visium_brain)</span></code></pre></div>
<p><img src="img/01_session5/13-tSNE.png" width="80%" style="display: block; margin: auto;" /></p>
<p>UMAP generally provides a more balanced view of spatial relationships and biological context, while t-SNE can be more effective in highlighting specific clusters or spatially proximal groups.</p>
</div>
</div>
<div id="clustering" class="section level2 hasAnchor" number="6.10">
<h2><span class="header-section-number">6.10</span> Clustering<a href="visium-part-i.html#clustering" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Shared Nearest Neighbor (sNN) emphasizes local coherence by considering spots that share nearest neighbors, highlighting cohesive spatial structures in the data. k-Nearest Neighbors (kNN) defines spatial relationships based on direct proximity, focusing on the nearest neighbors of each spot without considering shared neighborhood patterns as explicitly as sNN.</p>
<ul>
<li>Create a sNN network (default)</li>
</ul>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="visium-part-i.html#cb33-1" tabindex="-1"></a>visium_brain <span class="ot">&lt;-</span> <span class="fu">createNearestNetwork</span>(<span class="at">gobject =</span> visium_brain, </span>
<span id="cb33-2"><a href="visium-part-i.html#cb33-2" tabindex="-1"></a>                                     <span class="at">dimensions_to_use =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>, </span>
<span id="cb33-3"><a href="visium-part-i.html#cb33-3" tabindex="-1"></a>                                     <span class="at">k =</span> <span class="dv">15</span>)</span></code></pre></div>
<ul>
<li>Create a kNN network</li>
</ul>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="visium-part-i.html#cb34-1" tabindex="-1"></a>visium_brain <span class="ot">&lt;-</span> <span class="fu">createNearestNetwork</span>(<span class="at">gobject =</span> visium_brain, </span>
<span id="cb34-2"><a href="visium-part-i.html#cb34-2" tabindex="-1"></a>                                     <span class="at">dimensions_to_use =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>, </span>
<span id="cb34-3"><a href="visium-part-i.html#cb34-3" tabindex="-1"></a>                                     <span class="at">k =</span> <span class="dv">15</span>,</span>
<span id="cb34-4"><a href="visium-part-i.html#cb34-4" tabindex="-1"></a>                                     <span class="at">type =</span> <span class="st">&quot;kNN&quot;</span>)</span></code></pre></div>
<div id="calculate-leiden-clustering" class="section level3 hasAnchor" number="6.10.1">
<h3><span class="header-section-number">6.10.1</span> Calculate Leiden clustering<a href="visium-part-i.html#calculate-leiden-clustering" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="visium-part-i.html#cb35-1" tabindex="-1"></a>visium_brain <span class="ot">&lt;-</span> <span class="fu">doLeidenCluster</span>(<span class="at">gobject =</span> visium_brain, </span>
<span id="cb35-2"><a href="visium-part-i.html#cb35-2" tabindex="-1"></a>                                <span class="at">resolution =</span> <span class="fl">0.4</span>, </span>
<span id="cb35-3"><a href="visium-part-i.html#cb35-3" tabindex="-1"></a>                                <span class="at">n_iterations =</span> <span class="dv">1000</span>)</span></code></pre></div>
<ul>
<li>Visualization</li>
</ul>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="visium-part-i.html#cb36-1" tabindex="-1"></a><span class="fu">plotPCA</span>(<span class="at">gobject =</span> visium_brain,</span>
<span id="cb36-2"><a href="visium-part-i.html#cb36-2" tabindex="-1"></a>        <span class="at">cell_color =</span> <span class="st">&quot;leiden_clus&quot;</span>)</span></code></pre></div>
<p><img src="img/01_session5/14-PCA.png" width="80%" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="visium-part-i.html#cb37-1" tabindex="-1"></a><span class="fu">plotUMAP</span>(<span class="at">gobject =</span> visium_brain,</span>
<span id="cb37-2"><a href="visium-part-i.html#cb37-2" tabindex="-1"></a>         <span class="at">cell_color =</span> <span class="st">&quot;leiden_clus&quot;</span>, </span>
<span id="cb37-3"><a href="visium-part-i.html#cb37-3" tabindex="-1"></a>         <span class="at">show_NN_network =</span> <span class="cn">FALSE</span>, </span>
<span id="cb37-4"><a href="visium-part-i.html#cb37-4" tabindex="-1"></a>         <span class="at">point_size =</span> <span class="fl">2.5</span>)</span></code></pre></div>
<p><img src="img/01_session5/15-UMAP.png" width="80%" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb38-1"><a href="visium-part-i.html#cb38-1" tabindex="-1"></a><span class="fu">plotUMAP</span>(<span class="at">gobject =</span> visium_brain,</span>
<span id="cb38-2"><a href="visium-part-i.html#cb38-2" tabindex="-1"></a>         <span class="at">cell_color =</span> <span class="st">&quot;leiden_clus&quot;</span>, </span>
<span id="cb38-3"><a href="visium-part-i.html#cb38-3" tabindex="-1"></a>         <span class="at">show_NN_network =</span> <span class="cn">TRUE</span>, </span>
<span id="cb38-4"><a href="visium-part-i.html#cb38-4" tabindex="-1"></a>         <span class="at">point_size =</span> <span class="fl">2.5</span>)</span></code></pre></div>
<p><img src="img/01_session5/16-UMAP.png" width="80%" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="visium-part-i.html#cb39-1" tabindex="-1"></a><span class="fu">plotTSNE</span>(<span class="at">gobject =</span> visium_brain,</span>
<span id="cb39-2"><a href="visium-part-i.html#cb39-2" tabindex="-1"></a>         <span class="at">cell_color =</span> <span class="st">&quot;leiden_clus&quot;</span>,</span>
<span id="cb39-3"><a href="visium-part-i.html#cb39-3" tabindex="-1"></a>         <span class="at">point_size =</span> <span class="fl">2.5</span>)</span></code></pre></div>
<p><img src="img/01_session5/17-tSNE.png" width="80%" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="visium-part-i.html#cb40-1" tabindex="-1"></a><span class="fu">plotTSNE</span>(<span class="at">gobject =</span> visium_brain,</span>
<span id="cb40-2"><a href="visium-part-i.html#cb40-2" tabindex="-1"></a>         <span class="at">cell_color =</span> <span class="st">&quot;leiden_clus&quot;</span>,</span>
<span id="cb40-3"><a href="visium-part-i.html#cb40-3" tabindex="-1"></a>         <span class="at">point_size =</span> <span class="fl">2.5</span>,</span>
<span id="cb40-4"><a href="visium-part-i.html#cb40-4" tabindex="-1"></a>         <span class="at">show_NN_network =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p><img src="img/01_session5/18-tSNE.png" width="80%" style="display: block; margin: auto;" /></p>
<ul>
<li>Dimension plots grouped by cluster</li>
</ul>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="visium-part-i.html#cb41-1" tabindex="-1"></a><span class="fu">spatPlot2D</span>(visium_brain, </span>
<span id="cb41-2"><a href="visium-part-i.html#cb41-2" tabindex="-1"></a>           <span class="at">cell_color =</span> <span class="st">&quot;leiden_clus&quot;</span>)</span></code></pre></div>
<p><img src="img/01_session5/19-spatPlot2D.png" width="80%" style="display: block; margin: auto;" /></p>
</div>
<div id="calculate-louvain-clustering" class="section level3 hasAnchor" number="6.10.2">
<h3><span class="header-section-number">6.10.2</span> Calculate Louvain clustering<a href="visium-part-i.html#calculate-louvain-clustering" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="visium-part-i.html#cb42-1" tabindex="-1"></a>visium_brain <span class="ot">&lt;-</span> <span class="fu">doLouvainCluster</span>(visium_brain)</span></code></pre></div>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb43-1"><a href="visium-part-i.html#cb43-1" tabindex="-1"></a><span class="fu">spatPlot2D</span>(visium_brain, </span>
<span id="cb43-2"><a href="visium-part-i.html#cb43-2" tabindex="-1"></a>           <span class="at">cell_color =</span> <span class="st">&quot;louvain_clus&quot;</span>)</span></code></pre></div>
<p><img src="img/01_session5/20-spatPlot2D.png" width="80%" style="display: block; margin: auto;" /></p>
<p>Leiden clustering is generally preferred for larger and more complex datasets, where as Louvain clustering may be suitable for more smaller datasets or scenarios where computational efficiency is a priority.</p>
</div>
</div>
<div id="save-the-object" class="section level2 hasAnchor" number="6.11">
<h2><span class="header-section-number">6.11</span> Save the object<a href="visium-part-i.html#save-the-object" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="visium-part-i.html#cb44-1" tabindex="-1"></a><span class="fu">saveGiotto</span>(visium_brain, <span class="st">&quot;visium_brain_object&quot;</span>)</span></code></pre></div>
</div>
<div id="session-info" class="section level2 hasAnchor" number="6.12">
<h2><span class="header-section-number">6.12</span> Session info<a href="visium-part-i.html#session-info" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb45-1"><a href="visium-part-i.html#cb45-1" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code></pre></div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="creating-a-giotto-object.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="visium-part-ii.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
