<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>11 Working with multiple samples | Workshop: Spatial multi-omics data analysis with Giotto Suite</title>
  <meta name="description" content="11 Working with multiple samples | Workshop: Spatial multi-omics data analysis with Giotto Suite" />
  <meta name="generator" content="bookdown 0.40 and GitBook 2.6.7" />

  <meta property="og:title" content="11 Working with multiple samples | Workshop: Spatial multi-omics data analysis with Giotto Suite" />
  <meta property="og:type" content="book" />
  
  
  <meta name="github-repo" content="drieslab/giotto_workshop_2024" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="11 Working with multiple samples | Workshop: Spatial multi-omics data analysis with Giotto Suite" />
  
  
  

<meta name="author" content="Ruben Dries, Jiaji George Chen, Joselyn Cristina Chávez-Fuentes, Junxiang Xu ,Edward Ruiz, Jeff Sheridan, Iqra Amin, Wen Wang" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="spatial-proteomics-multiplex-if.html"/>
<link rel="next" href="spatial-multi-modal-analysis.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Giotto Suite Workshop 2024</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#instructors"><i class="fa fa-check"></i><b>1.1</b> Instructors</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#topics-and-schedule"><i class="fa fa-check"></i><b>1.2</b> Topics and Schedule:</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#license"><i class="fa fa-check"></i><b>1.3</b> License</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html"><i class="fa fa-check"></i><b>2</b> Spatial omics technologies</a>
<ul>
<li class="chapter" data-level="2.1" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#presentation"><i class="fa fa-check"></i><b>2.1</b> Presentation</a></li>
<li class="chapter" data-level="2.2" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#short-summary"><i class="fa fa-check"></i><b>2.2</b> Short summary</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#why-do-we-need-spatial-omics-technologies"><i class="fa fa-check"></i><b>2.2.1</b> Why do we need spatial omics technologies?</a></li>
<li class="chapter" data-level="2.2.2" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#what-is-spatial-omics"><i class="fa fa-check"></i><b>2.2.2</b> What is spatial omics?</a></li>
<li class="chapter" data-level="2.2.3" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#what-are-the-main-spatial-omics-technologies"><i class="fa fa-check"></i><b>2.2.3</b> What are the main spatial omics technologies?</a></li>
<li class="chapter" data-level="2.2.4" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#other-spatial-omics-atac-seq-cuttag-lipidomics-etc"><i class="fa fa-check"></i><b>2.2.4</b> Other Spatial omics: ATAC-seq, CUT&amp;Tag, lipidomics, etc</a></li>
<li class="chapter" data-level="2.2.5" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#what-are-the-different-types-of-spatial-downstream-analyses"><i class="fa fa-check"></i><b>2.2.5</b> What are the different types of spatial downstream analyses?</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html"><i class="fa fa-check"></i><b>3</b> Introduction to the Giotto package</a>
<ul>
<li class="chapter" data-level="3.1" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html#presentation-1"><i class="fa fa-check"></i><b>3.1</b> Presentation</a></li>
<li class="chapter" data-level="3.2" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html#ecosystem"><i class="fa fa-check"></i><b>3.2</b> Ecosystem</a></li>
<li class="chapter" data-level="3.3" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html#installation-python-environment"><i class="fa fa-check"></i><b>3.3</b> Installation + python environment</a></li>
<li class="chapter" data-level="3.4" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html#giotto-instructions"><i class="fa fa-check"></i><b>3.4</b> Giotto instructions</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html"><i class="fa fa-check"></i><b>4</b> Data formatting and Pre-processing</a>
<ul>
<li class="chapter" data-level="4.1" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#data-formats"><i class="fa fa-check"></i><b>4.1</b> Data formats</a></li>
<li class="chapter" data-level="4.2" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#pre-processing"><i class="fa fa-check"></i><b>4.2</b> Pre-processing</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html"><i class="fa fa-check"></i><b>5</b> Creating a Giotto object</a>
<ul>
<li class="chapter" data-level="5.0.1" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#used-packages"><i class="fa fa-check"></i><b>5.0.1</b> Used packages</a></li>
<li class="chapter" data-level="5.1" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#overview"><i class="fa fa-check"></i><b>5.1</b> Overview</a></li>
<li class="chapter" data-level="5.2" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#from-matrix-locations"><i class="fa fa-check"></i><b>5.2</b> From matrix + locations</a></li>
<li class="chapter" data-level="5.3" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#from-subcellular-raw-data-transcripts-or-images-polygons"><i class="fa fa-check"></i><b>5.3</b> From subcellular raw data (transcripts or images) + polygons</a></li>
<li class="chapter" data-level="5.4" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#from-piece-wise"><i class="fa fa-check"></i><b>5.4</b> From piece-wise</a></li>
<li class="chapter" data-level="5.5" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#using-convenience-functions-for-popular-technologies-vizgen-xenium-cosmx"><i class="fa fa-check"></i><b>5.5</b> Using convenience functions for popular technologies (Vizgen, Xenium, CosMx, …)</a></li>
<li class="chapter" data-level="5.6" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#spatial-plots"><i class="fa fa-check"></i><b>5.6</b> Spatial plots</a></li>
<li class="chapter" data-level="5.7" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#subsetting"><i class="fa fa-check"></i><b>5.7</b> Subsetting</a></li>
<li class="chapter" data-level="5.8" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#mini-objects-giottodata"><i class="fa fa-check"></i><b>5.8</b> Mini objects &amp; GiottoData</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="visium-part-i.html"><a href="visium-part-i.html"><i class="fa fa-check"></i><b>6</b> Visium Part I</a>
<ul>
<li class="chapter" data-level="6.1" data-path="visium-part-i.html"><a href="visium-part-i.html#introduction-to-the-spatial-dataset"><i class="fa fa-check"></i><b>6.1</b> Introduction to the spatial dataset</a></li>
<li class="chapter" data-level="6.2" data-path="visium-part-i.html"><a href="visium-part-i.html#download-dataset"><i class="fa fa-check"></i><b>6.2</b> Download dataset</a></li>
<li class="chapter" data-level="6.3" data-path="visium-part-i.html"><a href="visium-part-i.html#create-the-giotto-object"><i class="fa fa-check"></i><b>6.3</b> Create the Giotto object</a></li>
<li class="chapter" data-level="6.4" data-path="visium-part-i.html"><a href="visium-part-i.html#subset-on-spots-that-were-covered-by-tissue"><i class="fa fa-check"></i><b>6.4</b> Subset on spots that were covered by tissue</a></li>
<li class="chapter" data-level="6.5" data-path="visium-part-i.html"><a href="visium-part-i.html#quality-control"><i class="fa fa-check"></i><b>6.5</b> Quality control</a></li>
<li class="chapter" data-level="6.6" data-path="visium-part-i.html"><a href="visium-part-i.html#filtering"><i class="fa fa-check"></i><b>6.6</b> Filtering</a></li>
<li class="chapter" data-level="6.7" data-path="visium-part-i.html"><a href="visium-part-i.html#normalization"><i class="fa fa-check"></i><b>6.7</b> Normalization</a></li>
<li class="chapter" data-level="6.8" data-path="visium-part-i.html"><a href="visium-part-i.html#feature-selection"><i class="fa fa-check"></i><b>6.8</b> Feature selection</a>
<ul>
<li class="chapter" data-level="6.8.1" data-path="visium-part-i.html"><a href="visium-part-i.html#highly-variable-features"><i class="fa fa-check"></i><b>6.8.1</b> Highly Variable Features:</a></li>
</ul></li>
<li class="chapter" data-level="6.9" data-path="visium-part-i.html"><a href="visium-part-i.html#dimension-reduction"><i class="fa fa-check"></i><b>6.9</b> Dimension Reduction</a>
<ul>
<li class="chapter" data-level="6.9.1" data-path="visium-part-i.html"><a href="visium-part-i.html#pca"><i class="fa fa-check"></i><b>6.9.1</b> PCA</a></li>
<li class="chapter" data-level="6.9.2" data-path="visium-part-i.html"><a href="visium-part-i.html#umap"><i class="fa fa-check"></i><b>6.9.2</b> UMAP</a></li>
<li class="chapter" data-level="6.9.3" data-path="visium-part-i.html"><a href="visium-part-i.html#t-sne"><i class="fa fa-check"></i><b>6.9.3</b> t-SNE</a></li>
</ul></li>
<li class="chapter" data-level="6.10" data-path="visium-part-i.html"><a href="visium-part-i.html#clustering"><i class="fa fa-check"></i><b>6.10</b> Clustering</a>
<ul>
<li class="chapter" data-level="6.10.1" data-path="visium-part-i.html"><a href="visium-part-i.html#calculate-leiden-clustering"><i class="fa fa-check"></i><b>6.10.1</b> Calculate Leiden clustering</a></li>
<li class="chapter" data-level="6.10.2" data-path="visium-part-i.html"><a href="visium-part-i.html#calculate-louvain-clustering"><i class="fa fa-check"></i><b>6.10.2</b> Calculate Louvain clustering</a></li>
</ul></li>
<li class="chapter" data-level="6.11" data-path="visium-part-i.html"><a href="visium-part-i.html#save-the-object"><i class="fa fa-check"></i><b>6.11</b> Save the object</a></li>
<li class="chapter" data-level="6.12" data-path="visium-part-i.html"><a href="visium-part-i.html#session-info"><i class="fa fa-check"></i><b>6.12</b> Session info</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="visium-part-ii.html"><a href="visium-part-ii.html"><i class="fa fa-check"></i><b>7</b> Visium Part II</a>
<ul>
<li class="chapter" data-level="7.1" data-path="visium-part-ii.html"><a href="visium-part-ii.html#load-the-object"><i class="fa fa-check"></i><b>7.1</b> Load the object</a></li>
<li class="chapter" data-level="7.2" data-path="visium-part-ii.html"><a href="visium-part-ii.html#differential-expression"><i class="fa fa-check"></i><b>7.2</b> Differential expression</a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="visium-part-ii.html"><a href="visium-part-ii.html#gini-markers"><i class="fa fa-check"></i><b>7.2.1</b> Gini markers</a></li>
<li class="chapter" data-level="7.2.2" data-path="visium-part-ii.html"><a href="visium-part-ii.html#scran-markers"><i class="fa fa-check"></i><b>7.2.2</b> Scran markers</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="visium-part-ii.html"><a href="visium-part-ii.html#enrichment-deconvolution"><i class="fa fa-check"></i><b>7.3</b> Enrichment &amp; Deconvolution</a>
<ul>
<li class="chapter" data-level="7.3.1" data-path="visium-part-ii.html"><a href="visium-part-ii.html#pagerank"><i class="fa fa-check"></i><b>7.3.1</b> PAGE/Rank</a></li>
<li class="chapter" data-level="7.3.2" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatialdwls"><i class="fa fa-check"></i><b>7.3.2</b> SpatialDWLS</a></li>
</ul></li>
<li class="chapter" data-level="7.4" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatial-expression-patterns"><i class="fa fa-check"></i><b>7.4</b> Spatial expression patterns</a>
<ul>
<li class="chapter" data-level="7.4.1" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatial-variable-genes"><i class="fa fa-check"></i><b>7.4.1</b> Spatial variable genes</a></li>
<li class="chapter" data-level="7.4.2" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatial-co-expression-modules"><i class="fa fa-check"></i><b>7.4.2</b> Spatial co-expression modules</a></li>
</ul></li>
<li class="chapter" data-level="7.5" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatially-informed-clusters"><i class="fa fa-check"></i><b>7.5</b> Spatially informed clusters</a></li>
<li class="chapter" data-level="7.6" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatial-domains-hmrf"><i class="fa fa-check"></i><b>7.6</b> Spatial domains HMRF</a></li>
<li class="chapter" data-level="7.7" data-path="visium-part-ii.html"><a href="visium-part-ii.html#interactive-tools"><i class="fa fa-check"></i><b>7.7</b> Interactive tools</a></li>
<li class="chapter" data-level="7.8" data-path="visium-part-ii.html"><a href="visium-part-ii.html#session-info-1"><i class="fa fa-check"></i><b>7.8</b> Session info</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="visium-hd.html"><a href="visium-hd.html"><i class="fa fa-check"></i><b>8</b> Visium HD</a>
<ul>
<li class="chapter" data-level="8.1" data-path="visium-hd.html"><a href="visium-hd.html#objective"><i class="fa fa-check"></i><b>8.1</b> Objective</a></li>
<li class="chapter" data-level="8.2" data-path="visium-hd.html"><a href="visium-hd.html#background"><i class="fa fa-check"></i><b>8.2</b> Background</a>
<ul>
<li class="chapter" data-level="8.2.1" data-path="visium-hd.html"><a href="visium-hd.html#visium-hd-technology"><i class="fa fa-check"></i><b>8.2.1</b> Visium HD Technology</a></li>
<li class="chapter" data-level="8.2.2" data-path="visium-hd.html"><a href="visium-hd.html#colorectal-cancer-sample"><i class="fa fa-check"></i><b>8.2.2</b> Colorectal Cancer Sample</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="visium-hd.html"><a href="visium-hd.html#data-ingestion"><i class="fa fa-check"></i><b>8.3</b> Data Ingestion</a>
<ul>
<li class="chapter" data-level="8.3.1" data-path="visium-hd.html"><a href="visium-hd.html#visium-hd-output-data-format"><i class="fa fa-check"></i><b>8.3.1</b> Visium HD output data format</a></li>
<li class="chapter" data-level="8.3.2" data-path="visium-hd.html"><a href="visium-hd.html#read-in-raw-data"><i class="fa fa-check"></i><b>8.3.2</b> Read in raw data</a></li>
<li class="chapter" data-level="8.3.3" data-path="visium-hd.html"><a href="visium-hd.html#giotto-visium-hd-convenience-function"><i class="fa fa-check"></i><b>8.3.3</b> Giotto Visium HD convenience function</a></li>
</ul></li>
<li class="chapter" data-level="8.4" data-path="visium-hd.html"><a href="visium-hd.html#tiling-and-aggregation"><i class="fa fa-check"></i><b>8.4</b> Tiling and aggregation</a></li>
<li class="chapter" data-level="8.5" data-path="visium-hd.html"><a href="visium-hd.html#scalability-and-projection-functions"><i class="fa fa-check"></i><b>8.5</b> Scalability and projection functions</a></li>
<li class="chapter" data-level="8.6" data-path="visium-hd.html"><a href="visium-hd.html#spatial-expression-patterns-1"><i class="fa fa-check"></i><b>8.6</b> Spatial expression patterns</a></li>
<li class="chapter" data-level="8.7" data-path="visium-hd.html"><a href="visium-hd.html#spatial-co-expression-modules-1"><i class="fa fa-check"></i><b>8.7</b> Spatial co-expression modules</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="xenium.html"><a href="xenium.html"><i class="fa fa-check"></i><b>9</b> Xenium</a>
<ul>
<li class="chapter" data-level="9.1" data-path="xenium.html"><a href="xenium.html#introduction-to-spatial-dataset"><i class="fa fa-check"></i><b>9.1</b> Introduction to spatial dataset</a></li>
<li class="chapter" data-level="9.2" data-path="xenium.html"><a href="xenium.html#additional-package-requirements"><i class="fa fa-check"></i><b>9.2</b> Additional package requirements</a></li>
<li class="chapter" data-level="9.3" data-path="xenium.html"><a href="xenium.html#introduce-mini-object"><i class="fa fa-check"></i><b>9.3</b> Introduce mini object</a></li>
<li class="chapter" data-level="9.4" data-path="xenium.html"><a href="xenium.html#read-in-raw-data-1"><i class="fa fa-check"></i><b>9.4</b> Read in raw data</a></li>
<li class="chapter" data-level="9.5" data-path="xenium.html"><a href="xenium.html#overlap-txs-polygons"><i class="fa fa-check"></i><b>9.5</b> Overlap txs &amp; polygons</a></li>
<li class="chapter" data-level="9.6" data-path="xenium.html"><a href="xenium.html#transcript-enrichment-gsea"><i class="fa fa-check"></i><b>9.6</b> Transcript enrichment GSEA</a></li>
<li class="chapter" data-level="9.7" data-path="xenium.html"><a href="xenium.html#spatial-location-analysis"><i class="fa fa-check"></i><b>9.7</b> Spatial location analysis</a></li>
<li class="chapter" data-level="9.8" data-path="xenium.html"><a href="xenium.html#visualizations"><i class="fa fa-check"></i><b>9.8</b> Visualizations</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="spatial-proteomics-multiplex-if.html"><a href="spatial-proteomics-multiplex-if.html"><i class="fa fa-check"></i><b>10</b> Spatial proteomics: multiplex IF</a>
<ul>
<li class="chapter" data-level="10.1" data-path="spatial-proteomics-multiplex-if.html"><a href="spatial-proteomics-multiplex-if.html#read-in-raw-data-2"><i class="fa fa-check"></i><b>10.1</b> Read in raw data</a></li>
<li class="chapter" data-level="10.2" data-path="spatial-proteomics-multiplex-if.html"><a href="spatial-proteomics-multiplex-if.html#overlap-intensity-workflows"><i class="fa fa-check"></i><b>10.2</b> Overlap intensity &amp; workflows</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html"><i class="fa fa-check"></i><b>11</b> Working with multiple samples</a>
<ul>
<li class="chapter" data-level="11.1" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#objective-1"><i class="fa fa-check"></i><b>11.1</b> Objective</a></li>
<li class="chapter" data-level="11.2" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#background-1"><i class="fa fa-check"></i><b>11.2</b> Background</a></li>
<li class="chapter" data-level="11.3" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#create-individual-giotto-objects"><i class="fa fa-check"></i><b>11.3</b> Create individual giotto objects</a>
<ul>
<li class="chapter" data-level="11.3.1" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#download-the-data"><i class="fa fa-check"></i><b>11.3.1</b> Download the data</a></li>
<li class="chapter" data-level="11.3.2" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#create-giotto-instructions"><i class="fa fa-check"></i><b>11.3.2</b> create giotto instructions</a></li>
<li class="chapter" data-level="11.3.3" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#load-visium-data-into-giotto"><i class="fa fa-check"></i><b>11.3.3</b> load visium data into Giotto</a></li>
<li class="chapter" data-level="11.3.4" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#healthy-prostate-tissue-coverage"><i class="fa fa-check"></i><b>11.3.4</b> Healthy prostate tissue coverage</a></li>
<li class="chapter" data-level="11.3.5" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#adenocarcinoma-prostate-tissue-coverage"><i class="fa fa-check"></i><b>11.3.5</b> Adenocarcinoma prostate tissue coverage</a></li>
<li class="chapter" data-level="11.3.6" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#showing-the-data-strucutre-for-the-inidividual-objects"><i class="fa fa-check"></i><b>11.3.6</b> Showing the data strucutre for the inidividual objects</a></li>
</ul></li>
<li class="chapter" data-level="11.4" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#join-giotto-objects"><i class="fa fa-check"></i><b>11.4</b> Join Giotto Objects</a></li>
<li class="chapter" data-level="11.5" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#visualizing-combined-datasets"><i class="fa fa-check"></i><b>11.5</b> Visualizing combined datasets</a>
<ul>
<li class="chapter" data-level="11.5.1" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#vizualizing-in-the-same-plot"><i class="fa fa-check"></i><b>11.5.1</b> Vizualizing in the same plot</a></li>
<li class="chapter" data-level="11.5.2" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#vizualizing-on-separate-plots"><i class="fa fa-check"></i><b>11.5.2</b> Vizualizing on separate plots</a></li>
</ul></li>
<li class="chapter" data-level="11.6" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#splitting-combined-dataset"><i class="fa fa-check"></i><b>11.6</b> Splitting combined dataset</a></li>
<li class="chapter" data-level="11.7" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#analyzing-joined-objects"><i class="fa fa-check"></i><b>11.7</b> Analyzing joined objects</a>
<ul>
<li class="chapter" data-level="11.7.1" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#normalization-and-adding-statistics"><i class="fa fa-check"></i><b>11.7.1</b> Normalization and adding statistics</a></li>
<li class="chapter" data-level="11.7.2" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#clustering-the-datasets"><i class="fa fa-check"></i><b>11.7.2</b> Clustering the datasets</a></li>
<li class="chapter" data-level="11.7.3" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#vizualizing-spatial-location-of-clusters"><i class="fa fa-check"></i><b>11.7.3</b> Vizualizing spatial location of clusters</a></li>
<li class="chapter" data-level="11.7.4" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#vizualizing-tissue-contribution-to-clusters"><i class="fa fa-check"></i><b>11.7.4</b> Vizualizing tissue contribution to clusters</a></li>
</ul></li>
<li class="chapter" data-level="11.8" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#perform-harmony-and-default-workflows"><i class="fa fa-check"></i><b>11.8</b> Perform Harmony and default workflows</a>
<ul>
<li class="chapter" data-level="11.8.1" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#clustering-harmonized-object"><i class="fa fa-check"></i><b>11.8.1</b> Clustering harmonized object</a></li>
<li class="chapter" data-level="11.8.2" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#vizualizing-the-tissue-contribution"><i class="fa fa-check"></i><b>11.8.2</b> Vizualizing the tissue contribution</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="12" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html"><i class="fa fa-check"></i><b>12</b> Spatial multi-modal analysis</a>
<ul>
<li class="chapter" data-level="12.1" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#overview-1"><i class="fa fa-check"></i><b>12.1</b> Overview</a></li>
<li class="chapter" data-level="12.2" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#spatial-manipulation"><i class="fa fa-check"></i><b>12.2</b> Spatial manipulation</a>
<ul>
<li class="chapter" data-level="12.2.1" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#spatial-transforms"><i class="fa fa-check"></i><b>12.2.1</b> Spatial transforms:</a></li>
<li class="chapter" data-level="12.2.2" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#spatial-utilities"><i class="fa fa-check"></i><b>12.2.2</b> Spatial utilities:</a></li>
<li class="chapter" data-level="12.2.3" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#spatial-classes"><i class="fa fa-check"></i><b>12.2.3</b> Spatial classes:</a></li>
</ul></li>
<li class="chapter" data-level="12.3" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#image-transforms"><i class="fa fa-check"></i><b>12.3</b> Image transforms</a></li>
<li class="chapter" data-level="12.4" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#co-registration-of-datasets"><i class="fa fa-check"></i><b>12.4</b> Co-registration of datasets</a></li>
<li class="chapter" data-level="12.5" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#examples-in-giotto-suite-manuscript"><i class="fa fa-check"></i><b>12.5</b> Examples in giotto suite manuscript</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html"><i class="fa fa-check"></i><b>13</b> Multi-omics integration</a>
<ul>
<li class="chapter" data-level="13.1" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#introduction-to-the-spatial-dataset-1"><i class="fa fa-check"></i><b>13.1</b> Introduction to the spatial dataset</a></li>
<li class="chapter" data-level="13.2" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#download-dataset-1"><i class="fa fa-check"></i><b>13.2</b> Download dataset</a></li>
<li class="chapter" data-level="13.3" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#create-the-giotto-object-1"><i class="fa fa-check"></i><b>13.3</b> Create the Giotto object</a></li>
<li class="chapter" data-level="13.4" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#subset-on-spots-that-were-covered-by-tissue-1"><i class="fa fa-check"></i><b>13.4</b> Subset on spots that were covered by tissue</a></li>
<li class="chapter" data-level="13.5" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#rna-processing"><i class="fa fa-check"></i><b>13.5</b> RNA processing</a></li>
<li class="chapter" data-level="13.6" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#protein-processing"><i class="fa fa-check"></i><b>13.6</b> Protein processing</a></li>
<li class="chapter" data-level="13.7" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#multi-omics-integration-1"><i class="fa fa-check"></i><b>13.7</b> Multi-omics integration</a></li>
<li class="chapter" data-level="13.8" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#session-info-2"><i class="fa fa-check"></i><b>13.8</b> Session info</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html"><i class="fa fa-check"></i><b>14</b> Interoperability with other frameworks</a>
<ul>
<li class="chapter" data-level="14.1" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#load-giotto-object"><i class="fa fa-check"></i><b>14.1</b> Load Giotto object</a></li>
<li class="chapter" data-level="14.2" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#seurat"><i class="fa fa-check"></i><b>14.2</b> Seurat</a>
<ul>
<li class="chapter" data-level="14.2.1" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#conversion-of-giotto-object-to-seurat-object"><i class="fa fa-check"></i><b>14.2.1</b> Conversion of Giotto Object to Seurat Object</a></li>
<li class="chapter" data-level="14.2.2" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#conversion-of-seurat-object-back-to-giotto-object"><i class="fa fa-check"></i><b>14.2.2</b> Conversion of Seurat object Back to Giotto Object</a></li>
</ul></li>
<li class="chapter" data-level="14.3" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#spatialexperiment"><i class="fa fa-check"></i><b>14.3</b> SpatialExperiment</a>
<ul>
<li class="chapter" data-level="14.3.1" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#convert-giotto-object-to-spatialexperiment-object"><i class="fa fa-check"></i><b>14.3.1</b> Convert Giotto Object to SpatialExperiment Object</a></li>
<li class="chapter" data-level="14.3.2" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#conversion-of-spatialexperiment-object-back-to-giotto"><i class="fa fa-check"></i><b>14.3.2</b> Conversion of SpatialExperiment object back to Giotto</a></li>
</ul></li>
<li class="chapter" data-level="14.4" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#anndata"><i class="fa fa-check"></i><b>14.4</b> AnnData</a>
<ul>
<li class="chapter" data-level="14.4.1" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#load-required-libraries"><i class="fa fa-check"></i><b>14.4.1</b> Load Required Libraries</a></li>
<li class="chapter" data-level="14.4.2" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#specify-path-for-results"><i class="fa fa-check"></i><b>14.4.2</b> Specify Path for Results</a></li>
<li class="chapter" data-level="14.4.3" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#giotto-to-anndata"><i class="fa fa-check"></i><b>14.4.3</b> Giotto To AnnData</a></li>
<li class="chapter" data-level="14.4.4" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#convert-anndata-to-giotto"><i class="fa fa-check"></i><b>14.4.4</b> Convert AnnData to Giotto</a></li>
</ul></li>
<li class="chapter" data-level="14.5" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#create-mini-vizgen-object"><i class="fa fa-check"></i><b>14.5</b> Create mini Vizgen object</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html"><i class="fa fa-check"></i><b>15</b> Interoperability with isolated tools</a>
<ul>
<li class="chapter" data-level="15.1" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#spatial-niche-trajectory-analysis"><i class="fa fa-check"></i><b>15.1</b> Spatial niche trajectory analysis</a>
<ul>
<li class="chapter" data-level="15.1.1" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#dataset-download"><i class="fa fa-check"></i><b>15.1.1</b> Dataset download</a></li>
<li class="chapter" data-level="15.1.2" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#create-the-giotto-object-2"><i class="fa fa-check"></i><b>15.1.2</b> Create the Giotto object</a></li>
<li class="chapter" data-level="15.1.3" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#visualization"><i class="fa fa-check"></i><b>15.1.3</b> visualization</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="16" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html"><i class="fa fa-check"></i><b>16</b> Interactivity with the R/Spatial ecosystem</a>
<ul>
<li class="chapter" data-level="16.1" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#kriging"><i class="fa fa-check"></i><b>16.1</b> Kriging</a>
<ul>
<li class="chapter" data-level="16.1.1" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#dataset"><i class="fa fa-check"></i><b>16.1.1</b> Dataset</a></li>
</ul></li>
<li class="chapter" data-level="16.2" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#importing-visium-data"><i class="fa fa-check"></i><b>16.2</b> Importing visium data</a>
<ul>
<li class="chapter" data-level="16.2.1" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#identifying-spatially-organized-features"><i class="fa fa-check"></i><b>16.2.1</b> Identifying spatially organized features</a></li>
</ul></li>
<li class="chapter" data-level="16.3" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#adding-cell-polygons-to-giotto-object"><i class="fa fa-check"></i><b>16.3</b> Adding cell polygons to Giotto object</a>
<ul>
<li class="chapter" data-level="16.3.1" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#read-in-the-poly-information"><i class="fa fa-check"></i><b>16.3.1</b> Read in the poly information</a></li>
<li class="chapter" data-level="16.3.2" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#vizualizing-polygons"><i class="fa fa-check"></i><b>16.3.2</b> Vizualizing polygons</a></li>
<li class="chapter" data-level="16.3.3" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#showing-giotto-object-prior-to-polygon-addition"><i class="fa fa-check"></i><b>16.3.3</b> Showing Giotto object prior to polygon addition</a></li>
<li class="chapter" data-level="16.3.4" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#adding-polygons-to-giotto-object"><i class="fa fa-check"></i><b>16.3.4</b> Adding polygons to giotto object</a></li>
<li class="chapter" data-level="16.3.5" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#check-polygon-information"><i class="fa fa-check"></i><b>16.3.5</b> Check polygon information</a></li>
</ul></li>
<li class="chapter" data-level="16.4" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#performing-kriging"><i class="fa fa-check"></i><b>16.4</b> Performing kriging</a>
<ul>
<li class="chapter" data-level="16.4.1" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#interpolating-features"><i class="fa fa-check"></i><b>16.4.1</b> Interpolating features</a></li>
<li class="chapter" data-level="16.4.2" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#expression-overlap"><i class="fa fa-check"></i><b>16.4.2</b> Expression overlap</a></li>
</ul></li>
<li class="chapter" data-level="16.5" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#reading-in-larger-dataset"><i class="fa fa-check"></i><b>16.5</b> Reading in larger dataset</a></li>
<li class="chapter" data-level="16.6" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#analyzing-interpolated-features"><i class="fa fa-check"></i><b>16.6</b> Analyzing interpolated features</a>
<ul>
<li class="chapter" data-level="16.6.1" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#filter-and-normalization"><i class="fa fa-check"></i><b>16.6.1</b> Filter and normalization</a></li>
<li class="chapter" data-level="16.6.2" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#visualizing-gene-expression-from-interpolated-expression"><i class="fa fa-check"></i><b>16.6.2</b> Visualizing gene expression from interpolated expression</a></li>
<li class="chapter" data-level="16.6.3" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#run-pca"><i class="fa fa-check"></i><b>16.6.3</b> Run PCA</a></li>
<li class="chapter" data-level="16.6.4" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#clustering-1"><i class="fa fa-check"></i><b>16.6.4</b> Clustering</a></li>
<li class="chapter" data-level="16.6.5" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#visualizing-clustering"><i class="fa fa-check"></i><b>16.6.5</b> Visualizing clustering</a></li>
<li class="chapter" data-level="16.6.6" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#cropping-objects"><i class="fa fa-check"></i><b>16.6.6</b> Cropping objects</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="17" data-path="contributing-to-giotto.html"><a href="contributing-to-giotto.html"><i class="fa fa-check"></i><b>17</b> Contributing to Giotto</a>
<ul>
<li class="chapter" data-level="17.1" data-path="contributing-to-giotto.html"><a href="contributing-to-giotto.html#contribution-guideline"><i class="fa fa-check"></i><b>17.1</b> Contribution guideline</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Workshop: Spatial multi-omics data analysis with Giotto Suite</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="working-with-multiple-samples" class="section level1 hasAnchor" number="11">
<h1><span class="header-section-number">11</span> Working with multiple samples<a href="working-with-multiple-samples.html#working-with-multiple-samples" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Jeff Sheridan</p>
<p>August 7th 2024</p>
<div id="objective-1" class="section level2 hasAnchor" number="11.1">
<h2><span class="header-section-number">11.1</span> Objective<a href="working-with-multiple-samples.html#objective-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Giotto allows us to group objects into a single object that we can analyze together as a single object. Datasets can be spatially distributed in the x, y or z directions. This can be used to create 3D datasets using the Z plane or analyze grouped datasets such as multiple replicates or samples of the same type. Whilst we can integrate multiple datasets, batch effects from samples can impair integration. More complex integration is sometimes required to integrate and cluster samples as one. An example of an integration method is <a href="https://www.nature.com/articles/s41592-019-0619-0">Harmony</a> integration, which will be explained in greater detail later in this tutorial.</p>
<p>The following tutorial will explore integrating two visium datasets together</p>
</div>
<div id="background-1" class="section level2 hasAnchor" number="11.2">
<h2><span class="header-section-number">11.2</span> Background<a href="working-with-multiple-samples.html#background-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>For this tutorial we will be using two prostate visium datasets produced by 10X Genomics, one an <a href="https://www.10xgenomics.com/datasets/human-prostate-cancer-adenocarcinoma-with-invasive-carcinoma-ffpe-1-standard-1-3-0">Adenocarcinoma with Invasive Carcinoma</a> and the other a <a href="https://www.10xgenomics.com/datasets/normal-human-prostate-ffpe-1-standard-1-3-0">normal prostate</a> sample. Details on visium technology can be found in section 6.</p>
</div>
<div id="create-individual-giotto-objects" class="section level2 hasAnchor" number="11.3">
<h2><span class="header-section-number">11.3</span> Create individual giotto objects<a href="working-with-multiple-samples.html#create-individual-giotto-objects" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="download-the-data" class="section level3 hasAnchor" number="11.3.1">
<h3><span class="header-section-number">11.3.1</span> Download the data<a href="working-with-multiple-samples.html#download-the-data" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>You need to download the expression matrix and spatial information by running these commands:</p>
<div class="sourceCode" id="cb104"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb104-1"><a href="working-with-multiple-samples.html#cb104-1" tabindex="-1"></a>data_dir <span class="ot">&lt;-</span> <span class="st">&#39;data&#39;</span></span>
<span id="cb104-2"><a href="working-with-multiple-samples.html#cb104-2" tabindex="-1"></a></span>
<span id="cb104-3"><a href="working-with-multiple-samples.html#cb104-3" tabindex="-1"></a><span class="fu">dir.create</span>(<span class="fu">file.path</span>(data_dir, <span class="st">&#39;Visium_FFPE_Human_Prostate_Cancer&#39;</span>), <span class="at">showWarnings =</span> T, <span class="at">recursive =</span> T)</span>
<span id="cb104-4"><a href="working-with-multiple-samples.html#cb104-4" tabindex="-1"></a></span>
<span id="cb104-5"><a href="working-with-multiple-samples.html#cb104-5" tabindex="-1"></a><span class="co"># Spatial data adenocarcinom prostate</span></span>
<span id="cb104-6"><a href="working-with-multiple-samples.html#cb104-6" tabindex="-1"></a><span class="fu">download.file</span>(<span class="at">url =</span> <span class="st">&#39;https://cf.10xgenomics.com/samples/spatial-exp/1.3.0/Visium_FFPE_Human_Prostate_Cancer/Visium_FFPE_Human_Prostate_Cancer_spatial.tar.gz&#39;</span>, </span>
<span id="cb104-7"><a href="working-with-multiple-samples.html#cb104-7" tabindex="-1"></a>              <span class="at">destfile =</span> <span class="fu">file.path</span>(data_dir, <span class="st">&#39;Visium_FFPE_Human_Prostate_Cancer/Visium_FFPE_Human_Prostate_Cancer_spatial.tar.gz&#39;</span>))</span>
<span id="cb104-8"><a href="working-with-multiple-samples.html#cb104-8" tabindex="-1"></a></span>
<span id="cb104-9"><a href="working-with-multiple-samples.html#cb104-9" tabindex="-1"></a><span class="co"># Deat matrix adenocarcinoma prostate</span></span>
<span id="cb104-10"><a href="working-with-multiple-samples.html#cb104-10" tabindex="-1"></a><span class="fu">download.file</span>(<span class="at">url =</span> <span class="st">&#39;https://cf.10xgenomics.com/samples/spatial-exp/1.3.0/Visium_FFPE_Human_Prostate_Cancer/Visium_FFPE_Human_Prostate_Cancer_raw_feature_bc_matrix.tar.gz&#39;</span>, </span>
<span id="cb104-11"><a href="working-with-multiple-samples.html#cb104-11" tabindex="-1"></a>              <span class="at">destfile =</span> <span class="fu">file.path</span>(data_dir, <span class="st">&#39;Visium_FFPE_Human_Prostate_Cancer/Visium_FFPE_Human_Prostate_Cancer_raw_feature_bc_matrix.tar.gz&#39;</span>))</span>
<span id="cb104-12"><a href="working-with-multiple-samples.html#cb104-12" tabindex="-1"></a></span>
<span id="cb104-13"><a href="working-with-multiple-samples.html#cb104-13" tabindex="-1"></a><span class="fu">dir.create</span>(<span class="fu">file.path</span>(data_dir, <span class="st">&#39;Visium_FFPE_Human_Normal_Prostate&#39;</span>), <span class="at">showWarnings =</span> F, <span class="at">recursive =</span> T)</span>
<span id="cb104-14"><a href="working-with-multiple-samples.html#cb104-14" tabindex="-1"></a></span>
<span id="cb104-15"><a href="working-with-multiple-samples.html#cb104-15" tabindex="-1"></a><span class="co"># Spatial data normal prostate</span></span>
<span id="cb104-16"><a href="working-with-multiple-samples.html#cb104-16" tabindex="-1"></a><span class="fu">download.file</span>(<span class="at">url =</span> <span class="st">&#39;https://cf.10xgenomics.com/samples/spatial-exp/1.3.0/Visium_FFPE_Human_Normal_Prostate/Visium_FFPE_Human_Normal_Prostate_spatial.tar.gz&#39;</span>,</span>
<span id="cb104-17"><a href="working-with-multiple-samples.html#cb104-17" tabindex="-1"></a>              <span class="at">destfile =</span> <span class="fu">file.path</span>(data_dir, <span class="st">&#39;Visium_FFPE_Human_Normal_Prostate/Visium_FFPE_Human_Normal_Prostate_spatial.tar.gz&#39;</span>))</span>
<span id="cb104-18"><a href="working-with-multiple-samples.html#cb104-18" tabindex="-1"></a></span>
<span id="cb104-19"><a href="working-with-multiple-samples.html#cb104-19" tabindex="-1"></a><span class="co"># Deat matrix normal prostate</span></span>
<span id="cb104-20"><a href="working-with-multiple-samples.html#cb104-20" tabindex="-1"></a><span class="fu">download.file</span>(<span class="at">url =</span> <span class="st">&#39;https://cf.10xgenomics.com/samples/spatial-exp/1.3.0/Visium_FFPE_Human_Normal_Prostate/Visium_FFPE_Human_Normal_Prostate_raw_feature_bc_matrix.tar.gz&#39;</span>,</span>
<span id="cb104-21"><a href="working-with-multiple-samples.html#cb104-21" tabindex="-1"></a>              <span class="at">destfile =</span> <span class="fu">file.path</span>(data_dir, <span class="st">&#39;Visium_FFPE_Human_Normal_Prostate/Visium_FFPE_Human_Normal_Prostate_raw_feature_bc_matrix.tar.gz&#39;</span>))</span></code></pre></div>
</div>
<div id="create-giotto-instructions" class="section level3 hasAnchor" number="11.3.2">
<h3><span class="header-section-number">11.3.2</span> create giotto instructions<a href="working-with-multiple-samples.html#create-giotto-instructions" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb105"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb105-1"><a href="working-with-multiple-samples.html#cb105-1" tabindex="-1"></a><span class="fu">library</span>(Giotto)</span>
<span id="cb105-2"><a href="working-with-multiple-samples.html#cb105-2" tabindex="-1"></a></span>
<span id="cb105-3"><a href="working-with-multiple-samples.html#cb105-3" tabindex="-1"></a>save_dir <span class="ot">&lt;-</span> <span class="st">&quot;~/Documents/GitHub/giotto_workshop_2024/img/03_session1&quot;</span></span>
<span id="cb105-4"><a href="working-with-multiple-samples.html#cb105-4" tabindex="-1"></a></span>
<span id="cb105-5"><a href="working-with-multiple-samples.html#cb105-5" tabindex="-1"></a>instrs <span class="ot">=</span> <span class="fu">createGiottoInstructions</span>(<span class="at">save_dir =</span> save_dir,</span>
<span id="cb105-6"><a href="working-with-multiple-samples.html#cb105-6" tabindex="-1"></a>                                  <span class="at">save_plot =</span> <span class="cn">TRUE</span>,</span>
<span id="cb105-7"><a href="working-with-multiple-samples.html#cb105-7" tabindex="-1"></a>                                  <span class="at">show_plot =</span> <span class="cn">TRUE</span>,</span>
<span id="cb105-8"><a href="working-with-multiple-samples.html#cb105-8" tabindex="-1"></a>                                  <span class="at">python_path =</span> <span class="cn">NULL</span>)</span></code></pre></div>
</div>
<div id="load-visium-data-into-giotto" class="section level3 hasAnchor" number="11.3.3">
<h3><span class="header-section-number">11.3.3</span> load visium data into Giotto<a href="working-with-multiple-samples.html#load-visium-data-into-giotto" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>First we need to load in the datasets as individual Giotto objects.</p>
<div class="sourceCode" id="cb106"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb106-1"><a href="working-with-multiple-samples.html#cb106-1" tabindex="-1"></a><span class="fu">print</span>(data_dir)</span>
<span id="cb106-2"><a href="working-with-multiple-samples.html#cb106-2" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">file.path</span>(data_dir, <span class="st">&#39;Visium_FFPE_Human_Normal_Prostate&#39;</span>))</span>
<span id="cb106-3"><a href="working-with-multiple-samples.html#cb106-3" tabindex="-1"></a><span class="do">## Healthy prostate</span></span>
<span id="cb106-4"><a href="working-with-multiple-samples.html#cb106-4" tabindex="-1"></a>N_pros <span class="ot">=</span> <span class="fu">createGiottoVisiumObject</span>(</span>
<span id="cb106-5"><a href="working-with-multiple-samples.html#cb106-5" tabindex="-1"></a>  <span class="at">visium_dir =</span> <span class="fu">file.path</span>(data_dir, <span class="st">&#39;Visium_FFPE_Human_Normal_Prostate&#39;</span>),</span>
<span id="cb106-6"><a href="working-with-multiple-samples.html#cb106-6" tabindex="-1"></a>  <span class="at">expr_data =</span> <span class="st">&#39;raw&#39;</span>,</span>
<span id="cb106-7"><a href="working-with-multiple-samples.html#cb106-7" tabindex="-1"></a>  <span class="at">png_name =</span> <span class="st">&#39;tissue_lowres_image.png&#39;</span>,</span>
<span id="cb106-8"><a href="working-with-multiple-samples.html#cb106-8" tabindex="-1"></a>  <span class="at">gene_column_index =</span> <span class="dv">2</span>,</span>
<span id="cb106-9"><a href="working-with-multiple-samples.html#cb106-9" tabindex="-1"></a>  <span class="at">instructions =</span> instrs</span>
<span id="cb106-10"><a href="working-with-multiple-samples.html#cb106-10" tabindex="-1"></a>)</span>
<span id="cb106-11"><a href="working-with-multiple-samples.html#cb106-11" tabindex="-1"></a></span>
<span id="cb106-12"><a href="working-with-multiple-samples.html#cb106-12" tabindex="-1"></a><span class="do">## Adenocarcinoma </span></span>
<span id="cb106-13"><a href="working-with-multiple-samples.html#cb106-13" tabindex="-1"></a>C_pros <span class="ot">=</span> <span class="fu">createGiottoVisiumObject</span>(</span>
<span id="cb106-14"><a href="working-with-multiple-samples.html#cb106-14" tabindex="-1"></a>  <span class="at">visium_dir =</span> <span class="fu">file.path</span>(data_dir, <span class="st">&#39;Visium_FFPE_Human_Prostate_Cancer&#39;</span>),</span>
<span id="cb106-15"><a href="working-with-multiple-samples.html#cb106-15" tabindex="-1"></a>  <span class="at">expr_data =</span> <span class="st">&#39;raw&#39;</span>,</span>
<span id="cb106-16"><a href="working-with-multiple-samples.html#cb106-16" tabindex="-1"></a>  <span class="at">png_name =</span> <span class="st">&#39;tissue_lowres_image.png&#39;</span>,</span>
<span id="cb106-17"><a href="working-with-multiple-samples.html#cb106-17" tabindex="-1"></a>  <span class="at">gene_column_index =</span> <span class="dv">2</span>,</span>
<span id="cb106-18"><a href="working-with-multiple-samples.html#cb106-18" tabindex="-1"></a>  <span class="at">instructions =</span> instrs</span>
<span id="cb106-19"><a href="working-with-multiple-samples.html#cb106-19" tabindex="-1"></a>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-159"></span>
<img src="img/03_session1/03_ses1_single_gobject.png" alt="Structure of Giotto object containing a single dataset." width="40%" />
<p class="caption">
Figure 11.1: Structure of Giotto object containing a single dataset.
</p>
</div>
</div>
<div id="healthy-prostate-tissue-coverage" class="section level3 hasAnchor" number="11.3.4">
<h3><span class="header-section-number">11.3.4</span> Healthy prostate tissue coverage<a href="working-with-multiple-samples.html#healthy-prostate-tissue-coverage" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb107"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb107-1"><a href="working-with-multiple-samples.html#cb107-1" tabindex="-1"></a><span class="fu">spatPlot2D</span>(<span class="at">gobject =</span> N_pros, <span class="at">cell_color =</span> <span class="st">&#39;in_tissue&#39;</span>,</span>
<span id="cb107-2"><a href="working-with-multiple-samples.html#cb107-2" tabindex="-1"></a>           <span class="at">show_image =</span> T, <span class="at">point_size =</span> <span class="fl">2.5</span>,<span class="at">cell_color_code =</span> <span class="fu">c</span>(<span class="st">&#39;black&#39;</span>, <span class="st">&#39;red&#39;</span>),</span>
<span id="cb107-3"><a href="working-with-multiple-samples.html#cb107-3" tabindex="-1"></a>           <span class="at">point_alpha =</span> <span class="fl">0.5</span>, <span class="at">save_param =</span> <span class="fu">list</span>(<span class="at">save_name =</span> <span class="st">&quot;03_ses1_normal_prostate_tissue&quot;</span>))</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-161"></span>
<img src="img/03_session1/03_ses1_normal_prostate_tissue.png" alt="Tissue coverage for the normal prostate sample." width="70%" />
<p class="caption">
Figure 11.2: Tissue coverage for the normal prostate sample.
</p>
</div>
</div>
<div id="adenocarcinoma-prostate-tissue-coverage" class="section level3 hasAnchor" number="11.3.5">
<h3><span class="header-section-number">11.3.5</span> Adenocarcinoma prostate tissue coverage<a href="working-with-multiple-samples.html#adenocarcinoma-prostate-tissue-coverage" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb108"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb108-1"><a href="working-with-multiple-samples.html#cb108-1" tabindex="-1"></a><span class="fu">spatPlot2D</span>(<span class="at">gobject =</span> C_pros, <span class="at">cell_color =</span> <span class="st">&#39;in_tissue&#39;</span>,</span>
<span id="cb108-2"><a href="working-with-multiple-samples.html#cb108-2" tabindex="-1"></a>           <span class="at">show_image =</span> T, <span class="at">point_size =</span> <span class="fl">2.5</span>,<span class="at">cell_color_code =</span> <span class="fu">c</span>(<span class="st">&#39;black&#39;</span>, <span class="st">&#39;red&#39;</span>), </span>
<span id="cb108-3"><a href="working-with-multiple-samples.html#cb108-3" tabindex="-1"></a>           <span class="at">point_alpha =</span> <span class="fl">0.5</span>, <span class="at">save_param =</span> <span class="fu">list</span>(<span class="at">save_name =</span> <span class="st">&quot;03_ses1_adeno_prostate_tissue&quot;</span>))</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-163"></span>
<img src="img/03_session1/03_ses1_adeno_prostate_tissue.png" alt="Tissue coverage for the adenocarcinoma prostate sample." width="70%" />
<p class="caption">
Figure 11.3: Tissue coverage for the adenocarcinoma prostate sample.
</p>
</div>
</div>
<div id="showing-the-data-strucutre-for-the-inidividual-objects" class="section level3 hasAnchor" number="11.3.6">
<h3><span class="header-section-number">11.3.6</span> Showing the data strucutre for the inidividual objects<a href="working-with-multiple-samples.html#showing-the-data-strucutre-for-the-inidividual-objects" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb109"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb109-1"><a href="working-with-multiple-samples.html#cb109-1" tabindex="-1"></a><span class="co"># Printing the file structure for the individual datasets</span></span>
<span id="cb109-2"><a href="working-with-multiple-samples.html#cb109-2" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">head</span>(<span class="fu">pDataDT</span>(N_pros)))</span>
<span id="cb109-3"><a href="working-with-multiple-samples.html#cb109-3" tabindex="-1"></a><span class="fu">print</span>(N_pros)</span></code></pre></div>
</div>
</div>
<div id="join-giotto-objects" class="section level2 hasAnchor" number="11.4">
<h2><span class="header-section-number">11.4</span> Join Giotto Objects<a href="working-with-multiple-samples.html#join-giotto-objects" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>joining with x_shift has the advantage that you can join both 2D and 3D data
x_padding determines how much distance is between each dataset
if x_shift = NULL, then the total shift will be guessed from the giotto image</p>
<div class="sourceCode" id="cb110"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb110-1"><a href="working-with-multiple-samples.html#cb110-1" tabindex="-1"></a>combined_pros <span class="ot">=</span> <span class="fu">joinGiottoObjects</span>(<span class="at">gobject_list =</span> <span class="fu">list</span>(N_pros, C_pros),</span>
<span id="cb110-2"><a href="working-with-multiple-samples.html#cb110-2" tabindex="-1"></a>                              <span class="at">gobject_names =</span> <span class="fu">c</span>(<span class="st">&#39;NP&#39;</span>, <span class="st">&#39;CP&#39;</span>),</span>
<span id="cb110-3"><a href="working-with-multiple-samples.html#cb110-3" tabindex="-1"></a>                              <span class="at">join_method =</span> <span class="st">&#39;shift&#39;</span>, <span class="at">x_padding =</span> <span class="dv">1000</span>)</span>
<span id="cb110-4"><a href="working-with-multiple-samples.html#cb110-4" tabindex="-1"></a></span>
<span id="cb110-5"><a href="working-with-multiple-samples.html#cb110-5" tabindex="-1"></a><span class="co"># Printing the file structure for the individual datasets</span></span>
<span id="cb110-6"><a href="working-with-multiple-samples.html#cb110-6" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">head</span>(<span class="fu">pDataDT</span>(combined_pros)))</span>
<span id="cb110-7"><a href="working-with-multiple-samples.html#cb110-7" tabindex="-1"></a><span class="fu">print</span>(combined_pros)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-166"></span>
<img src="img/03_session1/03_ses1_combined_data.png" alt="Structure of Giotto object containing two datasets (left) and cell metadata on the left. Note the addition of multiple images and the addition of the list_ID column to define the dataset." width="100%" />
<p class="caption">
Figure 11.4: Structure of Giotto object containing two datasets (left) and cell metadata on the left. Note the addition of multiple images and the addition of the list_ID column to define the dataset.
</p>
</div>
</div>
<div id="visualizing-combined-datasets" class="section level2 hasAnchor" number="11.5">
<h2><span class="header-section-number">11.5</span> Visualizing combined datasets<a href="working-with-multiple-samples.html#visualizing-combined-datasets" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The combined dataset can either visualized in the same space or in two separate plots through the group_by variable. To show images both the show_image variable and the image_name variable containing both image names needs to be used.</p>
<div id="vizualizing-in-the-same-plot" class="section level3 hasAnchor" number="11.5.1">
<h3><span class="header-section-number">11.5.1</span> Vizualizing in the same plot<a href="working-with-multiple-samples.html#vizualizing-in-the-same-plot" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb111"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb111-1"><a href="working-with-multiple-samples.html#cb111-1" tabindex="-1"></a><span class="fu">spatPlot2D</span>(<span class="at">gobject =</span> combined_pros, <span class="at">cell_color =</span> <span class="st">&#39;in_tissue&#39;</span>, <span class="at">cell_color_code =</span> <span class="fu">c</span>(<span class="st">&#39;black&#39;</span>, <span class="st">&#39;red&#39;</span>),</span>
<span id="cb111-2"><a href="working-with-multiple-samples.html#cb111-2" tabindex="-1"></a>           <span class="at">show_image =</span> T, <span class="at">image_name =</span> <span class="fu">c</span>(<span class="st">&quot;NP-image&quot;</span>, <span class="st">&quot;CP-image&quot;</span>), <span class="at">point_size =</span> <span class="dv">1</span>, </span>
<span id="cb111-3"><a href="working-with-multiple-samples.html#cb111-3" tabindex="-1"></a>           <span class="at">point_alpha =</span> <span class="fl">0.5</span>, <span class="at">save_param =</span> <span class="fu">list</span>(<span class="at">save_name =</span> <span class="st">&quot;03_ses1_combined_tissue&quot;</span>))</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-168"></span>
<img src="img/03_session1/03_ses1_combined_tissue.png" alt="Structure of Giotto object containing two datasets (left) and cell metadata on the left. Note the addition of multiple images and the addition of the list_ID column to define the dataset." width="100%" />
<p class="caption">
Figure 11.5: Structure of Giotto object containing two datasets (left) and cell metadata on the left. Note the addition of multiple images and the addition of the list_ID column to define the dataset.
</p>
</div>
</div>
<div id="vizualizing-on-separate-plots" class="section level3 hasAnchor" number="11.5.2">
<h3><span class="header-section-number">11.5.2</span> Vizualizing on separate plots<a href="working-with-multiple-samples.html#vizualizing-on-separate-plots" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb112"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb112-1"><a href="working-with-multiple-samples.html#cb112-1" tabindex="-1"></a><span class="fu">spatPlot2D</span>(<span class="at">gobject =</span> combined_pros, <span class="at">cell_color =</span> <span class="st">&#39;in_tissue&#39;</span>, <span class="at">cell_color_code =</span> <span class="fu">c</span>(<span class="st">&#39;black&#39;</span>, <span class="st">&#39;pink&#39;</span>),</span>
<span id="cb112-2"><a href="working-with-multiple-samples.html#cb112-2" tabindex="-1"></a>           <span class="at">show_image =</span> T, <span class="at">image_name =</span> <span class="fu">c</span>(<span class="st">&quot;NP-image&quot;</span>, <span class="st">&quot;CP-image&quot;</span>),</span>
<span id="cb112-3"><a href="working-with-multiple-samples.html#cb112-3" tabindex="-1"></a>           <span class="at">group_by =</span> <span class="st">&#39;list_ID&#39;</span>, <span class="at">point_alpha =</span> <span class="fl">0.5</span>, <span class="at">point_size =</span> <span class="fl">0.5</span>, <span class="at">cow_n_col =</span> <span class="dv">1</span>,</span>
<span id="cb112-4"><a href="working-with-multiple-samples.html#cb112-4" tabindex="-1"></a>           <span class="at">save_param =</span> <span class="fu">list</span>(<span class="at">save_name =</span> <span class="st">&quot;03_ses1_combined_tissue_group&quot;</span>))</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-170"></span>
<img src="img/03_session1/03_ses1_combined_tissue_group.png" alt="Structure of Giotto object containing two datasets (left) and cell metadata on the left. Note the addition of multiple images and the addition of the list_ID column to define the dataset." width="100%" />
<p class="caption">
Figure 11.6: Structure of Giotto object containing two datasets (left) and cell metadata on the left. Note the addition of multiple images and the addition of the list_ID column to define the dataset.
</p>
</div>
</div>
</div>
<div id="splitting-combined-dataset" class="section level2 hasAnchor" number="11.6">
<h2><span class="header-section-number">11.6</span> Splitting combined dataset<a href="working-with-multiple-samples.html#splitting-combined-dataset" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The individual objects can be split into single objects again through subsetting the cell metadata.</p>
<div class="sourceCode" id="cb113"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb113-1"><a href="working-with-multiple-samples.html#cb113-1" tabindex="-1"></a><span class="co"># Getting the cell information</span></span>
<span id="cb113-2"><a href="working-with-multiple-samples.html#cb113-2" tabindex="-1"></a>combined_cells <span class="ot">&lt;-</span> <span class="fu">pDataDT</span>(combined_pros)</span>
<span id="cb113-3"><a href="working-with-multiple-samples.html#cb113-3" tabindex="-1"></a>np_cells <span class="ot">&lt;-</span> combined_cells[list_ID <span class="sc">==</span> <span class="st">&#39;NP&#39;</span>]</span>
<span id="cb113-4"><a href="working-with-multiple-samples.html#cb113-4" tabindex="-1"></a></span>
<span id="cb113-5"><a href="working-with-multiple-samples.html#cb113-5" tabindex="-1"></a>np_split <span class="ot">&lt;-</span> <span class="fu">subsetGiotto</span>(combined_pros, </span>
<span id="cb113-6"><a href="working-with-multiple-samples.html#cb113-6" tabindex="-1"></a>                         <span class="at">cell_ids =</span> np_cells<span class="sc">$</span>cell_ID,</span>
<span id="cb113-7"><a href="working-with-multiple-samples.html#cb113-7" tabindex="-1"></a>                         <span class="at">poly_info =</span> np_cells<span class="sc">$</span>cell_ID,</span>
<span id="cb113-8"><a href="working-with-multiple-samples.html#cb113-8" tabindex="-1"></a>                         <span class="at">spat_unit =</span> <span class="st">&#39;cell&#39;</span>)</span>
<span id="cb113-9"><a href="working-with-multiple-samples.html#cb113-9" tabindex="-1"></a></span>
<span id="cb113-10"><a href="working-with-multiple-samples.html#cb113-10" tabindex="-1"></a><span class="fu">spatPlot2D</span>(<span class="at">gobject =</span> np_split, <span class="at">cell_color =</span> <span class="st">&#39;in_tissue&#39;</span>, <span class="at">cell_color_code =</span> <span class="fu">c</span>(<span class="st">&#39;black&#39;</span>, <span class="st">&#39;red&#39;</span>),</span>
<span id="cb113-11"><a href="working-with-multiple-samples.html#cb113-11" tabindex="-1"></a>           <span class="at">show_image =</span> T, <span class="at">point_alpha =</span> <span class="fl">0.5</span>, <span class="at">point_size =</span> <span class="fl">0.5</span>, </span>
<span id="cb113-12"><a href="working-with-multiple-samples.html#cb113-12" tabindex="-1"></a>           <span class="at">save_param =</span> <span class="fu">list</span>(<span class="at">save_name =</span> <span class="st">&quot;03_ses1_split_object&quot;</span>))</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-172"></span>
<img src="img/03_session1/03_ses1_split_object.png" alt="Structure of Giotto object containing two datasets (left) and cell metadata on the left. Note the addition of multiple images and the addition of the list_ID column to define the dataset." width="100%" />
<p class="caption">
Figure 11.7: Structure of Giotto object containing two datasets (left) and cell metadata on the left. Note the addition of multiple images and the addition of the list_ID column to define the dataset.
</p>
</div>
</div>
<div id="analyzing-joined-objects" class="section level2 hasAnchor" number="11.7">
<h2><span class="header-section-number">11.7</span> Analyzing joined objects<a href="working-with-multiple-samples.html#analyzing-joined-objects" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The joined objects can be analyzed as if they were a single dataset as seen in section 6.</p>
<div id="normalization-and-adding-statistics" class="section level3 hasAnchor" number="11.7.1">
<h3><span class="header-section-number">11.7.1</span> Normalization and adding statistics<a href="working-with-multiple-samples.html#normalization-and-adding-statistics" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb114"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb114-1"><a href="working-with-multiple-samples.html#cb114-1" tabindex="-1"></a><span class="co"># subset on in-tissue spots</span></span>
<span id="cb114-2"><a href="working-with-multiple-samples.html#cb114-2" tabindex="-1"></a>metadata <span class="ot">=</span> <span class="fu">pDataDT</span>(combined_pros)</span>
<span id="cb114-3"><a href="working-with-multiple-samples.html#cb114-3" tabindex="-1"></a>in_tissue_barcodes <span class="ot">=</span> metadata[in_tissue <span class="sc">==</span> <span class="dv">1</span>]<span class="sc">$</span>cell_ID</span>
<span id="cb114-4"><a href="working-with-multiple-samples.html#cb114-4" tabindex="-1"></a>combined_pros <span class="ot">=</span> <span class="fu">subsetGiotto</span>(combined_pros, <span class="at">cell_ids =</span> in_tissue_barcodes)</span>
<span id="cb114-5"><a href="working-with-multiple-samples.html#cb114-5" tabindex="-1"></a></span>
<span id="cb114-6"><a href="working-with-multiple-samples.html#cb114-6" tabindex="-1"></a><span class="do">## filter</span></span>
<span id="cb114-7"><a href="working-with-multiple-samples.html#cb114-7" tabindex="-1"></a>combined_pros <span class="ot">&lt;-</span> <span class="fu">filterGiotto</span>(<span class="at">gobject =</span> combined_pros,</span>
<span id="cb114-8"><a href="working-with-multiple-samples.html#cb114-8" tabindex="-1"></a>                          <span class="at">expression_threshold =</span> <span class="dv">1</span>,</span>
<span id="cb114-9"><a href="working-with-multiple-samples.html#cb114-9" tabindex="-1"></a>                          <span class="at">feat_det_in_min_cells =</span> <span class="dv">50</span>,</span>
<span id="cb114-10"><a href="working-with-multiple-samples.html#cb114-10" tabindex="-1"></a>                          <span class="at">min_det_feats_per_cell =</span> <span class="dv">500</span>,</span>
<span id="cb114-11"><a href="working-with-multiple-samples.html#cb114-11" tabindex="-1"></a>                          <span class="at">expression_values =</span> <span class="fu">c</span>(<span class="st">&#39;raw&#39;</span>),</span>
<span id="cb114-12"><a href="working-with-multiple-samples.html#cb114-12" tabindex="-1"></a>                          <span class="at">verbose =</span> T)</span>
<span id="cb114-13"><a href="working-with-multiple-samples.html#cb114-13" tabindex="-1"></a></span>
<span id="cb114-14"><a href="working-with-multiple-samples.html#cb114-14" tabindex="-1"></a><span class="do">## normalize</span></span>
<span id="cb114-15"><a href="working-with-multiple-samples.html#cb114-15" tabindex="-1"></a>combined_pros <span class="ot">&lt;-</span> <span class="fu">normalizeGiotto</span>(<span class="at">gobject =</span> combined_pros, <span class="at">scalefactor =</span> <span class="dv">6000</span>)</span>
<span id="cb114-16"><a href="working-with-multiple-samples.html#cb114-16" tabindex="-1"></a></span>
<span id="cb114-17"><a href="working-with-multiple-samples.html#cb114-17" tabindex="-1"></a><span class="do">## add gene &amp; cell statistics</span></span>
<span id="cb114-18"><a href="working-with-multiple-samples.html#cb114-18" tabindex="-1"></a>combined_pros <span class="ot">&lt;-</span> <span class="fu">addStatistics</span>(<span class="at">gobject =</span> combined_pros, <span class="at">expression_values =</span> <span class="st">&#39;raw&#39;</span>)</span>
<span id="cb114-19"><a href="working-with-multiple-samples.html#cb114-19" tabindex="-1"></a></span>
<span id="cb114-20"><a href="working-with-multiple-samples.html#cb114-20" tabindex="-1"></a><span class="do">## visualize</span></span>
<span id="cb114-21"><a href="working-with-multiple-samples.html#cb114-21" tabindex="-1"></a><span class="fu">spatPlot2D</span>(<span class="at">gobject =</span> combined_pros, </span>
<span id="cb114-22"><a href="working-with-multiple-samples.html#cb114-22" tabindex="-1"></a>           <span class="at">cell_color =</span> <span class="st">&#39;nr_feats&#39;</span>, <span class="at">color_as_factor =</span> F, <span class="at">point_size =</span> <span class="dv">1</span>, </span>
<span id="cb114-23"><a href="working-with-multiple-samples.html#cb114-23" tabindex="-1"></a>           <span class="at">show_image =</span> T, <span class="at">image_name =</span> <span class="fu">c</span>(<span class="st">&quot;NP-image&quot;</span>, <span class="st">&quot;CP-image&quot;</span>),</span>
<span id="cb114-24"><a href="working-with-multiple-samples.html#cb114-24" tabindex="-1"></a>           <span class="at">save_param =</span> <span class="fu">list</span>(<span class="at">save_name =</span> <span class="st">&quot;ses3_1_feat_expression&quot;</span>))</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-174"></span>
<img src="img/03_session1/ses3_1_feat_expression.png" alt="Unique feat expression for visium spots for both prostate samples." width="100%" />
<p class="caption">
Figure 11.8: Unique feat expression for visium spots for both prostate samples.
</p>
</div>
</div>
<div id="clustering-the-datasets" class="section level3 hasAnchor" number="11.7.2">
<h3><span class="header-section-number">11.7.2</span> Clustering the datasets<a href="working-with-multiple-samples.html#clustering-the-datasets" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb115"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb115-1"><a href="working-with-multiple-samples.html#cb115-1" tabindex="-1"></a><span class="do">## PCA ##</span></span>
<span id="cb115-2"><a href="working-with-multiple-samples.html#cb115-2" tabindex="-1"></a>combined_pros <span class="ot">&lt;-</span> <span class="fu">calculateHVF</span>(<span class="at">gobject =</span> combined_pros)</span>
<span id="cb115-3"><a href="working-with-multiple-samples.html#cb115-3" tabindex="-1"></a></span>
<span id="cb115-4"><a href="working-with-multiple-samples.html#cb115-4" tabindex="-1"></a>combined_pros <span class="ot">&lt;-</span> <span class="fu">runPCA</span>(<span class="at">gobject =</span> combined_pros, <span class="at">center =</span> <span class="cn">TRUE</span>, <span class="at">scale_unit =</span> <span class="cn">TRUE</span>)</span>
<span id="cb115-5"><a href="working-with-multiple-samples.html#cb115-5" tabindex="-1"></a></span>
<span id="cb115-6"><a href="working-with-multiple-samples.html#cb115-6" tabindex="-1"></a><span class="do">## cluster and run UMAP ##</span></span>
<span id="cb115-7"><a href="working-with-multiple-samples.html#cb115-7" tabindex="-1"></a><span class="co"># sNN network (default)</span></span>
<span id="cb115-8"><a href="working-with-multiple-samples.html#cb115-8" tabindex="-1"></a>combined_pros <span class="ot">&lt;-</span> <span class="fu">createNearestNetwork</span>(<span class="at">gobject =</span> combined_pros,</span>
<span id="cb115-9"><a href="working-with-multiple-samples.html#cb115-9" tabindex="-1"></a>                                  <span class="at">dim_reduction_to_use =</span> <span class="st">&#39;pca&#39;</span>, <span class="at">dim_reduction_name =</span> <span class="st">&#39;pca&#39;</span>,</span>
<span id="cb115-10"><a href="working-with-multiple-samples.html#cb115-10" tabindex="-1"></a>                                  <span class="at">dimensions_to_use =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>, <span class="at">k =</span> <span class="dv">15</span>)</span>
<span id="cb115-11"><a href="working-with-multiple-samples.html#cb115-11" tabindex="-1"></a></span>
<span id="cb115-12"><a href="working-with-multiple-samples.html#cb115-12" tabindex="-1"></a><span class="co"># Leiden clustering</span></span>
<span id="cb115-13"><a href="working-with-multiple-samples.html#cb115-13" tabindex="-1"></a>combined_pros <span class="ot">&lt;-</span> <span class="fu">doLeidenCluster</span>(<span class="at">gobject =</span> combined_pros, <span class="at">resolution =</span> <span class="fl">0.2</span>, <span class="at">n_iterations =</span> <span class="dv">1000</span>)</span>
<span id="cb115-14"><a href="working-with-multiple-samples.html#cb115-14" tabindex="-1"></a></span>
<span id="cb115-15"><a href="working-with-multiple-samples.html#cb115-15" tabindex="-1"></a><span class="co"># UMAP</span></span>
<span id="cb115-16"><a href="working-with-multiple-samples.html#cb115-16" tabindex="-1"></a>combined_pros <span class="ot">=</span> <span class="fu">runUMAP</span>(combined_pros)</span></code></pre></div>
</div>
<div id="vizualizing-spatial-location-of-clusters" class="section level3 hasAnchor" number="11.7.3">
<h3><span class="header-section-number">11.7.3</span> Vizualizing spatial location of clusters<a href="working-with-multiple-samples.html#vizualizing-spatial-location-of-clusters" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb116"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb116-1"><a href="working-with-multiple-samples.html#cb116-1" tabindex="-1"></a><span class="fu">spatDimPlot2D</span>(<span class="at">gobject =</span> combined_pros,</span>
<span id="cb116-2"><a href="working-with-multiple-samples.html#cb116-2" tabindex="-1"></a>              <span class="at">cell_color =</span> <span class="st">&#39;leiden_clus&#39;</span>, <span class="at">show_image =</span> <span class="cn">TRUE</span>, </span>
<span id="cb116-3"><a href="working-with-multiple-samples.html#cb116-3" tabindex="-1"></a>              <span class="at">image_name =</span> <span class="fu">c</span>(<span class="st">&quot;NP-image&quot;</span>, <span class="st">&quot;CP-image&quot;</span>),</span>
<span id="cb116-4"><a href="working-with-multiple-samples.html#cb116-4" tabindex="-1"></a>              <span class="at">save_param =</span> <span class="fu">list</span>(<span class="at">save_name =</span> <span class="st">&quot;ses3_1_leiden_clus&quot;</span>))</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-177"></span>
<img src="img/03_session1/ses3_1_leiden_clus.png" alt="Leiden clustering for the normal prostate (left) and the adenocarcinoma prostate sample (right)." width="100%" />
<p class="caption">
Figure 11.9: Leiden clustering for the normal prostate (left) and the adenocarcinoma prostate sample (right).
</p>
</div>
</div>
<div id="vizualizing-tissue-contribution-to-clusters" class="section level3 hasAnchor" number="11.7.4">
<h3><span class="header-section-number">11.7.4</span> Vizualizing tissue contribution to clusters<a href="working-with-multiple-samples.html#vizualizing-tissue-contribution-to-clusters" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>We can see that the samples are clustered distinctly within the UMAP.</p>
<div class="sourceCode" id="cb117"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb117-1"><a href="working-with-multiple-samples.html#cb117-1" tabindex="-1"></a><span class="fu">spatDimPlot2D</span>(<span class="at">gobject =</span> combined_pros,</span>
<span id="cb117-2"><a href="working-with-multiple-samples.html#cb117-2" tabindex="-1"></a>              <span class="at">cell_color =</span> <span class="st">&#39;list_ID&#39;</span>,<span class="at">show_image =</span> <span class="cn">TRUE</span>, </span>
<span id="cb117-3"><a href="working-with-multiple-samples.html#cb117-3" tabindex="-1"></a>              <span class="at">image_name =</span> <span class="fu">c</span>(<span class="st">&quot;NP-image&quot;</span>, <span class="st">&quot;CP-image&quot;</span>),</span>
<span id="cb117-4"><a href="working-with-multiple-samples.html#cb117-4" tabindex="-1"></a>              <span class="at">save_param =</span> <span class="fu">list</span>(<span class="at">save_name =</span> <span class="st">&quot;ses3_1_tissue_contribution&quot;</span>))</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-179"></span>
<img src="img/03_session1/ses3_1_tissue_contribution.png" alt="Tissue contribution for leiden clustering for the normal prostate (left) and the adenocarcinoma prostate sample (right)." width="100%" />
<p class="caption">
Figure 11.10: Tissue contribution for leiden clustering for the normal prostate (left) and the adenocarcinoma prostate sample (right).
</p>
</div>
</div>
</div>
<div id="perform-harmony-and-default-workflows" class="section level2 hasAnchor" number="11.8">
<h2><span class="header-section-number">11.8</span> Perform Harmony and default workflows<a href="working-with-multiple-samples.html#perform-harmony-and-default-workflows" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We can use harmony to integrate multiple datasets, grouping equivelent cell types between samples. Harmony is an algorithm that iteratively adjusts cell coordinates in a reduced-dimensional space to correct for dataset-specific effects. It uses fuzzy clustering to assign cells to multiple clusters, calculates dataset-specific correction factors, and applies these corrections to each cell, repeating the process until the influence of the dataset diminishes.</p>
<div class="sourceCode" id="cb118"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb118-1"><a href="working-with-multiple-samples.html#cb118-1" tabindex="-1"></a><span class="fu">library</span>(harmony)</span>
<span id="cb118-2"><a href="working-with-multiple-samples.html#cb118-2" tabindex="-1"></a></span>
<span id="cb118-3"><a href="working-with-multiple-samples.html#cb118-3" tabindex="-1"></a><span class="do">## run harmony integration</span></span>
<span id="cb118-4"><a href="working-with-multiple-samples.html#cb118-4" tabindex="-1"></a>combined_pros <span class="ot">=</span> <span class="fu">runGiottoHarmony</span>(combined_pros, <span class="at">vars_use =</span> <span class="st">&#39;list_ID&#39;</span>, <span class="at">do_pca =</span> F)</span></code></pre></div>
<div id="clustering-harmonized-object" class="section level3 hasAnchor" number="11.8.1">
<h3><span class="header-section-number">11.8.1</span> Clustering harmonized object<a href="working-with-multiple-samples.html#clustering-harmonized-object" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb119"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb119-1"><a href="working-with-multiple-samples.html#cb119-1" tabindex="-1"></a><span class="do">## sNN network (default)</span></span>
<span id="cb119-2"><a href="working-with-multiple-samples.html#cb119-2" tabindex="-1"></a>combined_pros <span class="ot">&lt;-</span> <span class="fu">createNearestNetwork</span>(<span class="at">gobject =</span> combined_pros,</span>
<span id="cb119-3"><a href="working-with-multiple-samples.html#cb119-3" tabindex="-1"></a>                                  <span class="at">dim_reduction_to_use =</span> <span class="st">&#39;harmony&#39;</span>, <span class="at">dim_reduction_name =</span> <span class="st">&#39;harmony&#39;</span>, <span class="at">name =</span> <span class="st">&#39;NN.harmony&#39;</span>,</span>
<span id="cb119-4"><a href="working-with-multiple-samples.html#cb119-4" tabindex="-1"></a>                                  <span class="at">dimensions_to_use =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>, <span class="at">k =</span> <span class="dv">15</span>)</span>
<span id="cb119-5"><a href="working-with-multiple-samples.html#cb119-5" tabindex="-1"></a></span>
<span id="cb119-6"><a href="working-with-multiple-samples.html#cb119-6" tabindex="-1"></a><span class="do">## Leiden clustering</span></span>
<span id="cb119-7"><a href="working-with-multiple-samples.html#cb119-7" tabindex="-1"></a>combined_pros <span class="ot">&lt;-</span> <span class="fu">doLeidenCluster</span>(<span class="at">gobject =</span> combined_pros,</span>
<span id="cb119-8"><a href="working-with-multiple-samples.html#cb119-8" tabindex="-1"></a>                             <span class="at">network_name =</span> <span class="st">&#39;NN.harmony&#39;</span>, <span class="at">resolution =</span> <span class="fl">0.2</span>, <span class="at">n_iterations =</span> <span class="dv">1000</span>, <span class="at">name =</span> <span class="st">&#39;leiden_harmony&#39;</span>)</span>
<span id="cb119-9"><a href="working-with-multiple-samples.html#cb119-9" tabindex="-1"></a></span>
<span id="cb119-10"><a href="working-with-multiple-samples.html#cb119-10" tabindex="-1"></a><span class="co"># UMAP dimension reduction</span></span>
<span id="cb119-11"><a href="working-with-multiple-samples.html#cb119-11" tabindex="-1"></a>combined_pros <span class="ot">=</span> <span class="fu">runUMAP</span>(combined_pros, <span class="at">dim_reduction_name =</span> <span class="st">&#39;harmony&#39;</span>, <span class="at">dim_reduction_to_use =</span> <span class="st">&#39;harmony&#39;</span>, <span class="at">name =</span> <span class="st">&#39;umap_harmony&#39;</span>)</span>
<span id="cb119-12"><a href="working-with-multiple-samples.html#cb119-12" tabindex="-1"></a></span>
<span id="cb119-13"><a href="working-with-multiple-samples.html#cb119-13" tabindex="-1"></a><span class="fu">spatDimPlot2D</span>(<span class="at">gobject =</span> combined_pros,</span>
<span id="cb119-14"><a href="working-with-multiple-samples.html#cb119-14" tabindex="-1"></a>              <span class="at">dim_reduction_to_use =</span> <span class="st">&#39;umap&#39;</span>, <span class="at">dim_reduction_name =</span> <span class="st">&#39;umap_harmony&#39;</span>,</span>
<span id="cb119-15"><a href="working-with-multiple-samples.html#cb119-15" tabindex="-1"></a>              <span class="at">cell_color =</span> <span class="st">&#39;leiden_harmony&#39;</span>, <span class="at">show_image =</span> T, <span class="at">image_name =</span> <span class="fu">c</span>(<span class="st">&quot;NP-image&quot;</span>, <span class="st">&quot;CP-image&quot;</span>),</span>
<span id="cb119-16"><a href="working-with-multiple-samples.html#cb119-16" tabindex="-1"></a>              <span class="at">spat_point_size =</span> <span class="dv">1</span>, <span class="at">save_param =</span> <span class="fu">list</span>(<span class="at">save_name =</span> <span class="st">&quot;leiden_clustering_harmony&quot;</span>))</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-182"></span>
<img src="img/03_session1/leiden_clustering_harmony.png" alt="Leiden clustering after harmony was performed for the normal prostate (left) and the adenocarcinoma prostate sample (right)." width="100%" />
<p class="caption">
Figure 11.11: Leiden clustering after harmony was performed for the normal prostate (left) and the adenocarcinoma prostate sample (right).
</p>
</div>
</div>
<div id="vizualizing-the-tissue-contribution" class="section level3 hasAnchor" number="11.8.2">
<h3><span class="header-section-number">11.8.2</span> Vizualizing the tissue contribution<a href="working-with-multiple-samples.html#vizualizing-the-tissue-contribution" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>We can see that after performing harmony that the clusters from the two tissue samples are now clustered together. There is still a cluster that is unique to the adenocarcinoma sample, however this is expected as this represents the visium spots that cover the tumor regions of the tissue, which are not found in the normal tissue.</p>
<div class="sourceCode" id="cb120"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb120-1"><a href="working-with-multiple-samples.html#cb120-1" tabindex="-1"></a> <span class="fu">spatDimPlot2D</span>(<span class="at">gobject =</span> combined_pros,</span>
<span id="cb120-2"><a href="working-with-multiple-samples.html#cb120-2" tabindex="-1"></a>               <span class="at">dim_reduction_to_use =</span> <span class="st">&#39;umap&#39;</span>, <span class="at">dim_reduction_name =</span> <span class="st">&#39;umap_harmony&#39;</span>,</span>
<span id="cb120-3"><a href="working-with-multiple-samples.html#cb120-3" tabindex="-1"></a>               <span class="at">cell_color =</span> <span class="st">&#39;list_ID&#39;</span>, <span class="at">save_plot =</span> <span class="cn">TRUE</span>,</span>
<span id="cb120-4"><a href="working-with-multiple-samples.html#cb120-4" tabindex="-1"></a>               <span class="at">save_param =</span> <span class="fu">list</span>(<span class="at">save_name =</span> <span class="st">&quot;leiden_clustering_harmony_contribution&quot;</span>))</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-184"></span>
<img src="img/03_session1/leiden_clustering_harmony_contribution.png" alt="Tissue contribution for leiden clustering after harmony for the normal prostate (left) and the adenocarcinoma prostate sample (right)." width="100%" />
<p class="caption">
Figure 11.12: Tissue contribution for leiden clustering after harmony for the normal prostate (left) and the adenocarcinoma prostate sample (right).
</p>
</div>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="spatial-proteomics-multiplex-if.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="spatial-multi-modal-analysis.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
